#To force non-scientific notation in figures, i.e. 1,000,000 not 1xE6
require(scales)
then either scale_y_continuous(labels = comma) or scale_x_continuous(labels = comma)


plp1 <- ggplot(bins_and_read_counts, aes(x="Bins", y=bins_and_read_counts$name)) + geom_boxplot() + xlab("Bins") + ylab("Number of reads per Bin") + labs(title = "Reads in each bin")
plp2 <- ggplot(bins_species_count, aes(x="Bins", y=bins_species_count$name)) + geom_boxplot() + xlab("Bins") + ylab("Number of species detected per Bin") + labs(title = "Kaiju species* in each bin")
plp3 <- ggplot(bins_read_count_to_base_count, aes(x=bins_read_count_to_base_count$name, y=bins_read_count_to_base_count$Read)) + geom_point() + theme_classic() + xlab("Reads in bin") + ylab("Bases in bin") + labs(title = "Reads per bin to bases per bin")
plp4 <- ggplot(bins_percent_reads_classified_unclassified_unidentified, aes(x=bins_percent_reads_classified_unclassified_unidentified$name, y=bins_percent_reads_classified_unclassified_unidentified$read)) + geom_point() + theme_classic() + xlab("% reads classified by Kaiju") + ylab("% reads unidentified by Kaiju") + labs(title = "Percent reads classified by Kaiju per bin to percent reads unidentified*")
plp5 <- ggplot(all_bins_total_reads_and_classified_reads, aes(x=all_bins_total_reads_and_classified_reads$Read.count, y=all_bins_total_reads_and_classified_reads$Classified.read.count)) + geom_point() + theme_classic() + xlab("Total reads") + ylab("Classified reads") + labs(title = "Total reads per bin to classified reads per bin (species*)")
plp6 <- ggplot(all_bins_total_reads_and_classified_read_percentages, aes(x=all_bins_total_reads_and_classified_read_percentages$Read.count, y=all_bins_total_reads_and_classified_read_percentages$Classified.read.percentage.all.reads)) + geom_point() + theme_classic() + xlab("Total reads") + ylab("Reads classified(%)") + labs(title = "Total reads per bin to percentage reads classified (species*)")
plp7 <- ggplot(all_bins_percent_reads_classified_to_percent_reads_classified_as_most_common_species, aes(x=all_bins_percent_reads_classified_to_percent_reads_classified_as_most_common_species$Classified.read.percentage.all.reads, y=all_bins_percent_reads_classified_to_percent_reads_classified_as_most_common_species$Largest.percentage.TaxID.actual.percentage)) + geom_point() + theme_classic() + xlab("% Reads classified") + ylab("% Reads classified to most abundant species") + labs(title = "% reads classified to % reads classified as most abundant species in Bin")
multiplot(plp1, plp2, plp3, plp4, plp5, plp6, plp7, cols=3)

ggplot(all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name, aes(x=all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Read.count, y=all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Largest.percentage.TaxID.actual.percentage)) + geom_point() + theme_classic() + xlab("Reads per bin") + ylab("% reads classified as most abundant species") + labs(title = "Reads per bin to percent reads classified as most abundant species")

ggplot(all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name, aes(x=all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Read.count, y=all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Largest.percentage.TaxID.actual.percentage)) + geom_point() + theme_classic() + xlab("Reads per bin") + ylab("% reads classified as most abundant species") + labs(title = "Reads per bin to percent reads classified as most abundant species") + geom_label_repel(aes(label=ifelse(all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Largest.percentage.TaxID.actual.percentage>87,as.character(all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Largest.percentage.TaxID.scientific.name),'')), box.padding = 0.35, point.padding = 0.5, segment.color = 'grey50')

ggplot(all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name, aes(x=all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Read.count, y=all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Largest.percentage.TaxID.actual.percentage)) + geom_point(aes(x=all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Read.count, y=all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Largest.percentage.TaxID.actual.percentage, color = all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Largest.percentage.TaxID.actual.percentage > 75 & all_bins_largest_species_read_percentage_scientific_name_reads_in_the_bin_and_bin_name$Read.count > 400)) + scale_color_manual(name = 'MAS > 75% AND Reads in bin > 400', values= setNames(c('blue','orange'),c(T, F))) + theme_classic() + xlab("Reads per bin") + ylab("% reads classified as most abundant species") + labs(title = "Reads per bin to percent reads classified as most abundant species")

ggplot(data = bases_and_longest_contigs_in_Bins, aes(x = Total_bases, y = Percentage_bases_in_longest_contig)) + geom_point(aes(color = ifelse(Percentage_bases_in_longest_contig >= 10, "blue", ifelse(Percentage_bases_in_longest_contig <=10, "dark orange", "orange")) + geom_smooth(method = "lm", se = FALSE, col = "dark orange") + theme_classic()+ scale_color_manual(name = 'MAS > 10% of bases', values= setNames(c('blue','orange'),c(T, F))

ggplot(data = bases_and_longest_contigs_in_Bins, aes(x = Total_bases, y = Percentage_bases_in_longest_contig)) + geom_point(aes(color = ifelse(Percentage_bases_in_longest_contig >= 10, "blue", ifelse(Percentage_bases_in_longest_contig <=10, "dark orange", "light orange")))) + geom_smooth(method = "lm", se = FALSE, col = "dark orange") + theme_classic() + xlab("Total bases in Bin") + ylab("Percentage bases in longest contig") + scale_color_manual(name = 'MAS > 10% of bases', values= setNames(c(T, F)))

ggplot(data = bases_and_longest_contigs_in_Bins, aes(x = Total_bases, y = Percentage_bases_in_longest_contig)) + geom_point(aes(color = ifelse(Percentage_bases_in_longest_contig >= 10, "blue", ifelse(Percentage_bases_in_longest_contig <=10, "dark orange", "light orange")))) + geom_smooth(method = "lm", se = FALSE, col = "dark orange") + xlab("Total bases in Bin") + ylab("Percentage bases in longest contig") + scale_color_manual(name = 'Longest contig contains > 10% of bases', values= setNames(c('blue','orange','dark orange'),c(T, F))) + theme(panel.background = element_blank())

 ggplot(data = bases_and_longest_contigs_in_Bins, aes(x = Total_bases, y = Percentage_bases_in_longest_contig)) + geom_point(aes(color = ifelse(Percentage_bases_in_longest_contig >= 10, "blue", ifelse(Percentage_bases_in_longest_contig <=10, "dark orange", "light orange")))) + geom_smooth(method = "lm", se = FALSE, col = "dark orange") + xlab("Total bases in Bin") + ylab("Percentage bases in longest contig") + scale_color_manual(name = 'MAS > 10% of bases', values= setNames(c('blue','orange','dark orange'),c(T, F, F))) + theme(panel.background = element_blank()) + labs(title = "Bases in bin to % bases in longest contig")

ggplot() + geom_point(data = PID.0578.r.16aA_S24_R1_001.read_lengths_and_occurences_of_that_length, aes(x = PID.0578.r.16aA_S24_R1_001.read_lengths_and_occurences_of_that_length$read_length, y = PID.0578.r.16aA_S24_R1_001.read_lengths_and_occurences_of_that_length$count, color="orange")) + geom_point(data = PID.0578.r.16aB_S25_R1_001.read_lengths_and_occurences_of_that_length, aes(x = PID.0578.r.16aB_S25_R1_001.read_lengths_and_occurences_of_that_length$read_length, y = PID.0578.r.16aB_S25_R1_001.read_lengths_and_occurences_of_that_length$count, color="blue"))

#Note the h = bit gives the colours blue for true and orange for false here, seems that those are the values on the colour wheel though this could break if there is more than a T\F binary it has to colour
ggplot(data = bases_and_longest_contigs_in_Bins, aes(x = Total_bases, y = Percentage_bases_in_longest_contig)) + geom_point(aes(color = ifelse(Percentage_bases_in_longest_contig >= 10, "blue", ifelse(Percentage_bases_in_longest_contig <=10, "dark orange", "light orange")))) + geom_smooth(method = "lm", se = FALSE, col = "turquoise") + xlab("Total bases in Bin") + ylab("Percentage bases in longest contig") + scale_color_hue(name = 'Longest contig contains > 10% of bases', labels = c("True", "False"), h = c(175, 50)) + theme(panel.background = element_blank())

ggplot(data = contig_lengths_culture_13E_S5) + aes(x="Bins", y = contig_lengths_culture_13E_S5$X13E_S5_R1_001) + geom_violin(color="black", fill="dark orange") + theme_classic() + ylab("Contig length (bases)") + labs(title = "Contig lengths in Megahit assembly")

ggplot(data = contig_lengths_culture_13E_S5) + aes(x="Bins", y = contig_lengths_culture_13E_S5$X13E_S5_R1_001) + geom_violin(color="black", fill="dark orange") + ylab("Contig length (bases)") + labs(title = "Contig lengths in Megahit assembly") + geom_boxplot(color="black", fill="grey", width=0.1) + theme_classic()

ggplot(data = unique_species_detected_by_Kaiju_all_samples, aes(x = unique_species_detected_by_Kaiju_all_samples$Total_reads_in_file, y = unique_species_detected_by_Kaiju_all_samples$Number_unique_species_detected, size = unique_species_detected_by_Kaiju_all_samples$Difference_in_reads_between_first_and_second_most_abundant_species)) + geom_point(alpha=0.5) + theme_classic() + scale_size(range = c(1,200), name="Difference between most and second most abundant species (reads)"))

ggplot(data = unique_species_detected_by_Kaiju_all_samples, aes(x = unique_species_detected_by_Kaiju_all_samples$Total_reads_in_file, y = unique_species_detected_by_Kaiju_all_samples$Number_unique_species_detected)) + geom_jitter(aes(size = unique_species_detected_by_Kaiju_all_samples$Difference_in_reads_between_first_and_second_most_abundant_species), alpha=0.5) + scale_size_continuous(range = c(1,15), name="MAS - 2nd MAS (reads)") + theme_linedraw() + xlab("Total reads in file") + ylab("No. unique species detected") + labs(title = "Kaiju analysis of Lac culture read files")


ggplot(data = bins_read_count_to_base_count, aes(x = "Bins", y = bins_read_count_to_base_count$Read)) + geom_violin(color="black", fill="dark orange") + geom_boxplot(color="black", fill="grey", width=0.1) + theme_classic() + ylab("No. bases per bin") + xlab("Bins") + labs(title = "Bases in all Bins")
orangered
dodgerblue3

ggplot(data = bins_species_count, aes(x = "Bins", y = `name`)) + geom_violin(color="black", fill="dark orange") + geom_boxplot(color="black", fill="grey", width=0.1) + theme_classic() + xlab("Bins") + ylab("No. species detected by Kaiju") + labs(title = "Number of species detected by Kaiju per bin")
p<-ggplot(data=df, aes(x=dose, y=len)) +
  geom_bar(stat="identity")
p

# Horizontal bar plot
p + coord_flip()

ggplot(data = AR47N_mapping, aes(x = `Mapping_confidence`, y = `Contigs`)) + geom_bar(stat = "identity") + theme_classic()

ggplot(data = all_bins_genus_level_info_classified_unclassified_viral_most_and_second_most_abundant_contig_number, aes(x = `Unclassified_or_viral_contigs_as_percent`, y = `Most_abundant_genus_as_percent`)) + geom_jitter(aes(size = all_bins_genus_level_info_classified_unclassified_viral_most_and_second_most_abundant_contig_number$Contigs_in_bin), alpha=0.15, color = "darkorange2") + scale_size_continuous(range = c(1,15), name="No. contigs in bin") + theme_linedraw()

ggplot(data = all_bins_genus_level_info_classified_unclassified_viral_most_and_second_most_abundant_contig_number, aes(x = `Most_abundant_genus_as_percent`, y = `Unclassified_or_viral_contigs_as_percent`)) + geom_jitter(aes(size = all_bins_genus_level_info_classified_unclassified_viral_most_and_second_most_abundant_contig_number$Contigs_in_bin), alpha=0.15, color = "darkorange2") + scale_size_continuous(range = c(1,15), name="No. contigs in bin") + theme_linedraw() + xlab("% contigs classified with most abundant genus") + ylab()

ggplot(data = all_bins_genus_level_info_classified_unclassified_viral_most_and_second_most_abundant_contig_number, aes(x = `Most_abundant_genus_as_percent`, y = `X2nd_most_abundant_genus_as_percent`)) + geom_jitter(aes(size = all_bins_genus_level_info_classified_unclassified_viral_most_and_second_most_abundant_contig_number$Contigs_in_bin), alpha=0.25, color = "darkorange2") + scale_size_continuous(range = c(1,15), name="No. contigs in bin") + theme_linedraw() + xlab("% contigs classified with most abundant genus") + ylab("% contigs classified with 2nd most abundant genus") + labs(title = "Comparing percent of contigs per bin which belong to most or second most abundant genus")

ggplot(data = all_bins_genus_level_info_classified_unclassified_viral_most_and_second_most_abundant_contig_number, aes(x = `Most_abundant_genus_as_percent`, y = `X2nd_most_abundant_genus_as_percent`)) + geom_jitter(aes(size = all_bins_genus_level_info_classified_unclassified_viral_most_and_second_most_abundant_contig_number$Difference_between_most_and_second_most_abundant_genera_as_percent), alpha=0.25, color = "dodgerblue3") + scale_size_continuous(range = c(1,15), name="Difference between most and second most abundant genus") + theme_linedraw() + xlab("% contigs classified with most abundant genus") + ylab("% contigs classified with 2nd most abundant genus") + labs(title = "Comparing percent of contigs per bin which belong to most or second most abundant genus")

ggplot(data = all_bins_genus_level_info_classified_unclassified_viral_most_and_second_most_abundant_contig_number, aes(x = `Most_abundant_genus_as_percent`, y = `X2nd_most_abundant_genus_as_percent`)) + geom_jitter(aes(size = all_bins_genus_level_info_classified_unclassified_viral_most_and_second_most_abundant_contig_number$Unclassified_or_viral_contigs_as_percent), alpha=0.25, color = "dimgrey") + scale_size_continuous(range = c(1,15), name="% contigs unclassified or viral") + theme_linedraw() + xlab("% contigs classified with most abundant genus") + ylab("% contigs classified with 2nd most abundant genus") + labs(title = "Comparing percent of contigs per bin which belong to most or second most abundant genus")

ggplot(data = culture_reads_read_counts, aes(x = "Samples", y = `Reads`)) + geom_violin(color = "black", fill="darkorange2") + geom_boxplot(color = "black", fill="grey", width=0.1) + theme_classic()


ggplot(data = trial_combined_mapping_actually_AC16J_col2_AC16N_col3, aes(x = `AC16J`, y = `AC16N`)) + geom_point() +theme_classic() + xlab("AC16J mapping") + ylab("AC16N mapping")


contig_mapping_counts.pca <- prcomp(all_contigs_all_samples_mapping_should_have_right_count_for_each_contig[,c(2:79)], center = TRUE, scale. = TRUE)

ggbiplot(contig_mapping_counts.pca, ellipse=TRUE, obs.scale = 1, var.scale = 1, var.axes=FALSE, circle = TRUE) + theme_classic()
ggbiplot(contig_mapping_counts.pca, ellipse=TRUE, circle = TRUE) + theme_classic()
ggbiplot(contig_mapping_counts.pca, labels=trial_taxons_collected_columns$Samples, ellipse=TRUE, obs.scale = 1, var.scale = 1, var.axes=FALSE, circle = TRUE, alpha = 1, ) + theme_classic()

ggplot(data = collected_taxons_as_rows, aes(x = `AC14J_subsampled`, y = `AC14N_subsampled`)) + geom_point() + theme_classic()

ggplot(data = total_counts_each_taxon_all_samples_per_host_species, aes(x = `AC_total_count`, y = `AR_total_count`)) + geom_point() + theme_linedraw() + scale_y_continuous(labels = comma) + scale_x_continuous(labels = comma)

ggbiplot(contig_mapping_counts.pca, ellipse=TRUE, obs.scale = 1, var.scale = 1, var.axes=FALSE, circle = TRUE, alpha = 1, color= blue ) + theme_classic()

ggbiplot(excluding_pilot_samples.pca, ellipse=TRUE, obs.scale = 1, var.scale = 1, var.axes=FALSE, circle = TRUE, alpha = 1, color= blue ) + theme_linedraw() + labs(title = "Excluding pilot samples, all taxons")

dodgerblue3

ggbiplot(contig_mapping_counts.pca, labels=(trial_taxons_collected_columns$Samples, color = red), ellipse=TRUE, obs.scale = 1, var.scale = 1, var.axes=FALSE, circle = TRUE, alpha = 1, color= blue ) + theme_linedraw() + labs(title = "All samples, all taxons")
ggbiplot(excluding_pilot_samples.pca, labels=rownames(V2_collected_taxons_as_columns_excluding_pilot_samples), ellipse=TRUE, obs.scale = 1, var.scale = 1, var.axes=FALSE, circle = TRUE, alpha = 1, color= blue ) + theme_linedraw() + labs(title = "Excluding pilot samples, all taxons")

ggplot(data = total_counts_each_taxon_all_samples_per_host_species, aes(x = `AC_total_count`, y = `AR_total_count`)) + geom_point() + theme_linedraw() + scale_y_continuous(labels = comma) + scale_x_continuous(labels = comma) + labs(title = 'All samples, summed taxon counts by host species') + xlab('Acomys cahirinus summed totals') + ylab('Acomys russatus summed totals') + geom_text(aes(label=ifelse(`AC_total_count`>50000000,as.character(`Taxa`),'')), hjust=1.5,vjust=1)

ggplot(data = total_counts_each_taxon_all_samples_per_season, aes(x = `June_total_count`, y = `November_total_count`)) + geom_point() + theme_linedraw() + scale_y_continuous(labels = comma) + scale_x_continuous(labels = comma) + labs(title = 'All samples, summed taxon counts by season') + xlab('June  summed totals') + ylab('November summed totals') + geom_text(aes(label=ifelse(`AC_total_count`>50000000,as.character(`Taxa`),'')), hjust=1.5,vjust=1)

plp1 <- ggplot(data = total_counts_each_taxon_all_samples_per_host_and_season, aes(x = `AC_June_total_count`, y = `AR_June_total_count`)) + geom_point() + theme_linedraw() + scale_y_continuous(labels = comma) + scale_x_continuous(labels = comma) + labs(title = 'All June samples, summed count by host species') + xlab('Acomys cahirinus June samples, summed taxon counts') + ylab('Acomys russatus June samples, summed taxon counts')
plp2 <- ggplot(data = total_counts_each_taxon_all_samples_per_host_and_season, aes(x = `AC_November_total_count`, y = `AR_November_total_count`)) + geom_point() + theme_linedraw() + scale_y_continuous(labels = comma) + scale_x_continuous(labels = comma) + labs(title = 'All November samples, summed count by host species') + xlab('Acomys cahirinus November samples, summed taxon counts') + ylab('Acomys russatus November samples, summed taxon counts')

row_pca.pca <- prcomp(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated`[,c(2:6352)], center = TRUE, scale. = TRUE)
ggthemr_reset()
ggbiplot(row_pca.pca, ellipse=TRUE, obs.scale = 1, var.scale = 1, var.axes = FALSE, circle = TRUE) + theme_classic()

trial_df_from_columns_file <- read.csv(file = '2021_05_17-Modified_just_taxons_right_order_single_row_comma_separated.csv')

datalong_from_columns_trial <- pivot_longer(trial_df_from_columns_file, cols = Abditibacterium.utsteinense:uncultured.rumen.bacterium, names_to = "name")

ggplot(data = datalong_from_columns_trial) + geom_col(aes(x = `Microorganism`, y = value, fill = `name`)) + theme_classic()
contig_mapping_counts.pca <- prcomp(all_contigs_all_samples_mapping_should_have_right_count_for_each_contig[,c(2:79)], center = TRUE, scale. = TRUE)
ggbiplot(contig_mapping_counts.pca, ellipse=TRUE, obs.scale = 1, var.scale = 1, var.axes=FALSE, circle = TRUE, alpha = 1, color= blue ) + theme_classic()
ggbiplot(host_and_season_average_mapping_counts.pca, obs.scale = 1, var.scale = 1, group=host_and_season_average_mapping_counts.Microorganism, varname.size = 3, labels.size=3, ellipse = TRUE, circle = TRUE) + theme(legend.direction ="horizontal", legend.position = "top") + scale_color_manual(name="Host and season", values=c("orange", "blue", "dark grey")) + scale_shape_manual(name="Host and season", values=c(17:19)) + geom_point(aes(color=host_and_season_average_mapping_counts.Microorganism, shape=host_and_season_average_mapping_counts.Microorganism, size = 3))

ggbiplot::ggbiplot(host_season_averages.pca, ellipse=TRUE, obs.scale = 1, var.scale = 1, var.axes=FALSE, circle = TRUE, alpha = 1, color= blue) + theme_linedraw()

ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = `ACJ`, y = `ACN`)) + geom_point()

ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = `ACJ`, y = `ACN`, label = `Microorganism`)) + geom_point() + geom_text_repel(aes(label=ifelse(`ACJ`>3000 | `ACN`>3000,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines"))) + theme_linedraw() + geom_smooth(method = "lm", se = TRUE, col = "dodgerblue3")

ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = `ARJ`, y = `ARN`, label = `Microorganism`)) + geom_point(color="dodgerblue3") + geom_text_repel(aes(label=ifelse(`ARJ`>2500 | `ARN`>2500,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines"))) + theme_linedraw() + xlab('Average mapping count for ARJ samples') + ylab('Average mapping count for ARN samples') + labs(title="Average mapping score for all ARJ and ARN samples to all detected unique taxons")

ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = `ACJ`, y = `ARJ`, label = `Microorganism`)) + geom_point() + geom_text_repel(aes(label=ifelse(`ACJ`>2500 | `ARJ`>2500,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines"))) + theme_linedraw() + xlab('Average mapping count for ACJ samples') + ylab('Average mapping count for ARJ samples') + labs(title="Average mapping score for all ACJ and ARJ samples to all detected unique taxons")
ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = `ACN`, y = `ARN`, label = `Microorganism`)) + geom_point(color="dodgerblue3") + geom_text_repel(aes(label=ifelse(`ACN`>2500 | `ARN`>2500,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines"))) + theme_linedraw() + xlab('Average mapping count for ACN samples') + ylab('Average mapping count for ARN samples') + labs(title="Average mapping score for all ACN and ARN samples to all detected unique taxons")
ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = `ACJ`, y = `ARN`, label = `Microorganism`)) + geom_point(color="dark grey") + geom_text_repel(aes(label=ifelse(`ACJ`>2500 | `ARN`>2500,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines"))) + theme_linedraw() + xlab('Average mapping count for ACJ samples') + ylab('Average mapping count for ARN samples') + labs(title="Average mapping score for all ACJ and ARN samples to all detected unique taxons")
ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = `ACN`, y = `ARJ`, label = `Microorganism`)) + geom_point(color="Sea Green 3") + geom_text_repel(aes(label=ifelse(`ACN`>2500 | `ARJ`>2500,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines"))) + theme_linedraw() + xlab('Average mapping count for ACN samples') + ylab('Average mapping count for ARJ samples') + labs(title="Average mapping score for all ACN and ARJ samples to all detected unique taxons")

`2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$AC_differences <- `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$ACJ - `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$ACN
`2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$AR_differences <- `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$ARJ - `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$ARN

`2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$AC_differences <- `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$ACN - `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$ACJ
`2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$AR_differences <- `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$ARN - `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`$ARJ

`2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`<-`2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`[-c(6,7)]

ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = `AC_differences`, y = `AR_differences`, label = `Microorganism`)) + geom_point() + geom_text_repel(aes(label=ifelse(`AC_differences`>100 | `AC_differences`< -500,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines"))) + theme_linedraw() + xlab('Per taxon difference between ACJ and ACN average mapping counts') + ylab('Per taxon difference between ARJ and ARN average mapping counts') + labs(title="Difference per unique taxon between average mapping scores from June to November per sample") + geom_text_repel(aes(label=ifelse(`AR_differences`>100 | `AR_differences`< -500,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines")))

ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = `AC_differences`, y = `AR_differences`, label = `Microorganism`)) + geom_point() + geom_text_repel(aes(label=ifelse(`AC_differences`>100 | `AC_differences`< -500 | `AR_differences`>100 | `AR_differences`< -500,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines"))) + theme_linedraw() + xlab('Per taxon difference between ACJ and ACN average mapping counts') + ylab('Per taxon difference between ARJ and ARN average mapping counts') + labs(title="Difference per unique taxon between average mapping scores from June to November per sample") + geom_smooth(method = "lm", se = FALSE, col = "dark orange")

ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = "ACJ average", y = `ACJ`)) + geom_boxplot() + geom_violin(color="black", fill="dark orange") + theme_linedraw()

ggplot(data = `2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows_includes_mean_and_medians_across_samples_for_each_taxon`, aes(x = `AC11J`, y = `AC11N`, label = `Microorganism`)) + geom_point() + theme_linedraw() + geom_point(data =`2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows_includes_mean_and_medians_across_samples_for_each_taxon`, aes(x = `AC14J`, y = `AC14N`), color = "dodgerblue3")

ggplot(data = `2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows_includes_mean_and_medians_across_samples_for_each_taxon`, aes(x = `AC11J`, y = `AC11N`, label = `Microorganism`)) + geom_point() + theme_linedraw() + geom_point(aes(x = `AC14J`, y = `AC14N`))

ggplot(data = `detailed_bin_notes`, aes(x = `Read.count`, y = `Base.count`)) + geom_point() + theme_linedraw() + xlab("No. contigs per Bin") + ylab("Total bases per bin") + labs(title="Contigs per bin to total base count per bin")

ggplot(data = `detailed_bin_notes`, aes(x = `Classified.read.percentage.all.reads`, y = `Unidentified.read.percentage`, label = `Largest.percentage.TaxID.scientific.name`)) + geom_point(color = "dark orange") + theme_linedraw() + xlab("% contigs classified by Kaiju") + ylab("% contigs unidentified by Kaiju") + geom_text_repel(aes(label=ifelse(`Classified.read.percentage.all.reads`>65,as.character(`Largest.percentage.TaxID.scientific.name`),''), box.padding = unit(0.45, "lines")))

ggplot(data = `detailed_bin_notes`, aes(x = `Classified.read.percentage.all.reads`, y = `Unidentified.read.percentage`, label = `Largest.percentage.TaxID.scientific.name`)) + geom_point(color = "dark orange") + theme_linedraw() + xlab("% contigs classified by Kaiju") + ylab("% contigs unidentified by Kaiju") + geom_text_repel(aes(label=ifelse(`Largest.percentage.TaxID.actual.percentage`>65,as.character(`Largest.percentage.TaxID.scientific.name`),''), max.overlaps = 10, box.padding = unit(0.45, "lines")))

ggplot(data = `detailed_bin_notes`, aes(x = `Classified.read.percentage.all.reads`, y = `Unidentified.read.percentage`, label = `Largest.percentage.TaxID.scientific.name`)) + geom_point(color = "dark orange") + theme_linedraw() + xlab("% contigs classified by Kaiju") + ylab("% contigs unidentified by Kaiju") + geom_text_repel(aes(label=ifelse(`Largest.percentage.TaxID.actual.percentage`>65,as.character(`Largest.percentage.TaxID.scientific.name`), max.overlaps = 10, box.padding = 0.45))

ggplot(data = `detailed_bin_notes`, aes(x = `Classified.read.percentage.all.reads`, y = `Unidentified.read.percentage`, label = `Largest.percentage.TaxID.scientific.name`)) + geom_point(color = "dark orange") + theme_linedraw() + xlab("% contigs classified by Kaiju") + ylab("% contigs unidentified by Kaiju") + geom_text_repel(data = subset(`detailed_bin_notes`, `Classified.read.percentage.all.reads` > 85), aes(label = `Largest.percentage.TaxID.scientific.name`), max.overlaps = 10, box.padding = 0.45)

ggplot(data = `detailed_bin_notes`, aes(x = `Classified.read.percentage.all.reads`, y = `Unidentified.read.percentage`, label = `Largest.percentage.TaxID.scientific.name`)) + geom_point(color = "dark orange") + theme_linedraw() + xlab("% contigs classified by Kaiju") + ylab("% contigs unidentified by Kaiju") + geom_text_repel(data = subset(`detailed_bin_notes`, `Classified.read.percentage.all.reads` > 85), aes(label = `Largest.percentage.TaxID.scientific.name`), max.overlaps = 10, box.padding = 0.45, hjust = -0.05)

ggplot(data = `detailed_bin_notes`, aes(x = `Unidentified.read.percentage`, y = `Classified.read.percentage.all.reads`, label = `Largest.percentage.TaxID.scientific.name`)) + geom_point(color = "dark orange") + theme_linedraw() + xlab("% contigs unidentified by Kaiju") + ylab("% contigs classified by Kaiju") + geom_text_repel(data = subset(`detailed_bin_notes`, `Classified.read.percentage.all.reads` > 85), aes(label = `Largest.percentage.TaxID.scientific.name`), max.overlaps = 40, box.padding = 10, hjust = -0.05)

ggplot(data = `detailed_bin_notes`, aes(x = `Unidentified.read.percentage`, y = `Classified.read.percentage.all.reads`, label = `Largest.percentage.TaxID.scientific.name`)) + geom_point(color = "dark orange") + theme_linedraw() + xlab("% contigs unidentified by Kaiju") + ylab("% contigs classified by Kaiju") + geom_text_repel(data = subset(`detailed_bin_notes`, `Classified.read.percentage.all.reads` > 90), aes(label = `Largest.percentage.TaxID.scientific.name`), max.overlaps = 20, box.padding = 1, hjust = 0.5, point.padding = 1, size = 3.5, nudge_x = 1, nudge_y=0.5, force = 3) + labs(title = "Percentage of contigs in each bin classified by Kaiju against unidentified by Kaiju", caption = "Named taxa are most abundant classifications from bins with over 90% of contigs classified")

ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = `AC_differences`, y = `AR_differences`, label = `Microorganism`)) + geom_point() + geom_text_repel(aes(label=ifelse(`AC_differences`>100 | `AC_differences`< -500,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines"))) + theme_linedraw() + xlab('Per taxon difference between ACJ and ACN average mapping counts') + ylab('Per taxon difference between ARJ and ARN average mapping counts') + labs(title="Difference per unique taxon between average mapping scores from June to November per sample") + geom_text_repel(aes(label=ifelse(`AR_differences`>100 | `AR_differences`< -500,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines")))

ggplot(data = `2021_05_17.All_taxa_adjusted_scores_by_host_species_and_season`, aes(x = `AC_differences`, y = `AR_differences`, label = `Microorganism`)) + geom_point(color = "dark orange") + theme_linedraw() + xlab('Per taxon difference between ACJ and ACN average mapping counts') + ylab('Per taxon difference between ARJ and ARN average mapping counts') + labs(title="Difference per unique taxon between average mapping scores from June to November per sample") + geom_text_repel(aes(label=ifelse(`AC_differences`>100 | `AR_differences`>100,as.character(`Microorganism`),''), box.padding = unit(0.45, "lines")))

ggplot(data = culture_read_files_name_read_count_percent_reads_most_abundant_genus_and_most_abundant_genus, aes(x = `Percent_most_abundant_genus`, y = `Reads`)) + geom_point(color = "dark orange", size = 3) + theme_linedraw() + xlab('Percent of reads in most abundant genus') + ylab('No. reads per sample') + labs(title = "Number of reads from sample file to percent reads classified by Kaiju as most abundant genus") + scale_y_continuous(labels = comma)

ggplot(data = `Culture_samples_assembly_contig_counts_to_eggnog_predicted_genes_from_assemblies`, aes(x = `Contigs_in_assembly`, y = `Predicted_genes_EGGNOG_MAPPER`)) + geom_point(color = "dark orange", size = 3) + theme_linedraw() + xlab('Number of contigs in Megahit assembly from sample') + ylab('Number of predicted genes in assembly by Eggnog Mapper') + labs(title="Number of contigs in Megahit assembly to no. genes predicted by Eggnog Mapper")

ggplot(data = `Culture_samples_assembly_contig_counts_to_eggnog_predicted_genes_from_assemblies`, aes(x = `Predicted_genes_EGGNOG_MAPPER`, y = `Named_predicted_proteins_EGGNOG_MAPPER`)) + geom_point(color = "#00ff8c", size = 3) + theme_linedraw() + xlab('Number of predicted genes in assembly by Eggnog Mapper') + ylab('Number of predicted named proteins from Eggnog Mapper') + labs(title="Number of Eggnog Mapper predicted genes to named predicted proteins")

ggplot(data = `Culture_samples_assembly_contig_counts_to_eggnog_predicted_genes_from_assemblies`, aes(x = `Contigs_in_assembly`, y = `Predicted_genes_EGGNOG_MAPPER`)) + geom_point(color = "dark orange", size = 3) + theme_linedraw() + xlab('Number of contigs in Megahit assembly from sample') + ylab('Number of predicted genes in assembly by Eggnog Mapper') + labs(title="Number of contigs in Megahit assembly to no. genes predicted by Eggnog Mapper") + geom_text_repel(aes(label=ifelse(`Contigs_in_assembly`>150,as.character(`Sample`),''), box.padding = unit(0.45, "lines")))

ggplot(data = `Culture_samples_assembly_contig_counts_to_eggnog_predicted_genes_from_assemblies`, aes(x = `Predicted_genes_EGGNOG_MAPPER`, y = `Named_predicted_proteins_EGGNOG_MAPPER`)) + geom_point(color = "#00ff8c", size = 3) + theme_linedraw() + xlab('Number of predicted genes in assembly by Eggnog Mapper') + ylab('Number of predicted named proteins from Eggnog Mapper') + labs(title="Number of Eggnog Mapper predicted genes to named predicted proteins") + geom_text_repel(aes(label=ifelse(`Named_predicted_proteins_EGGNOG_MAPPER`>50,as.character(`Sample`),''), box.padding = unit(0.45, "lines")))

nrow(`2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`[`2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$AC11J>10, ])

sum(`2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$AC11J > 10)

mean(`2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$AC11N)

ggplot(data = `2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`, aes( x = `AC11J`, y = `AC11N`)) + geom_point(color = "#00ff8c", size = 3) + theme_linedraw()
ggplot(data = `2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`, aes( x = `AC14J`, y = `AC14N`)) + geom_point(color = "#8c00ff", size = 3) + theme_linedraw()

June_to_November_changes_all_taxa_from_averages_of_all_host_by_season_combinations

longer_june_to_november_changes_all_taxons_from_average_values <- `June_to_November_changes_all_taxa_from_averages_of_all_host_by_season_combinations` %>% pivot_longer(`Abditibacterium.utsteinense`:`uncultured.rumen.bacterium`, names_to = "Microorganism", values_to = "Seasonal_change")
print longer_june_to_november_changes_all_taxons_from_average_values

ggplot(data = `Columnised_June_to_November_changes_all_taxa_from_averages_of_all_host_by_season_combinations`, aes( x = `AC_J_to_N`, y = `AR_J_to_N`)) + geom_point(color = "#ffb700", size = 1) + theme_linedraw() + xlab('Acomys cahirinus June to Nov. average mapping count change') + ylab('Acomys russatus June to Nov. average mapping count change') + labs(title="Changes in average mapping count from June to November for all unique taxons for both host species", caption="Named taxa see a change in average mapping count of 500 or greater whether positive or negative") + geom_text_repel(aes(label=ifelse(`AC_J_to_N`>500 | `AC_J_to_N`< -500 | `AR_J_to_N`>500 | `AR_J_to_N`< -500 ,as.character(`Season_change`),''), max.overlaps = 30, box.padding = unit(0.45, "lines")))

ggplot(data = `2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`, aes(x = `Microorganism`, y = `AC11J`)) + geom

stacked_trial_host <- c(rep("AC11J",6350))
stacked_trial_microorganism <- `2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Microorganism
stacked_trial_count <- `2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$AC11J
stacked_trial <- data.frame(stacked_trial_host,stacked_trial_microorganism,stacked_trial_count)
ggplot(data = `stacked_trial`, aes(fill = `stacked_trial_microorganism`, y = `stacked_trial_count`, x = `stacked_trial_host`)) + geom_bar(position="fill", stat="identity")

ggbiplot(host_and_season_average_mapping_counts.pca, obs.scale = 1, var.scale = 1, group=host_and_season_average_mapping_counts.Microorganism, varname.size = 3, labels.size=3, ellipse = TRUE, circle = TRUE) + theme(legend.direction ="horizontal", legend.position = "top") + scale_color_manual(name="Host and season", values=c("orange", "blue", "dark grey")) + scale_shape_manual(name="Host and season", values=c(17:19)) + geom_point(aes(color=host_and_season_average_mapping_counts.Microorganism, shape=host_and_season_average_mapping_counts.Microorganism, size = 3))

taxa_pca.pca <- prcomp(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns`[,c(2:6351)], center = TRUE, scale. = TRUE)
ggbiplot(`taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(color = "#8c00ff", size = 3)

ggbiplot(`taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = FALSE, var.axes = FALSE, labels=rownames(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns`), labels.size = 5) + theme_linedraw() + geom_point(color = "#8c00ff", size = 3)

ggbiplot(`taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = FALSE, var.axes = FALSE, labels=`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns`$Microorganism, labels.size = 4) + theme_linedraw() + geom_point(color = "#8c00ff", size = 1)

ggbiplot(`taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = FALSE, var.axes = FALSE, labels=`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns`$Microorganism, labels.size = 4) + theme_linedraw() + geom_point(color = "#8c00ff", size = 0.1)

ggbiplot(`taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = FALSE, var.axes = FALSE, labels=`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns`$Microorganism, labels.size = 4) + theme_linedraw()

taxa2_pca.pca <- prcomp(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`[,c(2:6351)], center = TRUE, scale. = TRUE)

ggbiplot(`taxa2_pca.pca`, obs.scale = 1, var.scale = 1, groups = `2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Host, ellipse = FALSE, circle = FALSE, var.axes = FALSE, labels=`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Microorganism, labels.size = 4) + theme_linedraw()

ggbiplot(`taxa2_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = FALSE, var.axes = FALSE, labels=2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns.Microorganism, labels.size = 4) + theme_linedraw()

ggbiplot(`taxa2_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = FALSE, var.axes = FALSE, labels=`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Microorganism, labels.size = 4) + theme_linedraw() + geom_point(aes(x, y, colour=ifelse(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Host = "AC", '#8c00ff', '#ffb700'), size = 4) + scale_color_identity())

ggbiplot(`taxa2_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = FALSE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Season), color=factor(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species")

ggbiplot(`taxa2_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = FALSE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Season), color=factor(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Host)), size = 4) + scale_color_manual(name = "Host and season", labels = c("AC-J", "AC-N", "AR-J", "AR-N"), values = c('#8c00ff','#8c00ff','#ffb700','#ffb700')) + scale_shape_manual(name ="Host and season", labels = c("AC-J", "AC-N", "AR-J", "AR-N"), values = c(17, 17, 16, 16)) + labs(shape="Season", colour="Host species")

ggbiplot(`taxa2_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = FALSE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Season), color=factor(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "PCA for all samples and all unique taxa identified by Kaiju, adjusted mapping counts")

confident_taxa_pca.pca <- prcomp(`2021_05_19_Confident_contigs_all_samples_kilobase_and_library_size_adjusted_scores_all_unique_taxa.ROWS_for_taxa_counts`[,c(2:1916)], center = TRUE, scale. = TRUE)

ggbiplot(`confident_taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`2021_05_19_Confident_contigs_all_samples_kilobase_and_library_size_adjusted_scores_all_unique_taxa.ROWS_for_taxa_counts`$Season), color=factor(`2021_05_19_Confident_contigs_all_samples_kilobase_and_library_size_adjusted_scores_all_unique_taxa.ROWS_for_taxa_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "PCA for all samples and all unique confident taxa identified by Kaiju, adjusted mapping counts", caption="Confident contigs are those where the Kaiju taxID assignment had a score over 50")

only_repeat_samples_confident_taxa_pca.pca <- prcomp(`Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`[,c(2:1916)], center = TRUE, scale. = TRUE)

ggbiplot(`only_repeat_samples_confident_taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`$Season), color=factor(`Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "PCA for repeat samples only, all unique confident taxa identified by Kaiju, adjusted mapping counts", caption="Confident contigs are those where the Kaiju taxID assignment had a score over 50")

pca_plot1 <- ggbiplot(`taxa2_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = FALSE, circle = FALSE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Season), color=factor(`2021_05_17.Modified_just_taxons_right_order_single_row_comma_separated_SAMPLEIDs_as_rows_taxa_as_columns_ALSO_has_AC_or_AR_and_J_or_N_as_last_two_columns`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species")

pca_plot2 <- ggbiplot(`confident_taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`2021_05_19_Confident_contigs_all_samples_kilobase_and_library_size_adjusted_scores_all_unique_taxa.ROWS_for_taxa_counts`$Season), color=factor(`2021_05_19_Confident_contigs_all_samples_kilobase_and_library_size_adjusted_scores_all_unique_taxa.ROWS_for_taxa_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(caption="Confident contigs are those where the Kaiju taxID assignment had a score over 50")

pca_plot_3 <- ggbiplot(`only_repeat_samples_confident_taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`$Season), color=factor(`Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(caption="Confident contigs from individuals with J and N samples")
library(grid)
library(gridExtra)
grid.arrange(pca_plot1, pca_plot2, pca_plot_3, ncol = 2)

repeats_only_no_pilots_confident_taxa_pca.pca <- prcomp(`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`[,c(2:1916)], center = TRUE, scale. = TRUE)
ggbiplot(`repeats_only_no_pilots_confident_taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`$Season), color=factor(`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "PCA for repeat samples only, all unique confident taxa identified by Kaiju, adjusted mapping counts, excluding pilot samples")

ggplot(data = `cultured_reads_those_which_assembled_kaiju_percent_contigs_most_common_species_to_fastANI_scores`, aes(x = `Percent_contigs_classed_as_most_abundant_species_by_Kaiju`, y = `FastANI_ANI_estimate`)) + geom_point(color = "#00ff8c", size = 2.5) + theme_linedraw() + xlab("% contigs classed as most abundant species by Kaiju") + ylab("FastANI ANI estimate") + labs(title = "Assemblies from cultured reads, percent contigs classified as most abundant species to ANI estimate", caption = "Named samples have ANI score above 90") + geom_text_repel(aes(label=ifelse(`FastANI_ANI_estimate`>90,as.character(`Sample`),''), max.overlaps = 10, box.padding = unit(0.45, "lines"))) + xlim(0, 100) + ylim(0, 100)

ggplot(data = bins_read_count_to_base_count, aes(x = "Bins", y = bins_read_count_to_base_count$Read)) + geom_violin(color="black", fill="dark orange") + geom_boxplot(color="black", fill="grey", width=0.1) + theme_classic() + ylab("No. bases per bin") + xlab("Bins") + labs(title = "Bases in all Bins")

long_form_improved_confident_contigs_only_repeat_samples_excluding_pilot_changes_in_kilobase_and_library_size_adjusted_mapping_scores_from_samtools_idxstats <- melt(`improved_confident_contigs_only_repeat_samples_excluding_pilot_changes_in_kilobase_and_library_size_adjusted_mapping_scores_from_samtools_idxstats`)

ggplot(data = `long_form_improved_confident_contigs_only_repeat_samples_excluding_pilot_changes_in_kilobase_and_library_size_adjusted_mapping_scores_from_samtools_idxstats`, aes(x = `variable`, y = `value`)) + geom_violin(color="black", fill='#8c00ff') + geom_boxplot(color="black", fill='#ffb700', width=0.5) + theme_classic()

ggplot(data = `long_form_improved_confident_contigs_only_repeat_samples_excluding_pilot_changes_in_kilobase_and_library_size_adjusted_mapping_scores_from_samtools_idxstats`, aes(x = `variable`, y = `value`)) + geom_violin(color="black", fill='#8c00ff') + geom_boxplot(color="black", fill='#ffb700', width=0.5) + theme_classic() + xlab("Sample") + ylab("Change in mapping count") + labs(title = "Change from June to Nov. in confident taxa mapping count, same individuals, excluding pilot samples")

ggplot(data = `long_form_improved_confident_contigs_only_repeat_samples_excluding_pilot_changes_in_kilobase_and_library_size_adjusted_mapping_scores_from_samtools_idxstats`, aes(x = `variable`, y = `value`)) + geom_violin(color="black", fill='#8c00ff') + geom_dotplot(binaxis = 'y', stackdir = 'center', position = position_dodge()) + theme_classic() + xlab("Sample") + ylab("Change in mapping count") + labs(title = "Change from June to Nov. in confident taxa mapping count, same individuals, excluding pilot samples")

ggplot(data = `long_form_improved_confident_contigs_only_repeat_samples_excluding_pilot_changes_in_kilobase_and_library_size_adjusted_mapping_scores_from_samtools_idxstats`, aes(x = `variable`, y = `value`)) + geom_violin(color="black", fill='#8c00ff') + geom_dotplot(binaxis = 'y', stackdir = 'center', position = position_dodge()) + theme_classic() + xlab("Sample") + ylab("Change in mapping count") + labs(title = "Change from June to Nov. in confident taxa mapping count, same individuals, excluding pilot samples")


just_AC_confident_contigs_repeat_samples_excludes_pilot_CHANGES.df <- select(`improved_confident_contigs_only_repeat_samples_excluding_pilot_changes_in_kilobase_and_library_size_adjusted_mapping_scores_from_samtools_idxstats`, Microorganism, AC11, AC14, AC15, AC23, AC25, AC28, AC32, AC33, AC36, AC37, AC40, AC44, AC5, AC6)

just_AR_confident_contigs_repeat_samples_excludes_pilot_CHANGES.df <- select(`improved_confident_contigs_only_repeat_samples_excluding_pilot_changes_in_kilobase_and_library_size_adjusted_mapping_scores_from_samtools_idxstats`, Microorganism, AC11, AC14, AC15, AC23, AC25, AC28, AC32, AC33, AC36, AC37, AC40, AC44, AC5, AC6)


just_AR_confident_contigs_repeat_samples_excludes_pilot.df <- select(`improved_confident_contigs_only_repeat_samples_excluding_pilot_changes_in_kilobase_and_library_size_adjusted_mapping_scores_from_samtools_idxstats`, Microorganism, AR10, AR1, AR20, AR21, AR26, AR2, AR30, AR42, AR43, AR45, AR46, AR47, AR48, AR9)

LONG_FORM_just_AC_confident_contigs_repeat_samples_excludes_pilot_CHANGES.df <- melt(`just_AC_confident_contigs_repeat_samples_excludes_pilot_CHANGES.df`)

ggplot(data = `LONG_FORM_just_AC_confident_contigs_repeat_samples_excludes_pilot_CHANGES.df`, aes(x = `variable`, y = `value`)) + geom_violin(color="black", fill='#8c00ff') + geom_boxplot(color="black", fill='#ffb700', width=0.5) + theme_classic() + ylab("Change from June to November in mapping count") + xlab("Sampled individual") + labs(title = "Change in mapping count, Acomys cahirinus, repeat sampled, confident contigs, excludes pilot samples")

LONG_FORM_just_AR_confident_contigs_repeat_samples_excludes_pilot.df <- melt(`just_AR_confident_contigs_repeat_samples_excludes_pilot.df`)

ggplot(data = `LONG_FORM_just_AR_confident_contigs_repeat_samples_excludes_pilot.df`, aes(x = `variable`, y = `value`)) + geom_violin(color="black", fill='#8c00ff') + geom_boxplot(color="black", fill='#ffb700', width=0.5) + theme_classic() + ylab("Change from June to November in mapping count") + xlab("Sampled individual") + labs(title = "Change in mapping count, Acomys russatus, repeat sampled, confident contigs, excludes pilot samples") + geom_point(color = '#8c00ff')

LF_AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot <- melt(`AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`)

ggplot(`AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`, aes(x = c(2, 4, 6, 8 ,10, 12, 14), y = c(3, 5, 7, 9, 11, 13, 15))) + geom_point()

df <- `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`(y1 = c())

recreated_LF_AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot.df <- spread(`LF_AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`, key = variable, value = value)

A = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`(x = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`$2, y = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`$3)
B = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`(x = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`$4, y = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`$5)

ggplot(data = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`) + geom_point(aes(x = J, y = N), color = '#ffff00') + geom_point(aes(x = J.1, y = N.1), color = '#ffccff') + theme_linedraw() + xlab("June mapping count") + ylab("November mapping count") + geom_point(aes(x = J.2, y = N.2), color = '#ffcccc') + geom_point(aes(x = J.3, y = N.3), color = '#ff9999') + geom_point(aes(x = J.4, y = N.4), color = '#ff9900') + geom_point(aes(x = J.5, y = N.5), color = '#ff6600') + geom_point(aes(x = J.6, y = N.6), color = '#ff3300') + geom_point(aes(x = J.7, y = N.7), color = '#ccff00') + geom_point(aes(x = J.8, y = N.8), color = '#ccccff') + geom_point(aes(x = J.9, y = N.9), color = '#cc9966') + geom_point(aes(x = J.10, y = N.10), color = '#cc6600') + geom_point(aes(x = J.11, y = N.11), color = '#cc3399') + geom_point(aes(x = J.12, y = N.12), color = '#99ccff') + geom_point(aes(x = J.13, y = N.13), color = '#6666ff')

ggplot(data = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`) + geom_point(aes(x = J, y = N)) + geom_point(aes(x = J.1, y = N.1)) + theme_linedraw() + xlab("June mapping count") + ylab("November mapping count") + geom_point(aes(x = J.2, y = N.2)) + geom_point(aes(x = J.3, y = N.3)) + geom_point(aes(x = J.4, y = N.4)) + geom_point(aes(x = J.5, y = N.5)) + geom_point(aes(x = J.6, y = N.6)) + geom_point(aes(x = J.7, y = N.7)) + geom_point(aes(x = J.8, y = N.8)) + geom_point(aes(x = J.9, y = N.9)) + geom_point(aes(x = J.10, y = N.10)) + geom_point(aes(x = J.11, y = N.11)) + geom_point(aes(x = J.12, y = N.12)) + geom_point(aes(x = J.13, y = N.13))

ggplot(data = `Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`) + geom_point(aes(x = AR10J, y = AR10N)) + geom_point(aes(x = AR1J, y = AR1N)) + theme_linedraw() + xlab("June mapping count") + ylab("November mapping count") + geom_point(aes(x = AR20J, y = AR20N)) + geom_point(aes(x = AR21J, y = AR21N)) + geom_point(aes(x = AR26J, y = AR26N)) + geom_point(aes(x = AR2J, y = AR2N)) + geom_point(aes(x = AR30J, y = AR30N)) + geom_point(aes(x = AR42J, y = AR42N)) + geom_point(aes(x = AR43J, y = AR43N)) + geom_point(aes(x = AR45J, y = AR45N)) + geom_point(aes(x = AR46J, y = AR46N)) + geom_point(aes(x = AR47J, y = AR47N)) + geom_point(aes(x = AR48J, y = AR48N)) + geom_point(aes(x = AR9J, y = AR9N))

aco_cah_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts <- ggplot(data = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`) + geom_point(aes(x = J, y = N), color = '#0073ff') + geom_point(aes(x = J.1, y = N.1), color = '#0073ff') + theme_linedraw() + xlab("June mapping count") + ylab("November mapping count") + geom_point(aes(x = J.2, y = N.2), color = '#0073ff') + geom_point(aes(x = J.3, y = N.3), color = '#0073ff') + geom_point(aes(x = J.4, y = N.4), color = '#0073ff') + geom_point(aes(x = J.5, y = N.5), color = '#0073ff') + geom_point(aes(x = J.6, y = N.6), color = '#0073ff') + geom_point(aes(x = J.7, y = N.7), color = '#0073ff') + geom_point(aes(x = J.8, y = N.8), color = '#0073ff') + geom_point(aes(x = J.9, y = N.9), color = '#0073ff') + geom_point(aes(x = J.10, y = N.10), color = '#0073ff') + geom_point(aes(x = J.11, y = N.11), color = '#0073ff') + geom_point(aes(x = J.12, y = N.12), color = '#0073ff') + geom_point(aes(x = J.13, y = N.13), color = '#0073ff') + labs(title = "A")

aco_rus_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts <- ggplot(data = `Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`) + geom_point(aes(x = AR10J, y = AR10N), color = '#ffb700') + geom_point(aes(x = AR1J, y = AR1N), color = '#ffb700') + theme_linedraw() + xlab("June mapping count") + ylab("November mapping count") + geom_point(aes(x = AR20J, y = AR20N), color = '#ffb700') + geom_point(aes(x = AR21J, y = AR21N), color = '#ffb700') + geom_point(aes(x = AR26J, y = AR26N), color = '#ffb700') + geom_point(aes(x = AR2J, y = AR2N), color = '#ffb700') + geom_point(aes(x = AR30J, y = AR30N), color = '#ffb700') + geom_point(aes(x = AR42J, y = AR42N), color = '#ffb700') + geom_point(aes(x = AR43J, y = AR43N), color = '#ffb700') + geom_point(aes(x = AR45J, y = AR45N), color = '#ffb700') + geom_point(aes(x = AR46J, y = AR46N), color = '#ffb700') + geom_point(aes(x = AR47J, y = AR47N), color = '#ffb700') + geom_point(aes(x = AR48J, y = AR48N), color = '#ffb700') + geom_point(aes(x = AR9J, y = AR9N), color = '#ffb700') + labs(title = "B")

aco_cah_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts <- ggplot(data = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`) + geom_point(aes(x = J, y = N), color = '#0073ff') + geom_point(aes(x = J.1, y = N.1), color = '#0073ff') + theme_linedraw()+ geom_point(aes(x = J.2, y = N.2), color = '#0073ff') + geom_point(aes(x = J.3, y = N.3), color = '#0073ff') + geom_point(aes(x = J.4, y = N.4), color = '#0073ff') + geom_point(aes(x = J.5, y = N.5), color = '#0073ff') + geom_point(aes(x = J.6, y = N.6), color = '#0073ff') + geom_point(aes(x = J.7, y = N.7), color = '#0073ff') + geom_point(aes(x = J.8, y = N.8), color = '#0073ff') + geom_point(aes(x = J.9, y = N.9), color = '#0073ff') + geom_point(aes(x = J.10, y = N.10), color = '#0073ff') + geom_point(aes(x = J.11, y = N.11), color = '#0073ff') + geom_point(aes(x = J.12, y = N.12), color = '#0073ff') + geom_point(aes(x = J.13, y = N.13), color = '#0073ff') + labs(title = "A") + xlab('') + ylab('')

aco_rus_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts <- ggplot(data = `Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`) + geom_point(aes(x = AR10J, y = AR10N), color = '#ffb700') + geom_point(aes(x = AR1J, y = AR1N), color = '#ffb700') + theme_linedraw() + geom_point(aes(x = AR20J, y = AR20N), color = '#ffb700') + geom_point(aes(x = AR21J, y = AR21N), color = '#ffb700') + geom_point(aes(x = AR26J, y = AR26N), color = '#ffb700') + geom_point(aes(x = AR2J, y = AR2N), color = '#ffb700') + geom_point(aes(x = AR30J, y = AR30N), color = '#ffb700') + geom_point(aes(x = AR42J, y = AR42N), color = '#ffb700') + geom_point(aes(x = AR43J, y = AR43N), color = '#ffb700') + geom_point(aes(x = AR45J, y = AR45N), color = '#ffb700') + geom_point(aes(x = AR46J, y = AR46N), color = '#ffb700') + geom_point(aes(x = AR47J, y = AR47N), color = '#ffb700') + geom_point(aes(x = AR48J, y = AR48N), color = '#ffb700') + geom_point(aes(x = AR9J, y = AR9N), color = '#ffb700') + labs(title = "B") + xlab('') + ylab('')

grid.arrange(aco_cah_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts, aco_rus_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts, ncol = 2, top = "Repeat sample individuals, confident contig taxa only, excluding pilot samples, June versus November mapping counts", bottom = "June mapping count", left = "November mapping count")

ggplot(data = `Excluding_pilot_samples_only_repeats_confident_contigs_changes_from_june_to_november_each_individual_and_average_for_host`, aes(x = `AC_average`, y = `AR_average`)) + theme_linedraw() + geom_point(color = '#8c00ff') + geom_text_repel(aes(label=ifelse(`AC_average`>500 | `AR_average`>500 | `AC_average`< -500 | `AR_average`< -500,as.character(`Microorganism`),''), max.overlaps = 10, box.padding = unit(0.45, "lines"))) + xlab("Acomys cahirinus") + ylab("Acomys russatus") + labs(title = "Average change in mapping count from June to November for bacterial taxa", caption = "Repeat sampled individuals, excluding pilot samples, confident contigs only")

install.packages("labdsv")
install.packages("BiodiversityR")

TRIAL_Vegan_alpha_diversity <- diversity(`ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples`[, `ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples`$Microorganism], index = "invsimpson", MARGIN = 2)

diversity(`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`$AC11J[, `Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`$Microorganism], MARGIN = 1, index = "invsimpson")

TESTING_melted_Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns <- melt(`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`)
TESTING_melted_ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples <- melt(`ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples`)

diversity(`NO_HEADERS_Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`, index = "simpson")

diversity(`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`[`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`$AC11J, `Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`$AC11N], index = "invsimpson")

diversity(`ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples`, index = "simpson")

ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples[] <- lapply(`ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples`, function (x) {if(is.factor(x)) as.numeric(as.character(x)) else x})

trial_alpha_diversity <- diversity(`NO_HEADERS_Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`, index = "simpson")
hist(trial_alpha_diversity$alpha)
NO_HEADERS_Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns

simpson_alpha_diversity <- diversity(ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples[, (2:1916)], index = "simpson")

ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples$simpson_alpha_diversity <- diversity(ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples[, (2:1916)], index = "simpson")

ggplot(data = ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples, aes(x = `Microorganism`, y = `simpson_alpha_diversity`)) + geom_boxplot() + theme_classic()

ggplot(data = ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples, aes(x = `simpson_alpha_diversity`)) + theme_classic() + geom_histogram(color="black", fill="#8c00ff") + ylab("Count") + xlab("Simpson")

inverse_simpson <- diversity(ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples[, (2:1916)], index = "invsimpson")

ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples$inverse_simpson_alpha_diversity <- diversity(ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples[, (2:1916)], index = "invsimpson")

ggplot(data = ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples, aes(x = HOST, y = simpson_alpha_diversity)) + geom_boxplot() + theme_linedraw() + ylab("Simpson's index of diversity") + labs(title = "Alpha diversity by host species, repeat samples and confident contigs only, excluding pilot samples") + xlab("Host species")

ggplot(data = ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples, aes(x = HOST, y = inverse_simpson_alpha_diversity)) + theme_classic() + geom_boxplot() + theme_linedraw()

contigs_to_bases <- ggplot(data = cultured_assemblies_bases_contigs_percent_contigs_in_longest_contig_N50, aes(x = Contigs, y = Bases)) + geom_point(color = '#0073ff', size = 2.5) + theme_linedraw() + scale_y_continuous(labels = comma) + xlab("Number of contigs in assembly") + ylab("Number of bases in assembly") + labs(title = "A")
contigs_to_percent_bases_in_longest_contig <- ggplot(data = cultured_assemblies_bases_contigs_percent_contigs_in_longest_contig_N50, aes(x = Contigs, y = Percent_bases_in_longest_contig)) + geom_point(color = '#ffb700', size = 2.5) + theme_linedraw() + scale_y_continuous(labels = comma) + xlab("Number of contigs in assembly") + ylab("Percent bases in longest contig") + labs(title = "B")
grid.arrange(contigs_to_bases, contigs_to_percent_bases_in_longest_contig, contigs_to_L50, ncol = 3)
contigs_to_L50 <- ggplot(data = cultured_assemblies_bases_contigs_percent_contigs_in_longest_contig_N50, aes(x = Contigs, y = L50.kb.)) + geom_point(color = '#8c00ff', size = 2.5) + theme_linedraw() + scale_y_continuous(labels = comma) + xlab("Number of contigs in assembly") + ylab("L50 (kb)") + labs(title = "C")
grid.arrange(contigs_to_bases, contigs_to_percent_bases_in_longest_contig, contigs_to_L50, ncol = 3)

species_richness_melted <- melt(species_richness_confident_contigs_only_repeats_excludes_pilot_samples)
ggplot(data = species_richness_melted) + geom_boxplot(aes(x = variable, y = value, color = variable)) + theme_linedraw() + xlab("Sample group") + ylab("No. species mapped to >0") + labs(title="Species richness for individuals with repeated samples, confident contig taxa and excluding pilot samples") + scale_color_manual(name="Host species, season and combinations", values=c('#00f3ff', '#ffb700', '#8c00ff', '#00ff8c', '#000000', '#8f8f8f', '#ac72ac', '#72ac72'))

ggplot(data = species_richness_melted) + geom_violin(aes(x = variable, y = value, color = variable), width=2.2) + geom_boxplot(aes(x = variable, y = value, color = variable), width = 0.35, alpha = 0.2) + theme_linedraw() + xlab("Sample group") + ylab("No. species mapped to >0") + labs(title="Species richness for individuals with repeated samples, confident contig taxa and excluding pilot samples") + scale_color_manual(name="Host species, season and combinations", values=c('#00f3ff', '#ffb700', '#8c00ff', '#00ff8c', '#000000', '#8f8f8f', '#ac72ac', '#72ac72')) + scale_fill_manual(values=c('#00f3ff', '#ffb700', '#8c00ff', '#00ff8c', '#000000', '#8f8f8f', '#ac72ac', '#72ac72'))

melted_Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns.df <- melt(Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns)

melted_Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns.df <- melt(Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns, id.vars=)

barebones_sampleid_as_first_row_taxa_counts_as_columns.df <- melt(Just_ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples, id.vars=1)

melt(Just_COLUMNS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples, id.vars = "Microorganism", measure.vars = 2:1916)

diversity(No_characters_in_species_ID_Just_COLUMNS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples, index = "simpson")

ggplot(data = melted_No_characters_in_species_ID_Just_COLUMNS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples) + geom_boxplot(aes(x = variable, y = value)) + theme_linedraw()

ggplot(Just_COLUMNS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples) + geom_jitter(aes(x = "AC11J", y = `AC11J`), color = '#00f3ff') + geom_jitter(aes(x = "AC11N", y = `AC11N`), color = '#0074ff') + theme_linedraw() + ylab("Mapping count per taxon") + xlab("Sample") + geom_jitter(aes(x = "AR10J", y = `AR10J`), color = '#ffb700') + geom_jitter(aes(x = "AR10N", y = `AR10N`), color = '#b36200')

ggplot(Just_COLUMNS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples) + geom_dotplot(aes(x = "AC11J", y = `AC11J`), color = '#00f3ff') + geom_dotplot(aes(x = "AC11N", y = `AC11N`), color = '#0074ff') + theme_linedraw() + ylab("Mapping count per taxon") + xlab("Sample") + geom_dotplot(aes(x = "AR10J", y = `AR10J`), color = '#ffb700') + geom_dotplot(aes(x = "AR10N", y = `AR10N`), color = '#b36200')

ggplot(melted_Just_COLUMNS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = variable))

ggplot(Just_COLUMNS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples) + geom_jitter(aes(x = "AC11J", y = `AC11J`), color = '#00f3ff') + geom_jitter(aes(x = "AC11N", y = `AC11N`), color = '#0074ff') + theme_linedraw() + ylab("Mapping count per taxon") + xlab("Sample") + geom_jitter(aes(x = "AR10J", y = `AR10J`), color = '#ffb700') + geom_jitter(aes(x = "AR10N", y = `AR10N`), color = '#b36200') + geom_jitter(aes(x = "AC14J", y = `AC14J`), color = '#00f3ff') + geom_jitter(aes(x = "AC14N", y = `AC14N`), color = '#0074ff')+ geom_jitter(aes(x = "AR1J", y = `AR1J`), color = '#ffb700') + geom_jitter(aes(x = "AR1N", y = `AR1N`), color = '#b36200')

melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID <- melt(`2021_05_24.Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID`)

ggplot(data = melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID) + theme_linedraw() + geom_jitter(aes(x = HOST_SEASON, y = value, color = HOST_SEASON)) + ylab("Mapping counts of taxa") + xlab("") + scale_color_manual(name="Host species by season combinations", values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + labs(title = "Repeat samples only, excluding pilot samples, confident contig taxa mapping counts by host and season")

summary_of_melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID <- summaryBy(value ~ HOST_SEASON + variable , melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID, FUN = c(mean, median, sd))

ggplot(data = melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID) + theme_linedraw() + geom_jitter(aes(x = HOST_SEASON, y = value, color = HOST_SEASON)) + ylab("Mapping counts of taxa") + xlab("") + scale_color_manual(name="Host species by season combinations", values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + labs(title = "Repeat samples only, excluding pilot samples, confident contig taxa mapping counts by host and season") + geom_crossbar(data = summary_of_melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID, aes(x = HOST_SEASON, ymin = value.median, ymax = value.median))

diversity(no_taxa_headers_taxa_counts_per_sample_as_rows_taxa_ordered_by_largest_average_across_all_samples_though_averages_column_is_removed[,1:1916], index = "invsimpson")
species_per_sample <- specnumber(no_taxa_headers_taxa_counts_per_sample_as_rows_taxa_ordered_by_largest_average_across_all_samples_though_averages_column_is_removed)

inverse_simpson_per_sample <- diversity(no_taxa_headers_taxa_counts_per_sample_as_rows_taxa_ordered_by_largest_average_across_all_samples_though_averages_column_is_removed[,2:1916], index = "invsimpson")
shannon_simpson_per_sample <- diversity(no_taxa_headers_taxa_counts_per_sample_as_rows_taxa_ordered_by_largest_average_across_all_samples_though_averages_column_is_removed[,2:1916], index = "shannon")
simpson_per_sample <- diversity(no_taxa_headers_taxa_counts_per_sample_as_rows_taxa_ordered_by_largest_average_across_all_samples_though_averages_column_is_removed[,2:1916], index = "simpson")
mds_test <- metaMDS(no_taxa_headers_taxa_counts_per_sample_as_rows_taxa_ordered_by_largest_average_across_all_samples_though_averages_column_is_removed[,2:1916])
mds_test_P2 <- as.data.frame(mds_test$points)

mds_test_P2$SampleID <- rownames(no_taxa_headers_taxa_counts_per_sample_as_rows_taxa_ordered_by_largest_average_across_all_samples_though_averages_column_is_removed)

diversity(Just_ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples[,2:1916], index = "simpson")
Just_ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples$Shannon <- diversity(Just_ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples[,2:1916], index = "shannon")
Just_ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples$Species_number <- specnumber(Just_ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples[,2:1916])
ggplot(data = Just_ROWS_for_sample_counts_only_repeat_samples_confident_contigs_excluding_pilot_samples, aes(x = Inv_Simpson)) + geom_histogram(aes(y=..density..),binwidth=1, color="black", fill="white") + theme_linedraw() + geom_density(alpha=.2, fill='#8c00ff')
just_host_and_season_as_ID_in_column1_taxa_counts_as_columns$simpson <- diversity(just_host_and_season_as_ID_in_column1_taxa_counts_as_columns[,2:1916], index = "simpson")
just_host_and_season_as_ID_in_column1_taxa_counts_as_columns$invsimpson <- diversity(just_host_and_season_as_ID_in_column1_taxa_counts_as_columns[,2:1916], index = "invsimpson")
just_host_and_season_as_ID_in_column1_taxa_counts_as_columns$shannon <- diversity(just_host_and_season_as_ID_in_column1_taxa_counts_as_columns[,2:1916], index = "shannon")

host_season_and_alpha_diversity_metrics <- just_host_and_season_as_ID_in_column1_taxa_counts_as_columns %>% select(1,1917,1918,1919)
melt(host_season_and_alpha_diversity_metrics)

melted_host_season_and_alpha_diversity_metrics_df <- melt(host_season_and_alpha_diversity_metrics)

host_season_and_inverse_simpson_diversity_metrics <- just_host_and_season_as_ID_in_column1_taxa_counts_as_columns %>% select(1,1918)

ggplot(data = melted_host_season_and_inverse_simpson_diversity_metrics, aes(x = Microorganism, y = value)) + geom_boxplot()

ggplot(data = melted_host_season_and_inverse_simpson_diversity_metrics, aes(x = Microorganism, y = value, fill = Microorganism)) + stat_boxplot(geom ='errorbar') + geom_boxplot() + xlab("Sample group") + ylab("Inverse Simpson's diversity") + labs(title="Inverse Simpson's diversity for individuals with repeated samples, confident contig taxa and excluding pilot samples") + theme_linedraw() + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + labs(fill = "Group")

ggplot(data = all_bins_classified_versus_unclassified_contig_percentages, aes(x = "Bins", y = `Unidentified_contigs`)) + theme_linedraw() + geom_violin(color="black", fill='#8c00ff', alpha = 0.3) + stat_boxplot(geom ='errorbar')+ geom_boxplot(width = 0.4) + xlab("") + ylab("Percent contigs in bin unidentified") + labs(title="For each bin, percent contigs which could not be identified by Kaiju at species level")

ggplot(data = all_bins_classified_versus_unclassified_contig_percentages, aes(x = "Bins", y = `V1`)) + theme_linedraw() + geom_violin(color="black", fill='#8c00ff', alpha = 0.3) + stat_boxplot(geom ='errorbar')+ geom_boxplot(width = 0.4) + xlab("") + ylab("Percent contigs in bin unidentified") + labs(title="For each bin, percent contigs which could not be identified by Kaiju at genus level")

ggplot(data = temporr1, aes(x = "Bins", y = V1)) + theme_linedraw() + geom_violin(color="black", fill='#8c00ff', alpha = 0.3) + stat_boxplot(geom ='errorbar')+ geom_boxplot(width = 0.4) + xlab("") + ylab("Percent contigs in bin unidentified") + labs(title="For each bin, percent contigs which could not be identified by Kaiju at genus level")

bins_spec_unident <- ggplot(data = all_bins_classified_versus_unclassified_contig_percentages, aes(x = "", y = `Unidentified_contigs`)) + theme_linedraw() + geom_violin(color="black", fill='#8c00ff', alpha = 0.3) + stat_boxplot(geom ='errorbar')+ geom_boxplot(width = 0.4) + xlab("") + ylab("") + labs(title="Species")
bins_genus_unident <- ggplot(data = temporr1, aes(x = "", y = V1)) + theme_linedraw() + geom_violin(color="black", fill='#00ff8c', alpha = 0.3) + stat_boxplot(geom ='errorbar')+ geom_boxplot(width = 0.4) + xlab("") + ylab("") + labs(title="Genera")
grid.arrange(bins_spec_unident, bins_genus_unident, ncol = 3, top = "Contigs unidentified after Kaiju analysis of all bins", bottom = "Bins", left = "Percent of contigs unidentified")

bins_spec_unident <- ggplot(data = all_bins_classified_versus_unclassified_contig_percentages, aes(x = "", y = `Unidentified_contigs`)) + theme_linedraw() + geom_violin(color="black", fill='#ffb700') + stat_boxplot(geom ='errorbar')+ geom_boxplot(width = 0.4, fill = 'grey') + xlab("") + ylab("") + labs(title="Species")
bins_genus_unident <- ggplot(data = temporr1, aes(x = "", y = V1)) + theme_linedraw() + geom_violin(color="black", fill='#ffb700') + stat_boxplot(geom ='errorbar')+ geom_boxplot(width = 0.4, fill = 'grey') + xlab("") + ylab("") + labs(title="Genera")
grid.arrange(bins_spec_unident, bins_genus_unident, ncol = 3, top = "Contigs unidentified after Kaiju analysis of all bins", bottom = "Bins", left = "Percent of contigs unidentified")
ggplot(data = number_of_contigs_matching_taxon_across_all_contigs_all_bins, aes(x = "Taxa", y = `Contigs`)) + theme_linedraw() + geom_violin(color="black", fill='#ffb700') + stat_boxplot(geom ='errorbar')+ geom_boxplot(width = 0.4, fill = 'grey') + xlab("") + ylab("No. contigs to") + labs(title="Number of contigs from across all bins assigned to Taxa")

ggplot(data = `2021_05_25_confident_contigs_TAXID_to_number_of_contigs_assigned_them`, aes(x = "Taxa", y = `V2`)) + theme_linedraw() + geom_violin(color="black", fill='#ffb700') + stat_boxplot(geom ='errorbar')+ geom_boxplot(width = 0.4, fill = 'grey') + xlab("") + ylab("No. contigs") + labs(title="Number of confident contigs from across all bins assigned to Taxa")

host_season_and_shannon_diversity_metrics <- just_host_and_season_as_ID_in_column1_taxa_counts_as_columns %>% select(1,1919)
melted_host_season_and_shannon_diversity_metrics <- melt(host_season_and_shannon_diversity_metrics)
ggplot(data = melted_host_season_and_shannon_diversity_metrics, aes(x = Microorganism, y = value, fill = Microorganism)) + stat_boxplot(geom ='errorbar') + geom_boxplot() + xlab("Sample group") + ylab("Shannon's diversity") + labs(title="Shannon's diversity for individuals with repeated samples, confident contig taxa and excluding pilot samples") + theme_linedraw() + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + labs(fill = "Group")


logged_melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID <- log(melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID$value)

ggplot(data = melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID) + theme_linedraw() + geom_jitter(aes(x = HOST_SEASON, y = value, color = HOST_SEASON)) + ylab("Mapping counts of taxa") + xlab("") + scale_color_manual(name="Host species by season combinations", values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + labs(title = "Repeat samples only, excluding pilot samples, confident contig taxa mapping counts by host and season") + geom_crossbar(data = summary_of_melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID, aes(x = HOST_SEASON, ymin = value.median, ymax = value.median)) + scale_x_continuous(trans="log10", limits=c(NA,1)) + scale_y_continuous(trans="log10", limits=c(NA,1))

melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID$logged_value <- log(melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID$value)

summary_of_melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID_2 <- summaryBy(logged_value ~ HOST_SEASON + variable , melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID, FUN = c(mean, median, sd))

ggplot(data = melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID) + theme_linedraw() + geom_jitter(aes(x = HOST_SEASON, y = logged_value, color = HOST_SEASON)) + ylab("Logged mapping counts of taxa") + xlab("") + scale_color_manual(name="Host species by season combinations", values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + labs(title = "Repeat samples only, excluding pilot samples, confident contig taxa mapping counts by host and season") + geom_crossbar(data = summary_of_melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID_2, aes(x = HOST_SEASON, ymin = logged_value.median, ymax = logged_value.median))

ggplot(data = melted_Taxa_as_columns_counts_as_rows_just_species_and_season_code_for_sample_ID) + theme_linedraw() + geom_jitter(aes(x = HOST_SEASON, y = logged_value, color = HOST_SEASON)) + ylab("Logged mapping counts of taxa") + xlab("") + scale_color_manual(name="Host species by season combinations", values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + labs(title = "Repeat samples only, excluding pilot samples, confident contig taxa mapping counts by host and season")

aco_cah_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts <- ggplot(data = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`) + geom_point(aes(x = J, y = N), color = '#0073ff') + geom_point(aes(x = J.1, y = N.1), color = '#0073ff') + theme_linedraw()+ geom_point(aes(x = J.2, y = N.2), color = '#0073ff') + geom_point(aes(x = J.3, y = N.3), color = '#0073ff') + geom_point(aes(x = J.4, y = N.4), color = '#0073ff') + geom_point(aes(x = J.5, y = N.5), color = '#0073ff') + geom_point(aes(x = J.6, y = N.6), color = '#0073ff') + geom_point(aes(x = J.7, y = N.7), color = '#0073ff') + geom_point(aes(x = J.8, y = N.8), color = '#0073ff') + geom_point(aes(x = J.9, y = N.9), color = '#0073ff') + geom_point(aes(x = J.10, y = N.10), color = '#0073ff') + geom_point(aes(x = J.11, y = N.11), color = '#0073ff') + geom_point(aes(x = J.12, y = N.12), color = '#0073ff') + geom_point(aes(x = J.13, y = N.13), color = '#0073ff') + labs(title = "A") + xlab('') + ylab('') + scale_x_log10() + scale_y_log10()

aco_rus_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts <- ggplot(data = `Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`) + geom_point(aes(x = AR10J, y = AR10N), color = '#ffb700') + geom_point(aes(x = AR1J, y = AR1N), color = '#ffb700') + theme_linedraw() + geom_point(aes(x = AR20J, y = AR20N), color = '#ffb700') + geom_point(aes(x = AR21J, y = AR21N), color = '#ffb700') + geom_point(aes(x = AR26J, y = AR26N), color = '#ffb700') + geom_point(aes(x = AR2J, y = AR2N), color = '#ffb700') + geom_point(aes(x = AR30J, y = AR30N), color = '#ffb700') + geom_point(aes(x = AR42J, y = AR42N), color = '#ffb700') + geom_point(aes(x = AR43J, y = AR43N), color = '#ffb700') + geom_point(aes(x = AR45J, y = AR45N), color = '#ffb700') + geom_point(aes(x = AR46J, y = AR46N), color = '#ffb700') + geom_point(aes(x = AR47J, y = AR47N), color = '#ffb700') + geom_point(aes(x = AR48J, y = AR48N), color = '#ffb700') + geom_point(aes(x = AR9J, y = AR9N), color = '#ffb700') + labs(title = "B") + xlab('') + ylab('') + scale_x_log10() + scale_y_log10()

aco_cah_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts <- ggplot(data = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`) + geom_point(aes(x = J, y = N), color = '#0073ff') + geom_point(aes(x = J.1, y = N.1), color = '#0073ff') + theme_linedraw()+ geom_point(aes(x = J.2, y = N.2), color = '#0073ff') + geom_point(aes(x = J.3, y = N.3), color = '#0073ff') + geom_point(aes(x = J.4, y = N.4), color = '#0073ff') + geom_point(aes(x = J.5, y = N.5), color = '#0073ff') + geom_point(aes(x = J.6, y = N.6), color = '#0073ff') + geom_point(aes(x = J.7, y = N.7), color = '#0073ff') + geom_point(aes(x = J.8, y = N.8), color = '#0073ff') + geom_point(aes(x = J.9, y = N.9), color = '#0073ff') + geom_point(aes(x = J.10, y = N.10), color = '#0073ff') + geom_point(aes(x = J.11, y = N.11), color = '#0073ff') + geom_point(aes(x = J.12, y = N.12), color = '#0073ff') + geom_point(aes(x = J.13, y = N.13), color = '#0073ff') + labs(title = "A") + xlab('') + ylab('') + scale_x_continuous(trans="log10", limits=c(NA,1)) + scale_y_continuous(trans="log10", limits=c(NA,1))

aco_rus_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts <- ggplot(data = `Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`) + geom_point(aes(x = AR10J, y = AR10N), color = '#ffb700') + geom_point(aes(x = AR1J, y = AR1N), color = '#ffb700') + theme_linedraw() + geom_point(aes(x = AR20J, y = AR20N), color = '#ffb700') + geom_point(aes(x = AR21J, y = AR21N), color = '#ffb700') + geom_point(aes(x = AR26J, y = AR26N), color = '#ffb700') + geom_point(aes(x = AR2J, y = AR2N), color = '#ffb700') + geom_point(aes(x = AR30J, y = AR30N), color = '#ffb700') + geom_point(aes(x = AR42J, y = AR42N), color = '#ffb700') + geom_point(aes(x = AR43J, y = AR43N), color = '#ffb700') + geom_point(aes(x = AR45J, y = AR45N), color = '#ffb700') + geom_point(aes(x = AR46J, y = AR46N), color = '#ffb700') + geom_point(aes(x = AR47J, y = AR47N), color = '#ffb700') + geom_point(aes(x = AR48J, y = AR48N), color = '#ffb700') + geom_point(aes(x = AR9J, y = AR9N), color = '#ffb700') + labs(title = "B") + xlab('') + ylab('') + scale_x_log10() + scale_y_log10()

aco_cah_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts <- ggplot(data = `AC_samples_only_all_labelled_just_by_season_confident_contigs_only_repeat_sampled_indviduals_excluding_pilot`) + geom_point(aes(x = J, y = N), color = '#0073ff') + geom_point(aes(x = J.1, y = N.1), color = '#0073ff') + theme_linedraw()+ geom_point(aes(x = J.2, y = N.2), color = '#0073ff') + geom_point(aes(x = J.3, y = N.3), color = '#0073ff') + geom_point(aes(x = J.4, y = N.4), color = '#0073ff') + geom_point(aes(x = J.5, y = N.5), color = '#0073ff') + geom_point(aes(x = J.6, y = N.6), color = '#0073ff') + geom_point(aes(x = J.7, y = N.7), color = '#0073ff') + geom_point(aes(x = J.8, y = N.8), color = '#0073ff') + geom_point(aes(x = J.9, y = N.9), color = '#0073ff') + geom_point(aes(x = J.10, y = N.10), color = '#0073ff') + geom_point(aes(x = J.11, y = N.11), color = '#0073ff') + geom_point(aes(x = J.12, y = N.12), color = '#0073ff') + geom_point(aes(x = J.13, y = N.13), color = '#0073ff') + labs(title = "A") + xlab('') + ylab('') + scale_x_continuous(trans="log10", limits=c(NA,1)) + scale_y_continuous(trans="log10", limits=c(NA,1))

ggplot(data = `Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`) + geom_point(aes(x =log10(AR10J), y =log10(AR10N), color = '#ffb700') + geom_point(aes(x =log10(AR1J), y =log10(AR1N), color = '#ffb700') + theme_linedraw() + geom_point(aes(x =log10(AR20J), y =log10(AR20N), color = '#ffb700') + geom_point(aes(x =log10(AR21J), y =log10(AR21N), color = '#ffb700') + geom_point(aes(x =log10(AR26J), y =log10(AR26N), color = '#ffb700') + geom_point(aes(x =log10(AR2J), y =log10(AR2N), color = '#ffb700') + geom_point(aes(x =log10(AR30J), y =log10(AR30N), color = '#ffb700') + geom_point(aes(x =log10(AR42J), y =log10(AR42N), color = '#ffb700') + geom_point(aes(x =log10(AR43J), y =log10(AR43N), color = '#ffb700') + geom_point(aes(x =log10(AR45J), y =log10(AR45N), color = '#ffb700') + geom_point(aes(x =log10(AR46J), y =log10(AR46N), color = '#ffb700') + geom_point(aes(x =log10(AR47J), y =log10(AR47N), color = '#ffb700') + geom_point(aes(x =log10(AR48J), y =log10(AR48N), color = '#ffb700') + geom_point(aes(x =log10(AR9J), y =log10(AR9N), color = '#ffb700') + labs(title = "B") + xlab('') + ylab('')

repeats_only_no_pilots_confident_taxa_pca.pca <- prcomp(`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`[,c(2:1916)], center = TRUE, scale. = TRUE)
ggbiplot(`repeats_only_no_pilots_confident_taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`$Season), color=factor(`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "PCA for repeat samples only, all unique confident taxa identified by Kaiju, adjusted mapping counts, excluding pilot samples")

ggbiplot(`repeats_only_no_pilots_confident_taxa_pca.pca`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`$Season), color=factor(`Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_columns_taxa_as_rows_also_has_AC_or_AR_and_J_or_N_at_bottom_as_approriate`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "PCA for repeat samples only, all unique confident taxa identified by Kaiju, adjusted mapping counts, excluding pilot samples")

ggbiplot(`confident_taxa_pca.pca`, obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`2021_05_19_Confident_contigs_all_samples_kilobase_and_library_size_adjusted_scores_all_unique_taxa.ROWS_for_taxa_counts`$Season), color=factor(`2021_05_19_Confident_contigs_all_samples_kilobase_and_library_size_adjusted_scores_all_unique_taxa.ROWS_for_taxa_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "PCA for all samples and all unique confident taxa identified by Kaiju, adjusted mapping counts", caption="Confident contigs are those where the Kaiju taxID assignment had a score over 50")

all_samples_all_contigs_pca.pca <- prcomp(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`[,c(2:6351)], center = TRUE, scale. = TRUE)

ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species")

ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species")


Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows %>% select(2:6352)

ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species")

ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not)), size = 4) + scale_shape_manual(values = c("Pilot" = 17, "Not_pilot" = 16)) + labs(shape="Sequencing run", colour="")


pca_plot_1_all_samples_components_1_and_2 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host_and_season)), size = 4) + scale_color_manual(values = c("ACJ" = '#00f3ff', "ARJ" = '#ffb700', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_shape_manual(values = c("Not_pilot" = 17, "Pilot" = 16)) + labs(shape="Sequence run", colour="Host species and season") + labs(title = "A")

pca_plot_1_all_samples_components_1_and_3 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host_and_season)), size = 4) + scale_color_manual(values = c("ACJ" = '#00f3ff', "ARJ" = '#ffb700', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_shape_manual(values = c("Not_pilot" = 17, "Pilot" = 16)) + labs(shape="Sequence run", colour="Host species and season") + labs(title = "B")

pca_plot_1_all_samples_components_2_and_3 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host_and_season)), size = 4) + scale_color_manual(values = c("ACJ" = '#00f3ff', "ARJ" = '#ffb700', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_shape_manual(values = c("Not_pilot" = 17, "Pilot" = 16)) + labs(shape="Sequence run", colour="Host species and season") + labs(title = "C")

ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#8c00ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species")

pca_plot_2 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "B")

pca_plot_3 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "C")

pca_plot_4 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "D")

pca_plot_1 <- ggbiplot::ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not)), size = 4) + scale_shape_manual(values = c("Pilot" = 17, "Not_pilot" = 16)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#00ff8c')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "A")

data.frame(pc1 = `all_samples_all_contigs_pca.pca`$x[,1])
loadings <- `all_samples_all_contigs_pca.pca`$rotation
sdev <- `all_samples_all_contigs_pca.pca`$sdev

ggplot(as.data.frame(`all_samples_all_contigs_pca.pca`$loadings[,1:2]), aes(x=Comp.1, y=Comp.2))+geom_point()

ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not)), size = 4) + scale_shape_manual(values = c("Pilot" = 17, "Not_pilot" = 16)) + labs(shape="Sequencing run", colour="")

logged_all_samples_all_contigs <- log(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`[,c(2:6351)])
all_samples_all_contigs_types <- `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`[,1,6352:6355]

logged_all_samples_all_contigs_pca.pca <- prcomp(`logged_all_samples_all_contigs`[,c(1:6350)], center = TRUE, scale. = TRUE)

sum(is.na(`logged_all_samples_all_contigs`))

ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species")

ggbiplot(`all_samples_all_contigs_pca.pca`, select = 1:2, circle = FALSE, circle.prob = 0.69, plot.loadings = TRUE, label.loadings = FALSE, sub.loadings = 1:nrow(`loadings`), loadings.cutoff = 10, label.offset = 0, label.size = 4.5, scale.loadings = 1, col.loadings = scales::muted("red"), alpha = 1, col = grouping, shape = NULL, group.ellipse = FALSE, scale.ellipse = 1, group.cloud = FALSE, xlab = "", ylab = "", equalcoord = TRUE, size = 3, size.loadings = 1)

grid.arrange(pca_plot_1, pca_plot_2, pca_plot_3, pca_plot_4, ncol = 2)

ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host_and_season)), size = 4) + scale_color_manual(values = c("ACJ" = '#00f3ff', "ARJ" = '#ffb700', "ACN" = '#0073ff', "ARN" = '#b36200', "Pilot" = '#8c00ff', "Not_pilot" = '#00ff8c')) + scale_shape_manual(values = c("Not_pilot" = 17, "Pilot" = 16)) + labs(shape="Sequence run", colour="Host species and season") + labs(title = "A")

ggsave(filename = "~/Google Drive/PhD/Scripts/R_working_directory/combined_pca_all_samples_all_contigs_2.svg")

pca_plot_2 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "B")

pca_plot_3 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "C")

pca_plot_4 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "D")

pca_plot_1 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not)), size = 2.75) + scale_shape_manual(values = c("Pilot" = 5, "Not_pilot" = 7)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#008046')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "A")

ihs <- function(x) {
  y <- log(x + sqrt(x^2 + 1))
  return(y)
}

IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end_v2 <- data.frame(lapply(working_copy_rows_all_taxa_all_contigs_includes_strings_at_end[2:6351], ihs) )

IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end <- data.frame(lapply(working_copy_rows_all_taxa_all_contigs_includes_strings_at_end[2:6351], ihs) )

PCA_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end.pca <- prcomp(IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end[,c(5:6354)], center = TRUE, scale. = TRUE)

ggbiplot::ggbiplot(`PCA_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not), color=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not)), size = 2.75) + scale_shape_manual(values = c("Pilot" = 5, "Not_pilot" = 7)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#008046')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "A")

IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end <- cbind(IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end, SampleID = just_strings_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end$Microorganism)
IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end <- cbind(IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end, Pilot_study_or_not = just_strings_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end$Pilot_study_or_not)
IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end <- cbind(IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end, Host = just_strings_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end$Host)
IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end <- cbind(IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end, Season = just_strings_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end$Season)
IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end <- cbind(IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end, Host_and_season = just_strings_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end$Host_and_season)

PCA_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end.pca <- prcomp(IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end[,c(1:6350)], center = TRUE, scale. = TRUE)

grid.arrange(pca_plot_1, pca_plot_2, pca_plot_3, pca_plot_4, ncol = 2, top = "PCAs employing IHS transformed mapped segment counts")


pca_plot_1 <- ggbiplot::ggbiplot(`PCA_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not), color=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not)), size = 2.75) + scale_shape_manual(values = c("Pilot" = 5, "Not_pilot" = 7)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#008046')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "A")
pca_plot_2 <- ggbiplot::ggbiplot(`PCA_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Season), color=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "B")
pca_plot_3 <- ggbiplot::ggbiplot(`PCA_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end.pca`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Season), color=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "C")
pca_plot_4 <- ggbiplot::ggbiplot(`PCA_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end.pca`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Season), color=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "D")

inverse hyperbolic sine

ggsave(filename = "~/Google Drive/PhD/Scripts/R_working_directory/inverse_hyperbolic_sine_combined_pca_all_samples_all_contigs.svg")

library(bestNormalize)
data.matrix(IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end_v2, rownames.force = NA)

log1p

log1p_version_working_copy_rows_all_taxa_all_contigs <- log1p(working_copy_rows_all_taxa_all_contigs_includes_strings_at_end[2:6351])
log1p_version_working_copy_rows_all_taxa_all_contigs <- cbind(log1p_version_working_copy_rows_all_taxa_all_contigs, SampleID = just_strings_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end$Microorganism)
log1p_version_working_copy_rows_all_taxa_all_contigs <- cbind(log1p_version_working_copy_rows_all_taxa_all_contigs, Pilot_study_or_not = just_strings_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end$Pilot_study_or_not)
log1p_version_working_copy_rows_all_taxa_all_contigs <- cbind(log1p_version_working_copy_rows_all_taxa_all_contigs, Host = just_strings_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end$Host)
log1p_version_working_copy_rows_all_taxa_all_contigs <- cbind(log1p_version_working_copy_rows_all_taxa_all_contigs, Season = just_strings_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end$Season)
log1p_version_working_copy_rows_all_taxa_all_contigs <- cbind(log1p_version_working_copy_rows_all_taxa_all_contigs, Host_and_season = just_strings_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end$Host_and_season)

log1p_version_working_copy_rows_all_taxa_all_contigs_pca.pca <- prcomp(log1p_version_working_copy_rows_all_taxa_all_contigs[,c(1:6350)], center = TRUE, scale. = TRUE)

logp1_pca_plot_1 <- ggbiplot::ggbiplot(`log1p_version_working_copy_rows_all_taxa_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `log1p_version_working_copy_rows_all_taxa_all_contigs`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Pilot_study_or_not), color=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Pilot_study_or_not)), size = 2.75) + scale_shape_manual(values = c("Pilot" = 5, "Not_pilot" = 7)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#008046')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "A")
logp1_pca_plot_2 <- ggbiplot::ggbiplot(`log1p_version_working_copy_rows_all_taxa_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `log1p_version_working_copy_rows_all_taxa_all_contigs`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Season), color=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "B")
logp1_pca_plot_3 <- ggbiplot::ggbiplot(`log1p_version_working_copy_rows_all_taxa_all_contigs_pca.pca`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `log1p_version_working_copy_rows_all_taxa_all_contigs`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Season), color=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "C")
logp1_pca_plot_4 <- ggbiplot::ggbiplot(`log1p_version_working_copy_rows_all_taxa_all_contigs_pca.pca`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `log1p_version_working_copy_rows_all_taxa_all_contigs`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Season), color=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "D")

grid.arrange(logp1_pca_plot_1, logp1_pca_plot_2, logp1_pca_plot_3, logp1_pca_plot_4, ncol = 2, top = "PCAs employing log(x+1) transformed mapped segment counts")

combined_pcas_IHS_transformed_all_samples_all_contigs
ggsave(filename = "~/Google Drive/PhD/Scripts/R_working_directory/combined_pcas_IHS_transformed_all_samples_all_contigs.svg")

pca_plot_2 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "B")

pca_plot_3 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "C")

pca_plot_4 <- ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "D")

pca_plot_1 <- ggbiplot::ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not)), size = 4) + scale_shape_manual(values = c("Pilot" = 5, "Not_pilot" = 7)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#008046')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "A")

pilot_raw_pca <- ggbiplot::ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not)), size = 4) + scale_shape_manual(values = c("Pilot" = 5, "Not_pilot" = 7)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#008046')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "A")
pilot_IHS_pca <- ggbiplot::ggbiplot(`PCA_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not), color=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not)), size = 2.75) + scale_shape_manual(values = c("Pilot" = 5, "Not_pilot" = 7)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#008046')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "B")
pilot_logp1_pca <- ggbiplot::ggbiplot(`log1p_version_working_copy_rows_all_taxa_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `log1p_version_working_copy_rows_all_taxa_all_contigs`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Pilot_study_or_not), color=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Pilot_study_or_not)), size = 2.75) + scale_shape_manual(values = c("Pilot" = 5, "Not_pilot" = 7)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#008046')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "C")
grid.arrange(pilot_raw_pca, pilot_IHS_pca, pilot_logp1_pca, ncol = 2, top = "PCAs examining potential relationships between mapped read segment counts in data due to sequencing run", common.legend = TRUE, legend="bottom")

pilot_raw_pca <- ggbiplot::ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Pilot_study_or_not)), size = 4) + scale_shape_manual(values = c("Pilot" = 5, "Not_pilot" = 7)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#008046')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "A") + theme(legend.position = "none")
pilot_IHS_pca <- ggbiplot::ggbiplot(`PCA_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not), color=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Pilot_study_or_not)), size = 2.75) + scale_shape_manual(values = c("Pilot" = 5, "Not_pilot" = 7)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#008046')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "B") + theme(legend.position = "none")
pilot_logp1_pca <- ggbiplot::ggbiplot(`log1p_version_working_copy_rows_all_taxa_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `log1p_version_working_copy_rows_all_taxa_all_contigs`$Pilot_study_or_not, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Pilot_study_or_not), color=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Pilot_study_or_not)), size = 2.75) + scale_shape_manual(values = c("Pilot" = 5, "Not_pilot" = 7)) + scale_color_manual(values = c("Pilot" = '#8c00ff', "Not_pilot" = '#008046')) + labs(shape="Sequencing run", colour="Sequencing run") + labs(title = "C") + theme(legend.position = "none")
grid.arrange(pilot_raw_pca, pilot_IHS_pca, pilot_logp1_pca, ncol = 2, common.legend = TRUE, legend="bottom", top = "PCAs examining potential relationships between mapped read segment counts in data due to sequencing run")

g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}

mylegend <- g_legend(pilot_raw_pca)
combined_pca_plots_pilot_or_not <- grid.arrange(arrangeGrob(pilot_raw_pca + theme(legend.position="none"),
  pilot_IHS_pca + theme(legend.position="none"),
  pilot_logp1_pca + theme(legend.position="none"),
  nrow = 2),
  mylegend, nrow=2,heights=c(15,2), top = "PCAs examining potential relationships between mapped read segment counts in data due to sequencing run")

rm(pilot_raw_pca)
rm(pilot_IHS_pca)
rm(pilot_logp1_pca)

ggsave(filename = "~/Google Drive/PhD/Scripts/R_working_directory/pilot_study_combined_pcas_all_samples_all_contigs_raw_counts_and_two_transform.svg")

raw_pca_plot <- ggbiplot::ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "A")
IHS_pca_plot_2 <- ggbiplot::ggbiplot(`PCA_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Season), color=factor(`IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "B") + theme(legend.position = "none")
logp1_pca_plot_2 <- ggbiplot::ggbiplot(`log1p_version_working_copy_rows_all_taxa_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `log1p_version_working_copy_rows_all_taxa_all_contigs`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Season), color=factor(`log1p_version_working_copy_rows_all_taxa_all_contigs`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "C") + theme(legend.position = "none")

mylegend <- g_legend(raw_pca_plot)

combined_pca_plots_PC1_and_2 <- grid.arrange(arrangeGrob(raw_pca_plot + theme(legend.position="none"),
  IHS_pca_plot_2 + theme(legend.position="none"),
  logp1_pca_plot_2 + theme(legend.position="none"),
  nrow = 2, ncol = 2),
  mylegend, nrow=2, ncol=2, heights=c(15,4), top = "PCAs examining potential relationships between mapped read segment counts in data by host species and season")

grid.arrange(raw_pca_plot + theme(legend.position="none", IHS_pca_plot_2 + theme(legend.position="none", logp1_pca_plot_2 + theme(legend.position="none"), plotted_legend, ncol = 2, top = "PCAs examining potential relationships between mapped read segment counts in data by host species and season")

combined_pca_plots_PC1_and_2 <- grid.arrange(raw_pca_plot, IHS_pca_plot_2, logp1_pca_plot_2, ncol = 2, top = "PCAs examining potential relationships between mapped read segment counts in data by host species and season")
grid.arrange(logp1_pca_plot_1, logp1_pca_plot_2, logp1_pca_plot_3, logp1_pca_plot_4, ncol = 2, top = "PCAs employing log(x+1) transformed mapped segment counts")

legender <- get_legend(raw_pca_plot)
plotted_legend <- as_ggplot(legender)

ggsave(filename = "~/Google Drive/PhD/Scripts/R_working_directory/first_and_second_component_study_combined_pcas_all_samples_all_contigs_raw_counts_and_two_transform.svg")

molten_Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows <- melt(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`)
melter_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end <- melt(IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end)
melted_log1p_version_working_copy_rows_all_taxa_all_contigs
ggplot(data = molten_Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows) + theme_linedraw() + geom_jitter(aes(x = Host_and_season, y = value, color = Host_and_season)) + ylab("Mapped read segment counts for all taxa in group samples") + xlab("") + scale_color_manual(name="Host species by season combinations", values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + labs(title = "Untransformed counts")
ggplot(data = melter_IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end) + theme_linedraw() + geom_jitter(aes(x = Host_and_season, y = value, color = Host_and_season)) + ylab("IHS transformed, mapped read segment counts for all taxa in group samples") + xlab("") + scale_color_manual(name="Host species by season combinations", values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + labs(title = "IHS transformed counts")
ggplot(data = melted_log1p_version_working_copy_rows_all_taxa_all_contigs) + theme_linedraw() + geom_jitter(aes(x = Host_and_season, y = value, color = Host_and_season)) + ylab("Log(x+1) transformed, mapped read segment counts for all taxa in group samples") + xlab("") + scale_color_manual(name="Host species by season combinations", values=c('#00f3ff', '#0074ff', '#ffb700', '#b36200')) + labs(title = "Log(x+1) transformed counts")

aco_cah_june_and_nov_counts <- ggplot(data = `working_copy_rows_all_taxa_all_contigs_includes_strings_at_end`) + geom_point(aes(x = J, y = N), color = '#0073ff') + geom_point(aes(x = J.1, y = N.1), color = '#0073ff') + theme_linedraw()+ geom_point(aes(x = J.2, y = N.2), color = '#0073ff') + geom_point(aes(x = J.3, y = N.3), color = '#0073ff') + geom_point(aes(x = J.4, y = N.4), color = '#0073ff') + geom_point(aes(x = J.5, y = N.5), color = '#0073ff') + geom_point(aes(x = J.6, y = N.6), color = '#0073ff') + geom_point(aes(x = J.7, y = N.7), color = '#0073ff') + geom_point(aes(x = J.8, y = N.8), color = '#0073ff') + geom_point(aes(x = J.9, y = N.9), color = '#0073ff') + geom_point(aes(x = J.10, y = N.10), color = '#0073ff') + geom_point(aes(x = J.11, y = N.11), color = '#0073ff') + geom_point(aes(x = J.12, y = N.12), color = '#0073ff') + geom_point(aes(x = J.13, y = N.13), color = '#0073ff') + labs(title = "A") + xlab('') + ylab('')

aco_rus_repeat_only_confident_contigs_exclude_pilot_june_and_nov_counts <- ggplot(data = `Excluding_pilot_samples_Only_individuals_with_both_samples_confident_contigs_samples_as_rows_taxa_as_columns`) + geom_point(aes(x = AR10J, y = AR10N), color = '#ffb700') + geom_point(aes(x = AR1J, y = AR1N), color = '#ffb700') + theme_linedraw() + geom_point(aes(x = AR20J, y = AR20N), color = '#ffb700') + geom_point(aes(x = AR21J, y = AR21N), color = '#ffb700') + geom_point(aes(x = AR26J, y = AR26N), color = '#ffb700') + geom_point(aes(x = AR2J, y = AR2N), color = '#ffb700') + geom_point(aes(x = AR30J, y = AR30N), color = '#ffb700') + geom_point(aes(x = AR42J, y = AR42N), color = '#ffb700') + geom_point(aes(x = AR43J, y = AR43N), color = '#ffb700') + geom_point(aes(x = AR45J, y = AR45N), color = '#ffb700') + geom_point(aes(x = AR46J, y = AR46N), color = '#ffb700') + geom_point(aes(x = AR47J, y = AR47N), color = '#ffb700') + geom_point(aes(x = AR48J, y = AR48N), color = '#ffb700') + geom_point(aes(x = AR9J, y = AR9N), color = '#ffb700') + labs(title = "B") + xlab('') + ylab('')

ggplot(data = molten_Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows) + theme_linedraw() + geom_jitter(aes(x = Pilot_study_or_not, y = value, color = Pilot_study_or_not)) + ylab("Mapped read segment counts for all taxa in group samples") + xlab("") + scale_color_manual(name="Sequencing run", values=c('#8c00ff', '#008046')) + labs(title = "Untransformed counts")
ggplot(data = melted_log1p_version_working_copy_rows_all_taxa_all_contigs) + theme_linedraw() + geom_jitter(aes(x = Pilot_study_or_not, y = value, color = Pilot_study_or_not)) + ylab("Counts for all taxa in group samples") + xlab("") + scale_color_manual(name="Sequencing run", values=c('#8c00ff', '#008046')) + labs(title = "Log(x+1) transformed, mapped read segement counts") + theme(text=element_text(family="Tex Gyre Heroes"))
+ theme(text=element_text(family="Tex Gyre Heroes"))

2021-06-03
classifiable <- ggplot(data = `2021_06_03_all_bins_kaiju_results_only_percents_classified_excluding_viruses_and_unclassifieds`, aes(x = "", y = `Percent_reads_classified_as_taxon`)) + geom_violin(color="black", fill='#af4dff') + geom_boxplot(color="black", fill="grey", width=0.1) + theme_classic() + xlab("Bins") + ylab("Percent") + labs(title = "A")
unclassifiable <- ggplot(data = `2021_06_03_distribution_of_cannot_be_classified_to_non_viral_genus_all_bins`, aes(x = "", y = `V1`)) + geom_violin(color="black", fill='#af4dff') + geom_boxplot(color="black", fill="grey", width=0.1) + theme_classic() + xlab("Bins") + ylab("Percent") + labs(title = "B")
grid.arrange(classifiable + theme(legend.position="none", unclassifiable + theme(legend.position="none", ncol = 2, top = "Kaiju classification of Bins at species level")

classifiable <- ggplot(data = `2021_06_03_all_bins_kaiju_results_only_percents_classified_excluding_viruses_and_unclassifieds`, aes(x = "", y = `Percent_reads_classified_as_taxon`)) + geom_violin(color="black", fill='#af4dff') + geom_boxplot(color="black", fill="grey", width=0.1) + theme_classic() + xlab("") + ylab("") + labs(title = "A")
unclassifiable <- ggplot(data = `2021_06_03_distribution_of_cannot_be_classified_to_non_viral_genus_all_bins`, aes(x = "", y = `V1`)) + geom_violin(color="black", fill='#ff00f3') + geom_boxplot(color="black", fill="grey", width=0.1) + theme_classic() + xlab("") + ylab("") + labs(title = "B")
grid.arrange(classifiable + theme(legend.position="none"), unclassifiable + theme(legend.position="none"), ncol = 2, top = "Kaiju classification of Bins at species level", bottom = "Bins", left = "Percent contigs in bin")

ggplot(data = `2021_06_03.all_bins_and_number_species_detected_by_kaiju`, aes(x = "Bins", y = `Species_detected`)) + geom_violin(color="black", fill='#af4dff') + geom_boxplot(color="black", fill="grey", width=0.1) + theme_classic() + xlab("") + ylab("Number of species") + labs(title = "Number of species detected at any level by Kaiju in bin")

ggplot(data = `2021_06_03_all_bins_kaiju_occurence_of_taxon_average_percent_of_contigs_in_bin_and_standard_deviation`, aes(x = `Occurrence_across_all_bins`, y = `Average_percent_contigs_in_a_bin`)) + theme_linedraw() + geom_point()

ggplot(data = `2021_06_03.all_bins_and_number_species_detected_by_kaiju`, aes(x = `Species_detected`)) + geom_histogram(bins = 50, color = "black", fill = '#ab77d6', alpha = 0.4) + geom_vline(aes(xintercept = mean(`Species_detected`)), linetype = "dashed", size = 0.75) + theme_linedraw() + xlab("Number of species") + ylab("Number of bins") + labs(title = "Number of species detected by Kaiju in each bin")

ggplot(data = `2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`, aes(x = ))

melted_trial <- melt(`2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`, id=c("Taxon", "Occurrence_across_all_bins"))

ggplot(data = `2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`, aes(x = `Taxon`, y = "value")) + geom_dotplot(binaxis='y', stackdir='center') + theme_linedraw()

subsetted <- subset(`2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`, Occurrence_across_all_bins >= 75, select=c(Occurrence_across_all_bins:Percent_reads.264))

subsetted_melted_trial <- melt(`subsetted`, id=c("Taxon", "Occurrence_across_all_bins"))
ggplot(data = `subsetted_melted_trial`, aes(x = `Taxon`, y = `value`)) + geom_dotplot(binaxis='y', stackdir='center') + theme_linedraw()

ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins >= 130), aes(x = `Taxon`, y = `value`)) + geom_dotplot(binaxis='y', stackdir='centerwhole', binwidth = 0.35, stackratio = 0.85, color = '#ab77d6') + theme_linedraw() + geom_jitter()

ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins >= 130), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(color = '#4b0089')

ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins >= 130 | value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(color = '#4b0089') + scale_y_continuous(trans='log10')

ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins >= 130 & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.25, color = '#4b0089') + scale_y_continuous(trans='log10') + xlab("Taxon") + ylab("Log10 transformed percent contigs in a bin classified") + labs(title = "A")
ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins >= 117 & Occurrence_across_all_bins < 130  & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.15, color = '#00894b') + scale_y_continuous(trans='log10') + xlab("Taxon") + ylab("Log10 transformed percent contigs in a bin classified") + labs(title = "B")
ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins >= 110 & Occurrence_across_all_bins < 117  & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.15, color = '#890700') + scale_y_continuous(trans='log10') + xlab("Taxon") + ylab("Log10 transformed percent contigs in a bin classified") + labs(title = "C")
ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins >= 100 & Occurrence_across_all_bins < 110  & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.15, color = '#543635') + scale_y_continuous(trans='log10') + xlab("Taxon") + ylab("Log10 transformed percent contigs in a bin classified") + labs(title = "D")

jit_1 <- ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins >= 130 & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.25, color = '#4b0089', size = 0.75) + scale_y_continuous(trans='log10') + xlab("") + ylab("") + labs(title = "A") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
jit_2 <- ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins >= 117 & Occurrence_across_all_bins < 130  & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.15, color = '#00894b', size = 0.75) + scale_y_continuous(trans='log10') + xlab("Taxon") + ylab("") + xlab("") + labs(title = "B") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
jit_3 <- ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins >= 110 & Occurrence_across_all_bins < 117  & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.15, color = '#890700', size = 0.75) + scale_y_continuous(trans='log10') + xlab("Taxon") + ylab("") + xlab("") + labs(title = "C") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
jit_4 <- ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins >= 100 & Occurrence_across_all_bins < 110  & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.15, color = '#543635', size = 0.75) + scale_y_continuous(trans='log10') + xlab("Taxon") + ylab("") + xlab("") + labs(title = "D") + theme(axis.text.x = element_text(angle = 45, hjust = 1))
multi_jit <- grid.arrange(jit_1 + theme(legend.position="none"), jit_2 + theme(legend.position="none"), jit_3 + theme(legend.position="none"), jit_4 + theme(legend.position="none"), ncol = 2, nrow=2, top = "Percent contigs classified as taxon by Kaiju in a bin", left = "Log10 transformed percent contigs classified", bottom = "Taxon")
rm(jit_1)
rm(jit_2)
rm(jit_3)
rm(jit_4)
rm(multi_jit)

ggplot(subset(`subsetted_melted_trial`, Occurrence_across_all_bins < 100 & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.25, color = '#4b0089', size = 0.75) + scale_y_continuous(trans='log10') + xlab("") + ylab("") + labs(title = "A") + theme(axis.text.x = element_text(angle = 45, hjust = 1))

2021-06-04

ggplot(subset(`melted_trial`, Occurrence_across_all_bins >= 1 & Occurrence_across_all_bins < 10  & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.10, color = '#543635', size = 0.25) + scale_y_continuous(trans='log10') + xlab("Taxon") + ylab("") + xlab("") + labs(title = "D") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
ggplot(subset(`melted_trial`, Occurrence_across_all_bins >= 1 & Occurrence_across_all_bins < 10  & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.10, color = '#543635', size = 0.25) + xlab("Taxon") + ylab("") + xlab("") + labs(title = "D") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
ggplot(subset(`melted_trial`, Occurrence_across_all_bins >= 1 & value > 75), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.10, color = '#543635', size = 2.75) + xlab("Taxon") + scale_y_continuous(trans='log10') + ylab("") + xlab("") + labs(title = "D") + theme(axis.text.x = element_text(angle = 45, hjust = 1))

subsetted2 <- subset(`melted_trial`, value >= 75, select=c(Taxon:value))


ggplot(subset(`melted_trial`, Taxon == "Roseburia sp 1XD4269" & value > 0 | Taxon == "Lachnospiraceae bacterium" & value > 0), aes(x = `Taxon`, y = `value`)) + theme_linedraw() + geom_jitter(width = 0.10, color = '#543635', size = 2.75) + xlab("Taxon") + scale_y_continuous(trans='log10') + ylab("") + xlab("") + labs(title = "D") + theme(axis.text.x = element_text(angle = 45, hjust = 1))

`2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`$Mean_percent_contigs <- rowMeans(`2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`[ , c(3:267)])

`2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`$Standard_deviation_percent_contigs <- rowSds(as.matrix(`2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`[ , c(3:267)]), na.rm=TRUE)

ggplot(subset(`2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`, Mean_percent_contigs > 0), aes(x = Occurrence_across_all_bins, y = Mean_percent_contigs)) + geom_point(size = 0.75, color = '#543635') + theme_linedraw() + scale_y_continuous(trans='log10', labels = comma)

ggplot(subset(`2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`, Mean_percent_contigs > 0), aes(x = Occurrence_across_all_bins, y = Mean_percent_contigs)) + geom_point(size = 0.75, color = '#543635') + theme_linedraw() + scale_y_continuous(trans='log10', labels = comma) + geom_errorbar(aes(ymin=Mean_percent_contigs-Standard_deviation_percent_contigs, ymax=Mean_percent_contigs+Standard_deviation_percent_contigs), width = 0.35, size = 1)
ggplot(subset(`2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`, Mean_percent_contigs > 0 & Occurrence_across_all_bins > 50), aes(x = Occurrence_across_all_bins, y = Mean_percent_contigs)) + geom_point(size = 0.75, color = '#543635') + theme_linedraw() + geom_errorbar(aes(ymin=Mean_percent_contigs-Standard_deviation_percent_contigs, ymax=Mean_percent_contigs+Standard_deviation_percent_contigs), width = 0.35, size = 0.25)

ggplot(subset(`2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`, Occurrence_across_all_bins =1), aes(x = "Taxa", y = Percent_reads)) + geom_boxplot() + geom_violin() + theme_linedraw()
ggplot(data = `2021_06_03_all_unique_taxids_count_of_occurence_across_all_bins_and_the_taxon_name_and_each_percent_contig_classification_occuring`, aes(x = `Occurrence_across_all_bins`)) + geom_histogram(bins = 100, color = "black", fill = '#ab77d6', alpha = 0.4) + geom_vline(aes(xintercept = mean(`Occurrence_across_all_bins`)), linetype = "dashed", size = 0.75) + theme_linedraw() + xlab("Number of bins") + ylab("Number of taxa") + labs(title = "For unique, lowest taxomoic level available Kaiju classifications,the number of bins in which a contig\nwas classified as that taxon")

Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows

melted_mapping_counts <- melt(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`,id=c("Microorganism", "Pilot_study_or_not", "Host", "Season", "Host_and_season"))
ggplot(subset(`melted_mapping_counts`, value > 0)) + theme_linedraw() + geom_jitter(aes(x = Host_and_season, y = value, color = Host_and_season), width = 0.45, size = 0.75) + xlab("Samples") + ylab("Log10 transformed, mapped read segments") + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_y_continuous(trans='log10', labels = comma)

ggplot(subset(`melted_mapping_counts`, value > 0)) + theme_linedraw() + geom_jitter(aes(x = Host_and_season, y = value, color = Host_and_season), width = 0.45, size = 0.75) + xlab("Samples") + ylab("Log10 transformed, mapped read segments") + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_y_continuous(trans='log10', labels = comma) + guides(color = guide_legend(override.aes = list(size=3)))

ggplot(subset(`melted_mapping_counts`, value > 0)) + theme_linedraw() + geom_jitter(aes(x = Host_and_season, y = value, color = Host_and_season), width = 0.45, size = 0.75) + xlab("Samples") + ylab("Mapped read segment count") + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + guides(color = guide_legend(override.aes = list(size=3))) + labs(title = "For each host species by season combination, gives the mapped read segment count for each taxon\nwhich was present")

Making legend only symbols bigger: + guides(color = guide_legend(override.aes = list(size=3)))

ggplot(subset(`melted_mapping_counts`, value > 0)) + theme_linedraw() + geom_boxplot(aes(x = Host_and_season, y = value, color = Host_and_season), width = 0.35, alpha = 0.2) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Inverse_Simpson <- diversity(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`[,2:6351], index = "invsimpson")
`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Simpson <- diversity(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`[,2:6351], index = "simpson")
`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Shannon <- diversity(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`[,2:6351], index = "shannon")

`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`

host_by_season_and_diversity_indices <- select(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`, Microorganism, Host_and_season, Inverse_Simpson, Simpson, Shannon)
melted_host_by_season_and_diversity_indices <- melt(host_by_season_and_diversity_indices, id=c("Host_and_season", "Microorganism"))

temp_plot_1 <- ggplot(data = subset(melted_host_by_season_and_diversity_indices, variable == "Inverse_Simpson"), aes(x = Host_and_season, y = value, fill = Host_and_season)) + theme_linedraw() + stat_boxplot(geom = 'errorbar') + geom_boxplot() + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + xlab("Samples") + ylab("Inverse Simpson's diversity score") + labs(title = "For mapped read segment count scores, shows Inverse Simpson's diversity for each host species\nand season combination")
ggplot(data = subset(melted_host_by_season_and_diversity_indices, variable == "Simpson"), aes(x = Host_and_season, y = value, fill = Host_and_season)) + theme_linedraw() + stat_boxplot(geom = 'errorbar') + geom_boxplot() + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + xlab("Samples") + ylab("Simpson's diversity score") + labs(title = "For mapped read segment count scores, shows Simpson's diversity for each host species\nand season combination")
ggplot(data = subset(melted_host_by_season_and_diversity_indices, variable == "Shannon"), aes(x = Host_and_season, y = value, fill = Host_and_season)) + theme_linedraw() + stat_boxplot(geom = 'errorbar') + geom_boxplot() + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + xlab("Samples") + ylab("Shannon's diversity score") + labs(title = "For mapped read segment count scores, shows Shannon's diversity for each host species\nand season combination")

invsimpplot <- ggplot(data = subset(melted_host_by_season_and_diversity_indices, variable == "Inverse_Simpson"), aes(x = Host_and_season, y = value, fill = Host_and_season)) + theme_linedraw() + stat_boxplot(geom = 'errorbar') + geom_boxplot() + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + xlab("Samples") + ylab("Inverse Simpson's diversity score") + labs(title = "A")
simpplot <- ggplot(data = subset(melted_host_by_season_and_diversity_indices, variable == "Simpson"), aes(x = Host_and_season, y = value, fill = Host_and_season)) + theme_linedraw() + stat_boxplot(geom = 'errorbar') + geom_boxplot() + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + xlab("Samples") + ylab("Simpson's diversity score") + labs(title = "B")
shanplot <- ggplot(data = subset(melted_host_by_season_and_diversity_indices, variable == "Shannon"), aes(x = Host_and_season, y = value, fill = Host_and_season)) + theme_linedraw() + stat_boxplot(geom = 'errorbar') + geom_boxplot() + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + xlab("Samples") + ylab("Shannon's diversity score") + labs(title = "C")
plotted_diversity_legend <- as_ggplot(diversity_legend)


mylegend <- g_legend(raw_pca_plot)


g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)}
diversity_legend <- g_legend(temp_plot_1)


invsimpplot <- ggplot(data = subset(melted_host_by_season_and_diversity_indices, variable == "Inverse_Simpson"), aes(x = Host_and_season, y = value, fill = Host_and_season)) + theme_linedraw() + stat_boxplot(geom = 'errorbar') + geom_boxplot() + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + xlab("") + ylab("") + labs(title = "A") + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
simpplot <- ggplot(data = subset(melted_host_by_season_and_diversity_indices, variable == "Simpson"), aes(x = Host_and_season, y = value, fill = Host_and_season)) + theme_linedraw() + stat_boxplot(geom = 'errorbar') + geom_boxplot() + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + xlab("") + ylab("") + labs(title = "B") + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
shanplot <- ggplot(data = subset(melted_host_by_season_and_diversity_indices, variable == "Shannon"), aes(x = Host_and_season, y = value, fill = Host_and_season)) + theme_linedraw() + stat_boxplot(geom = 'errorbar') + geom_boxplot() + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + xlab("") + ylab("") + labs(title = "C") + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
plotted_diversity_legend <- as_ggplot(diversity_legend)
grid.arrange(invsimpplot + theme(legend.position="none"), simpplot + theme(legend.position="none"), shanplot + theme(legend.position="none"), plotted_diversity_legend, ncol = 2, top = "Diversity metric scores for mapped read segment counts by host species and season combination", left = "Metric score", bottom = "Sample group")


melted_host_by_season_and_diversity_indices <- melt(host_by_season_and_diversity_indices, id=c("Host_and_season", "Microorganism"))

2021_06_04
transposed <- transpose(`2021_06_04_all_contigs_all_samples_mapped_segment_counts.COPY`, make.names = "V1")
View(transposed)
melted_transposed <- melt(transposed, id=c("Contig"))
View(melted_transposed)
melted_transposed_all_samples_all_contigs <- melt(`transposed_2021_06_04_all_contigs_all_samples_mapped_segment_counts.COPY`, id=c("Contig"))

transposed <- transpose(`2021_06_06_all_contigs_all_samples_mapped_segment_counts_and_ID_things_for_each_sample`, make.names = "V1")
melted_transposed <- melt(transposed, id=c("SampleID", "Sequence_run", "Host", "Season", "Host_by_season", "Contig"))

all_samples_all_contigs_pca.pca <- prcomp(transposed[,c(2:132974)], center = TRUE, scale. = TRUE)
contig_mapping_counts.pca <- prcomp(all_contigs_all_samples_mapping_should_have_right_count_for_each_contig[,c(2:79)], center = TRUE, scale. = TRUE)

transposed[, c(2:132974)] <- sapply(transposed[, c(2:132974)], as.numeric)
ggbiplot::ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `transposed`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`transposed`$Season), color=factor(`transposed`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "A")

log1p_version_transposed_all_samples_all_contigs <- log1p(transposed[2:132974])
ihs <- function(x) {
  y <- log(x + sqrt(x^2 + 1))
  return(y)
}
log_transposed_all_samples_all_contigs_pca <- prcomp(log1p_version_transposed_all_samples_all_contigs[,c(1:132974)], center = TRUE, scale. = TRUE)
ggbiplot::ggbiplot(`log_transposed_all_samples_all_contifs_pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `log1p_version_transposed_all_samples_all_contigs`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_version_transposed_all_samples_all_contigs`$Season), color=factor(`log1p_version_transposed_all_samples_all_contigs`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "B")

IHS_version_transposed_all_samples_all_contigs <- data.frame(lapply(transposed[2:132974], ihs) )
IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end <- cbind(IHS_version_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end, SampleID = just_strings_working_copy_rows_all_taxa_all_contigs_includes_strings_at_end$Microorganism)

2021_06_09

transposed <- transpose(`2021_06_09_all_samples_all_contigs_kilobase_and_library_size_adjusted_last_bits_are_strings_for_ID_host_etc`, make.names = "Sample")
melted_transposed_all_samples_all_contigs <- melt(transposed, id=c("Sample_ID", "Sequence_run", "Host", "Season", "Host_by_season"))

all_samples_all_contigs_pca.pca <- prcomp(transposed[,c(1:132973)], center = TRUE, scale. = TRUE)
transposed[, c(1:132973)] <- sapply(transposed[, c(1:132973)], as.numeric)

temporary_taxa_levels_for_poster_figures

melted_taxa <- melt(temporary_taxa_levels_for_poster_figures, id=c("Microorganism", "Pilot_study_or_not.1", "Host.1", "Season.1", "Host_and_season.1"))

all_taxa_pca <- prcomp(temporary_taxa_levels_for_poster_figures[,c(2,12690)], center = TRUE, scale. = TRUE)

ggbiplot::ggbiplot(`all_taxa_pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `temporary_taxa_levels_for_poster_figures`$Host.1, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`temporary_taxa_levels_for_poster_figures`$Season.1), color=factor(`temporary_taxa_levels_for_poster_figures`$Host.1)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "B")

ggbiplot::ggbiplot(`all_taxa_pca`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `temporary_taxa_levels_for_poster_figures`$Host.1, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`temporary_taxa_levels_for_poster_figures`$Season.1), color=factor(`temporary_taxa_levels_for_poster_figures`$Host.1)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "B")

`2021_06_11_TEMP`
`2021_06_10_all_samples_all_contigs_kilobase_and_library_size_adjusted`
library(data.table)

transposed <- transpose(`2021_06_10_all_samples_all_contigs_kilobase_and_library_size_adjusted`, make.names = "Sample")
all_taxa_pca <- prcomp(transposed[,c(1,132973)], center = TRUE, scale. = TRUE)
transposed[, c(1:132973)] <- sapply(transposed[, c(1:132973)], as.numeric)
ggbiplot::ggbiplot(`all_taxa_pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `transposed`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`transposed`$Month), color=factor(`transposed`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "B")
ggbiplot::ggbiplot(`all_taxa_pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `transposed`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`transposed`$Month), color=factor(`transposed`$Host)), size = 2.75) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 16, "N" = 18)) + labs(shape="Season", colour="Host species") + labs(title = "All samples, all contigs")

all_taxa_pca_2 <- prcomp(`2021_06_11_TEMP`[,c(2,80)], center = TRUE, scale. = TRUE)
ggbiplot::ggbiplot(`all_taxa_pca_2`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + labs(title = "Not transposed, all contigs")

log1p__transposed_all_samples_all_contigs <- log1p(transposed[1:132973])
pca_counts <- prcomp(trimmed4, center = TRUE, scale. = TRUE)

trimmed3 <- sapply(trimmed3[, c(1:79)], as.numeric(as.character(trimmed3)))

trimmed4 <- trimmed3[-c(132973, 132974, 132975, 132976), ]
trimmed4 <- trimmed4 %>% mutate_all(as.character) %>% mutate_all(as.numeric)
ggbiplot::ggbiplot(`pca_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw()

pca_counts <- prcomp(transposed_v5, center = TRUE, scale. = TRUE)

ggbiplot::ggbiplot(`pca_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw()

ggbiplot(`all_samples_all_contigs_pca.pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Season), color=factor(`Rows_for_sampleIDs_May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species")

ggbiplot::ggbiplot(`pca_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`transposed_v3`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`transposed_v3`$Month), color=factor(`transposed_v3`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species")

hope_rows <- read.csv("/Users/osbourne/Google Drive/PhD/Scripts/R_working_directory/first_row.COPY.csv", header = TRUE)

pca_counts <- prcomp(hope_rows[,c(2:132974)], center = TRUE, scale. = TRUE)
ggbiplot::ggbiplot(`pca_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`hope_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`hope_rows`$Month), color=factor(`hope_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month", colour="Host species") + labs(title = "All samples, all contigs, contig and library size adjusted mapped read segment counts")

logged <- log1p(hope_rows[2:132974])
log_pca <- prcomp(logged, center = TRUE, scale. = TRUE)
ggbiplot::ggbiplot(`log_pca`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`hope_rows`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`hope_rows`$Month), color=factor(`hope_rows`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month", colour="Host species") + labs(title = "All samples, all contigs, contig and library size adjusted mapped read segment counts, Log+1 transformed")

melted_taxa <- melt(temporary_taxa_levels_for_poster_figures, id=c("Microorganism", "Pilot_study_or_not.1", "Host.1", "Season.1", "Host_and_season.1"))

ggplot(subset(`melted_taxa`, value > 0)) + theme_linedraw() + geom_jitter(aes(x = Host.1, y = value, color = Host.1), width = 0.45, size = 0.75) + xlab("Samples") + ylab("Mapped read segment counts") + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species", values=c('#00f3ff', '#ff8c00')) + scale_fill_manual(values=c('#00f3ff', '#ff8c00'))

ggplot(subset(`melted_taxa`, value > 0)) + theme_linedraw() + geom_jitter(aes(x = Host_and_season.1, y = value, color = Host_and_season.1), width = 0.45, size = 0.75) + xlab("Samples") + ylab("Log10 transformed, mapped read segments") + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

taxa_counts_only_season_and_host <- subset(temporary_taxa_levels_for_poster_figures, select = -c(Microorganism, Pilot_study_or_not.1, Host.1, Season.1))

melted_just_host_by_season <- melt(taxa_counts_only_season_and_host, id=c("Host_and_season.1"))

ggplot(subset(`melted_just_host_by_season`, value > 0)) + theme_linedraw() + geom_jitter(aes(x = Host_and_season.1, y = value, color = Host_and_season.1), width = 0.45, size = 0.75) + xlab("Samples") + ylab("Log10 transformed, mapped read segments") + theme(axis.text.x = element_text(hjust = 1)) + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))
RNGkind(sample.kind = "Rejection")

2021-06-20
melted <- melt(TRANSPOSED_2021_06_20_all_subsampled_reads_mapped_to_culture_SPADES_assemblies_kilobase_and_library_size_adjusted_scores, id=c("Scaffolds__", "Host", "Season", "Host_by_season", "Sequence_run"))

pca_scaffolds <- prcomp(TRANSPOSED_2021_06_20_all_subsampled_reads_mapped_to_culture_SPADES_assemblies_kilobase_and_library_size_adjusted_scores[,c(2:5801)], center = TRUE, scale. = TRUE)
require('fBasics')
colStdevs(TRANSPOSED_2021_06_20_all_subsampled_reads_mapped_to_culture_SPADES_assemblies_kilobase_and_library_size_adjusted_scores[,c(2:5801)])

melted_combo <- melt(TRANSPOSED_2021_06_20_per_culture_combined_scaffold_mapped_read_segment_count_scores_kilobase_and_library_size_adjusted_for_all_subsampled_reads, id=c("Cultures", "Host", "Season", "Host_by_season", "Sequence_run"))
pca_cultures <- prcomp(TRANSPOSED_2021_06_20_per_culture_combined_scaffold_mapped_read_segment_count_scores_kilobase_and_library_size_adjusted_for_all_subsampled_reads[,c(2:29)], center = TRUE, scale. = TRUE)

ggbiplot::ggbiplot(`pca_cultures`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`TRANSPOSED_2021_06_20_per_culture_combined_scaffold_mapped_read_segment_count_scores_kilobase_and_library_size_adjusted_for_all_subsampled_reads`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`TRANSPOSED_2021_06_20_per_culture_combined_scaffold_mapped_read_segment_count_scores_kilobase_and_library_size_adjusted_for_all_subsampled_reads`$Season), color=factor(`TRANSPOSED_2021_06_20_per_culture_combined_scaffold_mapped_read_segment_count_scores_kilobase_and_library_size_adjusted_for_all_subsampled_reads`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Mapped read segment counts for subsampled reads to SPADES assembled scaffolds from Lac cultures")

ggbiplot::ggbiplot(`pca_cultures`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`TRANSPOSED_2021_06_20_per_culture_combined_scaffold_mapped_read_segment_count_scores_kilobase_and_library_size_adjusted_for_all_subsampled_reads`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`TRANSPOSED_2021_06_20_per_culture_combined_scaffold_mapped_read_segment_count_scores_kilobase_and_library_size_adjusted_for_all_subsampled_reads`$Season), color=factor(`TRANSPOSED_2021_06_20_per_culture_combined_scaffold_mapped_read_segment_count_scores_kilobase_and_library_size_adjusted_for_all_subsampled_reads`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Mapped read segment counts for subsampled reads to SPADES assembled scaffolds from Lac cultures")

library(factoextra)
fviz_eig(pca_cultures)

fviz_pca_ind(pca_cultures, col.ind = "cos2", gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"), repel = TRUE, max.overlaps = Inf)

fviz_pca_biplot(pca_cultures, col.var = "#2E9FDF", col.ind = "#696969")

ggplot(data = `melted_combo`, aes(x = `variable`, y = `value`)) + geom_boxplot(aes(x = variable, y = value, fill = variable), width=0.6) + geom_violin(aes(x = variable, y = value), width=0.4) + theme_classic() + xlab("Culture sample") + ylab("Mapped read segment counts") + labs(title = "Mapped read segment counts for Lac. cultures to subsampled reads") + theme(axis.text.x = element_text(angle = 90))

ggplot(data = `melted_combo`, aes(x = `variable`, y = `value`)) + geom_boxplot(aes(x = variable, y = value), width=0.6) + geom_violin(aes(x = variable, y = value, fill = "black"), width=0.4) + theme_classic() + xlab("Culture sample") + ylab("Mapped read segment counts") + labs(title = "Mapped read segment counts for Lac. cultures to subsampled reads") + theme(axis.text.x = element_text(angle = 90)) + theme(legend.position="none")

2021-06-28
df_all_sam_all_contigs_all_adjustments <- `2021_06_10_all_samples_all_contigs_kilobase_and_library_size_adjusted`[!row.names(`2021_06_10_all_samples_all_contigs_kilobase_and_library_size_adjusted`) %in% c("Sequence_run", "Host", "Month", "Host_by_season"),]

copy_dataframe <- `2021_06_10_all_samples_all_contigs_kilobase_and_library_size_adjusted`[-c(132974:132977), ]

copy_dataframe <- `May_not_be_reliable_only_use_for_annual_review_2021_05_17.All_samples_kilobase_and_library_size_adjusted_all_taxons_taxa_as_rows`[-c(6351:6354), ]

write.csv(copy_dataframe,"/Users/osbourne/Google Drive/PhD/Scripts/R_working_directory")

2021-07-26
ggplot(data = `2021_07_25.all_contigs_from_reference_how_many_samples_mapped_to_them_at_all`, aes(x = `V2`)) + geom_histogram(aes(y=..density..), binwidth = 1) + theme_linedraw()
ggplot(data = `2021_07_25.all_contigs_from_reference_how_many_samples_mapped_to_them_at_all`, aes(x = `V2`)) + geom_histogram(aes(y=..count..), binwidth = 1) + theme_linedraw() + xlab("Samples mapping to contig") + ylab("Number of contigs") + labs(title = "Number of reference culture contigs to number of samples mapping to them at any value")

log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp <- log1p(TRANSPOSED.2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts[2:1308])
log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$SampleID <- TRANSPOSED.2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts$Scaffolds
melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp <- melt(log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp, id=c("SampleID"))
ggplot(data = `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value), width = 0.15, size = 0.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments")

library(data.table)
ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, SampleID=="AC11")) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value), width = 0.15, size = 0.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments")
ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl("AC11", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID))) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value), width = 0.15, size = 0.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments")
ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl("AC", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) &  grepl("J", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID))) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value), width = 0.15, size = 0.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments")

ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl("AC", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & grepl("J", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 1)) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value), width = 0.25, size = 1.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl("AC", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & grepl("J", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 1)) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value, color = SampleID), width = 0.25, size = 1.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Sample ID") + labs(title = "For contigs mapped with adjusted count > 1 in Acomys cahirinus June samples")

ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl("AC1", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & grepl("J", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 0.1)) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value, color = SampleID), width = 0.25, size = 1.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Sample ID") + labs(title = "For contigs mapped with adjusted count > 0.1 in select Acomys cahirinus June samples")

ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl("AC1", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & grepl("N", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 0.1)) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value, color = SampleID), width = 0.25, size = 1.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Sample ID") + labs(title = "For contigs mapped with adjusted count > 0.1 in select Acomys cahirinus November samples")

ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl("AC11", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 0.1)) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value, color = SampleID), width = 0.25, size = 1.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Sample ID") + labs(title = "For contigs mapped with adjusted count > 0.1 in selected Acomys cahirinus samples")

ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl("AC11", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) | grepl("AC14", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 0.1)) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value, color = SampleID), width = 0.25, size = 1.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Sample ID") + labs(title = "For contigs mapped with adjusted count > 0.1 in selected Acomys cahirinus samples")

paired_AC_to_match <- c("11", "14", "15", "16", "18", "23", "25", "28", "32", "33", "36", "37", "40", "44", "5", "6")

ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl(paste(paired_AC_to_match,collapse="|"), `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & grepl("AC1", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 0.1)) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value, color = SampleID), width = 0.25, size = 1.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Sample ID") + labs(title = "For contigs mapped with adjusted count > 0.1 in selected Acomys cahirinus samples")

ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl(paste(paired_AC_to_match,collapse="|"), `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & grepl("AC1", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID))) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value, color = SampleID), width = 0.25, size = 1.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Sample ID") + labs(title = "Log1p transformed, library and contig size adjusted mapped read segment counts in selected Acomys cahirinus samples")

ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 3)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = SampleID), width = 0.25, size = 1.25) + xlab("Contig") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Sample ID") + labs(title = "For contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment count > 3\n in at least one sample")

melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp <- melt(log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp, id=c("SampleID"))

colnames(TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts) <- gsub("X","",colnames(TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts))
log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts <- log1p(TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts[2:1308])
log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts$SampleID <- TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts$SampleID
log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts$Host <- TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts$Host
log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts$Season <- TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts$Season
log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts$Host_by_Season <- TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts$HostSeason
melted_log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts <- melt(log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts, id=c("SampleID", "Host", "Season", "Host_by_Season"))

ggplot(subset(`melted_log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`, melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 3)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_Season), width = 0.15, size = 2.25) + xlab("Contig") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Host by season") + labs(title = "For contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment count > 3\n in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(subset(`melted_log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`, melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 1)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_Season), width = 0.15, size = 2.25) + xlab("Contig") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment count > 1\n in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))


2021-07-27
ggplot(subset(`melted_log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`, melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 1)) + theme_linedraw() + geom_jitter(aes(x = Host_by_Season, y = value, color = Host_by_Season), width = 0.15, size = 2.25) + xlab("Contig") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment count > 1\n in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(subset(`melted_log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`, melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 0.1)) + theme_linedraw() + geom_jitter(aes(x = Host_by_Season, y = value, color = Host_by_Season), width = 0.15, size = 2.25) + geom_boxplot(notch = TRUE, aes(x = Host_by_Season, y = value, color = Host_by_Season, fill = Host_by_Season, alpha=0.3), outlier.shape=NA, show.legend = F) + xlab("Host species and collection month") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment count") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

pca_log1p_transposed_subsampled_reads_to_culture_mapped_read_segment_counts <- prcomp(log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts[,c(1:1307)], center = TRUE, scale. = TRUE)

ggbiplot::ggbiplot()

ggbiplot::ggbiplot(`pca_log1p_transposed_subsampled_reads_to_culture_mapped_read_segment_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Season), color=factor(`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Mapped read segment counts for subsampled reads to SPADES assembled scaffolds from Lac cultures")

ggbiplot::ggbiplot(`pca_log1p_transposed_subsampled_reads_to_culture_mapped_read_segment_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Season), color=factor(`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Mapped read segment counts for subsampled reads to SPADES assembled scaffolds from Lac cultures") + scale_fill_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700'))

ggbiplot::ggbiplot(`pca_log1p_transposed_subsampled_reads_to_culture_mapped_read_segment_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Season), color=factor(`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Mapped read segment counts for subsampled reads to SPADES assembled scaffolds from Lac cultures") + guides(fill = guide_legend(override.aes = list(shape = 21)))

ggbiplot::ggbiplot(`pca_log1p_transposed_subsampled_reads_to_culture_mapped_read_segment_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Season), color=factor(`log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted mapped read segment counts\nfor subsampled reads to SPADES assemblies from Lac cultures") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%")

library(factoextra)
fviz_eig(`pca_log1p_transposed_subsampled_reads_to_culture_mapped_read_segment_counts`)

eig.val <- get_eigenvalue(`pca_log1p_transposed_subsampled_reads_to_culture_mapped_read_segment_counts`)
res.var <- get_pca_var(`pca_log1p_transposed_subsampled_reads_to_culture_mapped_read_segment_counts`)
res.var$contrib
res.var$coord
res.ind <- get_pca_ind(`pca_log1p_transposed_subsampled_reads_to_culture_mapped_read_segment_counts`)

res.var$contrib
summary(`pca_log1p_transposed_subsampled_reads_to_culture_mapped_read_segment_counts`)
head(res.var$coord)
loadings <- pca_log1p_transposed_subsampled_reads_to_culture_mapped_read_segment_counts$rotation

ggplot(subset(`melted_log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts`, melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 0)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_Season), width = 0.05, size = 0.025) + xlab("Contig") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment count > 0\n in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + theme(axis.text.x=element_blank())

2021-08-07
log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts$Sequencing <- `2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`$Sequencing
pca_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts <- prcomp(log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts[,c(1:6357)], center = TRUE, scale. = TRUE)

ggbiplot::ggbiplot(`pca_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`$Season), color=factor(`2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p transformed, mapped read segment counts for individual TaxIDs in subsampled reads", caption="Counts were adjusted for contig and library size then summed for each TaxID from identified contigs")

Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts <- melt(log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts, id=c("SampleID", "Host", "Season", "Host_by_season", "Sequencing"))

2021-08-08
ggplot(subset(`Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`, Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts$value > 7)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.15, size = 2.25) + xlab("TaxID") + ylab("Log1p transformed, adjusted and summed mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "TaxIDs mapped with Log1p transformed, library and contig size adjusted, summed mapped read segment count > 7\n in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts, aes(x=value)) + geom_histogram(binwidth = 1, color = "#20003b", fill = "#8100eb") + geom_vline(aes(xintercept=mean(Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts$value)), color = "#00b061", linetype = "dashed", size = 1) + scale_y_continuous(labels = comma) + xlab("Log1p transformed, summed, contig and library size adjusted mapped read counts") + ylab("Count across all samples") + labs(title = "Distribution of TaxID counts across all samples") + theme_linedraw()

ggplot(data = `Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`) + theme_linedraw() + geom_jitter(aes(x = Host_by_season, y = value), width = 0.15, size = 0.25) + xlab("Host species and season") + ylab("Log1p transformed, summed, conting and library size adjusted mapped read segments")

ggplot(subset(`Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`, Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts$value > 7.5)) + theme_linedraw() + geom_jitter(aes(x = Host_by_season, y = value, color = Host_by_season), width = 0.25, size = 1.15) + xlab("Host species and collection month") + ylab("Segment counts") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Log1p transformed, summed, contig and library size adjusted mapped read segment counts per TaxID across all samples\nwith counts greater than 7.5") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(subset(`Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`, Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts$value > 7.5)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.3, size = 1.75) + xlab("TaxID") + ylab("Log1p transformed, adjusted and summed mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "TaxIDs mapped with Log1p transformed, library and contig size adjusted, summed mapped read segment count > 7.5\n in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(subset(`Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`, Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts$value > 5 & Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts$value < 7.5)) + theme_linedraw() + geom_jitter(aes(x = Host_by_season, y = value, color = Host_by_season), width = 0.25, size = 1.15) + xlab("Host species and collection month") + ylab("Segment counts") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Log1p transformed, summed, contig and library size adjusted mapped read segment counts per TaxID across all samples\nwith counts greater than 5 but lower than 7.5") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

2021-08-09
ggplot(data = `Melted_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`) + theme_linedraw() + geom_violin(aes(x = Host_by_season, y = value, fill = Host_by_season, color = Host_by_season)) + xlab("Host species and collection month") + ylab("Segment counts") + theme(axis.text.x = element_text(angle = 0, hjust = 0, size = 8)) + labs(color="Host by season") + labs(title = "Log1p transformed, summed, contig and library size adjusted mapped read segment counts per TaxID across all samples") + scale_color_manual(name="Host_by_season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

2021-08-11
log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts <- log1p(`2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`[2:6358])

log1p_TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts <- log1p(TRANSPOSED_trimmed_names_2021_07_26.Contigs_from_combined_unique_cultures_over_500bp_in_size_combined_reference.all_contigs_kilobase_and_library_size_adjusted_mapped_segment_read_counts[2:1308])
`log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Host <- `2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Host
`log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Season <- `2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Season
`log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Host_by_season <- `2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Host_by_season
`log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$SampleID <- `2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$SampleID

2021-08-12
MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts <- melt(`log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, id=c("Host", "Season", "Host_by_season", "Sequencing", "SampleID"))

ggplot(subset(`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts$value > 7.5)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host and Season") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Host by season") + labs(title = "For taxids mapped with Log1p transformed, library and contig size adjusted mapped read segment counts summed > 7.5\n in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

pca_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts <- prcomp(`log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`[,c(1:6357)], center = TRUE, scale. = TRUE)
fviz_pca_var(`pca_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, alpha.var="contrib") + theme_linedraw()
fviz_pca_var(`pca_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, select.var = list(cos2 = 0.75)) + theme_classic()

2021-08-17
library(ggalt)
ggplot(data = `2021.08.17_All_cultures_both_sets_hosts_percent_ANI_to_MIGA_nearest_match_and_nearest_culture_match`, aes(x = Percent_to_database_closest_match, xend = FastANI_to_other_assembly_nearest_match_percent, y = Sample, group = Sample)) + geom_dumbbell(color = "#ec8000", alpha = 0.75, size = 2.25, colour_x = "#8000ec", colour_xend = "#00ec80") + theme_linedraw() + xlab("Percent ANI score to match") + labs(title = "Per sample ANI percent to nearest MIGA database match and nearest other culture match", caption = "Purple dot is ANI score for closest MIGA database hit, green is for ANI score to closest culture isolate from either Acomys or LI/UK mice", hjust = 0)

2021-08-19
2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted
`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted` <- log1p(`2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`[2:1182])

`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$Host <- `2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$Host
`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$Season <- `2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$Season
`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$Host_by_season <- `2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$Host_by_season
`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$Sequencing <- `2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$Sequencing
`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$SampleID <- `2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$SampleID

`MELTED_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted` <- melt(`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`, id=c("Host", "Season", "Host_by_season", "Sequencing", "SampleID"))

`pca_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted` <- prcomp(`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`[,c(1:1181)], center = TRUE, scale. = TRUE)

`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`[,!apply(`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`, MARGIN = 2, function(x) max(x, na.rm = TRUE) == min(x, na.rm = TRUE))]

`Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted` <- which_are_constant(`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`, keep_cols = TRUE)

`Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted` <- dplyr::select(`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`, -c("S124.NODE_44", "S124.NODE_49", "S125.NODE_156", "S125.NODE_166", "S125.NODE_283", "S125.NODE_285", "S125.NODE_413", "S125.NODE_422", "S125.NODE_424", "S125.NODE_434", "S125.NODE_460", "S125.NODE_506", "S125.NODE_523", "S125.NODE_560", "S125.NODE_609", "S125.NODE_634", "S125.NODE_657", "S125.NODE_693", "S125.NODE_746", "S125.NODE_764", "S125.NODE_792", "S125.NODE_828", "S125.NODE_835", "S125.NODE_866", "S125.NODE_885", "S125.NODE_886", "S125.NODE_902", "S126.NODE_25", "S128.NODE_47", "S128.NODE_48", "S128.NODE_51"))

rm(`Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`)

`Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted` <- subset(`LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`, select=-c(S124.NODE_44, S124.NODE_49, S125.NODE_156, S125.NODE_166, S125.NODE_283, S125.NODE_285, S125.NODE_413, S125.NODE_422, S125.NODE_424, S125.NODE_434, S125.NODE_460, S125.NODE_506, S125.NODE_523, S125.NODE_560, S125.NODE_609, S125.NODE_634, S125.NODE_657, S125.NODE_693, S125.NODE_746, S125.NODE_764, S125.NODE_792, S125.NODE_828, S125.NODE_835, S125.NODE_866, S125.NODE_885, S125.NODE_886, S125.NODE_902, S126.NODE_25, S128.NODE_47, S128.NODE_48, S128.NODE_51))

`pca_Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted` <- prcomp(`Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`[,c(1:1150)], center = TRUE, scale. = TRUE)
`MELTED_Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted` <- melt(`Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`, id=c("Host", "Season", "Host_by_season", "Sequencing", "SampleID"))

ggbiplot::ggbiplot(`pca_Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$Season), color=factor(`Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted mapped read segment counts\nfor subsampled reads to SPADES assemblies from LI/UK mice Lactobacillus cultures") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%")

ggplot(subset(`MELTED_Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`, MELTED_Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted$value > 5)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host and Season") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Host by season") + labs(title = "Contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment counts > 5\nin at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(data = `MELTED_Only_variable_columns_LOG1P_2021.08.18.TRANSPOSED_Magda_provided_isolates_SPADES_assemblies_combined_Contigs_min_size_500bp.ALL_subsampled_mapped_read_counts_contig_and_library_size_adjusted`) + theme_linedraw() + geom_violin(aes(x = Host_by_season, y = value, fill = Host_by_season, color = Host_by_season)) + xlab("Host species and collection month") + ylab("Segment counts") + theme(axis.text.x = element_text(angle = 0, hjust = 0, size = 8)) + labs(color="Host by season") + labs(title = "Log1p transformed, summed, contig and library size adjusted mapped read segment counts across all\nsamples to LI/UK mice Lactobacilli isolates") + scale_color_manual(name="Host_by_season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

2021-08-20
`2021.08.20.TRANSPOSED.All_subsampled_reads_mapping_all_cultures_both_Acomys_and_LI_and_UK_mice_isolates_careful_SPADES_assemblies_min_500bp_contig_size_combined_reference`
which_are_constant(`2021.08.20.TRANSPOSED.All_subsampled_reads_mapping_all_cultures_both_Acomys_and_LI_and_UK_mice_isolates_careful_SPADES_assemblies_min_500bp_contig_size_combined_reference`, keep_cols = TRUE)
[1] "which_are_constant: TRUE aren't columns of the table, i do nothing for those variables"
[1] "which_are_constant: S125.NODE_460 is constant."
[1] "which_are_constant: S125.NODE_620 is constant."
[1] "which_are_constant: S125.NODE_628 is constant."
[1] "which_are_constant: S125.NODE_634 is constant."

shitty_temp.df <- `2021.08.20.TRANSPOSED.All_subsampled_reads_mapping_all_cultures_both_Acomys_and_LI_and_UK_mice_isolates_careful_SPADES_assemblies_min_500bp_contig_size_combined_reference`[, c(2:2489)]

log1p_shitty_temp_only_variable <- subset(log1p_shitty_temp, select=-c(S125.NODE_460, S125.NODE_620, S125.NODE_628, S125.NODE_634))
`log1p_shitty_temp_only_variable`$SampleID <- `2021.08.20.TRANSPOSED.All_subsampled_reads_mapping_all_cultures_both_Acomys_and_LI_and_UK_mice_isolates_careful_SPADES_assemblies_min_500bp_contig_size_combined_reference`$SampleID
`log1p_shitty_temp_only_variable`$Host <- `2021.08.20.TRANSPOSED.All_subsampled_reads_mapping_all_cultures_both_Acomys_and_LI_and_UK_mice_isolates_careful_SPADES_assemblies_min_500bp_contig_size_combined_reference`$Host
`log1p_shitty_temp_only_variable`$Season <- `2021.08.20.TRANSPOSED.All_subsampled_reads_mapping_all_cultures_both_Acomys_and_LI_and_UK_mice_isolates_careful_SPADES_assemblies_min_500bp_contig_size_combined_reference`$Season
`log1p_shitty_temp_only_variable`$Host_by_season <- `2021.08.20.TRANSPOSED.All_subsampled_reads_mapping_all_cultures_both_Acomys_and_LI_and_UK_mice_isolates_careful_SPADES_assemblies_min_500bp_contig_size_combined_reference`$Host_by_season
`log1p_shitty_temp_only_variable`$Sequencing <- `2021.08.20.TRANSPOSED.All_subsampled_reads_mapping_all_cultures_both_Acomys_and_LI_and_UK_mice_isolates_careful_SPADES_assemblies_min_500bp_contig_size_combined_reference`$Sequencing

PCA_shitty_log1p_only_variable <- prcomp(`log1p_shitty_temp_only_variable`[,c(1:2484)], center = TRUE, scale. = TRUE)

ggbiplot::ggbiplot(`PCA_shitty_log1p_only_variable`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_shitty_temp_only_variable`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_shitty_temp_only_variable`$Season), color=factor(`log1p_shitty_temp_only_variable`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted mapped read segment counts for\nsubsampled reads to SPADES assemblies from Acomys and LI/UK mice Lactobacillus cultures") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%")

ggbiplot::ggbiplot(`PCA_shitty_log1p_only_variable`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_shitty_temp_only_variable`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_shitty_temp_only_variable`$Season), color=factor(`log1p_shitty_temp_only_variable`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted mapped read segment counts for\nsubsampled reads to SPADES assemblies from Acomys and LI/UK mice Lactobacillus cultures") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%")

`mltd_log1p_shitty_temp_only_variable` <- melt(`log1p_shitty_temp_only_variable`, id=c("Host", "Season", "Host_by_season", "Sequencing", "SampleID"))

ggplot(data = `mltd_log1p_shitty_temp_only_variable`) + theme_linedraw() + geom_violin(aes(x = Host_by_season, y = value, fill = Host_by_season, color = Host_by_season)) + xlab("Host species and collection month") + ylab("Segment counts") + theme(axis.text.x = element_text(angle = 0, hjust = 0, size = 8)) + labs(color="Host by season") + labs(title = "Log1p transformed, summed, contig and library size adjusted mapped read segment counts across all\nsamples to Acomys and LI/UK mice Lac. isolates") + scale_color_manual(name="Host_by_season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(subset(`mltd_log1p_shitty_temp_only_variable`, mltd_log1p_shitty_temp_only_variable$value > 5)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host and Season") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Host by season") + labs(title = "Contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment counts > 5\nin at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl(paste(paired_AC_to_match,collapse="|"), `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & grepl("AC1", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID))) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value, color = SampleID), width = 0.25, size = 1.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Sample ID") + labs(title = "Log1p transformed, library and contig size adjusted mapped read segment counts in selected Acomys cahirinus samples")

ggplot(subset(`mltd_log1p_shitty_temp_only_variable`, grepl("X", mltd_log1p_shitty_temp_only_variable$variable) & mltd_log1p_shitty_temp_only_variable$value > 5.5)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host and Season") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 270)) + labs(color="Host by season") + labs(title = "Acomys isolate contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment\ncounts > 5.5 in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(subset(`mltd_log1p_shitty_temp_only_variable`, grepl("S121", mltd_log1p_shitty_temp_only_variable$variable) & mltd_log1p_shitty_temp_only_variable$value > 2)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host and Season") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + labs(color="Host by season") + labs(title = "UK/LI isolate contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment\ncounts > 2 in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

p1 <- ggplot(subset(`mltd_log1p_shitty_temp_only_variable`, grepl("X", mltd_log1p_shitty_temp_only_variable$variable) & mltd_log1p_shitty_temp_only_variable$value > 5.5)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host and Season") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 270)) + labs(color="Host by season") + labs(title = "Acomys isolate contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment\ncounts > 5.5 in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))
p2 <- ggplot(subset(`mltd_log1p_shitty_temp_only_variable`, grepl("S121", mltd_log1p_shitty_temp_only_variable$variable) & mltd_log1p_shitty_temp_only_variable$value > 2)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host and Season") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + labs(color="Host by season") + labs(title = "UK/LI isolate contigs mapped with Log1p transformed, library and contig size adjusted mapped read segment\ncounts > 2 in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

adjusted_by_isolate_sums <- `2021.08.20_TRANSPOSED.Combined_reference_all_cultures_all_subsampled_mapped_read_segment_counts_contig_and_library_size_adjusted_summed_to_sample_by_source_of_isolate_contig`[, c(7,8,9)]
log1p_transformed_adjusted_by_isolate_sums <- log1p(adjusted_by_isolate_sums)

log1p_transformed_adjusted_by_isolate_sums$SampleID <- `2021.08.20_TRANSPOSED.Combined_reference_all_cultures_all_subsampled_mapped_read_segment_counts_contig_and_library_size_adjusted_summed_to_sample_by_source_of_isolate_contig`$SampleID
log1p_transformed_adjusted_by_isolate_sums$Sequencing <- `2021.08.20_TRANSPOSED.Combined_reference_all_cultures_all_subsampled_mapped_read_segment_counts_contig_and_library_size_adjusted_summed_to_sample_by_source_of_isolate_contig`$Sequencing
log1p_transformed_adjusted_by_isolate_sums$Host <- `2021.08.20_TRANSPOSED.Combined_reference_all_cultures_all_subsampled_mapped_read_segment_counts_contig_and_library_size_adjusted_summed_to_sample_by_source_of_isolate_contig`$Host
log1p_transformed_adjusted_by_isolate_sums$Host_by_season <- `2021.08.20_TRANSPOSED.Combined_reference_all_cultures_all_subsampled_mapped_read_segment_counts_contig_and_library_size_adjusted_summed_to_sample_by_source_of_isolate_contig`$Host_by_season
log1p_transformed_adjusted_by_isolate_sums$Season <- `2021.08.20_TRANSPOSED.Combined_reference_all_cultures_all_subsampled_mapped_read_segment_counts_contig_and_library_size_adjusted_summed_to_sample_by_source_of_isolate_contig`$Season

melted_log1p_transformed_adjusted_by_isolate_sums <- melt(`log1p_transformed_adjusted_by_isolate_sums`, id=c("Host", "Season", "Host_by_season", "Sequencing", "SampleID"))

ggplot(data = `melted_log1p_transformed_adjusted_by_isolate_sums`) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host source of isolates") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts per sample,\nsummed by isolate source")

geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.15, size = 2.25)

bin_map_counts <- `2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`[, c(2:6358)]

log1p_bin_map_counts <- log1p(bin_map_counts)
melted_log1p_bin_map_counts <- melt(`log1p_bin_map_counts`, id=c("Host", "Season", "Host_by_season", "Sequencing", "SampleID"))
pca_log1p_bin_map_counts <- prcomp(`log1p_bin_map_counts`[,c(1:6357)], center = TRUE, scale. = TRUE)

ggplot(subset(`melted_log1p_bin_map_counts`, melted_log1p_bin_map_counts$value > 5 & melted_log1p_bin_map_counts$value < 7.5)) + theme_linedraw() + geom_jitter(aes(x = Host_by_season, y = value, color = Host_by_season), width = 0.25, size = 1.15) + xlab("Host species and collection month") + ylab("Segment counts") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Log1p transformed, summed, contig and library size adjusted mapped read segment counts per Taxon across all samples\nwith counts greater than 5 but lower than 7.5") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(data = `melted_log1p_bin_map_counts`) + theme_linedraw() + geom_violin(aes(x = Host_by_season, y = value, color = Host_by_season, fill = Host_by_season)) + xlab("Host species and collection month") + ylab("Segment counts") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Log1p transformed, summed, contig and library size adjusted mapped read segment counts per Taxon across all samples") + scale_color_manual(name="Host species and collection month", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + labs(fill = "Host species and collection month")

ggplot(subset(`melted_log1p_bin_map_counts`,`melted_log1p_bin_map_counts`$value > 7.5)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.3, size = 1.75) + xlab("Taxon") + ylab("Log1p transformed, adjusted and summed mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Taxa mapped with Log1p transformed, library and contig size adjusted, summed mapped read segment count > 7.5\n in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggbiplot::ggbiplot(`pca_log1p_bin_map_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_bin_map_counts`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_bin_map_counts`$Season), color=factor(`log1p_bin_map_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed taxa mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%")

fviz_pca_var(`pca_log1p_bin_map_counts`, alpha.var="contrib") + theme_linedraw()
fviz_pca_var(`pca_log1p_bin_map_counts`, select.var = list(cos2 = 0.75)) + theme_classic()

2021-08-24
fviz_pca_var(`pca_log1p_bin_map_counts`, select.var = list(cos2 = 0.80), repel = TRUE) + theme_classic()

factoextra::fviz_pca_var(`pca_log1p_bin_map_counts`, select.var = list(cos2 = 0.80), gradient.cols = c("#00AFBB", "#E7B800", "#FC4E07"), repel = TRUE) + theme_classic()

fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 1, top = 15) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))
fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 2, top = 15)

set.seed(124)
km.res <- kmeans(scale(log1p_bin_map_counts[,c(1:6357)]), 4, nstart = 25)
res_df <- scale(log1p_bin_map_counts[,c(1:6357)])
fviz_cluster(km.res, data = `res_df`, palette = c("#00AFBB","#2E9FDF", "#E7B800", "#FC4E07"), ggtheme = theme_linedraw(), main = "Who knows what this is")

km.res <- kmeans(scale(log1p_bin_map_counts[,c(1:6357)]), 8, nstart = 10)
res_df <- scale(log1p_bin_map_counts[,c(1:6357)])
fviz_cluster(km.res, data = `res_df`, palette = c("#00AFBB","#2E9FDF", "#E7B800", "#FC4E07", "#ff00bf", "#4000ff", "#ff8000", "#00802b"), ggtheme = theme_linedraw(), main = "Who knows what this is")

ggplot(subset(`melted_log1p_bin_map_counts`,`melted_log1p_bin_map_counts`$value > 7.5)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.3, size = 1.75) + xlab("Taxon") + ylab("Log1p transformed, adjusted and summed mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Taxa mapped with Log1p transformed, library and contig size adjusted, summed mapped read segment count > 7.5\n in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(subset(`melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`, grepl("AC11", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID))) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value), width = 0.15, size = 0.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments")

ggplot(subset(`melted_log1p_bin_map_counts`, grepl("ACJ", `melted_log1p_bin_map_counts`$Host_by_season) & `melted_log1p_bin_map_counts`$value == 0)) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.3, size = 1.75) + xlab("Taxon") + ylab("Log1p transformed, adjusted and summed mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Taxa mapped with Log1p transformed, library and contig size adjusted, summed mapped read segment count of 0\n in at least one sample") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ACJ_absent.df <- (subset(`melted_log1p_bin_map_counts`, grepl("ACJ", `melted_log1p_bin_map_counts`$Host_by_season) & `melted_log1p_bin_map_counts`$value == 0))
ACN_absent.df <- (subset(`melted_log1p_bin_map_counts`, grepl("ACN", `melted_log1p_bin_map_counts`$Host_by_season) & `melted_log1p_bin_map_counts`$value == 0))
ARJ_absent.df <- (subset(`melted_log1p_bin_map_counts`, grepl("ARJ", `melted_log1p_bin_map_counts`$Host_by_season) & `melted_log1p_bin_map_counts`$value == 0))
ARN_absent.df <- (subset(`melted_log1p_bin_map_counts`, grepl("ARN", `melted_log1p_bin_map_counts`$Host_by_season) & `melted_log1p_bin_map_counts`$value == 0))
unique(ACJ_absent.df[c("variable")])
unique(ACN_absent.df[c("variable")])
ACJ_absent.df <- (subset(`melted_log1p_bin_map_counts`, grepl("ACJ", `melted_log1p_bin_map_counts`$Host_by_season) & `melted_log1p_bin_map_counts`$value == 0))
log1p_bin_map_counts
write.csv(log1p_bin_map_counts, "/Users/osbourne/Google Drive/PhD/Scripts/R_working_directory/2021-08-24_log1p_bin_map_counts.csv", row.names = FALSE)

fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 1, top = 15) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))
fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 2, top = 15)
fviz_pca_var(`pca_log1p_bin_map_counts`, select.var = list(contrib = 0.25), repel = TRUE) + theme_classic()
fviz_pca_biplot(`pca_log1p_bin_map_counts`, repel = TRUE)
fviz_pca_var(`pca_log1p_bin_map_counts`, select.var = list(contrib = 0.55), repel = TRUE) + theme_classic()
fviz_pca_var(`pca_log1p_bin_map_counts`, select.var = list(contrib = 0.55), repel = TRUE) + theme_classic()

2021-08-25
get_eigenvalue(`pca_log1p_bin_map_counts`)
fviz_eig(`pca_log1p_bin_map_counts`)
ggbiplot::ggbiplot(`pca_log1p_bin_map_counts`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_bin_map_counts`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_bin_map_counts`$Season), color=factor(`log1p_bin_map_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed taxa mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%")
var <- get_pca_var(`pca_log1p_bin_map_counts`)
head(var$coord)
head(var$coord, 5)
fviz_pca_var(`pca_log1p_bin_map_counts`, col.var = "red")
corrplot(var$cos2, is.corr=FALSE, 5)
head(var$contrib)
get_pca_var(pca_log1p_bin_map_counts)

2021-08-26
ggbiplot::ggbiplot(`pca_log1p_bin_map_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_bin_map_counts`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_bin_map_counts`$Season), color=factor(`log1p_bin_map_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed taxa mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%") + theme(text = element_text(family = "serif"))

to change from default sans serif font in ggplot use: + theme(text = element_text(family = "serif"))
library(showtext)
font_paths()
font_files() %>% tibble()

Good example of piping and searching outputs here as well
font_files() %>% tibble() %>% filter(str_detect(file, "Tex Gyre Heroes"))

font_add(family ="Tex Gyre Heroes", regular = "Helvetica.ttc")
showtext_auto() - is necessary after the prior line for ggplot to use the font
ggbiplot::ggbiplot(`pca_log1p_bin_map_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_bin_map_counts`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_bin_map_counts`$Season), color=factor(`log1p_bin_map_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed taxa mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%") + theme(text = element_text(family = "Tex Gyre Heroes"))
Can put the element_text(family = "Tex Gyre Heroes") into title, label functions specifically as well

ggbiplot::ggbiplot(`pca_log1p_bin_map_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_bin_map_counts`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE, family = "Tex Gyre Heroes") + theme_linedraw() + geom_point(aes(shape=factor(`log1p_bin_map_counts`$Season), color=factor(`log1p_bin_map_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed taxa mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%") + theme(text = element_text(family = "Tex Gyre Heroes"))

2021-08-29
ggplot(subset(`melted_log1p_bin_map_counts`, grepl("Bacteria", `melted_log1p_bin_map_counts`$variable))) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.3, size = 1.75) + xlab("Taxon") + ylab("Log1p transformed, adjusted and summed mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Log1p transformed, contig and library size adjusted summed counts across sample for 'Bacteria'\nTaxID. by host species and season") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

ggplot(subset(`melted_log1p_bin_map_counts`, grepl("Lachnospiraceae.bacterium.3.1", `melted_log1p_bin_map_counts`$variable))) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.3, size = 1.75) + xlab("Taxon") + ylab("Log1p transformed, adjusted and summed mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Log1p transformed, contig and library size adjusted summed counts across sample for given\nTaxID. by host species and season") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 1, top = 15) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))
fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 2, top = 15) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))

fviz_pca_var(`pca_log1p_bin_map_counts`, select.var = list(contrib = 0.51), repel = TRUE) + theme_classic()

ggplot(subset(`melted_log1p_bin_map_counts`, grepl("^Firmicutes$", `melted_log1p_bin_map_counts`$variable) | grepl("^Bacteroidetes$", `melted_log1p_bin_map_counts`$variable))) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.3, size = 1.75) + xlab("Taxon") + ylab("Log1p transformed, adjusted and summed mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Log1p transformed, contig and library size adjusted summed counts across sample for the'Firmicutes'\nand 'Bacteroidetes' TaxIDs. by host species and season") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100'))

vegan::vegdist(`log1p_bin_map_counts`[,c(1:6357)])

log1p_bin_map_counts_MATRIX <- as.matrix(log1p_bin_map_counts[,c(1:6357)])

untransposed_log1p_bin_map_counts <- `2021.08.29_Untransposed_log1p_adjusted_summed_mapped_read_counts_for_taxa` %>% slice(1:6357)
untransposed_log1p_bin_map_counts2 <- `2021.08.29_Untransposed_log1p_adjusted_summed_mapped_read_counts_for_taxa`[2:80]

untransposed_log1p_bin_map_counts_MATRIX <- as.matrix(sapply(untransposed_log1p_bin_map_counts2, as.numeric))
untransposed_log1p_bin_map_counts_MATRIX <- data.matrix(`untransposed_log1p_bin_map_counts2`)
untransposed_log1p_bin_map_counts_MATRIX_v2 <- na.omit(untransposed_log1p_bin_map_counts_MATRIX)

summary(log1p_bin_map_counts_MATRIX)
colSums(log1p_bin_map_counts_MATRIX)
rowSums(log1p_bin_map_counts_MATRIX)
density(log1p_bin_map_counts_MATRIX[,1])

summary(untransposed_log1p_bin_map_counts_MATRIX_v2)
density(untransposed_log1p_bin_map_counts_MATRIX_v2[,1])

plot(density(untransposed_log1p_bin_map_counts_MATRIX_v2[,1]))
hist(untransposed_log1p_bin_map_counts_MATRIX_v2[,1], nclass = 100)
plot(frequency(untransposed_log1p_bin_map_counts_MATRIX_v2[,1]))

log1p_bin_map_count_densities = apply(untransposed_log1p_bin_map_counts_MATRIX_v2, 2, density)

lapply(log1p_bin_map_count_densities,lines)

plot(1,1,type="n",ylim=c(0,5),xlim=c(0,10),xlab="Log1p transformed, contig and library size adjusted, summed mapped read counts per taxon", ylab="Density")

shan_div <- apply(untransposed_log1p_bin_map_counts_MATRIX_v2, 2, vegan::diversity, index = "shannon")
which.min(shan_div)
which.max(shan_div)
lines(density(untransposed_log1p_bin_map_counts_MATRIX_v2[,"AC16N"]), col = "blue")
lines(density(untransposed_log1p_bin_map_counts_MATRIX_v2[,"AC16J"]), col = "green")
lines(density(untransposed_log1p_bin_map_counts_MATRIX_v2[,"AC14J"]), col = "orange")
lines(density(untransposed_log1p_bin_map_counts_MATRIX_v2[,"AC14N"]), col = "red")

pilot_excluded_log1p_bin_map_counts <- `2021.08.29_Excluding_pilot_Untransposed_log1p_adjusted_summed_mapped_read_counts_for_taxa`
pilot_excluded_log1p_bin_map_counts_MATRIX <- as.matrix(pilot_excluded_log1p_bin_map_counts)
summary(pilot_excluded_log1p_bin_map_counts_MATRIX)
pilot_excluded_log1p_bin_map_count_densities = apply(pilot_excluded_log1p_bin_map_counts_MATRIX, 2, density)

lapply(pilot_excluded_log1p_bin_map_count_densities,lines)
shan_div_exclude_pilot <- apply(pilot_excluded_log1p_bin_map_counts_MATRIX, 2, vegan::diversity, index = "shannon")
which.min(shan_div_exclude_pilot)
which.max(shan_div_exclude_pilot)

OnceOrMoreOftenPresent = pilot_excluded_log1p_bin_map_counts_MATRIX>0
richness_pilot_excluded_log1p_bin_map_counts_MATRIX = apply(OnceOrMoreOftenPresent, 2, sum)
order(richness_pilot_excluded_log1p_bin_map_counts_MATRIX)

plot(richness_pilot_excluded_log1p_bin_map_counts_MATRIX)
which.min(richness_pilot_excluded_log1p_bin_map_counts_MATRIX)

ttest_trial_host_df <- `melted_log1p_bin_map_counts` %>% filter(Host == "AC" | Host == "AR") %>% select(Host, value)
summary(ttest_trial_host_df %>% filter(Host == "AC") %>% .$value)
summary(ttest_trial_host_df %>% filter(Host == "AR") %>% .$value)
ggplot(ttest_trial_host_df, aes(Host, value)) + geom_boxplot()
t.test(value ~ Host, data = ttest_trial_host_df)
wilcox.test(value ~ Host, data = ttest_trial_host_df)


ttest_trial_season_df <- `melted_log1p_bin_map_counts` %>% filter(Season == "J" | Season == "N") %>% select(Season, value)
t.test(value ~ Season, data = ttest_trial_season_df)
wilcox.test(value ~ Season, data = ttest_trial_season_df)

#Bit below would need initial data to only be the paired samples so both seasons had same number of samples
ttest_trial_AChostXseason_df <- `melted_log1p_bin_map_counts` %>% filter(Host_by_season == "ACJ" | Host_by_season == "ACN") %>% select(Host_by_season, value)
ggplot(ttest_trial_AChostXseason_df, aes(Host_by_season, value)) + geom_boxplot()
t.test(value ~ Host_by_season, data = ttest_trial_AChostXseason_df, paired = TRUE)

library(ggpubr)
ggboxplot(ttest_trial_host_df$value, width = 0.5, add = c("mean", "jitter")) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

gghistogram(ttest_trial_host_df, x = "value", y = "..density..", fill = "#d39dff", bins = 100, add_density = TRUE) + xlabs("Log1p transformed, contig and library size adjusted, summed mapped read segment counts per taxon in all samples")

ggplot(data = ttest_trial_host_df, aes(x = value)) + geom_histogram(aes(y = ..density..), color = "black", fill = "#d39dff", bins = 100, alpha = 0.75) + geom_density(alpha=.25, fill="#ffb151") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Log1p transformed, summed, contig and library size adjusted mapped read counts across all samples") + ylab("Density") + labs(title = "Mapped read segment counts for each taxon across all samples")
ggplot(data = ttest_trial_host_df, aes(x = value)) + geom_density(alpha=.85, fill="#ffb151") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Log1p transformed, summed, contig and library size adjusted mapped read counts across all samples") + ylab("Density") + labs(title = "Mapped read segment counts for each taxon across all samples")

tapply(value,Host,median,)
ttest_trial_host_df

par(mfrow=c(1,2))
hist(ttest_trial_host_df["GROUP" == 'AC'], main = "Histogram for Acomys cahirinus", xlab='Mapped read segment counts*')

ggplot(subset(`ttest_trial_host_df`, Host == "AC")) + geom_histogram(aes(x = value), binwidth = 0.5, color = "black", fill = "#0475ff") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("Count") + xlab("Mapped read segment counts") + labs(title = "Acomys cahirinus") + ylim(0, 125000)
ggplot(subset(`ttest_trial_host_df`, Host == "AR")) + geom_histogram(aes(x = value), binwidth = 0.5, color = "black", fill = "#ff8e04") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("Count") + xlab("Mapped read segment counts") + labs(title = "Acomys russatus") + ylim(0, 125000)

ggplot(melted_log1p_bin_map_counts, aes(value, fill = Host)) + geom_density(alpha = 0.45) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("Log1p transformed, summed per taxon, contig and library size adjusted mapped read segment counts") + ylab("Density") + labs(title = "Distribution of mapping counts across all samples for all taxa by host species")

ggplot(melted_log1p_bin_map_counts, aes(value, fill = Season)) + geom_density(alpha = 0.45) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("Log1p transformed, summed per taxon, contig and library size adjusted mapped read segment counts") + ylab("Density") + labs(title = "Distribution of mapping counts across all samples for all taxa by season")

ggplot(melted_log1p_bin_map_counts, aes(value, fill = Host_by_season)) + geom_density(alpha = 0.65) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("Log1p transformed, summed per taxon, contig and library size adjusted mapped read segment counts") + ylab("Density") + labs(title = "Distribution of mapping counts across all samples for all taxa by host species and season")

hist(subset(melted_log1p_bin_map_counts, melted_log1p_bin_map_counts$Host == "AC"))

wilcox.test(value ~ Host, data = ttest_trial_host_df)
wilcox.test(value ~ Season, data = ttest_trial_season_df)

cor.mat <- round(cor(`log1p_bin_map_counts`[,c(1:6357)]),2)
corrplot(cor.mat, type="upper", order="hclust", tl.col="black", tl.srt=45)
library(lares)

corr_cross(cor.mat, max_pvalue = 0.05, top = 10)

corr_var(cor.mat, Root, top = 5)
summary(cor.mat)

2021-08-30
library(ggcorrplot)
library(quantmod)
library(corrr)

cor2_trial <- cor(as.matrix(`log1p_bin_map_counts`[,c(1:6357)]))
cor2_trial <- as.data.frame(cor2_trial)

corr_var(cor2_trial, Root, top = 5)

ggcorrplot(cor2_trial, hc.order = TRUE, type = "lower", lab = TRUE)

fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 1, top = 5) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + theme(text = element_text(family = "Tex Gyre Heroes"))
fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 2, top = 5) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + theme(text = element_text(family = "Tex Gyre Heroes"))
fviz_pca_var(`pca_log1p_bin_map_counts`, select.var = list(contrib = 0.51), repel = TRUE) + theme_classic()
fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 1, top = 15) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))
fviz_pca_var(`pca_log1p_bin_map_counts`, alpha.var="contrib") + theme_linedraw()

fviz_pca_var(`pca_log1p_bin_map_counts`, top = 10) + theme_linedraw()

corr_var(cor2_trial, Lachnospiraceae.bacterium.3.1, top = 10)

`2021.08.30_Averages_and_medians_all_samples_and_by_host`

median_plot1 <- ggplot(data = `2021.08.30_Averages_and_medians_all_samples_and_by_host`, aes(x = AC.Median, y = AR.Median)) + geom_point(color = "#8c00ff") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("AC median count") + ylab("AR median count") + labs(title = "Host species median counts")
median_plot2 <- ggplot(data = `2021.08.30_Averages_and_medians_all_samples_and_by_host`, aes(x = June.Median, y = November.median)) + geom_point(color = "#ff0073") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("June median count") + ylab("November median count") + labs(title = "Sampling month median counts")
median_plot3 <- ggplot(data = `2021.08.30_Averages_and_medians_all_samples_and_by_host`, aes(x = ACJ.Median, y = ACN.Median)) + geom_point(color = "#4d9dff") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("ACJ median count") + ylab("ACN median count") + labs(title = "Acomys cahirinus sampling month median counts")
median_plot4 <- ggplot(data = `2021.08.30_Averages_and_medians_all_samples_and_by_host`, aes(x = ARJ.Median, y = ARN.Median)) + geom_point(color = "#ff8c00") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("ARJ median count") + ylab("ARN median count") + labs(title = "Acomys russatus sampling month median counts")
grid.arrange(median_plot1, median_plot2, median_plot3, median_plot4, ncol = 2)


ggbiplot::ggbiplot(`pca_log1p_bin_map_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_bin_map_counts`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_bin_map_counts`$Season), color=factor(`log1p_bin_map_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed taxa mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%") + theme(text = element_text(family = "Tex Gyre Heroes"))

fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 1, top = 10, color = "black", fill = "#8c00ff") + theme_linedraw() + theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 11)) + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Ten greatest contributing variables to Dimension-1") + xlab("Taxon")
fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 2, top = 8, color = "black", fill = "#00f2ff") + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 11)) + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Ten greatest contributing variables to Dimension-2") + xlab("Taxon")
fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 3, top = 10, color = "black", fill = "#f2ff00") + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 10)) + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Ten greatest contributing variables to Dimension-3") + xlab("Taxon")

fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 2, top = 30, color = "black", fill = "#00f2ff") + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 11)) + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Ten greatest contributing variables to Dimension-2") + xlab("Taxon")

Lachnospiraceae.bacterium.3.1
melted_log1p_bin_map_counts

mann-whitney AKA wilcox.test(y,x) # where y and x are numeric

lachnospiraceae_bacterium_3_1_grouped_by_host

ggpubr::ggboxplot(subset(`melted_log1p_bin_map_counts`, grepl("Lachnospiraceae.bacterium.3.1", `melted_log1p_bin_map_counts`$variable)), x = `melted_log1p_bin_map_counts`$Host, y = `melted_log1p_bin_map_counts`$value, color = `melted_log1p_bin_map_counts`$Host, palette = c("#0073ff", "#ff8c00"), ylab = "Log1p transformed mapping count", xlab = "Host species")

ggpubr::ggboxplot(subset(`melted_log1p_bin_map_counts`, grepl("Lachnospiraceae.bacterium.3.1", `melted_log1p_bin_map_counts`$variable)), x = "Host", y = "value", color = `melted_log1p_bin_map_counts`$Host, palette = c("#0073ff", "#ff8c00"), ylab = "Log1p transformed mapping count", xlab = "Host species")

lach_host_test_df <- subset(`melted_log1p_bin_map_counts`, grepl("^Lachnospiraceae.bacterium.3.1$", `melted_log1p_bin_map_counts`$variable))

ggpubr::ggboxplot(lach_host_test_df, x = "Host", y = "value", color = "Host", palette = c("#0073ff", "#ff8c00"), ylab = "Log1p transformed mapping count", xlab = "Host species") + stat_compare_means(aes(label = ..p.signif..))
wilcox.test(value ~ Host, data = lach_host_test_df, exact = FALSE)

"^Firmicutes$"

Pygmaiobacter.massiliensis

Pygmaiobacter_massiliensis_test_df <- subset(`melted_log1p_bin_map_counts`, grepl("^Pygmaiobacter.massiliensis$", `melted_log1p_bin_map_counts`$variable))
ggpubr::ggboxplot(Pygmaiobacter_massiliensis_test_df, x = "Host", y = "value", color = "Host", palette = c("#0073ff", "#ff8c00"), ylab = "Log1p transformed mapping count", xlab = "Host species") + stat_compare_means()

lach_sig_plot_1 <- ggpubr::ggboxplot(lach_host_test_df, x = "Host", y = "value", color = "Host", palette = c("#0073ff", "#ff8c00"), ylab = "Log1p transformed mapping count", xlab = "Host species") + stat_compare_means()
lach_sig_plot_2 <- ggpubr::ggboxplot(lach_host_test_df, x = "Season", y = "value", color = "Season", palette = c("#00f3ff", "#4d2a00"), ylab = "Log1p transformed mapping count", xlab = "Season collected") + stat_compare_means()
grid.arrange(lach_sig_plot_1, lach_sig_plot_2, ncol = 2, top = "Lachnospiraceae bacterium 3-1 log1p transformed mapping counts")

host_season_comparisons <- list( c("ACJ", "ACN"), c("ARJ", "ARN"), c("ACJ", "ARJ"), c("ACN", "ARN"))
ggpubr::ggboxplot(lach_host_test_df, x = "Host_by_season", y = "value", color = "Host_by_season", palette = c("#00f3ff", "#0073ff", "#ff8c00", "#4d2a00"), ylab = "Log1p transformed mapping count", xlab = "Host species and collection month") + stat_compare_means(comparisons = host_season_comparisons) + labs(title = "Lachnospiraceae bacterium 3-1 log1p transformed mapping counts by group") + theme(text = element_text(family = "Tex Gyre Heroes"))

pyg_sig_plot_1 <- ggpubr::ggboxplot(Pygmaiobacter_massiliensis_test_df, x = "Host", y = "value", color = "Host", palette = c("#0073ff", "#ff8c00"), ylab = "Log1p transformed mapping count", xlab = "Host species") + stat_compare_means() + theme(text = element_text(family = "Tex Gyre Heroes"))
pyg_sig_plot_2 <- ggpubr::ggboxplot(Pygmaiobacter_massiliensis_test_df, x = "Season", y = "value", color = "Season", palette = c("#00f3ff", "#4d2a00"), ylab = "Log1p transformed mapping count", xlab = "Season collected") + stat_compare_means() + theme(text = element_text(family = "Tex Gyre Heroes"))
grid.arrange(pyg_sig_plot_1, pyg_sig_plot_2, ncol = 2, top = "Pygmaiobacter massiliensis log1p transformed mapping counts") + theme(text = element_text(family = "Tex Gyre Heroes"))
ggpubr::ggboxplot(Pygmaiobacter_massiliensis_test_df, x = "Host_by_season", y = "value", color = "Host_by_season", palette = c("#00f3ff", "#0073ff", "#ff8c00", "#4d2a00"), ylab = "Log1p transformed mapping count", xlab = "Host species and collection month") + stat_compare_means(comparisons = host_season_comparisons) + labs(title = "Pygmaiobacter massiliensis log1p transformed mapping counts by group") + theme(text = element_text(family = "Tex Gyre Heroes"))

tic()
corr_var(cor2_trial, Pygmaiobacter.massiliensis, top = 10)
toc()

Anaerofilum_sp_An_201_test_df <- subset(`melted_log1p_bin_map_counts`, grepl("^Anaerofilum.sp..An201$", `melted_log1p_bin_map_counts`$variable))
ggpubr::ggboxplot(Anaerofilum_sp_An_201_test_df, x = "Host_by_season", y = "value", color = "Host_by_season", palette = c("#00f3ff", "#0073ff", "#ff8c00", "#4d2a00"), ylab = "Log1p transformed mapping count", xlab = "Host species and collection month") + stat_compare_means(comparisons = host_season_comparisons) + labs(title = "Anaerofilum sp. An201 log1p transformed mapping counts by group") + theme(text = element_text(family = "Tex Gyre Heroes"))

fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 2, top = 30, color = "black", fill = "#00f2ff") + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 11)) + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Ten greatest contributing variables to Dimension-2") + xlab("Taxon")

fviz_pca_var(`pca_log1p_bin_map_counts`, select.var = list(contrib = 0.51), repel = TRUE) + theme_classic()

fviz_pca_var(`pca_log1p_bin_map_counts`, geom = c("point", "text"), select.var = list(cos2 = 0.80), repel = TRUE, col.var = "#0073ff") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

fviz_pca_var(`pca_log1p_bin_map_counts`, geom = c("point", "text"), select.var = list(cos2 = -0.50), repel = TRUE, col.var = "#0073ff") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

fviz_pca_var(`pca_log1p_bin_map_counts`, geom = c("point", "text"), col.var = "#ff8c00", repel = TRUE, force = 2, max.overlaps = 50) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

fviz_pca_var(`pca_log1p_bin_map_counts`, geom = c("point", "text"), select.var = list(cos2 = 0.85), repel = TRUE, col.var = "#0073ff", jitter = "label") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

fviz_pca_biplot(`pca_log1p_bin_map_counts`, label = "ind", invisible = "var")

tic()
corr_var(cor2_trial, Pygmaiobacter.massiliensis, bottom = 10)
toc()

ggpubr::ggboxplot(Anaerofilum_sp_An_201_test_df, x = "Host_by_season", y = "value", color = "Host_by_season", palette = c("#00f3ff", "#0073ff", "#ff8c00", "#4d2a00"), ylab = "Log1p transformed mapping count", xlab = "Host species and collection month") + stat_compare_means(aes(label = ..p.signif..)) + labs(title = "Anaerofilum sp. An201 log1p transformed mapping counts by group") + theme(text = element_text(family = "Tex Gyre Heroes"))
ggpubr::ggboxplot(lach_host_test_df, x = "Host_by_season", y = "value", color = "Host_by_season", palette = c("#00f3ff", "#0073ff", "#ff8c00", "#4d2a00"), ylab = "Log1p transformed mapping count", xlab = "Host species and collection month") + stat_compare_means(aes(label = ..p.signif..)) + labs(title = "Lachnospiraceae bacterium 3-1 log1p transformed mapping counts by group") + theme(text = element_text(family = "Tex Gyre Heroes"))

fviz_pca_var(`pca_log1p_bin_map_counts`, geom = c("point", "text"), select.var = list(cos2 = 0.75), repel = TRUE, col.var = "#0073ff", jitter = "label") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(2,3), geom = c("point", "text"), select.var = list(cos2 = 0.65), repel = TRUE, col.var = "#0073ff", jitter = "label") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(1,2), geom = c("point", "text"), select.var = list(cos2 = 0.85), repel = TRUE, col.var = "#003300", labelsize = 3) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Taxa with cos2 of 0.85 or greater for Dimensions 1 or 2") + theme(axis.title = element_text(size = 7.5), axis.text = element_text(size = 9.5))
fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(1,3), geom = c("point", "text"), select.var = list(cos2 = 0.80), repel = TRUE, col.var = "#003333", labelsize = 3) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Taxa with cos2 of 0.8 or greater for Dimensions 1 or 3") + theme(axis.title = element_text(size = 7.5), axis.text = element_text(size = 9.5))
fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(2,3), geom = c("point", "text"), select.var = list(cos2 = 0.75), repel = TRUE, col.var = "#330033", labelsize = 3) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Taxa with cos2 of 0.75 or greater for Dimensions 2 or 3") + theme(axis.title = element_text(size = 7.5), axis.text = element_text(size = 9.5))
fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(3,4), geom = c("point", "text"), select.var = list(cos2 = 0.55), repel = TRUE, col.var = "#330000", labelsize = 3) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Taxa with cos2 of 0.55 or greater for Dimensions 3 or 4") + theme(axis.title = element_text(size = 7.5), axis.text = element_text(size = 9.5))


bacteroidales_bacterium_St1_host_test_df <- subset(`melted_log1p_bin_map_counts`, grepl("^Bacteroidales.bacterium.St1$", `melted_log1p_bin_map_counts`$variable))

ggpubr::ggboxplot(bacteroidales_bacterium_St1_host_test_df, x = "Host", y = "value", color = "Host", palette = c("#0073ff", "#ff8c00"), ylab = "Log1p transformed mapping count", xlab = "Host species") + stat_compare_means(aes(label = ..p.signif..))
wilcox.test(value ~ Host, data = bacteroidales_bacterium_St1_host_test_df, exact = FALSE, correct = FALSE)
ggpubr::ggboxplot(bacteroidales_bacterium_St1_host_test_df, x = "Host_by_season", y = "value", color = "Host_by_season", palette = c("#00f3ff", "#0073ff", "#ff8c00", "#4d2a00"), ylab = "Log1p transformed mapping count", xlab = "Host species and collection month") + stat_compare_means(comparisons = host_season_comparisons, aes(label = ..p.signif..)) + labs(title = "Bacteroidales bacterium St1 log1p transformed mapping counts by group") + theme(text = element_text(family = "Tex Gyre Heroes"))

2021-08-31
tic()
corr_var(cor2_trial, bacterium.0.1xD8.82, top = 10)
toc()

Bifidobacterium.pseudolongum

bif_psu_longum_df <- subset(`melted_log1p_bin_map_counts`, grepl("^Bifidobacterium.pseudolongum$", `melted_log1p_bin_map_counts`$variable))
wilcox.test(value ~ Season, data = bif_psu_longum_df, exact = FALSE, correct = FALSE)
ggpubr::ggboxplot(bif_psu_longum_df, x = "Season", y = "value", color = "Season", palette = c("#0073ff", "#ff8c00"), ylab = "Log1p transformed mapping count", xlab = "Collection month") + stat_compare_means(aes(label = ..p.signif..)) + labs(title = "Bifidobacterium pseudolongum seasonal change across all samples") + theme(text = element_text(family = "Tex Gyre Heroes"))
ggpubr::ggboxplot(bif_psu_longum_df, x = "Season", y = "value", color = "Season", palette = c("#0073ff", "#ff8c00"), ylab = "Log1p transformed mapping count", xlab = "Collection month", conf.int = TRUE) + stat_compare_means(aes(label = ..p.signif..)) + stat_compare_means(label.y = 10, label.x = 1) + labs(title = "Bifidobacterium pseudolongum seasonal change across all samples") + theme(text = element_text(family = "Tex Gyre Heroes"))

ggpubr::ggboxplot(bif_psu_longum_df, x = "Host_by_season", y = "value", color = "Host_by_season", palette = c("#00f3ff", "#0073ff", "#ff8c00", "#4d2a00"), ylab = "Log1p transformed mapping count", xlab = "Host species and collection month") + stat_compare_means(comparisons = host_season_comparisons, aes(label = ..p.signif..)) + labs(title = "Bifidobacterium pseudolongum log1p transformed mapping counts by group") + theme(text = element_text(family = "Tex Gyre Heroes"))
ggpubr::ggboxplot(bif_psu_longum_df, x = "Host_by_season", y = "value", color = "Host_by_season", palette = c("#00f3ff", "#0073ff", "#ff8c00", "#4d2a00"), ylab = "Log1p transformed mapping count", xlab = "Host species and collection month") + stat_compare_means(comparisons = host_season_comparisons, aes(label = ..p.signif..))+ stat_compare_means(label.y = 10, label.x = 1) + labs(title = "Bifidobacterium pseudolongum log1p transformed mapping counts by group") + theme(text = element_text(family = "Tex Gyre Heroes"))

tic()
corr_var(cor2_trial, Bifidobacterium.pseudolongum, top = 20)
toc()

ggplot(subset(`melted_log1p_bin_map_counts`, grepl("^Bifidobacterium.pseudolongum$", `melted_log1p_bin_map_counts`$variable))) + theme_linedraw() + geom_jitter(aes(x = Host_by_season, y = value, color = Host_by_season), width = 0.3, size = 2.75) + xlab("Taxon") + ylab("Log1p transformed, adjusted and summed mapped read segments") + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + labs(color="Host by season") + labs(title = "Log1p transformed, contig and library size adjusted summed counts across sample for Bifidobacterium\npseudolongum by host species and season") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2021.08.30_Averages_and_medians_all_samples_and_by_host`, aes(x = June.Median, y = November.median)) + geom_point(color = "#ff0073") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("June median count") + ylab("November median count") + labs(title = "Sampling month median counts") + geom_text(data = subset(`2021.08.30_Averages_and_medians_all_samples_and_by_host`, SampleID == "Bifidobacterium.pseudolongum"), aes(June.Median, November.median, label = SampleID), hjust = 0.5, vjust = -1, family = "Tex Gyre Heroes")

fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(1,2), geom = c("point"), select.var = list(cos2 = -0.90), repel = TRUE, col.var = "#00b3ff") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))
fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(1,3), geom = c("point"), select.var = list(cos2 = -0.90), repel = TRUE, col.var = "#00ff4c") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))
fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(2,3), geom = c("point"), select.var = list(cos2 = -0.90), repel = TRUE, col.var = "#ff00b3") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))
fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(3,4), geom = c("point"), select.var = list(cos2 = -0.90), repel = TRUE, col.var = "#ff4c00") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(1,2), geom = c("point", "text"), select.var = "Bifidobacterium.pseudolongum"), repel = TRUE, col.var = "#003300", labelsize = 3) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Taxa with cos2 of 0.85 or greater for Dimensions 1 or 2") + theme(axis.title = element_text(size = 7.5), axis.text = element_text(size = 9.5))

fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(1,2), geom = c("point", "text"), select.var = list(cos2 = 0.85), repel = TRUE, col.var = "#003300", labelsize = 3) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Taxa with cos2 of 0.85 or greater for Dimensions 1 or 2") + theme(axis.title = element_text(size = 7.5), axis.text = element_text(size = 9.5))


2021-09-01
select.ind = list(name = c("23", "42", "119")))
fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(1,2), geom = c("point", "text"), select.var = list(name = c("Leptospira", "Root", "Bacteria", "Lachnospiraceae.bacterium.3.1")), repel = TRUE, col.var = "#009999", labelsize = 3) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Positions of indicated taxa on Dimensions 1 and 2") + theme(axis.title = element_text(size = 7.5), axis.text = element_text(size = 9.5))

fviz_contrib(`pca_log1p_bin_map_counts`, choice = "var", axes = 2, top = 25) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + theme(text = element_text(family = "Tex Gyre Heroes"))

tic()
corr_var(cor2_trial, Root, top = 10)
toc()

fviz_pca_var(`pca_log1p_bin_map_counts`, axes = c(1,2), geom = c("point", "text"), select.var = list(name = c("Bacteria", "Firmicutes", "Ruminococcaceae", "Treponema.socranskii.subsp..paredis.ATCC.35535", "Symbiodinium.microadriaticum")), repel = TRUE, col.var = "#009999", labelsize = 3) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Positions of indicated taxa on Dimensions 1 and 2") + theme(axis.title = element_text(size = 7.5), axis.text = element_text(size = 9.5))

AC11_single_paired_sample_trial_df <- melt(`2021.09.01_TRANSPOSED_testing_a_paired_sample_data_AC11`, id = c("SampleID", "Host", "Season", "Host_by_season", "Sequencing"))

melted_log1p_transformed_adjusted_by_isolate_sums <- melt(`log1p_transformed_adjusted_by_isolate_sums`, id=c("Host", "Season", "Host_by_season", "Sequencing", "SampleID"))

ggplot(data = `AC11_single_paired_sample_trial_df`, aes(Season, value, fill = Season)) + geom_boxplot() + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(fill = Season, group = variable), position = position_dodge(0.2), size = 0.15) + geom_line(aes(group = variable), position = position_dodge(0.2))

ggplot(subset(`AC11_single_paired_sample_trial_df`, grepl("^Root$", `AC11_single_paired_sample_trial_df`$variable)), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(fill = Season, group = variable), position = position_dodge(0.2), size = 0.15) + geom_line(aes(group = variable), position = position_dodge(0.2))

ggplot(subset(`AC11_single_paired_sample_trial_df`, grepl("^Bacteria$|^Root$|^Lachnospiraceae", `AC11_single_paired_sample_trial_df`$variable)), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(fill = Season, group = variable), position = position_dodge(0.2), size = 0.15) + geom_line(aes(group = variable), position = position_dodge(0.2))

ggplot(subset(`AC11_single_paired_sample_trial_df`, `AC11_single_paired_sample_trial_df`$value > 5.5), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(fill = Season, group = variable), position = position_dodge(0.2), size = 0.15) + geom_line(aes(group = variable), position = position_dodge(0.2))

wilcox.test(value ~ Season, data = `AC11_single_paired_sample_trial_df`, exact = FALSE)

AC11_lm_model = lm(Host ~ Season, `AC11_single_paired_sample_trial_df`)

2021-09-02
ggplot(subset(`AC11_single_paired_sample_trial_df`, `AC11_single_paired_sample_trial_df`$value > 2.5), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(fill = Season, group = variable), position = position_dodge(0.2), size = 0.15) + geom_line(aes(group = variable), position = position_dodge(0.2))

group_by_bins <- `2021.09.02_TRANSPOSED.all_subsampled_reads_mapped_to_bins_contig_and_library_size_adjusted_scores_summed_by_bin_contig_is_from`
log1p_group_by_bins <- log1p(`group_by_bins`[2:504])
log1p_group_by_bins$SampleID <- `group_by_bins`$SampleID
log1p_group_by_bins$Host <- `group_by_bins`$Host
log1p_group_by_bins$Season <- `group_by_bins`$Season
log1p_group_by_bins$Sequencing <- `group_by_bins`$Sequencing
log1p_group_by_bins$Host_by_season <- `group_by_bins`$Host_by_season

PCA_log1p_group_by_bins <- prcomp(`log1p_group_by_bins`[,c(1:503)], center = TRUE, scale. = TRUE)
melted_log1p_group_by_bins <- reshape2::melt(`log1p_group_by_bins`, id=c("Host", "Season", "Host_by_season", "Sequencing", "SampleID"))

ggbiplot::ggbiplot(`PCA_log1p_group_by_bins`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_group_by_bins`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_group_by_bins`$Season), color=factor(`log1p_group_by_bins`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed Bin level mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%") + theme(text = element_text(family = "Tex Gyre Heroes"))

set.seed(124)
km.res.grouped_by_bins <- kmeans(scale(`log1p_group_by_bins`[,c(1:503)]), 4, nstart = 25)
grouped_by_bins_res_df <- scale(`log1p_group_by_bins`[,c(1:503)])
fviz_cluster(km.res.grouped_by_bins, data = `grouped_by_bins_res_df`, palette = c("#00AFBB","#2E9FDF", "#E7B800", "#FC4E07"), ggtheme = theme_linedraw(), main = "Clustering with kmeans of summed read counts grouped by Bin of origin for contigs")

fviz_contrib(`PCA_log1p_group_by_bins`, choice = "var", axes = 2, top = 25) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + theme(text = element_text(family = "Tex Gyre Heroes"))
fviz_pca_var(`PCA_log1p_group_by_bins`, axes = c(1,2), geom = c("point", "text"), select.var = list(name = c("Bin_c544")), repel = TRUE, col.var = "#009999", labelsize = 3) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Positions of indicated taxa on Dimensions 1 and 2") + theme(axis.title = element_text(size = 7.5), axis.text = element_text(size = 9.5))
fviz_contrib(`PCA_log1p_group_by_bins`, choice = "var", axes = 1, top = 25) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + theme(text = element_text(family = "Tex Gyre Heroes"))
fviz_pca_var(`PCA_log1p_group_by_bins`, axes = c(1,2), geom = c("point", "text"), select.var = list(name = c("Bin_c544", "Bin_c1319")), repel = TRUE, col.var = "#009999", labelsize = 3) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Positions of indicated taxa on Dimensions 1 and 2") + theme(axis.title = element_text(size = 7.5), axis.text = element_text(size = 9.5))

cor2_bin_groups <- cor(as.matrix(`log1p_group_by_bins`[,c(1:503)]))
cor2_bin_groups <- as.data.frame(cor2_bin_groups)
ggcorrplot(cor2_bin_groups, hc.order = TRUE, type = "lower", lab = TRUE)
toc()
tic()
corr_var(cor2_trial, Bin_c544, top = 10)
toc()

2021-09-03
tmpplot1 <- ggbiplot::ggbiplot(`pca_log1p_bin_map_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_bin_map_counts`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_bin_map_counts`$Season), color=factor(`log1p_bin_map_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed taxa mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%") + theme(text = element_text(family = "Tex Gyre Heroes"))
tmpplot2 <- ggbiplot::ggbiplot(`PCA_log1p_group_by_bins`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_group_by_bins`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_group_by_bins`$Season), color=factor(`log1p_group_by_bins`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed Bin level mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%") + theme(text = element_text(family = "Tex Gyre Heroes"))

grid.arrange(tmpplot1, tmpplot2, ncol = 1)

ggbiplot::ggbiplot(`pca_log1p_bin_map_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_bin_map_counts`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_bin_map_counts`$Sequencing), color=factor(`log1p_bin_map_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("Novogene" = 17, "Pilot" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed taxa mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%") + theme(text = element_text(family = "Tex Gyre Heroes"))

2021-09-06
write.csv(`log1p_bin_map_counts`, "/Users/osbourne/Google Drive/PhD/Scripts/R_working_directory/2021-09-06_log1p_bin_map_counts.csv", row.names = FALSE)
updated_log1p_bin_counts <- `2021.09.06_log1p_bin_map_counts`
pca_updated_log1p_bin_counts <- prcomp(`updated_log1p_bin_counts`[,c(1:6357)], center = TRUE, scale. = TRUE)

ggbiplot::ggbiplot(`pca_updated_log1p_bin_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`updated_log1p_bin_counts`$Sequencing, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE, family = "Tex Gyre Heroes") + theme_linedraw() + geom_point(aes(shape=factor(`updated_log1p_bin_counts`$Season), color=factor(`updated_log1p_bin_counts`$Sequencing)), size = 4) + scale_color_manual(values = c("Pilot" = "red", "Novogene" = "Black")) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Sequencing") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed taxa mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%") + theme(text = element_text(family = "Tex Gyre Heroes"))
ggbiplot::ggbiplot(`pca_updated_log1p_bin_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`updated_log1p_bin_counts`$Sequencing, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE, family = "Tex Gyre Heroes") + theme_linedraw() + geom_point(aes(shape=factor(`updated_log1p_bin_counts`$Host), color=factor(`updated_log1p_bin_counts`$Sequencing)), size = 4) + scale_color_manual(values = c("Pilot" = "red", "Novogene" = "Black")) + scale_shape_manual(values = c("AR" = 17, "AC" = 16)) + labs(shape="Season", colour="Sequencing") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed taxa mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%") + theme(text = element_text(family = "Tex Gyre Heroes"))

2021.09.06.AC_paired_non_pilot_log1p_counts
melted_2021.09.06.AC_paired_non_pilot_log1p_counts <- reshape2::melt(`2021.09.06.AC_paired_non_pilot_log1p_counts`, id=c("Pairing", "Host", "Season", "Host_by_season", "Sequencing", "SampleID"))

ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$value > 6.5), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(fill = Season, group = variable), position = position_dodge(0.2), size = 0.15) + geom_line(aes(group = variable), position = position_dodge(0.2))

ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$value > 6.5), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(fill = Season, group = variable), position = position_dodge(0.2), size = 0.15)

ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$value > 6.5), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(fill = Season, group = variable), position = position_dodge(0.2), size = 0.15) + geom_line(aes(group = variable), position = position_dodge(0.2))

ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$value > 6.5), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(y = value, fill = Season, group = variable), position = position_dodge(0.2), size = 0.15) + geom_line(aes(group = Pairing), position = position_dodge(0.2))
scale_x_discrete()
ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$variable == "Nitrosomonadales.bacterium.Swamp67"), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(fill = Season, group = Pairing), position = position_dodge(0.2), size = 0.15) + geom_line(aes(group = Pairing), position = position_dodge(0.2))

ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$variable == "Nitrosomonadales.bacterium.Swamp67"), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(fill = Season, group = Pairing), position = position_dodge(0.2), size = 0.15) + geom_line(aes(group = Pairing), position = position_dodge(0.2)) + geom_boxplot(aes(y = value, fill = Season, group = Season), alpha = 0.8) + scale_fill_manual(values = c("J" = "#E69F00", "N" = "#56B4E9"))


ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$variable == "Nitrosomonadales.bacterium.Swamp67" | `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$variable == "Unclassified"), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(fill = Season, group = variable), position = position_dodge(0.2), size = 0.15) + geom_line(aes(group = variable), position = position_dodge(0.2)) + geom_boxplot(aes(y = value, fill = Season, group = variable), alpha = 0.8) + scale_fill_manual(values = c("J" = "#E69F00", "N" = "#56B4E9"))


ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$variable == "Bacteria" | `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$variable == "Rhodospirillaceae.bacterium.KN72"), aes(Season, value, fill = variable)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(color = variable, group = Pairing), position = position_dodge(0.2), size = 0.15) + geom_boxplot(aes(y = value, fill = variable, group = variable), alpha = 0.8)

ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$variable == "Bacteria"), aes(Season, value, fill = variable)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(color = variable, group = Pairing), position = position_dodge(0.2), size = 1.25) + geom_boxplot(aes(y = value, fill = variable, group = Season), alpha = 0.4) + geom_line(aes(group = Pairing), position = position_dodge(0.2))


ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$value > 9), aes(Season, value, fill = variable)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(color = variable, group = Pairing), position = position_dodge(0.2), size = 1.25) + geom_boxplot(aes(y = value, fill = variable, group = Season), alpha = 0.4) + geom_line(aes(group = Pairing), position = position_dodge(0.2))
ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$value > 9), aes(Season, value, fill = variable)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(color = variable, group = Pairing), position = position_dodge(0.2), size = 1.25) + geom_boxplot(aes(y = value, fill = Season, group = variable), alpha = 0.4) + geom_line(aes(group = Pairing), position = position_dodge(0.2))

ggplot(subset(`melted_2021.09.06.AC_paired_non_pilot_log1p_counts`, `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$variable == "Nitrosomonadales.bacterium.Swamp67" | `melted_2021.09.06.AC_paired_non_pilot_log1p_counts`$variable == "Unclassified"), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(color = Season, group = Pairing), position = position_dodge(0.2), size = 1.25) + geom_boxplot(aes(y = value, fill = Season, group = variable), alpha = 0.3) + geom_line(aes(group = Pairing), position = position_dodge(0.2))


melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts <- reshape2::melt(`2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, id=c("Pairing", "Host", "Season", "Host_by_season", "Sequencing", "SampleID"))

ggplot(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Bacteria"), aes(Season, value, fill = Host)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(y = value, fill = Season, group = Host), alpha = 0.4)

ggplot(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Bacteria"), aes(Season, value, fill = Season)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(y = value, x = Host, fill = Season), alpha = 0.4)

ggplot(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Bacteria"), aes(Season, value, fill = Host)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(y = value, x = Season, fill = Host), alpha = 0.4) + geom_jitter(aes(y = value, x = Season, color = Host), width = 0.3, size = 1.75)
set.seed(1916)
ggplot(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Bacteria"), aes(Season, value, fill = Host)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(y = value, x = Season, fill = Host), alpha = 0.4) + geom_point(aes(y = value, x = Season, color = Host), size=2, shape=21) + geom_line(aes(color = Host, group = Pairing))

ggplot(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Bacteria" | `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Bacteroides.cellulosilyticus"), aes(Season, value, fill = Host)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(y = value, x = Season, fill = Host), alpha = 0.4) + geom_point(aes(y = value, x = Season, color = Host), size=2, shape=21) + geom_line(aes(color = Host, group = Pairing))

ggplot(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Bacteria" | `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Bacteroides.cellulosilyticus"), aes(Season, value, fill = Host)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(y = value, x = Season, fill = Host), alpha = 0.4)

ggplot(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Bacteria" | `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Bacteroides.cellulosilyticus" | `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "bacterium.0.1xD8.82"), aes(Season, value, fill = Host)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(y = value, x = Season, fill = Host, group = Pairing), alpha = 0.4)

ggplot(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$value > 9), aes(Season, value, color = Host)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(y = value, x = Season, color = Host), size = 1.5, shape = 21, position = position_dodge(0.2)) + geom_line(aes(color = Host, group = Pairing), position = position_dodge(0.2))

2021-09-08
#loaded workspace
#loaded all packages from the cautious load list

rm(test_LM_model_all_samples_median_June_to_November)
view(`melted_log1p_group_by_bins`)

ggplot(data = `melted_log1p_group_by_bins`) + theme_linedraw() + geom_jitter(aes(x = variable, y = value, color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts per sample,\nsummed by host and season")

ggplot(data = `melted_log1p_group_by_bins`, aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts per sample,\nsummed by Bin")

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$value > 8.5), aes(x = variable, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.15, size = 2.25) + xlab("Bins") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin with score greater than 8.5 in at least one sample") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_c634"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for Bin_c634 by host and season") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_c362"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.15, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for Bin_c362 by host and season") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))


ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_c362" | `melted_log1p_group_by_bins`$variable == "Bin_m1254" | `melted_log1p_group_by_bins`$variable == "Bin_c1319"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season, shape = variable), width = 0.35, size = 2.65) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for indicated Bins by host and season") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))

ggdensity(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1254"), x = "value", fill = "#800079", color = "black", add = "mean", rug = TRUE) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggdensity(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1254" | `melted_log1p_group_by_bins`$variable == "Bin_c1319"), x = "value", fill = "variable", palette = c("#800079", "#007980"), color = "black", add = "mean", rug = TRUE) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggdensity(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1254" | `melted_log1p_group_by_bins`$variable == "Bin_c1319" | `melted_log1p_group_by_bins`$variable == "Bin_m1590"), x = "value", fill = "variable", palette = c("#800079", "#007980", "#798000"), color = "black", add = "mean", rug = TRUE) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

Bin_m1590

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_c610" | `melted_log1p_group_by_bins`$variable == "Bin_m1197" | `melted_log1p_group_by_bins`$variable == "Bin_c1732" | `melted_log1p_group_by_bins`$variable == "Bin_m1265" | `melted_log1p_group_by_bins`$variable == "Bin_c771" | `melted_log1p_group_by_bins`$variable == "Bin_c1126" | `melted_log1p_group_by_bins`$variable == "Bin_c13" | `melted_log1p_group_by_bins`$variable == "Bin_m372" | `melted_log1p_group_by_bins`$variable == "Bin_c1235" | `melted_log1p_group_by_bins`$variable == "Bin_m1474" | `melted_log1p_group_by_bins`$variable == "Bin_c51" | `melted_log1p_group_by_bins`$variable == "Bin_m1642" | `melted_log1p_group_by_bins`$variable == "Bin_c1004" | `melted_log1p_group_by_bins`$variable == "Bin_m1794" | `melted_log1p_group_by_bins`$variable == "Bin_m691" | `melted_log1p_group_by_bins`$variable == "Bin_m1050" | `melted_log1p_group_by_bins`$variable == "Bin_c393" | `melted_log1p_group_by_bins`$variable == "Bin_m1569" | `melted_log1p_group_by_bins`$variable == "Bin_c770" | `melted_log1p_group_by_bins`$variable == "Bin_c1138"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season, shape = variable), width = 0.35, size = 2.65) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for indicated Bins by host and season") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + scale_shape_manual(values = 0:19)

#First 20 from file of bins with at least 50 percent contigs same species
`melted_log1p_group_by_bins`$variable == "Bin_c610" | `melted_log1p_group_by_bins`$variable == "Bin_m1197" | `melted_log1p_group_by_bins`$variable == "Bin_c1732" | `melted_log1p_group_by_bins`$variable == "Bin_m1265" | `melted_log1p_group_by_bins`$variable == "Bin_c771" | `melted_log1p_group_by_bins`$variable == "Bin_c1126" | `melted_log1p_group_by_bins`$variable == "Bin_c13" | `melted_log1p_group_by_bins`$variable == "Bin_m372" | `melted_log1p_group_by_bins`$variable == "Bin_c1235" | `melted_log1p_group_by_bins`$variable == "Bin_m1474" | `melted_log1p_group_by_bins`$variable == "Bin_c51" | `melted_log1p_group_by_bins`$variable == "Bin_m1642" | `melted_log1p_group_by_bins`$variable == "Bin_c1004" | `melted_log1p_group_by_bins`$variable == "Bin_m1794" | `melted_log1p_group_by_bins`$variable == "Bin_m691" | `melted_log1p_group_by_bins`$variable == "Bin_m1050" | `melted_log1p_group_by_bins`$variable == "Bin_c393" | `melted_log1p_group_by_bins`$variable == "Bin_m1569" | `melted_log1p_group_by_bins`$variable == "Bin_c770" | `melted_log1p_group_by_bins`$variable == "Bin_c1138"

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_c1167" | `melted_log1p_group_by_bins`$variable == "Bin_m1590" | `melted_log1p_group_by_bins`$variable == "Bin_c943" | `melted_log1p_group_by_bins`$variable == "Bin_c1194" | `melted_log1p_group_by_bins`$variable == "Bin_m1595" | `melted_log1p_group_by_bins`$variable == "Bin_m1461" | `melted_log1p_group_by_bins`$variable == "Bin_c828" | `melted_log1p_group_by_bins`$variable == "Bin_c633" | `melted_log1p_group_by_bins`$variable == "Bin_c998" | `melted_log1p_group_by_bins`$variable == "Bin_c416" | `melted_log1p_group_by_bins`$variable == "Bin_c1736" | `melted_log1p_group_by_bins`$variable == "Bin_c513" | `melted_log1p_group_by_bins`$variable == "Bin_m147" | `melted_log1p_group_by_bins`$variable == "Bin_c59" | `melted_log1p_group_by_bins`$variable == "Bin_m21" | `melted_log1p_group_by_bins`$variable == "Bin_m1454" | `melted_log1p_group_by_bins`$variable == "Bin_c1115" | `melted_log1p_group_by_bins`$variable == "Bin_m216" | `melted_log1p_group_by_bins`$variable == "Bin_c298" | `melted_log1p_group_by_bins`$variable == "Bin_m731"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season, shape = variable), width = 0.35, size = 2.65) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for indicated Bins by host and season") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) + scale_shape_manual(values = 0:19)

#First 20 from file of bins with under 50 percent contigs same species
`melted_log1p_group_by_bins`$variable == "Bin_c1167" | `melted_log1p_group_by_bins`$variable == "Bin_m1590" | `melted_log1p_group_by_bins`$variable == "Bin_c943" | `melted_log1p_group_by_bins`$variable == "Bin_c1194" | `melted_log1p_group_by_bins`$variable == "Bin_m1595" | `melted_log1p_group_by_bins`$variable == "Bin_m1461" | `melted_log1p_group_by_bins`$variable == "Bin_c828" | `melted_log1p_group_by_bins`$variable == "Bin_c633" | `melted_log1p_group_by_bins`$variable == "Bin_c998" | `melted_log1p_group_by_bins`$variable == "Bin_c416" | `melted_log1p_group_by_bins`$variable == "Bin_c1736" | `melted_log1p_group_by_bins`$variable == "Bin_c513" | `melted_log1p_group_by_bins`$variable == "Bin_m147" | `melted_log1p_group_by_bins`$variable == "Bin_c59" | `melted_log1p_group_by_bins`$variable == "Bin_m21" | `melted_log1p_group_by_bins`$variable == "Bin_m1454" | `melted_log1p_group_by_bins`$variable == "Bin_c1115" | `melted_log1p_group_by_bins`$variable == "Bin_m216" | `melted_log1p_group_by_bins`$variable == "Bin_c298" | `melted_log1p_group_by_bins`$variable == "Bin_m731"

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$value > 8.5 | `melted_log1p_group_by_bins`$value < 2.5), aes(x = variable, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.15, size = 2.25) + xlab("Bins") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00f3ff', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin with score greater than 8.5 or less than 2.5 in at least one sample") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8))

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_c610" | `melted_log1p_group_by_bins`$variable == "Bin_m1197" | `melted_log1p_group_by_bins`$variable == "Bin_c1732" | `melted_log1p_group_by_bins`$variable == "Bin_m1265" | `melted_log1p_group_by_bins`$variable == "Bin_c771" | `melted_log1p_group_by_bins`$variable == "Bin_c1126" | `melted_log1p_group_by_bins`$variable == "Bin_c13" | `melted_log1p_group_by_bins`$variable == "Bin_m372" | `melted_log1p_group_by_bins`$variable == "Bin_c1235" | `melted_log1p_group_by_bins`$variable == "Bin_m1474" | `melted_log1p_group_by_bins`$variable == "Bin_c51" | `melted_log1p_group_by_bins`$variable == "Bin_m1642" | `melted_log1p_group_by_bins`$variable == "Bin_c1004" | `melted_log1p_group_by_bins`$variable == "Bin_m1794" | `melted_log1p_group_by_bins`$variable == "Bin_m691" | `melted_log1p_group_by_bins`$variable == "Bin_m1050" | `melted_log1p_group_by_bins`$variable == "Bin_c393" | `melted_log1p_group_by_bins`$variable == "Bin_m1569" | `melted_log1p_group_by_bins`$variable == "Bin_c770" | `melted_log1p_group_by_bins`$variable == "Bin_c1138" | `melted_log1p_group_by_bins`$variable == "Bin_m1269" | `melted_log1p_group_by_bins`$variable == "Bin_m1093" | `melted_log1p_group_by_bins`$variable == "Bin_c1242" | `melted_log1p_group_by_bins`$variable == "Bin_c1570" | `melted_log1p_group_by_bins`$variable == "Bin_c228" | `melted_log1p_group_by_bins`$variable == "Bin_c951" | `melted_log1p_group_by_bins`$variable == "Bin_m1301" | `melted_log1p_group_by_bins`$variable == "Bin_c35" | `melted_log1p_group_by_bins`$variable == "Bin_c892" | `melted_log1p_group_by_bins`$variable == "Bin_m1254" | `melted_log1p_group_by_bins`$variable == "Bin_c112" | `melted_log1p_group_by_bins`$variable == "Bin_c362" | `melted_log1p_group_by_bins`$variable == "Bin_m334" | `melted_log1p_group_by_bins`$variable == "Bin_c524" | `melted_log1p_group_by_bins`$variable == "Bin_c1158" | `melted_log1p_group_by_bins`$variable == "Bin_c1007" | `melted_log1p_group_by_bins`$variable == "Bin_c199" | `melted_log1p_group_by_bins`$variable == "Bin_c1268" | `melted_log1p_group_by_bins`$variable == "Bin_m780" | `melted_log1p_group_by_bins`$variable == "Bin_c751" | `melted_log1p_group_by_bins`$variable == "Bin_c498" | `melted_log1p_group_by_bins`$variable == "Bin_c1280" | `melted_log1p_group_by_bins`$variable == "Bin_c82" | `melted_log1p_group_by_bins`$variable == "Bin_m837" | `melted_log1p_group_by_bins`$variable == "Bin_c1476" | `melted_log1p_group_by_bins`$variable == "Bin_c1039" | `melted_log1p_group_by_bins`$variable == "Bin_c1319"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 1.65) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for host and season groupings") + theme(text = element_text(family = "Tex Gyre Heroes"))

`melted_log1p_group_by_bins`$variable == "Bin_c610" | `melted_log1p_group_by_bins`$variable == "Bin_m1197" | `melted_log1p_group_by_bins`$variable == "Bin_c1732" | `melted_log1p_group_by_bins`$variable == "Bin_m1265" | `melted_log1p_group_by_bins`$variable == "Bin_c771" | `melted_log1p_group_by_bins`$variable == "Bin_c1126" | `melted_log1p_group_by_bins`$variable == "Bin_c13" | `melted_log1p_group_by_bins`$variable == "Bin_m372" | `melted_log1p_group_by_bins`$variable == "Bin_c1235" | `melted_log1p_group_by_bins`$variable == "Bin_m1474" | `melted_log1p_group_by_bins`$variable == "Bin_c51" | `melted_log1p_group_by_bins`$variable == "Bin_m1642" | `melted_log1p_group_by_bins`$variable == "Bin_c1004" | `melted_log1p_group_by_bins`$variable == "Bin_m1794" | `melted_log1p_group_by_bins`$variable == "Bin_m691" | `melted_log1p_group_by_bins`$variable == "Bin_m1050" | `melted_log1p_group_by_bins`$variable == "Bin_c393" | `melted_log1p_group_by_bins`$variable == "Bin_m1569" | `melted_log1p_group_by_bins`$variable == "Bin_c770" | `melted_log1p_group_by_bins`$variable == "Bin_c1138" | `melted_log1p_group_by_bins`$variable == "Bin_m1269" | `melted_log1p_group_by_bins`$variable == "Bin_m1093" | `melted_log1p_group_by_bins`$variable == "Bin_c1242" | `melted_log1p_group_by_bins`$variable == "Bin_c1570" | `melted_log1p_group_by_bins`$variable == "Bin_c228" | `melted_log1p_group_by_bins`$variable == "Bin_c951" | `melted_log1p_group_by_bins`$variable == "Bin_m1301" | `melted_log1p_group_by_bins`$variable == "Bin_c35" | `melted_log1p_group_by_bins`$variable == "Bin_c892" | `melted_log1p_group_by_bins`$variable == "Bin_m1254" | `melted_log1p_group_by_bins`$variable == "Bin_c112" | `melted_log1p_group_by_bins`$variable == "Bin_c362" | `melted_log1p_group_by_bins`$variable == "Bin_m334" | `melted_log1p_group_by_bins`$variable == "Bin_c524" | `melted_log1p_group_by_bins`$variable == "Bin_c1158" | `melted_log1p_group_by_bins`$variable == "Bin_c1007" | `melted_log1p_group_by_bins`$variable == "Bin_c199" | `melted_log1p_group_by_bins`$variable == "Bin_c1268" | `melted_log1p_group_by_bins`$variable == "Bin_m780" | `melted_log1p_group_by_bins`$variable == "Bin_c751" | `melted_log1p_group_by_bins`$variable == "Bin_c498" | `melted_log1p_group_by_bins`$variable == "Bin_c1280" | `melted_log1p_group_by_bins`$variable == "Bin_c82" | `melted_log1p_group_by_bins`$variable == "Bin_m837" | `melted_log1p_group_by_bins`$variable == "Bin_c1476" | `melted_log1p_group_by_bins`$variable == "Bin_c1039" | `melted_log1p_group_by_bins`$variable == "Bin_c1319"

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_c610" & `melted_log1p_group_by_bins`$Host_by_season == "ACJ" | `melted_log1p_group_by_bins`$variable == "Bin_c610" & `melted_log1p_group_by_bins`$Host_by_season == "ACN"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 1.65) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for host and season groupings") + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_c610" & `melted_log1p_group_by_bins`$Host_by_season == "ACJ" | `melted_log1p_group_by_bins`$variable == "Bin_c610" & `melted_log1p_group_by_bins`$Host_by_season == "ACN"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for host and season groupings") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Host_by_season), alpha = 0.25) + stat_compare_means(method = "wilcox.test")

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1474" & `melted_log1p_group_by_bins`$Host_by_season == "ACJ" | `melted_log1p_group_by_bins`$variable == "Bin_m1474" & `melted_log1p_group_by_bins`$Host_by_season == "ACN"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for host and season groupings") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Host_by_season), alpha = 0.25) + stat_compare_means(method = "wilcox.test")

host_and_season_within_host_comparisons <- list( c("ACJ", "ACN"), c("ARJ", "ARN"))

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1474" & `melted_log1p_group_by_bins`$Host_by_season == "ARJ" | `melted_log1p_group_by_bins`$variable == "Bin_m1474" & `melted_log1p_group_by_bins`$Host_by_season == "ARN"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#ff8c00', '#b16100')) + scale_fill_manual(values=c('#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for host and season groupings") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Host_by_season), alpha = 0.25) + stat_compare_means(method = "wilcox.test")

ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1474"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for host and season groupings") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Host_by_season), alpha = 0.25) + stat_compare_means(comparisons = host_and_season_within_host_comparisons)

ggbetweenstats(data = subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1474"), x = Host_by_season, y = value, title = "Distribution of summed Bin_m1474 log1p transformed, adjusted, mapped read counts across all samples") + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100'))


ggbetweenstats(data = subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1474" & `melted_log1p_group_by_bins`$Host == "AC"), x = Host_by_season, y = value, title = "Distribution of summed Bin_m1474 log1p transformed, adjusted, mapped read counts across all Acomys cahirinus samples") + theme(text = element_text(family = "Tex Gyre Heroes")) + + scale_color_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100'))
ggwithinstats(data = subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1474" & `melted_log1p_group_by_bins`$Host == "AC"), x = Host_by_season, y = value, title = "Distribution of summed Bin_m1474 log1p transformed, adjusted, mapped read counts across all Acomys cahirinus samples") + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100'))


ggbetweenstats(data = subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1474" & `melted_log1p_group_by_bins`$Host == "AC"), x = Host_by_season, y = value, title = "Distribution of summed Bin_m1474 log1p transformed, adjusted, mapped read counts across all Acomys cahirinus samples", type = "np", conf.level = 0.99) + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#ff8c00', '#b16100'))

temp_var_1 <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1474" & `melted_log1p_group_by_bins`$Host_by_season == "ACJ")$value
temp_var_2 <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1474" & `melted_log1p_group_by_bins`$Host_by_season == "ACN")$value

wilcox.test(temp_var_1, temp_var_2, alternative = "two.sided")$p.value

wilcox.test(temp_var_1, temp_var_2, alternative = "two.sided", exact =FALSE)$p.value

wilcox.test(temp_var_1, temp_var_2, alternative = "two.sided", exact =FALSE)$p.value

`2021_09_08_unique_bins_list` <- list(unique(`melted_log1p_group_by_bins`$variable))

small_list_test <- list(c("Bin_c610", "Bin_m1569", "Bin_c1039"))

for (p in `small_list_test`) {print(p)}

for (i in 1:length(`small_list_test`)) {print(`small_list_test`[[i]])}

for (bin in `small_list_test`){ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == small_list_test[[bin]] & `melted_log1p_group_by_bins`$Host == "AR"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for host and season groupings") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Host_by_season), alpha = 0.25) + stat_compare_means(comparisons = host_and_season_within_host_comparisons)}

temp_var_1 <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ACJ")$value ; temp_var_2 <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ACN")$value ; PLACEHOLDER_Wilcox_test_AC_June_versus_November_pvalue <- wilcox.test(temp_var_1, temp_var_2, alternative = "two.sided")$p.value ;

ggbetweenstats(data = subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host == "AC"), x = Host_by_season, y = value, title = "Distribution of summed PLACEHOLDER log1p transformed, adjusted, mapped read counts across all Acomys cahirinus samples", type = "np", conf.level = 0.99) + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#ff8c00', '#b16100'))

ggbetweenstats(data = subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_m1569" & `melted_log1p_group_by_bins`$Host == "AC"), x = Host_by_season, y = value, title = "Distribution of summed PLACEHOLDER log1p transformed, adjusted, mapped read counts across all Acomys cahirinus samples", type = "np", conf.level = 0.99) + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#ff8c00', '#b16100'))


PLACEHOLDER_ACJ_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ACJ")$value
PLACEHOLDER_ACN_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ACN")$value
PLACEHOLDER_ARJ_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ARJ")$value
PLACEHOLDER_ARN_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ARN")$value

PLACEHOLDER_AC_J_vs_N_pvalue <- wilcox.test(PLACEHOLDER_ACJ, PLACEHOLDER_ACN, alternative = "two.sided", exact =FALSE)$p.value
PLACEHOLDER_AR_J_vs_N_pvalue <- wilcox.test(PLACEHOLDER_ARJ, PLACEHOLDER_ARN, alternative = "two.sided", exact =FALSE)$p.value
PLACEHOLDER_ACJ_vs_ARJ_pvalue <- wilcox.test(PLACEHOLDER_ACJ, PLACEHOLDER_ARJ, alternative = "two.sided", exact =FALSE)$p.value
PLACEHOLDER_ACN_vs_ARN_pvalue <- wilcox.test(PLACEHOLDER_ACN, PLACEHOLDER_ARN, alternative = "two.sided", exact =FALSE)$p.value

`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons` <- data.frame(Bin=character(), AC_seasonal=integer(), AR_seasonal=integer(), June_across_hosts=integer(), November_across_hosts=integer(), stringsasFactors = FALSE)
test_colClasses <- c("character", "integer", "integer", "integer", "integer")
test_df <- read.csv(text="Bin,AC_seasonal,AR_seasonal, June_across_hosts, November_across_hosts", colClasses = test_colClasses)
View(test_df)
test_df[nrow(test_df)+1,] <- c(trial_name, trial_value_1, trial_value_2, trial_value_3, trial_value_4)
View(test_df)
str(test_df)
test_df[, c(2:5)] <- sapply(test_df[, c(2:5)], as.numeric)
View(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`)

`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons` <- read.csv(text="Bin,AC_seasonal,AR_seasonal, June_across_hosts, November_across_hosts", colClasses = test_colClasses)
str(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`)
PLACEHOLDER_ACJ_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ACJ")$value ; PLACEHOLDER_ACN_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ACN")$value ; PLACEHOLDER_ARJ_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ARJ")$value ; PLACEHOLDER_ARN_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ARN")$value ; PLACEHOLDER_bin_name <- "PLACEHOLDER" ; PLACEHOLDER_AC_J_vs_N_pvalue <- wilcox.test(PLACEHOLDER_ACJ_counts, PLACEHOLDER_ACN_counts, alternative = "two.sided", exact =FALSE)$p.value ; PLACEHOLDER_AR_J_vs_N_pvalue <- wilcox.test(PLACEHOLDER_ARJ_counts, PLACEHOLDER_ARN_counts, alternative = "two.sided", exact =FALSE)$p.value ; PLACEHOLDER_ACJ_vs_ARJ_pvalue <- wilcox.test(PLACEHOLDER_ACJ_counts, PLACEHOLDER_ARJ_counts, alternative = "two.sided", exact =FALSE)$p.value ; PLACEHOLDER_ACN_vs_ARN_pvalue <- wilcox.test(PLACEHOLDER_ACN_counts, PLACEHOLDER_ARN_counts, alternative = "two.sided", exact =FALSE)$p.value ; `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[nrow(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_J_vs_N_pvalue, PLACEHOLDER_AR_J_vs_N_pvalue, PLACEHOLDER_ACJ_vs_ARJ_pvalue, PLACEHOLDER_ACN_vs_ARN_pvalue) ; `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[, c(2:5)] <- sapply(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[, c(2:5)], as.numeric) ; rm(PLACEHOLDER_ACJ_counts) ; rm(PLACEHOLDER_ACN_counts) ; rm(PLACEHOLDER_ARJ_counts) ; rm(PLACEHOLDER_ARN_counts) ; rm(PLACEHOLDER_AC_J_vs_N_pvalue) ; rm(PLACEHOLDER_AR_J_vs_N_pvalue) ; rm(PLACEHOLDER_ACJ_vs_ARJ_pvalue) ; rm(PLACEHOLDER_ACN_vs_ARN_pvalue)

host_and_season_within_and_across_comparisons <- list( c("ACJ", "ACN"), c("ARJ", "ARN"), c("ACJ", "ARJ"), c("ACN", "ARN"))


ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "Bin_c0"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for host and season groupings") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Host_by_season), alpha = 0.25) + stat_compare_means(comparisons = host_and_season_within_and_across_comparisons)

Bin_m1301

PLACEHOLDER_ggplot_boxplot_and_wilcox_test <- ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for PLACEHOLDER according to host and season groupings", caption = "Wilcox test by stat_compare_means") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Host_by_season), alpha = 0.25) + stat_compare_means(method = "wilcox.test", comparisons = host_and_season_within_and_across_comparisons)

#This is the bit which you use with Bash and sed to make the long list of individual commands to run
PLACEHOLDER_ACJ_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ACJ")$value ; PLACEHOLDER_ACN_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ACN")$value ; PLACEHOLDER_ARJ_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ARJ")$value ; PLACEHOLDER_ARN_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ARN")$value ; PLACEHOLDER_bin_name <- "PLACEHOLDER" ; PLACEHOLDER_AC_J_vs_N_pvalue <- wilcox.test(PLACEHOLDER_ACJ_counts, PLACEHOLDER_ACN_counts, alternative = "two.sided", exact =FALSE)$p.value ; PLACEHOLDER_AR_J_vs_N_pvalue <- wilcox.test(PLACEHOLDER_ARJ_counts, PLACEHOLDER_ARN_counts, alternative = "two.sided", exact =FALSE)$p.value ; PLACEHOLDER_ACJ_vs_ARJ_pvalue <- wilcox.test(PLACEHOLDER_ACJ_counts, PLACEHOLDER_ARJ_counts, alternative = "two.sided", exact =FALSE)$p.value ; PLACEHOLDER_ACN_vs_ARN_pvalue <- wilcox.test(PLACEHOLDER_ACN_counts, PLACEHOLDER_ARN_counts, alternative = "two.sided", exact =FALSE)$p.value ; `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[nrow(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_J_vs_N_pvalue, PLACEHOLDER_AR_J_vs_N_pvalue, PLACEHOLDER_ACJ_vs_ARJ_pvalue, PLACEHOLDER_ACN_vs_ARN_pvalue) ; `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[, c(2:5)] <- sapply(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[, c(2:5)], as.numeric) ; rm(PLACEHOLDER_ACJ_counts) ; rm(PLACEHOLDER_ACN_counts) ; rm(PLACEHOLDER_ARJ_counts) ; rm(PLACEHOLDER_ARN_counts) ; rm(PLACEHOLDER_AC_J_vs_N_pvalue) ; rm(PLACEHOLDER_AR_J_vs_N_pvalue) ; rm(PLACEHOLDER_ACJ_vs_ARJ_pvalue) ; rm(PLACEHOLDER_ACN_vs_ARN_pvalue) ; PLACEHOLDER_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host <- ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for PLACEHOLDER according to host and season groupings", caption = "Wilcox test by stat_compare_means") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Host_by_season), alpha = 0.25) + stat_compare_means(method = "wilcox.test", comparisons = host_and_season_within_and_across_comparisons) ; ggsave("PLACEHOLDER_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host.pdf") ; ggsave("PLACEHOLDER_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host.eps") ; rm(`PLACEHOLDER_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host`) ; rm(PLACEHOLDER_bin_name)

rm(ls(pattern='_ACJ_counts$'))

ggplot(data = `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`) + geom_point(aes(x = AC_seasonal, y = AR_seasonal), size = 1.75, color = "#8c00ff") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "P values for Wilcox test for significant difference between mean of adjusted and log1p transformed mapping counts\nsummed by Bin for seasonal change within each host species") + xlab("P value for June to November change in Acomys cahirinus") + ylab("P value for June to November change in Acomys russatus")

ggplot(data = `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, aes(x = "P value", y = "P value")) + geom_point(aes(x = June_across_hosts, y = November_across_hosts), size = 1.75, color = "#5ccc00") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal < 0.05)) + geom_point(aes(x = AC_seasonal, y = AR_seasonal), size = 1.75, color = "#8c00ff") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "P values for Wilcox test for significant difference between mean of adjusted and log1p transformed mapping counts\nsummed by Bin for seasonal change within each host species", caption = "Showing Bins with P value < 0.05 in both species") + xlab("P value for June to November change in Acomys cahirinus") + ylab("P value for June to November change in Acomys russatus")

ggplot(subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts > 0.05)) + geom_point(aes(x = AC_seasonal, y = AR_seasonal), size = 1.75, color = "#8c00ff") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "P values for Wilcox test for significant difference between mean of adjusted and log1p transformed mapping counts\nsummed by Bin for seasonal change within each host species", caption = "Showing Bins with P value < 0.05 within each species for seasonal change\nbut > 0.05 for across species same season difference") + xlab("P value for June to November change in Acomys cahirinus") + ylab("P value for June to November change in Acomys russatus") + geom_label_repel(aes(x = AC_seasonal, y = AR_seasonal, label = Bin), family = "Tex Gyre Heroes", box.padding = 0.35, point.padding = 0.5)
ggplot(subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts < 0.05)) + geom_point(aes(x = June_across_hosts, y = November_across_hosts), size = 1.75, color = "#5ccc00") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "P values for Wilcox test for significant difference between mean of adjusted and log1p transformed mapping counts\nsummed by Bin for each season between the two host species", caption = "Showing Bins with P value < 0.05 for each month across species\nbut > 0.05 for change across season within same species") + xlab("P value for difference between June samples of each host species") + ylab("P value for difference between November samples of each host species") + geom_label_repel(aes(x = June_across_hosts, y = November_across_hosts, label = Bin), family = "Tex Gyre Heroes", box.padding = 4.35, point.padding = 0.5)

Seasonal_difference_within_each_species_but_not_same_season_across_species <- subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts > 0.05)

Difference_across_two_species_same_season_but_not_within_species_as_season_changes <- subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts < 0.05)

season_change_only_plot <- ggplot(subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts > 0.05)) + geom_point(aes(x = AC_seasonal, y = AR_seasonal), size = 1.75, color = "#8c00ff") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("P value for June to November change in Acomys cahirinus") + ylab("P value for June to November change in Acomys russatus")
host_change_only_plot <- ggplot(subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts < 0.05)) + geom_point(aes(x = June_across_hosts, y = November_across_hosts), size = 1.75, color = "#5ccc00") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("P value for difference between June samples of each host species") + ylab("P value for difference between November samples of each host species")

grid.arrange(season_change_only_plot, host_change_only_plot, ncol = 2, top = "Wilcox test indicated significant bins", bottom = "Purple are bins with significant change within host across season\ngreen are bins with significant difference between host species in same month")

2021-09-09

p.adjust(2.644304e-03, method = "bonferroni", n = length(4))
p.adjust(2.644304e-03, method = "hochberg", n = length(4))
p.adjust(2.644304e-03, method = "BY", n = length(4))

0.0001242229
p.adjust(0.06307245, method = "bonferroni", n = length(4))
p.adjust(3.632750e-06, method = "hochberg", n = length(4))
p.adjust(3.632750e-06, method = "BY", n = length(4))

p.adjust(0.113, method = "bonferroni")

p.adjust(0.06, method = "bonferroni", n = length(4))

#2 is the number of comparisons, since you are comparing within species change and across species same month change
p.adjust(3.632750e-06, method = "BH", n = length(2))

2021-09-08
bin_summing_AC_BH_adjusted_Wilcox_pvalues_within_season_change <- p.adjust(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal, "BH", n = length(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal))
`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal_BH_corrected = `bin_summing_AC_BH_adjusted_Wilcox_pvalues_within_season_change`

bin_summing_AR_BH_adjusted_Wilcox_pvalues_within_season_change <- p.adjust(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal, "BH", n = length(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal))
`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal_BH_corrected = `bin_summing_AR_BH_adjusted_Wilcox_pvalues_within_season_change`

bin_summing_June_across_hosts_Wilcox_pvalues_BH_adjusted <- p.adjust(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts, "BH", n = length(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts))
`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts_BH_corrected = `bin_summing_June_across_hosts_Wilcox_pvalues_BH_adjusted`

bin_summing_November_across_hosts_Wilcox_pvalues_BH_adjusted <- p.adjust(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts, "BH", n = length(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts))
`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts_BH_corrected = `bin_summing_November_across_hosts_Wilcox_pvalues_BH_adjusted`

BH_corrected_season_change_only_plot <- ggplot(subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal_BH_corrected < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal_BH_corrected < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts_BH_corrected > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts_BH_corrected > 0.05)) + geom_point(aes(x = AC_seasonal_BH_corrected, y = AR_seasonal_BH_corrected), size = 1.75, color = "#8c00ff") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for June to November change in Acomys cahirinus") + ylab("BH corrected P value for June to November change in Acomys russatus")

BH_corrected_host_change_only_plot <- ggplot(subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal_BH_corrected > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal_BH_corrected > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts_BH_corrected < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts_BH_corrected < 0.05)) + geom_point(aes(x = June_across_hosts_BH_corrected, y = November_across_hosts_BH_corrected), size = 1.75, color = "#5ccc00") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for difference between June samples of each species") + ylab("BH corrected P value for November samples of each species")

grid.arrange(BH_corrected_season_change_only_plot, BH_corrected_host_change_only_plot, ncol = 2, top = "Wilcox test indicated significant bins after BH correction of P values", bottom = "Purple are bins with significant change within host across season\ngreen are bins with significant difference between host species in same month")

`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts` <- reshape2::melt(`2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, id=c("Pairing", "Host", "Season", "Host_by_season", "Sequencing", "SampleID"))


ggbetweenstats(data = subset(`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$variable == "Root"), x = Host_by_season, y = value, title = "Distribution of summed Root log1p transformed, adjusted, mapped read counts across all samples") + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100'))

ggbetweenstats(data = subset(`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$variable == "Root" & `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Host == "AC"), x = Host_by_season, y = value, title = "Distribution of summed Root log1p transformed, adjusted, mapped read counts across all Acomys cahirinus samples") + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#00c2cc', '#0074ff'))
ggbetweenstats(data = subset(`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$variable == "Root" & `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Host == "AR"), x = Host_by_season, y = value, title = "Distribution of summed Root log1p transformed, adjusted, mapped read counts across all Acomys russatus samples") + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#ff8c00', '#b16100'))

2021-09-10 (after total crash and lost of everything written on this day which hadn't been saved)

`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`

`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons` <- read.csv(text="Taxon, AC_seasonal, AR_seasonal, June_across_hosts, November_across_hosts", colClasses = test_colClasses)
view(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`)

melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts$variable

AC_seasonal_P_values_BH_adjusted <- p.adjust(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal, method = "BH", n = length(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal))
AR_seasonal_P_values_BH_adjusted <- p.adjust(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal, method = "BH", n = length(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal))
June_across_hosts_P_values_BH_adjusted <- p.adjust(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts, method = "BH", n = length(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts))
November_across_hosts_P_values_BH_adjusted <- p.adjust(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts, method = "BH", n = length(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts))
`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal <- AC_seasonal_P_values_BH_adjusted
`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal <- AR_seasonal_P_values_BH_adjusted
`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts <- June_across_hosts_P_values_BH_adjusted
`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts <- November_across_hosts_P_values_BH_adjusted

write.csv(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, "2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons_including_BH_adjusted_values.csv", row.names = FALSE)

`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`

ac_seasonal_values_bin_sums_P_values_BH_adjusted <- p.adjust(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal, method = "BH", n = length(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal))
ar_seasonal_values_bin_sums_P_values_BH_adjusted <- p.adjust(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal, method = "BH", n = length(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AR_seasonal))
summed_by_bins_June_across_samples_BH_corrected_P_values <- p.adjust(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts, method = "BH", n = length(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$June_across_hosts))
summed_by_bins_November_across_samples_BH_corrected_P_values <- p.adjust(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts, method = "BH", n = length(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$November_across_hosts))
`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal <- ac_seasonal_values_bin_sums_P_values_BH_adjusted
`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal <- ar_seasonal_values_bin_sums_P_values_BH_adjusted
`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts <- summed_by_bins_June_across_samples_BH_corrected_P_values
`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts <- summed_by_bins_November_across_samples_BH_corrected_P_values

write.csv(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, "2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons", row.names = FALSE)


BH_corrected_host_change_only_plot <- ggplot(subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts < 0.05)) + geom_point(aes(x = BH_adjusted_June_across_hosts, y = BH_adjusted_November_across_hosts), size = 1.75, color = "#5ccc00") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for difference between June samples of each species") + ylab("BH corrected P value for November samples of each species")

BH_corrected_season_change_only_plot <- ggplot(subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts > 0.05)) + geom_point(aes(x = BH_adjusted_AC_seasonal, y = BH_adjusted_AR_seasonal), size = 1.75, color = "#8c00ff") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for June to November change in Acomys cahirinus") + ylab("BH corrected P value for June to November change in Acomys russatus")

taxa_grouped_BH_corrected_host_change_only <- ggplot(subset(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal > 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal > 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts < 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts < 0.05)) + geom_point(aes(x = BH_adjusted_June_across_hosts, y = BH_adjusted_November_across_hosts), size = 1.75, color = "#5ccc00") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for difference between June samples of each species") + ylab("BH corrected P value for November samples of each species")

taxa_grouped_BH_corrected_seasonal_change_only <- ggplot(subset(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal < 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal < 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts > 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts > 0.05)) + geom_point(aes(x = BH_adjusted_June_across_hosts, y = BH_adjusted_November_across_hosts), size = 1.75, color = "#8c00ff") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for difference between June samples of each species") + ylab("BH corrected P value for November samples of each species")


2021-09-11
`2021_09_11_only_seasonal_change_significantly_different_after_BH_correction_data_frame` <- subset(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal < 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal < 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts > 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts > 0.05)

`2021_09_11_only_host_change_significantly_different_after_BH_correction_data_frame` <- subset(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal > 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal > 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts < 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts < 0.05)

`2021_09_11_summed_by_Bins_only_seasonal_change_significantly_different_after_BH_correction_data_frame` <- subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts > 0.05)

`2021_09_11_summed_by_Bins_only_host_change_significantly_different_after_BH_correction_data_frame` <- subset(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal > 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts < 0.05 & `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts < 0.05)

write.csv(`2021_09_11_only_seasonal_change_significantly_different_after_BH_correction_data_frame`, "2021_09_11_summed_by_taxa_only_seasonal_change_significantly_different_after_BH_correction.csv", row.names = FALSE)
write.csv(`2021_09_11_only_host_change_significantly_different_after_BH_correction_data_frame`, "2021_09_11_summed_by_taxa_only_host_change_significantly_different_after_BH_correction.csv", row.names = FALSE)

write.csv(`2021_09_11_summed_by_Bins_only_seasonal_change_significantly_different_after_BH_correction_data_frame`, "2021_09_11_summed_by_Bins_only_seasonal_change_significantly_different_after_BH_correction_data_frame.csv", row.names = FALSE)
write.csv(`2021_09_11_summed_by_Bins_only_host_change_significantly_different_after_BH_correction_data_frame`, "2021_09_11_summed_by_Bins_only_host_change_significantly_different_after_BH_correction_data_frame.csv", row.names = FALSE)

ggbetweenstats(data = subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Citrobacter" & `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$Host == "AC"), x = Host_by_season, y = value, title = "Distribution of summed Citrobacter log1p transformed, adjusted, mapped read counts across paired Acomys cahirinus samples") + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#00c2cc', '#0074ff'))
ggbetweenstats(data = subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Citrobacter" & `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$Host == "AR"), x = Host_by_season, y = value, title = "Distribution of summed Citrobacter log1p transformed, adjusted, mapped read counts across paired Acomys russatus samples") + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#ff8c00', '#b16100'))

Actinobacteria
ggbetweenstats(data = subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Actinobacteria" & `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$Host == "AC"), x = Host_by_season, y = value, title = "Distribution of summed Actinobacteria log1p transformed, adjusted, mapped read counts across paired Acomys cahirinus samples") + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#00c2cc', '#0074ff'))
ggbetweenstats(data = subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Actinobacteria" & `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$Host == "AR"), x = Host_by_season, y = value, title = "Distribution of summed Actinobacteria log1p transformed, adjusted, mapped read counts across paired Acomys russatus samples") + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values=c('#ff8c00', '#b16100'))

ggplot(data = `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`) + geom_point(aes(x = BH_adjusted_June_across_hosts, y = BH_adjusted_November_across_hosts), size = 1.75, color = "#8c00ff") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for difference between June samples of each species") + ylab("BH corrected P value for November samples of each species")
ggplot(data = `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`) + geom_point(aes(x = BH_adjusted_AC_seasonal, y = BH_adjusted_AR_seasonal), size = 1.75, color = "#5ccc00") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for difference between June and November Acomys cahirinus samples") + ylab("BH corrected P value for June and November samples of Acomys russatus")


ggplot(data = `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`) + geom_point(aes(x = BH_adjusted_June_across_hosts, y = BH_adjusted_November_across_hosts), size = 1.75, color = "#8c00ff") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for difference between June samples of each species") + ylab("BH corrected P value for November samples of each species")

#Note that I believe, from following back the chain of what files made what that the pvalues and wilcox tests have been done on the paired, non-pilot samples
view(`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`)
`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts` <- reshape2::melt(`2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, id=c("Pairing", "Host", "Season", "Host_by_season", "Sequencing", "SampleID"))

Unclassified_ACJ_counts <- subset(`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$variable == "Unclassified" & `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Host_by_season == "ACJ")$value ; Unclassified_ACN_counts <- subset(`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$variable == "Unclassified" & `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Host_by_season == "ACN")$value  ; Unclassified_ARJ_counts <- subset(`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$variable == "Unclassified" & `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Host_by_season == "ARJ")$value ; Unclassified_ARN_counts <- subset(`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$variable == "Unclassified" & `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$Host_by_season == "ARN")$value ; Unclassified_taxa_name <- "Unclassified" ; Unclassified_AC_J_vs_N_pvalue <- wilcox.test(Unclassified_ACJ_counts, Unclassified_ACN_counts, alternative = "two.sided", exact =FALSE)$p.value ; Unclassified_AR_J_vs_N_pvalue <- wilcox.test(Unclassified_ARJ_counts, Unclassified_ARN_counts, alternative = "two.sided", exact =FALSE)$p.value ; Unclassified_ACJ_vs_ARJ_pvalue <- wilcox.test(Unclassified_ACJ_counts, Unclassified_ARJ_counts, alternative = "two.sided", exact =FALSE)$p.value ; Unclassified_ACN_vs_ARN_pvalue <- wilcox.test(Unclassified_ACN_counts, Unclassified_ARN_counts, alternative = "two.sided", exact =FALSE)$p.value ; `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[nrow(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`)+1,] <- c(Unclassified_taxa_name, Unclassified_AC_J_vs_N_pvalue, Unclassified_AR_J_vs_N_pvalue, Unclassified_ACJ_vs_ARJ_pvalue, Unclassified_ACN_vs_ARN_pvalue) ; `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[, c(2:5)] <- sapply(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[, c(2:5)], as.numeric) ; rm(Unclassified_ACJ_counts) ; rm(Unclassified_ACN_counts) ; rm(Unclassified_ARJ_counts) ; rm(Unclassified_ARN_counts) ; rm(Unclassified_AC_J_vs_N_pvalue) ; rm(Unclassified_AR_J_vs_N_pvalue) ; rm(Unclassified_ACJ_vs_ARJ_pvalue) ; rm(Unclassified_ACN_vs_ARN_pvalue) ; Unclassified_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host <- ggplot(subset(`MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`, `MELTED_log1p_2021.08.11.Transposed_Combined_contig_and_library_size_adjusted_Taxa_names_summed_mapped_read_segment_counts`$variable == "Unclassified"), aes(x = Host_by_season, y = value)) + theme_classic() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by taxon\nfor Unclassified according to host and season groupings", caption = "Wilcox test by stat_compare_means") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Host_by_season), alpha = 0.25) + stat_compare_means(method = "wilcox.test", comparisons = host_and_season_within_and_across_comparisons) ; ggsave("Unclassified_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host.pdf") ; ggsave("Unclassified_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host.eps") ; rm(`Unclassified_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host`) ; rm(Unclassified_taxa_name)

taxa_grouped_BH_corrected_seasonal_change_only <- ggplot(subset(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal < 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal < 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts > 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts > 0.05)) + geom_point(aes(x = BH_adjusted_June_across_hosts, y = BH_adjusted_November_across_hosts), size = 1.75, color = "#8c00ff") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for difference between June samples of each species") + ylab("BH corrected P value for November samples of each species")


subset(`2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`, `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AC_seasonal < 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_AR_seasonal > 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_June_across_hosts > 0.05 & `2021_09_08_summed_to_taxa_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$BH_adjusted_November_across_hosts > 0.05)

2021-09-13

#Example of code used for BH correction
p.adjust(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal, "BH", n = length(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal))
`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`$AC_seasonal_BH_corrected = `bin_summing_AC_BH_adjusted_Wilcox_pvalues_within_season_change`

test_colClasses2 <- c("character", "integer", "integer")
`2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons` <- read.csv(text="Taxon, AC_vs_AR, June_vs_November", colClasses = test_colClasses2)

AC_vs_AR_summed_to_taxa_wilcox_test_P_values_BH_adjusted <- p.adjust(`2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$AC_vs_AR, method = "BH", n = length(`2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$AC_vs_AR))
`2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$BH_adjusted_AC_vs_AR <- AC_vs_AR_summed_to_taxa_wilcox_test_P_values_BH_adjusted
J_vs_N_summed_to_taxa_wilcox_test_P_values_BH_adjusted <- p.adjust(`2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$June_vs_November, method = "BH", n = length(`2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$June_vs_November))
`2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$BH_adjusted_J_vs_N <- `J_vs_N_summed_to_taxa_wilcox_test_P_values_BH_adjusted`

ggplot(subset(`2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`, `2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$BH_adjusted_AC_vs_AR < 0.05 & `2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$BH_adjusted_J_vs_N > 0.05)) + geom_point(aes(x = BH_adjusted_AC_vs_AR, y = BH_adjusted_J_vs_N), size = 1.75, color = "#5ccc00") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for difference between each species") + ylab("BH corrected P value for difference between each month")
ggplot(subset(`2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`, `2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$BH_adjusted_AC_vs_AR > 0.05 & `2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$BH_adjusted_J_vs_N < 0.05)) + geom_point(aes(x = BH_adjusted_AC_vs_AR, y = BH_adjusted_J_vs_N), size = 1.75, color = "#8c00ff") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("BH corrected P value for difference between each species") + ylab("BH corrected P value for difference between each month")
`2021-09-13_Summed_to_taxa_significant_after_BH_correction_Wilcox_tested_P_values_taxa_different_between_hosts_but_not_between_seasons` <- subset(`2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`, `2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$BH_adjusted_AC_vs_AR < 0.05 & `2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$BH_adjusted_J_vs_N > 0.05)
`2021-09-13_Summed_to_taxa_significant_after_BH_correction_Wilcox_tested_P_values_taxa_different_between_seasons_but_not_between_hosts` <- subset(`2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`, `2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$BH_adjusted_AC_vs_AR > 0.05 & `2021_09_13_summed_to_taxa_wilcox_test_P_values_across_host_and_across_season_comparisons`$BH_adjusted_J_vs_N < 0.05)
#note, need to make a melted dataframe for the mapping to bins containing only the paired reads - and you need to make the original dataframe for that first
`2021_09_13_summed_to_Bins_wilcox_test_P_values_across_host_and_across_season_comparisons` <- read.csv(text="Bin, AC_vs_AR, June_vs_November", colClasses = test_colClasses2)

subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Unclassified")

`2021_09_13_summed_to_taxa_foldchanges_and_logratios_only_paired_samples_log1p_transformed_contig_and_library_size_adjusted_no_pilot_samples` <- read.csv(text="Taxon, AC_versus_AR_fold_change, AC_versus_AR_logratio, June_versus_November_fold_change, June_versus_November_logratio", colClasses = test_colClasses3)
test_colClasses3 <- c("character", "integer", "integer", "integer", "integer")

PLACEHOLDER_taxa_name <- "PLACEHOLDER"
PLACEHOLDER_mean_AC_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa <- mean(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "PLACEHOLDER" & `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$Host == "AC")$value)
PLACEHOLDER_mean_AR_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa <- mean(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "PLACEHOLDER" & `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$Host == "AR")$value)
`2021-09-13_PLACEHOLDER_foldchange_AC_to_AR_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa` <- foldchange(PLACEHOLDER_mean_AC_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa, PLACEHOLDER_mean_AR_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa)
PLACEHOLDER_mean_J_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa <- mean(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "PLACEHOLDER" & `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$Season == "J")$value)
PLACEHOLDER_mean_N_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa <- mean(subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "PLACEHOLDER" & `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$Season == "N")$value)
`2021-09-13_PLACEHOLDER_foldchange_J_to_N_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa` <- foldchange(PLACEHOLDER_mean_J_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa, PLACEHOLDER_mean_N_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa)
`2021-09-13_PLACEHOLDER_logratio_AC_to_AR_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa` <- foldchange2logratio(`2021-09-13_PLACEHOLDER_foldchange_AC_to_AR_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`)
`2021-09-13_PLACEHOLDER_logratio_J_to_N_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa` <- foldchange2logratio(`2021-09-13_PLACEHOLDER_foldchange_J_to_N_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`)
`2021_09_13_summed_to_taxa_foldchanges_and_logratios_only_paired_samples_log1p_transformed_contig_and_library_size_adjusted_no_pilot_samples`[nrow(`2021_09_13_summed_to_taxa_foldchanges_and_logratios_only_paired_samples_log1p_transformed_contig_and_library_size_adjusted_no_pilot_samples`)+1,] <- c(`PLACEHOLDER_taxa_name`, `2021-09-13_PLACEHOLDER_foldchange_AC_to_AR_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`, `2021-09-13_PLACEHOLDER_logratio_AC_to_AR_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`, `2021-09-13_PLACEHOLDER_foldchange_J_to_N_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`, `2021-09-13_PLACEHOLDER_logratio_J_to_N_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`)
`2021_09_13_summed_to_taxa_foldchanges_and_logratios_only_paired_samples_log1p_transformed_contig_and_library_size_adjusted_no_pilot_samples`[, c(2:5)] <- sapply(`2021_09_13_summed_to_taxa_foldchanges_and_logratios_only_paired_samples_log1p_transformed_contig_and_library_size_adjusted_no_pilot_samples`[, c(2:5)], as.numeric)
rm(`PLACEHOLDER_mean_AC_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`)
rm(`PLACEHOLDER_mean_AR_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`)
rm(`PLACEHOLDER_taxa_name`)
rm(`2021-09-13_PLACEHOLDER_foldchange_AC_to_AR_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`)
rm(`PLACEHOLDER_mean_J_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`)
rm(`PLACEHOLDER_mean_N_paired_samples_only_no_pilots_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`)
rm(`2021-09-13_PLACEHOLDER_foldchange_J_to_N_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`)
rm(`2021-09-13_PLACEHOLDER_logratio_AC_to_AR_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`)
rm(`2021-09-13_PLACEHOLDER_logratio_J_to_N_paired_samples_no_pilot_log1p_transformed_library_and_contig_size_adjusted_summed_by_taxa`)

qplot(`2021_09_13_summed_to_taxa_foldchanges_and_logratios_only_paired_samples_log1p_transformed_contig_and_library_size_adjusted_no_pilot_samples`$AC_versus_AR_fold_change)
qplot(`2021_09_13_summed_to_taxa_foldchanges_and_logratios_only_paired_samples_log1p_transformed_contig_and_library_size_adjusted_no_pilot_samples`$AC_versus_AR_logratio)

2022-01-10
ggplot(data = melted_paper_1_percent_reads, aes(fill = Sample, y = value, x = variable)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Sample") + ylab("Percent reads classified") + guides(fill=guide_legend(title="Classifier used")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + coord_cartesian(ylim = c(0, 100)) + labs(title="Percentage of reads classified by each taxonomic classifier at phylum level or below")

ggplot(data = melted_paper_1_percent_reads, aes(fill = Sample, y = value, x = variable)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Sample") + ylab("Percent reads classified") + guides(fill=guide_legend(title="Classifier used")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + coord_cartesian(ylim = c(0, 100)) + labs(title="Percentage of reads classified by each taxonomic classifier at phylum level or below")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-01-10_pilot_samples_subsampled_QCed_cleansed_and_human_mock_QCed_and_cleansed_percent_reads_classified_any_level_by_kaiju_kraken_and_metaphlan.svg")

2022-01-23
ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Human_mock_percent_reads_classified, x = Classifier_and_parameter)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90"))

ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Human_mock_percent_reads_classified, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + ylab("Percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "Difference in percent reads classified by different taxonomic classifiers using different parameters\nfrom human mock metagenomic sample") + guides(fill=guide_legend(title="Classifier used"))

ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Average_percent_reads_classified_Acomys_cahirinus_samples, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + ylab("Average percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "Difference in average percent reads classified by different taxonomic classifiers using different parameters\nfrom Acomys cahirinus samples") + guides(fill=guide_legend(title="Classifier used"))

ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Average_percent_reads_classified_Acomys_russatus_samples, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + ylab("Average percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "Difference in average percent reads classified by different taxonomic classifiers using different parameters\nfrom Acomys russatus samples") + guides(fill=guide_legend(title="Classifier used"))

ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Average_percent_reads_classified_June_samples, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + ylab("Average percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "Difference in average percent reads classified by different taxonomic classifiers using different parameters\nfrom both Acomys species June samples") + guides(fill=guide_legend(title="Classifier used"))

ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Average_percent_reads_classified_November_samples, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + ylab("Average percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "Difference in average percent reads classified by different taxonomic classifiers using different parameters\nfrom both Acomys species November samples") + guides(fill=guide_legend(title="Classifier used"))

melted_2022_01_23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters <- reshape2::melt(`2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, id=c("Classifier_and_parameter"))

ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Average_percent_reads_classified_Acomys_cahirinus_samples, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + ylab("Average percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "Difference in average percent reads classified by different taxonomic classifiers using different parameters\nfrom Acomys cahirinus samples") + guides(fill=guide_legend(title="Classifier used"))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022_01_23_human_mock_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters.svg")

avg_cah_plot <- ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Average_percent_reads_classified_Acomys_cahirinus_samples, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + ylab("Average percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "A") + guides(fill=guide_legend(title="Classifier used")) + theme(legend.position = "none", axis.title.x = element_blank())

avg_rus_plot <- ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Average_percent_reads_classified_Acomys_russatus_samples, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + ylab("Average percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "B") + guides(fill=guide_legend(title="Classifier used")) + theme(axis.title.x = element_blank(), axis.title.y = element_blank())

avg_rus_plot <- ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Average_percent_reads_classified_Acomys_russatus_samples, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + ylab("Average percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "B") + guides(fill=guide_legend(title="Classifier used")) + theme(legend.position = "none", axis.title.x = element_blank(), axis.title.y = element_blank())


grid.arrange(avg_cah_plot, avg_rus_plot, species_averages_multiple_classifier_parameters_legend, ncol = 3, widths=c(4.5, 4.5, 1), top = "Average percent reads classified in Acomys samples by taxonomic classifiers using different parameters", bottom = "A gives mean values for Acomys cahirinus samples, B gives mean values for Acomys russatus samples")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022_01_23_pilot_samples_combined_figures_average_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters_by_host_species.svg")


fake_kaiju_false_positive_taxa_df <- data.frame(Kaiju_errors_allowed = c("0","1","2","3","4","5","10","20"), False_positive_taxa = c(3, 5, 7, 15, 35, 51, 129, 297))

fake_kraken_false_positive_taxa_df <- data.frame(Kraken_confidence_score_percentage = c("95","75","66","50","25","10"), False_positive_taxa = c(27, 93, 187, 589, 2037, 15786))


avg_jun_plot <- ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Average_percent_reads_classified_June_samples, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + ylab("Average percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "A") + guides(fill=guide_legend(title="Classifier used")) + theme(legend.position = "none", axis.title.x = element_blank())

avg_nov_plot <- ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Average_percent_reads_classified_November_samples, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#ff1e90")) + ylab("Average percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "B") + guides(fill=guide_legend(title="Classifier used")) + theme(legend.position = "none", axis.title.x = element_blank(), axis.title.y = element_blank())

grid.arrange(avg_jun_plot, avg_nov_plot, species_averages_multiple_classifier_parameters_legend, ncol = 3, widths=c(4.5, 4.5, 1), top = "Average percent reads classified in Acomys samples by taxonomic classifiers using different parameters", bottom = "A gives mean values for June samples of both species, B gives mean values for November samples of both species")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022_01_23_pilot_samples_combined_figures_average_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters_by_sample_collection_month.svg")


fake_kaiju_false_positive_taxa_df <- data.frame(Kaiju_errors_allowed = c("0","1","2","3","4","5","10","20"), False_positive_taxa = c(3, 5, 7, 15, 35, 51, 129, 297))

fake_kraken_false_positive_taxa_df <- data.frame(Kraken_confidence_score_percentage = c("95","75","66","50","25","10"), False_positive_taxa = c(27, 93, 187, 589, 2037, 15786))

ggplot(data = fake_kaiju_false_positive_taxa_df, aes(x = Kaiju_errors_allowed, y = False_positive_taxa)) + geom_line(linetype = "dashed", color = "#1e90ff") + geom_point() + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Errors allowed") + ylab("False positive taxa") + labs(title = "False taxa detections by Kaiju using different allowances for errors in processed mock human metagenomic community")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022_01_23_FAKE_DATA_ILLUSTRATIVE_ONLY-False_kaiju_detections_different_error_allowances_human_mock.svg")

ggplot(data = fake_kraken_false_positive_taxa_df, aes(x = Kraken_confidence_score_percentage, y = False_positive_taxa)) + geom_line(linetype = "dashed", color = "#ff8d1e") + geom_point() + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Confidence score (%)") + ylab("False positive taxa") + labs(title = "False taxa detections by Kraken using different confidence scores in processed mock human metagenomic community")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022_01_23_FAKE_DATA_ILLUSTRATIVE_ONLY-False_kraken_detections_different_confidence_scores_human_mock.svg")

melted_2022_01_23_FAKE_DATA_Pilot_samples_each_sample_percent_reads_same_classification_between_within_or_not_tools

ggplot(data = `melted_2022_01_23_FAKE_DATA_Pilot_samples_each_sample_percent_reads_same_classification_between_within_or_not_tools`, aes(fill = variable, y = value, x = Sample)) + geom_bar(position="stack", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#66023c")) + ylab("Percent of reads") + guides(fill=guide_legend(title="Classification agreement status")) + labs(title = "Percent of reads in Kaiju and Kraken output from multiple parameter runs by degree of consensus on TaxID\nfrom processed Acomys faecal samples")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022_01_23_FAKE_DATA_ILLUSTRATIVE_ONLY-Pilot_samples_taxid_agreement_status_Kaiju_and_Kraken_within_and_across_parameters_and_tools.svg")

2022-01-24
`2021.09.02_TRANSPOSED.all_subsampled_reads_mapped_to_bins_contig_and_library_size_adjusted_scores_summed_by_bin_contig_is_from`
`pca_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`

ggbiplot::ggbiplot(`PCA_log1p_group_by_bins`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`log1p_group_by_bins`$Host, ellipse.prob = 0.25, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`log1p_group_by_bins`$Season), color=factor(`log1p_group_by_bins`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p adjusted, kilobase and library size adjusted, summed Bin level mapped read segment counts for\nsubsampled reads to MAGs from Acomys") + guides(fill = guide_legend(override.aes = list(shape = 21))) + labs(caption = "Probability ellipse indicating 25%") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_text(aes(label=ifelse(`log1p_group_by_bins`$SampleID=="AC19J", "AC19J", " ")), hjust=0,vjust=0)

ggbiplot::ggbiplot(`pca_log1p_transformed_2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups=`2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE) + theme_linedraw() + geom_point(aes(shape=factor(`2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`$Season), color=factor(`2021.08.06_Transposed.Combined_contig_and_library_size_adjusted_taxid_summed_mapped_read_segment_counts`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ffb700')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Season", colour="Host species") + labs(title = "Log1p transformed, mapped read segment counts for individual TaxIDs in subsampled reads", caption="Counts were adjusted for contig and library size then summed for each TaxID from identified contigs")

2022-02-01
`2022.01.30_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`

`melted_2022.01.30_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores` <- reshape2::melt(`2022.01.30_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`, id=c("Sample", "Host_species",))

`melted_2022.01.30_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`$variable <- gsub("X", "", as.character(`melted_2022.01.30_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`$variable))

`melted_2022.01.30_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`$variable <- gsub(".", "", as.character(`melted_2022.01.30_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`$variable))

ggplot(data = `melted_2022.01.30_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`, aes(x = Sample, y = value)) + theme_linedraw() + geom_jitter(aes(color = variable), width = 0.35, size = 2.25) + scale_color_brewer(palette = "Dark2") + scale_fill_brewer(palette = "Dark2") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Sample), alpha = 0.25)

PLACEHOLDER_ggplot_boxplot_and_wilcox_test <- ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for PLACEHOLDER according to host and season groupings", caption = "Wilcox test by stat_compare_means") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Host_by_season), alpha = 0.25) + stat_compare_means(method = "wilcox.test", comparisons = host_and_season_within_and_across_comparisons)

`2022.01.30_Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`

`2022.01.30_melted-Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores` <- reshape2::melt(`2022.01.30_Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`, id=c("Host_species", "Sample_ID", "X10.00", "X25.00", "X50.00", "X66.00", "X75.00", "X95.00"))

`2022.01.30_melted-Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores` %>% dplyr::rename("10%" = "X10.00")
names(`2022.01.30_melted-Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`)[3] <- "10%"
names(`2022.01.30_melted-Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`)[4] <- "25%"
names(`2022.01.30_melted-Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`)[5] <- "50%"
names(`2022.01.30_melted-Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`)[6] <- "66%"
names(`2022.01.30_melted-Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`)[7] <- "75%"
names(`2022.01.30_melted-Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`)[8] <- "95%"
names(`2022.01.30_melted-Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`)[1] <- "Host species"
names(`2022.01.30_melted-Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`)[2] <- "Sample ID"

ggplot(data = `2022.01.30_melted-Transposed_Kraken_all_samples_percent_reads_unclassified_all_confidence_scores`, aes(x = 10%, y = value)) + theme_linedraw() + geom_jitter(aes(color = variable), width = 0.35, size = 2.25) + scale_color_brewer(palette = "Dark2") + scale_fill_brewer(palette = "Dark2") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Sample), alpha = 0.25)

2022-02-02
`2022.02.02_pilot_samples_percent_classified_from_direct_output_files_Kaiju_Kraken_multiple_parameters_and_Metaphlan`

melted_temp <- reshape2::melt(`2022.02.02_pilot_samples_percent_classified_from_direct_output_files_Kaiju_Kraken_multiple_parameters_and_Metaphlan`, id=c("Host_species", "Sample_ID", "Kraken_95_percent_confidence", "Kraken_75_percent_confidence", "Kraken_66_percent_confidence", "Kraken_50_percent_confidence", "Kraken_25_percent_confidence", "Kraken_10_percent_confidence", "Kaiju_0_errors_allowed", "Kaiju_1_error_allowed", "Kaiju_2_errors_allowed", "Kaiju_3_errors_allowed", "Kaiju_4_errors_allowed", "Kaiju_5_errors_allowed", "Kaiju_10_errors_allowed", "Kaiju_20_errors_allowed", "Sample", "Metaphlan"))

ggplot(melted_temp, aes(x = Host_species, y = Kraken_95_percent_confidence, color = Host_species)) + geom_boxplot(aes(fill = Host_species), alpha = 0.25) + geom_jitter(width = 0.35) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="hot") + scale_color_posborne(palette="hot")

melted_temp_2 <- reshape2::melt(`2022.02.02_pilot_samples_percent_classified_from_direct_output_files_Kaiju_Kraken_multiple_parameters_and_Metaphlan`, id=c("Host_species", "Sample_ID", "Sample"), measure.vars=c("Kraken_95_percent_confidence", "Kraken_75_percent_confidence", "Kraken_66_percent_confidence", "Kraken_50_percent_confidence", "Kraken_25_percent_confidence", "Kraken_10_percent_confidence", "Kaiju_0_errors_allowed", "Kaiju_1_error_allowed", "Kaiju_2_errors_allowed", "Kaiju_3_errors_allowed", "Kaiju_4_errors_allowed", "Kaiju_5_errors_allowed", "Kaiju_10_errors_allowed", "Kaiju_20_errors_allowed", "Metaphlan"))


ggplot(melted_temp_2, aes(x = Host_species, y = value, color = variable)) + geom_boxplot(alpha = 0.25) + geom_jitter(width = 0.35) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble")


ggplot(melted_temp_2, aes(x = Host_species, y = value, color = variable)) + geom_boxplot(alpha = 0.25) + geom_jitter(width = 0.35) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble")

subset(`melted_temp_2`, grepl(glob2rx("Kaiju*"), variable) )

subset(`melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`, `melted_2021.09.06.both_host_species_paired_non_pilot_log1p_counts`$variable == "Unclassified")

ggplot(subset(`melted_temp_2`, grepl(glob2rx("Kaiju*"), variable) ), aes(x = Host_species, y = value, color = variable)) + geom_boxplot(alpha = 0.25) + geom_jitter(width = 0.15) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble")

`melted_temp_2`$variable <- gsub("Kaiju_20_errors_allowed", "Kaiju 20 allowed errors", `melted_temp_2`$variable)

ggplot(subset(`melted_temp_2`, grepl(glob2rx("Kaiju*"), variable) ), aes(x = Host_species, y = value, color = variable, fill = variable)) + geom_boxplot(alpha = 0.25, notch = FALSE) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble") + xlab("Host species") + ylab("Percent reads classified") + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + labs(title = "Percent of subsampled Acomys faecal sample reads classified by Kaiju using different parameters for each host species") + stat_boxplot(geom = 'errorbar')

ggplot(subset(`melted_temp_2`, grepl(glob2rx("Kaiju*"), variable) ), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble") + xlab("Host species") + ylab("Percent reads classified") + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + labs(title = "Percent of subsampled Acomys faecal sample reads classified by Kaiju using different parameters for each host species")

ggplot(data = `2022.01.23_human_mock_and_pilot_sample_averages_by_species_or_season_percent_reads_classified_all_classifiers_Kaiju_and_Kraken_multiple_parameters`, aes(y = Human_mock_percent_reads_classified, x = Classifier_and_parameter, fill = Classifier_for_ggplot)) + geom_bar(position="dodge", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + ylab("Percent reads classified") + xlab("Classifier and parameter adjusted") + labs(title = "Difference in percent reads classified by different taxonomic classifiers using different parameters\nfrom human mock metagenomic sample") + guides(fill=guide_legend(title="Classifier used")) + ggpubr::rotate_x_text()

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-02_Human_mock_all_classifiers_all_parameters_percent_reads_classified.svg")

ggplot(subset(`melted_temp_2`, grepl(glob2rx("Kaiju*"), variable) ), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble") + xlab("Sample ID") + ylab("Percent reads classified") + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + labs(title = "Percent of subsampled Acomys faecal sample reads classified by Kaiju using different parameters for each sample")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-02_Pilot_samples_percent_reads_classified_Kaiju_different_parameters_per_sample.svg")

ggplot(subset(`melted_temp_2`, grepl(glob2rx("Kraken*|Metaphlan"), variable) ), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble") + xlab("Sample ID") + ylab("Percent reads classified") + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + labs(title = "Percent of subsampled Acomys faecal sample reads classified by Kraken or Metaphlan using different parameters\nper sample")

2022-02-02_pilot_reads_percent_classified_metaphlan_or_kraken_multiple_parameters
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-02_pilot_reads_percent_classified_metaphlan_or_kraken_multiple_parameters.svg")

`melted_temp_2`$variable <- gsub(" %", "%", `melted_temp_2`$variable)

ggplot(melted_temp_2, aes(x = Host_species, y = value, color = variable, fill = variable)) + geom_boxplot(alpha = 0.35) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble") + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + ylab("Percent reads classified") + xlab("Host species")

ggplot(subset(`melted_temp_2`, grepl(glob2rx("AC"), Host_species)), aes(x = Host_species, y = value, color = variable, fill = variable)) + geom_boxplot(alpha = 0.35) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble") + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + ylab("Percent reads classified") + xlab("Host species")

ggplot(melted_temp_2, aes(x = Sample, y = value, color = variable, fill = variable)) + geom_boxplot(alpha = 0.35) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble") + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + ylab("Percent reads classified") + xlab("Host species") + facet_wrap( ~ Sample, scales="free") + ggtitle("Percent Acomys reads classified by different tools using multiple parameters per host species")

ggplot(melted_temp_2, aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble") + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + ylab("Percent reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free") + ggtitle("Percent Acomys reads classified by different tools using multiple parameters in each sample")  + theme(strip.background = element_blank(), strip.text.x = element_blank())

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-02_all_pilot_samples_all_classifiers_all_parameters_percent_reads_classified.svg")

ggplot(melted_temp_2, aes(x = Host_species, y = value, color = variable, fill = variable)) + geom_boxplot(width = 1, alpha = 0.35) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble") + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + ylab("Percent reads classified") + xlab("Host species") + facet_wrap( ~ Host_species, scales="free") + ggtitle("Percent Acomys reads classified by different tools using multiple parameters per host species") + theme(strip.background = element_blank(), strip.text.x = element_blank())

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-02_all_pilot_samples_all_classifiers_all_parameters_percentage_reads_classified.svg")

ggplot(data = `2022.02.02_human_mock_true_positives_kraken_confidence_scores`, aes(x = Confidence.score)) + geom_line(aes(y = True.positives), linetype = "dashed", color = "blue") + geom_line(aes(y = FAKE.False.positives / kraken_false_pos_coeff), color = "orange") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Confidence score (%)") + ylab("True positive taxa") + labs(title = "True positive taxa detected using different confidence scores in processed mock human metagenomic community by Kraken2") + ylim(0, 20)

ggplot(data = `2022.02.02_human_mock_true_positives_kraken_confidence_scores`) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_line(aes(x = Confidence.score, y = True.positives), color = "#002fa7", linetype = "dashed") + geom_line(aes(x = Confidence.score, y = FAKE.False.positives), color = "#ff4f00", linetype = "dotdash") + geom_point(aes(x = Confidence.score, y = True.positives), color = "#002fa7") + geom_point(aes(x = Confidence.score, y = FAKE.False.positives), color = "#ff4f00") + theme(legend.position="top")

`melted_2022.02.02_human_mock_true_positives_kraken_confidence_scores` <- reshape2::melt(`2022.02.02_human_mock_true_positives_kraken_confidence_scores`, id=c("Confidence.score"), measure.vars=c("True.positives", "FAKE.False.positives"))

melted_2022.02.02_human_mock_true_positives_kraken_confidence_scores

`2022.02.02_pilot_samples_percent_classified_from_direct_output_files_Kaiju_Kraken_multiple_parameters_and_Metaphlan` <- melted_temp_2

`melted_2022.02.02_human_mock_true_positives_kraken_confidence_scores` <- reshape2::melt(`2022.02.02_V2_human_mock_true_positives_kraken_confidence_scores`, id=c("Status"), measure.vars=c("Confidence.score", "Species"))

ggplot(data = `melted_2022.02.02_human_mock_true_positives_kraken_confidence_scores`, aes(x = "Confidence.score")) + geom_line(aes(y = value, color = Status)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))


ggplot() + geom_line(data = subset(`2022.02.02_V2_human_mock_true_positives_kraken_confidence_scores`, grepl(glob2rx("True*"), Status)), aes(x = Confidence.score, y = Species, color = Status, linetype = Status)) + geom_line(data = subset(`2022.02.02_V2_human_mock_true_positives_kraken_confidence_scores`, grepl(glob2rx("False*"), Status)), aes(x = Confidence.score, y = Species, color = Status, linetype = Status)) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_posborne(palette="common") + scale_x_continuous(name = "Confidence score (%)", limits = c(0, 100), breaks = scales::pretty_breaks(n = 10))

ggplot() + geom_line(data = subset(`2022.02.02_V2_human_mock_true_positives_kraken_confidence_scores`, grepl(glob2rx("True*"), Status)), aes(x = Confidence.score, y = Species, color = Status, linetype = Status), size = 0.75) + geom_line(data = subset(`2022.02.02_V2_human_mock_true_positives_kraken_confidence_scores`, grepl(glob2rx("False*"), Status)), aes(x = Confidence.score, y = Species, color = Status, linetype = Status), size = 0.75) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_posborne(palette="common") + scale_x_continuous(name = "Confidence score (%)", limits = c(10, 100), breaks = scales::pretty_breaks(n = 9)) + ylab("No. species detected") + labs(title = "Number of true and false positive species level detections by Kraken2 at different confidence levels from\nmock human microbiota data")

ggplot() + geom_line(data = subset(`2022.02.02_V2_human_mock_true_positives_kraken_confidence_scores`, grepl(glob2rx("True*"), Status)), aes(x = Confidence.score, y = Species, color = Status, linetype = Status), size = 0.75) + geom_line(data = subset(`2022.02.02_V2_human_mock_true_positives_kraken_confidence_scores`, grepl(glob2rx("False*"), Status)), aes(x = Confidence.score, y = Species, color = Status, linetype = Status), size = 0.75) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_posborne(palette="common") + scale_x_continuous(name = "Confidence score (%)", limits = c(10, 100), breaks = scales::pretty_breaks(n = 9)) + ylab("No. species detected") + labs(title = "Number of true and false positive species level detections by Kraken2 at different confidence levels from\nmock human microbiota data") + scale_y_continuous(limits = c(0, 100), breaks = scales::pretty_breaks(n = 10))


ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-02_Kraken_multiple_confidence_scores_human_mock_true_and_false_positive_species_level_detections.svg")

2022-02-03

`2022.02.02_Kaiju_human_mock_species_true_and_false_positive_counts_all_error_allowances`
`2022.02.02_MELTED_Kaiju_human_mock_species_true_and_false_positive_counts_all_error_allowances` <- reshape2::melt(`2022.02.02_Kaiju_human_mock_species_true_and_false_positive_counts_all_error_allowances`, id=c("Status"), measure.vars=c("Errors.allowed", "Species"))

ggplot() + geom_line(data = subset(`2022.02.02_Kaiju_human_mock_species_true_and_false_positive_counts_all_error_allowances`, grepl(glob2rx("True*"), Status)), aes(x = Errors.allowed, y = Species, color = Status, linetype = Status), size = 0.75) + geom_line(data = subset(`2022.02.02_Kaiju_human_mock_species_true_and_false_positive_counts_all_error_allowances`, grepl(glob2rx("False*"), Status)), aes(x = Errors.allowed, y = Species, color = Status, linetype = Status), size = 0.75) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_posborne(palette="common") + ylab("No. species detected") + labs(title = "Number of true and false positive species level detections by Kaiju with different numbers of match errors allowed from\nmock human microbiota data") + scale_y_continuous(trans='log10')

ggplot() + geom_line(data = subset(`2022.02.02_Kaiju_human_mock_species_true_and_false_positive_counts_all_error_allowances`, grepl(glob2rx("True*"), Status)), aes(x = Errors.allowed, y = Species, color = Status, linetype = Status), size = 0.75) + geom_line(data = subset(`2022.02.02_Kaiju_human_mock_species_true_and_false_positive_counts_all_error_allowances`, grepl(glob2rx("False*"), Status)), aes(x = Errors.allowed, y = Species, color = Status, linetype = Status), size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_posborne(palette="common") + ylab("No. species detected") + labs(title = "Number of true and false positive species level detections by Kaiju with different numbers of match errors allowed from\nmock human microbiota data") + scale_y_continuous(trans='log10') + annotation_logticks(sides = "l") + xlab("Allowed errors in match by Kaiju")

2022-02-03_Human_mock_Kaiju_multiple_parameters_number_species_detected_true_and_false_positives
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-03_Human_mock_Kaiju_multiple_parameters_number_species_detected_true_and_false_positives.svg")

krak_plot <- ggplot() + geom_line(data = subset(`2022.02.02_V2_human_mock_true_positives_kraken_confidence_scores`, grepl(glob2rx("True*"), Status)), aes(x = Confidence.score, y = Species, color = Status, linetype = Status), size = 0.75) + geom_line(data = subset(`2022.02.02_V2_human_mock_true_positives_kraken_confidence_scores`, grepl(glob2rx("False*"), Status)), aes(x = Confidence.score, y = Species, color = Status, linetype = Status), size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_posborne(palette="common") + scale_x_continuous(name = "Confidence score (%)", limits = c(10, 100), breaks = scales::pretty_breaks(n = 9)) + ylab("No. species detected") + labs(title = "A") + scale_y_continuous(limits = c(0, 100), breaks = scales::pretty_breaks(n = 10)) + theme(legend.position = "none")
kaiju_plot <- ggplot() + geom_line(data = subset(`2022.02.02_Kaiju_human_mock_species_true_and_false_positive_counts_all_error_allowances`, grepl(glob2rx("True*"), Status)), aes(x = Errors.allowed, y = Species, color = Status, linetype = Status), size = 0.75) + geom_line(data = subset(`2022.02.02_Kaiju_human_mock_species_true_and_false_positive_counts_all_error_allowances`, grepl(glob2rx("False*"), Status)), aes(x = Errors.allowed, y = Species, color = Status, linetype = Status), size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_posborne(palette="common") + ylab("No. species detected") + labs(title = "B") + scale_y_continuous(trans='log10') + annotation_logticks(sides = "l") + xlab("Allowed errors in match by Kaiju")

grid.arrange(krak_plot, kaiju_plot, ncol = 2, widths = c(4.5, 5), top = "True and false positive species detections by Kraken and Kaiju using different parameters from human mock microbiota")

grid.arrange(avg_jun_plot, avg_nov_plot, species_averages_multiple_classifier_parameters_legend, ncol = 3, widths=c(4.5, 4.5, 1), top = "Average percent reads classified in Acomys samples by taxonomic classifiers using different parameters", bottom = "A gives mean values for June samples of both species, B gives mean values for November samples of both species")

kaiju_plot <- ggplot() + geom_line(data = subset(`2022.02.02_Kaiju_human_mock_species_true_and_false_positive_counts_all_error_allowances`, grepl(glob2rx("True*"), Status)), aes(x = Errors.allowed, y = Species, color = Status, linetype = Status), size = 0.75) + geom_line(data = subset(`2022.02.02_Kaiju_human_mock_species_true_and_false_positive_counts_all_error_allowances`, grepl(glob2rx("False*"), Status)), aes(x = Errors.allowed, y = Species, color = Status, linetype = Status), size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_posborne(palette="common") + ylab("No. species detected") + labs(title = "B") + scale_y_continuous(trans='log10') + annotation_logticks(sides = "l") + xlab("Allowed errors in match by Kaiju") + theme(legend.position = "none")

ggarrange(krak_plot, kaiju_plot, ncol=2, nrow=1, common.legend = TRUE, legend = "bottom", font.label = list(size = 14, font = "helvetica"))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-03_Human_mock_Kaiju__and_kraken_combined_multiple_parameters_number_species_detected_true_and_false_positives.svg")

working_temp_fake <- `2022.02.03.FAKE_DATA_illustrative_purposes_only_pilot_samples_agreement_within_tools_across_tools_per_read_Kaiju_Kraken`

melted_working_temp_fake <- reshape2::melt(working_temp_fake, id=c("Sample"))

ggplot(data = melted_working_temp_fake) + geom_bar(aes(x = Sample, y = value, fill = variable), position="stack", stat="identity")

`melted_working_temp_fake`$variable <- gsub("_", " ", as.character(`melted_working_temp_fake`$variable))

ggplot(data = melted_working_temp_fake) + geom_bar(aes(x = Sample, y = value, fill = variable), position="stack", stat="identity") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + ylab("% reads in sample") + xlab("Sample ID") + guides(fill = guide_legend(title = "Classification agreement status")) + labs(title = "ILLUSTRATIVE DATA ONLY\nPercent reads in sample by classification agreement status from Kaiju and Kraken runs with different parameters")

2022-02-03_FAKE_DATA_Pilot_samples_percent_reads_in_each_sample_classification_agreement_within_across_tools_and_parameters

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-03_FAKE_DATA_Pilot_samples_percent_reads_in_each_sample_classification_agreement_within_across_tools_and_parameters.svg")

2022-02-13

working_file <- `2022.02.09_All_processed_sample_files_percent_classified_reads_from_raw_files_Kaiju_Kraken_from_report_Metaphlan`

melted_working_file <- reshape2::melt(`working_file`, id=c("Sample", "Host_species", "Sample_ID"), measure.vars=c("Metaphlan", "Kaiju_0_errors_allowed", "Kaiju_1_error_allowed", "Kaiju_2_errors_allowed", "Kaiju_3_errors_allowed", "Kaiju_4_errors_allowed", "Kaiju_5_errors_allowed", "Kaiju_10_errors_allowed", "Kaiju_20_errors_allowed", "Kraken_95_percent_confidence", "Kraken_75_percent_confidence", "Kraken_66_percent_confidence", "Kraken_50_percent_confidence", "Kraken_25_percent_confidence", "Kraken_10_percent_confidence"))

ggplot(melted_working_file, aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_posborne(palette="jumble") + scale_color_posborne(palette="jumble") + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + ylab("Percent reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free") + ggtitle("Percent reads classified by different tools using multiple parameters in each sample")  + theme(strip.background = element_blank(), strip.text.x = element_blank())

`melted_working_file`$variable <- gsub("_", " ", as.character(`melted_working_file`$variable))
`melted_working_file`$variable <- gsub("percent", "%", as.character(`melted_working_file`$variable))

ggplot(melted_working_file, aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + ylab("Percent reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free") + ggtitle("Percent reads classified by different tools using multiple parameters in each sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Metaphlan" = "#124E78", "Kaiju_0_errors_allowed" = "#0e0d0a", "Kaiju_1_error_allowed" = "#3d3829", "Kaiju_2_errors_allowed" = "#6c6349", "Kaiju_3_errors_allowed" = "#8f8460", "Kaiju_4_errors_allowed" = "#6f560b", "Kaiju_5_errors_allowed" = "#a29775", "Kaiju_10_errors_allowed" = "#c2baa4", "Kaiju_20_errors_allowed" = "#e9e6de", "Kraken_95_percent_confidence" = "#de541e", "Kraken_75_percent_confidence" = "#aa4017", "Kraken_66_percent_confidence" = "#e3602d", "Kraken_50_percent_confidence" = "#e87a50", "Kraken_25_percent_confidence" = "#e87a50", "Kraken_10_percent_confidence" = "#f8d6c9")) + scale_color_manual(values = c("Metaphlan" = "#124E78", "Kaiju_0_errors_allowed" = "#0e0d0a", "Kaiju_1_error_allowed" = "#3d3829", "Kaiju_2_errors_allowed" = "#6c6349", "Kaiju_3_errors_allowed" = "#8f8460", "Kaiju_4_errors_allowed" = "#6f560b", "Kaiju_5_errors_allowed" = "#a29775", "Kaiju_10_errors_allowed" = "#c2baa4", "Kaiju_20_errors_allowed" = "#e9e6de", "Kraken_95_percent_confidence" = "#de541e", "Kraken_75_percent_confidence" = "#aa4017", "Kraken_66_percent_confidence" = "#e3602d", "Kraken_50_percent_confidence" = "#e87a50", "Kraken_25_percent_confidence" = "#e87a50", "Kraken_10_percent_confidence" = "#f8d6c9"))

ggplot(subset(`melted_working_file`, grepl(glob2rx("Kraken*"), variable)), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free") + ggtitle("Percent reads classified by Kraken using different confidence scores per sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))

ggplot(subset(`melted_working_file`, grepl(glob2rx("Kaiju*"), variable)), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors"))

ggplot(subset(`melted_working_file`, grepl(glob2rx("Kaiju*"), variable)), aes(x = reorder(variable, `kaiju_x_axis_level_order`), y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors"))

kaiju_x_axis_level_order <- c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')

ggplot(subset(`melted_working_file`, grepl(glob2rx("Kaiju*"), variable)), aes(x = fct_relevel(Sample, "Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed"), y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors"))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-13-All_samples_percent_reads_classified_by_Kraken_all_confidence_scores.svg")

ggplot(subset(`melted_working_file`, grepl(glob2rx("Kaiju*"), variable)), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + scale_fill_discrete(limits = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed")) + scale_color_discrete(limits = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed"))

for_sorting_melted_working <- subset(`melted_working_file`, grepl(glob2rx("Kaiju*"), variable))

`for_sorting_melted_working`$variable <- gsub("Kaiju", "", as.character(`for_sorting_melted_working`$variable))
`for_sorting_melted_working`$variable <- gsub("errors allowed", "", as.character(`for_sorting_melted_working`$variable))
`for_sorting_melted_working`$variable <- gsub("error allowed", "", as.character(`for_sorting_melted_working`$variable))

`for_sorting_melted_working`$variable <- factor(`for_sorting_melted_working`$variable, levels = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed"))

ggplot(`for_sorting_melted_working`, aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors"))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-13-All_samples_percent_reads_classified_by_Kaiju_all_allowed_errors.svg")

ggplot(subset(`melted_working_file`, grepl(glob2rx("Metaphlan"), variable)), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + ggtitle("Percent reads classified by Metaphlan using different confidence scores per sample") + scale_fill_manual(values = c("Metaphlan" = "#141414")) + scale_color_manual(values = c("Metaphlan" = "#141414")) + theme(legend.position = "none") + theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-13-All_samples_percent_reads_classified_by_Metaphlan.svg")

2022-02-18
feb_18th_working_file <- `2022.02.18_All_processed_sample_files_only_more_recent_lab_mouse_samples_percent_classified_reads_from_raw_files_Kaiju_Kraken_from_report_Metaphlan`

melted_feb_18th_working_file <- reshape2::melt(feb_18th_working_file, id=c("Sample", "Host_species", "Sample_ID"), measure.vars=c("Metaphlan", "Kaiju_0_errors_allowed", "Kaiju_1_error_allowed", "Kaiju_2_errors_allowed", "Kaiju_3_errors_allowed", "Kaiju_4_errors_allowed", "Kaiju_5_errors_allowed", "Kaiju_10_errors_allowed", "Kaiju_20_errors_allowed", "Kraken_95_percent_confidence", "Kraken_75_percent_confidence", "Kraken_66_percent_confidence", "Kraken_50_percent_confidence", "Kraken_25_percent_confidence", "Kraken_10_percent_confidence"))

`melted_feb_18th_working_file`$variable <- gsub("_", " ", as.character(`melted_feb_18th_working_file`$variable))
`melted_feb_18th_working_file`$variable <- gsub("percent", "%", as.character(`melted_feb_18th_working_file`$variable))
`melted_feb_18th_working_file`$Sample_ID <- gsub("_", " ", as.character(`melted_feb_18th_working_file`$Sample_ID))
`melted_feb_18th_working_file`$Sample <- gsub("_", " ", as.character(`melted_feb_18th_working_file`$Sample))

ggplot(subset(`melted_feb_18th_working_file`, grepl(glob2rx("Metaphlan"), variable)), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + ggtitle("Percent reads classified by Metaphlan using different confidence scores per sample") + scale_fill_manual(values = c("Metaphlan" = "#141414")) + scale_color_manual(values = c("Metaphlan" = "#141414")) + theme(legend.position = "none") + theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(subset(`melted_feb_18th_working_file`, grepl(glob2rx("Metaphlan"), variable)), aes(x = Sample, y = value, color = Host_species, fill = Host_species)) + geom_bar(stat = "identity", position = position_dodge()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + ggtitle("Percent reads classified by Metaphlan using different confidence scores per sample") + scale_fill_manual(values = c("HS" = "#010305", "MM" = "#16468d", "AC" = "#5b92e5", "AR" = "#7998c7")) + scale_color_manual(values = c("HS" = "#010305", "MM" = "#16468d", "AC" = "#5b92e5", "AR" = "#7998c7")) + theme(legend.position = "none") + theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(subset(`melted_feb_18th_working_file`, grepl(glob2rx("Metaphlan"), variable)), aes(x = Sample, y = value, color = Host_species, fill = Host_species)) + geom_bar(stat = "identity", position = position_dodge()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + ggtitle("Percent reads classified by Metaphlan per sample by host species") + scale_fill_manual(values = c("HS" = "#010305", "MM" = "#16468d", "AC" = "#5b92e5", "AR" = "#9e9fa2")) + scale_color_manual(values = c("HS" = "#010305", "MM" = "#16468d", "AC" = "#5b92e5", "AR" = "#9e9fa2")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color = "Host species") + labs(fill = "Host species")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-18-All_samples_only_more_recent_downloaded_lab_mice_percent_reads_classified_by_Metaphlan.svg")

ggplot(subset(`melted_feb_18th_working_file`, grepl(glob2rx("Kraken*"), variable)), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales = "free_x") + ggtitle("Percent reads classified by Kraken using different confidence scores per sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))

ggplot(subset(`melted_feb_18th_working_file`, grepl(glob2rx("Kraken*"), variable | `melted_feb_18th_working_file`$Host_species != "HS")), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales = "free_x") + ggtitle("Percent reads classified by Kraken using different confidence scores per sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))

ggplot(subset(`melted_feb_18th_working_file`, grepl(glob2rx("Kraken*"), variable)), subset = `melted_feb_18th_working_file`$Host_species != "HS", aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales = "free_x") + ggtitle("Percent reads classified by Kraken using different confidence scores per sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))

ggplot(subset(`p`, grepl(paste(paired_AC_to_match,collapse="|"), `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & grepl("AC1", `melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp`$SampleID) & melted_log1p_transposed_kilobase_and_library_size_adjusted_mapping_counts_unique_original_cultures_contigs_over_500bp$value > 0.1)) + theme_linedraw() + geom_jitter(aes(x = SampleID, y = value, color = SampleID), width = 0.25, size = 1.25) + xlab("Samples") + ylab("Log1p transformed, mapped read segments") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color="Sample ID") + labs(title = "For contigs mapped with adjusted count > 0.1 in selected Acomys cahirinus samples")

subset(`melted_feb_18th_working_file`, `melted_feb_18th_working_file`$Host_species != "HS")

`melted_feb_18th_working_file_excluding_human` <- subset(`melted_feb_18th_working_file`, `melted_feb_18th_working_file`$Host_species != "HS")

ggplot(subset(`melted_feb_18th_working_file_excluding_human`, grepl(glob2rx("Kraken*"), variable)), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales = "free_x") + ggtitle("Percent reads classified by Kraken using different confidence scores per rodent sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-18-All_RODENT_samples_only_more_recent_downloaded_lab_mice_percent_reads_classified_by_Kraken.svg")

ggplot(subset(`melted_feb_18th_working_file`, grepl(glob2rx("Kaiju*"), variable)), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + scale_fill_discrete(limits = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed")) + scale_color_discrete(limits = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed"))

ggplot(subset(`melted_feb_18th_working_file_excluding_human`, grepl(glob2rx("Kaiju*"), variable)), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + scale_fill_discrete(limits = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed")) + scale_color_discrete(limits = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed"))

`melted_feb_18th_working_file_excluding_human_kaiju_only` <- subset(`melted_feb_18th_working_file_excluding_human`, grepl(glob2rx("Kaiju*"), variable))

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors"))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-18-All_RODENT_samples_only_more_recent_downloaded_lab_mice_percent_reads_classified_by_Kaiju.svg")

`melted_feb_18th_working_file_excluding_human_kaiju_only`$variable <- factor(`melted_feb_18th_working_file_excluding_human_kaiju_only`$variable, levels = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed"))

`melted_feb_18th_working_file_ONLY_human` <- subset(`melted_feb_18th_working_file`, `melted_feb_18th_working_file`$Host_species == "HS")

`melted_feb_18th_working_file_ONLY_human`$variable <- factor(`melted_feb_18th_working_file_ONLY_human`$variable, levels = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed", "Kraken 95 % confidence", "Kraken 75 % confidence", "Kraken 66 % confidence", "Kraken 50 % confidence", "Kraken 25 % confidence", "Kraken 10 % confidence", "Metaphlan"))

ggplot(subset(`melted_feb_18th_working_file_ONLY_human`, `melted_feb_18th_working_file_ONLY_human`$variable != "Metaphlan"), aes(x = variable, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Tool and parameter") + ggtitle("Percent reads classified by Kaiju and Kraken using different parameters in human mock community") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5", "Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5", "Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter"))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-18-Mock_human_sample_only_percent_reads_classified_by_Kaiju_and_kraken_all_parameters.svg")

ggplot(subset(`melted_feb_18th_working_file_ONLY_human`, `melted_feb_18th_working_file_ONLY_human`$variable != "Metaphlan"), aes(x = variable, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Tool and parameter") + ggtitle("Percent reads classified by Kaiju and Kraken using different parameters in human mock community") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5", "Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5", "Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Tool and parameter"), color = guide_legend(title = "Tool and parameter")) + theme(legend.position = "none")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-18-Mock_human_sample_only_percent_reads_classified_by_Kaiju_and_kraken_all_parameters-No_legend.svg")

2022-02-21

ggplot(subset(`melted_feb_18th_working_file`, grepl(glob2rx("Metaphlan"), variable)), aes(x = Sample, y = value, color = Host_species, fill = Host_species)) + geom_bar(stat = "identity", position = position_dodge()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + ggtitle("Percent reads classified by Metaphlan per sample by host species") + scale_fill_manual(values = c("HS" = "#050505", "MM" = "#2d2d2d", "AC" = "#858585", "AR" = "#c9c9c9")) + scale_color_manual(values = c("HS" = "#050505", "MM" = "#2d2d2d", "AC" = "#858585", "AR" = "#c9c9c9")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color = "Host species") + labs(fill = "Host species")
2022-02-18-All_samples_only_more_recent_downloaded_lab_mice_percent_reads_classified_by_Metaphlan
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-18-All_samples_only_more_recent_downloaded_lab_mice_percent_reads_classified_by_Metaphlan.svg")


ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Sample, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors"))

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Host_species, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + theme(strip.background = element_blank(), strip.text.x = element_blank()) + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors"))

`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species <- gsub("AC", "Acomys cahirinus", as.character(`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species))
`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species <- gsub("AR", "Acomys russatus", as.character(`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species))
`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species <- gsub("MM", "Mus musculus", as.character(`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species))

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Host_species, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic"))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-21-All_rodent_samples_only_more_recent_downloaded_lab_mice_percent_reads_classified_by_Kaiju_grouped_facets_by_host_species.svg")

`melted_feb_18th_working_file_excluding_human`$Host_species <- gsub("AC", "Acomys cahirinus", as.character(`melted_feb_18th_working_file_excluding_human`$Host_species))
`melted_feb_18th_working_file_excluding_human`$Host_species <- gsub("AR", "Acomys russatus", as.character(`melted_feb_18th_working_file_excluding_human`$Host_species))
`melted_feb_18th_working_file_excluding_human`$Host_species <- gsub("MM", "Mus musculus", as.character(`melted_feb_18th_working_file_excluding_human`$Host_species))

ggplot(subset(`melted_feb_18th_working_file_excluding_human`, grepl(glob2rx("Kraken*"), variable)), aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Host_species, scales = "free_x") + ggtitle("Percent reads classified by Kraken using different confidence scores per rodent sample") + theme(strip.text.x = element_text(face = "italic")) + scale_fill_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-21-All_rodent_samples_only_more_recent_downloaded_lab_mice_percent_reads_classified_by_Kraken_grouped_facets_by_host_species.svg")

2022-02-23
ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Sample, y = value, color = variable, fill = variable)) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Sample ID") + facet_wrap( ~ Host_species, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic"))

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Sample, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median)

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Sample, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median) + geom_bar(stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic"))

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median, geom="bar") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic"))

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median, geom="bar") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median, geom="errorbar")

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Sample, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Sample, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median, geom="bar") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + stat_summary(mapping = aes(x = Sample, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median, geom="errorbar")

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median, geom="bar") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median, geom="errorbar")

posdoge <- position_dodge(0.3)

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species")

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median, position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x") + ggtitle("Percent reads classified by Kaiju using different number of allowed errors per rodent sample") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species")

ggplot(subset(`melted_feb_18th_working_file_excluding_human`, grepl(glob2rx("Kraken*"), variable)), aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median, position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Host species") + facet_wrap( ~ Host_species, scales = "free_x") + ggtitle("Percent reads classified by Kraken using different confidence scores per rodent sample") + scale_fill_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score")) + theme(strip.text.x = element_text(face = "italic"))

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median, position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank())

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-23-All_rodent_samples_only_more_recent_downloaded_lab_mice_median_percent_reads_classified_by_Kaiju_grouped_facets_by_host_species_with_95_percent_confidence_intervals.svg")

ggplot(subset(`melted_feb_18th_working_file_excluding_human`, grepl(glob2rx("Kraken*"), variable)), aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.95) }, fun.max= function(z) { quantile(z,1.05) }, fun = median, position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Host species") + facet_wrap( ~ Host_species, scales = "free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kraken using different confidence scores from all samples by host species") + scale_fill_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank())

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-23-All_rodent_samples_only_more_recent_downloaded_lab_mice_median_percent_reads_classified_by_Kraken_grouped_facets_by_host_species_with_95_percent_confidence_intervals.svg")

quantile_cl <- function(y, q=0.5, conf.level = 0.95, na.rm=TRUE) {
  alpha <- 1 - conf.level
  if (na.rm) y <- y[!is.na(y)]
  n <- length(y)
  l <- qbinom(alpha/2, size=n, prob = q)
  u <- 1 + n - l
  ys <- sort.int(c(-Inf, y, Inf), partial = c(1 + l, 1 + u))
  data.frame(
    y = quantile(y, probs = q, na.rm=na.rm, type = 8),
    ymin = ys[1 + l],
    ymax = ys[1 + u]
  )
}

median_cl <- function(y, conf.level=0.95, na.rm=TRUE) quantile_cl(y, q=0.5, conf.level=conf.level, na.rm=na.rm)

ggplot(subset(`melted_feb_18th_working_file_excluding_human`, grepl(glob2rx("Kraken*"), variable)), aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.data=median_cl, position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + xlab("Host species") + facet_wrap( ~ Host_species, scales = "free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kraken using different confidence scores from all samples by host species") + scale_fill_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank())

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.data=median_cl, position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank()) + stat_summary(fun.data = mean_cl_normal, geom="crossbar", aes(x = Host_species, y = value, color = variable, fill = variable), position=position_dodge(width=1.25))

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.05) }, fun.max= function(z) { quantile(z,0.05) }, fun = median, position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank())

`melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(variable) %>% dplyr::summarise(GroupMedianValue = median(value), .groups = 'drop')

`melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(variable) %>% dplyr::summarise(GroupMedianValue = median(value), .groups = 'keep')

`2022_02_23_dplyr_medians_and_confidence_scores_testing_rodent_only_samples_kaiju_and_kraken_percent_classified_reads` <- `melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(GroupMedianValue = median(value), .groups = 'keep')

DescTools::MedianCI(`melted_feb_18th_working_file_excluding_human_kaiju_only`$value, method = "exact", conf.level = 0.95)

rm(`2022_02_23_dplyr_medians_and_confidence_scores_testing_rodent_only_samples_kaiju_and_kraken_percent_classified_reads`)

`2022_02_23_rodent_only_kaiju_working_file` <- `melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, method =c("exact"))[1], lower_confidence_interval = MedianCI(value, method =c("exact"))[2], upper_confidence_interval = MedianCI(value, method =c("exact"))[3])

`2022_02_23_rodent_only_kaiju_working_file` <- `melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[3])

rm(`2022_02_23_rodent_only_kaiju_working_file`)

DescTools::MedianCI(`melted_feb_18th_working_file_excluding_human_kaiju_only`$value, method = "exact", conf.level = 0.95)

`melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[3])

`melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("exact"))[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("exact"))[3])

`2022_02_23_rodent_only_kaiju_working_file`

ggplot(`melted_feb_18th_working_file_excluding_human_kaiju_only`, aes(x = Host_species, y = value, color = variable, fill = variable)) + stat_summary(mapping = aes(x = Host_species, y = value, color = variable, fill = variable), fun.min= function(z) { quantile(z,0.25) }, fun.max= function(z) { quantile(z,0.75) }, fun = median, position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank())

ggplot(`2022_02_23_rodent_only_kaiju_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=0.75)) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank())
scale_y_continuous(oob=scales::discard)

ggplot(`2022_02_23_rodent_only_kaiju_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=0.75)) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank()) + scale_y_continuous(oob=scales::discard)

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-23-All_rodent_samples_only_more_recent_downloaded_lab_mice_median_percent_reads_classified_by_Kaiju_grouped_facets_by_host_species_with_95_percent_confidence_intervals.svg")

`melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.75, sides = c("two.sided"), method =c("exact"))[1], lower_confidence_interval = MedianCI(value, conf.level = 0.75, method =c("exact"))[2], upper_confidence_interval = MedianCI(value, conf.level = 0.75, method =c("exact"))[3]) %>% View()

`2022_02_23_rodent_only_kaiju_working_file` <- `melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("exact"))[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("exact"))[3])

`2022_02_23_rodent_only_kaiju_working_file` <- `melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("boot"), R = 999)[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("boot"), R = 999)[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("boot"), R = 999)[3])

`2022_02_23_rodent_only_kraken_working_file` <- subset(`melted_feb_18th_working_file_excluding_human`, grepl(glob2rx("Kraken*"), variable) %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("boot"), R = 999)[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("boot"), R = 999)[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("boot"), R = 999)[3])

ggplot(`2022_02_23_rodent_only_kraken_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=0.75)) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kraken using different classification confidence scores from all samples by host species") + scale_fill_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + scale_color_manual(values = c("Kraken 95 % confidence" = "#000001", "Kraken 75 % confidence" = "#0a2d45", "Kraken 66 % confidence" = "#124e78", "Kraken 50 % confidence" = "#1c7abc", "Kraken 25 % confidence" = "#3399e0", "Kraken 10 % confidence" = "#88c4ed")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))+ guides(fill = guide_legend(title = "Kraken classification confidence score"), color = guide_legend(title = "Kraken classification confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank()) + scale_y_continuous(oob=scales::discard)

`2022_02_23_rodent_only_kraken_working_file`$variable <- factor(`2022_02_23_rodent_only_kraken_working_file`$variable, levels = c("Kraken 10 % confidence", "Kraken 25 % confidence", "Kraken 50 % confidence", "Kraken 66 % confidence", "Kraken 75 % confidence", "Kraken 95 % confidence"))

ggplot(`2022_02_23_rodent_only_kraken_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=0.75)) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=0.75)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kraken using different classification confidence scores\nfrom all samples by host species") + scale_fill_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + scale_color_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))+ guides(fill = guide_legend(title = "Kraken classification confidence score"), color = guide_legend(title = "Kraken classification confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank()) + scale_y_continuous(oob=scales::discard)

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-23-All_rodent_samples_only_more_recent_downloaded_lab_mice_median_percent_reads_classified_by_Kraken_grouped_facets_by_host_species_with_95_percent_confidence_intervals.svg")

2022-02-24
ggplot(`2022_02_23_rodent_only_kaiju_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank()) + scale_y_continuous(oob=scales::discard)

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-23-All_rodent_samples_only_more_recent_downloaded_lab_mice_median_percent_reads_classified_by_Kaiju_grouped_facets_by_host_species_with_95_percent_confidence_intervals.svg")

ggplot(`2022_02_23_rodent_only_kraken_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kraken using different classification confidence scores\nfrom all samples by host species") + scale_fill_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + scale_color_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))+ guides(fill = guide_legend(title = "Kraken classification confidence score"), color = guide_legend(title = "Kraken classification confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank()) + scale_y_continuous(oob=scales::discard)

ggplot(`2022.02.24_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`) + geom_point(aes(x = Kaiju_allowed_errors, y = Species_level_below_hits),color = "#9b3b15", fill = "#9b3b15") + geom_line(color = "#9b3b15", fill = "#9b3b15") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylim(0, 5000) + geom_line(`2022.02.24_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`, aes(x = Kaiju_allowed_errors, y = Genus_level_below_hits), color = "#f5c7b5") + ylabs("Number of taxids")

ggplot(`2022.02.24_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`) + geom_point(aes(x = Kaiju_allowed_errors, y = Species_level_below_hits),color = "#9b3b15", fill = "#9b3b15") + geom_point(aes(x = Kaiju_allowed_errors, y = Genus_level_below_hits),color = "#f5c7b5", fill = "#f5c7b5") + geom_line(aes(x = Kaiju_allowed_errors, y = Species_level_below_hits),color = "#9b3b15", fill = "#9b3b15") + geom_line(aes(x = Kaiju_allowed_errors, y = Genus_level_below_hits),color = "#f5c7b5", fill = "#f5c7b5") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylim(0, 5000) + ylab("Number of taxids detected at taxonomic level threshold")

`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds` <- reshape2::melt(`2022.02.24_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`, id=as.factor("Kaiju_allowed_errors"), measure.vars=c("Species_exact_true_positive", "Unique_taxids_identified", "Species_level_below_hits", "Genus_level_below_hits"))

ggplot(`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`, aes(x = Kaiju_allowed_errors, y = value, colour = variable, fill = variable, shape = variable)) + geom_point() + geom_line()

`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`$variable <- factor(`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`$variable, levels = c("Unique_taxids_identified", "Genus_level_below_hits", "Species_level_below_hits", "Species_exact_true_positive"))

`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`$variable <- gsub("Species_exact_true_positive", "Exact known member species taxID", as.character(`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`$variable))
`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`$variable <- gsub("Unique_taxids_identified", "Unique taxids detected", as.character(`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`$variable))
`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`$variable <- gsub("Species_level_below_hits", "Species and lower taxon taxids", as.character(`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`$variable))
`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`$variable <- gsub("Genus_level_below_hits", "Genera and lower taxon taxids", as.character(`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`$variable))

ggplot(`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`, aes(x = Kaiju_allowed_errors, y = value, colour = variable, fill = variable, shape = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#9b3b15", "Genera and lower taxon taxids" = "#e67549", "Unique taxids detected" = "#848482")) + scale_color_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#9b3b15", "Genera and lower taxon taxids" = "#e67549", "Unique taxids detected" = "#848482")) + scale_shape_manual(name = "Kaiju allowed errors", labels = c("Exact known member species taxID", "Species and lower taxon taxids", "Genera and lower taxon taxids", "Unique taxids detected"), values = c ("15", "17", "17", "15")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors"), shape = guide_legend(title = "Kaiju allowed errors"))

ggplot(`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`, aes(x = Kaiju_allowed_errors, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#9b3b15", "Genera and lower taxon taxids" = "#e67549", "Unique taxids detected" = "#848482")) + scale_color_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#9b3b15", "Genera and lower taxon taxids" = "#e67549", "Unique taxids detected" = "#848482")) + guides(fill = guide_legend(title = "Known taxid threshold"), color = guide_legend(title = "Known taxid threshold")) + ylab("Number of taxids") + xlab("Number of allowed errors in Kaiju match") + labs(title = "Number of taxids detected by Kaiju at given errors level for specific threshold in mock\nhuman microbiota")

`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds` <- reshape2::melt(`2022.02.24_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`, id=as.factor("Kraken_percent_confidence_score"), measure.vars=c("Species_exact_true_positive", "Unique_taxids_identified", "Species_level_below_hits", "Genus_level_below_hits"))

`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`$variable <- factor(`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`$variable, levels = c("Unique_taxids_identified", "Genus_level_below_hits", "Species_level_below_hits", "Species_exact_true_positive"))

`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`$variable <- gsub("Species_exact_true_positive", "Exact known member species taxID", as.character(`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`$variable))
`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`$variable <- gsub("Unique_taxids_identified", "Unique taxids detected", as.character(`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`$variable))
`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`$variable <- gsub("Species_level_below_hits", "Species and lower taxon taxids", as.character(`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`$variable))
`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`$variable <- gsub("Genus_level_below_hits", "Genera and lower taxon taxids", as.character(`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`$variable))

ggplot(`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`, aes(x = Kraken_percent_confidence_score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#1c7abc", "Genera and lower taxon taxids" = "#88c4ed", "Unique taxids detected" = "#848482")) + scale_color_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#1c7abc", "Genera and lower taxon taxids" = "#88c4ed", "Unique taxids detected" = "#848482")) + guides(fill = guide_legend(title = "Known taxid threshold"), color = guide_legend(title = "Known taxid threshold")) + ylab("Number of taxids") + xlab("Confidence score of Kraken classification (%)") + labs(title = "Number of taxids detected by Kraken at given required confidence levels for specific threshold in mock\nhuman microbiota")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-24-All_rodent_samples_only_more_recent_downloaded_lab_mice_median_percent_reads_classified_by_Kaiju_grouped_facets_by_host_species_with_95_percent_confidence_intervals.svg")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-24-All_rodent_samples_only_more_recent_downloaded_lab_mice_median_percent_reads_classified_by_Kraken_grouped_facets_by_host_species_with_95_percent_confidence_intervals.svg")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-24-human_mock_taxid_based_kaiju_number_exact_species_species_and_lower_genus_and_lower_and_all_unique_taxids.svg")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-24-human_mock_taxid_based_kraken_number_exact_species_species_and_lower_genus_and_lower_and_all_unique_taxids.svg")

2022-02-28
ggplot(`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`, aes(x = Kraken_percent_confidence_score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#1c7abc", "Genera and lower taxon taxids" = "#88c4ed", "Unique taxids detected" = "#848482")) + scale_color_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#1c7abc", "Genera and lower taxon taxids" = "#88c4ed", "Unique taxids detected" = "#848482")) + guides(fill = guide_legend(title = "Known taxid threshold"), color = guide_legend(title = "Known taxid threshold")) + ylab("Number of taxids") + xlab("Confidence score of Kraken classification (%)") + labs(title = "Number of taxids detected by Kraken at given required confidence levels for specific threshold in mock\nhuman microbiota") + scale_y_continuous(breaks=pretty_breaks(n=12))

ggplot(`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`, aes(x = Kaiju_allowed_errors, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#9b3b15", "Genera and lower taxon taxids" = "#e67549", "Unique taxids detected" = "#848482")) + scale_color_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#9b3b15", "Genera and lower taxon taxids" = "#e67549", "Unique taxids detected" = "#848482")) + guides(fill = guide_legend(title = "Known taxid threshold"), color = guide_legend(title = "Known taxid threshold")) + ylab("Number of taxids") + xlab("Number of allowed errors in Kaiju match") + labs(title = "Number of taxids detected by Kaiju at given errors level for specific threshold in mock\nhuman microbiota") + scale_y_continuous(breaks=pretty_breaks(n=10))

ggplot(`2022_02_23_rodent_only_kaiju_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank()) + scale_y_continuous(oob=scales::discard) + scale_y_continuous(limits = c(0,100))

ggplot(`2022_02_23_rodent_only_kraken_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kraken using different classification confidence scores\nfrom all samples by host species") + scale_fill_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + scale_color_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))+ guides(fill = guide_legend(title = "Kraken classification confidence score"), color = guide_legend(title = "Kraken classification confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank()) + scale_y_continuous(oob=scales::discard) + scale_y_continuous(limits = c(0,100))

ggplot(`2022_02_23_rodent_only_kaiju_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#9b3b15", "Kaiju 1 error allowed" = "#b14318", "Kaiju 2 errors allowed" = "#c84c1b", "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#e67549", "Kaiju 10 errors allowed" = "#e9865f", "Kaiju 20 errors allowed" = "#f5c7b5")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank()) + scale_y_continuous(oob=scales::discard)

ggplot(`2022_02_23_rodent_only_kaiju_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank()) + scale_y_continuous(oob=scales::discard)

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-24-All_rodent_samples_only_more_recent_downloaded_lab_mice_median_percent_reads_classified_by_Kaiju_grouped_facets_by_host_species_with_95_percent_confidence_intervals.svg")

ggplot(`2022.02.24_Melted_Kraken_only_human_mock_true_taxids_species_and_genus_thresholds`, aes(x = Kraken_percent_confidence_score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#1c7abc", "Genera and lower taxon taxids" = "#88c4ed", "Unique taxids detected" = "#848482")) + scale_color_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#1c7abc", "Genera and lower taxon taxids" = "#88c4ed", "Unique taxids detected" = "#848482")) + guides(fill = guide_legend(title = "Known taxid threshold"), color = guide_legend(title = "Known taxid threshold")) + ylab("Number of taxids") + xlab("Confidence score of Kraken classification (%)") + labs(title = "Number of taxids detected by Kraken at given required confidence levels for specific threshold in mock\nhuman microbiota") + scale_y_continuous(breaks=pretty_breaks(n=12))

ggplot(`2022.02.24_Melted_Kaiju_only_human_mock_true_taxids_species_and_genus_thresholds`, aes(x = Kaiju_allowed_errors, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#9b3b15", "Genera and lower taxon taxids" = "#e67549", "Unique taxids detected" = "#848482")) + scale_color_manual(values = c("Exact known member species taxID" = "#0f0f0f", "Species and lower taxon taxids" = "#9b3b15", "Genera and lower taxon taxids" = "#e67549", "Unique taxids detected" = "#848482")) + guides(fill = guide_legend(title = "Known taxid threshold"), color = guide_legend(title = "Known taxid threshold")) + ylab("Number of taxids") + xlab("Number of allowed errors in Kaiju match") + labs(title = "Number of taxids detected by Kaiju at given errors level for specific threshold in mock\nhuman microbiota") + scale_y_continuous(breaks=pretty_breaks(n=10))

2022-03-07

ggplot(`2022.03.06_Kraken_human_mock_detailed_taxID_known_members_breakdown`, aes(x = Confidence.score, y = Unique.taxIDs, colour = "#000001", fill = "#000001")) + geom_line() + geom_point() + scale_y_continuous(limits = c(0,300)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_line(aes(x = Confidence.score, y = TaxIDs.with.more.than.100.reads, color = "#88c4ed" , fill = "#88c4ed")) + geom_point(aes(x = Confidence.score, y = TaxIDs.with.more.than.100.reads, color = "#88c4ed" , fill = "#88c4ed"))

ggplot(`2022_02_23_rodent_only_kraken_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kraken using different classification confidence scores\nfrom all samples by host species") + scale_fill_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + scale_color_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))+ guides(fill = guide_legend(title = "Kraken classification confidence score"), color = guide_legend(title = "Kraken classification confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank()) + scale_y_continuous(oob=scales::discard) + scale_y_continuous(limits = c(0,100))

ggplot(`2022.03.06_Kraken_human_mock_detailed_taxID_known_members_breakdown`, aes(x = Confidence.score, y = "TaxIDs")) + geom_line(aes(x = Confidence.score, y = Unique.taxIDs, color = "#000001", fill = "#000001")) + geom_point(aes(x = Confidence.score, y = Unique.taxIDs, color = "#000001", fill = "#000001")) + scale_y_continuous(limits = c(0,300)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_line(aes(x = Confidence.score, y = TaxIDs.with.more.than.100.reads, color = "#88c4ed" , fill = "#88c4ed")) + geom_point(aes(x = Confidence.score, y = TaxIDs.with.more.than.100.reads, color = "#88c4ed" , fill = "#88c4ed"))

`2022_03_07_melted_krak_hum_mock_detailed` <- reshape2::melt(`2022.03.06_Kraken_human_mock_detailed_taxID_known_members_breakdown`, id=as.factor("Confidence.score"), measure.vars=c("Unique.taxIDs", "TaxIDs.with.more.than.100.reads", "Exact.strain.occurrences", "Exact.species.only.taxID.occurrences", "Fuzzy.false.positive.taxID.occurrences", "Genus.false.positive.taxID.occurrences", "Exact.strain.reads.classified", "Exact.species.only.reads.classified", "Fuzzy.false.positive.reads.classified", "Genus.false.positive.reads.classified", "None.same.genus.taxids.detected", "None.same.genus.reads.classified"))

`2022_03_07_melted_krak_hum_mock_detailed`$variable <- gsub("[.]", " ", as.character(`2022_03_07_melted_krak_hum_mock_detailed`$variable))

`2022_03_07_melted_krak_hum_mock_detailed`

ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, "Unique taxIDs" == variable | "TaxIDs with more than 100 reads" == variable), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, "Unique taxIDs" == variable | "TaxIDs with more than 100 reads" == variable), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Confidence score (%)") + ylab("TaxIDs") + ggtitle("Number unique taxIDs detected in total and those with more than 100 reads assigned by Kraken\nat multiple confidence scores") + scale_fill_manual(values = c("TaxIDs with more than 100 reads" = "#000001", "Unique taxIDs" = "#88c4ed")) + scale_color_manual(values = c("TaxIDs with more than 100 reads" = "#000001", "Unique taxIDs" = "#88c4ed")) + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10))

ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, "Exact strain occurrences" == variable | "Exact species only taxID occurrences" == variable), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Confidence score (%)") + ylab("TaxIDs") + ggtitle("Number Exact strain occurrences detected in total and those with more than 100 reads assigned by Kraken\nat multiple confidence scores") + scale_fill_manual(values = c("Exact species only taxID occurrences" = "#3399e0", "Exact strain occurrences" = "#0a2d45")) + scale_color_manual(values = c("Exact species only taxID occurrences" = "#3399e0", "Exact strain occurrences" = "#0a2d45")) + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10))

ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, "Fuzzy false positive taxID occurrences" == variable | "Genus false positive taxID occurrences" == variable | "None same genus taxids detected" == variable), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Confidence score (%)") + ylab("TaxIDs") + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10)) + scale_color_manual(values = c("Genus false positive taxID occurrences" = "#124e78", "Fuzzy false positive taxID occurrences" = "#1c7abc", "None same genus taxids detected" = "#6e7f80")) + scale_fill_manual(values = c("Genus false positive taxID occurrences" = "#124e78", "Fuzzy false positive taxID occurrences" = "#1c7abc", "None same genus taxids detected" = "#6e7f80")) + ggtitle("Number of false positive taxids detected by Kraken at multiple confidence scores")

ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, grepl(glob2rx("*reads classified"), variable)), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(labels = comma) + scale_x_continuous(breaks=pretty_breaks(n=10)) + theme(legend.title=element_blank()) + ggtitle("Number of reads assigned by Kraken at different confidence scores by relationship to known members") + scale_color_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_fill_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_y_continuous(trans = "log10") + xlab("Confidence score (%)") + ylab("Number of reads")

plp1 <- ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, "Unique taxIDs" == variable | "TaxIDs with more than 100 reads" == variable), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Confidence score (%)") + ylab("TaxIDs") + scale_fill_manual(values = c("TaxIDs with more than 100 reads" = "#000001", "Unique taxIDs" = "#88c4ed")) + scale_color_manual(values = c("TaxIDs with more than 100 reads" = "#000001", "Unique taxIDs" = "#88c4ed")) + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10))
plp2 <- ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, "Exact strain occurrences" == variable | "Exact species only taxID occurrences" == variable), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Confidence score (%)") + ylab("TaxIDs") + scale_fill_manual(values = c("Exact species only taxID occurrences" = "#3399e0", "Exact strain occurrences" = "#0a2d45")) + scale_color_manual(values = c("Exact species only taxID occurrences" = "#3399e0", "Exact strain occurrences" = "#0a2d45")) + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10))
plp3 <- ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, "Fuzzy false positive taxID occurrences" == variable | "Genus false positive taxID occurrences" == variable | "None same genus taxids detected" == variable), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Confidence score (%)") + ylab("TaxIDs") + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10)) + scale_color_manual(values = c("Genus false positive taxID occurrences" = "#124e78", "Fuzzy false positive taxID occurrences" = "#1c7abc", "None same genus taxids detected" = "#6e7f80")) + scale_fill_manual(values = c("Genus false positive taxID occurrences" = "#124e78", "Fuzzy false positive taxID occurrences" = "#1c7abc", "None same genus taxids detected" = "#6e7f80"))
plp4 <- ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, grepl(glob2rx("*reads classified"), variable)), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(labels = comma) + scale_x_continuous(breaks=pretty_breaks(n=10)) + theme(legend.title=element_blank()) + scale_color_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_fill_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_y_continuous(trans = "log10") + xlab("Confidence score (%)") + ylab("Number of reads")

ggarrange(plp1, plp2, plp3, plp4, labels = c("A", "B", "C", "D"), ncol = 2, nrow = 2)

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-03-07-Human_mock_Kraken_detailed_breakdown.svg")

2022-03-16
plp1 <- ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, "Unique taxIDs" == variable | "TaxIDs with more than 100 reads" == variable), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Confidence score (%)") + ylab("TaxIDs") + scale_fill_manual(values = c("TaxIDs with more than 100 reads" = "#000001", "Unique taxIDs" = "#88c4ed")) + scale_color_manual(values = c("TaxIDs with more than 100 reads" = "#000001", "Unique taxIDs" = "#88c4ed")) + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10)) + scale_y_continuous(limits = c(0, 300), breaks=pretty_breaks(n=20)) + geom_hline(yintercept = 20, linetype = "dashed", color = "#ed872d", size = 1) + theme(legend.position =c(0.8,0.9))
plp2 <- ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, "Exact strain occurrences" == variable | "Exact species only taxID occurrences" == variable), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point(shape = 15, size = 2.5) + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Confidence score (%)") + ylab("TaxIDs") + scale_fill_manual(values = c("Exact species only taxID occurrences" = "#3399e0", "Exact strain occurrences" = "#0a2d45")) + scale_color_manual(values = c("Exact species only taxID occurrences" = "#3399e0", "Exact strain occurrences" = "#0a2d45")) + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10)) + scale_y_continuous(limits = c(0, 20), breaks=pretty_breaks(n=5)) + geom_hline(yintercept = 20, linetype = "dashed", color = "#ed872d", size = 1) + theme(legend.position="top") + theme(legend.position =c(0.8,0.6))
plp3 <- ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, "Fuzzy false positive taxID occurrences" == variable | "Genus false positive taxID occurrences" == variable | "None same genus taxids detected" == variable), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point(shape = 17, size = 2.5) + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Confidence score (%)") + ylab("TaxIDs") + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10)) + scale_color_manual(values = c("Genus false positive taxID occurrences" = "#124e78", "Fuzzy false positive taxID occurrences" = "#1c7abc", "None same genus taxids detected" = "#6e7f80")) + scale_fill_manual(values = c("Genus false positive taxID occurrences" = "#124e78", "Fuzzy false positive taxID occurrences" = "#1c7abc", "None same genus taxids detected" = "#6e7f80")) + scale_y_continuous(limits = c(0, 300), breaks=pretty_breaks(n=20)) + theme(legend.position =c(0.8,0.9))

plp1 + plp2 + plp3 + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') + plot_annotation(title = 'Kraken 2 analysis of human mock microbiota at different confidence levels')

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-03-16-Human_mock_Kraken_detailed_breakdown.svg")

ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, grepl(glob2rx("*reads classified"), variable)), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(labels = comma) + scale_x_continuous(breaks=pretty_breaks(n=10)) + theme(legend.title=element_blank()) + scale_color_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_fill_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_y_continuous(trans = "log10", labels = comma) + xlab("Confidence score (%)") + ylab("Number of reads") + ggtitle("Number of reads classified to taxids of different type by Kraken 2 at different confidence scores")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-03-16-human_mock_kraken2_read_counts_different_taxid_groups.svg")

2022-03-20
`2022_02_03_map_to_reads_lib_contig_adjusted_RAW` <- `2022.03.20_combined_per_read_library_and_contig_size_adjusted_mapping_scores_all_samples_mapping_to_concatenated_CQ_and_SR_bins`
read_names_and_variables <- `2022_02_03_map_to_reads_lib_contig_adjusted_RAW`$Read
`2022_02_03_map_to_reads_lib_contig_adjusted_RAW_TRANSPOSED` <- as.data.frame(t(`2022_02_03_map_to_reads_lib_contig_adjusted_RAW`[,-1]))
colnames(`2022_02_03_map_to_reads_lib_contig_adjusted_RAW_TRANSPOSED`) <- read_names_and_variables

2022-03-23
read.csv("2022-03-23_Transposed-contig_and_library_size_adjusted_summed_to_Kaiju_taxids_mapping_counts_all_samples.csv", header = TRUE, check.names = FALSE)

2022_03_21-Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid` <- read.csv("2022-03-23_Transposed-contig_and_library_size_adjusted_summed_to_Kaiju_taxids_mapping_counts_all_samples.csv", header = TRUE, check.names = FALSE)

log1p(`2022_03_21-Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$5)

`2022_03_21_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid` <- log1p(`2022_03_21-Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`[2:6775])

`2022_03_21_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Species <- `2022_03_21-Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Species
`2022_03_21_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Season <- `2022_03_21-Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Season
`2022_03_21_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Sequencing <- `2022_03_21-Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Sequencing
`2022_03_21_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Species_and_season <- `2022_03_21-Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Species_x_season
`2022_03_21_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Sample_ID <- `2022_03_21-Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Kaiju_taxids

`2022_03_21_PCA_Melted_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid` <- prcomp(`2022_03_21_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`[,c(1:6774)], center = TRUE, scale. = TRUE)

ggbiplot(`2022_03_21_PCA_Melted_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`, ellipse=TRUE, obs.scale = 1, var.scale = 1, var.axes=FALSE, circle = TRUE) + theme_classic()
ggbiplot(`2022_03_21_PCA_Melted_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`, labels=`2022_03_21_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Sample_ID, ellipse=TRUE, obs.scale = 1, var.scale = 1, var.axes=FALSE, circle = TRUE, alpha = 1, ) + theme_classic()

ggbiplot(`2022_03_21_PCA_Melted_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022_03_21_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022_03_21_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Season), color=factor(`2022_03_21_log1p_transformed_Transposed_all_samples_contig_size_and_library_sized_adjusted_mapped_read_counts_summed_to_Kaiju_taxid`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff4f00')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species")

2022-03-23_contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples
`2022.03.23_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples` <- read.csv("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-03-23_Transposed-contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples.csv")
`2022.03.23_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples` <- log1p(`2022.03.23_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`[2:504])
`2022.03.23_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Species <- `2022.03.23_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Species
`2022.03.23_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Season <- `2022.03.23_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Season
`2022.03.23_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Sequencing <- `2022.03.23_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Sequencing
`2022.03.23_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Species_and_season <- `2022.03.23_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Species_x_season
`2022.03.23_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Sample_ID <- `2022.03.23_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Bin_name

`2022.03.23_PCA_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples` <- prcomp(`2022.03.23_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`[,c(1:503)], center = TRUE, scale. = TRUE)

ggbiplot(`2022.03.23_PCA_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.03.23_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.03.23_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Season), color=factor(`2022.03.23_log1p_transformed_Transposed.contig_and_library_size_adjusted_summed_to_bins_mapping_counts_all_samples`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff4f00')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species")

2022-04-01
`2022.04.01_Kraken_human_mock_detailed_taxID_known_members_breakdown` <- read.csv("~/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-04-01_Kraken_human_mock_detailed_taxID_known_members_breakdown.csv")

`no_empty_rows_2022.04.01_Kraken_human_mock_detailed_taxID_known_members_breakdown` <- na.omit(`2022.04.01_Kraken_human_mock_detailed_taxID_known_members_breakdown`)

`melted_no_empty_rows_2022.04.01_Kraken_human_mock_detailed_taxID_known_members_breakdown` <- reshape2::melt(no_empty_rows_2022.04.01_Kraken_human_mock_detailed_taxID_known_members_breakdown, id=as.factor("Confidence.score"), measure.vars=c("Unique.taxIDs", "TaxIDs.with.more.than.100.reads", "Exact.strain.occurrences", "Exact.species.only.taxID.occurrences", "Fuzzy.false.positive.taxID.occurrences", "Genus.false.positive.taxID.occurrences", "Exact.strain.reads.classified", "Exact.species.only.reads.classified", "Fuzzy.false.positive.reads.classified", "Genus.false.positive.reads.classified", "None.same.genus.taxids.detected", "None.same.genus.reads.classified"))

`melted_no_empty_rows_2022.04.01_Kraken_human_mock_detailed_taxID_known_members_breakdown`$variable <- gsub("[.]", " ", as.character(`melted_no_empty_rows_2022.04.01_Kraken_human_mock_detailed_taxID_known_members_breakdown`$variable))

ggplot(subset(`melted_no_empty_rows_2022.04.01_Kraken_human_mock_detailed_taxID_known_members_breakdown`, grepl(glob2rx("*reads classified"), variable)), aes(x = Confidence.score, y = value, colour = variable, fill = variable, shape = variable)) + geom_point(size = 2.5) + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(labels = comma) + scale_x_continuous(breaks=pretty_breaks(n=10)) + theme(legend.title=element_blank()) + scale_color_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_fill_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_y_continuous(trans = "log10", labels = comma) + xlab("Confidence score (%)") + ylab("Number of reads") + ggtitle("Number of reads classified to taxids of different type by Kraken 2 at different confidence scores")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-04-01-human_mock_kraken2_read_counts_different_taxid_groups.svg")


2022-04-04
`2022.04.04_Kaiju_human_mock_detailed_taxID_known_members_breakdown` <- read.csv("~/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-04-04_Kaiju_human_mock_detailed_taxID_known_members_breakdown.csv")
`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown` <- reshape2::melt(`2022.04.04_Kaiju_human_mock_detailed_taxID_known_members_breakdown`, id=as.factor("Allowed.errors"), measure.vars=c("Unique.taxIDs", "TaxIDs.with.more.than.100.reads", "Exact.strain.occurrences", "Exact.species.only.taxID.occurrences", "Fuzzy.false.positive.taxID.occurrences", "Genus.false.positive.taxID.occurrences", "Exact.strain.reads.classified", "Exact.species.only.reads.classified", "Fuzzy.false.positive.reads.classified", "Genus.false.positive.reads.classified", "None.same.genus.taxids.detected", "None.same.genus.reads.classified"))

`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`$variable <- gsub("[.]", " ", as.character(`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`$variable))

plp1 <- ggplot(subset(`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`, "Unique taxIDs" == variable | "TaxIDs with more than 100 reads" == variable), aes(x = Allowed.errors, y = value, colour = variable, fill = variable, shape = variable)) + geom_point(size = 2.5) + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Allowed errors") + ylab("TaxIDs") + scale_fill_manual(values = c("TaxIDs with more than 100 reads" = "#000001", "Unique taxIDs" = "#f5c7b5")) + scale_color_manual(values = c("TaxIDs with more than 100 reads" = "#000001", "Unique taxIDs" = "#f5c7b5")) + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10)) + scale_y_continuous(limits = c(0, 15000), breaks=pretty_breaks(n=20)) + geom_hline(yintercept = 20, linetype = "dashed", color = "#1c7abc", size = 1) + theme(legend.position =c(0.8,0.9))

plp2 <- ggplot(subset(`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`, "Exact strain occurrences" == variable | "Exact species only taxID occurrences" == variable), aes(x = Allowed.errors, y = value, colour = variable, fill = variable, shape = variable)) + geom_point(size = 2.5) + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Allowed errors") + ylab("TaxIDs") + scale_fill_manual(values = c("Exact species only taxID occurrences" = "#e9865f", "Exact strain occurrences" = "#9b3b15")) + scale_color_manual(values = c("Exact species only taxID occurrences" = "#e9865f", "Exact strain occurrences" = "#9b3b15")) + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10)) + scale_y_continuous(limits = c(0, 20), breaks=pretty_breaks(n=10)) + geom_hline(yintercept = 20, linetype = "dashed", color = "#1c7abc", size = 1) + theme(legend.position="top") + theme(legend.position =c(0.8,0.6))

plp3 <- ggplot(subset(`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`, "Fuzzy false positive taxID occurrences" == variable | "Genus false positive taxID occurrences" == variable | "None same genus taxids detected" == variable), aes(x = Allowed.errors, y = value, colour = variable, fill = variable, shape = variable)) + geom_point(size = 2.5) + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Allowed errors") + ylab("TaxIDs") + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10)) + scale_shape_manual() + scale_color_manual(values = c("Genus false positive taxID occurrences" = "#9b3b15", "Fuzzy false positive taxID occurrences" = "#e67549", "None same genus taxids detected" = "#c84c1b")) + scale_fill_manual(values = c("Genus false positive taxID occurrences" = "#9b3b15", "Fuzzy false positive taxID occurrences" = "#e67549", "None same genus taxids detected" = "#c84c1b")) + scale_y_continuous(limits = c(0, 8000), breaks=pretty_breaks(n=20)) + theme(legend.position =c(0.8,0.8))

plp1 + plp2 + plp3 + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') + plot_annotation(title = 'Kaiju analysis of human mock microbiota with different number of allowed errors')

ggplot(`2022_02_23_rodent_only_kaiju_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c( "Kaiju 3 errors allowed" = "#de541e", "Kaiju 4 errors allowed" = "#e36432", "Kaiju 5 errors allowed" = "#b14318"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2023-01-07-human_mock_kaiju_some_detailed_bits.svg")


ggplot(subset(`2022_03_07_melted_krak_hum_mock_detailed`, grepl(glob2rx("*reads classified"), variable)), aes(x = Confidence.score, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(labels = comma) + scale_x_continuous(breaks=pretty_breaks(n=10)) + theme(legend.title=element_blank()) + scale_color_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_fill_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_y_continuous(trans = "log10", labels = comma) + xlab("Confidence score (%)") + ylab("Number of reads") + ggtitle("Number of reads classified to taxids of different type by Kraken 2 at different confidence scores")


`2022.04.04_Melted_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values` <- reshape2::melt(`2022.04.04_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, id=as.factor("Taxon"), measure.vars=c("Percentage_abundance", "Confidence_score"))
`2022.04.04_Melted_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values` <- reshape2::melt(`2022.04.04_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, id=as.factor("Confidence_score", "Taxon"), measure.vars=c("Percentage_abundance"))
ggplot(data = `2022.04.04_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, aes(x = Confidence_score, y = Percentage_abundance)) + geom_point()
ggplot(data = `2022.04.04_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, aes(x = Confidence_score, y = Percentage_abundance)) + geom_line()

`2022.04.04_Melted_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`

ggplot(data = `2022.04.04_Melted_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, aes(x = Confidence_score, y = value, color = Confidence_score)) + geom_point()

2022-04-06
ggplot(data = `2022.04.04_Melted_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, aes(x = value, y = Confidence_score, color = Confidence_score)) + geom_point()
`2022.04.04_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`
ggplot(data = `2022.04.04_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, aes(x = Confidence_score, y = Percentage_abundance)) + geom_point()

ggplot(data = `2022.04.04_Melted_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, aes(x = Confidence_score, y = value, group = Confidence_score)) + geom_boxplot() + geom_jitter(shape=16, position=position_jitter(0.2))

ggplot(data = `2022.04.04_Melted_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, aes(x = Confidence_score, y = value, group = Confidence_score)) + geom_violin() + geom_jitter(shape=16, position=position_jitter(0.2))

ggplot(data = `2022.04.04_Melted_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, aes(x = value, y = Confidence_score, color = Confidence_score, size = value)) + geom_point() + scale_size(range = c(.25, 25), name = "Percentage abundance")
ggplot(data = `2022.04.04_Melted_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, aes(x = Confidence_score, y = value, color = Confidence_score)) + geom_point() + theme_linedraw() + xlab("Confidence score (%)") + ylab("Abundance (%)") + scale_y_continuous(limits = c(0, 100), breaks=pretty_breaks(n=10)) + scale_x_continuous(limits = c(0, 100), breaks=pretty_breaks(n=10))

ggplot(data = `2022.04.04_Melted_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, aes(x = Confidence_score, y = value, color = Confidence_score)) + geom_jitter(shape=16, position=position_jitter(0.5)) + theme_linedraw() + xlab("Confidence score (%)") + ylab("Abundance (%)") + scale_y_continuous(limits = c(0, 100), breaks=pretty_breaks(n=10)) + scale_x_continuous(limits = c(0, 100), breaks=pretty_breaks(n=10))

ggplot(data = `2022.04.04_Melted_Kraken_human_mock_report_file_outputs_percent_abundance_to_the_taxa_at_confidence_values`, aes(x = Confidence_score, y = value, color = Confidence_score)) + geom_jitter(shape=16, position=position_jitter(0.05)) + theme_linedraw() + xlab("Confidence score (%)") + ylab("Abundance (%)") + scale_y_continuous(limits = c(0, 100), breaks=pretty_breaks(n=10))

font_add(family = "Tex Gyre Heroes", regular = "/Users/osbourne/Google Drive/Fonts/TeX-Gyre-Heros/texgyreheros-regular.otf")
showtext_auto()
font_add(family = "Hasegawa font", regular = "/Users/osbourne/Google Drive/Fonts/Hasegawa_font/Hasegawa-Regular.otf")

ggplot(subset(`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`, grepl(glob2rx("*reads classified"), variable)), aes(x = Allowed.errors, y = value, colour = variable, fill = variable)) + geom_point() + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(labels = comma) + scale_x_continuous(breaks=pretty_breaks(n=10)) + theme(legend.title=element_blank()) + scale_color_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_fill_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_y_continuous(trans = "log10", labels = comma) + xlab("Allowed errors") + ylab("Number of reads") + ggtitle("Number of reads classified to taxids of different type by Kaiju with different number of allowed errors")

ggplot(subset(`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`, grepl(glob2rx("*reads classified"), variable)), aes(x = Allowed.errors, y = value, colour = variable, fill = variable, shape = variable)) + geom_point(size = 2.5) + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_x_continuous(breaks=pretty_breaks(n=10)) + theme(legend.title=element_blank()) + scale_color_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_fill_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_y_continuous(trans = "log10", labels = comma, limits = c(1, 8000000)) + xlab("Allowed errors") + ylab("Number of reads") + ggtitle("Number of reads classified to taxids of different type by Kaiju with different number of allowed errors")

2022-04-08


product mapping stuff
View(`2022.04.07_transposed_all_samples_summed_mapping_contig_size_and_library_size_adjusted_scores_all_unique_highest_scoring_DFAST_products_all_bins`)

temporr <- `2022.04.07_transposed_all_samples_summed_mapping_contig_size_and_library_size_adjusted_scores_all_unique_highest_scoring_DFAST_products_all_bins`
`temporr_log1p` <- log1p(temporr[,c(2:4707)])
`temporr_log1p`$Species <- `temporr`$Species
`temporr_log1p`$Season <- `temporr`$Season
`temporr_log1p`$Sequencing <- `temporr`$Sequencing
`temporr_log1p`$Species_x_season <- `temporr`$Species_x_season

`PCA_2022_04_07_log1p_adjusted_transposed_all_samples_summed_mapping_contig_size_and_library_size_adjusted_scores_all_unique_highest_scoring_DFAST_products_all_bins` <- prcomp(`temporr_log1p`[,c(1:4706)], center = TRUE, scale. = TRUE)

`PCA_2022_04_07_log1p_adjusted_transposed_all_samples_summed_mapping_contig_size_and_library_size_adjusted_scores_all_unique_highest_scoring_DFAST_products_all_bins` <- prcomp(`temporr_log1p`[,c(1:4706)], center = TRUE, scale. = FALSE)

ggbiplot(`PCA_2022_04_07_log1p_adjusted_transposed_all_samples_summed_mapping_contig_size_and_library_size_adjusted_scores_all_unique_highest_scoring_DFAST_products_all_bins`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `temporr_log1p`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`temporr_log1p`$Season), color=factor(`temporr_log1p`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species")

ggbiplot(`PCA_2022_04_07_log1p_adjusted_transposed_all_samples_summed_mapping_contig_size_and_library_size_adjusted_scores_all_unique_highest_scoring_DFAST_products_all_bins`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `temporr_log1p`$Species, ellipse.prob = 0.95, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`temporr_log1p`$Season), color=factor(`temporr_log1p`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species")

plot(`PCA_2022_04_07_log1p_adjusted_transposed_all_samples_summed_mapping_contig_size_and_library_size_adjusted_scores_all_unique_highest_scoring_DFAST_products_all_bins`)

fviz_pca_biplot(`PCA_2022_04_07_log1p_adjusted_transposed_all_samples_summed_mapping_contig_size_and_library_size_adjusted_scores_all_unique_highest_scoring_DFAST_products_all_bins`, repel = TRUE, select.var = list(contrib = 5))

2022-04-12
plp1 <- ggplot(subset(`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`, "Unique taxIDs" == variable | "TaxIDs with more than 100 reads" == variable), aes(x = Allowed.errors, y = value, colour = variable, fill = variable, shape = variable)) + geom_point(size = 2.5) + geom_line(position = position_dodge(width = 0.2)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Allowed errors") + ylab("TaxIDs") + scale_fill_manual(values = c("TaxIDs with more than 100 reads" = "#000001", "Unique taxIDs" = "#ec9472")) + scale_color_manual(values = c("TaxIDs with more than 100 reads" = "#000001", "Unique taxIDs" = "#ec9472")) + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=5)) + scale_y_continuous(limits = c(0, 14000), breaks=pretty_breaks(n=20)) + geom_hline(yintercept = 20, linetype = "dashed", color = "#1c7abc", size = 1) + theme(legend.position =c(0.8,0.9)) + scale_shape_manual(values = c("TaxIDs with more than 100 reads" = 0, "Unique taxIDs" = 17))

plp2 <- ggplot(subset(`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`, "Exact strain occurrences" == variable | "Exact species only taxID occurrences" == variable), aes(x = Allowed.errors, y = value, colour = variable, fill = variable, shape = variable)) + geom_point(size = 2.5) + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Allowed errors") + ylab("TaxIDs") + scale_fill_manual(values = c("Exact species only taxID occurrences" = "#e9865f", "Exact strain occurrences" = "#9b3b15")) + scale_color_manual(values = c("Exact species only taxID occurrences" = "#e9865f", "Exact strain occurrences" = "#9b3b15")) + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10)) + scale_y_continuous(limits = c(0, 20), breaks=pretty_breaks(n=10)) + geom_hline(yintercept = 20, linetype = "dashed", color = "#1c7abc", size = 0.5) + theme(legend.position="top") + theme(legend.position =c(0.8,0.6)) + scale_shape_manual(values = c("Exact species only taxID occurrences" = 9, "Exact strain occurrences" = 11))

plp3 <- ggplot(subset(`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`, "Fuzzy false positive taxID occurrences" == variable | "Genus false positive taxID occurrences" == variable | "None same genus taxids detected" == variable), aes(x = Allowed.errors, y = value, colour = variable, fill = variable, shape = variable)) + geom_point(size = 2.5) + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Allowed errors") + ylab("TaxIDs") + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10)) + scale_shape_manual(values = c("Genus false positive taxID occurrences" = 1, "Fuzzy false positive taxID occurrences" = 6, "None same genus taxids detected" = 4)) + scale_color_manual(values = c("Genus false positive taxID occurrences" = "#9b3b15", "Fuzzy false positive taxID occurrences" = "#e67549", "None same genus taxids detected" = "#c84c1b")) + scale_fill_manual(values = c("Genus false positive taxID occurrences" = "#9b3b15", "Fuzzy false positive taxID occurrences" = "#e67549", "None same genus taxids detected" = "#c84c1b")) + scale_y_continuous(limits = c(0, 8000), breaks=pretty_breaks(n=20)) + theme(legend.position =c(0.8,0.8))

ggplot(subset(`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`, "Fuzzy false positive taxID occurrences" == variable | "Genus false positive taxID occurrences" == variable | "None same genus taxids detected" == variable), aes(x = Allowed.errors, y = value, colour = variable, fill = variable, shape = variable)) + geom_point(size = 2.5) + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Allowed errors") + ylab("TaxIDs") + theme(legend.title=element_blank()) + scale_x_continuous(breaks=pretty_breaks(n=10)) + scale_color_manual(values = c("Genus false positive taxID occurrences" = "#9b3b15", "Fuzzy false positive taxID occurrences" = "#e67549", "None same genus taxids detected" = "#c84c1b")) + scale_fill_manual(values = c("Genus false positive taxID occurrences" = "#9b3b15", "Fuzzy false positive taxID occurrences" = "#e67549", "None same genus taxids detected" = "#c84c1b")) + scale_y_continuous(limits = c(0, 6500), breaks=pretty_breaks(n=10)) + theme(legend.position =c(0.8,0.8)) + scale_shape_manual(values = c("Genus false positive taxID occurrences" = 12, "Fuzzy false positive taxID occurrences" = 13, "None same genus taxids detected" = 10))

plp1 + plp2 + plp3 + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') + plot_annotation(title = 'Kaiju analysis of human mock microbiota with different number of allowed errors')

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-04-12-Human_mock_Kaiju_detailed_breakdown.svg")

ggplot(subset(`2022.04.04_Melted_Kaiju_human_mock_detailed_taxID_known_members_breakdown`, grepl(glob2rx("*reads classified"), variable)), aes(x = Allowed.errors, y = value, colour = variable, fill = variable, shape = variable)) + geom_point(size = 2.5) + geom_line() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_x_continuous(breaks=pretty_breaks(n=10)) + theme(legend.title=element_blank()) + scale_color_manual(values = c("Exact species only reads classified" = "#000001", "Exact strain reads classified" = "#ec9472", "Fuzzy false positive reads classified" = "#e9865f", "Genus false positive reads classified" = "#9b3b15", "None same genus reads classified" = "#e67549")) + scale_fill_manual(values = c("Exact species only reads classified" = "#3399e0", "Exact strain reads classified" = "#0a2d45", "Fuzzy false positive reads classified" = "#1c7abc", "Genus false positive reads classified" = "#124e78", "None same genus reads classified" = "#6e7f80")) + scale_y_continuous(trans = "log10", labels = comma, limits = c(1, 8000000)) + xlab("Allowed errors") + ylab("Number of reads") + ggtitle("Number of reads classified to taxids of different type by Kaiju with different number of allowed errors")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-04-11_human_mock_kaiju_read_counts_different_taxid_groups.svg")

`2022.04.12_metaphlan_all_samples_only_most_recent_lab_mouse_downloads_reads_in_processed_fastq_file_estimated_reads_classified_by_metaphlan_and_percent_of_former_which_is_latter` <- read.delim("~/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-04-12_metaphlan_all_samples_only_most_recent_lab_mouse_downloads_reads_in_processed_fastq_file_estimated_reads_classified_by_metaphlan_and_percent_of_former_which_is_latter.txt")
View(`2022.04.12_metaphlan_all_samples_only_most_recent_lab_mouse_downloads_reads_in_processed_fastq_file_estimated_reads_classified_by_metaphlan_and_percent_of_former_which_is_latter`)

`2022.04.12_MELTED_metaphlan_all_samples_most_recent_lab_mouse_estimated_reads_classified_as_percent_total_reads_processed_file` <- reshape2::melt(`2022.04.12_metaphlan_all_samples_only_most_recent_lab_mouse_downloads_reads_in_processed_fastq_file_estimated_reads_classified_by_metaphlan_and_percent_of_former_which_is_latter`, id=c("Sample", "Host.species"), measure.vars=c("Estimated.reads.classified.by.Metaphlan", "Estimated.number.reads.classified.as.percentage.of.total.reads.in.file", "Reads.in.processed.file"))

ggplot(data = subset(`2022.04.12_MELTED_metaphlan_all_samples_most_recent_lab_mouse_estimated_reads_classified_as_percent_total_reads_processed_file`, grepl(glob2rx("*percentage*"), variable)), aes(x = Sample, y = value, color = Host.species, fill = Host.species)) + geom_bar(colour = "#0f0f0f", stat = "identity", position = position_dodge()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("Estimated number of reads mapped to know clades\nas a percentage of total reads in analysed file (%)") + xlab("Sample ID") + ggtitle("Percent reads classified by Metaphlan per sample by host species") + scale_fill_manual(values = c("HS" = "#8d1eff", "MM" = "#1a1a1a", "AC" = "#858585", "AR" = "#c9c9c9")) + scale_color_manual(values = c("HS" = "#8d1eff", "MM" = "#1a1a1a", "AC" = "#858585", "AR" = "#c9c9c9")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color = "Host species") + labs(fill = "Host species") + scale_y_continuous(limits = c(0, 120), breaks=pretty_breaks(n=12))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-04-11_metaphlanall_samples_only_most_recent_lab_mouse_downloads_reads_in_processed_fastq_file_estimated_reads_classified_by_metaphlan_and_percent_of_former_which_is_latter.svg")

View(`2022.04.11_transposed_all_samples_mapping_scores_to_bins_with_less_than_5_percent_contamination_kilobase_and_library_size_adjusted`)

`2022.04.11.mapping_to_good_bins` <- `2022.04.11_transposed_all_samples_mapping_scores_to_bins_with_less_than_5_percent_contamination_kilobase_and_library_size_adjusted`
`2022.04.11.logp1_adjusted_mapping_to_good_bins` <- log1p(`2022.04.11.mapping_to_good_bins`[,c(2:424)])
`2022.04.11.logp1_adjusted_mapping_to_good_bins`$Species <- `2022.04.11.mapping_to_good_bins`$Species
`2022.04.11.logp1_adjusted_mapping_to_good_bins`$Season <- `2022.04.11.mapping_to_good_bins`$Season
`2022.04.11.logp1_adjusted_mapping_to_good_bins`$Sequencing <- `2022.04.11.mapping_to_good_bins`$Sequencing
`2022.04.11.logp1_adjusted_mapping_to_good_bins`$Species_x_season <- `2022.04.11.mapping_to_good_bins`$Species_x_season
`2022.04.11.logp1_adjusted_mapping_to_good_bins`$Bins <- `2022.04.11.mapping_to_good_bins`$Bins

`2022.04.11.PCA.logp1_adjusted_mapping_to_good_bins` <- prcomp(`2022.04.11.logp1_adjusted_mapping_to_good_bins`[,c(1:423)], center = TRUE, scale. = FALSE)

ggbiplot(`2022.04.11.PCA.logp1_adjusted_mapping_to_good_bins`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.11.logp1_adjusted_mapping_to_good_bins`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.04.11.logp1_adjusted_mapping_to_good_bins`$Season), color=factor(`2022.04.11.logp1_adjusted_mapping_to_good_bins`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species")

plot(`2022.04.11.PCA.logp1_adjusted_mapping_to_good_bins`)

fviz_pca_biplot(`2022.04.11.PCA.logp1_adjusted_mapping_to_good_bins`, repel = TRUE, select.var = list(contrib = 5))

2022-04-13
View(`2022.04.10_CheckM_output_all_bins`)
hist(`2022.04.10_CheckM_output_all_bins`$Completeness)

checkm_bin_plot1 <- ggplot(data = `2022.04.10_CheckM_output_all_bins`, aes(x = Completeness)) + geom_histogram(binwidth = 5, fill = "#676767", color = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Completeness (%)") + ylab("Frequency")

checkm_bin_plot2 <- ggplot(data = `2022.04.10_CheckM_output_all_bins`, aes(x = Contamination)) + geom_histogram(binwidth = 5, fill = "#5b92e5", color = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Contamination (%)") + ylab("Frequency")

checkm_bin_plot3 <- ggplot(data = `2022.04.10_CheckM_output_all_bins`, aes(x = Completeness, y = Contamination, color = Strain.heterogeneity)) + geom_point() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_gradient(low = "#0f0f0f", high = "#ff830a") + xlab("Completeness (%)") + ylab("Contamination (%)") + labs(color = "Strain heterogeneity (%)")

checkm_bin_plot1 + checkm_bin_plot2 + checkm_bin_plot3 + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') + plot_annotation(title = 'CheckM analysis of 503 bin files')

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-13_all_bins_CheckM_output_completeness_contamination_frequency_and_completeness_to_contamination_points_with_strain_heterogeneity_colour.svg")

`2022.04.10_CheckM_output_all_bins`[ which ( `2022.04.10_CheckM_output_all_bins`$Completeness < 80 & `2022.04.10_CheckM_output_all_bins`$Contamination < 5) , ]
`2022.04.10_CheckM_output_all_bins_with_completeness_below_80_and_contamination_below_5` <- `2022.04.10_CheckM_output_all_bins`[ which ( `2022.04.10_CheckM_output_all_bins`$Completeness < 80 & `2022.04.10_CheckM_output_all_bins`$Contamination < 5) , ]

 View(`2022.04.13_all_bins_GTDB.TK_output_number_of_bins_classified_to_that_level`)

View(`2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`)
`2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`

ggplot(data = `2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`, aes(x = GTDB.TK.Classification.level, y = CheckM.Completeness, color = CheckM.Contamination)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(width = 0.5, outlier.shape = NA) + geom_jitter(width = 0.15, size = 2.35, alpha = 0.45) + scale_color_gradient(low = "#0f0f0f", high = "#ff830a")

ggplot(data = `2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`, aes(x = GTDB.TK.Classification.level, y = CheckM.Completeness, color = CheckM.Strain.heterogeneity)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(width = 0.5, outlier.shape = NA, fill = "#0a86ff", alpha = 0.25) + geom_jitter(width = 0.35, size = 2, alpha = 0.55) + scale_color_gradient(low = "#0f0f0f", high = "#ff830a") + ylab("Completeness (%)") + xlab("Lowest classification level") + labs(color = "Strain heterogeneity (%)")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-13_all_bins_CheckM_and_GTDB-TK_outputs_completeness_at_lowest_classified_taxonomic_level_with_strain_heterogeneity.svg")

ggplot(data = `2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`, aes(x = GTDB.TK.Classification.level, y = CheckM.Contamination, color = CheckM.Strain.heterogeneity)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(width = 0.5, outlier.shape = NA, fill = "#0a86ff", alpha = 0.25) + geom_jitter(width = 0.35, size = 2, alpha = 0.55) + scale_color_gradient(low = "#0f0f0f", high = "#ff830a") + ylab("Contamination (%)") + xlab("Lowest classification level") + labs(color = "Strain heterogeneity (%)")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-13_all_bins_CheckM_and_GTDB-TK_outputs_contamination_at_lowest_classified_taxonomic_level_with_strain_heterogeneity.svg")

ggplot(data = `2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[which ( `2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Species") , ], aes(x = GTDB.TK.Classification, y = CheckM.Completeness, color = CheckM.Strain.heterogeneity)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(width = 0.5, outlier.shape = NA, fill = "#0a86ff", alpha = 0.25) + geom_jitter(width = 0.35, size = 2, alpha = 0.55) + scale_color_gradient(low = "#0f0f0f", high = "#ff830a") + ylab("Completeness (%)") + xlab("Lowest classification level") + labs(color = "Strain heterogeneity (%)")

ggplot(data = `2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[which ( `2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Species") , ], aes(x = CheckM.Completeness, y = CheckM.Contamination, color = CheckM.Strain.heterogeneity)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(size = 2) + scale_color_gradient(low = "#0f0f0f", high = "#ff830a") + ylab("Completeness (%)") + xlab("Lowest classification level") + labs(color = "Strain heterogeneity (%)")

ggplot(data = `2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`, aes(x = CheckM.Completeness, y = CheckM.Contamination, color = CheckM.Strain.heterogeneity, shape = GTDB.TK.Classification.level)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point() + scale_color_gradient(low = "#0f0f0f", high = "#ff830a")

ggplot(data = `2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`, aes(x = GTDB.TK.Classification, y = CheckM.Completeness, color = CheckM.Contamination)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(width = 0.5, outlier.shape = NA) + geom_jitter(width = 0.15) + scale_color_gradient(low = "#0f0f0f", high = "#ff830a")

View(`2022.04.13_all_products_all_GOOD_bins_product_and_contig_counts_to_them`)

hist(`2022.04.13_all_products_all_GOOD_bins_product_and_contig_counts_to_them`$V2)

`2022.04.13_DENSITY_all_products_all_GOOD_bins_product_and_contig_counts_to_them` <- density(`2022.04.13_all_products_all_GOOD_bins_product_and_contig_counts_to_them`$V2)

ggplot(data = `2022.04.13_all_products_all_GOOD_bins_product_and_contig_counts_to_them`, aes(y = V2, x = "Products")) + geom_boxplot(width = 0.5, outlier.shape = NA, fill = "#0a86ff", alpha = 0.25) + geom_jitter(width = 0.85, size = 2, alpha = 0.55) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.04.13_all_products_all_GOOD_bins_product_and_contig_counts_to_them`, aes(x = V2)) + geom_histogram(binwidth = 5, fill = "#676767", color = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Number of contigs assigned to product") + ylab("Frequency")

2022-04-14
View(`2022.04.08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size`)

ggplot(data = `2022.04.08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size`, aes(x = Completeness, y = Contamination, color = Strain.heterogeneity)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(size = 4, position = position_dodge(width = 0.5)) + scale_color_gradient(low = "#0f0f0f", high = "#ff830a") + xlab("Completeness (%)") + ylab("Contamination (%)") + labs(color = "Strain heterogeneity (%)") + ggtitle("All isolate assemblies percent completeness to contamination with strain heterogeneity")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-04-08_CheckM_all_isolates_points_completeness_contamination_strain_heterogeneity.svg")

View(`2022.04.09_ANI_breakdown_all_cultures`)
View(`2022.04.05_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`)

2022-04-15

`2022.04.05_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[ , !names(`2022.04.05_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`) %in% c("S125")]

`2022.04.10_DFAST_all_potential_product_detections_all_isolates_except_S125` <- `2022.04.05_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[ , !names(`2022.04.05_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`) %in% c("S125")]

`2022.04.10_melted_DFAST_all_potential_product_detections_all_isolates_except_S125` <- reshape2::melt(`2022.04.10_DFAST_all_potential_product_detections_all_isolates_except_S125`, id=c("Products"))

ggplot(data = `2022.04.10_melted_DFAST_all_potential_product_detections_all_isolates_except_S125`, aes(x = variable, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_jitter()

ggplot(subset(`2022.04.10_melted_DFAST_all_potential_product_detections_all_isolates_except_S125`, grepl(glob2rx("X*"), variable) & `2022.04.10_melted_DFAST_all_potential_product_detections_all_isolates_except_S125`$value < 100), aes(x = variable, y = value))  + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_jitter()

ggplot(subset(`2022.04.10_melted_DFAST_all_potential_product_detections_all_isolates_except_S125`$value > 100)

2022-04-16
View(`2022.04.16_all_GOOD_bins_contig_count_and_DFAST_output_total_products_found_including_multiple_products_per_contig_and_total_unique_products_found`)

ggplot(data = `2022.04.16_all_GOOD_bins_contig_count_and_DFAST_output_total_products_found_including_multiple_products_per_contig_and_total_unique_products_found`, aes(x = Number.of.contigs, y = Total.number.products.found, colour = Total.number.unique.products.found)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point() + scale_color_gradient(low = "#0f0f0f", high = "#ff830a") + xlab("No. contigs in bin") + ylab("Products or potential products identified in Bin") + labs(color = "Number of Unique products\n(or potential products)\nidentified in Bin")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-16_all_GOOD_bins_DFAST_analysis_num_contigs_per_bin_to_num_products_identified_in_bin_with_num_unique_products_in_bin_as_colour_gradient.svg")

ggplot(data = `2022.04.16_all_GOOD_bins_contig_count_and_DFAST_output_total_products_found_including_multiple_products_per_contig_and_total_unique_products_found`, aes(x = Total.number.products.found, y =Total.number.unique.products.found)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(colour = "#ff830a") + xlab("Products/potential products identified in Bin") + ylab("Unique products/potential products identified in Bin")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-16_all_GOOD_bins_DFAST_analysis_num_products_identified_in_bin_to_num_unique_products_in_bin.svg")

View(`2022.04.16_all_unique_products_DFAST_analysis_GOOD_bins_single_highest_scoring_per_number_of_contigs_to_product`)
ggplot(data = `2022.04.16_all_unique_products_DFAST_analysis_GOOD_bins_single_highest_scoring_per_number_of_contigs_to_product`, aes(x = "Products and potential products", y = V2)) + geom_boxplot(width = 0.5, outlier.shape = NA, fill = "#0a86ff", alpha = 0.25) + geom_jitter(width = 0.85, size = 2, alpha = 0.55) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.04.16_all_unique_products_DFAST_analysis_GOOD_bins_single_highest_scoring_per_number_of_contigs_to_product`, aes(x = "Products/potential products", y = V2)) + geom_jitter(width = 0.55, size = 1.5, alpha = 0.45) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(trans = "log10", labels = comma, limits = c(1, 13000)) + ylab("Detections across all bins") + xlab("")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-16_all_GOOD_bins_DFAST_analysis_number_of_detections_for_each_unique_highest_scoring_product_across_all_bins_log_10_y_scale_jitter_plot.svg")

ggplot(data = `2022.04.16_all_unique_products_DFAST_analysis_GOOD_bins_single_highest_scoring_per_number_of_contigs_to_product`, aes(x = V2)) + geom_histogram(binwidth = 10, fill = "#676767", color = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Product/potential product total\ndetections across all bins") + ylab("Frequency")

ggplot(data = `2022.04.16_all_unique_products_DFAST_analysis_GOOD_bins_single_highest_scoring_per_number_of_contigs_to_product`, aes(x = V2)) + geom_freqpoly(binwidth = 10, fill = "#676767", color = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Product/potential product total\ndetections across all bins") + ylab("Frequency")

View(`2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`)

2022-04-17
ggplot(data = `2022.04.16_all_GOOD_bins_contig_count_and_DFAST_output_total_products_found_including_multiple_products_per_contig_and_total_unique_products_found`, aes(x = "", y = Number.of.contigs)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(width = 0.5, outlier.shape = NA, fill = "#0a86ff", alpha = 0.25) + geom_jitter(color = "#ff830a", width = 0.25, size = 2, alpha = 0.75) + ylab("Number of contigs in Bin") + xlab("Bins")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-17_all_good_bins_number_of_contigs_in_bin_boxplot_and_jitter.svg")

View(`2022.04.14_concatenated_low_contamination_high_completeness_bins.List_of_contigs_and_their_lengths`)

ggplot(data = `2022.04.14_concatenated_low_contamination_high_completeness_bins.List_of_contigs_and_their_lengths`, aes(x = "", y = Length)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_violin(color = "#0f0f0f", fill = "#1e90ff") + ylab("Length of contig") + xlab("Contigs") + scale_y_continuous(labels = comma) + geom_jitter(data = subset(`2022.04.14_concatenated_low_contamination_high_completeness_bins.List_of_contigs_and_their_lengths`, Length > 50000), aes(x = "", y = Length), color = "#ff830a", width = 0.55, size = 2.5, alpha = 0.35) + labs(caption="Only contig lengths greater than 50kb shown with points")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-17_all_good_bins_all_contigs_length_violin_plot_with_jitter_points_for_lengths_over_50kb_as_well.svg")

`2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`

`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level` <- reshape2::melt(`2022.04.13_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`, id=c("Lowest.classification", "GTDB.TK.Classification.level", "GTDB.TK.Classification", "Bin.id"), measure.vars=c("CheckM.Completeness", "CheckM.Contamination", "CheckM.Strain.heterogeneity"))

`2022-04-17_good_bins_GTDB-TK_species_level_completeness_plot` <- ggplot(data = `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$variable == "CheckM.Completeness" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Species", ], aes(x = Lowest.classification, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_jitter(color = "#545454", size = 2, width = 0) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("CheckM Completeness (%)") + xlab("GTDB-TK Detected Species") + theme(axis.text.x = element_text(size = 8)) + scale_y_continuous(name = "CheckM Completeness (%)", limits = c(0, 100), breaks = scales::pretty_breaks(n = 10))

`2022-04-17_good_bins_GTDB-TK_species_level_contamination_plot` <- ggplot(data = `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$variable == "CheckM.Contamination" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Species", ], aes(x = Lowest.classification, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_jitter(color = "#545454", size = 2, width = 0) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("CheckM Contamination (%)") + xlab("GTDB-TK Detected Species") + theme(axis.text.x = element_text(size = 8)) + scale_y_continuous(name = "CheckM Contamination (%)", limits = c(0, 100), breaks = scales::pretty_breaks(n = 10))

`2022-04-17_good_bins_GTDB-TK_species_level_completeness_plot` + `2022-04-17_good_bins_GTDB-TK_species_level_contamination_plot` + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-17_all_good_bins_GTDB-TK_species_level_classifications_combined_two_plots_CheckM_completeness_and_contamination.svg")

`2022-04-17_good_bins_GTDB-TK_family_level_completeness_plot` <- ggplot(data = `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$variable == "CheckM.Completeness" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Family", ], aes(x = Lowest.classification, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(fill = "#ff8d1e", width = 0.5, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("CheckM Completeness (%)") + xlab("GTDB-TK Detected Family") + scale_y_continuous(name = "CheckM Completeness (%)", limits = c(0, 100), breaks = scales::pretty_breaks(n = 10)) + theme(axis.text.x = element_text(size = 8))

`2022-04-17_good_bins_GTDB-TK_family_level_contamination_plot` <- ggplot(data = `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$variable == "CheckM.Contamination" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Family", ], aes(x = Lowest.classification, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(fill = "#ff8d1e", width = 0.5, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("CheckM Contamination (%)") + xlab("GTDB-TK Detected Family") + scale_y_continuous(name = "CheckM Contamination (%)", limits = c(0, 100), breaks = scales::pretty_breaks(n = 10)) + theme(axis.text.x = element_text(size = 8))

`2022-04-17_good_bins_GTDB-TK_family_level_completeness_plot` + `2022-04-17_good_bins_GTDB-TK_family_level_contamination_plot` + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-17_all_good_bins_GTDB-TK_family_level_classifications_combined_two_plots_CheckM_completeness_and_contamination.svg")

`%notin%` <- Negate(`%in%`)

`2022-04-17_list_nice_name_genera_all_good_bins_GTDB-TK_lowest_classification_levels` <- c("Acetatifactor", "Actinomyces", "Acutalibacter", "Aeriscardovia", "Agathobacter", "Alistipes", "Allobaculum", "Amulumruptor", "Anaerosacchariphilus", "Anaerotruncus", "Bifidobacterium", "Bilophila", "Butyrivibrio", "Cellulosilyticum", "Desulfovibrio", "Duncaniella", "Eisenbergiella", "Emergencia", "Eubacterium F", "Eubacterium G", "Eubacterium J", "Eubacterium R", "Faecalibaculum", "Helicobacter C", "Helicobacter D", "Ileibacterium", "Kineothrix", "Lachnospira", "Lactobacillus", "Limosilactobacillus", "Mailhella", "Monoglobus", "Muribaculum", "Odoribacter", "Parabacteroides", "Paramuribaculum", "Paraprevotella", "Phocaeicola", "Prevotella", "Prevotellamassilia", "Pseudoruminococcus", "Pygmaiobacter", "Rikenella", "Ruminiclostridium E", "Ruminococcus", "Ruminococcus C", "Ruminococcus E", "Ruthenibacterium")

`2022-04-17_list_alphanumeric_name_genera_all_good_bins_GTDB-TK_lowest_classification_levels` <- c("ASF356", "BX12", "C-19", "CAG-1031", "CAG-110", "CAG-115", "CAG-127", "CAG-1435", "CAG-180", "CAG-194", "CAG-269", "CAG-303", "CAG-353", "CAG-41", "CAG-433", "CAG-460", "CAG-462", "CAG-465", "CAG-485", "CAG-492", "CAG-495", "CAG-582", "CAG-590", "CAG-594", "CAG-603", "CAG-632", "CAG-710", "CAG-873", "CAG-882", "CAG-95", "CAG-964", "COE1", "D16-63", "HGM05190", "HGM13634", "JAAYNV01", "M3", "MD308", "NC2004", "Rs-D84", "RUG11198", "RUG115", "RUG14121", "RUG14199", "RUG762", "TF01-11", "UBA11940", "UBA1394", "UBA1405", "UBA1436", "UBA1762", "UBA1777", "UBA2821", "UBA2866", "UBA3263", "UBA3282", "UBA3789", "UBA3818", "UBA3855", "UBA4716", "UBA5578", "UBA6857", "UBA7001", "UBA7057", "UBA7067", "UBA7109", "UBA7173", "UBA7182", "UBA9475", "UBA9502", "UMGS1004", "UMGS1312", "UMGS1370", "UMGS1663", "UMGS1696", "UMGS1815", "UMGS1872", "UMGS268", "UMGS995", "V9D3004", "VSOB01", "ZJ304", "1XD42-69", "1XD8-76", "14-2", "51-20", "12844")

ggplot(data = `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$variable == "CheckM.Completeness" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Genus", ], aes(x = Lowest.classification, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(fill = "#1e90ff", width = 0.5, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("CheckM Completeness (%)") + xlab("GTDB-TK Detected Genus") + theme(axis.text.x = element_text(size = 8)) + scale_y_continuous(name = "CheckM Completeness (%)", limits = c(0, 100), breaks = scales::pretty_breaks(n = 10))

ggplot(data = `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$variable == "CheckM.Contamination" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Genus", ], aes(x = Lowest.classification, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(fill = "#1e90ff", width = 0.5, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("CheckM Contamination (%)") + xlab("GTDB-TK Detected Genus") + theme(axis.text.x = element_text(size = 8)) + scale_y_continuous(name = "CheckM Contamination (%)", limits = c(0, 100), breaks = scales::pretty_breaks(n = 10))

ggplot(data = `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$variable == "CheckM.Completeness" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Genus" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$Lowest.classification %in% `2022-04-17_list_nice_name_genera_all_good_bins_GTDB-TK_lowest_classification_levels`, ], aes(x = Lowest.classification, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(fill = "#1e90ff", width = 0.5, outlier.shape = 3) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("CheckM Completeness (%)") + xlab("GTDB-TK Detected Genus") + theme(axis.text.x = element_text(size = 8)) + scale_y_continuous(name = "CheckM Completeness (%)", limits = c(40, 100), breaks = scales::pretty_breaks(n = 10))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-17_all_good_bins_GTDB-TK_genus_level_classifications_plot_CheckM_completeness_only.svg")

ggplot(data = `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$variable == "CheckM.Contamination" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Genus" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$Lowest.classification %in% `2022-04-17_list_nice_name_genera_all_good_bins_GTDB-TK_lowest_classification_levels`, ], aes(x = Lowest.classification, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(fill = "#1e90ff", width = 0.5, outlier.shape = 3) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("CheckM Contamination (%)") + xlab("GTDB-TK Detected Genus") + theme(axis.text.x = element_text(size = 8)) + scale_y_continuous(name = "CheckM Contamination (%)", limits = c(0, 62), breaks = scales::pretty_breaks(n = 10))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-17_all_good_bins_GTDB-TK_genus_level_classifications_plot_CheckM_contamination_only.svg")

ggplot(data = `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$variable == "CheckM.Completeness" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Genus" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$Lowest.classification %in% `2022-04-17_list_alphanumeric_name_genera_all_good_bins_GTDB-TK_lowest_classification_levels`, ], aes(x = Lowest.classification, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(fill = "#a751ff", width = 0.5, outlier.shape = 4) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("CheckM Completeness (%)") + xlab("GTDB-TK Detected Genus") + theme(axis.text.x = element_text(size = 7)) + scale_y_continuous(name = "CheckM Completeness (%)", limits = c(40, 100), breaks = scales::pretty_breaks(n = 10))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-17_all_good_bins_GTDB-TK_other_genus_level_classifications_plot_CheckM_completeness_only.svg")

ggplot(data = `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$variable == "CheckM.Contamination" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Genus" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$Lowest.classification %in% `2022-04-17_list_alphanumeric_name_genera_all_good_bins_GTDB-TK_lowest_classification_levels`, ], aes(x = Lowest.classification, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(fill = "#a751ff", width = 0.5, outlier.shape = 4) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("CheckM Contamination (%)") + xlab("GTDB-TK Detected Genus") + theme(axis.text.x = element_text(size = 7)) + scale_y_continuous(name = "CheckM Contamination (%)", limits = c(0, 50), breaks = scales::pretty_breaks(n = 10))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-17_all_good_bins_GTDB-TK_other_genus_level_classifications_plot_CheckM_contamination_only.svg")

View(`2022.04.17_TRANSPOSED_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted`)

`2022.04.17.mapping_to_good_bins` <- `2022.04.17_TRANSPOSED_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted`
`2022.04.17.logp1_adjusted_mapping_to_good_bins` <- log1p(`2022.04.17.mapping_to_good_bins`[,c(2:349)])
`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species <- `2022.04.17.mapping_to_good_bins`$Species
`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Season <- `2022.04.17.mapping_to_good_bins`$Season
`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Sequencing <- `2022.04.17.mapping_to_good_bins`$Sequencing
`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species_x_season <- `2022.04.17.mapping_to_good_bins`$Species_x_season
`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Bins <- `2022.04.17.mapping_to_good_bins`$Bins

`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins` <- prcomp(`2022.04.17.logp1_adjusted_mapping_to_good_bins`[,c(1:348)], center = TRUE, scale. = FALSE)

`2022-04-17_PC1_vs_PC2_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` <- ggbiplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Season), color=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-04-17_PC1_vs_PC3_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` <- ggbiplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Season), color=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-04-17_PC2_vs_PC3_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` <- ggbiplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Season), color=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-04-17_PC1_vs_PC2_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` + `2022-04-17_PC1_vs_PC3_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` + `2022-04-17_PC2_vs_PC3_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_top_3_PCs_three_plots_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

fviz_eig(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`)
fviz_cluster(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`)

fviz_pca_biplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, repel = TRUE, select.var = list(contrib = 5))

fviz_pca_biplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, repel = TRUE, select.var = list(contrib = 5), font.family = "Tex Gyre Heroes")

fviz_pca_biplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, repel = TRUE, select.var = list(contrib = 15), font.family = "Tex Gyre Heroes", col.ind = `2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species, palette = c('#0073ff', '#ff8d1e'), addEllipses = TRUE, ellipse.type = "confidence", legend.title = "Host species")

fviz_pca_var(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, col.var = "contrib", gradient.cols = c("#0f0f0f", "#3b0078", "#6c00da", "#8916ff", "#a751ff", "#cf9fff"), repel = TRUE, select.var = list(contrib = 40), font.family = "Tex Gyre Heroes")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_PCA_var_plot_top_40_PCs_1_and_2_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

fviz_pca_var(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, col.var = "contrib", gradient.cols = c("#0f0f0f", "#3b0078", "#6c00da", "#8916ff", "#a751ff", "#cf9fff"), repel = TRUE, select.var = list(contrib = 30), axes = c(1, 3), font.family = "Tex Gyre Heroes")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_PCA_var_plot_top_30_PCs_1_and_3_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

fviz_pca_var(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, col.var = "contrib", gradient.cols = c("#0f0f0f", "#3b0078", "#6c00da", "#8916ff", "#a751ff", "#cf9fff"), repel = TRUE, select.var = list(contrib = 20), axes = c(2, 3), font.family = "Tex Gyre Heroes")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_PCA_var_plot_top_20_PCs_2_and_3_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins-Var` <- get_pca_var(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`)

corr_cross(`temp_table`, max_pvalue = 0.05, top = 10)

temp_table <- `2022.04.17.logp1_adjusted_mapping_to_good_bins`[,c(1:348)]

2022-04-18
View(`2022.04.16_all_GOOD_bins_contig_count_and_DFAST_output_total_products_found_including_multiple_products_per_contig_and_total_unique_products_found`)
ggplot(data = `2022.04.16_all_GOOD_bins_contig_count_and_DFAST_output_total_products_found_including_multiple_products_per_contig_and_total_unique_products_found`, aes(x = Number.of.contigs, y = Total.number.products.found, colour = Bin.size)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point() + scale_color_gradient(low = "#0f0f0f", high = "#b9dcff") + xlab("No. contigs in bin") + ylab("Products or potential products identified in Bin") + labs(color = "Size of bin (bp)")
ggplot(data = `2022.04.16_all_GOOD_bins_contig_count_and_DFAST_output_total_products_found_including_multiple_products_per_contig_and_total_unique_products_found`, aes(x = Number.of.contigs, y = Total.number.products.found, size = Bin.size, colour = Total.number.unique.products.found)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point() + scale_color_gradient(low = "#8d1eff", high = "#cee7ff", labels = comma) + xlab("No. contigs in bin") + ylab("Products or potential products identified in Bin") + labs(color = "Total number unique products detected in bin") + labs(size = "Bin size (bp)")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-all_good_bins_DFAST_output_number_products_per_bin_unique_products_per_bin_contig_number_per_bin_and_bin_size.svg")


View(`2022.04.17_TRANSPOSED_all_samples_mapping_to_completeness_min_80_percent_contamination_under_5_percent_concatenated_bins_summed_by_single_highest_scoring_product_per_contig_unique_products_from_DFAST_kilobase_and_library_size_adjusted`)
`2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted` <- `2022.04.17_TRANSPOSED_all_samples_mapping_to_completeness_min_80_percent_contamination_under_5_percent_concatenated_bins_summed_by_single_highest_scoring_product_per_contig_unique_products_from_DFAST_kilobase_and_library_size_adjusted`
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted` <- log1p(`2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`[,c(2:3723)])
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species <- `2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Season <- `2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Season
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Sequencing <- `2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Sequencing
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species_x_season <- `2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species_x_season
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Product <- `2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Product

`2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted` <- prcomp(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`[,c(1:3722)], center = TRUE, scale. = FALSE)

ggbiplot(`2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species, ellipse.prob = 0.50, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(shape=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Season), color=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_PCA_biplot_PC1_and_PC2_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_to_single_highest_scoring_unique_product_per_contig_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

ggbiplot(`2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species, ellipse.prob = 0.50, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(shape=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Season), color=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_PCA_biplot_PC1_and_PC3_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_to_single_highest_scoring_unique_product_per_contig_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

ggbiplot(`2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species, ellipse.prob = 0.50, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(shape=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Season), color=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_PCA_biplot_PC2_and_PC3_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_to_single_highest_scoring_unique_product_per_contig_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

`2022.4.18_product_mapping_PCA_content` <- get_pca_var(`2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`)
head(`2022.4.18_product_mapping_PCA_content`$coord)
head(`2022.4.18_product_mapping_PCA_content`$cos2)
head(`2022.4.18_product_mapping_PCA_content`$contrib)
head(`2022.4.18_product_mapping_PCA_content`$rotation)
head(`2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation)
`2022.4.18_product_mapping_PCA_rotations` <- `2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation

`2022.04.18_pca_factomineR_version` <- FactoMineR::PCA(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`[,c(1:3722)])

`contribs_PC1_and_PC2_2022.4.18_product_mapping_PCA` <- `2022.04.18_pca_factomineR_version`$var$contrib[,1:2]
`eigenvalues_PC1_and_PC2_2022.4.18_product_mapping_PCA` <- `2022.04.18_pca_factomineR_version`$eig[,1][1:2]
`contrib_totals_PC1_and_PC2_2022.4.18_product_mapping_PCA` <- apply(`contribs_PC1_and_PC2_2022.4.18_product_mapping_PCA`, 1, function(x) {sum(x*`eigenvalues_PC1_and_PC2_2022.4.18_product_mapping_PCA`)/sum(`eigenvalues_PC1_and_PC2_2022.4.18_product_mapping_PCA`)})
number_twenty = 20
names(sort(`contrib_totals_PC1_and_PC2_2022.4.18_product_mapping_PCA`, decreasing = TRUE)[1:number_twenty]) -> `2022.04.18_pca_factomineR_version_product_good_bin_mapping_log1p_transformed_lib_and_contig_size_adjusted_all_samples-top_20_PC1_and_PC2_contributing_variables`

fviz_pca_biplot(`2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, repel = TRUE, select.var = list(contrib = 15), font.family = "Tex Gyre Heroes", col.ind = `2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species, palette = c('#0073ff', '#ff8d1e'), addEllipses = TRUE, ellipse.type = "confidence", legend.title = "Host species")

fviz_pca_var(`2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, select.var = list(contrib = 10))

barplot(`2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation[,1], main = "PC 1 Loadings Plot", las = 2)
barplot(`2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation[,1][1:20], main = "PC 1 Loadings Plot", las = 2)

screeplot(`2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`)

`modable-2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted` <- `2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`
unclass(`modable-2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`)
screeplot(`modable-2022.04.18_PCA_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`)

`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted` <- prcomp(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`[,c(1:3722)], center = TRUE, scale. = TRUE)

ggbiplot(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species, ellipse.prob = 0.50, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(shape=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Season), color=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_SCALED_PCA_biplot_PC1_and_PC2_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_to_single_highest_scoring_unique_product_per_contig_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

ggbiplot(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species, ellipse.prob = 0.50, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(shape=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Season), color=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_SCALED_PCA_biplot_PC1_and_PC3_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_to_single_highest_scoring_unique_product_per_contig_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

ggbiplot(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species, ellipse.prob = 0.50, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_point(aes(shape=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Season), color=factor(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_SCALED_PCA_biplot_PC2_and_PC3_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_to_single_highest_scoring_unique_product_per_contig_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

screeplot(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`)
barplot(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation[,1], main = "PC 1 Loadings Plot", las = 2)

barplot(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation[,1], main = "PC 1 Loadings Plot", las = 2)

`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation[`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation > 0.02]

tomper <- as.data.frame(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation)
tomper <- cbind(Product.Names = rownames(`tomper`), `tomper`)

ggplot(data = `tomper`[`tomper`$PC1 > 0.19 | `tomper`$PC1 < -0.0001, ], aes(x = Product.Names, y = PC1)) + geom_jitter()
ggplot(data = `tomper`[`tomper`$PC1 > 0.018, ], aes(x = Product.Names, y = PC1)) + geom_jitter()


barplot(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation[`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation > 0.02][,1], main = "PC 1 Loadings Plot", las = 2)

ggplot(data = `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`[`2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$variable == "CheckM.Contamination" & `2022.04.13_Melted_All_bins_CheckM_contamination_completeness_and_strain_heterogeneity_with_GTDB.TK_classification_and_clade_level`$GTDB.TK.Classification.level == "Genus", ], aes(x = Lowest.classification, y = value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(fill = "#1e90ff", width = 0.5, outlier.shape = NA) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("CheckM Contamination (%)") + xlab("GTDB-TK Detected Genus") + theme(axis.text.x = element_text(size = 8)) + scale_y_continuous(name = "CheckM Contamination (%)", limits = c(0, 100), breaks = scales::pretty_breaks(n = 10))

---

View(`2022.04.05_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`)
list_colnames <- colnames(`2022.04.05_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`)
list_colnames_except_product <- list_colnames[!list_colnames %in% "Products"]
`2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results` <- reshape2::melt(`2022.04.05_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, id=c("Products"), measure.vars=c(list_colnames_except_product))

ggplot(data = `2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[`2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$value > 50, ], aes(x = Products, y = value)) + geom_boxplot()

ggplot(data = `2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, aes(x = "Products", value)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_violin(color = "#0f0f0f", fill = "#1e90ff") + geom_jitter(data = `2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[`2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$value > 50, ], aes(x = "Products", y = value), color = "#ff830a", width = 0.55, size = 2.5, alpha = 0.65) + ylab("Detections in assembly") + xlab("") + labs(caption = "Only products with at least 50 detections\nin one assembly show with points")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022.04.14_concatenated_low_contamination_high_completeness_bins.List_of_contigs_and_their_lengths_all_products_all_assemblies_violin_plot_with_jitter_points_any_count_over_50.svg")
`2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results` <- `2022.04.05_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[,-1]
rownames(`2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`) <- `2022.04.05_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[,1]

`transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results` <- data.table::transpose(`2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`)
colnames(`transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`) <- rownames(`2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`)
rownames(`transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`) <- colnames(`2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`)
`transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$assembly <- rownames(`transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`)

`2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results` <- reshape2::melt(`transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, id=c("Host", "assembly"))
`2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$value = as.numeric(as.character(`2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$value))

ggplot(data = `2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, aes(x = Host, y = value, colour = Host, fill = Host)) + geom_boxplot() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, aes(x = Host, y = value, colour = Host)) + geom_violin() + geom_jitter(data = `2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[`2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$value > 50, ], aes(x = Host, y = value, color = Host)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_color_manual(values = c("AC" = '#1e90ff', "AR" = '#ffb700', "MM" = '#5218fa'))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-04-05_all_products_all_assemblies_violin_plot_with_jitter_points_any_count_over_50_by_host_species_isolate_comes_from.svg")

`2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results` %>% group_by(Host, variable) %>% mutate(value, diff_value = value[Host == "AR"] - value[Host == "AC"])

df_2 <- mutate_all(`transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[,c(1:2432)], function(x) as.numeric(as.character(x)))
df_2$Host <- `transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$Host
df_2$Assembly <- `transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$assembly
df_2_log1p <- log1p(`df_2`[,c(1:2432)])
df_2_log1p$Host <- df_2$Host
df_2_log1p$Assembly <- df_2$Assembly
melted_df_2_log1p <- reshape2::melt(`df_2_log1p`, id=c("Host", "Assembly"))

attach(melted_df_2_log1p)
kruskal.test(value, g = Host)
kruskal.test(value ~ Host, data = melted_df_2_log1p)

kruskal.test(value ~ Host, data = melted_df_2_log1p[melted_df_2_log1p$variable == "AAA family ATPase", ])

for(i in unique(melted_df_2_log1p$variable))
{
out <- paste0("A thing here? ", i, ".")
print(out)
}

trial_kruskal_test_results_lop1_isolate_product_counts <- list()
for(i in unique(melted_df_2_log1p$variable)){
  results[[i]] <- kruskal.test(formula(paste(i, "~ Host", data = melted_df_2_log1p[melted_df_2_log1p])))
}

apply(melted_df_2_log1p[])

ggbetweenstats(data = `2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[`2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$variable == "antibiotic biosynthesis monooxygenase", ], x = Host, y = value, type = "wilcox")


df2 <- mutate_all(`transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[,c(1:2432)], function(x) as.numeric(as.character(x)))
df2$Host <- `transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$Host

`log1p_df2_transformed` <- log1p(df2[,c(1:2432)])
`log1p_df2_transformed`$Host <- df2$Host
`log1p_df2_transformed`$Host <- df2$Host


`log1p_transformed_isolate_product_detection_counts_COR` <- cor(`log1_df2_transformed`)
`DF_log1p_transformed_isolate_product_detection_counts_COR` <- as.data.frame(`log1p_transformed_isolate_product_detection_counts_COR`)
`Long_format-DF_log1p_transformed_isolate_product_detection_counts_COR` <- `DF_log1p_transformed_isolate_product_detection_counts_COR` %>% rownames_to_column() %>% gather(colname, value, -rowname)

corr_cross(as.data.frame(`log1p_transformed_isolate_product_detection_counts_COR`, max_pvalue = 0.05, top = 10))
corr_cross(`DF_log1p_transformed_isolate_product_detection_counts_COR`, max_pvalue = 0.05, top = 10))

`2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results` <- reshape2::melt(`2022.04.05_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, id=c("Products"), measure.vars=c(list_colnames_except_product))

`log1p_transformed_transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results` <- log1p(`transposed-2022-04-05_row_name_products_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[,c(1:2432)])

`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted` <- prcomp(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`[,c(1:3722)], center = TRUE, scale. = TRUE)

top_n(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation, 10, PC1)

`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted-rotations` <- `2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation
`2022.04.18_DF_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted-rotations` <- as.data.frame(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted-rotations`)
top_n(`2022.04.18_DF_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted-rotations`, 10, `2022.04.18_DF_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted-rotations`$PC1)

aload <- abs(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation)
head(sweep(aload, 2, colSums(aload), "/"))

View(`2022.04.17_TRANSPOSED_all_samples_mapping_to_completeness_min_80_percent_contamination_under_5_percent_concatenated_bins_summed_by_single_highest_scoring_product_per_contig_unique_products_from_DFAST_kilobase_and_library_size_adjusted`)
`2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted` <- `2022.04.17_TRANSPOSED_all_samples_mapping_to_completeness_min_80_percent_contamination_under_5_percent_concatenated_bins_summed_by_single_highest_scoring_product_per_contig_unique_products_from_DFAST_kilobase_and_library_size_adjusted`
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted` <- log1p(`2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`[,c(2:3723)])
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species <- `2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Season <- `2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Season
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Sequencing <- `2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Sequencing
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species_x_season <- `2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Species_x_season
`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Product <- `2022.04.18_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$Product

`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted` <- prcomp(`2022.04.18_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`[,c(1:3722)], center = TRUE, scale. = TRUE)

screeplot(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, type = "l")

fviz_eig(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, barfill = "#c994ff", barcolor = "#8d1eff", linecolor = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Log1p transformed, library and contig size adjusted mapping scores summed to product - PCA", x = "Principal components", y = "% of variances")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_PCA_fviz_eigenvalue_plot_percent_variance_explained_top_10_PCs_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_product_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/shite.svg")

ggplot(data = `2022.04.18_DF_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted-rotations`[`2022.04.18_DF_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted-rotations`$PC1 > 0.02 | `2022.04.18_DF_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted-rotations`$PC1 < -0.0315, ], aes(x = "Thing", y = PC1)) + geom_jitter()

marque2 <- `2022.04.18_DF_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted-rotations`
marque2 <- tibble::rownames_to_column(`2022.04.18_DF_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted-rotations`, "Product")
melted_marque2 <- reshape2::melt(marque2, id=c("Product"))

ggplot(data = melted_marque2[(melted_marque2$variable == "PC1" | melted_marque2$variable == "PC2") & (melted_marque2$value > 0.02 | melted_marque2$value < -0.03) , ], aes(x = variable, y = value, color = variable)) + geom_jitter()

limited_marque <- melted_marque2[melted_marque2$variable == "PC1" | melted_marque2$variable == "PC2", ]

ggplot(data = melted_marque2[melted_marque2$variable %in% c("PC1", "PC2", "PC3", "PC4", "PC5") & (melted_marque2$value > 0.02 | melted_marque2$value < -0.03), ], aes(x = variable, y = value, color = variable)) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) +

ggplot(data = melted_marque2[melted_marque2$variable %in% c("PC1", "PC2", "PC3", "PC4", "PC5"), ], aes(x = variable, y = value, color = variable)) + geom_jitter()

ggplot(data = melted_marque2[melted_marque2$variable %in% c("PC1") & (melted_marque2$value > 0.02 | melted_marque2$value < -0.03), ], aes(x = variable, y = value, color = variable, label = Product)) + geom_jitter()

2022-04-19
`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`$rotation[,1:5]

fviz_cos2(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, choice = "var", axes = 1, top = 10, fill = "#ff8d1e", colour = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Log1p transformed, library and contig size adjusted mapping scores\nsummed to product - Cos2 of variables to PC1") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + xlab("Product")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-19_PCA_log1p_transformed_contig_and_lib_size_adjusted_all_samples_mapping_scores_good_bin_single_unique_products_PC1_top_10_Cos2_products.svg")

fviz_pca_var(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, select.var = list(contrib = 5), col.var = "contrib", gradient.cols = c("#383aff", "#ff389d", "#ff8d1f"), repel = TRUE, labelsize = 3)+ theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Log1p transformed, library and contig size adjusted mapping scores\nsummed to product - PCA Variables") + labs(color = "Contribution")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-19_PCA_log1p_transformed_contig_and_lib_size_adjusted_all_samples_mapping_scores_good_bin_single_unique_products_PC1_PC2_top_5_contrib_products.svg")

fviz_pca_var(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, axes = c(1, 3), select.var = list(contrib = 5), col.var = "contrib", gradient.cols = c("#383aff", "#ff389d", "#ff8d1f"), repel = TRUE, labelsize = 3)+ theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Log1p transformed, library and contig size adjusted mapping scores\nsummed to product - PCA Variables") + labs(color = "Contribution")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-19_PCA_log1p_transformed_contig_and_lib_size_adjusted_all_samples_mapping_scores_good_bin_single_unique_products_PC1_PC3_top_5_contrib_products.svg")

fviz_pca_var(`2022.04.18_PCA_SCALED_log1p_good_bin_product_mapping_kilobase_and_lib_size_adjusted`, axes = c(2, 3), select.var = list(contrib = 5), col.var = "contrib", gradient.cols = c("#383aff", "#ff389d", "#ff8d1f"), repel = TRUE, labelsize = 3)+ theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "Log1p transformed, library and contig size adjusted mapping scores\nsummed to product - PCA Variables") + labs(color = "Contribution")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-19_PCA_log1p_transformed_contig_and_lib_size_adjusted_all_samples_mapping_scores_good_bin_single_unique_products_PC2_PC3_top_5_contrib_products.svg")

select.var = list(contrib = 10)

melted_df_2_log1p

`melted_df_2_log1p`$Assembly <- gsub("X", "", as.character(`melted_df_2_log1p`$Assembly))

ggplot(data = melted_df_2_log1p[melted_df_2_log1p$Assembly %in% c("16aC_S26", "13E_S5"), ], aes(x = Assembly, y = value, color = Host)) + geom_jitter(alpha = 0.5) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_colour_manual(values = c("AC" = "#1e90ff", "AR" = "#ff8d1e")) + ylab("Product detections (log1p transformed)")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-04-05_all_products_all_assemblies_two_assemblies_different_host_species_over_99_point_9_percent_ANI.svg")

ggplot(data = melted_marque2[melted_marque2$variable %in% c("PC1", "PC2", "PC3", "PC4", "PC5"), ], aes(x = variable, y = value, color = variable)) + geom_jitter()

kruskal.test(value ~ Host, data = melted_df_2_log1p[melted_df_2_log1p$variable == "AAA family ATPase", ])

ggbetweenstats(data = `2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[`2022.04.05_Melted_all_assemblies_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$variable == "antibiotic biosynthesis monooxygenase", ], x = Host, y = value, type = "nonparametric", pairwise.comparisons = TRUE)

2022-05-03
`2022.05.03_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-05-03_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results.txt")
View(`2022.05.03_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`)

`2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-05-03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results.txt")
View(`2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`)

`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results` <- reshape2::melt(`2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, id=c("Products", "Host.species"))

set.seed(909)
ggbetweenstats(data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, variable == "ABC.transporter"), x = Host.species, y = value)

ggbetweenstats(data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, variable == "MFS.transporter"), x = Host.species, y = value)

trial.aov <- aov(value ~ Host.species, data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, variable == "MFS.transporter"))
TukeyHSD(trial.aov)
summary(trial.aov)[[1]][[1,"Pr(>F)"]]

products_list <- unique(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[c("variable")])
num_unique_products <- "2432" #Got this just by looking at how long the list length was
list_form_products_list <- as.list(products_list)
for (p in list_form_products_list) {
  print(p)
}

as.list(levels(list_form_products_list$variable))
list_form_two_list_form_products_list <- as.list(levels(list_form_products_list$variable))

for (p in list_form_two_list_form_products_list) {
  print(p)
}

pvalues_table <- data.frame(Product=character(length=(num_unique_products)), pvalue=as.numeric(character(length=(num_unique_products))),stringsAsFactors=FALSE)

NO BELOW
for (p in list_form_two_list_form_products_list) {
  product_aov <- aov(value ~ Host.species, data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, variable == (p)))
  prod_pval <- summary(product_aov)[[1]][[1,"Pr(>F)"]]
  new_line <- print((p), `prod_pval`)
  pvalues_table = rbind(pvalues_table, new_line)
  rm(product)
  rm(product_aov)
  rm(prod_pval)
}

for (p in list_form_two_list_form_products_list) {
  product_aov <- aov(value ~ Host.species, data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, variable == (p)))
  prod_pval <- summary(product_aov)[[1]][[1,"Pr(>F)"]]
  print(p, prod_pval)
  rm(product_aov)
  rm(prod_pval)
}

products_as_single_column_data_frame <- data.frame(matrix(unlist(list_form_two_list_form_products_list), nrow=length(list_form_two_list_form_products_list), byrow = TRUE))

products_as_single_column_data_frame %>% rename(better = matrix.unlist.list_form_two_list_form_products_list...nrow...length.list_form_two_list_form_products_list...)

for (row in 1:nrow(products_as_single_column_data_frame)) {
  product <- products_as_single_column_data_frame[row, ]



  matrix.unlist.list_form_two_list_form_products_list...nrow...length.list_form_two_list_form_products_list...
  2261
  Zn.dependent.protease

NO BELOW
for (p in list_form_products_list) {
  product <- print(p)
  product_aov <- aov(value ~ Host.species, data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, variable == (p)))
  prod_pval <- summary(product_aov)[[1]][[1,"Pr(>F)"]]
  #`pvalues_table` %>% add_row(`pvalues_table`, Product = `product`, pvalue = `prod_pval`) -> `pvalues_table`
  #`pvalues_table` %>% add_row(Product = `product`, pvalue = `prod_pval`)
  #`pvalues_table` <- add_row(Product = `product`, pvalue = `prod_pval`)
  new_line <- print(`product`, `prod_pval`)
  pvalues_table = as.data.frame(rbind(pvalues_table, new_line), stringsAsFactors = FALSE)
  rm(product)
  rm(product_aov)
  rm(prod_pval)
}

NO BELOW - new bad
for (p in list_form_products_list) {
  product <- print(p)
  product_aov <- aov(value ~ Host.species, data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, variable == (p)))
  prod_pval <- summary(product_aov)[[1]][[1,"Pr(>F)"]]
  new_line <- c(product, prod_pval)
  pvalues_table[p, ] <- new_line
  rm(product)
  rm(product_aov)
  rm(prod_pval)
  rm(new_line)
}

NO BELOW
for (p in list_form_products_list) {
  product <- print(p)
  product_aov <- aov(value ~ Host.species, data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, variable == (p)))
  prod_pval <- summary(product_aov)[[1]][[1,"Pr(>F)"]]
  print(product, prod_pval)
  rm(product)
  rm(product_aov)
  rm(prod_pval)
}


product <- "ABC.transporter"
trial.aov <- aov(value ~ Host.species, data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, variable == `product`))
TukeyHSD(trial.aov)
prod_pval <- summary(trial.aov)[[1]][[1,"Pr(>F)"]]
`pvalues_table` %>% add_row(Product = `product`, pvalue = `prod_pval`) -> `pvalues_table`


2022-05-04
ggbetweenstats(data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, variable == "MFS.transporter"), x = Host.species, y = value)
trial.aov <- aov(value ~ Host.species, data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, variable == "MFS.transporter"))
TukeyHSD(trial.aov)
summary(trial.aov)[[1]][[1,"Pr(>F)"]]

pvalues_table <- data.frame(Product=character(length=(num_unique_products)), pvalue=as.numeric(character(length=(num_unique_products))),stringsAsFactors=FALSE)

names(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`)[names(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`) == "variable"] <- "Products"

for (p in list_form_two_list_form_products_list) {
  product_aov <- aov(value ~ Host.species, data = dplyr::filter(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`, Products == (p)))
  prod_pval <- summary(product_aov)[[1]][[1,"Pr(>F)"]]
  new_line <- print((p), `prod_pval`)
  pvalues_table = rbind(pvalues_table, new_line)
  rm(product)
  rm(product_aov)
  rm(prod_pval)
}

TESTING NOW
for (p in list_form_two_list_form_products_list) {
  product_aov <- aov(value ~ Host.species, data = `melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$Products == (p) ,] )
  prod_pval <- summary(product_aov)[[1]][[1,"Pr(>F)"]]
  #new_line <- print((p), `prod_pval`)
  new_line <- print(paste((p),`prod_pval`))
  pvalues_table = rbind(pvalues_table, new_line)
  rm(product)
  rm(product_aov)
  rm(prod_pval)
}


for (p in list_form_products_list) {
  product_aov <- aov(value ~ Host.species, data = `melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`$Products == (p) ,] )
  prod_pval <- summary(product_aov)[[1]][[1,"Pr(>F)"]]
  new_line <- print((p), `prod_pval`)
  pvalues_table = rbind(pvalues_table, new_line)
  rm(product)
  rm(product_aov)
  rm(prod_pval)
}

colnames(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`)[1] <- "Isolate"

products_list_2 <- as.character(unique(`melted-2022.05.03_Transposed_all_assemblies_excluding_S125_min_500bp_contig_size_detection_all_potential_products_from_all_faa_files_DFAST_results`[c("Products")]))

list_form_products_list_2 <- as.list(products_list_2)

for (p in list_form_products_list_2) {
  print(p)
}

ggplot(data = melted_marque2[(melted_marque2$variable == "PC1" | melted_marque2$variable == "PC2") & (melted_marque2$value > 0.02 | melted_marque2$value < -0.03) , ], aes(x = variable, y = value, color = variable)) + geom_jitter()

2022-05-13
#All the below comes from https://jmonlong.github.io/Hippocamplus/2017/09/19/mummerplots-with-ggplot2/
readDelta <- function(deltafile){
  lines = scan(deltafile, 'a', sep='\n', quiet=TRUE)
  lines = lines[-1]
  lines.l = strsplit(lines, ' ')
  lines.len = lapply(lines.l, length) %>% as.numeric
  lines.l = lines.l[lines.len != 1]
  lines.len = lines.len[lines.len != 1]
  head.pos = which(lines.len == 4)
  head.id = rep(head.pos, c(head.pos[-1], length(lines.l)+1)-head.pos)
  mat = matrix(as.numeric(unlist(lines.l[lines.len==7])), 7)
  res = as.data.frame(t(mat[1:5,]))
  colnames(res) = c('rs','re','qs','qe','error')
  res$qid = unlist(lapply(lines.l[head.id[lines.len==7]], '[', 2))
  res$rid = unlist(lapply(lines.l[head.id[lines.len==7]], '[', 1)) %>% gsub('^>', '', .)
  res$strand = ifelse(res$qe-res$qs > 0, '+', '-')
  res
}
mumgp = readDelta("13E_S5_against_16aC_S26.delta")
mumgp %>% head %>% kable

#Skip the filtering for now since I can't get the function to work
filterMum <- function(df, minl=1000, flanks=1e4){
    coord = df %>% filter(abs(re-rs)>minl) %>% group_by(qid, rid) %>%
        summarize(qsL=min(qs)-flanks, qeL=max(qe)+flanks, rs=median(rs)) %>%
        ungroup %>% arrange(desc(rs)) %>%
        mutate(qid=factor(qid, levels=unique(qid))) %>% dplyr::select(-rs)
    merge(df, coord) %>% filter(qs>qsL, qe<qeL) %>%
        mutate(qid=factor(qid, levels=levels(coord$qid))) %>% dplyr::select(-qsL, -qeL)
}
mumgp.filt = filterMum(mumgp, minl=1e4)
mumgp.filt %>% head %>% kable

ggplot(mumgp, aes(x=rs, xend=re, y=qs, yend=qe, colour=strand)) + geom_segment() +
    geom_point(alpha=.5) + facet_grid(qid~., scales='free', space='free', switch='y') +
    theme_bw() + theme(strip.text.y=element_text(angle=180, size=5),
                       legend.position=c(.99,.01), legend.justification=c(1,0),
                       strip.background=element_blank(),
                       axis.text.y=element_blank(), axis.ticks.y=element_blank()) +
    xlab('reference sequence') + ylab('assembly') + scale_colour_brewer(palette='Set1')

2022-06-09
`2022.06.07_metaphlan_estimated_reads_mapped_to_known_clades_as_percent_total_reads_in_QCed_cleansed_subsampled_pilot_and_novogene` <- read.csv("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-07_metaphlan_estimated_reads_mapped_to_known_clades_as_percent_total_reads_in_QCed_cleansed_subsampled_pilot_and_novogene.csv")
View(`2022.06.07_metaphlan_estimated_reads_mapped_to_known_clades_as_percent_total_reads_in_QCed_cleansed_subsampled_pilot_and_novogene`)

`06_09_met_read_percent` <- `2022.06.07_metaphlan_estimated_reads_mapped_to_known_clades_as_percent_total_reads_in_QCed_cleansed_subsampled_pilot_and_novogene`
ggplot(data = `06_09_met_read_percent`, aes(x = "Processed read files", y = Estimated_reads_mapped_to_known_clades_by_Metaphlan_as_percentage_reads_in_subsampled_read_file)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_violin(color="black", fill='#8a795d', alpha = 0.65) + geom_boxplot(width = 0.4, fill = '#f2f3f4') + xlab("") + ylab("Estimated mapped reads no. as\npercentage of total input reads (%)")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-09_all_samples_processed_Metaphlan_estimated_reads_mapped_to_known_clade_as_percent_input_reads.svg")

`2022.06.07_Kraken_report_files_percent_reads_unclassified_and_percent_reads_classified_Bacteria_all_min_conf_scores_all_processed_sample_files` <- read.csv("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-07_Kraken_report_files_percent_reads_unclassified_and_percent_reads_classified_Bacteria_all_min_conf_scores_all_processed_sample_files.csv")
View(`2022.06.07_Kraken_report_files_percent_reads_unclassified_and_percent_reads_classified_Bacteria_all_min_conf_scores_all_processed_sample_files`)

`06_09_krak_read_percent` <- `2022.06.07_Kraken_report_files_percent_reads_unclassified_and_percent_reads_classified_Bacteria_all_min_conf_scores_all_processed_sample_files`
`06_09_melted_krak_read_percent` <- reshape2::melt(`06_09_krak_read_percent`, id=c("Sample_ID", "Kraken_minimum_confidence_score_required"))
ggplot(data = `06_09_melted_krak_read_percent`[(`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria"), ], aes(x = Kraken_minimum_confidence_score_required, y = value, group = Kraken_minimum_confidence_score_required)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_violin(color="black", fill='#8a795d', alpha = 0.65, width = 12) + geom_boxplot(width = 1, fill = '#f2f3f4') + xlab("Minimum confidence score required for classification") + ylab("Reads classified as bacteria as percent\nof total input reads (%)") + scale_x_continuous(breaks=seq(0,100,5))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-09_all_samples_processed_kraken_reads_mapping_to_bacteria_as_percent_input_reads.svg")

2022-06-10
`2022.06.07_Kaiju_output_files_percent_reads_classified_and_number_reads_classified_and_unclassified_all_processed_samplescsv` <- read.csv("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-07_Kaiju_output_files_percent_reads_classified_and_number_reads_classified_and_unclassified_all_processed_samplescsv.csv")
View(`2022.06.07_Kaiju_output_files_percent_reads_classified_and_number_reads_classified_and_unclassified_all_processed_samplescsv`)
`06_10_kaiju_read_percent` <- `2022.06.07_Kaiju_output_files_percent_reads_classified_and_number_reads_classified_and_unclassified_all_processed_samplescsv`
`06_10_melted_kaiju_read_percent` <- reshape2::melt(`06_10_kaiju_read_percent`, id=c("Sample_ID", "Errors_allowed", "Classified_reads", "Unclassified_reads"))
ggplot(data = `06_10_melted_kaiju_read_percent`, aes(x = Errors_allowed, y = value, group = Errors_allowed)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_violin(color="black", fill='#8a795d', alpha = 0.65, width = 1) + geom_boxplot(width = 0.5, fill = '#f2f3f4') + scale_x_continuous(breaks=seq(0,20,1)) + xlab("Allowed errors in match") + ylab("Percent input reads reads classified (%)")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-10_all_samples_processed_kaiju_reads_classified_as_percent_input_reads.svg")

p1 <- ggplot(data = `06_09_met_read_percent`, aes(x = "Processed read files", y = Estimated_reads_mapped_to_known_clades_by_Metaphlan_as_percentage_reads_in_subsampled_read_file)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_violin(color="black", fill='#8a795d', alpha = 0.65) + geom_boxplot(width = 0.4, fill = '#f2f3f4') + xlab("") + ylab("Estimated mapped reads no. as\npercentage of total input reads (%)")
p2 <- ggplot(data = `06_09_melted_krak_read_percent`[(`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria"), ], aes(x = Kraken_minimum_confidence_score_required, y = value, group = Kraken_minimum_confidence_score_required)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_violin(color="black", fill='#8a795d', alpha = 0.65, width = 12) + geom_boxplot(width = 1, fill = '#f2f3f4') + xlab("Minimum confidence score required\nfor classification (%)") + ylab("Reads classified as bacteria as percent\nof total input reads (%)") + scale_x_continuous(breaks=seq(0,100,5), guide = guide_axis(n.dodge = 2))
p3 <- ggplot(data = `06_10_melted_kaiju_read_percent`, aes(x = Errors_allowed, y = value, group = Errors_allowed)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_violin(color="black", fill='#8a795d', alpha = 0.65, width = 1) + geom_boxplot(width = 0.5, fill = '#f2f3f4') + scale_x_continuous(breaks=seq(0,20,1), guide = guide_axis(n.dodge = 2)) + xlab("Allowed errors in match") + ylab("Percent input reads classified (%)")
ggarrange(p1, p2, p3, labels = c("A", "B", "C"), ncol = 3, nrow = 1)
grid.arrange(p1, p2, p3, nrow = 2)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-12_all_samples_processed_metaphlan_kraken_kaiju_percent_reads_classified.svg")

2022-06-20
setwd("~/Google Drive/PhD/2022-Paper_2-Chapter_3")
`095_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/095_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file.txt")
View(`095_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`)
transpose_trial_095 <- t(`095_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`)
rownames(`transpose_trial_095`) <- colnames(`095_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`)

melted_transposed_trial_095 <- reshape2::melt(`transpose_trial_095`, id=c("832", "833", "834", "835", "836"))

testor.df <- (`melted_transposed_trial_095`[(`melted_transposed_trial_095`$Var1 == "Taxid") & (`melted_transposed_trial_095`$Var2 == "2"), ])

ggplot(data = `06_09_melted_krak_read_percent`[(`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria"), ], aes(x = Kraken_minimum_confidence_score_required, y = value, group = Kraken_minimum_confidence_score_required)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_violin(color="black", fill='#8a795d', alpha = 0.65, width = 12) + geom_boxplot(width = 1, fill = '#f2f3f4') + xlab("Minimum confidence score required for classification") + ylab("Reads classified as bacteria as percent\nof total input reads (%)") + scale_x_continuous(breaks=seq(0,100,5))

V2_melted_transposed_trial_095 <- melted_transposed_trial_095 %>% row_to_names(row_number = 1)

melted_transposed_trial_095 <- reshape2::melt(`transpose_trial_095`, id=c("Taxid", "Taxid_rank", "834", "835", "836"))

2022-06-24
`2022.06.23_20_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxID_percentages` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-23_20_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxID_percentages.tsv")
View(`2022.06.23_20_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxID_percentages`)

max_20_errors_species_level_kaiju_report_percentages <- `2022.06.23_20_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxID_percentages`
`max_20_errors_species_level_kaiju_report_percentages` <- data.table::transpose(`max_20_errors_species_level_kaiju_report_percentages`)

`headers_max_20_errors_species_level_kaiju_report_percentages` <- janitor::row_to_names(`max_20_errors_species_level_kaiju_report_percentages`, 1, remove_row = TRUE, remove_rows_above = TRUE)

`melted_max_20_errors_species_level_kaiju_report_percentages` <- reshape2::melt(`headers_max_20_errors_species_level_kaiju_report_percentages`, id=c("Sample_ID", "Host", "Season", "Host_X_season", "Sequencing"))

2022-06-27
`2022.04.17_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-17_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted.tsv")

`2022.04.17.mapping_to_good_bins` <- `2022.04.17_TRANSPOSED_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted`
`2022.04.17.logp1_adjusted_mapping_to_good_bins` <- log1p(`2022.04.17.mapping_to_good_bins`[,c(2:349)])
`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species <- `2022.04.17.mapping_to_good_bins`$Species
`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Season <- `2022.04.17.mapping_to_good_bins`$Season
`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Sequencing <- `2022.04.17.mapping_to_good_bins`$Sequencing
`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species_x_season <- `2022.04.17.mapping_to_good_bins`$Species_x_season
`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Bins <- `2022.04.17.mapping_to_good_bins`$Bins

`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins` <- prcomp(`2022.04.17.logp1_adjusted_mapping_to_good_bins`[,c(1:348)], center = TRUE, scale. = FALSE)

`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins` <- prcomp(`2022.04.17.logp1_adjusted_mapping_to_good_bins`[,c(1:348)], center = TRUE, scale. = FALSE)

`2022-04-17_PC1_vs_PC2_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` <- ggbiplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Season), color=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-04-17_PC1_vs_PC3_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` <- ggbiplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Season), color=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-04-17_PC2_vs_PC3_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` <- ggbiplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Season), color=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-04-17_PC1_vs_PC2_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` + `2022-04-17_PC1_vs_PC3_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` + `2022-04-17_PC2_vs_PC3_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contgs` + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_top_3_PCs_three_plots_PCA_ggbiplot_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

fviz_eig(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`)
fviz_cluster(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`)

fviz_pca_biplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, repel = TRUE, select.var = list(contrib = 5))

fviz_pca_biplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, repel = TRUE, select.var = list(contrib = 5), font.family = "Tex Gyre Heroes")

fviz_pca_biplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, repel = TRUE, select.var = list(contrib = 15), font.family = "Tex Gyre Heroes", col.ind = `2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species, palette = c('#0073ff', '#ff8d1e'), addEllipses = TRUE, ellipse.type = "confidence", legend.title = "Host species")

fviz_pca_var(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, col.var = "contrib", gradient.cols = c("#0f0f0f", "#3b0078", "#6c00da", "#8916ff", "#a751ff", "#cf9fff"), repel = TRUE, select.var = list(contrib = 40), font.family = "Tex Gyre Heroes")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_PCA_var_plot_top_40_PCs_1_and_2_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

fviz_pca_var(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, col.var = "contrib", gradient.cols = c("#0f0f0f", "#3b0078", "#6c00da", "#8916ff", "#a751ff", "#cf9fff"), repel = TRUE, select.var = list(contrib = 30), axes = c(1, 3), font.family = "Tex Gyre Heroes")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_PCA_var_plot_top_30_PCs_1_and_3_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

fviz_pca_var(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, col.var = "contrib", gradient.cols = c("#0f0f0f", "#3b0078", "#6c00da", "#8916ff", "#a751ff", "#cf9fff"), repel = TRUE, select.var = list(contrib = 20), axes = c(2, 3), font.family = "Tex Gyre Heroes")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-18_PCA_var_plot_top_20_PCs_2_and_3_all_samples_log1p_transformed_kilobase_and_lib_size_adjusted_summed_mapping_scores_min_80_percent_complete_less_5_percent_contaminated_bin_contigs.svg")

`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins-Var` <- get_pca_var(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`)

corr_cross(`temp_table`, max_pvalue = 0.05, top = 10)

temp_table <- `2022.04.17.logp1_adjusted_mapping_to_good_bins`[,c(1:348)]

`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`
print(`2022.04.17.logp1_adjusted_mapping_to_good_bins`(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, columns = data_scaled, index = ['PC-1', 'PC-2']))

print (`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`.explained_variance_ratio)
summary(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`)


`2022.04.17_TRANSPOSED_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-17_TRANSPOSED_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted.txt")
 View(`2022.04.17_TRANSPOSED_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted`)
`2022-06-27_TRANSPOSED_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted` <- `2022.04.17_TRANSPOSED_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted`

`2022-06-27_TRANSPOSED_all` <- `2022-06-27_TRANSPOSED_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted`

`2022-06-27_TRANSPOSED_all_proto` <- `2022-06-27_TRANSPOSED_all`[,-1]
rownames(`2022-06-27_TRANSPOSED_all_proto`) <- `2022-06-27_TRANSPOSED_all`[,1]
tmm_norm_test <- calcNormFactors(`2022-06-27_TRANSPOSED_all_proto`[,c(1:348)], method = "TMM")

2022-06-28
1.389811
0.2250855

2022-06-28
rm(tmm_norm_test)
AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats.txt", header=FALSE)
View(AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats)

proto_AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats <- AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats[,-1]
rownames(proto_AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats) <- AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats[,1]
tmm_norm_test <- calcNormFactors(proto_AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats, method = "TMM")
only_counts_proto_AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats <- as.matrix(proto_AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats[,2])
rownames(only_counts_proto_AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats) <- AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats[,1]

yarp <- DGEList(counts=AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats$V3, genes=AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats(Length=V2))

AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats.txt", header=FALSE, row.names=1)

colnames(AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats) <- c("Contig_length", "Mapped_read_segments", "Unmapped_read_segments")
dgelist <- AC19N_BWA_against_concatenated_GOOD_bins.Mapped.Sorted.Only_reads.IDX_stats %>% as.data.frame() %>% DGEList() %>% calcNormFactors()

`2022.06.28_all_subsampled_reads_unadjusted_per_contig_mapped_segment_counts_all_subsampled_read_files` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-28_all_subsampled_reads_unadjusted_per_contig_mapped_segment_counts_all_subsampled_read_files.tsv", row.names=1)

2022-06-29
view(`2022.04.17_TRANSPOSED_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted`)

`2022_06_29_transposed_summed_lib_cont_adjusted_all_samples` <- `2022.04.17_TRANSPOSED_summing_mapping_contigs_all_GOOD_bins_all_subsampled_sample_files_contig_and_library_size_adjusted`

`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples` <- reshape2::melt(`2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`, id=c("Bins", "Species", "Season", "Sequencing", "Species_x_season"))

`2022_06_29_PCA_transposed_summed_lib_cont_adjusted_all_samples` <- prcomp(`2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`[,c(2:349)], center = TRUE, scale. = FALSE)

ggbiplot(`2022_06_29_PCA_transposed_summed_lib_cont_adjusted_all_samples`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`$Season), color=factor(`2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "No transformation")

ggbiplot(`2022_06_29_PCA_transposed_summed_lib_cont_adjusted_all_samples`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`$Season), color=factor(`2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "No transformation")

ggbiplot(`2022_06_29_PCA_transposed_summed_lib_cont_adjusted_all_samples`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`$Season), color=factor(`2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "No transformation")

2022-06-29_log1p_transformed_lib_contig_adjusted_good_bin_mapping_counts

fviz_pca_biplot(`2022_06_29_PCA_transposed_summed_lib_cont_adjusted_all_samples`, repel = TRUE, select.var = list(contrib = 15), font.family = "Tex Gyre Heroes", col.ind = `2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`$Species, palette = c('#0073ff', '#ff8d1e'), addEllipses = TRUE, ellipse.type = "confidence", legend.title = "Host species")

ggbiplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Season), color=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#3178ce', "AR" = '#ce8731')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) +labs(title = "Log1p transformed")
ggbiplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Season), color=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#3178ce', "AR" = '#ce8731')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) +labs(title = "Log1p transformed")
ggbiplot(`2022.04.17.PCA.logp1_adjusted_mapping_to_good_bins`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Season), color=factor(`2022.04.17.logp1_adjusted_mapping_to_good_bins`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#3178ce', "AR" = '#ce8731')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) +labs(title = "Log1p transformed")

ggpubr::ggboxplot(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c113", ], x = "Species", y = "value", color = "Species", palette = c("#0073ff", "#ff8c00"), ylab = "Library and contig size adjusted mapping count", xlab = "Host species") + stat_compare_means(method = 'kruskal.test', label="p.signif")

host_x_season_comparisons <- list( c ("ACJ", "ACN"), c("ACJ", "ARJ"), c("ACJ", "ARN"), c("ACN", "ARJ"), c("ACN", "ARN"), c("ARJ", "ARN") )
ggpubr::ggboxplot(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m1981", ], x = "Species_x_season", y = "value", color = "Species_x_season", palette = c("#009eff", "#0024b3", "#ffcc00", "#e67e00"), ylab = "Library and contig size adjusted mapping count", xlab = "Host species") + stat_compare_means(comparisons = host_x_season_comparisons, method = 'kruskal.test', label="p.signif")

ggpubr::ggboxplot(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m1981", ], x = "Species_x_season", y = "value", color = "Species_x_season", palette = c("#009eff", "#0024b3", "#ffcc00", "#e67e00"), ylab = "Library and contig size adjusted mapping count", xlab = "Host species") + stat_compare_means(comparisons = host_x_season_comparisons, method = 'kruskal.test'

`2022_06_29_all_unique_bins` <- as.character(unique(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[c("variable")]))

`2022_06_29_proto_all_unique_bins` <- unique(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[c("variable")])

`2022_06_29_proto_all_unique_bins`$variable

`2022_06_29_list_all_unique_bins` <- as.list(`2022_06_29_proto_all_unique_bins`)
`2022_06_29_list_all_unique_bins` <- as.list(as.character(`2022_06_29_proto_all_unique_bins`))

print(`2022_06_29_list_all_unique_bins`)

for (i in 1:length(`2022_06_29_list_all_unique_bins`)){
  print(`2022_06_29_list_all_unique_bins`[[i][1]])
}

for (i in `2022_06_29_list_all_unique_bins`){
  print("This is a bin i")
}

kruskal.test(value ~ Species, data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c912", ])

ggbetweenstats(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c113", ], x = Species, y = value, theme(text = element_text(family = "Tex Gyre Heroes")), fill = "Species", palette = c("#0073ff", "#ff8c00"))
ggpubr::ggboxplot(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c113", ], x = "Species", y = "value", color = "Species", palette = c("#0073ff", "#ff8c00"), ylab = "Library and contig size adjusted mapping count", xlab = "Host species") + stat_compare_means(method = 'kruskal.test', label="p.signif")
ggpubr::ggviolin(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c113", ], x = "Species", y = "value", fill = "Species", palette = c("#0073ff", "#ff8c00"), add = "boxplot", add.params = list(fill = "white"), ylab = "Library and contig size adjusted mapping count", xlab = "Host species") + stat_compare_means(method = 'wilcox', label="p.signif", label.y = 10)

ggbetweenstats(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c113", ], x = Species, y = value)


ggpubr::ggboxplot(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c113", ], x = "Species_x_season", y = "value", color = "Species_x_season", palette = c("#009eff", "#0024b3", "#ffcc00", "#e67e00"), ylab = "Library and contig size adjusted mapping count", xlab = "Host species") + stat_compare_means(comparisons = host_x_season_comparisons, method = 'kruskal.test')

PLACEHOLDER_ACJ_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ACJ")$value ; PLACEHOLDER_ACN_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ACN")$value ; PLACEHOLDER_ARJ_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ARJ")$value ; PLACEHOLDER_ARN_counts <- subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER" & `melted_log1p_group_by_bins`$Host_by_season == "ARN")$value ; PLACEHOLDER_bin_name <- "PLACEHOLDER" ; PLACEHOLDER_AC_J_vs_N_pvalue <- wilcox.test(PLACEHOLDER_ACJ_counts, PLACEHOLDER_ACN_counts, alternative = "two.sided", exact =FALSE)$p.value ; PLACEHOLDER_AR_J_vs_N_pvalue <- wilcox.test(PLACEHOLDER_ARJ_counts, PLACEHOLDER_ARN_counts, alternative = "two.sided", exact =FALSE)$p.value ; PLACEHOLDER_ACJ_vs_ARJ_pvalue <- wilcox.test(PLACEHOLDER_ACJ_counts, PLACEHOLDER_ARJ_counts, alternative = "two.sided", exact =FALSE)$p.value ; PLACEHOLDER_ACN_vs_ARN_pvalue <- wilcox.test(PLACEHOLDER_ACN_counts, PLACEHOLDER_ARN_counts, alternative = "two.sided", exact =FALSE)$p.value ; `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[nrow(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_J_vs_N_pvalue, PLACEHOLDER_AR_J_vs_N_pvalue, PLACEHOLDER_ACJ_vs_ARJ_pvalue, PLACEHOLDER_ACN_vs_ARN_pvalue) ; `2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[, c(2:5)] <- sapply(`2021_09_08_summed_to_bins_wilcox_test_P_values_within_host_seasonal_and_across_host_same_season_comparisons`[, c(2:5)], as.numeric) ; rm(PLACEHOLDER_ACJ_counts) ; rm(PLACEHOLDER_ACN_counts) ; rm(PLACEHOLDER_ARJ_counts) ; rm(PLACEHOLDER_ARN_counts) ; rm(PLACEHOLDER_AC_J_vs_N_pvalue) ; rm(PLACEHOLDER_AR_J_vs_N_pvalue) ; rm(PLACEHOLDER_ACJ_vs_ARJ_pvalue) ; rm(PLACEHOLDER_ACN_vs_ARN_pvalue) ; PLACEHOLDER_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host <- ggplot(subset(`melted_log1p_group_by_bins`, `melted_log1p_group_by_bins`$variable == "PLACEHOLDER"), aes(x = Host_by_season, y = value)) + theme_linedraw() + geom_jitter(aes(color = Host_by_season), width = 0.35, size = 2.25) + xlab("Host and season grouping") + ylab("Log1p transformed counts") + scale_color_manual(name="Host species and season", values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + scale_fill_manual(values=c('#00c2cc', '#0074ff', '#ff8c00', '#b16100')) + labs(title = "Log1p transformed, contig and library size adjusted mapped read segment counts summed by bin of\norigin for PLACEHOLDER according to host and season groupings", caption = "Wilcox test by stat_compare_means") + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_boxplot(aes(color = Host_by_season), alpha = 0.25) + stat_compare_means(method = "wilcox.test", comparisons = host_and_season_within_and_across_comparisons) ; ggsave("PLACEHOLDER_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host.pdf") ; ggsave("PLACEHOLDER_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host.eps") ; rm(`PLACEHOLDER_ggplot_boxplot_and_wilcox_test_within_host_and_same_season_across_host`) ; rm(PLACEHOLDER_bin_name)

p.adjust(2.644304e-03, method = "bonferroni", n = length(4))
p.adjust(2.644304e-03, method = "hochberg", n = length(4))
p.adjust(2.644304e-03, method = "BY", n = length(4))

rm(ls(pattern='Bin_c113*'))

`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species` <- data.frame(matrix(ncol = 2, nrow = 0, dimnames = list(NULL, c("Bin name", "Uncorrected pvalue"))))
`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season` <- data.frame(matrix(ncol = 2, nrow = 0, dimnames = list(NULL, c("Bin name", "Uncorrected pvalue"))))


PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_host_kruskal <- kruskal.test(value ~ Species, data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "PLACEHOLDER", ])
PLACEHOLDER_season_kruskal <- kruskal.test(value ~ Season, data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "PLACEHOLDER", ])
PLACEHOLDER_host_pvalue <- PLACEHOLDER_host_kruskal$p.value
PLACEHOLDER_season_pvalue <- PLACEHOLDER_season_kruskal$p.value
`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[nrow(`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_host_pvalue)
`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[, c(2)] <- sapply(`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[, c(2)], as.numeric)
`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[nrow(`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_season_pvalue)
`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[, c(2)] <- sapply(`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[, c(2)], as.numeric)
rm(PLACEHOLDER_bin_name)
rm(PLACEHOLDER_host_kruskal)
rm(PLACEHOLDER_season_kruskal)
rm(PLACEHOLDER_host_pvalue)
rm(PLACEHOLDER_season_pvalue)


`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_x_season` <- data.frame(matrix(ncol = 2, nrow = 0, dimnames = list(NULL, c("Bin name", "Uncorrected pvalue"))))
PLACEHOLDER_host_x_season_kruskal <- kruskal.test(value ~ Species_x_season, data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "PLACEHOLDER", ])
PLACEHOLDER_host_x_season_pvalue <- PLACEHOLDER_host_x_season_kruskal$p.value

/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/
`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`

write.table(`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species.tsv", sep="\t", row.names=FALSE)
write.table(`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season.tsv", sep="\t", row.names=FALSE)

`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species` <- `2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[order(`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Uncorrected.pvalue),]
`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season` <- `2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[order(`2022_06_29_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Uncorrected.pvalue),]

`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues <- p.adjust(`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Uncorrected.pvalue, method = "BH")
`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$BH_corrected_pvalues <- p.adjust(`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Uncorrected.pvalue, method = "BH")

write.table(`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_06_29_BH_corrected_added_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season.tsv", sep="\t", row.names=FALSE)
write.table(`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_06_29_BH_corrected_added_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species.tsv", sep="\t", row.names=FALSE)

Bin_m1474

ggbetweenstats(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m1474", ], x = Season, y = value, fill = Season, palette = c("#E1BE6A", "#40B0A6"), theme(text = element_text(family = "Tex Gyre Heroes")))
ggpubr::ggboxplot(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m1474", ], x = "Season", y = "value", fill = "Season", palette = c("#E1BE6A", "#40B0A6"), ylab = "Library and contig size adjusted mapping count", xlab = "Sample season") + stat_compare_means(method = 'kruskal.test', label="p.signif")

`2022_06_29_BH_corrected_Pvalues_below_005_sampling_season_Kruskal_wallis_tested` <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$BH_corrected_pvalues < 0.05, ]
`2022_06_29_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested` <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05, ]

2022-06-30
Bin_m595

ggbetweenstats(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m595", ], x = Species_x_season, y = value, fill = Species_x_season, theme(text = element_text(family = "Tex Gyre Heroes")))

my_comparisons_host_x_season_limited <- list( c("ACJ", "ACN"), c("ARJ", "ARN"), c("ACJ", "ARJ"), c("ACN", "ARN"))

ggpubr::ggviolin(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1500", ], x = "Species_x_season", y = "value", fill = "Species_x_season", ylab = "Library and contig size adjusted mapping count", xlab = "Host species and sample season", add = "boxplot", add.params = list(fill = "#f2f3f4")) + stat_compare_means(comparisons = my_comparisons_host_x_season_limited, method = 'kruskal.test', label="p.signif") + stat_compare_means(label.y = 1550) + theme_linedraw()

kruskal.test(value ~ Species_x_season, data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1500", ])
Bin_m595

ggpubr::ggviolin(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m1474", ], x = "Species_x_season", y = "value", fill = "Species_x_season", ylab = "Library and contig size adjusted mapping count", xlab = "Host species and sample season", add = "boxplot", add.params = list(fill = "#f2f3f4")) + stat_compare_means(comparisons = my_comparisons_host_x_season_limited, method = 'kruskal.test', label="p.signif") + stat_compare_means(label.y = 1550) + theme_linedraw()

`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_host_species_comparisons` <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05, ]
`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_sampling_season_comparisons` <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$BH_corrected_pvalues < 0.05, ]

`%notin%`

subset(`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_host_species_comparisons`, (`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_host_species_comparisons`$Bin.name %notin% `2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_sampling_season_comparisons`$Bin.name))
Bin_c1687 should be host effect not season
`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season` <- subset(`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_host_species_comparisons`, (`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_host_species_comparisons`$Bin.name %notin% `2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_sampling_season_comparisons`$Bin.name))


ggpubr::ggviolin(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687", ], x = "Species_x_season", y = "value", fill = "Species_x_season", ylab = "Library and contig size adjusted mapping count", xlab = "Host species and sample season", add = "boxplot", add.params = list(fill = "#f2f3f4")) + stat_compare_means(method = 'kruskal.test', label="p.signif") + stat_compare_means(label.y = 1550) + theme_linedraw()

ggpubr::ggviolin(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687", ], x = "Species", y = "value", fill = "Species", ylab = "Library and contig size adjusted mapping count", xlab = "Host species", add = "boxplot", add.params = list(fill = "#f2f3f4")) + stat_compare_means(method = "kruskal.test", label="p.signif", paired = FALSE)
ggpubr::ggviolin(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1272", ], x = "Species", y = "value", fill = "Species", ylab = "Library and contig size adjusted mapping count", xlab = "Host species", add = "boxplot", add.params = list(fill = "#f2f3f4")) + stat_compare_means(method = "kruskal.test", label="p.signif", paired = FALSE)


`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host` <- subset(`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_sampling_season_comparisons`, (`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_sampling_season_comparisons`$Bin.name %notin% `2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_host_species_comparisons`$Bin.name))
Bin_c1272 should be season effect not host

ggpubr::ggviolin(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1272", ], x = "Species_x_season", y = "value", fill = "Species_x_season", ylab = "Library and contig size adjusted mapping count", xlab = "Host species and sample season", add = "boxplot", add.params = list(fill = "#f2f3f4")) + stat_compare_means(method = 'kruskal.test', label="p.signif") + stat_compare_means(label.y = 1550) + theme_linedraw()

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1272"), ], aes(x = Species_x_season, y = value)) + geom_boxplot()

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1272"), ], aes(x = Species, y = value)) + geom_boxplot()
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1272"), ], aes(x = Season, y = value)) + geom_boxplot() + stat_compare_means(method = "kruskal.test", label="p.signif") + stat_compare_means(label.y = 300)

kruskal.test(value ~ Species, data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1272", ])

ggpubr::ggviolin(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687", ], x = "Species", y = "value", fill = "Species", palette = c("#1e90ff", "#ff8d1e"), ylab = "Library and contig size adjusted mapping count", xlab = "Host species", add = "boxplot", add.params = list(fill = "#f2f3f4")) + stat_compare_means(method = "kruskal.test", label="p.signif", paired = FALSE)


ggpubr::ggviolin(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1500", ], x = "Species", y = "value", fill = "Species", palette = c("#1e90ff", "#ff8d1e"), group = "Bin.name", ylab = "Library and contig size adjusted mapping count", xlab = "Host species", add = "boxplot", add.params = list(fill = "#f2f3f4")) + stat_compare_means(method = "kruskal.test", label="p.signif", paired = FALSE)

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1500", ], aes(x = variable, y = value, group = Species, fill = Species)) + geom_boxplot()

mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687" & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR", ][,7])
mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687" & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC", ][,7])

mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1272" & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR", ][,7])
mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1272" & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC", ][,7])

ggplot(data = `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`[`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$BH_corrected_pvalues < 0.05])

subset(merge(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`, `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`, by.x = "variable", by.y = "BH_corrected_pvalues"), BH_corrected_pvalues < 0.05)

`2022-06-30_temp_merged_test` <- subset(merge(`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`, `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`), BH_corrected_pvalues < 0.05)


`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season` <- subset(`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_host_species_comparisons`, (`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_host_species_comparisons`$Bin.name %notin% `2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_sampling_season_comparisons`$Bin.name))
`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host` <- subset(`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_sampling_season_comparisons`, (`2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_sampling_season_comparisons`$Bin.name %notin% `2022_06_30_bins_BH_corrected_Kruskal_Wallis_pvalues_under_005_host_species_comparisons`$Bin.name))


write.table(`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season.tsv", sep="\t", row.names=FALSE)
write.table(`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host.tsv", sep="\t", row.names=FALSE)

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522", ], aes(x = value, y = Species)) + geom_density_ridges()
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522", ], aes(x = value, y = Species)) + geom_density_ridges2()
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522", ], aes(x = value, y = Species)) + geom_density_ridges2(aes(point_colour = Species, point_fill = Species, point_shape = Species), jittered_points = TRUE, position = "raincloud", alpha = 0.7) + scale_point_colour_hue(l = 40) + scale_discrete_manual(aesthetics = "point_shape", values = c("#ff8d1e", "#1e90ff"))
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522", ], aes(x = value, y = Species)) + geom_density_ridges2(aes(point_colour = Species, point_fill = Species, point_shape = Species), jittered_points = TRUE, alpha = 0.7) + scale_point_colour_hue(l = 40) + scale_discrete_manual(aesthetics = "point_shape", values = c("#ff8d1e", "#1e90ff"))
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522", ], aes(x = value, y = Species)) + geom_density_ridges2(aes(fill = Species, point_colour = Species, point_fill = Species, point_shape = Species), jittered_points = TRUE, alpha = 0.6, point_size = 4) + scale_fill_manual(values = c("#ff8d1e", "#1e90ff")) + scale_discrete_manual(aesthetics = "point_shape", values = c("#6c3600", "#00366c")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522", ], aes(x = value, y = Species, fill = Species, shape = Species, colour = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.70, point_size = 3.25) + scale_fill_manual(values = c("#ff8d1e", "#1e90ff")) + scale_colour_manual(values = c ("#6c3600", "#00366c")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522", ], aes(x = value, y = Species, fill = Species, shape = Species, colour = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.70, point_size = 3.25) + scale_fill_manual(values = c("#ff8d1e", "#1e90ff")) + scale_colour_manual(values = c ("#6c3600", "#00366c")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c665", ], aes(x = value, y = Species, group = variable, fill = Species, shape = Species, colour = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.70, point_size = 3.25) + scale_fill_manual(values = c("#ff8d1e", "#1e90ff")) + scale_colour_manual(values = c ("#6c3600", "#00366c")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c665", ], aes(x = value, y = Species, group_by = variable, fill = Species, shape = Species, colour = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.70, point_size = 3.25) + scale_fill_manual(values = c("#ff8d1e", "#1e90ff")) + scale_colour_manual(values = c ("#6c3600", "#00366c")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c665", ], aes(x = value, y = Species, fill = Species, shape = Species, colour = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.70, point_size = 3.25) + scale_fill_manual(values = c("#ff8d1e", "#1e90ff")) + scale_colour_manual(values = c ("#6c3600", "#00366c")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable)

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c665" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687", ], aes(x = value, y = Species, fill = Species, shape = Species, colour = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.70, point_size = 1.75) + scale_fill_manual(values = c("#ff8d1e", "#1e90ff")) + scale_colour_manual(values = c ("#6c3600", "#00366c")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable)


ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c665" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687", ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season, colour = Species_x_season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75) + scale_fill_manual(values = c("#ff8d1e", "#1e90ff", "#d48e49", "#5b8fc2")) + scale_colour_manual(values = c ("#6c3600", "#00366c", "#d48e49", "#5b8fc2")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable)

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c665" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687", ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75) + scale_fill_manual(values = c("#80c0ff", "#ff8d1e", "#5b8fc2", "#d48e49")) + scale_colour_manual(values = c ("#1e20ff", "#ff8d1e", "#5b8fc2", "#d48e49")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable)

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c665" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687", ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75) + scale_fill_manual(values = c("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + scale_colour_manual(values = c ("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable)

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c665" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687", ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75) + scale_fill_manual(values = c("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + scale_colour_manual(values = c ("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable) + scale_shape_manual(values = c("0", "15", "2", "17"))

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c665" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687", ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75) + scale_fill_manual(values = c("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + scale_colour_manual(values = c ("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable) + scale_shape_manual(values = c("0", "15", "2", "17")) + scale_y_discrete(expand = expand_scale(mult = c(0.01, .425)))

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c665" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687", ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75) + scale_fill_manual(values = c("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + scale_colour_manual(values = c ("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable) + scale_shape_manual(values = c("0", "15", "2", "17")) + scale_y_discrete(expand = expand_scale(mult = c(0.01, .425)))

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c665" | `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1687", ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75) + scale_fill_manual(values = c("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + scale_colour_manual(values = c ("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_discrete_manual(aesthetics = "point_shape", values = c("0", "15", "2", "17")) + facet_wrap(~variable) + scale_y_discrete(expand = expand_scale(mult = c(0.01, .425)))

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75) + scale_fill_manual(values = c("#bbddff", "#0a85ff")) + scale_colour_manual(values = c ("#bbddff", "#0a85ff")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_discrete_manual(aesthetics = "point_shape", values = c("0", "15")) + facet_wrap(~variable) + scale_y_discrete(expand = expand_scale(mult = c(0.01, .425)))

Bin_c9*

`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & grepl(glob2rx("Bin_c9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & grepl(glob2rx("Bin_c9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable) + scale_y_discrete(expand = expand_scale(mult = c(0.01, .425)))

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, nrow = 3) + scale_y_discrete(expand = expand_scale(mult = c(0.01, .425)))

#Exemplar host specific effect bins matching a grepl pattern and faceted
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Host species")

#Exemplar season specific effect bins matching a grepl pattern and faceted
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Sampling month")

2022-07-01
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin") + ylab("Host species")

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.60, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin") + ylab("Sampling month")

`2022.07.01_all_subsampled_contig_level_library_and_contig_size_adjusted_mapped_segment_counts` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-01_all_subsampled_contig_level_library_and_contig_size_adjusted_mapped_segment_counts.tsv", row.names=1)
View(`2022.07.01_all_subsampled_contig_level_library_and_contig_size_adjusted_mapped_segment_counts`)

trial_dataframe_transposed <- t(`2022.07.01_all_subsampled_contig_level_library_and_contig_size_adjusted_mapped_segment_counts`)

intermediate_transposed_dataframe <- as.data.frame(`trial_dataframe_transposed`)
safehold_intermediate_transposed_dataframe <- intermediate_transposed_dataframe
intermediate_transposed_dataframe <- intermediate_transposed_dataframe %>% mutate_at(c(1:79380), as.numeric)
intermediate_transposed_dataframe <- cbind(intermediate_transposed_dataframe, Sample_ID=`2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`$Bins)
`2022_07_01_safehold_intermediate_transposed_dataframe` <- safehold_intermediate_transposed_dataframe
`2022_07_01_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments` <- intermediate_transposed_dataframe
write.table(`2022_07_01_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_01_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments_evening_version.tsv", sep="\t", row.names=FALSE)
`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments` <- reshape2::melt(`2022_07_01_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`, id=c("Species", "Season", "Sequencing", "Species_x_season", "Sample_ID"))
write.table(`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments.tsv", sep="\t", row.names=FALSE)

`2022_07_01_PCA_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments` <- prcomp(`2022_07_01_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[,c(1:79380)], center = TRUE, scale. = FALSE)

ggbiplot(`2022_07_01_PCA_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022_07_01_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022_07_01_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$Season), color=factor(`2022_07_01_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "No transformation")

`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species` <- data.frame(matrix(ncol = 2, nrow = 0, dimnames = list(NULL, c("Contig name", "Uncorrected pvalue"))))
`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season` <- data.frame(matrix(ncol = 2, nrow = 0, dimnames = list(NULL, c("Contig name", "Uncorrected pvalue"))))

PLACEHOLDER_contig_name <- "PLACEHOLDER"
PLACEHOLDER_contig_host_kruskal <- kruskal.test(value ~ Species, data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable == "PLACEHOLDER", ])
PLACEHOLDER_contig_season_kruskal <- kruskal.test(value ~ Season, data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable == "PLACEHOLDER", ])
PLACEHOLDER_contig_host_pvalue <- PLACEHOLDER_contig_host_kruskal$p.value
PLACEHOLDER_contig_season_pvalue <- PLACEHOLDER_contig_season_kruskal$p.value
`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`[nrow(`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`)+1,] <- c(PLACEHOLDER_contig_name, PLACEHOLDER_contig_host_pvalue)
`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`[, c(2)] <- sapply(`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`[, c(2)], as.numeric)
`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`[nrow(`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`)+1,] <- c(PLACEHOLDER_contig_name, PLACEHOLDER_contig_season_pvalue)
`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`[, c(2)] <- sapply(`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`[, c(2)], as.numeric)
rm(PLACEHOLDER_contig_name)
rm(PLACEHOLDER_contig_host_kruskal)
rm(PLACEHOLDER_contig_season_kruskal)
rm(PLACEHOLDER_contig_host_pvalue)
rm(PLACEHOLDER_contig_season_pvalue)

ggplot(data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`, aes(x = Species, y = value, fill = Species)) + geom_boxplot() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Host species") + ylab("Contig and library size adjusted mapped segment counts")

2022-07-03
write.table(`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species", sep="\t", row.names=FALSE)
write.table(`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season", sep="\t", row.names=FALSE)

`2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species` <- `2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`[order(`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`$Uncorrected.pvalue),]
`2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season` <- `2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`[order(`2022_07_01_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`$Uncorrected.pvalue),]

`2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`$BH_corrected <- p.adjust(`2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`$Uncorrected.pvalue, method = "BH")
`2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`$BH_corrected <- p.adjust(`2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`$Uncorrected.pvalue, method = "BH")

`2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested` <- `2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`[`2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`$BH_corrected < 0.05, ]
`2022_07_03_BH_corrected_Pvalues_below_005_sampling_season_Kruskal_wallis_tested` <- `2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`[`2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`$BH_corrected < 0.05, ]

write.table(`2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species", sep="\t", row.names=FALSE)
write.table(`2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_sample_season", sep="\t", row.names=FALSE)

`2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested_not_sample_season` <- subset(`2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested`, (`2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested`$Contig.name %notin% `2022_07_03_BH_corrected_Pvalues_below_005_sampling_season_Kruskal_wallis_tested`$Contig.name))
`2022_07_03_BH_corrected_Pvalues_below_005_sample_season_Kruskal_wallis_tested_not_host_species` <- subset(`2022_07_03_BH_corrected_Pvalues_below_005_sampling_season_Kruskal_wallis_tested`, (`2022_07_03_BH_corrected_Pvalues_below_005_sampling_season_Kruskal_wallis_tested`$Contig.name %notin% `2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested`$Contig.name))

write.table(`2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested_not_sample_season`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested_not_sample_season", sep="\t", row.names=FALSE)
write.table(`2022_07_03_BH_corrected_Pvalues_below_005_sample_season_Kruskal_wallis_tested_not_host_species`, "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_03_BH_corrected_Pvalues_below_005_sample_season_Kruskal_wallis_tested_not_host_species", sep="\t", row.names=FALSE)

ggplot(data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable %in% `2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested_not_sample_season`$Contig.name & grepl(glob2rx("k141_3022*"), `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Mapped segment counts to contig, library and contig size adjusted") + ylab("Host species")

ggplot(data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable %in% `2022_07_03_BH_corrected_Pvalues_below_005_sample_season_Kruskal_wallis_tested_not_host_species`$Contig.name & grepl(glob2rx("k141_1103*"), `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable), ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Mapped segment counts to contig, library and contig size adjusted") + ylab("Sampling month")

###Below needs some work looking at the BH corrected values to avoid plotting loads
###but the code itself is fine
ggplot(data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable %in% `2022_07_03_BH_corrected_Pvalues_below_005_sample_season_Kruskal_wallis_tested_not_host_species`$Contig.name & `2022_07_03_BH_corrected_Pvalues_below_005_sample_season_Kruskal_wallis_tested_not_host_species`$BH_corrected < -0.000241, ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Mapped segment counts to contig, library and contig size adjusted") + ylab("Sampling month")
###


ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Host species")

Bin_m522

cohen.d(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable) & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC"),] , `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable) & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR"),])

effect_size_testing <- `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable),]

kruskal_effsize(effect_size_testing, formula = effect_size_testing$value ~ effect_size_testing$Species, ci = FALSE, conf.level = 0.95, ci.type = "perc", nboot = 1000)
kruskal_effsize(effect_size_testing, formula = effect_size_testing$value ~ effect_size_testing$Species, ci = TRUE, conf.level = 0.95, ci.type = "perc", nboot = 1000)

kruskal_effsize(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c1500", ], formula = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$value ~ `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species, ci = FALSE, conf.level = 0.95, ci.type = "perc", nboot = 1000)
kruskal_effsize(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_c748", ], formula = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$value ~ `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species, ci = FALSE, conf.level = 0.95, ci.type = "perc", nboot = 1000)
kruskal_effsize(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m1197", ], formula = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$value ~ `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species, ci = FALSE, conf.level = 0.95, ci.type = "perc", nboot = 1000)

`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`
kruskal_effsize(`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable == "k141_11034216",], formula = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$value ~ `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$Species, ci = FALSE, conf.level = 0.95, ci.type = "perc", nboot = 1000)
kruskal_effsize(`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable == "k141_11034216",], formula = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$value ~ `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$Season, ci = FALSE, conf.level = 0.95, ci.type = "perc", nboot = 1000)

ggplot(data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable %in% `2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested_not_sample_season`$Contig.name & grepl(glob2rx("k141_11034216*"), `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Mapped segment counts to contig, library and contig size adjusted") + ylab("Host species")
ggplot(data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable %in% `2022_07_03_BH_corrected_Pvalues_below_005_sample_season_Kruskal_wallis_tested_not_host_species`$Contig.name & grepl(glob2rx("k141_11034216*"), `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable), ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Mapped segment counts to contig, library and contig size adjusted") + ylab("Sampling month")

`2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested`[`2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested`$Contig.name == "k141_11034216", ]
`2022_07_03_BH_corrected_Pvalues_below_005_sampling_season_Kruskal_wallis_tested`[`2022_07_03_BH_corrected_Pvalues_below_005_sampling_season_Kruskal_wallis_tested`$Contig.name == "k141_11034216", ]

kruskal_effsize(`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable == "k141_11034216",], formula = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$value ~ `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$Species, ci = FALSE, conf.level = 0.95, ci.type = "perc", nboot = 1000)

2022-07-04
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Host species")
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Sampling month")


ggplot(data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable %in% `2022_07_03_BH_corrected_Pvalues_below_005_host_species_Kruskal_wallis_tested_not_sample_season`$Contig.name & grepl(glob2rx("k141_3022554*"), `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Mapped segment counts to contig, library and contig size adjusted") + ylab("Host species")
ggplot(data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable %in% `2022_07_03_BH_corrected_Pvalues_below_005_sample_season_Kruskal_wallis_tested_not_host_species`$Contig.name & grepl(glob2rx("k141_15636287*"), `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable), ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Mapped segment counts to contig, library and contig size adjusted") + ylab("Sampling month")


PLACEHOLDER_contig_name <- "PLACEHOLDER"
PLACEHOLDER_contig_host_kruskal <- kruskal.test(value ~ Species, data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable == "PLACEHOLDER", ])
PLACEHOLDER_contig_season_kruskal <- kruskal.test(value ~ Season, data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable == "PLACEHOLDER", ])
PLACEHOLDER_contig_host_pvalue <- PLACEHOLDER_contig_host_kruskal$p.value
PLACEHOLDER_contig_season_pvalue <- PLACEHOLDER_contig_season_kruskal$p.value

PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_bin_host_kruskal_effect_size_not_BH_corrected <- kruskal_effsize(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "PLACEHOLDER", ], formula = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$value ~ `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species, ci = FALSE, conf.level = 0.95, ci.type = "perc", nboot = 1000)
PLACEHOLDER_bin_season_kruskal_effect_size_not_BH_corrected <- kruskal_effsize(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "PLACEHOLDER", ], formula = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$value ~ `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season, ci = FALSE, conf.level = 0.95, ci.type = "perc", nboot = 1000)


kruskal_effsize(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522", ], formula = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$value ~ `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species, ci = FALSE, conf.level = 0.95, ci.type = "perc", nboot = 1000)
kruskal_effsize(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable == "Bin_m522", ], formula = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$value ~ `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species, ci = TRUE, conf.level = 0.95, ci.type = "perc", nboot = 500)

kruskal.test(value ~ Species_x_season, data = `2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`[`2022_07_01_melted_transposed_contig_level_lib_and_kilo_adjusted_mapped_segments`$variable == "k141_15636287", ])

foldchange()

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.65)

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Sampling month") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.65)

2022-07-05
foldchange(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value, `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)

`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value
`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value

nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])
[1] 41
nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])
[1] 40
nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species_x_season == "ACJ" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])
[1] 22
nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species_x_season == "ACN" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])
[1] 19
nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species_x_season == "ARJ" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])
[1] 21
nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species_x_season == "ARN" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])
[1] 19


`2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed` <- `2022_06_29_transposed_summed_lib_cont_adjusted_all_samples`

`2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed` %>% add_row(Bins = "AR_zero_values_logfold_balancer", Species = "AR", Season = "EMPTY", Sequencing = "EMPTY", Species_x_season = "EMPTY")

`V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed` <- `2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed` %>% add_row(Bins = "AR_zero_values_logfold_balancer", Species = "AR", Season = "EMPTY", Sequencing = "EMPTY", Species_x_season = "EMPTY")

`V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`[is.na(`V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`)] <- 0

`melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed` <- reshape2::melt(`V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`, id=c("Bins", "Species", "Season", "Sequencing", "Species_x_season"))

foldchange(`melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`[`melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$Species == "AC" & grepl(glob2rx("Bin_m522"), `melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$variable), ]$value, `melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`[`melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$Species == "AR" & grepl(glob2rx("Bin_m522"), `melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$variable), ]$value)

foldchange(`melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`[`melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$Species == "AC" & grepl(glob2rx("Bin_m1545"), `melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$variable), ]$value, `melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`[`melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$Species == "AR" & grepl(glob2rx("Bin_m522"), `melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$variable), ]$value)

`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`[`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name == "Bin_m522", ]$BH_corrected_pvalues

log10(`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`[`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name == "Bin_m522", ]$BH_corrected_pvalues)

log10(`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`[`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name == "Bin_c583", ]$BH_corrected_pvalues)

mean(foldchange(`melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`[`melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$Species == "AC" & grepl(glob2rx("Bin_m1545"), `melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$variable), ]$value, `melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`[`melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$Species == "AR" & grepl(glob2rx("Bin_m522"), `melted_V2_2022_07_05_species_numbers_equalised_with_zero_value_AR_row_transposed`$variable), ]$value)[is.infinite()])

-log10(`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`[`2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name == "Bin_m522", ]$BH_corrected_pvalues)

mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)

nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])

nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])

nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species_x_season == "ACJ" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])

nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species_x_season == "ACN" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])

nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species_x_season == "ARJ" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])

nrow(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species_x_season == "ARN" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ])

`2022-07-05_foldchange_details_table_species` <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.mean.mapping.count", "AR.mean.mapping.count", "Fold.change.species.mean", "BH.corrected.pvalue", "-log10.BH.corrected.pvalue"))))
`2022-07-05_foldchange_details_table_season` <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Bin.name", "June.mean.mapping.count", "November.mean.mapping.count", "Fold.change.species.mean", "BH.corrected.pvalue", "-log10.BH.corrected.pvalue"))))

PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_AR_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_logfold_change_means_host <- foldchange(PLACEHOLDER_AC_mean_mappings, PLACEHOLDER_AR_mean_mappings)
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_host <- -log10(PLACEHOLDER_BH_corrected_pvalue_host)
PLACEHOLDER_J_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_N_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_logfold_change_means_season <- foldchange(PLACEHOLDER_J_mean_mappings, PLACEHOLDER_N_mean_mappings)
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_season <- -log10(PLACEHOLDER_BH_corrected_pvalue_season)
`2022-07-05_foldchange_details_table_species`[nrow(`2022-07-05_foldchange_details_table_species`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_mean_mappings, PLACEHOLDER_AR_mean_mappings, PLACEHOLDER_logfold_change_means_host, PLACEHOLDER_BH_corrected_pvalue_host, PLACEHOLDER_negative_log10_BH_corrected_pvalue_host)
`2022-07-05_foldchange_details_table_season`[nrow(`2022-07-05_foldchange_details_table_season`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_J_mean_mappings, PLACEHOLDER_N_mean_mappings, PLACEHOLDER_logfold_change_means_season, PLACEHOLDER_BH_corrected_pvalue_season, PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)
`2022-07-05_foldchange_details_table_species`[, c(2:6)] <- sapply(`2022-07-05_foldchange_details_table_species`[, c(2:6)], as.numeric)
`2022-07-05_foldchange_details_table_season`[, c(2:6)] <- sapply(`2022-07-05_foldchange_details_table_season`[, c(2:6)], as.numeric)
rm(PLACEHOLDER_bin_name)
rm(PLACEHOLDER_AC_mean_mappings)
rm(PLACEHOLDER_AR_mean_mappings)
rm(PLACEHOLDER_logfold_change_means_host)
rm(PLACEHOLDER_BH_corrected_pvalue_host)
rm(PLACEHOLDER_J_mean_mappings)
rm(PLACEHOLDER_N_mean_mappings)
rm(PLACEHOLDER_logfold_change_means_season)
rm(PLACEHOLDER_BH_corrected_pvalue_season)
rm(PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)

rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))

ggplot(data = `2022-07-05_foldchange_details_table_species`, aes(x = Fold.change.species.mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-05_foldchange_of_means_species.svg")
ggplot(data = `2022-07-05_foldchange_details_table_season`, aes(x = Fold.change.species.mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Season effect bins")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-05_foldchange_of_means_season.svg")

ggplot(data = `2022-07-05_foldchange_details_table_species`, aes(x = log(Fold.change.species.mean), y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-05_log_foldchange_of_means_species.svg")
ggplot(data = `2022-07-05_foldchange_details_table_season`, aes(x = log(Fold.change.species.mean), y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Season effect bins")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-05_log_foldchange_of_means_season.svg")


ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.65)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-06_demo_box_and_violin_bin_contig_counts_by_host_specific_grepl_match.svg")

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host`$Bin.name & grepl(glob2rx("Bin_m5*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Sampling month") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.65)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-06_demo_box_and_violin_bin_contig_counts_by_season_specific_grepl_match.svg")

mean(log2(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value) - mean(log2(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)

ggplot(data = `2022-07-05_foldchange_details_table_species`, aes(x = log(Fold.change.species.mean), y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins")


median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)

log(median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value))
log(median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value))
foldchange(log(median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)), log(median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))

log(median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)) - log(median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value))

exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))

`2022-07-05_medians_foldchange_details_table_species` <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.median.mapping.count", "AR.median.mapping.count", "Fold.change.species.median", "BH.corrected.pvalue", "-log10.BH.corrected.pvalue"))))
`2022-07-05_medians_foldchange_details_table_season` <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Bin.name", "June.median.mapping.count", "November.median.mapping.count", "Fold.change.season.median", "BH.corrected.pvalue", "-log10.BH.corrected.pvalue"))))

PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_AR_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_fold_change_medians_host <- foldchange(PLACEHOLDER_AC_median_mappings, PLACEHOLDER_AR_median_mappings)
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_host <- -log10(PLACEHOLDER_BH_corrected_pvalue_host)
PLACEHOLDER_J_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_N_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_fold_change_medians_season <- foldchange(PLACEHOLDER_J_median_mappings, PLACEHOLDER_N_median_mappings)
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_season <- -log10(PLACEHOLDER_BH_corrected_pvalue_season)
`2022-07-05_medians_foldchange_details_table_species`[nrow(`2022-07-05_medians_foldchange_details_table_species`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_median_mappings, PLACEHOLDER_AR_median_mappings, PLACEHOLDER_logfold_change_medians_host, PLACEHOLDER_BH_corrected_pvalue_host, PLACEHOLDER_negative_log10_BH_corrected_pvalue_host)
`2022-07-05_medians_foldchange_details_table_season`[nrow(`2022-07-05_medians_foldchange_details_table_season`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_J_median_mappings, PLACEHOLDER_N_median_mappings, PLACEHOLDER_logfold_change_medians_season, PLACEHOLDER_BH_corrected_pvalue_season, PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)
`2022-07-05_medians_foldchange_details_table_species`[, c(2:6)] <- sapply(`2022-07-05_medians_foldchange_details_table_species`[, c(2:6)], as.numeric)
`2022-07-05_medians_foldchange_details_table_season`[, c(2:6)] <- sapply(`2022-07-05_medians_foldchange_details_table_season`[, c(2:6)], as.numeric)
rm(PLACEHOLDER_bin_name)
rm(PLACEHOLDER_AC_median_mappings)
rm(PLACEHOLDER_AR_median_mappings)
rm(PLACEHOLDER_fold_change_medians_host)
rm(PLACEHOLDER_BH_corrected_pvalue_host)
rm(PLACEHOLDER_J_median_mappings)
rm(PLACEHOLDER_N_median_mappings)
rm(PLACEHOLDER_fold_change_medians_season)
rm(PLACEHOLDER_BH_corrected_pvalue_season)
rm(PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)

rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))

ggplot(data = `2022-07-05_medians_foldchange_details_table_species`, aes(x = Fold.change.species.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins") + xlab("Fold change of median value for mapped segment count")
ggplot(data = `2022-07-05_medians_foldchange_details_table_season`, aes(x = Fold.change.season.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Season effect bins") + xlab("Fold change of median value for mapped segment count")
ggplot(data = `2022-07-05_medians_foldchange_details_table_species`, aes(x = log(Fold.change.species.median), y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins") + xlab("Log of fold change of median value for mapped segment count")
ggplot(data = `2022-07-05_medians_foldchange_details_table_season`, aes(x = log(Fold.change.season.median), y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Season effect bins") + xlab("Log of fold change of median value for mapped segment count")
ggplot(data = `2022-07-05_medians_foldchange_details_table_species`, aes(x = log2(Fold.change.species.median), y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins") + xlab("Log 2 of fold change of median value for mapped segment count")
ggplot(data = `2022-07-05_medians_foldchange_details_table_season`, aes(x = log2(Fold.change.season.median), y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Season effect bins") + xlab("Log 2 of fold change of median value for mapped segment count")
ggplot(data = `2022-07-05_medians_foldchange_details_table_species`, aes(x = log1p(Fold.change.species.median), y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins") + xlab("Log1p of fold change of median value for mapped segment count")
ggplot(data = `2022-07-05_medians_foldchange_details_table_season`, aes(x = log1p(Fold.change.season.median), y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Season effect bins") + xlab("Log1p of fold change of median value for mapped segment count")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-05_foldchange_of_medians_species.svg")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-05_foldchange_of_medians_season.svg")

is.nan(log(`2022-07-05_medians_foldchange_details_table_season`$Fold.change.season.median))
is.nan(log(`2022-07-05_medians_foldchange_details_table_season`$Fold.change.season.median + 1))
is.nan(log(`2022-07-05_medians_foldchange_details_table_season`$Fold.change.season.median + 5))

`2022-07-05_geometric_means_foldchange_details_table_species` <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.geometric_mean.mapping.count", "AR.geometric_mean.mapping.count", "Fold.change.species.geometric_mean", "BH.corrected.pvalue", "-log10.BH.corrected.pvalue"))))
`2022-07-05_geometric_means_foldchange_details_table_season` <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Bin.name", "June.geometric_mean.mapping.count", "November.geometric_mean.mapping.count", "Fold.change.season.geometric_mean", "BH.corrected.pvalue", "-log10.BH.corrected.pvalue"))))

PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_AR_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_fold_change_geometric_means_host <- foldchange(PLACEHOLDER_AC_geometric_means_mappings, PLACEHOLDER_AR_geometric_means_mappings)
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_host <- -log10(PLACEHOLDER_BH_corrected_pvalue_host)
PLACEHOLDER_J_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_N_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_fold_change_geometric_means_season <- foldchange(PLACEHOLDER_J_geometric_means_mappings, PLACEHOLDER_N_geometric_means_mappings)
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_season <- -log10(PLACEHOLDER_BH_corrected_pvalue_season)
`2022-07-05_geometric_means_foldchange_details_table_species`[nrow(`2022-07-05_geometric_means_foldchange_details_table_species`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_geometric_means_mappings, PLACEHOLDER_AR_geometric_means_mappings, PLACEHOLDER_fold_change_geometric_means_host, PLACEHOLDER_BH_corrected_pvalue_host, PLACEHOLDER_negative_log10_BH_corrected_pvalue_host)
`2022-07-05_geometric_means_foldchange_details_table_season`[nrow(`2022-07-05_geometric_means_foldchange_details_table_season`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_J_geometric_means_mappings, PLACEHOLDER_N_geometric_means_mappings, PLACEHOLDER_fold_change_geometric_means_season, PLACEHOLDER_BH_corrected_pvalue_season, PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)
`2022-07-05_geometric_means_foldchange_details_table_species`[, c(2:6)] <- sapply(`2022-07-05_geometric_means_foldchange_details_table_species`[, c(2:6)], as.numeric)
`2022-07-05_geometric_means_foldchange_details_table_season`[, c(2:6)] <- sapply(`2022-07-05_geometric_means_foldchange_details_table_season`[, c(2:6)], as.numeric)
rm(PLACEHOLDER_bin_name)
rm(PLACEHOLDER_AC_geometric_means_mappings)
rm(PLACEHOLDER_AR_geometric_means_mappings)
rm(PLACEHOLDER_fold_change_geometric_means_host)
rm(PLACEHOLDER_BH_corrected_pvalue_host)
rm(PLACEHOLDER_J_geometric_means_mappings)
rm(PLACEHOLDER_N_geometric_means_mappings)
rm(PLACEHOLDER_fold_change_geometric_means_season)
rm(PLACEHOLDER_BH_corrected_pvalue_season)
rm(PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)

rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))

ggplot(data = `2022-07-05_geometric_means_foldchange_details_table_species`, aes(x = Fold.change.species.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins") + xlab("Fold change of geometric mean value for mapped segment count")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-05_foldchange_of_geometric_means_species.svg")
ggplot(data = `2022-07-05_geometric_means_foldchange_details_table_season`, aes(x = Fold.change.season.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Season effect bins") + xlab("Fold change of geometric mean value for mapped segment count")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-05_foldchange_of_geometric_means_season.svg")

2022-07-07
ggplot(data = `2022-07-05_foldchange_details_table_species`, aes(x = Fold.change.species.mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins")
ggplot(data = `2022-07-05_geometric_means_foldchange_details_table_species`, aes(x = Fold.change.species.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins") + xlab("Fold change of geometric mean value for mapped segment count")
ggplot(data = `2022-07-05_medians_foldchange_details_table_species`, aes(x = log(abs(Fold.change.species.median)), y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins") + xlab("Log of fold change of median value for mapped segment count")

ggplot(data = `2022-07-05_foldchange_details_table_species`, aes(x = Fold.change.species.mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins")
ggplot(data = `2022-07-05_medians_foldchange_details_table_species`, aes(x = -log(Fold.change.species.median), y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins") + xlab("Log of fold change of median value for mapped segment count")

ggplot(data = `2022-07-05_foldchange_details_table_species`, aes(x = Fold.change.species.mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins")

`2022_07_01_sorted_Pvalues_lib_cont_adjusted_no_transformation_contig_level_mapped_segments_kruskal_test_P_values_host_species`


foldchange(log(median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)), log(median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))

(log(median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)) - log(median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)) / `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_m522"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)

(`2022-07-05_foldchange_details_table_species`[`2022-07-05_foldchange_details_table_species`$Bin.name == "Bin_m522" ,]$AR.mean.mapping.count - `2022-07-05_foldchange_details_table_species`[`2022-07-05_foldchange_details_table_species`$Bin.name == "Bin_m522" ,]$AC.mean.mapping.count) / `2022-07-05_foldchange_details_table_species`[`2022-07-05_foldchange_details_table_species`$Bin.name == "Bin_m522", ]$AC.mean.mapping.count
log((`2022-07-05_foldchange_details_table_species`[`2022-07-05_foldchange_details_table_species`$Bin.name == "Bin_m522" ,]$AR.mean.mapping.count - `2022-07-05_foldchange_details_table_species`[`2022-07-05_foldchange_details_table_species`$Bin.name == "Bin_m522" ,]$AC.mean.mapping.count) / `2022-07-05_foldchange_details_table_species`[`2022-07-05_foldchange_details_table_species`$Bin.name == "Bin_m522", ]$AC.mean.mapping.count)

`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.mean.mapping.count", "AR.mean.mapping.count", "Log1p.Fold.change.species.arithmetic_mean", "-log10.BH.corrected.pvalue"))))
PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_AR_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_log1pfold_change_means_host <- log1p((`PLACEHOLDER_AR_mean_mappings` - `PLACEHOLDER_AC_mean_mappings`) / `PLACEHOLDER_AC_mean_mappings`)
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_host <- -log10(PLACEHOLDER_BH_corrected_pvalue_host)
`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`[nrow(`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_mean_mappings, PLACEHOLDER_AR_mean_mappings, PLACEHOLDER_log1pfold_change_means_host, PLACEHOLDER_negative_log10_BH_corrected_pvalue_host)
`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`[, c(2:5)] <- sapply(`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`[, c(2:5)], as.numeric)

log((mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_c793"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value) - mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_c793"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value) / mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_c793"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))

mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_c793"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_c793"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)

(mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_c793"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value) - mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("Bin_c793"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value) / mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("Bin_c793"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value))

ggplot(data = `2022-07-07_arithmetic_means_manual_log_foldchange_table_species`, aes(x = Log.Fold.change.species.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(0.05), col="#c80815") + ggtitle("Host effect bins")
ggplot(data = `2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`, aes(x = Log1p.Fold.change.species.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins")

`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`

ggplot(data = `2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`, aes(x = Log1p.Fold.change.species.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-0.6, 0.6), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of arithmetic means per bin")

ggplot(data = `2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`, aes(x = Log1p.Fold.change.species.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of arithmetic means per bin")

`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.mean.mapping.count", "AR.mean.mapping.count", "Log1p.Fold.change.species.arithmetic_mean", "-log10.BH.corrected.pvalue"))))
`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_season` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "J.mean.mapping.count", "N.mean.mapping.count", "Log1p.Fold.change.season.arithmetic_mean", "-log10.BH.corrected.pvalue"))))
`2022-07-07_geometric_means_manual_log1p_foldchange_table_species` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.geometric.mean.mapping.count", "AR.geometric.mean.mapping.count", "Log1p.Fold.change.species.geometric_mean", "-log10.BH.corrected.pvalue"))))
`2022-07-07_geometric_means_manual_log1p_foldchange_table_season` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "J.geometric.mean.mapping.count", "N.geometric.mean.mapping.count", "Log1p.Fold.change.season.geometric_mean", "-log10.BH.corrected.pvalue"))))
`2022-07-07_medians_manual_log1p_foldchange_table_species` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.median.mapping.count", "AR.median.mapping.count", "Log1p.Fold.change.species.median", "-log10.BH.corrected.pvalue"))))
`2022-07-07_medians_manual_log1p_foldchange_table_season` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "J.median.mapping.count", "N.median.mapping.count", "Log1p.Fold.change.season.median", "-log10.BH.corrected.pvalue"))))

PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_AR_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_log1pfold_change_means_host <- log1p((`PLACEHOLDER_AR_mean_mappings` - `PLACEHOLDER_AC_mean_mappings`) / `PLACEHOLDER_AC_mean_mappings`)
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_host <- -log10(PLACEHOLDER_BH_corrected_pvalue_host)
PLACEHOLDER_J_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_N_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_log1pfold_change_means_season <- log1p((`PLACEHOLDER_N_mean_mappings` - `PLACEHOLDER_J_mean_mappings`) / `PLACEHOLDER_J_mean_mappings`)
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_season <- -log10(PLACEHOLDER_BH_corrected_pvalue_season)
`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`[nrow(`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_mean_mappings, PLACEHOLDER_AR_mean_mappings, PLACEHOLDER_log1pfold_change_means_host, PLACEHOLDER_negative_log10_BH_corrected_pvalue_host)
`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`[, c(2:5)] <- sapply(`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`[, c(2:5)], as.numeric)
`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_season`[nrow(`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_season`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_J_mean_mappings, PLACEHOLDER_N_mean_mappings, PLACEHOLDER_log1pfold_change_means_season, PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)
`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_season`[, c(2:5)] <- sapply(`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_season`[, c(2:5)], as.numeric)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))
PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_AR_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_log1pfold_change_geometric_means_host <- log1p((PLACEHOLDER_AR_geometric_means_mappings - PLACEHOLDER_AC_geometric_means_mappings) / PLACEHOLDER_AC_geometric_means_mappings)
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_host <- -log10(PLACEHOLDER_BH_corrected_pvalue_host)
PLACEHOLDER_J_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_N_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_log1pfold_change_geometric_means_season <- log1p((PLACEHOLDER_N_geometric_means_mappings - PLACEHOLDER_J_geometric_means_mappings) / PLACEHOLDER_J_geometric_means_mappings)
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_season <- -log10(PLACEHOLDER_BH_corrected_pvalue_season)
`2022-07-07_geometric_means_manual_log1p_foldchange_table_species`[nrow(`2022-07-07_geometric_means_manual_log1p_foldchange_table_species`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_geometric_means_mappings, PLACEHOLDER_AR_geometric_means_mappings, PLACEHOLDER_log1pfold_change_geometric_means_host, PLACEHOLDER_negative_log10_BH_corrected_pvalue_host)
`2022-07-07_geometric_means_manual_log1p_foldchange_table_species`[, c(2:5)] <- sapply(`2022-07-07_geometric_means_manual_log1p_foldchange_table_species`[, c(2:5)], as.numeric)
`2022-07-07_geometric_means_manual_log1p_foldchange_table_season`[nrow(`2022-07-07_geometric_means_manual_log1p_foldchange_table_season`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_J_geometric_means_mappings, PLACEHOLDER_N_geometric_means_mappings, PLACEHOLDER_log1pfold_change_geometric_means_season, PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)
`2022-07-07_geometric_means_manual_log1p_foldchange_table_season`[, c(2:5)] <- sapply(`2022-07-07_geometric_means_manual_log1p_foldchange_table_season`[, c(2:5)], as.numeric)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))
PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_AR_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_log1pfold_change_medians_host <- log1p((PLACEHOLDER_AR_median_mappings - PLACEHOLDER_AC_median_mappings) / PLACEHOLDER_AC_median_mappings)
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_host <- -log10(PLACEHOLDER_BH_corrected_pvalue_host)
PLACEHOLDER_J_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_N_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_log1pfold_change_medians_season <- log1p((PLACEHOLDER_N_median_mappings - PLACEHOLDER_J_median_mappings) / PLACEHOLDER_J_median_mappings)
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_season <- -log10(PLACEHOLDER_BH_corrected_pvalue_season)
`2022-07-07_medians_manual_log1p_foldchange_table_species`[nrow(`2022-07-07_medians_manual_log1p_foldchange_table_species`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_median_mappings, PLACEHOLDER_AR_median_mappings, PLACEHOLDER_log1pfold_change_medians_host, PLACEHOLDER_negative_log10_BH_corrected_pvalue_host)
`2022-07-07_medians_manual_log1p_foldchange_table_species`[, c(2:5)] <- sapply(`2022-07-07_medians_manual_log1p_foldchange_table_species`[, c(2:5)], as.numeric)
`2022-07-07_medians_means_manual_log1p_foldchange_table_season`[nrow(`2022-07-07_medians_means_manual_log1p_foldchange_table_season`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_J_median_mappings, PLACEHOLDER_N_median_mappings, PLACEHOLDER_log1pfold_change_medians_season, PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)
`2022-07-07_medians_means_manual_log1p_foldchange_table_season`[, c(2:5)] <- sapply(`2022-07-07_medians_means_manual_log1p_foldchange_table_season`[, c(2:5)], as.numeric)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))

rm(`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`)
rm(`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_season`)
rm(`2022-07-07_geometric_means_manual_log1p_foldchange_table_species`)
rm(`2022-07-07_geometric_means_manual_log1p_foldchange_table_season`)
rm(`2022-07-07_medians_manual_log1p_foldchange_table_species`)
rm(`2022-07-07_medians_means_manual_log1p_foldchange_table_season`)

ggplot(data = `2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`, aes(x = Log1p.Fold.change.species.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of arithmetic means per bin")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species.svg")

ggplot(data = `2022-07-07_geometric_means_manual_log1p_foldchange_table_species`, aes(x = Log1p.Fold.change.species.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of geometric means per bin")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-07_geometric_means_manual_log1p_foldchange_table_species.svg")

ggplot(data = `2022-07-07_medians_manual_log1p_foldchange_table_species`, aes(x = Log1p.Fold.change.species.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of median per bin")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-07_medians_manual_log1p_foldchange_table_species.svg")

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`, aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = FALSE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-07_ridgeline_distribution_per_species_all_mapping_counts_all_samples_contig_and_lib_adjusted.svg")


ggplot(data = `2022-07-07_arithmetic_means_manual_log1p_foldchange_table_season`, aes(x = Log1p.Fold.change.season.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of arithmetic means per bin")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-07_arithmetic_means_manual_log1p_foldchange_table_season.svg")

ggplot(data = `2022-07-07_geometric_means_manual_log1p_foldchange_table_season`, aes(x = Log1p.Fold.change.season.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of geometric means per bin")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-07_geometric_means_manual_log1p_foldchange_table_season.svg")

ggplot(data = `2022-07-07_medians_means_manual_log1p_foldchange_table_season`, aes(x = Log1p.Fold.change.season.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of median per bin")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-07_medians_means_manual_log1p_foldchange_table_season.svg")

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`, aes(x = value, y = Season, fill = Season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Season") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-07_box_and_violin_distribution_per_season_all_mapping_counts_all_samples_contig_and_lib_adjusted.svg")

temp_fig_1 <- ggplot(data = `2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`, aes(x = Log1p.Fold.change.species.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of arithmetic means per bin")
temp_fig_2 <- ggplot(data = `2022-07-07_geometric_means_manual_log1p_foldchange_table_species`, aes(x = Log1p.Fold.change.species.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of geometric means per bin")
temp_fig_3 <- ggplot(data = `2022-07-07_medians_manual_log1p_foldchange_table_species`, aes(x = Log1p.Fold.change.species.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of median per bin")
temp_fig_4 <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`, aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = FALSE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species")
temp_fig_5 <- grid.arrange(temp_fig_1, temp_fig_2, temp_fig_3, temp_fig_4)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-07_combined_species_plots_foldchanges.svg", `temp_fig_5`)

temp_fig_1 <- ggplot(data = `2022-07-07_arithmetic_means_manual_log1p_foldchange_table_season`, aes(x = Log1p.Fold.change.season.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of arithmetic means per bin")
temp_fig_2 <- ggplot(data = `2022-07-07_geometric_means_manual_log1p_foldchange_table_season`, aes(x = Log1p.Fold.change.season.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of geometric means per bin")
temp_fig_3 <- ggplot(data = `2022-07-07_medians_means_manual_log1p_foldchange_table_season`, aes(x = Log1p.Fold.change.season.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("Log1p fold change of median per bin")
temp_fig_4 <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`, aes(x = value, y = Season, fill = Season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Season") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75)
temp_fig_5 <- grid.arrange(temp_fig_1, temp_fig_2, temp_fig_3, temp_fig_4)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-07_combined_season_plots_foldchanges.svg", `temp_fig_5`)

`2022-07-07_geometric_means_manual_log1p_foldchange_table_season`[`2022-07-07_geometric_means_manual_log1p_foldchange_table_season`$X.log10.BH.corrected.pvalue > -log10(0.05) & `2022-07-07_geometric_means_manual_log1p_foldchange_table_season`$Log1p.Fold.change.season.geometric_mean > 1, ]
Bin_m1474
Bin_m1569
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host`$Bin.name & grepl(glob2rx("Bin_m1474"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Sampling month")
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host`$Bin.name & grepl(glob2rx("Bin_m1569"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Sampling month")

write.table(`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species`, "2022-07-07_arithmetic_means_manual_log1p_foldchange_table_species.tsv", sep="\t", row.names=FALSE)
write.table(`2022-07-07_arithmetic_means_manual_log1p_foldchange_table_season`, "2022-07-07_arithmetic_means_manual_log1p_foldchange_table_season.tsv", sep="\t", row.names=FALSE)
write.table(`2022-07-07_geometric_means_manual_log1p_foldchange_table_species`, "2022-07-07_geometric_means_manual_log1p_foldchange_table_species.tsv", sep="\t", row.names=FALSE)
write.table(`2022-07-07_geometric_means_manual_log1p_foldchange_table_season`, "2022-07-07_geometric_means_manual_log1p_foldchange_table_season.tsv", sep="\t", row.names=FALSE)
write.table(`2022-07-07_medians_manual_log1p_foldchange_table_species`, "2022-07-07_medians_manual_log1p_foldchange_table_species.tsv", sep="\t", row.names=FALSE)
write.table(`2022-07-07_medians_means_manual_log1p_foldchange_table_season`, "2022-07-07_medians_means_manual_log1p_foldchange_table_season.tsv", sep="\t", row.names=FALSE)

`2022-07-07_geometric_means_manual_log1p_foldchange_table_species`[`2022-07-07_geometric_means_manual_log1p_foldchange_table_species`$X.log10.BH.corrected.pvalue > -log10(0.05) & `2022-07-07_geometric_means_manual_log1p_foldchange_table_species`$Log1p.Fold.change.species.geometric_mean > 1, ]
Bin_c745
Bin_c1000
Bin_c1199
Bin_c1500
Bin_c1673
Bin_c1687
Bin_m522
Bin_m1066
Bin_m1648

`2022-07-07_geometric_means_manual_log1p_foldchange_table_species`[`2022-07-07_geometric_means_manual_log1p_foldchange_table_species`$X.log10.BH.corrected.pvalue > -log10(0.05) & `2022-07-07_geometric_means_manual_log1p_foldchange_table_species`$Log1p.Fold.change.species.geometric_mean < -1, ]
Bin_c892
Bin_m1093
Bin_m1122

2022-07-08
`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_species` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.mean.mapping.count", "AR.mean.mapping.count", "Log2.add.one.fold.change.species.arithmetic_mean", "-log10.BH.corrected.pvalue"))))
`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_season` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "J.mean.mapping.count", "N.mean.mapping.count", "Log2.add.one.fold.change.season.arithmetic_mean", "-log10.BH.corrected.pvalue"))))
`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.geometric.mean.mapping.count", "AR.geometric.mean.mapping.count", "Log2.add.one.fold.change.species.geometric_mean", "-log10.BH.corrected.pvalue"))))
`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "J.geometric.mean.mapping.count", "N.geometric.mean.mapping.count", "Log2.add.one.fold.change.season.geometric_mean", "-log10.BH.corrected.pvalue"))))
`2022-07-08_medians_manual_log2_add_1_foldchange_table_species` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.median.mapping.count", "AR.median.mapping.count", "Log2.add.one.fold.change.species.median", "-log10.BH.corrected.pvalue"))))
`2022-07-08_medians_manual_log2_add_1_foldchange_table_season` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Bin.name", "J.median.mapping.count", "N.median.mapping.count", "Log2.add.one.fold.change.season.median", "-log10.BH.corrected.pvalue"))))

PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_AR_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_log2_add_one_fold_change_means_host <- log2(((`PLACEHOLDER_AR_mean_mappings` - `PLACEHOLDER_AC_mean_mappings`) / `PLACEHOLDER_AC_mean_mappings`) + 1)
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_host <- -log10(PLACEHOLDER_BH_corrected_pvalue_host)
PLACEHOLDER_J_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_N_mean_mappings <- mean(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_log2_add_one_fold_change_means_season <- log2(((`PLACEHOLDER_N_mean_mappings` - `PLACEHOLDER_J_mean_mappings`) / `PLACEHOLDER_J_mean_mappings`) + 1)
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_season <- -log10(PLACEHOLDER_BH_corrected_pvalue_season)
`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_species`[nrow(`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_species`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_mean_mappings, PLACEHOLDER_AR_mean_mappings, PLACEHOLDER_log2_add_one_fold_change_means_host, PLACEHOLDER_negative_log10_BH_corrected_pvalue_host)
`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_species`[, c(2:5)] <- sapply(`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_species`[, c(2:5)], as.numeric)
`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_season`[nrow(`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_season`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_J_mean_mappings, PLACEHOLDER_N_mean_mappings, PLACEHOLDER_log2_add_one_fold_change_means_season, PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)
`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_season`[, c(2:5)] <- sapply(`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_season`[, c(2:5)], as.numeric)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))
PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_AR_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_log2_add_one_fold_change_geometric_means_host <- log2(((PLACEHOLDER_AR_geometric_means_mappings - PLACEHOLDER_AC_geometric_means_mappings) / PLACEHOLDER_AC_geometric_means_mappings) + 1)
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_host <- -log10(PLACEHOLDER_BH_corrected_pvalue_host)
PLACEHOLDER_J_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_N_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_log2_add_one_fold_change_geometric_means_season <- log2(((PLACEHOLDER_N_geometric_means_mappings - PLACEHOLDER_J_geometric_means_mappings) / PLACEHOLDER_J_geometric_means_mappings) + 1)
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_season <- -log10(PLACEHOLDER_BH_corrected_pvalue_season)
`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[nrow(`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_geometric_means_mappings, PLACEHOLDER_AR_geometric_means_mappings, PLACEHOLDER_log2_add_one_fold_change_geometric_means_host, PLACEHOLDER_negative_log10_BH_corrected_pvalue_host)
`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[, c(2:5)] <- sapply(`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[, c(2:5)], as.numeric)
`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`[nrow(`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_J_geometric_means_mappings, PLACEHOLDER_N_geometric_means_mappings, PLACEHOLDER_log2_add_one_fold_change_geometric_means_season, PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)
`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`[, c(2:5)] <- sapply(`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`[, c(2:5)], as.numeric)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))
PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_AR_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_log2_add_one_fold_change_medians_host <- log2(((PLACEHOLDER_AR_median_mappings - PLACEHOLDER_AC_median_mappings) / PLACEHOLDER_AC_median_mappings) + 1)
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_host <- -log10(PLACEHOLDER_BH_corrected_pvalue_host)
PLACEHOLDER_J_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_N_median_mappings <- median(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)
PLACEHOLDER_log2_add_one_fold_change_medians_season <- log2(((PLACEHOLDER_N_median_mappings - PLACEHOLDER_J_median_mappings) / PLACEHOLDER_J_median_mappings) + 1)
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_negative_log10_BH_corrected_pvalue_season <- -log10(PLACEHOLDER_BH_corrected_pvalue_season)
`2022-07-08_medians_manual_log2_add_1_foldchange_table_species`[nrow(`2022-07-08_medians_manual_log2_add_1_foldchange_table_species`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_median_mappings, PLACEHOLDER_AR_median_mappings, PLACEHOLDER_log2_add_one_fold_change_medians_host, PLACEHOLDER_negative_log10_BH_corrected_pvalue_host)
`2022-07-08_medians_manual_log2_add_1_foldchange_table_species`[, c(2:5)] <- sapply(`2022-07-08_medians_manual_log2_add_1_foldchange_table_species`[, c(2:5)], as.numeric)
`2022-07-08_medians_manual_log2_add_1_foldchange_table_season`[nrow(`2022-07-08_medians_manual_log2_add_1_foldchange_table_season`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_J_median_mappings, PLACEHOLDER_N_median_mappings, PLACEHOLDER_log2_add_one_fold_change_medians_season, PLACEHOLDER_negative_log10_BH_corrected_pvalue_season)
`2022-07-08_medians_manual_log2_add_1_foldchange_table_season`[, c(2:5)] <- sapply(`2022-07-08_medians_manual_log2_add_1_foldchange_table_season`[, c(2:5)], as.numeric)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))

rm(list=ls(pattern='2022-07-08_*'))

write.table(`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_species`, "2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_species.tsv", sep="\t", row.names=FALSE)
write.table(`2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_season`, "2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_season.tsv", sep="\t", row.names=FALSE)
write.table(`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`, "2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species.tsv", sep="\t", row.names=FALSE)
write.table(`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`, "2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season.tsv", sep="\t", row.names=FALSE)
write.table(`2022-07-08_medians_manual_log2_add_1_foldchange_table_species`, "2022-07-08_medians_manual_log2_add_1_foldchange_table_species.tsv", sep="\t", row.names=FALSE)
write.table(`2022-07-08_medians_manual_log2_add_1_foldchange_table_season`, "2022-07-08_medians_manual_log2_add_1_foldchange_table_season.tsv", sep="\t", row.names=FALSE)

temp_fig_1 <- ggplot(data = `2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_species`, aes(x = Log2.add.one.fold.change.species.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of arithmetic means per bin")
temp_fig_2 <- ggplot(data = `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`, aes(x = Log2.add.one.fold.change.species.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin")
temp_fig_3 <- ggplot(data = `2022-07-08_medians_manual_log2_add_1_foldchange_table_species`, aes(x = Log2.add.one.fold.change.species.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of median per bin")
temp_fig_4 <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`, aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = FALSE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species")
temp_fig_5 <- grid.arrange(temp_fig_1, temp_fig_2, temp_fig_3, temp_fig_4)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-08_combined_species_plots_log2_add_one_fold_changes.svg", `temp_fig_5`)

temp_fig_1 <- ggplot(data = `2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_season`, aes(x = Log2.add.one.fold.change.season.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of arithmetic means per bin")
temp_fig_2 <- ggplot(data = `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`, aes(x = Log2.add.one.fold.change.season.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin")
temp_fig_3 <- ggplot(data = `2022-07-08_medians_manual_log2_add_1_foldchange_table_season`, aes(x = Log2.add.one.fold.change.season.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of median per bin")
temp_fig_4 <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`, aes(x = value, y = Season, fill = Season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Season") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75)
temp_fig_5 <- grid.arrange(temp_fig_1, temp_fig_2, temp_fig_3, temp_fig_4)

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-08_combined_season_plots_log2_add_one_fold_changes.svg", `temp_fig_5`)

`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$X.log10.BH.corrected.pvalue > -log10(0.05) & `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$Log2.add.one.fold.change.season.geometric_mean > 1, ]
`2022-07-08_season_effect_bins_pvalue_significant_effect_size_over_1_or_under_negative_1` <- `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$X.log10.BH.corrected.pvalue > -log10(0.05) & `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$Log2.add.one.fold.change.season.geometric_mean > 1, ]
`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$X.log10.BH.corrected.pvalue > -log10(0.05) & `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$Log2.add.one.fold.change.season.geometric_mean < -1, ]
write.table(`2022-07-08_season_effect_bins_pvalue_significant_effect_size_over_1_or_under_negative_1`, "2022-07-08_season_effect_bins_pvalue_significant_effect_size_over_1_or_under_negative_1.tsv", sep="\t", row.names=FALSE)

`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$X.log10.BH.corrected.pvalue > -log10(0.05) & `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean > 1, ]
`2022-07-08_geometric_means_species_effect_bins_pvalue_significant_effect_size_over_1` <- `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$X.log10.BH.corrected.pvalue > -log10(0.05) & `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean > 1, ]
write.table(`2022-07-08_geometric_means_species_effect_bins_pvalue_significant_effect_size_over_1`, "2022-07-08_geometric_means_species_effect_bins_pvalue_significant_effect_size_over_1.tsv", sep="\t", row.names=FALSE)

`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$X.log10.BH.corrected.pvalue > -log10(0.05) & `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean < -1, ]
`2022-07-08_geometric_means_species_effect_bins_pvalue_significant_effect_size_under_negative_1` <- `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$X.log10.BH.corrected.pvalue > -log10(0.05) & `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean < -1, ]
write.table(`2022-07-08_geometric_means_species_effect_bins_pvalue_significant_effect_size_under_negative_1`, "2022-07-08_geometric_means_species_effect_bins_pvalue_significant_effect_size_under_negative_1.tsv", sep="\t", row.names=FALSE)

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & grepl(glob2rx("Bin_m798"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Host species")

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[grepl(glob2rx("Bin_c380"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Host species")

2022-07-09
#Exemplar host specific effect bins matching a grepl pattern and faceted
ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_host_significant_not_season`$Bin.name & grepl(glob2rx("Bin_m9*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Host species")


ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022_06_30_bins_BH_corrected_Pvalue_lower_005_only_present_season_significant_not_host`$Bin.name & grepl(glob2rx("Bin_m14*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Sampling month")

`2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1` <- `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$Log2.add.one.fold.change.season.geometric_mean > 1 & `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$BH_corrected_pvalues < 0.05 | `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$Log2.add.one.fold.change.season.geometric_mean < -1 & `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$BH_corrected_pvalues < 0.05, ]

`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1` <- `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean > 1 & `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05 | `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean < -1 & `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05, ]

setwd("~/Google Drive/PhD/2022-Paper_2-Chapter_3")
write.table(`2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`, "2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1.tsv", sep="\t", row.names=FALSE)
write.table(`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`, "2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1.tsv", sep="\t", row.names=FALSE)

2022-07-10
`2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Sampling month")

`temp_fig_5` <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Season") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-10_season_effect_bins_foldchange_greater_1_non-logged_BH_pvalues_under_005.svg", `temp_fig_5`)

`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Host species")

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4"))

`temp_fig_5` <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-10_host_effect_bins_foldchange_greater_1_or_less_than_negative_1_non-logged_BH_pvalues_under_005.svg", `temp_fig_5`)

`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean > 1 & `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05 | `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean < -1 & `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05, ]

`2022-07-10_true_host_effect_bins` <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05 & `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean > 1 | `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean < -1, ]

`2022-07-10_true_host_effect_bins` <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05 & `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean > 1 | `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05 & '2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean < -1, ]

species_filtered_one <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05, ]
species_filtered_intermediate <- `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean > 1 | `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean < -1, ]
species_filtered_two <- `species_filtered_one`[`species_filtered_one`$Bin.name %in% `species_filtered_intermediate`$Bin.name, ]
`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1` <- species_filtered_two

season_filtered_one <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$BH_corrected_pvalues < 0.05, ]
season_filtered_intermediate <- `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$Log2.add.one.fold.change.season.geometric_mean > 1 | `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$Log2.add.one.fold.change.season.geometric_mean < -1, ]
season_filtered_two <- `season_filtered_one`[`season_filtered_one`$Bin.name %in% `season_filtered_intermediate`$Bin.name, ]
`2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1` <- season_filtered_two
temp_df <- `2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`
`2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1` <- `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$Bin.name %in% `temp_df`$Bin.name, ]
write.table(`2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`, "2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1.tsv", sep="\t", row.names=FALSE)

`temp_fig_5` <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Season") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + ggtitle("Seasonal variation bins, logfold change over/under 1/-1 and BH corrected P-value below 0.05")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-10_season_effect_bins_foldchange_greater_1_non-logged_BH_pvalues_under_005.svg", `temp_fig_5`)

species_filtered_one <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05, ]
species_filtered_intermediate <- `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean > 1 | `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Log2.add.one.fold.change.species.geometric_mean < -1, ]
species_filtered_two <- `species_filtered_one`[`species_filtered_one`$Bin.name %in% `species_filtered_intermediate`$Bin.name, ]
`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1` <- species_filtered_two
temp_df <- `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`
`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1` <- `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`$Bin.name %in% `temp_df`$Bin.name, ]
write.table(`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`, "2022-07-09_species_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1.tsv", sep="\t", row.names=FALSE)

`temp_fig_5` <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4"))

`temp_fig_5` <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name & grepl(glob2rx("Bin_c*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable) & ! grepl(glob2rx("Bin_c1*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + ggtitle("1 of 3\nHost species variation bins, logfold change over/under 1/-1 and BH corrected\nP-value below 0.05")

`temp_fig_5` <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name & grepl(glob2rx("Bin_c1*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + ggtitle("2 of 3\nHost species variation bins, logfold change over/under 1/-1 and BH corrected\nP-value below 0.05")

`temp_fig_5` <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name & grepl(glob2rx("Bin_m*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + ggtitle("3 of 3\nHost species variation bins, logfold change over/under 1/-1 and BH corrected\nP-value below 0.05")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-10_C_bins_only_set_1_fig_host_effect_bins_foldchange_greater_1_or_less_than_negative_1_non-logged_BH_pvalues_under_005.svg", device = "svg", `temp_fig_5`)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-10_C_bins_only_set_2_fig_host_effect_bins_foldchange_greater_1_or_less_than_negative_1_non-logged_BH_pvalues_under_005.svg", device = "svg", `temp_fig_5`)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-10_M_bins_only_fig_host_effect_bins_foldchange_greater_1_or_less_than_negative_1_non-logged_BH_pvalues_under_005.svg", device = "svg", `temp_fig_5`)

`2022_07_11_temp_table_host_bins_to_get_pvalues_for_excel_table` <- temp_df

2022-07-11
`temp_fig_5` <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ], aes(x = value, y = Season, fill = Season, shape = Season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, log 10 transformed, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Season") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + ggtitle("Seasonal variation bins, logfold change over/under 1/-1 and BH corrected P-value below 0.05") + scale_x_continuous(trans="log10")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-11_season_effect_bins_foldchange_greater_1_non-logged_BH_pvalues_under_005_log_scale_mapping_counts.svg", `temp_fig_5`)
`temp_fig_5` <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name & grepl(glob2rx("Bin_c*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable) & ! grepl(glob2rx("Bin_c1*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, log10 transformed, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + ggtitle("1 of 3\nHost species variation bins, logfold change over/under 1/-1 and BH corrected\nP-value below 0.05")  + scale_x_continuous(trans="log10")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-11_C_bins_only_set_1_fig_host_effect_bins_foldchange_greater_1_or_less_than_negative_1_non-logged_BH_pvalues_under_005_log_scale_mapping_counts.svg", device = "svg", `temp_fig_5`)
`temp_fig_5` <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name & grepl(glob2rx("Bin_c1*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, log10 transformed, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + ggtitle("2 of 3\nHost species variation bins, logfold change over/under 1/-1 and BH corrected\nP-value below 0.05") + scale_x_continuous(trans="log10")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-11_C_bins_only_set_2_fig_host_effect_bins_foldchange_greater_1_or_less_than_negative_1_non-logged_BH_pvalues_under_005_log_scale_mapping_counts.svg", device = "svg", `temp_fig_5`)
`temp_fig_5` <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name & grepl(glob2rx("Bin_m*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, log10 transformed, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + ggtitle("3 of 3\nHost species variation bins, logfold change over/under 1/-1 and BH corrected\nP-value below 0.05") + scale_x_continuous(trans="log10")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-11_M_bins_only_fig_host_effect_bins_foldchange_greater_1_or_less_than_negative_1_non-logged_BH_pvalues_under_005_log_scale_mapping_counts.svg", device = "svg", `temp_fig_5`)

temp_fig_1 <- ggplot(data = `2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_species`, aes(x = Log2.add.one.fold.change.species.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of arithmetic means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_2 <- ggplot(data = `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`, aes(x = Log2.add.one.fold.change.species.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_3 <- ggplot(data = `2022-07-08_medians_manual_log2_add_1_foldchange_table_species`, aes(x = Log2.add.one.fold.change.species.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of median per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_4 <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`, aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = FALSE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species")
temp_fig_5 <- grid.arrange(temp_fig_1, temp_fig_2, temp_fig_3, temp_fig_4)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-11_combined_species_plots_log2_add_one_fold_changes.svg", `temp_fig_5`)

temp_fig_1 <- ggplot(data = `2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_season`, aes(x = Log2.add.one.fold.change.season.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of arithmetic means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_2 <- ggplot(data = `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`, aes(x = Log2.add.one.fold.change.season.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_3 <- ggplot(data = `2022-07-08_medians_manual_log2_add_1_foldchange_table_season`, aes(x = Log2.add.one.fold.change.season.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of median per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_4 <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`, aes(x = value, y = Season, fill = Season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Season") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75)
temp_fig_5 <- grid.arrange(temp_fig_1, temp_fig_2, temp_fig_3, temp_fig_4)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-11_combined_season_plots_log2_add_one_fold_changes.svg", `temp_fig_5`)


`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`[`2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`$Bin.name ]

view(`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`)
`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05, ]
nrow(`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05, ])
`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season` <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$BH_corrected_pvalues < 0.05, ]
write.table(`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`, "2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season.tsv", sep="\t", row.names=FALSE)

`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$BH_corrected_pvalues < 0.05, ]
nrow(`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$BH_corrected_pvalues < 0.05, ])
`2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host` <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$BH_corrected_pvalues < 0.05, ]
write.table(`2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`, "2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host.tsv", sep="\t", row.names=FALSE)

nrow(`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`[`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name %in% `2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name, ])
nrow(`2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`[`2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name %in% `2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name, ])

`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`[`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name %in% `2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name, ]
nrow(`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`[`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name %notin% `2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name, ])

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[grepl(glob2rx("Bin_c1233"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable),], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + scale_colour_manual(values = c ("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, log10 transformed, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species and sample month") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + scale_x_continuous(trans="log10") + scale_discrete_manual(aesthetics = "point_shape", values = c("0", "15", "2", "17")) + labs(fill = "Host species and sample month", colour = "Host species and sample month", shape = "Host species and sample month")

ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[grepl(glob2rx("Bin_c123*"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable),], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + scale_colour_manual(values = c ("#bbddff", "#0a85ff", "#ffc51e", "#d16700")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, log10 transformed, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species and sample month") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + scale_x_continuous(trans="log10") + scale_discrete_manual(aesthetics = "point_shape", values = c("0", "15", "2", "17")) + labs(fill = "Host species and sample month", colour = "Host species and sample month", shape = "Host species and sample month")

ggpubr::ggboxplot(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[grepl(glob2rx("Bin_c1233"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable),], x = "Species_x_season", y = "value", color = "Species_x_season", ylab = "Mapped segement read count, library and contig size adjusted", xlab = "Host species and sample month") + stat_compare_means(method = "kruskal.test", aes(label = ..p.signif..))

`2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins` <- `2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`[`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name %in% `2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name, ]
write.table(`2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`, "2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins.tsv", sep="\t", row.names=FALSE)

`2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`[`2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name %notin% `2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name, ]
`2022-07-11_BH_corrected_pvalues_under_005_season_effect_ONLY` <- `2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`[`2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name %notin% `2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name, ]
write.table(`2022-07-11_BH_corrected_pvalues_under_005_season_effect_ONLY`, "2022-07-11_BH_corrected_pvalues_under_005_season_effect_ONLY.tsv", sep="\t", row.names=FALSE)

nrow(`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`[`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name %notin% `2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name, ])
`2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY` <- `2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`[`2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name %notin% `2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name, ]
write.table(`2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY`, "2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY.tsv", sep="\t", row.names=FALSE)

View(`2022.07.11_all_strict_bins_and_GTDB.TK_taxonomy`)

2022-07-12
ggplot(data = `2022.07.11_all_strict_bins_and_GTDB.TK_taxonomy`, aes(x = GTDB.TK.taxonomy,)) + geom_bar()

`2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`
`2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`[`2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`$Bin.name %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name | `2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`$Bin.name %in% `2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ]

`2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY`
`2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY`[`2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY`$Bin.name %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ]$Log2.add.one.fold.change.species.geometric_mean
`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`[`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name %in% `2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY`$Bin.name, ]

`2022-07-11_BH_corrected_pvalues_under_005_season_effect_ONLY`
`2022-07-11_BH_corrected_pvalues_under_005_season_effect_ONLY`[`2022-07-11_BH_corrected_pvalues_under_005_season_effect_ONLY`$Bin.name %in% `2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ]

`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`

`2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`

2022-07-12
`2022.07.12_short_list_number_bins_by_category_terms_of_BH_pvalue_and_fold_change` <- read.delim2("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-12_short_list_number_bins_by_category_terms_of_BH_pvalue_and_fold_change.txt")
View(`2022.07.12_short_list_number_bins_by_category_terms_of_BH_pvalue_and_fold_change`)

ggplot(data = `2022.07.12_short_list_number_bins_by_category_terms_of_BH_pvalue_and_fold_change`[! grepl(glob2rx("Strict*"), `2022.07.12_short_list_number_bins_by_category_terms_of_BH_pvalue_and_fold_change`$Category), ], aes(x = Number.of.bins, y = Category, fill = Category)) + geom_bar(stat = 'identity') + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(legend.position = "none")

venn.diagram(x = list(`2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`$Bin.name, `2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY`$Bin.name, `2022-07-11_BH_corrected_pvalues_under_005_season_effect_ONLY`$Bin.name), category.names = c("Host and season effect bins", "Host effect only bins", "Season effect only bins"), filename = 'Test.png', output = TRUE)
venn.diagram(x = list(`2022.07.11_all_strict_bins_and_GTDB.TK_taxonomy`$Bin.name, `2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY`$Bin.name, `2022-07-11_BH_corrected_pvalues_under_005_season_effect_ONLY`$Bin.name), category.names = c("All bins", "Host only bins", "Season only bins"), filename = 'Test.png', output = TRUE)
venn.diagram(x = list(`2022.07.11_all_strict_bins_and_GTDB.TK_taxonomy`$Bin.name, `2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`$Bin.name, `2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY`$Bin.name, `2022-07-11_BH_corrected_pvalues_under_005_season_effect_ONLY`$Bin.name), category.names = c("All strict bins", "Host and season effect bins", "Host effect only bins", "Season effect only bins"), filename = 'Test.png', output = TRUE)

`2022-07-12_bins_host_and_season_significant_BH_corrected_and_logfold_over_1_at_least_one_group` <- `2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`[`2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`$Bin.name %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name | `2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`$Bin.name %in% `2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name, ]

2022-07-13
`2022.07.13_all_bins_which_have_any_category_significant_pvalue_including_those_for_both_species_and_season` <- read.csv("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-13_all_bins_which_have_any_category_significant_pvalue_including_those_for_both_species_and_season.txt", sep="")
palecolours <- brewer.pal(3, "Pastel2")
venn.diagram(x = list(`2022.07.11_all_strict_bins_and_GTDB.TK_taxonomy`$Bin.name, `2022.07.13_all_bins_which_have_any_category_significant_pvalue_including_those_for_both_species_and_season`$Bin.name, `2022.07.11_all_strict_bins_and_GTDB.TK_taxonomy`[`2022.07.11_all_strict_bins_and_GTDB.TK_taxonomy`$Bin.name %notin% `2022.07.13_all_bins_which_have_any_category_significant_pvalue_including_those_for_both_species_and_season`$Bin.name, ]), category.names = c("All strict bins", "Corrected P-value significant bins", "Not significant bins"), filename = 'Test.png', height = 1080, width = 1080, compression = "lzw", lwd = 2, lty = "blank", fill = palecolours, output = TRUE)

`2022.07.13_number_bins_just_by_BH_corrected_pvalue_significance_by_factor_or_combined_factor` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-13_number_bins_just_by_BH_corrected_pvalue_significance_by_factor_or_combined_factor.txt")
View(`2022.07.13_number_bins_just_by_BH_corrected_pvalue_significance_by_factor_or_combined_factor`)

ggplot(data = `2022.07.13_number_bins_just_by_BH_corrected_pvalue_significance_by_factor_or_combined_factor`, aes(fill = Category, y = Number.of.bins, x = "Bins and significance by category")) + geom_bar(position="fill", stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("") + ylab("Proportion of all strict bins") + scale_fill_manual(values = c("#66023c", "#848482", "#4ffa18", "#ff8d1e"))
temp_fig_5 <- ggplot(data = `2022.07.13_number_bins_just_by_BH_corrected_pvalue_significance_by_factor_or_combined_factor`, aes(fill = Category, y = Number.of.bins, x = "Bins and significance by category")) + geom_bar(position="fill", stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("") + ylab("Proportion of all strict bins") + scale_fill_manual(values = c("#66023c", "#848482", "#4ffa18", "#ff8d1e"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-13_number_bins_just_by_BH_corrected_pvalue_significance_by_factor_or_combined_factor_proportional_stacked_bar_chart.svg", `temp_fig_5`)


`2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY`[`2022-07-11_BH_corrected_pvalues_under_005_species_effect_ONLY`$Bin.name %in% `2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`$Bin.name ,]

`2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`[`2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`$Bin.name %in% ]

`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`

`2022-07-09_season_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`

`2022-07-09_host_effect_bins_BH_corrected_not_logged_pvalue_significant_effect_size_over_1_or_under_negative_1`

bin_names_list_venn_1 <- list("Significant bins" = `2022.07.13_all_bins_which_have_any_category_significant_pvalue_including_those_for_both_species_and_season`$Bin.name, "Significant bins" = `2022.07.13_all_bins_which_have_any_category_significant_pvalue_including_those_for_both_species_and_season`$Bin.name, "Not significant bins" = as.character(`2022.07.11_all_strict_bins_and_GTDB.TK_taxonomy`[`2022.07.11_all_strict_bins_and_GTDB.TK_taxonomy`$Bin.name %notin% `2022.07.13_all_bins_which_have_any_category_significant_pvalue_including_those_for_both_species_and_season`$Bin.name, ]))

ggVennDiagram(bin_names_list_venn_2)
venn.diagram(x = list("Host significant bins" = `2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name, "Season significant bins" = `2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name, "Host and season significant bins" = `2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`$Bin.name), filename = 'Test.png', height = 1080, width = 1080, compression = "lzw", lwd = 2, lty = "blank", fill = palecolours, output = TRUE)
bin_names_list_venn_2 <- list("Host significant bins" = `2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name, "Season significant bins" = `2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name, "Host and season significant bins" = `2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`$Bin.name)
gplots::venn(bin_names_list_venn_2)

createVennObj(nSets = 3, )
myV <- plotVenn(list("Host significant bins" = `2022-07-11_BH_corrected_pvalues_under_005_host_effect_all_bins_not_exclusive_with_season`$Bin.name, "Season significant bins" = `2022-07-11_BH_corrected_pvalues_under_005_season_effect_all_bins_not_exclusive_with_host`$Bin.name, "Host and season significant bins" = `2022-07-11_BH_corrected_pvalues_under_005_host_AND_season_bins`$Bin.name), nCycles = 10)
showSVG(nVennObj = myV, opacity = 0.5, borderWidth = 2)

`2022-07-13_geometric_means_all_values_different_things_and_taxID` <- data.frame(matrix(ncol = 10, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.geometric.mean.mapping.count", "AR.geometric.mean.mapping.count", "AC.to.AR.BH.corrected.P.value", "Log2.add.one.fold.change.species.geometric.mean", "June.geometric.mean.mapping.count", "November.geometric.mean.mapping.count", "June.to.November.BH.corrected.P.value", "Log2.add.one.fold.change.season.geometric.mean", "GTDB.TK.Taxonomy"))))
PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_AR_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_log2_add_one_fold_change_geometric_means_host <- log2(((PLACEHOLDER_AR_geometric_means_mappings - PLACEHOLDER_AC_geometric_means_mappings) / PLACEHOLDER_AC_geometric_means_mappings) + 1)
PLACEHOLDER_J_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_N_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_log2_add_one_fold_change_geometric_means_season <- log2(((PLACEHOLDER_N_geometric_means_mappings - PLACEHOLDER_J_geometric_means_mappings) / PLACEHOLDER_J_geometric_means_mappings) + 1)
PLACEHOLDER_taxonomy <- `2022.07.11_all_strict_bins_and_GTDB.TK_taxonomy`[`2022.07.11_all_strict_bins_and_GTDB.TK_taxonomy`$Bin.name == "PLACEHOLDER", ]$GTDB.TK.taxonomy
`2022-07-13_geometric_means_all_values_different_things_and_taxID`[nrow(`2022-07-13_geometric_means_all_values_different_things_and_taxID`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_geometric_means_mappings, PLACEHOLDER_AR_geometric_means_mappings, PLACEHOLDER_BH_corrected_pvalue_host, PLACEHOLDER_log2_add_one_fold_change_geometric_means_host, PLACEHOLDER_J_geometric_means_mappings, PLACEHOLDER_N_geometric_means_mappings, PLACEHOLDER_BH_corrected_pvalue_season, PLACEHOLDER_log2_add_one_fold_change_geometric_means_season, PLACEHOLDER_taxonomy)
`2022-07-13_geometric_means_all_values_different_things_and_taxID`[, c(2:9)] <- sapply(`2022-07-13_geometric_means_all_values_different_things_and_taxID`[, c(2:9)], as.numeric)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))

`2022-07-13_geometric_means_all_values_different_things_and_taxID`
write.table(`2022-07-13_geometric_means_all_values_different_things_and_taxID`, "2022-07-13_geometric_means_all_values_different_things_and_taxID.tsv", sep="\t", row.names=FALSE)

2022-07-14
`2022.07.13_geometric_means_all_values_different_things_and_taxID`[`2022.07.13_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022.07.13_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean > 1, ]

ggplot(data = `2022.07.13_geometric_means_all_values_different_things_and_taxID`[`2022.07.13_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022.07.13_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean > 1, ], aes(x = Genus.or.family.GTDB.TK, fill = Genus.or.family.GTDB.TK)) + geom_bar(stat = "count", position = "dodge")

ggplot(data = `2022.07.12_short_list_number_bins_by_category_terms_of_BH_pvalue_and_fold_change`[! grepl(glob2rx("Strict*"), `2022.07.12_short_list_number_bins_by_category_terms_of_BH_pvalue_and_fold_change`$Category), ], aes(x = Number.of.bins, y = Category, fill = Category)) + geom_bar(stat = 'identity') + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(legend.position = "none")

ggplot(data = `2022.07.13_geometric_means_all_values_different_things_and_taxID`[`2022.07.13_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022.07.13_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean > 1, ], aes(x = Genus.or.family.GTDB.TK, fill = Genus.or.family.GTDB.TK)) + geom_bar(stat = "count", position = "dodge")
ggplot(data = `2022.07.13_geometric_means_all_values_different_things_and_taxID`[`2022.07.13_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022.07.13_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean < -1, ], aes(x = Genus.or.family.GTDB.TK, fill = Genus.or.family.GTDB.TK)) + geom_bar(stat = "count", position = "dodge")

`2022.07.13_geometric_means_all_values_different_things_and_taxID`[`2022.07.13_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05, ]
view(`2022.07.13_geometric_means_all_values_different_things_and_taxID`[`2022.07.13_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05, ])

view(`2022.07.13_geometric_means_all_values_different_things_and_taxID`[`2022.07.13_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022.07.13_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean < -1 | `2022.07.13_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean > 1, ])

`2022-07-14_geometric_means_all_values_different_things_and_taxID` <- data.frame(matrix(ncol = 11, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.geometric.mean.mapping.count", "AR.geometric.mean.mapping.count", "AC.to.AR.BH.corrected.P.value", "Log2.add.one.fold.change.species.geometric.mean", "June.geometric.mean.mapping.count", "November.geometric.mean.mapping.count", "June.to.November.BH.corrected.P.value", "Log2.add.one.fold.change.season.geometric.mean", "GTDB.TK.Taxonomy", "GTDB.TK.Family.or.species"))))

PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_AR_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_log2_add_one_fold_change_geometric_means_host <- log2(((PLACEHOLDER_AR_geometric_means_mappings - PLACEHOLDER_AC_geometric_means_mappings) / PLACEHOLDER_AC_geometric_means_mappings) + 1)
PLACEHOLDER_J_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_N_geometric_means_mappings <- exp(mean(log(`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`[`2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`$variable), ]$value)))
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`[`2022_06_29_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_log2_add_one_fold_change_geometric_means_season <- log2(((PLACEHOLDER_N_geometric_means_mappings - PLACEHOLDER_J_geometric_means_mappings) / PLACEHOLDER_J_geometric_means_mappings) + 1)
PLACEHOLDER_taxonomy <- `2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`[`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`$Bin.name == "PLACEHOLDER", ]$Bin.long.taxa
PLACEHOLDER_short_taxonomy <- `2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`[`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`$Bin.name == "PLACEHOLDER", ]$Bin.family.or.genus
`2022-07-14_geometric_means_all_values_different_things_and_taxID`[nrow(`2022-07-14_geometric_means_all_values_different_things_and_taxID`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_geometric_means_mappings, PLACEHOLDER_AR_geometric_means_mappings, PLACEHOLDER_BH_corrected_pvalue_host, PLACEHOLDER_log2_add_one_fold_change_geometric_means_host, PLACEHOLDER_J_geometric_means_mappings, PLACEHOLDER_N_geometric_means_mappings, PLACEHOLDER_BH_corrected_pvalue_season, PLACEHOLDER_log2_add_one_fold_change_geometric_means_season, PLACEHOLDER_taxonomy, PLACEHOLDER_short_taxonomy)
`2022-07-14_geometric_means_all_values_different_things_and_taxID`[, c(2:9)] <- sapply(`2022-07-14_geometric_means_all_values_different_things_and_taxID`[, c(2:9)], as.numeric)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))

write.table(`2022-07-14_geometric_means_all_values_different_things_and_taxID`, "2022-07-14_geometric_means_all_values_different_things_and_taxID.tsv", sep="\t", row.names=FALSE)

2022-07-15
`2022-07-14_geometric_means_all_values_different_things_and_taxID`[`2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean < -1 | `2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean > 1, ]
view(`2022-07-14_geometric_means_all_values_different_things_and_taxID`[`2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean < -1 | `2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean > 1, ])
write.table(`2022-07-14_geometric_means_all_values_different_things_and_taxID`[`2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean < -1 | `2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean > 1, ], "2022-07-15_geometric_means_BH_significant_and_meaningful_effect_fold_change_species_detailed_table.tsv", sep = "\t", row.names = FALSE)

`2022-07-14_geometric_means_all_values_different_things_and_taxID`[`2022-07-14_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean < -1 | `2022-07-14_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean > 1, ]
write.table(`2022-07-14_geometric_means_all_values_different_things_and_taxID`[`2022-07-14_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean < -1 | `2022-07-14_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean > 1, ], "2022-07-15_geometric_means_BH_significant_and_meaningful_effect_fold_change_season_detailed_table.tsv", sep = "\t", row.names = FALSE)

temp_species <- `2022-07-14_geometric_means_all_values_different_things_and_taxID`[`2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean < -1 | `2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean > 1, ]
temp_season <- `2022-07-14_geometric_means_all_values_different_things_and_taxID`[`2022-07-14_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean < -1 | `2022-07-14_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean > 1, ]

temp_species[temp_species$Bin.name %in% temp_season$Bin.name ,]
temp_species[temp_species$Bin.name %notin% temp_season$Bin.name ,]
nrow(temp_species[temp_species$Bin.name %notin% temp_season$Bin.name ,])
temp_season[temp_season$Bin.name %in% temp_species$Bin.name, ]
temp_season[temp_season$Bin.name %notin% temp_species$Bin.name, ]
nrow(temp_season[temp_season$Bin.name %notin% temp_species$Bin.name, ])

`2022.07.15_all_November_significant_and_meaningful_bins_unique_potential_products_count` <- read.delim2("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-15_all_November_significant_and_meaningful_bins_unique_potential_products_count.tsv")
View(`2022.07.15_all_November_significant_and_meaningful_bins_unique_potential_products_count`)

rowMeans(`2022.07.15_all_November_significant_and_meaningful_bins_unique_potential_products_count`[,2:10, ])
temp_season <- `2022.07.15_all_November_significant_and_meaningful_bins_unique_potential_products_count`
temp_season$Mean <- rowMeans(`2022.07.15_all_November_significant_and_meaningful_bins_unique_potential_products_count`[,2:10, ])

`2022-07-15_products_list`$Products <- temp_season$Product
names(`2022-07-15_products_list`)[1] <- 'Products'
names(`2022-07-15_products_list`)[names(`2022-07-15_products_list`) == "x" ] <- "Products"

`2022-07-15_products_list` <- data.frame(matrix(ncol = 3, nrow = 0, dimnames = list(NULL, c("Products", "AC.arithemetic.mean.count", "AR.arithemetic.mean.count"))))
`2022-07-15_products_list`$Products[which(`2022-07-15_products_list`$Products <= 0)] <- temp_season$Product

temp_season$Mean(order(temp_season$Mean, decreasing = TRUE), 10)
temp_season[order(temp_season$Mean, decreasing = TRUE), ]
as.data.frame(temp_season[order(temp_season$Mean, decreasing = TRUE), ]) %>% dplyr::slice(1:10)
temp_season[order(temp_season$Mean, decreasing = TRUE), ] %>% dplyr::slice(1:10)

`2022.07.15_all_Acomys_cahirinus_significant_and_meaningful_bins_unique_potential_products_count` <- read.delim2("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-15_all_Acomys_cahirinus_significant_and_meaningful_bins_unique_potential_products_count.tsv")
View(`2022.07.15_all_Acomys_cahirinus_significant_and_meaningful_bins_unique_potential_products_count`)

`2022.07.15_all_Acomys_cahirinus_significant_and_meaningful_bins_unique_potential_products_count`$Arithmetic.mean <- rowMeans(`2022.07.15_all_Acomys_cahirinus_significant_and_meaningful_bins_unique_potential_products_count`[,2:18, ])

`2022.07.15_all_Acomys_russatus_significant_and_meaningful_bins_unique_potential_products_count` <- read.delim2("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-15_all_Acomys_russatus_significant_and_meaningful_bins_unique_potential_products_count.tsv")
View(`2022.07.15_all_Acomys_russatus_significant_and_meaningful_bins_unique_potential_products_count`)
`2022.07.15_all_Acomys_russatus_significant_and_meaningful_bins_unique_potential_products_count`$Arithmetic.mean <- rowMeans(`2022.07.15_all_Acomys_russatus_significant_and_meaningful_bins_unique_potential_products_count`[,2:32, ])

ggplot() + geom_point(aes(x = `2022.07.15_all_Acomys_cahirinus_significant_and_meaningful_bins_unique_potential_products_count`$Arithmetic.mean, y = `2022.07.15_all_Acomys_russatus_significant_and_meaningful_bins_unique_potential_products_count`$Arithmetic.mean)) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_log10() + scale_y_log10()
ggplot() + geom_point(aes(x = `2022.07.15_all_Acomys_cahirinus_significant_and_meaningful_bins_unique_potential_products_count`$Arithmetic.mean, y = `2022.07.15_all_Acomys_russatus_significant_and_meaningful_bins_unique_potential_products_count`$Arithmetic.mean)) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_log10(labels = comma) + scale_y_log10(labels = comma) + xlab("Arithmetic mean count in A. cahirinus enriched bins") + ylab("Arithmetic mean count in A. russatus enriched bins") + ggtitle("DFAST detections of products in Bins")

diff(`2022.07.15_all_Acomys_cahirinus_significant_and_meaningful_bins_unique_potential_products_count`$Arithmetic.mean, `2022.07.15_all_Acomys_russatus_significant_and_meaningful_bins_unique_potential_products_count`$Arithmetic.mean)

temp_fig_1 <- ggplot(data = `2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_species`, aes(x = Log2.add.one.fold.change.species.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of arithmetic means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_2 <- ggplot(data = `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_species`, aes(x = Log2.add.one.fold.change.species.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_3 <- ggplot(data = `2022-07-08_medians_manual_log2_add_1_foldchange_table_species`, aes(x = Log2.add.one.fold.change.species.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of median per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_4 <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`, aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = FALSE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Host species")
temp_fig_5 <- grid.arrange(temp_fig_1, temp_fig_2, temp_fig_3, temp_fig_4)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-11_combined_species_plots_log2_add_one_fold_changes.svg", `temp_fig_5`)

temp_fig_1 <- ggplot(data = `2022-07-08_arithmetic_means_manual_log2_add_1_foldchange_table_season`, aes(x = Log2.add.one.fold.change.season.arithmetic_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of arithmetic means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_2 <- ggplot(data = `2022-07-08_geometric_means_manual_log2_add_1_foldchange_table_season`, aes(x = Log2.add.one.fold.change.season.geometric_mean, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_3 <- ggplot(data = `2022-07-08_medians_manual_log2_add_1_foldchange_table_season`, aes(x = Log2.add.one.fold.change.season.median, y = X.log10.BH.corrected.pvalue)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of median per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_4 <- ggplot(data = `2022_06_29_melted_transposed_summed_lib_cont_adjusted_all_samples`, aes(x = value, y = Season, fill = Season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Season") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75)
temp_fig_5 <- grid.arrange(temp_fig_1, temp_fig_2, temp_fig_3, temp_fig_4)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-11_combined_season_plots_log2_add_one_fold_changes.svg", `temp_fig_5`)

`2022-07-14_geometric_means_all_values_different_things_and_taxID`
ggplot(data = `2022-07-14_geometric_means_all_values_different_things_and_taxID`, aes(x = Log2.add.one.fold.change.species.geometric.mean, y = -log10(AC.to.AR.BH.corrected.P.value), colour = GTDB.TK.Family.or.species)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")

ggplot(data = `2022-07-14_geometric_means_all_values_different_things_and_taxID`, aes(x = Log2.add.one.fold.change.species.geometric.mean, y = -log10(AC.to.AR.BH.corrected.P.value))) + geom_point(aes(colour = ifelse(Log2.add.one.fold.change.species.geometric.mean > 1 & -log10(AC.to.AR.BH.corrected.P.value) > -log10(0.05) | Log2.add.one.fold.change.species.geometric.mean < -1 & -log10(AC.to.AR.BH.corrected.P.value) > -log10(0.05), `GTDB.TK.Family.or.species`, "Black"))) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_color_identity()

ggplot(data = `2022-07-14_geometric_means_all_values_different_things_and_taxID`, aes(x = Log2.add.one.fold.change.species.geometric.mean, y = -log10(AC.to.AR.BH.corrected.P.value))) + geom_point(aes(colour = ifelse(Log2.add.one.fold.change.species.geometric.mean > 1 & -log10(AC.to.AR.BH.corrected.P.value) > -log10(0.05) | Log2.add.one.fold.change.species.geometric.mean < -1 & -log10(AC.to.AR.BH.corrected.P.value) > -log10(0.05), "blue", "Black"))) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_color_identity()

`2022-07-14_geometric_means_all_values_different_things_and_taxID` >%> mutate(Colour = )

`2022-07-14_geometric_means_all_values_different_things_and_taxID`[`2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean > 1 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 | `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean < -1 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05, ]

`2022-07-14_geometric_means_all_values_different_things_and_taxID`[`2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean > 1 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 | `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean < -1 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05, ]

unique(`2022-07-14_geometric_means_all_values_different_things_and_taxID`[`2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean > 1 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 | `2022-07-14_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean < -1 & `2022-07-14_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05, ]$GTDB.TK.Family.or.species)

2022-07-16
pie_chart_df_sig_bins_type <- data.frame(group = c("AC increased", "AR increased", "November increased", "No meaningful changes"), values=c(17, 31, 9, 199))

ggplot(data = pie_chart_df_sig_bins_type, aes(x = "", y = values, fill = group)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("") + xlab("") + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#848482", "#c318fa")) + ggtitle("Number of significant bins") + guides(fill = guide_legend(title = "Category")) + labs(caption = "Meaningful change was log2()+1 change of >1 or <-1")
temp_fig_5 <- ggplot(data = pie_chart_df_sig_bins_type, aes(x = "", y = values, fill = group)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("") + xlab("") + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#848482", "#c318fa")) + ggtitle("Number of significant bins") + guides(fill = guide_legend(title = "Category")) + labs(caption = "Meaningful change was log2()+1 change of >1 or <-1")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-16_number_significant_bins_by_category_of_meaningful_fold_change.svg", `temp_fig_5`)
`2022-07-16_number_significant_bins_by_category_of_meaningful_fold_change` <- pie_chart_df_sig_bins_type
write.table(`2022-07-16_number_significant_bins_by_category_of_meaningful_fold_change`, "2022-07-16_number_significant_bins_by_category_of_meaningful_fold_change.tsv", sep="\t", row.names=FALSE)

`temp_species_comparing_products` <- data.frame(matrix(ncol = 1, nrow = 7736, dimnames = list(NULL, c("placeholder"))))
`temp_species_comparing_products`$Products <- `2022.07.15_all_Acomys_russatus_significant_and_meaningful_bins_unique_potential_products_count`[1]
`temp_species_comparing_products`$Bin_m959 <- `2022.07.15_all_Acomys_cahirinus_significant_and_meaningful_bins_unique_potential_products_count`[18]
`temp_species_comparing_products`$Bin_m419 <- `2022.07.15_all_Acomys_russatus_significant_and_meaningful_bins_unique_potential_products_count`[25]
`temp_species_comparing_products`$second_from_first <- `temp_species_comparing_products`$Bin_m959 - `temp_species_comparing_products`$Bin_m419
`temp_species_comparing_products`$Bin_m959 <- unlist(`temp_species_comparing_products`$Bin_m959)
`temp_species_comparing_products`$Bin_m419 <- unlist(`temp_species_comparing_products`$Bin_m419)
`temp_species_comparing_products`$second_from_first <- unlist(`temp_species_comparing_products`$second_from_first)
ggplot(data = temp_species_comparing_products, aes(x = Bin_m959, Bin_m419)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_log10(labels = comma) + scale_y_log10(labels = comma)

ggplot(data = temp_species_comparing_products, aes(x = Bin_m959, y = Bin_m419, colour = second_from_first)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_log10(labels = comma) + scale_y_log10(labels = comma)

2022-07-18
`2022.07.18_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-18_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins.txt")
View(`2022.07.18_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`)
`2022.07.18_melted_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins` <- reshape2::melt(`2022.07.18_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`, id=c("Bins", "Species", "Season", "Sequencing", "Species_x_season"))
`2022.07.18_melted_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Sequencing == "Novogene", ]
`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins` <- `2022.07.18_melted_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Sequencing == "Novogene", ]
view(`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`)
`2022.07.18_PCA_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins` <- prcomp(`2022.07.18_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[,c(2:349)], center = TRUE, scale. = FALSE)
write.table(`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`, "2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins.tsv", sep="\t", row.names=FALSE)

filter(`2022.07.18_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`, Sequencing == "Novogene")
`2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins` <- filter(`2022.07.18_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`, Sequencing == "Novogene")
`2022.07.18_PCA_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins` <- prcomp(`2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[,c(2:349)], center = TRUE, scale. = FALSE)

ggbiplot(`2022.07.18_PCA_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Season), color=factor(`2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + labs(title = "No transformation")

`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene` <- data.frame(matrix(ncol = 2, nrow = 0, dimnames = list(NULL, c("Bin name", "Uncorrected pvalue"))))
`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene` <- data.frame(matrix(ncol = 2, nrow = 0, dimnames = list(NULL, c("Bin name", "Uncorrected pvalue"))))

PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_host_kruskal <- kruskal.test(value ~ Species, data = `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable == "PLACEHOLDER", ])
PLACEHOLDER_season_kruskal <- kruskal.test(value ~ Season, data = `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable == "PLACEHOLDER", ])
PLACEHOLDER_host_pvalue <- PLACEHOLDER_host_kruskal$p.value
PLACEHOLDER_season_pvalue <- PLACEHOLDER_season_kruskal$p.value
`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`[nrow(`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_host_pvalue)
`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`[, c(2)] <- sapply(`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`[, c(2)], as.numeric)
`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`[nrow(`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_season_pvalue)
`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`[, c(2)] <- sapply(`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`[, c(2)], as.numeric)
rm(PLACEHOLDER_bin_name)
rm(PLACEHOLDER_host_kruskal)
rm(PLACEHOLDER_season_kruskal)
rm(PLACEHOLDER_host_pvalue)
rm(PLACEHOLDER_season_pvalue)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))

`2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene` <- `2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`[order(`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`$Uncorrected.pvalue),]
`2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene` <- `2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`[order(`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`$Uncorrected.pvalue),]
`2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`$BH_corrected_pvalues <- p.adjust(`2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`$Uncorrected.pvalue, method = "BH")
`2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`$BH_corrected_pvalues <- p.adjust(`2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`$Uncorrected.pvalue, method = "BH")

`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID` <- data.frame(matrix(ncol = 11, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.geometric.mean.mapping.count", "AR.geometric.mean.mapping.count", "AC.to.AR.BH.corrected.P.value", "Log2.add.one.fold.change.species.geometric.mean", "June.geometric.mean.mapping.count", "November.geometric.mean.mapping.count", "June.to.November.BH.corrected.P.value", "Log2.add.one.fold.change.season.geometric.mean", "GTDB.TK.Taxonomy", "GTDB.TK.Family.or.species"))))

PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_geometric_means_mappings <- exp(mean(log(`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable), ]$value)))
PLACEHOLDER_AR_geometric_means_mappings <- exp(mean(log(`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable), ]$value)))
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`[`2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_log2_add_one_fold_change_geometric_means_host <- log2(((PLACEHOLDER_AR_geometric_means_mappings - PLACEHOLDER_AC_geometric_means_mappings) / PLACEHOLDER_AC_geometric_means_mappings) + 1)
PLACEHOLDER_J_geometric_means_mappings <- exp(mean(log(`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable), ]$value)))
PLACEHOLDER_N_geometric_means_mappings <- exp(mean(log(`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable), ]$value)))
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`[`2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_log2_add_one_fold_change_geometric_means_season <- log2(((PLACEHOLDER_N_geometric_means_mappings - PLACEHOLDER_J_geometric_means_mappings) / PLACEHOLDER_J_geometric_means_mappings) + 1)
PLACEHOLDER_taxonomy <- `2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`[`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`$Bin.name == "PLACEHOLDER", ]$Bin.long.taxa
PLACEHOLDER_short_taxonomy <- `2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`[`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`$Bin.name == "PLACEHOLDER", ]$Bin.family.or.genus
`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_geometric_means_mappings, PLACEHOLDER_AR_geometric_means_mappings, PLACEHOLDER_BH_corrected_pvalue_host, PLACEHOLDER_log2_add_one_fold_change_geometric_means_host, PLACEHOLDER_J_geometric_means_mappings, PLACEHOLDER_N_geometric_means_mappings, PLACEHOLDER_BH_corrected_pvalue_season, PLACEHOLDER_log2_add_one_fold_change_geometric_means_season, PLACEHOLDER_taxonomy, PLACEHOLDER_short_taxonomy)
`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[, c(2:9)] <- sapply(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[, c(2:9)], as.numeric)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))

view(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`)
write.table(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`, "2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID.tsv", sep="\t", row.names=FALSE)
nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05, ])
nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05, ])
nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05, ])

nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 | `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05, ])

temp_fig_2 <- ggplot(data = `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`, aes(x = Log2.add.one.fold.change.species.geometric.mean, y = -log10(AC.to.AR.BH.corrected.P.value))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
temp_fig_2 <- ggplot(data = `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`, aes(x = Log2.add.one.fold.change.season.geometric.mean, y = -log10(June.to.November.BH.corrected.P.value))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")

ggplot(data = `2022.07.18_strict_bins_significant_by_category`, aes(fill = Category, y = Bins, x = Category)) + geom_bar(stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("") + ylab("Number of bins by category") + scale_fill_manual(values = c("#66023c", "#848482", "#4ffa18", "#ff8d1e"))
temp_fig_5 <- ggplot(data = `2022.07.18_strict_bins_significant_by_category`, aes(fill = Category, y = Bins, x = Category)) + geom_bar(stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("") + ylab("Number of bins by category") + scale_fill_manual(values = c("#66023c", "#848482", "#4ffa18", "#ff8d1e")) + coord_flip() + theme(legend.position = "none")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_18_strict_bins_significant_by_category.svg", `temp_fig_5`)

temp_fig_4 <- ggplot(data = `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`, aes(x = value, y = Species, fill = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + scale_x_log10()
temp_fig_5 <- grid.arrange(temp_fig_2, temp_fig_4)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-18_combined_species_plots_log2_add_one_fold_change_geometric_mean_violin_boxplot_mappings_only_paired_novogene.svg", `temp_fig_5`)

temp_fig_4 <- ggplot(data = `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`, aes(x = value, y = Season, fill = Season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#c318fa", "#4ffa18")) + scale_colour_manual(values = c ("#c318fa", "#4ffa18")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + xlab("All samples, summed mapped segment counts for bins, library and contig size(s) adjusted") + ylab("Season") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + scale_x_log10()

temp_fig_5 <- grid.arrange(temp_fig_2, temp_fig_4)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-18_combined_season_plots_log2_add_one_fold_change_geometric_mean_violin_boxplot_mappings_only_paired_novogene.svg", `temp_fig_5`)

nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean > 1 | `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean < -1, ])

nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean > 1 | `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean < -1, ])

nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean > 1 | `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean < -1, ])

species_effect_bins_novogene_paired <- `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean > 1 | `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.species.geometric.mean < -1, ]

season_effect_bins_novogene_paired <- `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean > 1 | `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$Log2.add.one.fold.change.season.geometric.mean < -1, ]

nrow(species_effect_bins_novogene_paired[species_effect_bins_novogene_paired$Bin.name %notin% season_effect_bins_novogene_paired$Bin.name,])
nrow(season_effect_bins_novogene_paired[season_effect_bins_novogene_paired$Bin.name %notin% species_effect_bins_novogene_paired$Bin.name,])

nrow(species_effect_bins_novogene_paired[species_effect_bins_novogene_paired$Log2.add.one.fold.change.species.geometric.mean > 1,])
nrow(species_effect_bins_novogene_paired[species_effect_bins_novogene_paired$Log2.add.one.fold.change.species.geometric.mean < -1,])

nrow(season_effect_bins_novogene_paired[season_effect_bins_novogene_paired$Log2.add.one.fold.change.season.geometric.mean > 1,])
season_effect_bins_novogene_paired[season_effect_bins_novogene_paired$Bin.name %in% species_effect_bins_novogene_paired$Bin.name,]

nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05, ])
nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05, ])

nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$AC.to.AR.BH.corrected.P.value < 0.05 | `2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`$June.to.November.BH.corrected.P.value < 0.05, ])

ggplot(data = `2022.07.18_strict_bins_significant_and_fold_change_by_category`, aes(x = "", y = Bins, fill = Category)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("") + xlab("") + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#848482", "#0e0013", "#c318fa")) + ggtitle("Number of significant bins") + guides(fill = guide_legend(title = "Category")) + labs(caption = "Meaningful change was log2()+1 change of >1 or <-1")
temp_fig_5 <- ggplot(data = `2022.07.18_strict_bins_significant_and_fold_change_by_category`, aes(x = "", y = Bins, fill = Category)) + geom_bar(width = 1, stat = "identity") + coord_polar("y", start=0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("") + xlab("") + scale_fill_manual(values=c("#1e90ff", "#ff8d1e", "#848482", "#0e0013", "#c318fa")) + ggtitle("Number of significant bins") + guides(fill = guide_legend(title = "Category")) + labs(caption = "Meaningful change was log2()+1 change of >1 or <-1")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-18_only_paired_novogene_significant_bins_and_fold_change_meaning_pie_chart_by_category.svg", `temp_fig_5`)

2022-07-19
ggplot(data = season_effect_bins_novogene_paired, aes(x = GTDB.TK.Family.or.species, y = "", fill = GTDB.TK.Family.or.species)) + geom_bar(width = 1, stat = "identity")+ theme(text = element_text(family = "Tex Gyre Heroes")) +

species_effect_bins_novogene_paired

write.table(`species_effect_bins_novogene_paired`, "2022-07-19_species_effect_bins_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID.tsv", sep="\t", row.names=FALSE)
write.table(`season_effect_bins_novogene_paired`, "2022-07-19_season_effect_bins_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID.tsv", sep="\t", row.names=FALSE)

`2022.07.19_bins_in_families_by_category_significant_and_foldchange_species_and_season_only_paired_novogene_lib_and_contig_adjusted` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-19_bins_in_families_by_category_significant_and_foldchange_species_and_season_only_paired_novogene_lib_and_contig_adjusted.txt")
View(`2022.07.19_bins_in_families_by_category_significant_and_foldchange_species_and_season_only_paired_novogene_lib_and_contig_adjusted`)

`2022.07.19_melted_bins_in_families_by_category_significant_and_foldchange_species_and_season_only_paired_novogene_lib_and_contig_adjusted` <- reshape2::melt(`2022.07.19_bins_in_families_by_category_significant_and_foldchange_species_and_season_only_paired_novogene_lib_and_contig_adjusted`, id=c("Family"))

ggplot(data = `2022.07.19_melted_bins_in_families_by_category_significant_and_foldchange_species_and_season_only_paired_novogene_lib_and_contig_adjusted`, x = Family, y = value, fill = variable) + ggdotchart(sorting = "descending")

ggdotchart(`2022.07.19_melted_bins_in_families_by_category_significant_and_foldchange_species_and_season_only_paired_novogene_lib_and_contig_adjusted`, x = "Family", y = "value", colour = "variable", sorting = "descending", add = "segments", rotate = TRUE, palette = c("#1e90ff", "#ff8d1e", #c318fa"))

"#4ffa18"

ggplot(data = `2022.07.19_melted_bins_in_families_by_category_significant_and_foldchange_species_and_season_only_paired_novogene_lib_and_contig_adjusted`, aes(x = Family, y = value, fill = variable)) + geom_bar(position = "dodge", stat = "identity") + coord_flip() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_fill_manual(values = c("#1e90ff", "#ff8d1e", "#c318fa")) + ylab("Bins") + labs(fill = "Category") + ggtitle("Number of BOSM in each family by category")
temp_fig_5 <- ggplot(data = `2022.07.19_melted_bins_in_families_by_category_significant_and_foldchange_species_and_season_only_paired_novogene_lib_and_contig_adjusted`, aes(x = Family, y = value, fill = variable)) + geom_bar(position = "dodge", stat = "identity") + coord_flip() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_fill_manual(values = c("#1e90ff", "#ff8d1e", "#c318fa")) + ylab("Bins") + labs(fill = "Category") + ggtitle("Number of BOSM in each family by category")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-19_bins_in_families_species_and_season_significant_and_fold_change_only_novogene_paired_contig_and_lib_adjusted_summed_to_bins_by_category.svg", `temp_fig_5`)

temp_df_ac_up <- `species_effect_bins_novogene_paired`[`species_effect_bins_novogene_paired`[5] < -1,][1]
write.table(`temp_df_ac_up`, "2022-07-19_list_bins_AC_increased_significant_and_fold_change_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID.tsv", sep="\t", row.names=FALSE)
temp_df_ar_up <- `species_effect_bins_novogene_paired`[`species_effect_bins_novogene_paired`[5] > 1,][1]
write.table(`temp_df_ar_up`, "2022-07-19_list_bins_AR_increased_significant_and_fold_change_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID.tsv", sep="\t", row.names=FALSE)
temp_df_nov_up <- season_effect_bins_novogene_paired[season_effect_bins_novogene_paired[9] > 1,][1]
write.table(`temp_df_nov_up`, "2022-07-19_list_bins_November_increased_significant_and_fold_change_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID.tsv", sep="\t", row.names=FALSE)

`2022.07.19_all_AC_significant_and_meaningful_bins_unique_potential_products_count`
`2022.07.19_all_AR_significant_and_meaningful_bins_unique_potential_products_count`

`2022.07.19_all_AC_significant_and_meaningful_bins_unique_potential_products_count`$Rmeans <- rowMeans(`2022.07.19_all_AC_significant_and_meaningful_bins_unique_potential_products_count`)
rowMeans(`2022.07.19_all_AC_significant_and_meaningful_bins_unique_potential_products_count`[,c(2:43)])
rowMeans(`2022.07.19_all_AC_significant_and_meaningful_bins_unique_potential_products_count`[2:43])
rowMeans(`2022.07.19_all_AC_significant_and_meaningful_bins_unique_potential_products_count`)

`2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`
rowSums(`2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[2:349])

`2022.07.19_all_samples_total_mapping_count` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-19_all_samples_total_mapping_count.txt")

ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = Mapping, x = "")) + geom_boxplot()

ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = ((Mapping / 7600000) * 100), x = "")) + geom_boxplot()

2022-07-21
`2022.07.19_melted_all_samples_total_mapping_count` <- reshape2::melt(`2022.07.19_all_samples_total_mapping_count`, id=c("Sample", "Species", "Season", "Sequencing", "Species_x_season"))

ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = ((Unique.mappings / 7600000) * 100), x = "", colour = Species_x_season, fill = Species_x_season)) + geom_boxplot(colour = "black", alpha = 0.6, position = position_dodge(1)) + geom_jitter(width = 0.05, size = 3) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Paired novogene samples") + ggtitle("Unique mappings only") + scale_color_manual(values = c("ACJ" = '#00f3ff', "ARJ" = '#ffb700', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00f3ff', "ARJ" = '#ffb700', "ACN" = '#0073ff', "ARN" = '#b36200'))
ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = ((Multiple.mappings / 7600000) * 100), x = "", colour = Species_x_season, fill = Species_x_season)) + geom_boxplot(colour = "black", alpha = 0.6, position = position_dodge(1)) + geom_jitter() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Paired novogene samples") + ggtitle("Multiple mappings only")

ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sequencing == "Pilot", ], aes(y = ((Unique.mappings / 4460000) * 100), x = "", colour = Species_x_season, fill = Species_x_season)) + geom_boxplot(colour = "black", alpha = 0.6, position = position_dodge(1)) + geom_jitter() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Paired pilot samples") + ggtitle("Unique mappings only")
ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sequencing == "Pilot", ], aes(y = ((Multiple.mappings / 4460000) * 100), x = "", colour = Species_x_season, fill = Species_x_season)) + geom_boxplot(colour = "black", alpha = 0.6, position = position_dodge(1)) + geom_jitter() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Paired pilot samples") + ggtitle("Multiple mappings only")

ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %notin% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins & `2022.07.19_all_samples_total_mapping_count`$Sequencing == "Novogene", ], aes(y = ((Unique.mappings / 7600000) * 100), x = "", colour = Species_x_season, fill = Species_x_season)) + geom_boxplot(colour = "black", alpha = 0.6, position = position_dodge(1)) + geom_jitter() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Novogene unpaired samples") + ggtitle("Unique mappings only")
ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %notin% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins & `2022.07.19_all_samples_total_mapping_count`$Sequencing == "Novogene", ], aes(y = ((Multiple.mappings / 7600000) * 100), x = "",colour = Species_x_season, fill = Species_x_season)) + geom_boxplot(colour = "black", alpha = 0.6, position = position_dodge(1)) + geom_jitter() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Novogene unpaired samples") + ggtitle("Multiple mappings only")

ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = ((Unique.mappings / 7600000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4") + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Paired novogene samples") + ggtitle("Unique mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200'))
ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = ((Multiple.mappings / 7600000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4") + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Paired novogene samples") + ggtitle("Multiple mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200'))

ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sequencing == "Novogene" & `2022.07.19_all_samples_total_mapping_count`$Sample %notin% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = ((Unique.mappings / 7600000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4") + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Novogene, unpaired samples") + ggtitle("Unique mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200'))
ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sequencing == "Novogene" & `2022.07.19_all_samples_total_mapping_count`$Sample %notin% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = ((Multiple.mappings / 7600000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4") + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Novogene, unpaired samples") + ggtitle("Multiple mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200'))

ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sequencing == "Pilot", ], aes(y = ((Unique.mappings / 4460000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4") + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Pilot paired samples") + ggtitle("Unique mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200'))
ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sequencing == "Pilot", ], aes(y = ((Multiple.mappings / 4460000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4") + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Pilot paired samples") + ggtitle("Multiple mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200'))

temp_fig_A <- ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = ((Unique.mappings / 7600000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4", width = 0.35) + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Paired novogene samples") + ggtitle("Unique mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + theme(legend.position = "none")
temp_fig_B <- ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sequencing == "Novogene" & `2022.07.19_all_samples_total_mapping_count`$Sample %notin% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = ((Unique.mappings / 7600000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4", width = 0.35) + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Novogene, unpaired samples") + ggtitle("Unique mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + theme(legend.position = "none") + theme(plot.title = element_blank())
temp_fig_C <- ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sequencing == "Pilot", ], aes(y = ((Unique.mappings / 4460000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4", width = 0.35) + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Pilot paired samples") + ggtitle("Unique mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + theme(plot.title = element_blank())
unique_mappings_legend <- get_legend(temp_fig_C)
temp_fig_C <- temp_fig_C + theme(legend.position="none")
grid.arrange(temp_fig_A, temp_fig_B, temp_fig_C, unique_mappings_legend, nrow = 2)
temp_fig_D <- grid.arrange(temp_fig_A, temp_fig_B, temp_fig_C, unique_mappings_legend, nrow = 2)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-21_all_samples_UNIQUE_mappings_percentage_of_input_reads_by_group.svg", `temp_fig_D`)

temp_fig_A <- ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = ((Multiple.mappings / 7600000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4", width = 0.35) + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Paired novogene samples") + ggtitle("Multiple mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + theme(legend.position = "none")
temp_fig_B <- ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sequencing == "Novogene" & `2022.07.19_all_samples_total_mapping_count`$Sample %notin% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ], aes(y = ((Multiple.mappings / 7600000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4", width = 0.35) + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Novogene, unpaired samples") + ggtitle("Multiple mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + theme(legend.position = "none") + theme(plot.title = element_blank())
temp_fig_C <- ggplot(data = `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sequencing == "Pilot", ], aes(y = ((Multiple.mappings / 4460000) * 100), x = Species_x_season, colour = Species_x_season, fill = Species_x_season)) + geom_violin(alpha = 0.45) + geom_boxplot(colour = "black", position = position_dodge(1), fill = "#f2f3f4", width = 0.35) + geom_jitter(width = 0.35, size = 1.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + ylab("Percent of total reads in file") + xlab("Pilot paired samples") + ggtitle("Multiple mappings only") + scale_color_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + scale_fill_manual(values = c("ACJ" = '#00aab3', "ARJ" = '#e6a500', "ACN" = '#0073ff', "ARN" = '#b36200')) + theme(plot.title = element_blank())
multiple_mappings_legend <- get_legend(temp_fig_C)
temp_fig_C <- temp_fig_C + theme(legend.position="none")
grid.arrange(temp_fig_A, temp_fig_B, temp_fig_C, multiple_mappings_legend, nrow = 2)
temp_fig_D <- grid.arrange(temp_fig_A, temp_fig_B, temp_fig_C, multiple_mappings_legend, nrow = 2)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-21_all_samples_MULTIPLE_mappings_percentage_of_input_reads_by_group.svg", `temp_fig_D`)

`2022_07_21_transposed_UNIQUE_mappings_only_all_samples_processed_read_files_mapping_strict_bins_summed_to_bins_contig_and_lib_size_adjusted_plus_details` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_21_transposed_UNIQUE_mappings_only_all_samples_processed_read_files_mapping_strict_bins_summed_to_bins_contig_and_lib_size_adjusted_plus_details.txt")
View(`2022_07_21_transposed_UNIQUE_mappings_only_all_samples_processed_read_files_mapping_strict_bins_summed_to_bins_contig_and_lib_size_adjusted_plus_details`)
`unique_mappings_transposed` <- `2022_07_21_transposed_UNIQUE_mappings_only_all_samples_processed_read_files_mapping_strict_bins_summed_to_bins_contig_and_lib_size_adjusted_plus_details`
`unique_mappings_melted_transposed` <- reshape2::melt(`unique_mappings_transposed`, id=c("SampleID", "Species", "Season", "Sequencing", "Species_x_season", "Bins"))
`unique_mappings_only_paired_novogene_melted_transposed` <- `unique_mappings_melted_transposed`[`unique_mappings_melted_transposed`$Bins %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ]

2022-07-22
`2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ]
`2022.07.22.mapping_counts_only_paired_novogene` <- `2022.07.19_all_samples_total_mapping_count`[`2022.07.19_all_samples_total_mapping_count`$Sample %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins, ]
`2022.07.22.mapping_counts_only_paired_novogene` <- cbind(`2022.07.22.mapping_counts_only_paired_novogene`, Percentage.unique.mapping.reads = ((`2022.07.22.mapping_counts_only_paired_novogene`$Unique.mappings / 7600000) * 100 ))

2022-07-28
ggplot(data = AC4J_Salmon_quantification_unsorted_all_reads_including_multiple_mapping, aes(x = TPM, y = EffectiveLength)) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_log10() + scale_y_log10()
ggplot(data = AC4J_Salmon_quantification_unsorted_all_reads_including_multiple_mapping[AC4J_Salmon_quantification_unsorted_all_reads_including_multiple_mapping$TPM > 0,], aes(x = TPM, y = EffectiveLength)) + geom_point(alpha = 0.45) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_log10() + scale_y_log10()

`2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_28-all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction.txt")
View(`2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`)
str(`2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`)
`transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction_NAMES` <- `2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Name
`transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction_COLS` <- colnames(`2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`)
`transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction` <- as.data.frame(t(`2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[,-1]))
colnames(`transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`) <- `transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction_NAMES`
`transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[, c(1:79380)] <- as.data.frame(sapply(`transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[, c(1:79380)], as.numeric))

write.table(`transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`, "2022-07-28_all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction.tsv", sep="\t", row.names=FALSE)
`melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction` <- reshape2::melt(`transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`, id=c("Sample_ID", "Species", "Season", "Sequencing", "Species_x_season"))
write.table(`melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`, "2022-07-28_melted_transposed_all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction.tsv", sep="\t", row.names=FALSE)

ggplot(data = `melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`, aes(y = value, x = Species_x_season)) + geom_boxplot() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_log10()
k141_9137563
ggplot(data = `melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_9137563",], aes(y = value, x = Species_x_season)) + geom_boxplot() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_y_log10()
species_x_season_comparisons <- list( c("ACJ", "ACN"), c("ACJ", "ARJ"), c("ARJ", "ARN"), c("ACN", "ARN"), c("ARN", "ACJ"))
ggboxplot(data = `melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_9137563",] , x = "Species_x_season", y = "value", colour = "Species_x_season", add = "jitter", shape = "Species_x_season") + stat_compare_means(label.y = 32)
ggpubr::ggboxplot(data = `melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_15965529",] , x = "Species_x_season", y = "value", colour = "Species_x_season", add = "jitter", shape = "Species_x_season", fill = "Species_x_season",) + stat_compare_means(comparisons = species_x_season_comparisons, method = "kruskal.test", label.y = c(32, 33, 34, 35, 36))

`melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Sample_ID %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins,]
`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction` <- `melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Sample_ID %in% `2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Bins,]
unique(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Sample_ID)

ggpubr::ggboxplot(data = `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_15965529",] , x = "Species_x_season", y = "value", colour = "Species_x_season", add = "jitter", shape = "Species_x_season", fill = "Species_x_season") + stat_compare_means(comparisons = species_x_season_comparisons, method = "kruskal.test", label.y = c(32, 33, 34, 35, 36))

ggpubr::ggviolin(data = `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_15965529",] , x = "Species_x_season", y = "value", fill = "Species_x_season", palette = c("#009eff", "#0024b3", "#ffcc00", "#e67e00"), add = "boxplot", add.params = list(fill = "#f2f3f4")) + stat_compare_means(comparisons = species_x_season_comparisons, method = "kruskal.test", label.y = c(32, 33, 34, 35, 36)) + theme(text = element_text(family = "Tex Gyre Heroes"))
ggpubr::ggviolin(data = `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_15965529",] , x = "Species_x_season", y = "value", fill = "Species_x_season", palette = c("#009eff", "#0024b3", "#ffcc00", "#e67e00"), add = "boxplot", add.params = list(fill = "#f2f3f4")) + stat_compare_means(comparisons = species_x_season_comparisons, label.y = c(32, 33, 34, 35, 36)) + theme(text = element_text(family = "Tex Gyre Heroes"))
ggpubr::ggviolin(data = `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_15965529",] , x = "Species_x_season", y = "value", fill = "Species_x_season", palette = c("#009eff", "#0024b3", "#ffcc00", "#e67e00"), add = "boxplot", add.params = list(fill = "#f2f3f4")) + stat_compare_means(comparisons = species_x_season_comparisons, method = "kruskal.test"(), label.y = c(32, 33, 34, 35, 36)) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggstatsplot::ggbetweenstats(data = `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_15965529",], x = Species_x_season, y = value, title = "Distribution of TPM for k141_15965529 across\nhost species and season", p.adjust.method = "BH")

PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_host_kruskal <- kruskal.test(value ~ Species, data = `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable == "PLACEHOLDER", ])
PLACEHOLDER_season_kruskal <- kruskal.test(value ~ Season, data = `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable == "PLACEHOLDER", ])
PLACEHOLDER_host_pvalue <- PLACEHOLDER_host_kruskal$p.value
PLACEHOLDER_season_pvalue <- PLACEHOLDER_season_kruskal$p.value
`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`[nrow(`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_host_pvalue)
`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`[, c(2)] <- sapply(`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`[, c(2)], as.numeric)
`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`[nrow(`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_season_pvalue)
`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`[, c(2)] <- sapply(`2022_07_18_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`[, c(2)], as.numeric)
rm(PLACEHOLDER_bin_name)
rm(PLACEHOLDER_host_kruskal)
rm(PLACEHOLDER_season_kruskal)
rm(PLACEHOLDER_host_pvalue)
rm(PLACEHOLDER_season_pvalue)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))

PLACEHOLDER_bin_name <- "PLACEHOLDER"
PLACEHOLDER_AC_geometric_means_mappings <- exp(mean(log(`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable), ]$value)))
PLACEHOLDER_AR_geometric_means_mappings <- exp(mean(log(`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable), ]$value)))
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`[`2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_host_species_only_paired_novogene`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_log2_add_one_fold_change_geometric_means_host <- log2(((PLACEHOLDER_AR_geometric_means_mappings - PLACEHOLDER_AC_geometric_means_mappings) / PLACEHOLDER_AC_geometric_means_mappings) + 1)
PLACEHOLDER_J_geometric_means_mappings <- exp(mean(log(`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable), ]$value)))
PLACEHOLDER_N_geometric_means_mappings <- exp(mean(log(`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable), ]$value)))
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`[`2022_07_18_sorted_Pvalues_lib_cont_adjusted_no_transformation_summed_to_bins_kruskal_test_P_values_sample_season_only_paired_novogene`$Bin.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_log2_add_one_fold_change_geometric_means_season <- log2(((PLACEHOLDER_N_geometric_means_mappings - PLACEHOLDER_J_geometric_means_mappings) / PLACEHOLDER_J_geometric_means_mappings) + 1)
PLACEHOLDER_taxonomy <- `2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`[`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`$Bin.name == "PLACEHOLDER", ]$Bin.long.taxa
PLACEHOLDER_short_taxonomy <- `2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`[`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`$Bin.name == "PLACEHOLDER", ]$Bin.family.or.genus
`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[nrow(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`)+1,] <- c(PLACEHOLDER_bin_name, PLACEHOLDER_AC_geometric_means_mappings, PLACEHOLDER_AR_geometric_means_mappings, PLACEHOLDER_BH_corrected_pvalue_host, PLACEHOLDER_log2_add_one_fold_change_geometric_means_host, PLACEHOLDER_J_geometric_means_mappings, PLACEHOLDER_N_geometric_means_mappings, PLACEHOLDER_BH_corrected_pvalue_season, PLACEHOLDER_log2_add_one_fold_change_geometric_means_season, PLACEHOLDER_taxonomy, PLACEHOLDER_short_taxonomy)
`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[, c(2:9)] <- sapply(`2022-07-18_only_paired_novogene_summed_bins_library_and_contig_adjusted_geometric_means_all_values_different_things_and_taxID`[, c(2:9)], as.numeric)
rm(list=ls(pattern='Bin_m*'))
rm(list=ls(pattern='Bin_c*'))

testing_AC_geometric_mean <- exp(mean(log(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_2495" & `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Species == "AC",]$value + 1)))
testing_AR_geometric_mean <- exp(mean(log(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_2495" & `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Species == "AR",]$value + 1)))

PLACEHOLDER_host_kruskal <- kruskal.test(value ~ Species, data = `2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[`2022.07.18_melted_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`$variable == "PLACEHOLDER", ])

kruskal.test(value ~ Species_x_season, data = `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_15965529",])
k141_2495
exp(mean(log(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable), ]$value)))
max(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$value)
sample(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`, 5)
113.767225
`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$value == 113.767225,]

write.table(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`, "2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction.tsv", sep="\t", row.names=FALSE)

`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`


`2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction` <- read.delim("2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction.tsv")
2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction.tsv

kruskal.test(value ~ Species, data = `2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction`[`2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction`$variable == "k141_2495",])


`2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction` <- read.delim("2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction.tsv")
`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction` <- `2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction`
library(stats)
library(methods)
source("2022_07_29_strict_bin_contigs_salmon_tpms_GC_and_seq_bias_flags_on_subsampled_reads_only_paired_novogene_kruskal_test_EXECUTION_rscript.R", local = FALSE, encoding = 'UTF-8')

write.table(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`, "2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues.tsv", sep="\t", row.names=FALSE)
write.table(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`, "2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues.tsv", sep="\t", row.names=FALSE)

`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues` <- data.frame(matrix(ncol = 2, nrow = 0, dimnames = list(NULL, c("Contig name", "Uncorrected pvalue"))))
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues` <- data.frame(matrix(ncol = 2, nrow = 0, dimnames = list(NULL, c("Contig name", "Uncorrected pvalue"))))

`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`
PLACEHOLDER_contig_name <- "PLACEHOLDER"
PLACEHOLDER_host_kruskal <- kruskal.test(value ~ Species, data = `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "PLACEHOLDER", ])
PLACEHOLDER_season_kruskal <- kruskal.test(value ~ Season, data = `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == "PLACEHOLDER", ])
PLACEHOLDER_host_pvalue <- PLACEHOLDER_host_kruskal$p.value
PLACEHOLDER_season_pvalue <- PLACEHOLDER_season_kruskal$p.value
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`[nrow(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`)+1,] <- c(PLACEHOLDER_contig_name, PLACEHOLDER_host_pvalue)
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`[, c(2)] <- sapply(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`[, c(2)], as.numeric)
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`[nrow(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`)+1,] <- c(PLACEHOLDER_contig_name, PLACEHOLDER_season_pvalue)
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`[, c(2)] <- sapply(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`[, c(2)], as.numeric)
rm(PLACEHOLDER_contig_name)
rm(PLACEHOLDER_host_kruskal)
rm(PLACEHOLDER_season_kruskal)
rm(PLACEHOLDER_host_pvalue)
rm(PLACEHOLDER_season_pvalue)
rm(list=ls(pattern='k141_*'))

`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues` <- `2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`[order(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`$Uncorrected.pvalue),]
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues` <- `2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`[order(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`$Uncorrected.pvalue),]
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`$BH_corrected_pvalues <- p.adjust(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`$Uncorrected.pvalue, method = "BH")
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`$BH_corrected_pvalues <- p.adjust(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`$Uncorrected.pvalue, method = "BH")

`2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things` <- data.frame(matrix(ncol = 11, nrow = 0, dimnames = list(NULL, c("Bin.name", "AC.geometric.mean.mapping.count", "AR.geometric.mean.mapping.count", "AC.to.AR.BH.corrected.P.value", "Log2.add.one.fold.change.species.geometric.mean", "June.geometric.mean.mapping.count", "November.geometric.mean.mapping.count", "June.to.November.BH.corrected.P.value", "Log2.add.one.fold.change.season.geometric.mean"))))

PLACEHOLDER_contig_name <- "PLACEHOLDER"
PLACEHOLDER_AC_geometric_means_mappings <- exp(mean(log(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Species == "AC" & grepl(glob2rx("PLACEHOLDER"), `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable), ]$value)))
PLACEHOLDER_AR_geometric_means_mappings <- exp(mean(log(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Species == "AR" & grepl(glob2rx("PLACEHOLDER"), `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable), ]$value)))
PLACEHOLDER_BH_corrected_pvalue_host <- `2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues_SORTED`[`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues_SORTED`$contig.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_log2_add_one_fold_change_geometric_means_host <- log2(((PLACEHOLDER_AR_geometric_means_mappings - PLACEHOLDER_AC_geometric_means_mappings) / PLACEHOLDER_AC_geometric_means_mappings) + 1)
PLACEHOLDER_J_geometric_means_mappings <- exp(mean(log(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Season == "J" & grepl(glob2rx("PLACEHOLDER"), `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable), ]$value)))
PLACEHOLDER_N_geometric_means_mappings <- exp(mean(log(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Season == "N" & grepl(glob2rx("PLACEHOLDER"), `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable), ]$value)))
PLACEHOLDER_BH_corrected_pvalue_season <- `2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues_SORTED`[`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues_SORTED`$contig.name == "PLACEHOLDER", ]$BH_corrected_pvalues
PLACEHOLDER_log2_add_one_fold_change_geometric_means_season <- log2(((PLACEHOLDER_N_geometric_means_mappings - PLACEHOLDER_J_geometric_means_mappings) / PLACEHOLDER_J_geometric_means_mappings) + 1)
`2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things`[nrow(`2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things`)+1,] <- c(PLACEHOLDER_contig_name, PLACEHOLDER_AC_geometric_means_mappings, PLACEHOLDER_AR_geometric_means_mappings, PLACEHOLDER_BH_corrected_pvalue_host, PLACEHOLDER_log2_add_one_fold_change_geometric_means_host, PLACEHOLDER_J_geometric_means_mappings, PLACEHOLDER_N_geometric_means_mappings, PLACEHOLDER_BH_corrected_pvalue_season, PLACEHOLDER_log2_add_one_fold_change_geometric_means_season)
`2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things`[, c(2:9)] <- sapply(`2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things`[, c(2:9)], as.numeric)
rm(list=ls(pattern='k141_*'))

2022-07-31
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues` <- read.delim("/Volumes/hpc-home/2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues.tsv")
View(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`)
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues` <- read.delim("/Volumes/hpc-home/2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues.tsv")
View(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`)

`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues_SORTED` <- `2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`[order(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues`$Uncorrected.pvalue),]
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues_SORTED` <- `2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`[order(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues`$Uncorrected.pvalue),]

`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues_SORTED`$BH_corrected_pvalues <- p.adjust(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues_SORTED`$Uncorrected.pvalue, method = "BH")
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues_SORTED`$BH_corrected_pvalues <- p.adjust(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues_SORTED`$Uncorrected.pvalue, method = "BH")

write.table(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues_SORTED`, "2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues_SORTED.tsv", sep="\t", row.names=FALSE)
write.table(`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues_SORTED`, "2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues_SORTED.tsv", sep="\t", row.names=FALSE)

`2022.06.23_Transposed_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-23_Transposed_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages.txt")
View(`2022.06.23_Transposed_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages`)
`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages` <- reshape2::melt(`2022.06.23_Transposed_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages`, id=c("Taxon", "Species", "Season", "Sequencing", "Species_x_season", "Sample_ID"))

`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages` <- `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$Sample_ID %in% `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Sample_ID,]

row_num(unique(`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$Sample_ID))
`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`

`2022.06.23_0_list_of_just_phyla_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages` <- read.table("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-23_0_list_of_just_phyla_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages.txt", quote="\"", comment.char="")
View(`2022.06.23_0_list_of_just_phyla_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages`)

phyla_list_kaiju <- as.list(unique(`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable))
for(phylum in phyla_list_kaiju) {
  phylum_name <- paste(phylum)
  phylum_AC_kaiju_percent <- mean(`testing_AC_kaiju_phyla_novogene_paired_df`[`testing_AC_kaiju_phyla_novogene_paired_df`$variable == phylum_name,]$value)
  print(paste(phylum_name, phylum_AC_kaiju_percent))
  rm(phylum_name)
  rm(phylum_AC_kaiju_percent)
  }

`2022.08.01_Kaiju_phyla_paired_novogene_only_percentages_arithmetic_means` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Phylum.name", "Arithmetic.mean.AC", "Arithmetic.mean.AR", "Arithmetic.mean.J", "Arithmetic.mean.N"))))
for(phylum in phyla_list_kaiju) {
  phylum_name <- paste(phylum)
  phylum_AC_kaiju_percent <- mean(`testing_AC_kaiju_phyla_novogene_paired_df`[`testing_AC_kaiju_phyla_novogene_paired_df`$variable == phylum_name,]$value)
  phylum_AR_kaiju_percent <- mean(`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$Species == "AR" & `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == phylum_name,]$value)
  phylum_J_kaiju_percent <- mean(`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$Season == "J" & `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == phylum_name,]$value)
  phylum_N_kaiju_percent <- mean(`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$Season == "N" & `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == phylum_name,]$value)
  print(paste(phylum_name, phylum_AC_kaiju_percent, phylum_AR_kaiju_percent, phylum_J_kaiju_percent, phylum_N_kaiju_percent))
  `2022.08.01_Kaiju_phyla_paired_novogene_only_percentages_arithmetic_means`[nrow(`2022.08.01_Kaiju_phyla_paired_novogene_only_percentages_arithmetic_means`)+1,] <- c(phylum_name, phylum_AC_kaiju_percent, phylum_AR_kaiju_percent, phylum_J_kaiju_percent, phylum_N_kaiju_percent)
  `2022.08.01_Kaiju_phyla_paired_novogene_only_percentages_arithmetic_means`[, c(2:5)] <- sapply(`2022.08.01_Kaiju_phyla_paired_novogene_only_percentages_arithmetic_means`[, c(2:5)], as.numeric)
  rm(phylum_name)
  rm(phylum_AC_kaiju_percent)
  rm(phylum_AR_kaiju_percent)
  rm(phylum_J_kaiju_percent)
  rm(phylum_N_kaiju_percent)
}

2022-08-01

library(stats)
library(methods)
`2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction` <- read.delim("2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction.tsv")
`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction` <- `2022-07-29_melted_transposed_only_paired_novogene_Salmon_TPM_gc_and_seq_bias_correction`
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues_SORTED` <- read.delim("2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues_SORTED.tsv")
`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues_SORTED` <- read.delim("2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues_SORTED.tsv")
`2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things` <- data.frame(matrix(ncol = 9, nrow = 0, dimnames = list(NULL, c("Contig.name", "AC.geometric.TPM.count", "AR.geometric.TPM.count", "AC.to.AR.BH.corrected.P.value", "Log2.add.one.fold.change.species.geometric.mean", "June.geometric.TPM.count", "November.geometric.TPM.count", "June.to.November.BH.corrected.P.value", "Log2.add.one.fold.change.season.geometric.mean"))))
unique_contigs_list <- as.list(unique(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable))

for (contig in unique_contigs_list) {
  contig_name <- paste(contig)
  contig_name_AC_geometric_TPMs <- exp(mean(log(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Species == "AC" & `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == contig_name, ]$value)))
  contig_name_AR_geometric_TPMs <- exp(mean(log(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Species == "AR" & `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == contig_name, ]$value)))
  contig_name_BH_corrected_pvalue_host <- `2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues_SORTED`[`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_HOST_kruskal_test_pvalues_SORTED`$Contig.name == contig_name, ]$BH_corrected_pvalues
  contig_name_log2_add_one_fold_change_geometric_means_host <- log2(((contig_name_AR_geometric_TPMs - contig_name_AC_geometric_TPMs) / contig_name_AC_geometric_TPMs) + 1)
  contig_name_J_geometric_TPMs <- exp(mean(log(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Season == "J" & `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == contig_name, ]$value)))
  contig_name_N_geometric_TPMs <- exp(mean(log(`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`[`only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$Season == "N" & `only_paired_novogene_melted_transposed_2022_07_28.all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction`$variable == contig_name, ]$value)))
  contig_name_BH_corrected_pvalue_season <- `2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues_SORTED`[`2022_07_29_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bin_contigs_subsampled_reads_only_novogene_paired_SEASON_kruskal_test_pvalues_SORTED`$Contig.name == contig_name, ]$BH_corrected_pvalues
  contig_name_log2_add_one_fold_change_geometric_means_season <- log2(((contig_name_N_geometric_TPMs - contig_name_J_geometric_TPMs) / contig_name_J_geometric_TPMs) + 1)
  `2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things`[nrow(`2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things`)+1,] <- c(contig_name, contig_name_AC_geometric_TPMs, contig_name_AR_geometric_TPMs, contig_name_BH_corrected_pvalue_host, contig_name_log2_add_one_fold_change_geometric_means_host, contig_name_J_geometric_TPMs, contig_name_N_geometric_TPMs, contig_name_BH_corrected_pvalue_season, contig_name_log2_add_one_fold_change_geometric_means_season)
  `2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things`[, c(2:9)] <- sapply(`2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things`[, c(2:9)], as.numeric)
  rm(list=ls(pattern='k141_*'))
  rm(list=ls(pattern='contig_name*'))
}

write.table(`2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things`, "2022-07-30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv", sep="\t", row.names=FALSE)
Transposed_2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/Transposed_2022-06-23_0_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages.txt", header=TRUE)
View(Transposed_2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages)
melted_subsampled_all_files_kaiju_genera <- reshape2::melt(Transposed_2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages, id=c("Tax_ID", "Season", "Sequencing", "Species_x_season", "Sample_ID"))
ggplot(data = melted_subsampled_all_files_kaiju_genera[melted_subsampled_all_files_kaiju_genera$variable == "X1578" & melted_subsampled_all_files_kaiju_genera$Species_x_season == "ACJ",], aes(x = Sample_ID, y = value)) + geom_bar(stat = "identity")

`only_novogene_paired_melted_subsampled_all_files_kaiju_genera` <- `melted_subsampled_all_files_kaiju_genera`[`melted_subsampled_all_files_kaiju_genera`$Sample_ID %in% `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$Sample_ID,]
`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$value <- as.numeric(as.character(`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$value))
ggplot(data = `only_novogene_paired_melted_subsampled_all_files_kaiju_genera`[`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$variable == "X1578",], aes(x = Species_x_season, y = value, fill = Species_x_season)) + geom_boxplot()

`genus_list_only_novogene_paired_melted_subsampled_all_files_kaiju_genera` <- as.list(unique(`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$variable))

for(genus in genus_list_only_novogene_paired_melted_subsampled_all_files_kaiju_genera){
  genus_name <- paste(genus)
  ACJ_geometric_mean <- exp(mean(log(`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`[`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$Species_x_season == "ACJ" & `only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$variable == genus_name,]$value)))
  ACN_geometric_mean <- exp(mean(log(`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`[`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$Species_x_season == "ACN" & `only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$variable == genus_name,]$value)))
  ARJ_geometric_mean <- exp(mean(log(`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`[`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$Species_x_season == "ARJ" & `only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$variable == genus_name,]$value)))
  ARN_geometric_mean <- exp(mean(log(`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`[`only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$Species_x_season == "ARN" & `only_novogene_paired_melted_subsampled_all_files_kaiju_genera`$variable == genus_name,]$value)))
  print(paste(genus_name, ACJ_geometric_mean, ACN_geometric_mean, ARJ_geometric_mean, ARN_geometric_mean))
  rm(genus_name)
  rm(ACJ_geometric_mean)
  rm(ACN_geometric_mean)
  rm(ARJ_geometric_mean)
  rm(ARN_geometric_mean)
}

`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`
ggplot(data = `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Firmicutes",], aes(x = Species_x_season, y = value, fill = Species_x_season)) + geom_boxplot()
ggplot(data = `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Firmicutes" | `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Bacteroidetes" | `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Proteobacteria" | `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Actinobacteria" | `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Candidatus.Saccharibacteria",], aes(x = Species_x_season, y = value, fill = variable)) + geom_boxplot()

ggplot(data = `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Firmicutes" | `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Bacteroidetes" | `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Actinobacteria",], aes(x = Species_x_season, y = value, fill = variable)) + geom_boxplot()

ggplot(data = `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Firmicutes" | `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Bacteroidetes" | `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Actinobacteria",], aes(x = Species_x_season, y = value, fill = variable)) + geom_boxplot(colour = "black", width = 0.65) +  scale_color_manual(values = c("Bacteroidetes" = '#1e90ff', "Firmicutes" = '#ff8d1e', "Actinobacteria" = '#ff1e90')) + scale_fill_manual(values = c("Bacteroidetes" = '#1e90ff', "Firmicutes" = '#ff8d1e', "Actinobacteria" = '#ff1e90')) + xlab("Host species and sample month") + ylab("Percent reads in sample (%)") + guides(fill = guide_legend(title = "Phyla")) + ggtitle("Percent reads belonging to 3 most commonly detected Phyla from all\nsamples assessed by Kaiju") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()
temp_fig_A <- ggplot(data = `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Firmicutes" | `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Bacteroidetes" | `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Actinobacteria",], aes(x = Species_x_season, y = value, fill = variable)) + geom_boxplot(colour = "black", width = 0.65) +  scale_color_manual(values = c("Bacteroidetes" = '#1e90ff', "Firmicutes" = '#ff8d1e', "Actinobacteria" = '#ff1e90')) + scale_fill_manual(values = c("Bacteroidetes" = '#1e90ff', "Firmicutes" = '#ff8d1e', "Actinobacteria" = '#ff1e90')) + xlab("Host species and sample month") + ylab("Percent reads in sample (%)") + guides(fill = guide_legend(title = "Phyla")) + ggtitle("Percent reads belonging to 3 most commonly detected Phyla from all\nsamples assessed by Kaiju") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-08-01_paired_novogene_Kaiju_percentages_bacteroidetes_firmicutes_actinobacteria_subsampled_reads.svg", `temp_fig_A`)

Transposed_2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_family_level_results_all_potential_taxID_percentages <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/Transposed_2022-06-23_0_allowed_errors_combined_results_all_processed_subsampled_files_family_level_results_all_potential_taxID_percentages.txt")
View(Transposed_2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_family_level_results_all_potential_taxID_percentages)
`melted_Transposed_2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_family_level_results_all_potential_taxID_percentages` <- reshape2::melt(Transposed_2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_family_level_results_all_potential_taxID_percentages, id=c("Tax_ID", "Season", "Sequencing", "Species_x_season", "Sample_ID"))
`only_paired_novogene_melted_Transposed_2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_family_level_results_all_potential_taxID_percentages` <- `melted_Transposed_2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_family_level_results_all_potential_taxID_percentages`[`melted_Transposed_2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_family_level_results_all_potential_taxID_percentages`$Sample_ID %in% `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$Sample_ID,]

short_kaiju_fam_novogene_pair <- `only_paired_novogene_melted_Transposed_2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_family_level_results_all_potential_taxID_percentages`
`short_kaiju_fam_novogene_pair`$value <- as.numeric(as.character(`short_kaiju_fam_novogene_pair`$value))
ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Species_x_season, y = value, fill = Species_x_season)) + geom_boxplot()
ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_boxplot() + facet_wrap(vars(Species_x_season))
ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(stat = "identity") + facet_wrap(vars(Species_x_season))
ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(stat = "identity") + facet_wrap(vars(Species_x_season)) + coord_polar()
ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(width = 1, stat = "identity") + coord_polar()
ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(width = 1, stat = "identity") + coord_polar()
ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Species_x_season, y = mean(value), fill = Species_x_season)) + geom_bar(width = 1, stat = "identity", colour = "black") + coord_polar() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x = element_blank())
ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Species_x_season, y = mean(value), fill = Species_x_season)) + geom_bar(width = 1, stat = "identity", colour = "black") + geom_jitter(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(y = value, shape = Species_x_season)) + coord_polar() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x = element_blank())
ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(width = 2, stat = "identity", colour = "black") + ylim(-50,50) + facet_wrap(vars(Species_x_season)) + coord_polar() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x = element_blank())

ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(width = 1, stat = "identity", colour = "black") + coord_polar() + facet_wrap(vars(Species_x_season)) + theme(axis.text.x = element_blank())
ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(width = 2, stat = "identity", colour = "black") + coord_polar() + facet_wrap(vars(Species_x_season)) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x = element_blank()) + theme(legend.position = "none")
ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(width = 2, stat = "identity", colour = "black") + ylim(-100, 120) + coord_polar() + facet_wrap(vars(Species_x_season)) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x = element_blank()) + theme(legend.position = "none")

ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(width = 1, stat = "identity", colour = "black") + ylim(0, 50) + facet_wrap(vars(Season), scales = "free") + scale_fill_manual(values = c("ACJ" = '#00f3ff', "ARJ" = '#ffb700', "ACN" = '#0073ff', "ARN" = '#b36200')) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x = element_blank()) + ylab("Percentage reads classified by Kaiju in Lactobacillaceae (%)") + xlab("Samples") + theme(axis.ticks.x = element_blank()) + theme(legend.position = "none")
temp_fig_A <- ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(width = 1, stat = "identity", colour = "black") + ylim(0, 50) + facet_wrap(vars(Species_x_season), scales = "free") + scale_fill_manual(values = c("ACJ" = '#00f3ff', "ARJ" = '#ffb700', "ACN" = '#0073ff', "ARN" = '#b36200')) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x = element_blank()) + ylab("Percentage reads classified by Kaiju in Lactobacillaceae (%)") + xlab("Samples") + theme(axis.ticks.x = element_blank()) + theme(legend.position = "none")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-08-01_paired_novogene_Kaiju_percentages_Lactobacillaceae_subsampled_reads.svg", `temp_fig_A`)

`2022.04.10_CheckM_output_all_bins` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-10_CheckM_output_all_bins.txt")
View(`2022.04.10_CheckM_output_all_bins`)

ggplot(data = `2022.04.10_CheckM_output_all_bins`, aes(x = Completeness, y = Contamination, size = Strain.heterogeneity)) + geom_point()

ggplot(data = `2022.04.10_CheckM_output_all_bins`[`2022.04.10_CheckM_output_all_bins`$Completeness > 80 & `2022.04.10_CheckM_output_all_bins`$Contamination < 5,], aes(group = Completeness, y = Contamination)) + geom_boxplot() + facet_wrap(vars(Marker.lineage))
ggplot(data = `2022.04.10_CheckM_output_all_bins`[`2022.04.10_CheckM_output_all_bins`$Completeness > 80 & `2022.04.10_CheckM_output_all_bins`$Contamination < 5,], aes(x = Marker.lineage, y = Completeness)) + geom_boxplot()

ggplot(data = `2022.04.10_CheckM_output_all_bins`[`2022.04.10_CheckM_output_all_bins`$Completeness > 80 & `2022.04.10_CheckM_output_all_bins`$Contamination < 5 & `2022.04.10_CheckM_output_all_bins`$Marker.lineage %>% count(`2022.04.10_CheckM_output_all_bins`$Marker.lineage) %>% filter(n >= 5),], aes(x = Marker.lineage, y = Completeness)) + geom_boxplot()

ggplot(data = `2022.04.10_CheckM_output_all_bins`, aes(x = "", y = Completeness)) + geom_violin(fill = "#ff8d1e") + geom_boxplot(fill = "#1e90ff", alpha = 0.55, outlier.shape = NA, stat = "boxplot") + geom_jitter(width = 0.4, alpha = 0.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Bins") + ylab("Completeness (%)")

ggplot(data = `2022.04.10_CheckM_output_all_bins`, aes(x = "", y = Contamination)) + geom_violin(fill = "#1e90ff") + geom_boxplot(fill = "#ff8d1e", alpha = 0.75, outlier.shape = NA, stat = "boxplot") + geom_jitter(width = 0.4, alpha = 0.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Bins") + ylab("Contamination (%)")

temp_fig_A <- ggplot(data = `2022.04.10_CheckM_output_all_bins`, aes(x = "", y = Completeness)) + geom_violin(fill = "#ff8d1e") + geom_boxplot(fill = "#1e90ff", alpha = 0.55, outlier.shape = NA, stat = "boxplot") + geom_jitter(width = 0.4, alpha = 0.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Bins") + ylab("Completeness (%)") + theme(axis.title.x = element_blank(), axis.text.x = element_blank())
temp_fig_B <- ggplot(data = `2022.04.10_CheckM_output_all_bins`, aes(x = "", y = Contamination)) + geom_violin(fill = "#1e90ff") + geom_boxplot(fill = "#ff8d1e", alpha = 0.75, outlier.shape = NA, stat = "boxplot") + geom_jitter(width = 0.4, alpha = 0.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Bins") + ylab("Contamination (%)")

plot_grid(temp_fig_A, temp_fig_B, align = "v", nrow = 2)
temp_fig_C <- plot_grid(temp_fig_A, temp_fig_B, align = "v", nrow = 2)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-08-02_all_bins_checkm_output_two_plots_contamination_and_completeness.svg", `temp_fig_C`)

`2022.08.02_google_scholar_results_searching_arid_microbiome_by_one_year_period` <- read.delim("~/Google Drive/PhD/2022-08-02_google_scholar_results_searching_arid_microbiome_by_one_year_period.txt")
View(`2022.08.02_google_scholar_results_searching_arid_microbiome_by_one_year_period`)

ggplot(data = `2022.08.02_google_scholar_results_searching_arid_microbiome_by_one_year_period`, aes(x = Years, y = Arid.microbiome.hits)) + geom_line(group = 1) + geom_point(fill = "#ff8d1e", colour = "#ff8d1e", size = 3, shape = 15) + scale_y_log10() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) + xlab("Publication year") + ylab("Google scholar results for 'Arid microbiome'")
temp_fig_A <- ggplot(data = `2022.08.02_google_scholar_results_searching_arid_microbiome_by_one_year_period`, aes(x = Years, y = Arid.microbiome.hits)) + geom_line(group = 1) + geom_point(fill = "#ff8d1e", colour = "#ff8d1e", size = 3, shape = 15) + scale_y_log10() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1)) + xlab("Publication year") + ylab("Google scholar results for 'Arid microbiome'")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-08-02_google_scholar_results_for_arid_microbiome_each_year_2000_to_2021.svg", `temp_fig_A`)

`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-07-14_all_strict_bins_GTDB-TK_full_taxonomy_and_ACTUAL_family_or_genus.tsv")
View(`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`)
unique(`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`$Bin.family.or.genus)
unique(`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`$Bin.long.taxa)

`2022.08.02_count_strict_bins_by_genus` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-08-02_count_strict_bins_by_genus.txt")
View(`2022.08.02_count_strict_bins_by_genus`)

ggplot(data = `2022.08.02_count_strict_bins_by_genus`[`2022.08.02_count_strict_bins_by_genus`$Bins > 5,], aes(x = Genus, y = Bins, fill = Genus)) + geom_bar(stat = "identity")


`2022.08.02_count_strict_bins_by_genus_data_table` = as.data.table(`2022.08.02_count_strict_bins_by_genus`)
`2022.08.02_count_strict_bins_by_genus_data_table`[order(-Bins), if (.BY[[1]]) .SD else list("Other", sum(Bins)), by = 1:nrow(`2022.08.02_count_strict_bins_by_genus_data_table`) <= 7][, !"nrow", with = F]
cbbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
temp_bins_genus_top_7_and_other_strict_bins_count <- `2022.08.02_count_strict_bins_by_genus_data_table`[order(-Bins), if (.BY[[1]]) .SD else list("Other", sum(Bins)), by = 1:nrow(`2022.08.02_count_strict_bins_by_genus_data_table`) <= 7][, !"nrow", with = F]

ggplot(data = temp_bins_genus_top_7_and_other_strict_bins_count, aes(x = Genus, y = Bins, fill = Genus)) + geom_bar(stat = "identity") + scale_fill_manual(values=cbbPalette) + theme(text = element_text(family = "Tex Gyre Heroes")) + guides(fill = guide_legend(title = "Genera")) + theme_linedraw() + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + theme(text = element_text(size = 25)) + theme(legend.position = "left")
temp_fig_A <- ggplot(data = temp_bins_genus_top_7_and_other_strict_bins_count, aes(x = Genus, y = Bins, fill = Genus)) + geom_bar(stat = "identity") + scale_fill_manual(values=cbbPalette) + theme(text = element_text(family = "Tex Gyre Heroes")) + guides(fill = guide_legend(title = "Genera")) + theme_linedraw() + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + theme(text = element_text(size = 25))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-08-02_strict_bins_GTDB-TK_genus_top_7_and_other_bins_per_genus.svg", `temp_fig_A`)

2022-08-03
`2022_08_03_Transposed_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_08_03_Transposed_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins.txt")
View(`2022_08_03_Transposed_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins`)

good_tpms <- `2022_08_03_Transposed_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins`
melted_good_tpms <- reshape2::melt(`good_tpms`, id=c("Name", "Species", "Season", "Sequencing", "Species_x_season", "Sample_ID"))

only_novogene_paired_melted_good_tpms <- `melted_good_tpms`[`melted_good_tpms`$Sample_ID %in% `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$Sample_ID,]
strict_bins_list <- as.list(unique(only_novogene_paired_melted_good_tpms$variable))

ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(width = 1, stat = "identity", colour = "black") + ylim(0, 50) + facet_wrap(vars(Season), scales = "free") + scale_fill_manual(values = c("ACJ" = '#00f3ff', "ARJ" = '#ffb700', "ACN" = '#0073ff', "ARN" = '#b36200')) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x = element_blank()) + ylab("Percentage reads classified by Kaiju in Lactobacillaceae (%)") + xlab("Samples") + theme(axis.ticks.x = element_blank()) + guides(fill = guide_legend(title = "Host species and\nsample season")) + theme(text = element_text(size = 15)) + theme(legend.position = "left")
temp_fig_A <- ggplot(data = `short_kaiju_fam_novogene_pair`[`short_kaiju_fam_novogene_pair`$variable == "X33958",], aes(x = Sample_ID, y = value, fill = Species_x_season)) + geom_bar(width = 1, stat = "identity", colour = "black") + ylim(0, 50) + facet_wrap(vars(Season), scales = "free") + scale_fill_manual(values = c("ACJ" = '#00f3ff', "ARJ" = '#ffb700', "ACN" = '#0073ff', "ARN" = '#b36200')) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x = element_blank()) + ylab("Percentage reads classified by Kaiju in Lactobacillaceae (%)") + xlab("Samples") + theme(axis.ticks.x = element_blank()) + guides(fill = guide_legend(title = "Host species and\nsample season")) + theme(text = element_text(size = 15))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-08-01_paired_novogene_Kaiju_percentages_Lactobacillaceae_subsampled_reads.svg", `temp_fig_A`)


`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs` <- data.frame(matrix(ncol = 7, nrow = 0, dimnames = list(NULL, c("Bin", "AC.geometric.mean", "AR.geometric.mean", "June.geometric.mean", "November.geometric.mean", "Host.raw.pvalue", "Season.raw.pvalue"))))

for (bin in strict_bins_list){
  bin_name <- paste(bin)
  AC.geometric.mean <- exp(mean(log(`only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name & `only_novogene_paired_melted_good_tpms`$Species == "AC",]$value)))
  AR.geometric.mean <- exp(mean(log(`only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name & `only_novogene_paired_melted_good_tpms`$Species == "AR",]$value)))
  June.geometric.mean <- exp(mean(log(`only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name & `only_novogene_paired_melted_good_tpms`$Season == "J",]$value)))
  November.geometric.mean <- exp(mean(log(`only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name & `only_novogene_paired_melted_good_tpms`$Season == "N",]$value)))
  host_kruskal_raw_pvalue <- kruskal.test(value ~ Species, data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name,])$p.value
  month_kruskal_raw_pvalue <- kruskal.test(value ~ Season, data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name,])$p.value
  `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs`[nrow(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs`)+1,] <- c(bin_name, AC.geometric.mean, AR.geometric.mean, June.geometric.mean, November.geometric.mean, host_kruskal_raw_pvalue, month_kruskal_raw_pvalue)
  `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs`[, c(2:7)] <- sapply(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs`[, c(2:7)], as.numeric)
  rm(bin_name)
  rm(AC.geometric.mean)
  rm(AR.geometric.mean)
  rm(June.geometric.mean)
  rm(November.geometric.mean)
  rm(host_kruskal_raw_pvalue)
  rm(month_kruskal_raw_pvalue)
}

kruskal.test(value ~ Species, data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == "Bin_c1000",])

novogene_paired_salmon_TPMs_strict_bin_single_contigs_host_pvalues <- `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs`[order(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs`$Host.raw.pvalue),]
novogene_paired_salmon_TPMs_strict_bin_single_contigs_month_pvalues <- `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs`[order(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs`$Season.raw.pvalue),]
novogene_paired_salmon_TPMs_strict_bin_single_contigs_host_pvalues$BH.corrected.host.pvalues <- p.adjust(novogene_paired_salmon_TPMs_strict_bin_single_contigs_host_pvalues$Host.raw.pvalue, method = "BH")
novogene_paired_salmon_TPMs_strict_bin_single_contigs_month_pvalues$BH.corrected.season.pvalues <- p.adjust(novogene_paired_salmon_TPMs_strict_bin_single_contigs_host_pvalues$Season.raw.pvalue, method = "BH")

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change` <- data.frame(matrix(ncol = 12, nrow = 0, dimnames = list(NULL, c("Bin", "AC.geometric.mean", "AR.geometric.mean", "June.geometric.mean", "November.geometric.mean", "Host.raw.pvalue", "Season.raw.pvalue", "Host.corrected.pvalue", "Season.corrected.pvalue", "AC.to.AR.log2fold.add.one.change", "June.to.November.log2fold.add.one.change", "GTDB.TK.taxonomy"))))
for (bin in strict_bins_list){
  bin_name <- paste(bin)
  AC.geometric.mean <- exp(mean(log(`only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name & `only_novogene_paired_melted_good_tpms`$Species == "AC",]$value)))
  AR.geometric.mean <- exp(mean(log(`only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name & `only_novogene_paired_melted_good_tpms`$Species == "AR",]$value)))
  June.geometric.mean <- exp(mean(log(`only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name & `only_novogene_paired_melted_good_tpms`$Season == "J",]$value)))
  November.geometric.mean <- exp(mean(log(`only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name & `only_novogene_paired_melted_good_tpms`$Season == "N",]$value)))
  host_kruskal_raw_pvalue <- kruskal.test(value ~ Species, data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name,])$p.value
  month_kruskal_raw_pvalue <- kruskal.test(value ~ Season, data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name,])$p.value
  host_corrected_pvalue <- novogene_paired_salmon_TPMs_strict_bin_single_contigs_host_pvalues[novogene_paired_salmon_TPMs_strict_bin_single_contigs_host_pvalues$Bin == bin_name,]$BH.corrected.host.pvalues
  month_corrected_pvalue <- novogene_paired_salmon_TPMs_strict_bin_single_contigs_month_pvalues[novogene_paired_salmon_TPMs_strict_bin_single_contigs_month_pvalues$Bin == bin_name,]$BH.corrected.season.pvalues
  log2_add_one_host_fold_change <- log2(((AR.geometric.mean - AC.geometric.mean) / AC.geometric.mean) + 1)
  log2_add_one_month_fold_change <- log2(((November.geometric.mean - June.geometric.mean) / June.geometric.mean) + 1)
  GTDB_TK_taxonomy <- `2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`[`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`$Bin.name == bin_name, ]$Bin.long.taxa
  `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[nrow(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`)+1,] <- c(bin_name, AC.geometric.mean, AR.geometric.mean, June.geometric.mean, November.geometric.mean, host_kruskal_raw_pvalue, month_kruskal_raw_pvalue, host_corrected_pvalue, month_corrected_pvalue, log2_add_one_host_fold_change, log2_add_one_month_fold_change, GTDB_TK_taxonomy)
  `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[, c(2:11)] <- sapply(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[, c(2:11)], as.numeric)
  rm(bin_name)
  rm(AC.geometric.mean)
  rm(AR.geometric.mean)
  rm(June.geometric.mean)
  rm(November.geometric.mean)
  rm(host_kruskal_raw_pvalue)
  rm(month_kruskal_raw_pvalue)
  rm(host_corrected_pvalue)
  rm(month_corrected_pvalue)
  rm(log2_add_one_host_fold_change)
  rm(log2_add_one_month_fold_change)
  rm(GTDB_TK_taxonomy)
}

view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`)
write.table(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, "2022_08_03_Salmon_TPMs_GC_and_seq_bias_flags_on_strict_bins_single_combined_contig_per_bin_subsampled_reads_only_novogene_paired_all_numbers.tsv", sep="\t", row.names=FALSE)

ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")
ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = June.to.November.log2fold.add.one.change, y = -log10(Season.corrected.pvalue))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Season effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")

temp_fig_A <- ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + theme(text = element_text(size = 15))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-08-03_paired_novogene_strict_bins_single_combined_contig_per_bin_geometric_mean_fold_change_corrected_pvalues_HOST.svg", `temp_fig_A`)


`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$June.to.November.log2fold.add.one.change > 1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Season.corrected.pvalue < 0.05 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$June.to.November.log2fold.add.one.change < -1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Season.corrected.pvalue < 0.05,]

nrow(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$June.to.November.log2fold.add.one.change > 1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Season.corrected.pvalue < 0.05 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$June.to.November.log2fold.add.one.change < -1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Season.corrected.pvalue < 0.05,])

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05,]

nrow(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05,])

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 3 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -3 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05,]

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`
is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Bin,])

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Bin),]
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.geometric.mean),]
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AR.geometric.mean),]
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$June.geometric.mean),]
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$November.geometric.mean),]
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.raw.pvalue),]
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Season.raw.pvalue),]
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue),]
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Season.corrected.pvalue),]
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change),]
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$June.to.November.log2fold.add.one.change),]
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[is.na(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy),]

nrow(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$June.to.November.log2fold.add.one.change > 0.5 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Season.corrected.pvalue < 0.05 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$June.to.November.log2fold.add.one.change < -0.5 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Season.corrected.pvalue < 0.05,])

nrow(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 0.5 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -0.5 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05,])

Bin_334
ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + theme(text = element_text(size = 15))


ggplot(data = `only_novogene_paired_melted_good_tpms`[grepl(glob2rx("Bin_m334"), `only_novogene_paired_melted_good_tpms`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_density_ridges2(jittered_points = TRUE, alpha = 0.75, point_size = 1.75, quantile_lines = TRUE, quantile_fun=function(value,...)mean(value)) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Host species")

ggplot(data = `only_novogene_paired_melted_good_tpms`[grepl(glob2rx("Bin_m334"), `only_novogene_paired_melted_good_tpms`$variable), ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700", "black")) + scale_colour_manual(values = c ("#1ec8ff", "#006ad1", "#ff551e", "#d16700", "black")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("Summed mapped segment counts for all contigs in Bin, library and contig size(s) adjusted") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.65)

ggplot(data = `only_novogene_paired_melted_good_tpms`[grepl(glob2rx("Bin_c1687"), `only_novogene_paired_melted_good_tpms`$variable) | grepl(glob2rx("Bin_c892"), `only_novogene_paired_melted_good_tpms`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))

ggplot(data = `only_novogene_paired_melted_good_tpms`[grepl(glob2rx("Bin_c1687"), `only_novogene_paired_melted_good_tpms`$variable) | grepl(glob2rx("Bin_c892"), `only_novogene_paired_melted_good_tpms`$variable) | grepl(glob2rx("Bin_c1750"), `only_novogene_paired_melted_good_tpms`$variable) | grepl(glob2rx("Bin_c1500"), `only_novogene_paired_melted_good_tpms`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))


temp_fig_A <- ggplot(data = `only_novogene_paired_melted_good_tpms`[grepl(glob2rx("Bin_c1687"), `only_novogene_paired_melted_good_tpms`$variable) | grepl(glob2rx("Bin_c892"), `only_novogene_paired_melted_good_tpms`$variable) | grepl(glob2rx("Bin_c1750"), `only_novogene_paired_melted_good_tpms`$variable) | grepl(glob2rx("Bin_c1500"), `only_novogene_paired_melted_good_tpms`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-08-03_paired_novogene_strict_bins_single_combined_contig_per_bin_2_most_AC_and_2_most_AR_bins_tpm_violin_and_boxplots_HOST.svg", `temp_fig_A`)

ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 3,]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))
ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -3,]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))

2022-08-04
nrow(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -1,])

nrow(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1,])

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1,]$GTDB.TK.taxonomy
unique(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1,]$GTDB.TK.taxonomy)
`list_AR_up_gtdb-tk_full_taxa_strict_bins_novogene_paired_UNIQUE` <- as.list(unique(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1,]$GTDB.TK.taxonomy))

`list_AC_up_gtdb-tk_full_taxa_strict_bins_novogene_paired_UNIQUE` <- as.list(unique(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -1,]$GTDB.TK.taxonomy))

`list_AR_up_gtdb-tk_full_taxa_strict_bins_novogene_paired_UNIQUE`
lapply(`list_AC_up_gtdb-tk_full_taxa_strict_bins_novogene_paired_UNIQUE`, write, "AC_up_TPM_bins_novogene_paired_strict_bins_single_contig_per_bin_geometric_mean_changes.txt", append = TRUE, ncolumns=1)
lapply(`list_AR_up_gtdb-tk_full_taxa_strict_bins_novogene_paired_UNIQUE`, write, "AR_up_TPM_bins_novogene_paired_strict_bins_single_contig_per_bin_geometric_mean_changes.txt", append = TRUE, ncolumns=1)

for (taxon_name in `list_AC_up_gtdb-tk_full_taxa_strict_bins_novogene_paired_UNIQUE`){
  taxon <- paste(taxon_name)
  AC_up_bins <- nrow(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy == taxon,])
  AR_up_bins <- nrow(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy == taxon,])
  print(paste(taxon, AC_up_bins, AR_up_bins))
  rm(taxon)
  rm(AC_up_bins)
  rm(AR_up_bins)
}


2022-08-05
ggplot(data = temp_bins_genus_top_7_and_other_strict_bins_count, aes(x = Genus, y = Bins, fill = Genus)) + geom_bar(stat = "identity") + scale_fill_manual(values=cbbPalette) + theme(text = element_text(family = "Tex Gyre Heroes")) + guides(fill = guide_legend(title = "Genera")) + theme_linedraw() + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + theme(text = element_text(size = 25)) + theme(legend.position = "left")

2022-08-08
temp_melted_df_salmon_novogene_paired_counts.tsv

AC.geometric.mean <- exp(mean(log(`only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable == bin_name & `only_novogene_paired_melted_good_tpms`$Species == "AC",]$value)))

ggplot(data = `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`, aes(x = Log2.add.one.fold.change.species.geometric.mean, y = -log10(AC.to.AR.BH.corrected.P.value))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Contig effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")

exp(mean(log(temp_melted_df_salmon_novogene_paired_counts[temp_melted_df_salmon_novogene_paired_counts$variable == "k141_13085355" & temp_melted_df_salmon_novogene_paired_counts$Species == "AC",]$value)))

2022-08-09
only_novogene_paired_melted_good_tpms
dcast(only_novogene_paired_melted_good_tpms, Sample_ID ~ variable)
temp_for_edgeR <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_07_28-all_subsampled_files_Salmon_TPM_gc_and_seq_bias_correction.txt")
view(temp_for_edgeR)

`2022.08.09_only_paired_novogene_Salmon_bias_and_gc_adjusted_tpms_per_contig` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-08-09_only_paired_novogene_Salmon_bias_and_gc_adjusted_tpms_per_contig.txt")
View(`2022.08.09_only_paired_novogene_Salmon_bias_and_gc_adjusted_tpms_per_contig`)

`2022.08.09_Just_sampleID_and_group_or_factors_per_sample_only_paired_novogene_Salmon_bias_and_gc_adjusted_tpms_per_contig` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-08-09_Just_sampleID_and_group_or_factors_per_sample_only_paired_novogene_Salmon_bias_and_gc_adjusted_tpms_per_contig.txt")
View(`2022.08.09_Just_sampleID_and_group_or_factors_per_sample_only_paired_novogene_Salmon_bias_and_gc_adjusted_tpms_per_contig`)

countsTable <- `2022.08.09_only_paired_novogene_Salmon_bias_and_gc_adjusted_tpms_per_contig`
targets <- `2022.08.09_Just_sampleID_and_group_or_factors_per_sample_only_paired_novogene_Salmon_bias_and_gc_adjusted_tpms_per_contig`

group <- factor(paste(targets$))

2022-08-11
ggplot(data = `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`, aes(x = Log2.add.one.fold.change.species.geometric.mean, y = -log10(AC.to.AR.BH.corrected.P.value))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Contig effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")

`2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY.No_infinites` <- `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`[is.finite(rowSums(`2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`)),]

AC_only_contigs <- `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`[`2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`$AC.geometric.TPM.count > 0 & `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`$AR.geometric.TPM.count == 0,]
AR_only_contigs <- `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`[`2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`$AR.geometric.TPM.count > 0 & `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`$AC.geometric.TPM.count == 0,]
AC_enriched_contigs <- `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`[`2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`$AR.geometric.TPM.count > 0 & `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`$Log2.add.one.fold.change.species.geometric.mean < -1,]
AR_enriched_contigs <- `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`[`2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`$AC.geometric.TPM.count > 0 & `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`$AC.to.AR.BH.corrected.P.value < 0.05 & `2022.07.30_Salmon_TPMs_GC_and_seq_bias_flagged_only_paired_novogene_contigs_geometric_means_all_values_different_things.tsv.COPY`$Log2.add.one.fold.change.species.geometric.mean > 1,]

ggplot(data = `AR_enriched_contigs`, aes(x = Log2.add.one.fold.change.species.geometric.mean, y = -log10(AC.to.AR.BH.corrected.P.value))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Contig effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")

write.table(`AC_only_contigs`, "2022_08_11_Salmon_TPMs_GC_and_seq_bias_flags_on_paired_novogene_only_individual_contigs_AC_only_contigs.tsv", sep="\t", row.names=FALSE)
write.table(`AR_only_contigs`, "2022_08_11_Salmon_TPMs_GC_and_seq_bias_flags_on_paired_novogene_only_individual_contigs_AR_only_contigs.tsv", sep="\t", row.names=FALSE)
write.table(`AC_enriched_contigs`, "2022_08_11_Salmon_TPMs_GC_and_seq_bias_flags_on_paired_novogene_only_individual_contigs_AC_enriched_but_not_exclusive_contigs_corrected_pvalue_significant_and_log2_add_one_fold_change_over_pos_minus_one.tsv", sep="\t", row.names=FALSE)
write.table(`AR_enriched_contigs`, "2022_08_11_Salmon_TPMs_GC_and_seq_bias_flags_on_paired_novogene_only_individual_contigs_AR_enriched_but_not_exclusive_contigs_corrected_pvalue_significant_and_log2_add_one_fold_change_over_pos_minus_one.tsv", sep="\t", row.names=FALSE)

ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2_add_1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed")

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1,]

AR_enriched_or_exclusive_bins <- `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1,]
AC_enriched_or_exclusive_bins <- `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -1,]
view(AC_enriched_or_exclusive_bins)

write.table(`AC_enriched_or_exclusive_bins`, "2022_08_12_Salmon_TPMs_GC_and_seq_bias_flags_on_paired_novogene_only_concatenated_single_line_fasta_bins_AC_enriched_bins.tsv", sep="\t", row.names=FALSE)

2022-09-02
`2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file.txt")
View(`2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`)

`melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file` <- reshape2::melt(`2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`, id=c("Taxid", "Species", "Season", "Sequencing", "Species_x_season", "Sample_ID"))

`melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`

ggplot(data = `melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0', ], aes(x = Species_x_season, y = value, fill = Species_x_season)) + geom_violin() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700"))

ggplot(data = `melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0', ], aes(x = Species_x_season, y = value, fill = Species_x_season, colour = Species_x_season, shape = Species_x_season)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Host species and sampling month") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Host species and sampling month"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

ggplot(data = `melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0', ], aes(x = Species_x_season, y = value, fill = Species_x_season, colour = Species_x_season)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Host species and sampling month") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Host species and sampling month"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

temp_fig_A <- ggplot(data = `melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0', ], aes(x = Species_x_season, y = value, fill = Species_x_season, colour = Species_x_season)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + facet_wrap(~ Sequencing) + ylab("Percent reads classified as taxID (%)") + xlab("Host species and sampling month") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Host species and sampling month"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-04_all_samples_subsampled_depths_Kraken_percent_classifications_all_actual_taxids_by_host_x_season_log_scale_010_conf_score.svg", `temp_fig_A`)

ggplot(data = `melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0', ], aes(x = Sample_ID, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Sample ID") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + facet_grid(~ Species_x_season) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

ggplot(data = `melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0' & `melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$value > 1, ], aes(x = variable, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("TaxIDs classified with >1% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 9))

`2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file.txt")
`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file` <- reshape2::melt(`2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`, id=c("Taxid", "Species", "Season", "Sequencing", "Species_x_season", "Sample_ID"))
ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0', ], aes(x = Species_x_season, y = value, fill = Species_x_season, colour = Species_x_season)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Host species and sampling month") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Host species and sampling month"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + facet_wrap(~ Sequencing) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()
temp_fig_A <- ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0', ], aes(x = Species_x_season, y = value, fill = Species_x_season, colour = Species_x_season)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Host species and sampling month") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Host species and sampling month"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + facet_wrap(~ Sequencing) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-04_all_samples_subsampled_depths_Kraken_percent_classifications_all_actual_taxids_by_host_x_season_log_scale_050_conf_score.svg", `temp_fig_A`)

ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0' & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$value > 1, ], aes(x = variable, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("TaxIDs classified with >1% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 9))

ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0' & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$value > 1, ], aes(x = variable, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("TaxIDs classified with >1% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + theme(text = element_text(family = "Tex Gyre Heroes")) + facet_wrap(~ Sequencing) + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 9))

`2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file.txt")
`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file` <- reshape2::melt(`2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`, id=c("Taxon", "Species", "Season", "Sequencing", "Species_x_season", "Sample_ID"))

ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$variable != 'unclassified' & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 2.5, ], aes(x = variable, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Taxa classified by Kraken2 with >2.5% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + theme(text = element_text(family = "Tex Gyre Heroes")) + facet_wrap(~ Sequencing) + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 9))

ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$variable != 'unclassified' & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value >= 2.5, ], aes(x = variable, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Taxa classified by Kraken2 with >2.5% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + theme(text = element_text(family = "Tex Gyre Heroes")) + ylim(2.5,10.5) + facet_wrap(~ Sequencing) + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 7.5))

temp_fig_A <- ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$variable != 'unclassified' & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 2.5, ], aes(x = variable, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Taxa classified by Kraken2 with >2.5% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + theme(text = element_text(family = "Tex Gyre Heroes")) + ylim(2.5,10.5) + facet_wrap(~ Sequencing) + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 7.5))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-04_all_samples_subsampled_depths_Kraken_percent_classifications_all_actual_taxa_by_host_x_season_050_conf_score_percent_reads_over_2_point_5_at_least_one_sample.svg", `temp_fig_A`)

`2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages.txt")
View(`2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages`)

`melted_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages` <- reshape2::melt(`2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages`, id=c("Tax_ID", "Species", "Season", "Sequencing", "Species_x_season", "Sample_ID"))

ggplot(data = `melted_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages`, aes(x = Species_x_season, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Host species and sample month") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + facet_wrap(~ Sequencing) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

temp_fig_A <- ggplot(data = `melted_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_phylum_level_results_all_potential_taxons_percentages`, aes(x = Species_x_season, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Host species and sample month") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + facet_wrap(~ Sequencing) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-04_all_samples_subsampled_depths_Kaiju_percent_classifications_all_phyla_by_host_x_season_0_allowed_errors.svg", `temp_fig_A`)

`2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-23_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxID_percentages.tsv")

`actually_transposed_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages` <- t(`2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages`)
`actually_transposed_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages` %>% row_to_names(row_number = 1)
`colnames_actually_transposed_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages` <- `actually_transposed_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages` %>% row_to_names(row_number = 1)

`colnames_actually_transposed_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages`[, c(1:38553)] <- sapply(`colnames_actually_transposed_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages`[, c(1:38553)], as.numeric)
`colnames_actually_transposed_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages`[, c(1:38553)] <- as.data.frame(sapply(`colnames_actually_transposed_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages`[, c(1:38553)], as.numeric))

`working_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages` <- `2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages`
`transposed_working_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages` <- t(`working_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages`)

`transposed_working_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages` <- `transposed_working_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages` %>% row_to_names(row_number = 1)

`converted_transposed_working_2022.06.23_TRANSPOSED_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxons_percentages`[,c (1:38553)]

save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-04.RData")

2022-09-07
setwd("~/Google Drive/PhD/2022-Paper_2-Chapter_3")
`2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxa_percentages` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-06-23_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxa_percentages.txt")
View(`2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxa_percentages`)

`working_kaiju_taxa` <- `2022.06.23_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxa_percentages`
`transposed_working_kaiju_taxa` <- t(`working_kaiju_taxa`)
`colnames_transposed_working_kaiju_taxa` <- `transposed_working_kaiju_taxa` %>% row_to_names(row_number = 1)
colrange <- c(1:38553)
`tempor` <- `colnames_transposed_working_kaiju_taxa`
colnames <- names(`tempor`[1:38553])
`tempor`[colnames] <- lapply(`tempor`[colnames], as.numeric)

2022-09-08
`tempor` <- `transposed_working_kaiju_taxa`
colnames(`tempor`) <- `tempor`[1,]
`tempor` <- `tempor`[-1,]
`tempor`[, c(1:38553)] <- sapply(`tempor`[, c(1:38553)], function(x) as.numeric(as.character(x)))
`tempor` <- read.csv("~/Google Drive/PhD/2022-Paper_2-Chapter_3/V2_2022-06-23_0_allowed_errors_combined_results_all_processed_subsampled_files_species_level_results_all_potential_taxa_percentages.csv", stringsAsFactors=FALSE, strip.white = TRUE)

`tempor1` <- read.csv("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-08_just_numbers_Kaiju_0_errors_all_taxa_possible_species_level_report_percentages_all_subsampled_reads.csv", stringsAsFactors=FALSE, strip.white = TRUE)
ggplot(data = `tempor1`, aes(x = AR30J, y = AR30N)) + geom_point()

`tempor2` <- read.csv("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-08_just_strings_Kaiju_0_errors_all_taxa_possible_species_level_report_percentages_all_subsampled_reads.csv", stringsAsFactors=FALSE, strip.white = TRUE)

`tempor1_names` <- `tempor1`$Taxon
`tp_tempor1` <- as.data.frame(t(`tempor1`[,-1]))
colnames(`tp_tempor1`) <- `tempor1_names`
`tempor2_names` <- `tempor2`$Taxon
`tp_tempor2` <- as.data.frame(t(`tempor2`[,-1]))
colnames(`tp_tempor2`) <- `tempor2_names`


write.table(`tp_tempor1`, "2022-09-08_Kaiju_0_errors_species_level_all_possible_species_all_subsampled_read_files.tsv", sep="\t", row.names=FALSE)
write.table(`tp_tempor1`, "V2_2022-09-08_Kaiju_0_errors_species_level_all_possible_species_all_subsampled_read_files.tsv", sep="\t", row.names=TRUE)
write.table(`tp_tempor2`, "2022-09-08_Kaiju_0_errors_species_level_all_possible_species_all_subsampled_read_files-STRINGS_about_metadata.tsv", sep="\t", row.names=FALSE)

write.table(`AC_enriched_or_exclusive_bins`, "2022_08_12_Salmon_TPMs_GC_and_seq_bias_flags_on_paired_novogene_only_concatenated_single_line_fasta_bins_AC_enriched_bins.tsv", sep="\t", row.names=FALSE)

`tempor_3` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-08_all_subsampled_reads_Kaiju_0_errors_all_possible_taxa_report_percentages_species_level_concatenated.tsv", stringsAsFactors=FALSE, strip.white = TRUE)
`tempor_3` <- read.delim("~/2022-09-08_all_subsampled_reads_Kaiju_0_errors_all_possible_taxa_report_percentages_species_level_concatenated.tsv", stringsAsFactors=FALSE, strip.white = TRUE)

`copy_tempor_3` <- `tempor_3`

`melted_copy_tempor_3` <- reshape2::melt(`copy_tempor_3`, id=c("Sample_ID", "Season", "Sequencing", "Species_x_season", "Sample_ID.1"))

save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-08.RData")

write.table(`melted_copy_tempor_3`, "2022_09_08_melted_CONTAINS_NA_VALUES_Kaiju_0_errors_species_level_report_percentages_all_possible_species_all_subsampled_reads.tsv", sep="\t", row.names=FALSE)

ggplot(data = `melted_copy_tempor_3`[`melted_copy_tempor_3`$variable != FALSE & `melted_copy_tempor_3`$value >= 85, ], aes(x = variable, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Taxa classified by Kaiju with >85% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + theme(text = element_text(family = "Tex Gyre Heroes")) + facet_wrap(~ Sequencing) + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 7.5))

is.na(`melted_copy_tempor_3`)
which(is.na(`melted_copy_tempor_3`))

2022-09-09
checkm_bin_plot1 <- ggplot(data = `2022.04.10_CheckM_output_all_bins`, aes(x = Completeness)) + geom_histogram(binwidth = 5, fill = "#f2f3f4", color = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Completeness (%)") + ylab("Frequency") + geom_vline(xintercept=c(80), col="#1e90ff", linetype = "dashed", size = 1)

checkm_bin_plot2 <- ggplot(data = `2022.04.10_CheckM_output_all_bins`, aes(x = Contamination)) + geom_histogram(binwidth = 5, fill = "#848482", color = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Contamination (%)") + ylab("Frequency") + geom_vline(xintercept=c(5), col="#ff8d1e", linetype = "dashed", size = 1)

checkm_bin_plot3 <- ggplot(data = `2022.04.10_CheckM_output_all_bins`, aes(x = Completeness, y = Contamination, colour = ifelse(Completeness > 80 & Contamination < 5, 'black', 'red'))) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Completeness (%)") + ylab("Contamination (%)") + geom_vline(xintercept=c(80), col="#1e90ff", linetype = "dashed", size = 1.35) + geom_hline(yintercept=c(5), col="#ff8d1e", linetype = "dashed", size = 1.35) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba"))

ggplot(data = `2022.04.10_CheckM_output_all_bins`[`2022.04.10_CheckM_output_all_bins`$Contamination < 5 & `2022.04.10_CheckM_output_all_bins`$Completeness > 80, ], aes(x = Completeness, y = Contamination, colour = Marker.lineage)) + geom_jitter()


checkm_bin_plot1 + checkm_bin_plot2 + checkm_bin_plot3 + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') + plot_annotation(title = 'CheckM analysis of 503 bin fasta files')

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-04-13_all_bins_CheckM_output_completeness_contamination_frequency_and_completeness_to_contamination_points_with_strain_heterogeneity_colour.svg")

strict_bins <- `2022.04.10_CheckM_output_all_bins`[`2022.04.10_CheckM_output_all_bins`$Contamination < 5 & `2022.04.10_CheckM_output_all_bins`$Completeness > 80, ]$Bin.id
write.table(`strict_bins`, "2022_09_09_list_strict_bins.tsv", sep="\t", row.names=FALSE)

count(`2022.04.10_CheckM_output_all_bins`[`2022.04.10_CheckM_output_all_bins`$Contamination < 5 & `2022.04.10_CheckM_output_all_bins`$Completeness > 80, ]$Bin.id)
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-09.RData")

`2022.09.09_Barrnap_strict_bins_rRNA_detection_results_bins_per_detection_type_not_mutually_exclusive_detected_length_at_least_10_percent_expected_length` <- read.csv("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-09_Barrnap_strict_bins_rRNA_detection_results_bins_per_detection_type_not_mutually_exclusive_detected_length_at_least_10_percent_expected_length.csv")
View(`2022.09.09_Barrnap_strict_bins_rRNA_detection_results_bins_per_detection_type_not_mutually_exclusive_detected_length_at_least_10_percent_expected_length`)

barrnap_plot <- ggplot(data = `2022.09.09_Barrnap_strict_bins_rRNA_detection_results_bins_per_detection_type_not_mutually_exclusive_detected_length_at_least_10_percent_expected_length`, aes(x = Factor, y = Number.of.bins, fill = Factor)) + geom_bar(stat = 'identity') + scale_fill_manual(values = c("#0f0f0f", "#1e90ff", "#ff8d1e", "#525252", "#6cb7ff", "#ffb46c")) + guides(fill=guide_legend(title="rRNA detection type")) + ylab("Number of bins") + theme_linedraw() + theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank(), text = element_text(family = "Tex Gyre Heroes"), legend.text = element_text(size = 8))

checkm_bin_plot1 + checkm_bin_plot2 + checkm_bin_plot3 + barrnap_plot + plot_layout(guides = "keep") + plot_annotation(tag_levels = 'A')

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-09_CheckM_figs_A-C_Barrnap_fig_D_all_bins_CheckM_strict_bins_Barrnap_QC_figure_combined.svg")
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-09.RData")

2022-09-11
The Kaiju table is broken after melting, contains NAs and still has the value
column as a character column not numeric. Checking whether the table prior to
melting was broken as well

names(which(colSums(is.na(`copy_tempor_3`))>0))
[1] "Streptococcus.infantarius"    "Rhodococcus.pyridinivorans"   "Glaciozyma.antarctica"        "Haloferax.alexandrinus"       "Chromohalobacter.israelensis"
 [6] "Fulvimonas.soli"              "Crepidotus.variabilis"        "Knoellia.subterranea"         "Prototheca.stagnorum"         "X.Curtobacterium..plantarum"
[11] "Xenorhabdus.doucetiae"

For meeting with Wilfried will just replace those na's with 0 for the time being
but have set a script running again to try and make sure to get all the percentages
or zero if they aren't in the sample's kaiju report

`copy_tempor_4` <- `copy_tempor_3`
`copy_tempor_4`[is.na(`copy_tempor_4`)] <- 0

`melted_copy_tempor_4` <- reshape2::melt(`copy_tempor_4`, id=c("Sample_ID", "Season", "Sequencing", "Species", "Species_x_season", "Sample_ID.1"))

`melted_copy_tempor_4`$value <- as.numeric(as.character(`melted_copy_tempor_4`$value))

`melted_copy_tempor_4`[is.na(`melted_copy_tempor_4`$value), ]

ggplot(data = `melted_copy_tempor_4`[`melted_copy_tempor_4`$value >= 2.5, ], aes(x = variable, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Taxa classified by Kaiju with at least 2.5% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + theme(text = element_text(family = "Tex Gyre Heroes")) + facet_wrap(~ Sequencing) + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 7.5), legend.text = element_text(size = 8))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-11_DEMO_ONLY_DO_NOT_USE_Kaiju_0_errors_allowed_all_subsampled_reads_all_possible_taxa_min_2_point_5_percent_at_least_one_sample.svg")
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-11.RData")

2022-09-12
sum(`melted_copy_tempor_4`[`melted_copy_tempor_4`$Species == "AC",]$value > 0)
nrow(`melted_copy_tempor_4`[`melted_copy_tempor_4`$value > 0, ])

sampleID_list <- as.list(unique(`melted_copy_tempor_4`$Sample_ID))

`kaiju_species_0_errors_all_possible_species_all_subsampled_reads_number_species_report_percent_over_0` <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Sample_ID", "Species_over_0", "Host", "Month", "Host_x_month", "Sequencing"))))

for (sample_ID in sampleID_list){
  sample_ID <- paste(sample_ID)
  Species_over_0 <- nrow(`melted_copy_tempor_4`[`melted_copy_tempor_4`$Sample_ID == sample_ID & `melted_copy_tempor_4`$value > 0, ])
  Host <- unique(`melted_copy_tempor_4`[`melted_copy_tempor_4`$Sample_ID == sample_ID,]$Species)
  Month <- unique(`melted_copy_tempor_4`[`melted_copy_tempor_4`$Sample_ID == sample_ID,]$Season)
  Host_x_month <- unique(`melted_copy_tempor_4`[`melted_copy_tempor_4`$Sample_ID == sample_ID,]$Species_x_season)
  Sequencing <- unique(`melted_copy_tempor_4`[`melted_copy_tempor_4`$Sample_ID == sample_ID,]$Sequencing)
  `kaiju_species_0_errors_all_possible_species_all_subsampled_reads_number_species_report_percent_over_0`[nrow(`kaiju_species_0_errors_all_possible_species_all_subsampled_reads_number_species_report_percent_over_0`)+1,] <- c(sample_ID, Species_over_0, Host, Month, Host_x_month, Sequencing)
  `kaiju_species_0_errors_all_possible_species_all_subsampled_reads_number_species_report_percent_over_0`[, c(2)] <- sapply(`kaiju_species_0_errors_all_possible_species_all_subsampled_reads_number_species_report_percent_over_0`[, c(2)], as.numeric)
  rm(sample_ID)
  rm(Species_over_0)
  rm(Host)
  rm(Month)
  rm(Host_x_month)
  rm(Sequencing)
}

save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-12.RData")

ggboxplot(`kaiju_species_0_errors_all_possible_species_all_subsampled_reads_number_species_report_percent_over_0`, x = "Host_x_month", y = "Species_over_0", shape = "Sequencing", color = "Host_x_month", add = "jitter") + stat_compare_means(method = "anova", label.y = 1000) + stat_compare_means(label = "p.signif", method = "t.test", ref.group = ".all.") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("Number of species detected") + xlab("Host species and collection month") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + facet_wrap(~ Sequencing)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-12_DEMO_ONLY_DO_NOT_USE_Kaiju_0_errors_allowed_all_subsampled_reads_number_species_percent_reads_above_zero_per_sample.svg")

ggplot(data = temp_bins_genus_top_7_and_other_strict_bins_count, aes(x = Genus, y = Bins, fill = Genus)) + geom_bar(stat = "identity", colour = 'black') + scale_fill_manual(values=cbbPalette) + theme(text = element_text(family = "Tex Gyre Heroes")) + guides(fill = guide_legend(title = "Genera")) + theme_linedraw() + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + theme(legend.position = "left")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-12_GTDB-TK_strict_bins_classifications_genus_level_bar_charts_7_most_abundant_and_others.svg")

#/hpc-home/osbourne/good_salmon is the place where the bin based Salmon work was
#done and can be found

`2022_08_03_2nd_copy_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_08_03_2nd_copy_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins.tsv")
View(`2022_08_03_2nd_copy_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins`)

2022-09-13
`2022_08_03_TRANSPOSED_Meta_data_added_2nd_copy_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins.tsv` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_08_03_TRANSPOSED_Meta_data_added_2nd_copy_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins.tsv.txt")
view(`2022_08_03_TRANSPOSED_Meta_data_added_2nd_copy_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins.tsv`)
`2022-09-13_working_bin_level_salmon_bias_flagged` <- `2022_08_03_TRANSPOSED_Meta_data_added_2nd_copy_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins.tsv`
view(`2022-09-13_working_bin_level_salmon_bias_flagged`)

`2022-09-13_melted_working_bin_level_salmon_bias_flagged` <- reshape2::melt(`2022-09-13_working_bin_level_salmon_bias_flagged`, id=c("Name", "Species", "Season", "Sequencing", "Species_x_season", "Sample_ID"))
`2022-09-13_PCA_working_bin_level_salmon_bias_flagged` <- prcomp(`2022-09-13_working_bin_level_salmon_bias_flagged`[,c(2:349)], center = TRUE, scale. = FALSE)

`2022-09-13_var_explained_PCA_working_bin_level_salmon_bias_flagged_dataframe` <- data.frame(PC = paste0("PC", 1:81), var_explained = (`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`$sdev)^2/sum((`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`$sdev)^2))

`2022-09-13_screeplot_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggplot(data = `2022-09-13_var_explained_PCA_working_bin_level_salmon_bias_flagged_dataframe`[`2022-09-13_var_explained_PCA_working_bin_level_salmon_bias_flagged_dataframe`$var_explained > 0.010, ], aes(x = reorder(PC, -var_explained), y = var_explained, colour = "black", fill = ifelse(var_explained > 0.05, 'black', 'red'))) + geom_bar(stat = 'identity') + scale_fill_manual(values = c("#ff1e90", "#f2f3f4")) + scale_colour_manual(values = c("#0f0f0f")) + coord_flip() + theme_linedraw() + ylab("Proportion of variance explained") + xlab("Principal Components") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

`2022-09-13_PCA_PC1_vs_PC2_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-09-13_PCA_PC1_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-09-13_PCA_PC2_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-09-13_PCA_PC1_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(1,4), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-09-13_PCA_PC2_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(2,4), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-09-13_PCA_PC3_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(3,4), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-09-13_screeplot_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` + `2022-09-13_PCA_PC1_vs_PC2_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` + `2022-09-13_PCA_PC1_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` + plot_layout(guides = "collect") + guide_area() + plot_annotation(tag_levels = 'A')

`2022-09-13_PCA_PC1_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` + `2022-09-13_PCA_PC2_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` + plot_layout(guides = "collect") + plot_layout(ncol = 2) + plot_annotation(tag_levels = 'A')

`2022-09-13_PCA_PC2_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` + `2022-09-13_PCA_PC3_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` + plot_layout(guides = "collect") + plot_layout(ncol = 2) + plot_annotation(tag_levels = 'A')

grid.arrange(`2022-09-13_screeplot_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged`, `2022-09-13_PCA_PC1_vs_PC2_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged`, `2022-09-13_PCA_PC1_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged`, `2022-09-13_PCA_PC1_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged`, nrow = 2)

save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-13.RData")

`only_novogene_paired_melted_good_tpms`
`2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins` <- filter(`2022.07.18_transposed_only_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`, Sequencing == "Novogene")
`2022.07.18_PCA_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins` <- prcomp(`2022.07.18_transposed_only_novogene_paired_reads_mapping_to_bins_library_and_contig_size_adjusted_summed_to_bins`[,c(2:349)], center = TRUE, scale. = FALSE)

save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-13.RData")

2022-09-14
ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$variable != 'unclassified' & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value >= 2.5, ], aes(x = variable, y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Taxa classified by Kraken2 with >2.5% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + theme(text = element_text(family = "Tex Gyre Heroes")) + ylim(2.5,10.5) + facet_wrap(~ Sequencing) + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 7.5))

temp_fig_A <- ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$variable != 'unclassified' & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 2.5, ], aes(x = reorder(variable, -value), y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Taxa classified by Kraken2 with >2.5% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + theme(text = element_text(family = "Tex Gyre Heroes")) + ylim(2.5,10.5) + facet_wrap(~ Sequencing) + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 7.5))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-04_all_samples_subsampled_depths_Kraken_percent_classifications_all_actual_taxa_by_host_x_season_050_conf_score_percent_reads_over_2_point_5_at_least_one_sample.svg", `temp_fig_A`)

`2022-09-13_screeplot_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggplot(data = `2022-09-13_var_explained_PCA_working_bin_level_salmon_bias_flagged_dataframe`[`2022-09-13_var_explained_PCA_working_bin_level_salmon_bias_flagged_dataframe`$var_explained > 0.010, ], aes(x = reorder(PC, -var_explained), y = var_explained, colour = "black", fill = ifelse(var_explained > 0.05, 'black', 'red'))) + geom_bar(stat = 'identity') + scale_fill_manual(values = c("#ff1e90", "#f2f3f4")) + scale_colour_manual(values = c("#0f0f0f")) + coord_flip() + theme_linedraw() + ylab("Proportion of variance explained") + xlab("Principal Components") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-13_screeplot_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged.svg", `2022-09-13_screeplot_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged`)

`2022-09-13_PCA_PC1_vs_PC2_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-13_PCA_PC1_vs_PC2_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged.svg", `2022-09-13_PCA_PC1_vs_PC2_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged`)

`2022-09-13_PCA_PC1_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-13_PCA_PC1_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged.svg", `2022-09-13_PCA_PC1_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged`)

`2022-09-13_PCA_PC2_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-13_PCA_PC2_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged.svg", `2022-09-13_PCA_PC2_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged`)

`2022-09-13_PCA_PC1_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(1,4), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-09-13_PCA_PC1_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged.svg", `2022-09-13_PCA_PC1_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged`)

`2022-09-13_PCA_PC2_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(2,4), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-13_PCA_PC2_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged.svg", `2022-09-13_PCA_PC2_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged`)

`2022-09-13_PCA_PC3_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` <- ggbiplot::ggbiplot(`2022-09-13_PCA_working_bin_level_salmon_bias_flagged`, choices = c(3,4), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-13_working_bin_level_salmon_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Season), color=factor(`2022-09-13_working_bin_level_salmon_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-13_PCA_PC3_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged.svg", `2022-09-13_PCA_PC3_vs_PC4_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged`)

`2022_08_03_TRANSPOSED_Meta_data_added_2nd_copy_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins.tsv` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_08_03_TRANSPOSED_Meta_data_added_2nd_copy_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins.tsv.txt")
view(`2022_08_03_TRANSPOSED_Meta_data_added_2nd_copy_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins.tsv`)
`2022-09-13_working_bin_level_salmon_bias_flagged` <- `2022_08_03_TRANSPOSED_Meta_data_added_2nd_copy_Salmon_TPMs_gc_and_seq_bias_flags_on_all_subsampled_reads_minimap2_mapped_concatenated_single_contig_per_bin_all_strict_bins.tsv`
view(`2022-09-13_working_bin_level_salmon_bias_flagged`)
`2022-09-13_melted_working_bin_level_salmon_bias_flagged` <- reshape2::melt(`2022-09-13_working_bin_level_salmon_bias_flagged`, id=c("Name", "Species", "Season", "Sequencing", "Species_x_season", "Sample_ID"))
`2022-09-13_PCA_working_bin_level_salmon_bias_flagged` <- prcomp(`2022-09-13_working_bin_level_salmon_bias_flagged`[,c(2:349)], center = TRUE, scale. = FALSE)

`2022-09-13_working_bin_level_salmon_bias_flagged`[`2022-09-13_working_bin_level_salmon_bias_flagged`$Sequencing == 'Novogene',]

view(`2022-09-13_working_bin_level_salmon_bias_flagged`[`2022-09-13_working_bin_level_salmon_bias_flagged`$Sequencing == 'Novogene' & `2022-09-13_working_bin_level_salmon_bias_flagged`$Sample_ID %in% `only_novogene_paired_melted_good_tpms`$Sample_ID, ])

`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- `2022-09-13_working_bin_level_salmon_bias_flagged`[`2022-09-13_working_bin_level_salmon_bias_flagged`$Sequencing == 'Novogene' & `2022-09-13_working_bin_level_salmon_bias_flagged`$Sample_ID %in% `only_novogene_paired_melted_good_tpms`$Sample_ID, ]
`2022-09-14_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- prcomp(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`[,c(2:349)], center = TRUE, scale. = FALSE)

`2022-09-14_var_explained_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- data.frame(PC = paste0("PC", 1:58), var_explained = (`2022-09-14_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$sdev)^2/sum((`2022-09-14_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$sdev)^2))

`2022-09-14_var_explained_screeplot_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- ggplot(data = `2022-09-14_var_explained_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`[`2022-09-14_var_explained_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$var_explained > 0.010, ], aes(x = reorder(PC, -var_explained), y = var_explained, colour = "black", fill = ifelse(var_explained > 0.05, 'black', 'red'))) + geom_bar(stat = 'identity') + scale_fill_manual(values = c("#ff1e90", "#f2f3f4")) + scale_colour_manual(values = c("#0f0f0f")) + coord_flip() + theme_linedraw() + ylab("Proportion of variance explained") + xlab("Principal Components") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-14_var_explained_screeplot_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged.svg", `2022-09-14_var_explained_screeplot_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`)

`2022-09-14_PCA_PC1_vs_PC2_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- ggbiplot::ggbiplot(`2022-09-14_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Season), color=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-14_PCA_PC1_vs_PC2_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged.svg", `2022-09-14_PCA_PC1_vs_PC2_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`)

`2022-09-14_PCA_PC1_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- ggbiplot::ggbiplot(`2022-09-14_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`, choices = c(1,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Season), color=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-14_PCA_PC1_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged.svg", `2022-09-14_PCA_PC1_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`)

`2022-09-14_PCA_PC1_vs_PC4_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- ggbiplot::ggbiplot(`2022-09-14_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`, choices = c(1,4), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Season), color=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-14_PCA_PC1_vs_PC4_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged.svg", `2022-09-14_PCA_PC1_vs_PC4_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`)

`2022-09-14_PCA_PC1_vs_PC5_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- ggbiplot::ggbiplot(`2022-09-14_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`, choices = c(1,5), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Season), color=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-14_PCA_PC1_vs_PC5_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged.svg", `2022-09-14_PCA_PC1_vs_PC5_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`)

`2022-09-14_PCA_PC2_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- ggbiplot::ggbiplot(`2022-09-14_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`, choices = c(2,3), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Season), color=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-14_PCA_PC2_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged.svg", `2022-09-14_PCA_PC2_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`)

`2022-09-14_PCA_PC3_vs_PC4_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- ggbiplot::ggbiplot(`2022-09-14_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`, choices = c(3,4), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Season), color=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-14_PCA_PC3_vs_PC4_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged.svg", `2022-09-14_PCA_PC3_vs_PC4_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`)

`2022-09-14_PCA_PC4_vs_PC5_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- ggbiplot::ggbiplot(`2022-09-14_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`, choices = c(4,5), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Season), color=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-14_PCA_PC4_vs_PC5_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged.svg", `2022-09-14_PCA_PC4_vs_PC5_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`)

`2022-09-14_var_explained_screeplot_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` / `2022-09-14_PCA_PC1_vs_PC2_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` + plot_layout(guides = "keep") + plot_annotation(tag_levels = 'A')

`2022-09-13_screeplot_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` + `2022-09-13_PCA_PC1_vs_PC2_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` + `2022-09-13_PCA_PC1_vs_PC3_ALL_subsampled_reads_salmon_tpms_bin_level_strict_bins_only_bias_and_GC_flagged` + plot_layout(guides = "collect") + guide_area() + plot_annotation(tag_levels = 'A')

`2022-09-14_PCA_PC1_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` / `2022-09-14_PCA_PC1_vs_PC4_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` / `2022-09-14_PCA_PC1_vs_PC5_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

`2022-09-14_PCA_PC2_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` / `2022-09-14_PCA_PC3_vs_PC4_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` / `2022-09-14_PCA_PC4_vs_PC5_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2023-01-07_for_presentation_pcas.svg")


save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-13.RData")

2022-09-21
ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + theme(text = element_text(size = 14))

ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = June.to.November.log2fold.add.one.change, y = -log10(Season.corrected.pvalue))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Time effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + theme(text = element_text(size = 14))

`2022-09-21_only_novogene_paired_strict_bin_mapping_salmon_GC_and_seq_bias_flagged_bin_level_host_effect_foldchange_and_qvalue_fig` <- ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue), colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#0f0f0f", "#848482")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

`2022-09-21_only_novogene_paired_strict_bin_mapping_salmon_GC_and_seq_bias_flagged_bin_level_month_effect_foldchange_and_qvalue_fig` <- ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = June.to.November.log2fold.add.one.change, y = -log10(Season.corrected.pvalue), colour = ifelse((June.to.November.log2fold.add.one.change > 1 & -log10(Season.corrected.pvalue) > -log10(0.05)) | (June.to.November.log2fold.add.one.change < -1 & -log10(Season.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Time effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#848482", "#0f0f0f")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

`2022-09-21_host_and_season_effect_novogene_paired_only_mapping_salmon_strict_bins_fold_change_and_qvalue_figure_combined_both` <- `2022-09-21_only_novogene_paired_strict_bin_mapping_salmon_GC_and_seq_bias_flagged_bin_level_host_effect_foldchange_and_qvalue_fig` + `2022-09-21_only_novogene_paired_strict_bin_mapping_salmon_GC_and_seq_bias_flagged_bin_level_month_effect_foldchange_and_qvalue_fig`
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-21_host_and_season_effect_novogene_paired_only_mapping_salmon_strict_bins_fold_change_and_qvalue_figure_combined_both.svg", `2022-09-21_host_and_season_effect_novogene_paired_only_mapping_salmon_strict_bins_fold_change_and_qvalue_figure_combined_both`)

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05,]

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)]

ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -2,]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))

ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 3,]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))

save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-21.RData")

2022-09-22
view(`only_novogene_paired_melted_good_tpms`)

`AC_exclusive_only_novogene_paired_melted_good_tpms` <- `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$Species == "AC",]
`AR_exclusive_only_novogene_paired_melted_good_tpms` <- `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$Species == "AR",]

`2022-09-23_AC_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AC_seasonal_change_table` <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Bin", "ACJ.geometric.mean", "ACN.geometric.mean", "AC.seasonal.raw.pvalue", "log2.add.one.AC.seasonal.fold.change", "GTDB_TK_taxonomy"))))
tic()
for (bin in strict_bins_list){
  bin_name <- paste(bin)
  ACJ.geometric.mean <- exp(mean(log(`AC_exclusive_only_novogene_paired_melted_good_tpms`[`AC_exclusive_only_novogene_paired_melted_good_tpms`$Species_x_season == "ACJ" & `AC_exclusive_only_novogene_paired_melted_good_tpms`$variable == bin_name,]$value)))
  ACN.geometric.mean <- exp(mean(log(`AC_exclusive_only_novogene_paired_melted_good_tpms`[`AC_exclusive_only_novogene_paired_melted_good_tpms`$Species_x_season == "ACN" & `AC_exclusive_only_novogene_paired_melted_good_tpms`$variable == bin_name,]$value)))
  AC.seasonal.raw.pvalue <- wilcox.test(value ~ Species_x_season, paired = TRUE, data = `AC_exclusive_only_novogene_paired_melted_good_tpms`[`AC_exclusive_only_novogene_paired_melted_good_tpms`$variable == bin_name,])$p.value
  log2.add.one.AC.seasonal.fold.change <- log2(((`ACN.geometric.mean` - `ACJ.geometric.mean`) / `ACJ.geometric.mean`) +1)
  GTDB_TK_taxonomy <- `2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`[`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`$Bin.name == bin_name, ]$Bin.long.taxa
  `2022-09-23_AC_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AC_seasonal_change_table`[nrow(`2022-09-23_AC_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AC_seasonal_change_table`) +1,] <- c(bin_name, ACJ.geometric.mean, ACN.geometric.mean, AC.seasonal.raw.pvalue, log2.add.one.AC.seasonal.fold.change, GTDB_TK_taxonomy)
  `2022-09-23_AC_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AC_seasonal_change_table`[,c(2:5)] <- sapply(`2022-09-23_AC_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AC_seasonal_change_table`[,c(2:5)], as.numeric)
  rm(bin_name)
  rm(ACJ.geometric.mean)
  rm(ACN.geometric.mean)
  rm(AC.seasonal.raw.pvalue)
  rm(log2.add.one.AC.seasonal.fold.change)
  rm(GTDB_TK_taxonomy)
}
toc()

`2022-0-23_AC_seasonal_raw_pvalue_SORTED` <- `2022-09-23_AC_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AC_seasonal_change_table`[order(`2022-09-23_AC_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AC_seasonal_change_table`$AC.seasonal.raw.pvalue),]
`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues <- p.adjust(`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$AC.seasonal.raw.pvalue, method = "BH")

stripchart(`AC_exclusive_only_novogene_paired_melted_good_tpms`[`AC_exclusive_only_novogene_paired_melted_good_tpms`$Species_x_season == "ACJ" & `AC_exclusive_only_novogene_paired_melted_good_tpms`$variable == "Bin_c1000",]$value, pch = 19, method = "stack", col = "#85bb65")

stripchart(value ~ Species_x_season, group.names = c("ACJ", "ACN"), pch = 19, method = "jitter", jitter = 0.2, vertical = TRUE, data = `AC_exclusive_only_novogene_paired_melted_good_tpms`[`AC_exclusive_only_novogene_paired_melted_good_tpms`$Species == "AC" & `AC_exclusive_only_novogene_paired_melted_good_tpms`$variable == "Bin_c1000",])
stripchart(value ~ Species_x_season, group.names = c("ACJ", "ACN"), pch = 19, method = "jitter", jitter = 0.2, vertical = FALSE, col = c("#85bb65", "#bb6585"), data = `AC_exclusive_only_novogene_paired_melted_good_tpms`[`AC_exclusive_only_novogene_paired_melted_good_tpms`$Species == "AC" & `AC_exclusive_only_novogene_paired_melted_good_tpms`$variable == "Bin_c1000",])

tic()
`2022-09-23_AR_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AR_seasonal_change_table` <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Bin", "ARJ.geometric.mean", "ARN.geometric.mean", "AR.seasonal.raw.pvalue", "log2.add.one.AR.seasonal.fold.change", "GTDB_TK_taxonomy"))))
for (bin in strict_bins_list){
  bin_name <- paste(bin)
  ARJ.geometric.mean <- exp(mean(log(`AR_exclusive_only_novogene_paired_melted_good_tpms`[`AR_exclusive_only_novogene_paired_melted_good_tpms`$Species_x_season == "ARJ" & `AR_exclusive_only_novogene_paired_melted_good_tpms`$variable == bin_name,]$value)))
  ARN.geometric.mean <- exp(mean(log(`AR_exclusive_only_novogene_paired_melted_good_tpms`[`AR_exclusive_only_novogene_paired_melted_good_tpms`$Species_x_season == "ARN" & `AR_exclusive_only_novogene_paired_melted_good_tpms`$variable == bin_name,]$value)))
  AR.seasonal.raw.pvalue <- wilcox.test(value ~ Species_x_season, paired = TRUE, data = `AR_exclusive_only_novogene_paired_melted_good_tpms`[`AR_exclusive_only_novogene_paired_melted_good_tpms`$variable == bin_name,])$p.value
  log2.add.one.AR.seasonal.fold.change <- log2(((`ARN.geometric.mean` - `ARJ.geometric.mean`) / `ARJ.geometric.mean`) +1)
  GTDB_TK_taxonomy <- `2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`[`2022.07.14_all_strict_bins_GTDB.TK_full_taxonomy_and_ACTUAL_family_or_genus`$Bin.name == bin_name, ]$Bin.long.taxa
  `2022-09-23_AR_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AR_seasonal_change_table`[nrow(`2022-09-23_AR_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AR_seasonal_change_table`) +1,] <- c(bin_name, ARJ.geometric.mean, ARN.geometric.mean, AR.seasonal.raw.pvalue, log2.add.one.AR.seasonal.fold.change, GTDB_TK_taxonomy)
  `2022-09-23_AR_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AR_seasonal_change_table`[,c(2:5)] <- sapply(`2022-09-23_AR_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AR_seasonal_change_table`[,c(2:5)], as.numeric)
  rm(bin_name)
  rm(ARJ.geometric.mean)
  rm(ARN.geometric.mean)
  rm(AR.seasonal.raw.pvalue)
  rm(log2.add.one.AR.seasonal.fold.change)
  rm(GTDB_TK_taxonomy)
}

`2022-0-23_AR_seasonal_raw_pvalue_SORTED` <- `2022-09-23_AR_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AR_seasonal_change_table`[order(`2022-09-23_AR_paired_novogene_strict_bin_mapping_Salmon_GC_and_seq_bias_flagged_within_AR_seasonal_change_table`$AR.seasonal.raw.pvalue),]
`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues <- p.adjust(`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$AR.seasonal.raw.pvalue, method = "BH")
toc()

ggplot(data = `2022-0-23_AC_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AC.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), colour = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AC seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#1e90ff", "#b5b5b3")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

ggplot(data = `2022-0-23_AR_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AR.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), colour = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AR seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#ff8d1e", "#b5b5b3")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

ggplot(data = `2022-0-23_AR_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AR.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AR seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#ff8d1e", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_text(aes(label = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0)) + theme(legend.position = "none")

ggplot(data = `2022-0-23_AR_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AR.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AR seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#ff8d1e", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#9e4e00") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

ggplot(data = `2022-0-23_AC_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AC.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AC seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#1e90ff", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#004385") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

ggplot(data = `2022-0-23_AC_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AC.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AC seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#1e90ff", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#004385", min.segment.length = 0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

ggplot(data = `2022-0-23_AR_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AR.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AR seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#ff8d1e", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#9e4e00", min.segment.length = 0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")


`2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure` <- ggplot(data = `2022-0-23_AC_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AC.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AC seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#1e90ff", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#004385", min.segment.length = 0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure.svg", `2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure`)

`2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure` <- ggplot(data = `AC_exclusive_only_novogene_paired_melted_good_tpms`[`AC_exclusive_only_novogene_paired_melted_good_tpms`$variable %in% `2022-0-23_AC_seasonal_raw_pvalue_SORTED`[`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$log2.add.one.AC.seasonal.fold.change > 1 & -log10(`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05) | `2022-0-23_AC_seasonal_raw_pvalue_SORTED`$log2.add.one.AC.seasonal.fold.change < -1 & -log10(`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05),]$Bin, ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#004385")) + scale_colour_manual(values = c ("#1e90ff", "#004385")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("RPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure.svg", `2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure`)

`2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure` <- ggplot(data = `2022-0-23_AR_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AR.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AR seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#ff8d1e", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#9e4e00", min.segment.length = 0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure.svg", `2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure`)

`2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure` <- ggplot(data = `AR_exclusive_only_novogene_paired_melted_good_tpms`[`AR_exclusive_only_novogene_paired_melted_good_tpms`$variable %in% `2022-0-23_AR_seasonal_raw_pvalue_SORTED`[`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$log2.add.one.AR.seasonal.fold.change > 1 & -log10(`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05) | `2022-0-23_AR_seasonal_raw_pvalue_SORTED`$log2.add.one.AR.seasonal.fold.change < -1 & -log10(`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05),]$Bin, ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#ff8d1e", "#522800")) + scale_colour_manual(values = c ("#ff8d1e", "#522800")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("RPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure.svg", `2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure`)

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-23.RData")
toc()

2022-09-26
`2022.02.09_All_processed_sample_files_percent_classified_reads_from_raw_files_Kaiju_Kraken_from_report_Metaphlan` <- read.csv("~/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-02-09_All_processed_sample_files_percent_classified_reads_from_raw_files_Kaiju_Kraken_from_report_Metaphlan.csv")
View(`2022.02.09_All_processed_sample_files_percent_classified_reads_from_raw_files_Kaiju_Kraken_from_report_Metaphlan`)

feb_18th_working_file <- `2022.02.09_All_processed_sample_files_percent_classified_reads_from_raw_files_Kaiju_Kraken_from_report_Metaphlan`

melted_feb_18th_working_file <- reshape2::melt(feb_18th_working_file, id=c("Sample", "Host_species", "Sample_ID"), measure.vars=c("Metaphlan", "Kaiju_0_errors_allowed", "Kaiju_1_error_allowed", "Kaiju_2_errors_allowed", "Kaiju_3_errors_allowed", "Kaiju_4_errors_allowed", "Kaiju_5_errors_allowed", "Kaiju_10_errors_allowed", "Kaiju_20_errors_allowed", "Kraken_95_percent_confidence", "Kraken_75_percent_confidence", "Kraken_66_percent_confidence", "Kraken_50_percent_confidence", "Kraken_25_percent_confidence", "Kraken_10_percent_confidence"))

`melted_feb_18th_working_file`$variable <- gsub("_", " ", as.character(`melted_feb_18th_working_file`$variable))
`melted_feb_18th_working_file`$variable <- gsub("percent", "%", as.character(`melted_feb_18th_working_file`$variable))
`melted_feb_18th_working_file`$Sample_ID <- gsub("_", " ", as.character(`melted_feb_18th_working_file`$Sample_ID))
`melted_feb_18th_working_file`$Sample <- gsub("_", " ", as.character(`melted_feb_18th_working_file`$Sample))

subset(`melted_feb_18th_working_file`, `melted_feb_18th_working_file`$Host_species != "HS")

`melted_feb_18th_working_file_excluding_human` <- subset(`melted_feb_18th_working_file`, `melted_feb_18th_working_file`$Host_species != "HS")

`melted_feb_18th_working_file_excluding_human_kaiju_only` <- subset(`melted_feb_18th_working_file_excluding_human`, grepl(glob2rx("Kaiju*"), variable))

view(`melted_feb_18th_working_file_excluding_human_kaiju_only`[! `melted_feb_18th_working_file_excluding_human_kaiju_only`$Sample_ID == "SRR9205148" | `melted_feb_18th_working_file_excluding_human_kaiju_only`$Sample_ID == "SRR9205149",])
interim <- `melted_feb_18th_working_file_excluding_human_kaiju_only`[! `melted_feb_18th_working_file_excluding_human_kaiju_only`$Sample_ID == "SRR9205148",]
fonal <- `interim`[! `interim`$Sample_ID == "SRR9205149,]
`melted_feb_18th_working_file_excluding_human_kaiju_only` <- `fonal`
`melted_feb_18th_working_file_excluding_human_kaiju_only`$variable <- factor(`melted_feb_18th_working_file_excluding_human_kaiju_only`$variable, levels = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed"))

`melted_feb_18th_working_file_ONLY_human` <- subset(`melted_feb_18th_working_file`, `melted_feb_18th_working_file`$Host_species == "HS")

`melted_feb_18th_working_file_ONLY_human`$variable <- factor(`melted_feb_18th_working_file_ONLY_human`$variable, levels = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed", "Kraken 95 % confidence", "Kraken 75 % confidence", "Kraken 66 % confidence", "Kraken 50 % confidence", "Kraken 25 % confidence", "Kraken 10 % confidence", "Metaphlan"))

`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species <- gsub("AC", "Acomys cahirinus", as.character(`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species))
`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species <- gsub("AR", "Acomys russatus", as.character(`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species))
`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species <- gsub("MM", "Mus musculus", as.character(`melted_feb_18th_working_file_excluding_human_kaiju_only`$Host_species))

`melted_feb_18th_working_file_excluding_human`$Host_species <- gsub("AC", "Acomys cahirinus", as.character(`melted_feb_18th_working_file_excluding_human`$Host_species))
`melted_feb_18th_working_file_excluding_human`$Host_species <- gsub("AR", "Acomys russatus", as.character(`melted_feb_18th_working_file_excluding_human`$Host_species))
`melted_feb_18th_working_file_excluding_human`$Host_species <- gsub("MM", "Mus musculus", as.character(`melted_feb_18th_working_file_excluding_human`$Host_species))

`melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(variable) %>% dplyr::summarise(GroupMedianValue = median(value), .groups = 'drop')

`melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(variable) %>% dplyr::summarise(GroupMedianValue = median(value), .groups = 'keep')

`2022_02_23_dplyr_medians_and_confidence_scores_testing_rodent_only_samples_kaiju_and_kraken_percent_classified_reads` <- `melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(GroupMedianValue = median(value), .groups = 'keep')

rm(`2022_02_23_dplyr_medians_and_confidence_scores_testing_rodent_only_samples_kaiju_and_kraken_percent_classified_reads`)

`2022_02_23_rodent_only_kaiju_working_file` <- `melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, method =c("exact"))[1], lower_confidence_interval = MedianCI(value, method =c("exact"))[2], upper_confidence_interval = MedianCI(value, method =c("exact"))[3])

`2022_02_23_rodent_only_kaiju_working_file` <- `melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[3])

rm(`2022_02_23_rodent_only_kaiju_working_file`)

DescTools::MedianCI(`melted_feb_18th_working_file_excluding_human_kaiju_only`$value, method = "exact", conf.level = 0.95)

`melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[3])

`melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("exact"))[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("exact"))[3])

`melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.75, sides = c("two.sided"), method =c("exact"))[1], lower_confidence_interval = MedianCI(value, conf.level = 0.75, method =c("exact"))[2], upper_confidence_interval = MedianCI(value, conf.level = 0.75, method =c("exact"))[3]) %>% View()

`2022_02_23_rodent_only_kaiju_working_file` <- `melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("exact"))[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("exact"))[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("exact"))[3])

`2022_02_23_rodent_only_kaiju_working_file` <- `melted_feb_18th_working_file_excluding_human_kaiju_only` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("boot"), R = 999)[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("boot"), R = 999)[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("boot"), R = 999)[3])

ggplot(`2022_02_23_rodent_only_kaiju_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + scale_y_continuous(oob=scales::discard)

ggplot(`2022_02_23_rodent_only_kaiju_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable, label = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_text(y = 10, label = "Text here") + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + scale_y_continuous(oob=scales::discard)


theme(axis.text.x=element_blank()) + theme(axis.ticks.x=element_blank())

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-26.RData")
toc()

2022-09-27
ggplot(`2022_02_23_rodent_only_kaiju_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank()) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + scale_y_continuous(oob=scales::discard)

`2022_02_23_rodent_only_kaiju_working_file_annotations_dataframe` <- data.frame(
label = c("0", "1", "2", "3", "4", "5", "10", "20"),
variable = c("Kaiju 0 errors allowed", "Kaiju 1 error allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed")
)

ggplot(`2022_02_23_rodent_only_kaiju_working_file`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_annotations_dataframe`, mapping = aes(x = `2022_02_23_rodent_only_kaiju_working_file`$variable, y = 85, colour = variable, label = label)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank()) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + scale_y_continuous(oob=scales::discard)

`2022_02_23_rodent_only_kaiju_working_file`

write.table(`2022_02_23_rodent_only_kaiju_working_file`, "2022_09_27_chapter2_pilot_stuff_rodent_kaiju_medians_table.tsv", sep="\t", row.names=FALSE)

`2022_02_23_rodent_only_kaiju_working_file_v2` <- `2022_09_27_chapter2_pilot_stuff_rodent_kaiju_medians_table`

ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`[`2022_02_23_rodent_only_kaiju_working_file_v2`$Host_species == "Acomys cahirinus",], mapping = aes(x = variable, y = 85, label = Label), position=position_dodge(width=1.05)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank()) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + scale_y_continuous(oob=scales::discard)

ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`[`2022_02_23_rodent_only_kaiju_working_file_v2`$Host_species == "Acomys cahirinus",], aes(x = Host_species, y = 85, colour = variable, label = Label), position=position_dodge(width=1.05), size = 3) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank()) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + scale_y_continuous(oob=scales::discard)

`2022_02_23_rodent_only_kaiju_working_file_v2_level_order` <- c("Kaiju 0 errors allowed", "Kaiju 1 errors allowed", "Kaiju 2 errors allowed", "Kaiju 3 errors allowed", "Kaiju 4 errors allowed", "Kaiju 5 errors allowed", "Kaiju 10 errors allowed", "Kaiju 20 errors allowed")

ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`[`2022_02_23_rodent_only_kaiju_working_file_v2`$Host_species == "Acomys cahirinus",], aes(x = Host_species, y = 85, colour = variable, label = Label), position=position_dodge(width=1.05), size = 3) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank()) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all samples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + scale_y_continuous(oob=scales::discard)

ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`[`2022_02_23_rodent_only_kaiju_working_file_v2`$Host_species == "Acomys cahirinus",], aes(x = Host_species, y = 82.5, colour = variable, label = factor(Label, level = c('0', '1', '2', '3', '4', '5', '10', '20'))), position=position_dodge(width=1.05), size = 3) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank()) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all\nsamples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + scale_y_continuous(oob=scales::discard)

ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`[`2022_02_23_rodent_only_kaiju_working_file_v2`$Host_species == "Acomys cahirinus",], aes(x = Host_species, y = 82.5, colour = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), label = Label), position=position_dodge(width=1.05), size = 3) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank()) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all\nsamples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + scale_y_continuous(oob=scales::discard)

`2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kaiju_median_max_min_reads_classified_per_error_allowance_by_host_species` <- ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`[`2022_02_23_rodent_only_kaiju_working_file_v2`$Host_species == "Acomys cahirinus",], aes(x = Host_species, y = 82.5, colour = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), label = Label), position=position_dodge(width=1.05), size = 3) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank()) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="bottom") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all\nsamples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + scale_y_continuous(oob=scales::discard)

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kaiju_median_max_min_reads_classified_per_error_allowance_by_host_species.svg", `2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kaiju_median_max_min_reads_classified_per_error_allowance_by_host_species`)

`2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kaiju_median_max_min_reads_classified_per_error_allowance_by_host_species` <- ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`[`2022_02_23_rodent_only_kaiju_working_file_v2`$Host_species == "Acomys cahirinus",], aes(x = Host_species, y = 56, colour = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), label = Label), position=position_dodge(width=1.05), size = 3) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all\nsamples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + xlab("Host species") + scale_y_continuous(oob=scales::discard) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank(), strip.text.x = element_text(face = "italic"))

ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`[`2022_02_23_rodent_only_kaiju_working_file_v2`$Host_species == "Acomys cahirinus",], aes(x = Host_species, y = 56, colour = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), label = Label), position=position_dodge(width=1.05), size = 3) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all\nsamples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + xlab("Host species") + scale_y_continuous(oob=scales::discard) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank(), strip.text.x = element_text(face = "italic"))


save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-27.RData")

2022-09-29
`2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kaiju_median_max_min_reads_classified_per_error_allowance_by_host_species` <- ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`[`2022_02_23_rodent_only_kaiju_working_file_v2`$Host_species == "Acomys cahirinus",], aes(x = Host_species, y = 56, colour = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), label = Label), position=position_dodge(width=1.05), size = 3.5, show.legend = FALSE) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all\nsamples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + xlab("Host species") + scale_y_continuous(oob=scales::discard) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank(), strip.text.x = element_text(face = "italic"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kaiju_median_max_min_reads_classified_per_error_allowance_by_host_species.svg", `2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kaiju_median_max_min_reads_classified_per_error_allowance_by_host_species`)

`2022_02_23_rodent_only_kraken_working_file` <- subset(`melted_feb_18th_working_file_excluding_human`, grepl(glob2rx("Kraken*"), variable))

`2022_02_23_rodent_only_kraken_working_file` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("boot"), R = 999)[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("boot"), R = 999)[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("boot"), R = 999)[3])
`2022_02_23_rodent_only_kraken_working_file_medians` <- `2022_02_23_rodent_only_kraken_working_file` %>% dplyr::group_by(Host_species, variable) %>% dplyr::summarise(Classified_median_percent = MedianCI(value, conf.level = 0.95, sides = c("two.sided"), method =c("boot"), R = 999)[1], lower_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("boot"), R = 999)[2], upper_confidence_interval = MedianCI(value, conf.level = 0.95, method =c("boot"), R = 999)[3])

ggplot(`2022_02_23_rodent_only_kraken_working_file_medians`, aes(x = Host_species, color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=0.75)) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=0.75)) + theme_linedraw() + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + ggtitle("Median percent reads classified by Kraken using different classification confidence scores\nfrom all samples by host species") + scale_fill_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + scale_color_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))+ guides(fill = guide_legend(title = "Kraken classification confidence score"), color = guide_legend(title = "Kraken classification confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank(), axis.ticks.x=element_blank(), text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(oob=scales::discard)

write.table(`2022_02_23_rodent_only_kraken_working_file_medians`, file = "2022_09_27_chapter2_pilot_stuff_rodent_kraken_medians_table.tsv", sep="\t", row.names=FALSE)

`2022_02_23_rodent_only_kraken_working_file_medians_V2` <- read.delim("~/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022_09_27_chapter2_pilot_stuff_rodent_kraken_medians_table.tsv")

ggplot(`2022_02_23_rodent_only_kraken_working_file_medians_V2`, aes(x = Host_species, color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=0.75)) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=0.75)) + theme_linedraw() + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + ggtitle("Median percent reads classified by Kraken using different classification confidence scores\nfrom all samples by host species") + scale_fill_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + scale_color_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))+ guides(fill = guide_legend(title = "Kraken classification confidence score"), color = guide_legend(title = "Kraken classification confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank(), axis.ticks.x=element_blank(), text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(oob=scales::discard)

ggplot(`2022_02_23_rodent_only_kraken_working_file_medians_V2`, aes(x = Host_species, color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.25)) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.25)) + theme_linedraw() + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + geom_text(data = `2022_02_23_rodent_only_kraken_working_file_medians_V2`[`2022_02_23_rodent_only_kraken_working_file_medians_V2`$Host_species == "Acomys cahirinus",], aes(x = Host_species, y = -8, colour = factor(variable, level = c('Kraken 10 % confidence', 'Kraken 25 % confidence', 'Kraken 50 % confidence', 'Kraken 66 % confidence', 'Kraken 75 % confidence', 'Kraken 95 % confidence')), label = label), position=position_dodge(width=1.55), size = 3, show.legend = FALSE) + ggtitle("Median percent reads classified by Kraken using different classification confidence scores\nfrom all samples by host species") + scale_fill_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + scale_color_manual(values = c("Kraken 10 % confidence" = "#88c4ed", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))+ guides(fill = guide_legend(title = "Kraken classification confidence score"), color = guide_legend(title = "Kraken classification confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank(), axis.ticks.x=element_blank(), text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(oob=scales::discard)

`2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kraken_median_max_min_reads_classified_per_error_allowance_by_host_species` <- ggplot(`2022_02_23_rodent_only_kraken_working_file_medians_V2`, aes(x = Host_species, color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.25)) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.25)) + theme_linedraw() + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + geom_text(data = `2022_02_23_rodent_only_kraken_working_file_medians_V2`, aes(x = Host_species, y = -8, colour = factor(variable, level = c('Kraken 10 % confidence', 'Kraken 25 % confidence', 'Kraken 50 % confidence', 'Kraken 66 % confidence', 'Kraken 75 % confidence', 'Kraken 95 % confidence')), label = label), position=position_dodge(width=1.55), size = 3, show.legend = FALSE) + ggtitle("Median percent reads classified by Kraken using different classification confidence scores\nfrom all samples by host species") + scale_fill_manual(values = c("Kraken 10 % confidence" = "#7b66e8", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + scale_color_manual(values = c("Kraken 10 % confidence" = "#7b66e8", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))+ guides(fill = guide_legend(title = "Kraken classification confidence score"), color = guide_legend(title = "Kraken classification confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank(), axis.ticks.x=element_blank(), text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(oob=scales::discard)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kraken_median_max_min_reads_classified_per_error_allowance_by_host_species.svg", `2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kraken_median_max_min_reads_classified_per_error_allowance_by_host_species`)

`2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kraken_median_max_min_reads_classified_per_error_allowance_by_host_species` <- ggplot(`2022_02_23_rodent_only_kraken_working_file_medians_V2`, aes(x = Host_species, color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.25)) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.25)) + theme_linedraw() + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + geom_text(data = `2022_02_23_rodent_only_kraken_working_file_medians_V2`, aes(x = Host_species, y = -8, colour = factor(variable, level = c('Kraken 10 % confidence', 'Kraken 25 % confidence', 'Kraken 50 % confidence', 'Kraken 66 % confidence', 'Kraken 75 % confidence', 'Kraken 95 % confidence')), label = label), position=position_dodge(width=1.55), size = 4, show.legend = FALSE) + ggtitle("Median percent reads classified by Kraken using different classification confidence scores\nfrom all samples by host species") + scale_fill_manual(values = c("Kraken 10 % confidence" = "#7b66e8", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + scale_color_manual(values = c("Kraken 10 % confidence" = "#7b66e8", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + guides(fill = guide_legend(title = "Percent confidence score"), color = guide_legend(title = "Percent confidence score"))+ guides(fill = guide_legend(title = "Kraken classification confidence score"), color = guide_legend(title = "Kraken classification confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("Host species") + theme(axis.text.x=element_blank(), axis.ticks.x=element_blank(), text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(oob=scales::discard) + theme(strip.text.x = element_text(size = 14)) + theme(text = element_text(size = 14))

`2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kaiju_median_max_min_reads_classified_per_error_allowance_by_host_species` <- ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = 56, colour = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), label = Label), position=position_dodge(width=1.05), size = 3.5, show.legend = FALSE) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all\nsamples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + xlab("Host species") + scale_y_continuous(oob=scales::discard) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank(), strip.text.x = element_text(face = "italic"))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kaiju_median_max_min_reads_classified_per_error_allowance_by_host_species.svg", `2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kaiju_median_max_min_reads_classified_per_error_allowance_by_host_species`)

`2022-09-27_Chapter_2_pilot_rodent_only_most_recent_mus_only_Kaiju_median_max_min_reads_classified_per_error_allowance_by_host_species` <- ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = 56, colour = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), label = Label), position=position_dodge(width=1.05), size = 3.5, show.legend = FALSE) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all\nsamples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + xlab("Host species") + scale_y_continuous(oob=scales::discard) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank(), strip.text.x = element_text(face = "italic")) + theme(strip.text.x = element_text(size = 14)) + theme(text = element_text(size = 14))
ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = 56, colour = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), label = Label), position=position_dodge(width=1.05), size = 3.5, show.legend = FALSE) + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + ggtitle("Median percent reads classified by Kaiju using different number of allowed errors from all\nsamples by host species") + scale_fill_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + scale_color_manual(values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) + guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) + xlab("Host species") + scale_y_continuous(oob=scales::discard) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank(), strip.text.x = element_text(face = "italic")) + theme(strip.text.x = element_text(size = 14)) + theme(text = element_text(size = 14))


`2022.09.22_all_samples_read_counts_R1_each_QC_step` <- read.delim("~/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-09-22_all_samples_read_counts_R1_each_QC_step.txt")
View(`2022.09.22_all_samples_read_counts_R1_each_QC_step`)

`2022.09.22_all_samples_read_counts_R1_each_QC_step`[`2022.09.22_all_samples_read_counts_R1_each_QC_step`$Sample %notin% "Human_mock",]

`2022.09.22_all_samples_read_counts_R1_each_QC_step_non_human` <- `2022.09.22_all_samples_read_counts_R1_each_QC_step`[`2022.09.22_all_samples_read_counts_R1_each_QC_step`$Sample %notin% "Human_mock",]

`2022.09.22_all_samples_read_counts_R1_each_QC_step_non_human_melted` <- reshape2::melt(`2022.09.22_all_samples_read_counts_R1_each_QC_step_non_human`, id=c("Sample", "Raw.Read.Count", "Qced.read.count", "Qced.and.cleansed.read.count", "Percent.original.reads.in.subsampled.files", "Percent.Qced.and.cleansed.reads.in.subsampled.file", "Host.species"))

ggplot(data = `2022.09.22_all_samples_read_counts_R1_each_QC_step_non_human_melted`, aes(x = Host.species, y = ((Qced.and.cleansed.read.count / Raw.Read.Count)*100), colour = Host.species, fill = Host.species, shape = Host.species)) + geom_violin(colour = 'black', alpha = 0.85) + geom_boxplot(fill = "#F2F3F4", colour = 'black', width = 0.15, outlier.size = 2.5, show.legend = FALSE) + xlab("Host species") + ylab("Percent of reads in raw read file found in QCed and\nhost cleansed read file(%)") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes", size = 15)) + scale_fill_manual(name = "Host species", values = c("Mus musculus" = "#2d2d2d", "Acomys cahirinus" = "#1e90ff", "Acomys russatus" = "#ff8d1e"))

ggplot(data = `2022.09.22_all_samples_read_counts_R1_each_QC_step_non_human_melted`, aes(x = Host.species, y = Percent.Qced.and.cleansed.reads.in.subsampled.file, colour = Host.species, fill = Host.species, shape = Host.species)) + geom_violin(colour = 'black', alpha = 0.85) + geom_boxplot(fill = "#F2F3F4", colour = 'black', width = 0.15, outlier.size = 2.5, show.legend = FALSE) + xlab("Host species") + ylab("Percent of reads in QCed and host cleansed\nfile found in subsampled read file (%)") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes", size = 15)) + scale_fill_manual(name = "Host species", values = c("Mus musculus" = "#2d2d2d", "Acomys cahirinus" = "#1e90ff", "Acomys russatus" = "#ff8d1e"))

`2022-09-30_chapter_2_pilot_stuff_rodent_species_percent_qced_and_cleansed_reads_in_subsampled_file_by_host_species` <- ggplot(data = `2022.09.22_all_samples_read_counts_R1_each_QC_step_non_human_melted`, aes(x = Host.species, y = ((Qced.and.cleansed.read.count / Raw.Read.Count)*100), colour = Host.species, fill = Host.species, shape = Host.species)) + geom_violin(colour = 'black', alpha = 0.85) + geom_boxplot(fill = "#F2F3F4", colour = 'black', width = 0.15, outlier.size = 2.5, show.legend = FALSE) + xlab("") + ylab("Percent of reads in raw read file found in QCed and\nhost cleansed read file(%)") + theme_linedraw() + theme(axis.text.x = element_text(angle = 45, hjust = 1), text = element_text(family = "Tex Gyre Heroes", size = 12)) + scale_fill_manual(name = "Host species", values = c("Mus musculus" = "#2d2d2d", "Acomys cahirinus" = "#1e90ff", "Acomys russatus" = "#ff8d1e"))

`2022-09-30_chapter_2_pilot_stuff_rodent_species_percent_raw_reads_in_qced_and_cleansed_reads_file_by_host_species` <- ggplot(data = `2022.09.22_all_samples_read_counts_R1_each_QC_step_non_human_melted`, aes(x = Host.species, y = Percent.Qced.and.cleansed.reads.in.subsampled.file, colour = Host.species, fill = Host.species, shape = Host.species)) + geom_violin(colour = 'black', alpha = 0.85) + geom_boxplot(fill = "#F2F3F4", colour = 'black', width = 0.15, outlier.size = 2.5, show.legend = FALSE) + xlab("") + ylab("Percent of reads in QCed and host cleansed\nfile found in subsampled read file (%)") + theme_linedraw() + theme(axis.text.x = element_text(angle = 45, hjust = 1), text = element_text(family = "Tex Gyre Heroes", size = 12)) + scale_fill_manual(name = "Host species", values = c("Mus musculus" = "#2d2d2d", "Acomys cahirinus" = "#1e90ff", "Acomys russatus" = "#ff8d1e"))

`2022-09-30_chapter_2_pilot_stuff_rodent_species_percent_qced_and_cleansed_reads_in_subsampled_file_by_host_species` + `2022-09-30_chapter_2_pilot_stuff_rodent_species_percent_raw_reads_in_qced_and_cleansed_reads_file_by_host_species` + plot_annotation(tag_levels = "A") + plot_layout(guides = "collect")

`2022-09-30_combined_chapter_2_pilot_stuff_rodent_species_QC_metrics_percentages_raw_reads_in_QC_and_cleansed_and_percent_QC_and_cleansed_in_subsampled` <- `2022-09-30_chapter_2_pilot_stuff_rodent_species_percent_qced_and_cleansed_reads_in_subsampled_file_by_host_species` + `2022-09-30_chapter_2_pilot_stuff_rodent_species_percent_raw_reads_in_qced_and_cleansed_reads_file_by_host_species` + plot_annotation(tag_levels = "A") + plot_layout(guides = "collect")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-09-30_combined_chapter_2_pilot_stuff_rodent_species_QC_metrics_percentages_raw_reads_in_QC_and_cleansed_and_percent_QC_and_cleansed_in_subsampled.svg", `2022-09-30_combined_chapter_2_pilot_stuff_rodent_species_QC_metrics_percentages_raw_reads_in_QC_and_cleansed_and_percent_QC_and_cleansed_in_subsampled`)

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-09-27.RData")
toc()

2022-10-01
`2022.09.30_raw_read_classification_percentages_tools_compared_processed` <- read.csv("~/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022-09-30_raw_read_classification_percentages_tools_compared_processed.csv")
View(`2022.09.30_raw_read_classification_percentages_tools_compared_processed`)

`2022.09.30_pilot_raw_comparison` <- `2022.09.30_raw_read_classification_percentages_tools_compared_processed`
`2022.09.30_pilot_raw_comparison_melted` <- reshape2::melt(`2022.09.30_pilot_raw_comparison`, id = c("Sample", "Sample.1", "Host.species", "Metaphlan.estimated.reads.classified.from.raw", "Metaphlan.estimated.percent.reads.classified.from.raw", "Metaphlan.estimated.percent.reads.classified.from.processed", "Kaiju.20.errors.number.classified.reads.from.raw", "Kaiju.20.errors.percent.classified.reads.from.raw", "Kaiju.20.errors.percent.classified.reads.from.processed", "Kraken.10..confidence.number.classified.reads.from.raw", "Kraken.10..confidence.percent.classified.reads.from.raw", "Kraken.10..confidence.percent.classified.reads.from.processed"))

ggplot(data = `2022.09.30_pilot_raw_comparison_melted`, aes(x = Host.species, y =  Metaphlan.estimated.percent.reads.classified.from.raw, fill = Host.species, shape = Host.species, colour = "Raw")) + geom_jitter() + geom_jitter(data = `2022.09.30_pilot_raw_comparison_melted`, aes(x = Host.species, y =  Metaphlan.estimated.percent.reads.classified.from.processed, fill = Host.species, shape = Host.species, colour = "Processed")) + theme_linedraw()

ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[`2022.09.30_pilot_raw_comparison_melted`$Host.species %notin% "HS",], aes(x = Host.species, y =  Metaphlan.estimated.percent.reads.classified.from.raw, fill = "Raw", shape = Host.species, colour = Host.species)) + geom_boxplot() + geom_boxplot(data = `2022.09.30_pilot_raw_comparison_melted`[`2022.09.30_pilot_raw_comparison_melted`$Host.species %notin% "HS",], aes(x = Host.species, y =  Metaphlan.estimated.percent.reads.classified.from.processed, fill = "Processed", shape = Host.species, colour = Host.species)) + theme_linedraw()

`2022.09.30_pilot_raw_comparison_melted` <- reshape2::melt(`2022.09.30_pilot_raw_comparison`, id = c("Sample", "Sample.1", "Host.species"))

ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Metaphlan.estimated.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = Host.species, y = value, fill = variable, colour = variable, shape = Host.species)) + geom_boxplot() + theme_linedraw() + facet_wrap(~ Host.species)
ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Metaphlan.estimated.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value, fill = Host.species, colour = Host.species, shape = Host.species)) + geom_boxplot() + theme_linedraw()

ggboxplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Metaphlan.estimated.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], x = "Host.species", y = "value", fill = "variable", colour = "variable", add = "jitter", shape = "Host.species") + stat_compare_means() + theme_linedraw() + facet_wrap(~ Host.species)
ggbetweenstats(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Metaphlan.estimated.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], x = variable, y = value, fill = Host.species, colour = Host.species, shape = Host.species) + theme_linedraw() + facet_grid(~ Host.species)

grouped_ggbetweenstats(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Metaphlan.estimated.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], x = variable, y = value, fill = variable, grouping.var = Host.species, colour = variable, shape = variable) + theme_linedraw()

ggbetweenstats(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Metaphlan.estimated.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species == "AC",], x = variable, y = value, fill = variable, colour = variable, shape = variable) + theme_linedraw()

ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Metaphlan.estimated.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value, fill = Host.species)) + geom_boxplot(colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Estimated percent reads classified (%)") + ggtitle("Estimated percent reads classified by Metaphlan, raw and processed files") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e"))

ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Kaiju.20.errors.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value, fill = Host.species)) + geom_boxplot(colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Percent reads classified by Kaiju, 20 errors allowed,\nraw and processed files") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e")) + facet_wrap(~ Host.species)

ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Kraken.10..confidence.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value, fill = Host.species)) + geom_boxplot(colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Percent reads classified by Kraken, 10% minimum confidence score\nfor classification, raw and processed files") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e")) + facet_wrap(~ Host.species)

ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Kraken.10..confidence.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value, fill = Host.species)) + geom_boxplot(colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Percent reads classified by Kraken, 10% minimum confidence score\nfor classification, raw and processed files") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e")) + facet_wrap(~ Host.species)

`kraken_fig` <- ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Kraken.10..confidence.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host.species), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Percent reads classified by Kraken, 10% minimum confidence score\nfor classification, raw and processed files") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e")) + stat_compare_means(label = 'p.signif') + facet_wrap(~ Host.species)


grouped_ggbetweenstats(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Metaphlan.estimated.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], x = variable, y = value, fill = variable, grouping.var = Host.species, colour = "#0f0f0f") + theme_linedraw() + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Percent reads classified by Kraken, 10% minimum confidence score\nfor classification, raw and processed files") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e"))


2022-10-02
`metaphlan_fig` <- ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Metaphlan.estimated.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host.species), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Estimated percent reads classified (%)") + ggtitle("Estimated percent reads classified by Metaphlan, raw and processed files") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e")) + stat_compare_means(method = "wilcox.test", label.y = 25.5) + facet_wrap(~ Host.species)

`kaiju_fig` <- ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Kaiju.20.errors.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host.species), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Percent reads classified by Kaiju, 20 errors allowed,\nraw and processed files") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e")) + stat_compare_means(method = "wilcox.test", label.y = 60) + facet_wrap(~ Host.species)

`kraken_fig` <- ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Kraken.10..confidence.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host.species), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Percent reads classified by Kraken, 10% minimum confidence score\nfor classification, raw and processed files") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e")) + stat_compare_means(method = "wilcox.test", label.y = 25.5) + facet_wrap(~ Host.species)


`metaphlan_fig` <- ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Metaphlan.estimated.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host.species), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Estimated percent reads classified (%)") + ggtitle("Metaphlan") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e")) + stat_compare_means(method = "wilcox.test", label.y = 25.5, label = 'p.signif') + facet_wrap(~ Host.species) + theme(text = element_text(family = "Tex Gyre Heroes"))

`kaiju_fig` <- ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Kaiju.20.errors.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host.species), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Kaiju, 20 errors allowed") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e")) + stat_compare_means(method = "wilcox.test", label.y = 60, label = 'p.signif') + facet_wrap(~ Host.species) + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), text = element_text(family = "Tex Gyre Heroes"))

`kraken_fig` <- ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Kraken.10..confidence.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host.species), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Kraken, 10% minimum confidence score\nrequired") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e")) + stat_compare_means(method = "wilcox.test", label.y = 25.5, label = 'p.signif') + facet_wrap(~ Host.species) + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), text = element_text(family = "Tex Gyre Heroes"))

`metaphlan_fig` + `kaiju_fig` / `kraken_fig` + plot_layout(guides = 'collect') + plot_annotation(title = 'Comparison of raw and processed read file percent classification by three classifiers for all rodent samples', tag_levels = 'A')
`kaiju_fig` + `metaphlan_fig` / `kraken_fig` + plot_layout(guides = 'collect') + plot_annotation(title = 'Comparison of raw and processed read file percent classification by three classifiers for all rodent samples', tag_levels = 'A')


`2022_10_02_pilot_stuff_chapter_1_rodent_samples_raw_versus_most_generous_parameters_combined_figure_metaphlan_kaiju_kraken_wilcoxon_test_for_sig` <- `metaphlan_fig` + `kaiju_fig` / `kraken_fig` + plot_layout(guides = 'collect') + plot_annotation(title = 'Comparison of raw and processed read file percent classification by three classifiers for all rodent samples', tag_levels = 'A')
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2022_10_02_pilot_stuff_chapter_1_rodent_samples_raw_versus_most_generous_parameters_combined_figure_metaphlan_kaiju_kraken_wilcoxon_test_for_sig.svg", `2022_10_02_pilot_stuff_chapter_1_rodent_samples_raw_versus_most_generous_parameters_combined_figure_metaphlan_kaiju_kraken_wilcoxon_test_for_sig`)

ggplot(data = `2022.09.30_pilot_raw_comparison_melted`[grepl("Kaiju.20.errors.percent*", `2022.09.30_pilot_raw_comparison_melted`$variable) & `2022.09.30_pilot_raw_comparison_melted`$Host.species != "HS",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host.species), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Kaiju, 20 errors allowed") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species", values = c("MM" = "#2d2d2d", "AC" = "#1e90ff", "AR" = "#ff8d1e")) + stat_compare_means(method = "wilcox.test", label.y = 60) + facet_wrap(~ Host.species) + theme(axis.title.x = element_blank(), axis.title.y = element_blank(), text = element_text(family = "Tex Gyre Heroes"))

view(`2022.09.30_pilot_raw_comparison_melted`[`2022.09.30_pilot_raw_comparison_melted`$Host.species == "HS",])

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-10-02.RData")
toc()

2022-10-03
ggplot(data = Chapter_1.Sorted_all_level_byte_usage, aes(y = Size)) + geom_boxplot()

2022-10-09
`2022_10_09_read_counts_all_samples_each_processing_steps` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_10_09_read_counts_all_samples_each_processing_steps.txt")
View(`2022_10_09_read_counts_all_samples_each_processing_steps`)

`2022_10_09_chapter_3_all_samples_percent_qced_and_cleansed_reads_of_raw_reads` <- ggplot(data = `2022_10_09_read_counts_all_samples_each_processing_steps`, aes(x = Host, fill = Host)) + geom_violin(alpha = 0.95, aes(fill = `2022_10_09_read_counts_all_samples_each_processing_steps`$Host, x = Host, y = (`2022_10_09_read_counts_all_samples_each_processing_steps`$Cleansed_reads / `2022_10_09_read_counts_all_samples_each_processing_steps`$Raw_reads)*100)) + geom_boxplot(aes(y = (`2022_10_09_read_counts_all_samples_each_processing_steps`$Cleansed_reads / `2022_10_09_read_counts_all_samples_each_processing_steps`$Raw_reads)*100), fill = "#f2f3f4", width = 0.15) + scale_fill_manual(name = "Host species", values = c("AC" = "#1e90ff", "AR" = "#ff8d1e")) + ylab("Percent reads (%)") + theme_linedraw() + theme(axis.title.x = element_blank(), text = element_text(family = "Tex Gyre Heroes"))

`2022_10_09_chapter_3_all_samples_percent_subsampled_qced_and_cleansed_reads_of_raw_reads` <- ggplot(data = `2022_10_09_read_counts_all_samples_each_processing_steps`, aes(x = Host, fill = Host)) + geom_violin(alpha = 0.95, aes(fill = `2022_10_09_read_counts_all_samples_each_processing_steps`$Host, x = Host, y = (`2022_10_09_read_counts_all_samples_each_processing_steps`$Subsampled_reads / `2022_10_09_read_counts_all_samples_each_processing_steps`$Raw_reads)*100)) + geom_boxplot(aes(y = (`2022_10_09_read_counts_all_samples_each_processing_steps`$Subsampled_reads / `2022_10_09_read_counts_all_samples_each_processing_steps`$Raw_reads)*100), fill = "#f2f3f4", width = 0.15) + scale_fill_manual(name = "Host species", values = c("AC" = "#1e90ff", "AR" = "#ff8d1e")) + ylab("Percent reads (%)") + theme_linedraw() + theme(axis.title.x = element_blank(),text = element_text(family = "Tex Gyre Heroes"))

`2022_10_09_chapter_3_all_samples_percent_qced_and_cleansed_reads_of_raw_reads` + `2022_10_09_chapter_3_all_samples_percent_subsampled_qced_and_cleansed_reads_of_raw_reads` + plot_layout(guides = 'collect') + plot_annotation(title = 'Percentage of reads from raw read files after processing steps', tag_levels = 'A')

`2022_10_09_chapter_3_all_samples_percent_reads_processing_steps_of_raw_reads_combined_figure` <- `2022_10_09_chapter_3_all_samples_percent_qced_and_cleansed_reads_of_raw_reads` + `2022_10_09_chapter_3_all_samples_percent_subsampled_qced_and_cleansed_reads_of_raw_reads` + plot_layout(guides = 'collect') + plot_annotation(title = 'Percentage of reads from raw read files after processing steps', tag_levels = 'A')
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_10_09_chapter_3_all_samples_percent_reads_processing_steps_of_raw_reads_combined_figure.svg", `2022_10_09_chapter_3_all_samples_percent_reads_processing_steps_of_raw_reads_combined_figure`)


2022-10-10
`2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure` <- ggplot(data = `2022-0-23_AC_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AC.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AC seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#1e90ff", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#004385", min.segment.length = 0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure.svg", `2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure`)

`2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure` <- ggplot(data = `AC_exclusive_only_novogene_paired_melted_good_tpms`[`AC_exclusive_only_novogene_paired_melted_good_tpms`$variable %in% `2022-0-23_AC_seasonal_raw_pvalue_SORTED`[`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$log2.add.one.AC.seasonal.fold.change > 1 & -log10(`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05) | `2022-0-23_AC_seasonal_raw_pvalue_SORTED`$log2.add.one.AC.seasonal.fold.change < -1 & -log10(`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05),]$Bin, ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#004385")) + scale_colour_manual(values = c ("#1e90ff", "#004385")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("RPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure.svg", `2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure`)

`2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure` <- ggplot(data = `2022-0-23_AR_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AR.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AR seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#ff8d1e", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#9e4e00", min.segment.length = 0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure.svg", `2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure`)

`2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure` <- ggplot(data = `AR_exclusive_only_novogene_paired_melted_good_tpms`[`AR_exclusive_only_novogene_paired_melted_good_tpms`$variable %in% `2022-0-23_AR_seasonal_raw_pvalue_SORTED`[`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$log2.add.one.AR.seasonal.fold.change > 1 & -log10(`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05) | `2022-0-23_AR_seasonal_raw_pvalue_SORTED`$log2.add.one.AR.seasonal.fold.change < -1 & -log10(`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05),]$Bin, ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#ff8d1e", "#522800")) + scale_colour_manual(values = c ("#ff8d1e", "#522800")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("RPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure.svg", `2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure`)

`AC_exclusive_only_novogene_paired_melted_good_tpms`[`AC_exclusive_only_novogene_paired_melted_good_tpms`$variable %in% `2022-0-23_AC_seasonal_raw_pvalue_SORTED`[`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$log2.add.one.AC.seasonal.fold.change > 1 & -log10(`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05) | `2022-0-23_AC_seasonal_raw_pvalue_SORTED`$log2.add.one.AC.seasonal.fold.change < -1 & -log10(`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05),]$Bin, ]

`2022_10_10_within_AC_seasonal_change_bins_only_novogene_paired_samples` <- `2022-0-23_AC_seasonal_raw_pvalue_SORTED`[`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues < 0.05,]
write.table(`2022-0-23_AC_seasonal_raw_pvalue_SORTED`[`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues < 0.05,], file = "2022_10_10_within_AC_seasonal_change_bins_only_novogene_paired_samples.tsv", sep="\t", row.names=FALSE)

`2022_10_10_within_AR_seasonal_change_bins_only_novogene_paired_samples` <- `2022-0-23_AR_seasonal_raw_pvalue_SORTED`[`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues < 0.05,]
write.table(`2022-0-23_AR_seasonal_raw_pvalue_SORTED`[`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues < 0.05,], file = "2022_10_10_within_AR_seasonal_change_bins_only_novogene_paired_samples.tsv", sep="\t", row.names=FALSE)

`2022-09-21_only_novogene_paired_strict_bin_mapping_salmon_GC_and_seq_bias_flagged_bin_level_host_effect_foldchange_and_qvalue_fig` <- ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue), colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

`2022-09-21_only_novogene_paired_strict_bin_mapping_salmon_GC_and_seq_bias_flagged_bin_level_month_effect_foldchange_and_qvalue_fig` <- ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = June.to.November.log2fold.add.one.change, y = -log10(Season.corrected.pvalue), colour = ifelse((June.to.November.log2fold.add.one.change > 1 & -log10(Season.corrected.pvalue) > -log10(0.05)) | (June.to.November.log2fold.add.one.change < -1 & -log10(Season.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Time effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#b5b5b3", "#0f0f0f")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & (`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -1),]

`2022_10_10_host_significant_difference_and_greater_1_less_neg_1_fold_change_bins_only_novogene_paired_samples` <- `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & (`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -1),]
write.table(`2022_10_10_host_significant_difference_and_greater_1_less_neg_1_fold_change_bins_only_novogene_paired_samples`, file = "2022_10_10_host_significant_difference_and_greater_1_less_neg_1_fold_change_bins_only_novogene_paired_samples.tsv", sep="\t", row.names=FALSE)

2022-10-11
`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`

write.table(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, file = "2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change.tsv", sep="\t", row.names=FALSE)

`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field.tsv")
View(`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`)

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue), colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + theme(text = element_text(size = 14)) + theme(legend.position = "none") + geom_label_repel(aes(label = ifelse((AC.to.AR.log2fold.add.one.change > 2.5 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -2.5 & -log10(Host.corrected.pvalue) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#0f0f0f", min.segment.length = 0)

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue), colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + theme(text = element_text(size = 14)) + theme(legend.position = "none") + geom_label_repel(aes(label = ifelse((AC.to.AR.log2fold.add.one.change > 2.75 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -2.75 & -log10(Host.corrected.pvalue) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#0f0f0f", min.segment.length = 0, size = 2.75, max.overlaps = Inf) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`, aes(x = -log10(Host.corrected.pvalue), y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_jitter() + geom_hline(yintercept=c(-1, 1), col="#c80815") + geom_vline(xintercept=(-log10(0.05)), col="#c80815") + facet_grid(~ GTDB.TK.family) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05, ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_jitter() + geom_hline(yintercept=c(-1, 1), col="#c80815") + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + facet_wrap(~ GTDB.TK.family) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05, ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'orange', 'blue', 'red'))) + geom_jitter() + geom_hline(yintercept=c(-1, 1), col="#c80815") + scale_colour_manual(values = c("#ff8d1e", "#1e90ff", "#b5b5b3")) + facet_wrap(~ GTDB.TK.family) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05, ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse(AC.to.AR.log2fold.add.one.change > 1, AC.to.AR.log2fold.add.one.change < -1, 'orange', 'blue', 'red')) + geom_jitter() + geom_hline(yintercept=c(-1, 1), col="#c80815") + scale_colour_manual(values = c("#ff8d1e", "#1e90ff", "#b5b5b3")) + facet_wrap(~ GTDB.TK.family) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())

2022-10-12
`2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure` <- ggplot(data = `2022-0-23_AC_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AC.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AC seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#1e90ff", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#004385", min.segment.length = 0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

`2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure` <- ggplot(data = `AC_exclusive_only_novogene_paired_melted_good_tpms`[`AC_exclusive_only_novogene_paired_melted_good_tpms`$variable %in% `2022-0-23_AC_seasonal_raw_pvalue_SORTED`[`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$log2.add.one.AC.seasonal.fold.change > 1 & -log10(`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05) | `2022-0-23_AC_seasonal_raw_pvalue_SORTED`$log2.add.one.AC.seasonal.fold.change < -1 & -log10(`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05),]$Bin, ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#004385")) + scale_colour_manual(values = c ("#1e90ff", "#004385")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + labs(fill = "Species and month", shape = "Species and month") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("RPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))

`2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure` <- ggplot(data = `2022-0-23_AR_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AR.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AR seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#ff8d1e", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#9e4e00", min.segment.length = 0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

`2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_signifcant_bins_box_and_violin_figure` <- ggplot(data = `AR_exclusive_only_novogene_paired_melted_good_tpms`[`AR_exclusive_only_novogene_paired_melted_good_tpms`$variable %in% `2022-0-23_AR_seasonal_raw_pvalue_SORTED`[`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$log2.add.one.AR.seasonal.fold.change > 1 & -log10(`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05) | `2022-0-23_AR_seasonal_raw_pvalue_SORTED`$log2.add.one.AR.seasonal.fold.change < -1 & -log10(`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05),]$Bin, ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#ff8d1e", "#522800")) + scale_colour_manual(values = c ("#ff8d1e", "#522800")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + labs(fill = "Species and month", shape = "Species and month") + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("RPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) + theme(text = element_text(size = 15))

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05, ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_jitter() + geom_hline(yintercept=c(-1, 1), col="#c80815") + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + facet_wrap(~ GTDB.TK.family) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05 & is.finite(`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$AC.to.AR.log2fold.add.one.change), ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_jitter() + geom_hline(yintercept=c(-1, 1), col="#c80815") + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + facet_wrap(~ GTDB.TK.family) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05 & is.finite(`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$AC.to.AR.log2fold.add.one.change), ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_jitter() + geom_hline(yintercept=c(-1), col="#1e90ff") + geom_hline(yintercept=c(1), col="#ff8d1e") + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + facet_wrap(~ GTDB.TK.family) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())

geom_rect(aes(ymin = 1, ymax = Inf, xmin = -Inf, xmax = Inf ), colour = "white", fill = "#ff8d1e", alpha = 0.35)
geom_rect(aes(ymin = -1, ymax = -Inf, xmin = -Inf, xmax = Inf ), colour = "white", fill = "#1e90ff", alpha = 0.35)

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05 & is.finite(`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$AC.to.AR.log2fold.add.one.change), ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_jitter() + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + facet_wrap(~ GTDB.TK.family) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + annotate("rect", ymin = -1, ymax = -Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#1e90ff", alpha = 0.25) + annotate("rect", ymin = 1, ymax = Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#ff8d1e", alpha = 0.25) + geom_hline(yintercept=c(-1), col="#1e90ff") + geom_hline(yintercept=c(1), col="#ff8d1e")

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05 & is.finite(`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$AC.to.AR.log2fold.add.one.change), ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + ylab("Log2()+1 fold change of geometric means per bin") + facet_wrap(~ GTDB.TK.family) + theme_linedraw() + ggtitle("Log2()+1 fold change of significantly different bins by GTDB-TK assigned family") + theme(text = element_text(family = "Tex Gyre Heroes"), strip.text = element_text(face = "italic")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + annotate("rect", ymin = -1, ymax = -Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#1e90ff", alpha = 0.25) + annotate("rect", ymin = 1, ymax = Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#ff8d1e", alpha = 0.25) + geom_hline(yintercept=c(-1), col="#1e90ff") + geom_hline(yintercept=c(1), col="#ff8d1e") + geom_jitter()

ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05 & is.finite(`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$AC.to.AR.log2fold.add.one.change), ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + ylab("Log2()+1 fold change of geometric means per bin") + facet_wrap(~ GTDB.TK.family, scales = "free_x") + theme_linedraw() + ggtitle("Log2()+1 fold change of significantly different bins by GTDB-TK assigned family") + theme(text = element_text(family = "Tex Gyre Heroes"), strip.text = element_text(face = "italic")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + annotate("rect", ymin = -1, ymax = -Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#1e90ff", alpha = 0.25) + annotate("rect", ymin = 1, ymax = Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#ff8d1e", alpha = 0.25) + geom_hline(yintercept=c(-1), col="#1e90ff") + geom_hline(yintercept=c(1), col="#ff8d1e") + geom_jitter()

`2022_10_12_subsampled_processed_only_paired_novogene_reads_mapping_pvalue_significant_strict_bins_log2_plus_one_fold_change_figure` <- ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05 & is.finite(`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$AC.to.AR.log2fold.add.one.change), ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + ylab("Log2()+1 fold change of geometric means per bin") + facet_wrap(~ GTDB.TK.family, scales = "free_x") + theme_linedraw() + ggtitle("Log2()+1 fold change of significantly different bins by GTDB-TK assigned family") + theme(text = element_text(family = "Tex Gyre Heroes"), strip.text = element_text(face = "italic")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + annotate("rect", ymin = -1, ymax = -Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#1e90ff", alpha = 0.25) + annotate("rect", ymin = 1, ymax = Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#ff8d1e", alpha = 0.25) + geom_hline(yintercept=c(-1), col="#1e90ff") + geom_hline(yintercept=c(1), col="#ff8d1e") + geom_jitter()
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_10_12_subsampled_processed_only_paired_novogene_reads_mapping_pvalue_significant_strict_bins_log2_plus_one_fold_change_figure.svg", `2022_10_12_subsampled_processed_only_paired_novogene_reads_mapping_pvalue_significant_strict_bins_log2_plus_one_fold_change_figure`)


ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05 & is.finite(`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$AC.to.AR.log2fold.add.one.change), ], aes(x = value, y = ))

ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -2,]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma) + theme(text = element_text(size = 15))

ggplot(data = `AR_exclusive_only_novogene_paired_melted_good_tpms`[`AR_exclusive_only_novogene_paired_melted_good_tpms`$variable %in% `2022-0-23_AR_seasonal_raw_pvalue_SORTED`[`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$log2.add.one.AR.seasonal.fold.change > 1 & -log10(`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05) | `2022-0-23_AR_seasonal_raw_pvalue_SORTED`$log2.add.one.AR.seasonal.fold.change < -1 & -log10(`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues) > -log10(0.05),]$Bin, ], aes(x = value, y = Species_x_season, fill = Species_x_season, shape = Species_x_season)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#ff8d1e", "#522800")) + scale_colour_manual(values = c ("#ff8d1e", "#522800")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + labs(fill = "Species and month", shape = "Species and month") + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("RPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) + theme(text = element_text(size = 15))

2022-10-13

ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -2,]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) + theme(text = element_text(size = 15))

ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -2,]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) + theme(text = element_text(size = 15))

ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -2,]$Bin & grepl(glob2rx("Bin_c*"), `only_novogene_paired_melted_good_tpms`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) + theme(text = element_text(size = 15))

ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -2.85,]$Bin & grepl(glob2rx("Bin_c*"), `only_novogene_paired_melted_good_tpms`$variable), ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) + theme(text = element_text(size = 15))

ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -2.84 & is.finite(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change),]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) + theme(text = element_text(size = 15))

ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 2.6 & is.finite(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change),]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) + theme(text = element_text(size = 15))

`2022_10_13_ten_most_AC_enriched_bins_all_tpms_plots_box_and_violin` <- ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -2.84 & is.finite(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change),]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("RPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) + theme(text = element_text(size = 15))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_10_13_ten_most_AC_enriched_bins_all_tpms_plots_box_and_violin.svg", `2022_10_13_ten_most_AC_enriched_bins_all_tpms_plots_box_and_violin`)

`2022_10_13_ten_most_AR_enriched_bins_all_tpms_plots_box_and_violin` <- ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 2.6 & is.finite(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change),]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("RPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) + theme(text = element_text(size = 15))
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_10_13_ten_most_AR_enriched_bins_all_tpms_plots_box_and_violin.svg", `2022_10_13_ten_most_AR_enriched_bins_all_tpms_plots_box_and_violin`)

2022-10-16
checkm_bin_plot1 + checkm_bin_plot2 + checkm_bin_plot3 + plot_layout(guides = "keep") + plot_annotation(tag_levels = 'A')
`2022-10-17_checkm_all_bins_figure_no_barrnap` <- checkm_bin_plot1 + checkm_bin_plot2 + checkm_bin_plot3 + plot_layout(guides = "keep") + plot_annotation(tag_levels = 'A')

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-10-17_checkm_all_bins_figure_no_barrnap.svg", `2022-10-17_checkm_all_bins_figure_no_barrnap`)

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < 0 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & is.finite(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change),]$Bin

`2022.10.16_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts` <- read.delim2("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-10-16_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts.tsv")
View(`2022.10.16_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`)

`2022.10.16_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-10-16_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts.txt")
View(`2022.10.16_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`)

`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts` <- reshape2::melt(`2022.10.16_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`, id=c("Product", "Bin_status"))
view(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`)

ggplot(data = `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Product %in% `2022-0-23_AC_seasonal_raw_pvalue_SORTED`[`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues < 0.05,]$Bin,], aes(x = Product, y = value, fill = Product)) + geom_boxplot() + scale_y_log10() + theme_linedraw()
ggplot(data = `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Product %in% `2022-0-23_AR_seasonal_raw_pvalue_SORTED`[`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues < 0.05,]$Bin,], aes(x = Product, y = value)) + geom_boxplot()

ggplot(data = `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`, aes(x = Bin_status, y = value, fill = Bin_status, colour = Bin_status)) + geom_boxplot() + theme_linedraw()

X60.kDa.chaperonin.2

ggboxplot(data = `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "hypothetical.protein",], x = "Bin_status", y = "value")
ggbetweenstats(data = `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "hypothetical.protein",], x = Bin_status, y = value)

group_by(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "hypothetical.protein",], Bin_status) %>% dplyr::summarise(mean = mean(value, na.rm = TRUE), sd = sd(value, na.rm = TRUE), median = median(value, na.rm = TRUE), IQR = IQR(value, na.rm = TRUE), count = n())

`2022.10.16.Unique_product_names` <- unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable)
sample(`2022.10.16.Unique_product_names`, 8)

group_by(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "putative.TspO.MBR.related.protein.precursor",], Bin_status) %>% dplyr::summarise(mean = mean(value, na.rm = TRUE), sd = sd(value, na.rm = TRUE), median = median(value, na.rm = TRUE), IQR = IQR(value, na.rm = TRUE), count = n())
ggbetweenstats(data = `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "UPF0365.protein.YqfA",], x = Bin_status, y = value)

write.table(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`, file = "2022_10_16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts.tsv", sep="\t", row.names=FALSE)

strict_bins_list <- as.list(unique(only_novogene_paired_melted_good_tpms$variable))
`2022.10.16.Unique_product_names_LIST` <- as.list(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable))

`test_list` <- as.list(sample(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable), 50))

exp(mean(log(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "UPF0122.protein.YlxM" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None",]$value)))
exp(mean(log(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "hypothetical.protein" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None",]$value)))

tic()
`test_products_dataframe_table` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Product.name", "Shapiro.Wilk.Wvalue", "Shapiro.Wilk.Pvalue", "Levene.Fvalue", "Levene.Pvalue"))))
for (product in `2022.10.16.Unique_product_names`){
  product_name <- paste(product)
  shap.pvalue <- shapiro.test(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product_name,]$value)$p.value
  shap.Wvalue <- shapiro.test(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product_name,]$value)$statistic
  lave.pvalue <- leveneTest(value ~ Bin_status, `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product_name,])[,3]
  lave.Fvalue <- leveneTest(value ~ Bin_status, `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product_name,])[,2]
  `test_products_dataframe_table`[nrow(`test_products_dataframe_table`) +1,] <- c(product_name, shap.Wvalue, shap.pvalue, lave.Fvalue, lave.pvalue)
  `test_products_dataframe_table`[,c(2:5)] <- sapply(`test_products_dataframe_table`[,c(2:5)], as.numeric)
  rm(product_name)
  rm(`shap.pvalue`)
  rm(`lave.pvalue`)
  rm(`shap.Wvalue`)
  rm(`lave.Fvalue`)
}
view(`test_products_dataframe_table`)
`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status` <- `test_products_dataframe_table`
write.table(`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status`, file = "2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status.tsv", sep="\t", row.names=FALSE)
`2022_10_17-levene_test_pvalue_significant_all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status` <- `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status`[`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status`$Levene.pvalue <= 0.05,]
write.table(`2022_10_17-levene_test_pvalue_significant_all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status`, file = "2022_10_17-levene_test_pvalue_significant_all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status.tsv", sep="\t", row.names=FALSE)
toc()

tic()
`test_products_dataframe_table` <- data.frame(matrix(ncol = 10, nrow = 0, dimnames = list(NULL, c("Product.name", "Shapiro.Wilk.Wvalue", "Shapiro.Wilk.Pvalue", "Levene.Fvalue", "Levene.Pvalue", "Non-enriched.bin.geometric.mean.dfast.detections", "AC.enriched.bin.geometric.mean.dfast.detections", "AR.enriched.bin.geometric.mean.dfast.detections", "Within.AC.seasonally.different.bin.geometric.mean.dfast.detections", "Within.AR.seasonally.different.bin.geometric.mean.dfast.detections"))))
for (product in `2022.10.16.Unique_product_names`){
  product_name <- paste(product)
  shap_test <- shapiro.test(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == product_name,]$value)
  shap.pvalue <- shap_test$p.value
  shap.Wvalue <- shap_test$statistic
  lave_test <- leveneTest(value ~ Bin_status, `2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == product_name,])
  lave.pvalue <- lave_test[1,3]
  lave.Fvalue <- lave_test[1,2]
  noise.bins.geom.mean <- exp(mean(log(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == product_name & `2022_10_16_working_melted_dfast`$Bin_status == "None", ]$value)))
  AC.enriched.bins.geom.mean <- exp(mean(log(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == product_name & `2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs", ]$value)))
  AR.enriched.bins.geom.mean <- exp(mean(log(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == product_name & `2022_10_16_working_melted_dfast`$Bin_status == "AR_enriched_vs", ]$value)))
  within.AC.bins.geom.mean <- exp(mean(log(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == product_name & `2022_10_16_working_melted_dfast`$Bin_status == "in_AC_effect", ]$value)))
  within.AR.bins.geom.mean <- exp(mean(log(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == product_name & `2022_10_16_working_melted_dfast`$Bin_status == "in_AR_effect", ]$value)))
  `test_products_dataframe_table`[nrow(`test_products_dataframe_table`) +1,] <- c(product_name, shap.Wvalue, shap.pvalue, lave.Fvalue, lave.pvalue, noise.bins.geom.mean, AC.enriched.bins.geom.mean, AR.enriched.bins.geom.mean, within.AC.bins.geom.mean, within.AR.bins.geom.mean)
  `test_products_dataframe_table`[,c(2:10)] <- sapply(`test_products_dataframe_table`[,c(2:10)], as.numeric)
  rm(`product_name`)
  rm(`shap.pvalue`)
  rm(`lave.pvalue`)
  rm(`shap.Wvalue`)
  rm(`lave.Fvalue`)
  rm(`noise.bins.geom.mean`)
  rm(`AC.enriched.bins.geom.mean`)
  rm(`AR.enriched.bins.geom.mean`)
  rm(`within.AC.bins.geom.mean`)
  rm(`within.AR.bins.geom.mean`)
}
view(`test_products_dataframe_table`)
`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status` <- `test_products_dataframe_table`
write.table(`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status`, file = "2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status.tsv", sep="\t", row.names=FALSE)
`2022_10_17-levene_test_pvalue_significant_all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status` <- `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status`[`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status`$Levene.pvalue <= 0.05,]
write.table(`2022_10_17-levene_test_pvalue_significant_all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status`, file = "2022_10_17-levene_test_pvalue_significant_all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status.tsv", sep="\t", row.names=FALSE)
`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Shapiro_pvalue_sorted` <- `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status`[order(`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status`$Shapiro.Wilk.Pvalue),]
`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Shapiro_pvalue_sorted`$BH.corrected.Shapiro.Wilk.Pvalue <- p.adjust(`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Shapiro_pvalue_sorted`$Shapiro.Wilk.Pvalue, method = "BH")
`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted` <- `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Shapiro_pvalue_sorted`[order(`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Shapiro_pvalue_sorted`$Levene.Pvalue),]
`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$BH.corrected.Levene.Pvalue <- p.adjust(`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$Levene.Pvalue, method = "BH")
write.table(`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`, file = "2022_10_18-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_and_corrected_pvalues_with_geometric_mean_counts.tsv", sep="\t", row.names=FALSE)
toc()

exp(mean(log(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == "ABC.transporter.arginine.binding.protein.1" & `2022_10_16_working_melted_dfast`$Bin_status == "None", ]$value)))

`2022_10_16_working_melted_dfast` <- `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`
`2022_10_16_working_melted_dfast`$Bin_status <- as.factor(`2022_10_16_working_melted_dfast`$Bin_status)

leveneTest(value ~ Bin_status, `2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == "ABC.transporter.arginine.binding.protein.1",])[1,3]

leveneTest(value ~ Bin_status, `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "ABC.transporter.arginine.binding.protein.1",])

as.list(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < 0 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & is.finite(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change),]$Bin)


`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == "aminodeoxychorismate.lyase",]$value
`2022-10-19_working_melted_dfast_pvalues` <- `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`

ggplot(data = `2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable%in% `2022-10-19_working_melted_dfast_pvalues`[`2022-10-19_working_melted_dfast_pvalues`$BH.corrected.Shapiro.Wilk.Pvalue < 0.05 & `2022-10-19_working_melted_dfast_pvalues`$BH.corrected.Levene.Pvalue < 0.05,]$Product.name,], aes(x = Bin_status, y = value, colour = Bin_status)) + geom_jitter() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

ggplot(data = `2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable%in% `2022-10-19_working_melted_dfast_pvalues`[`2022-10-19_working_melted_dfast_pvalues`$BH.corrected.Shapiro.Wilk.Pvalue < 0.05 & `2022-10-19_working_melted_dfast_pvalues`$BH.corrected.Levene.Pvalue < 0.05,]$Product.name & `2022_10_16_working_melted_dfast`$value > 0,], aes(x = Bin_status, y = value, colour = Bin_status)) + geom_jitter() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

ggplot(data = `2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable%in% `2022-10-19_working_melted_dfast_pvalues`[`2022-10-19_working_melted_dfast_pvalues`$BH.corrected.Shapiro.Wilk.Pvalue < 0.0000005 & `2022-10-19_working_melted_dfast_pvalues`$BH.corrected.Levene.Pvalue < 0.0000005,]$Product.name & `2022_10_16_working_melted_dfast`$value > 0,], aes(x = Bin_status, y = value, colour = Bin_status)) + geom_jitter() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()


ggplot(data = `only_novogene_paired_melted_good_tpms`[`only_novogene_paired_melted_good_tpms`$variable %in% `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -2,]$Bin, ], aes(x = value, y = Species, fill = Species, shape = Species)) + geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e")) + scale_colour_manual(values = c ("#1e90ff", "#ff8d1e")) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + facet_wrap(~variable, strip.position="top") + scale_y_discrete(expand = expansion(mult = c(0.01, .425))) + theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) + theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) + xlab("TPMs from mapping of\nsubsampled reads to concatenated Bins reference") + ylab("Host species") + geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) + geom_jitter(width = 0.25, alpha = 0.75) + scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) + theme(text = element_text(size = 15))

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-10-16.RData")
toc()

2022-10-22
`2022.10.22_all_samples_comparing_raw_taxonomic_classifications_to_processed_metaphlan_kaiju_20_error_kraken_10_percent_conf` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-10-22_all_samples_comparing_raw_taxonomic_classifications_to_processed_metaphlan_kaiju_20_error_kraken_10_percent_conf.txt")
View(`2022.10.22_all_samples_comparing_raw_taxonomic_classifications_to_processed_metaphlan_kaiju_20_error_kraken_10_percent_conf`)

`2022-10-22_working` <- `2022.10.22_all_samples_comparing_raw_taxonomic_classifications_to_processed_metaphlan_kaiju_20_error_kraken_10_percent_conf`

`2022-10-22_working_melted` <- reshape2::melt(`2022-10-22_working`, id=c("SampleID", "Host", "Season", "Host_X_season", "Sequencing"))

ggplot(data = `2022-10-22_working_melted`[`2022-10-22_working_melted`$variable == "Raw.Metaphlan.estimated.percent.reads.classified" | `2022-10-22_working_melted`$variable == "Processed.Metaphlan.estimated.percent.reads.classified",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host_X_season), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Estimated percent reads classified (%)") + ggtitle("Metaphlan estimated percent reads classified pre- and post-processing") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species and\nsample month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + stat_compare_means(method = "wilcox.test", label.y = 25.5, label = 'p.signif') + facet_wrap(~ Host_X_season) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022_10_22_all_samples_raw_versus_processed_percent_classified_metaphlan_fig` <- ggplot(data = `2022-10-22_working_melted`[`2022-10-22_working_melted`$variable == "Raw.Metaphlan.estimated.percent.reads.classified" | `2022-10-22_working_melted`$variable == "Processed.Metaphlan.estimated.percent.reads.classified",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host_X_season), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Estimated percent reads classified (%)") + ggtitle("Metaphlan estimated percent reads classified pre- and post-processing") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species and\nsample month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + stat_compare_means(method = "wilcox.test", label.y = 25.5, label = 'p.signif') + facet_wrap(~ Host_X_season) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022-10-22_working_melted`[`2022-10-22_working_melted`$variable == "Raw.Kraken.percent.reads.classified" | `2022-10-22_working_melted`$variable == "Processed.Kraken.percent.reads.classified",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host_X_season), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Kraken 2, 10% minimum required confidence score, percent reads classified\npre- and post-processing") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species and\nsample month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + stat_compare_means(method = "wilcox.test", label.y = 25.5, label = 'p.signif') + facet_wrap(~ Host_X_season) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022_10_22_all_samples_raw_versus_processed_percent_classified_kraken_fig` <- ggplot(data = `2022-10-22_working_melted`[`2022-10-22_working_melted`$variable == "Raw.Kraken.percent.reads.classified" | `2022-10-22_working_melted`$variable == "Processed.Kraken.percent.reads.classified",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host_X_season), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Kraken 2, 10% minimum required confidence score, percent reads classified\npre- and post-processing") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species and\nsample month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + stat_compare_means(method = "wilcox.test", label.y = 25.5, label = 'p.signif') + facet_wrap(~ Host_X_season) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022-10-22_working_melted`[`2022-10-22_working_melted`$variable == "Raw.Kaiju.percent.reads.classified" | `2022-10-22_working_melted`$variable == "Processed.Kaiju.percent.reads.classified",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host_X_season), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Kaiju, 20 errors allowed, percent reads classified pre- and post-processing") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species and\nsample month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + stat_compare_means(method = "wilcox.test", label.y = 25.5, label = 'p.signif') + facet_wrap(~ Host_X_season) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022_10_22_all_samples_raw_versus_processed_percent_classified_kaiju_fig` <- ggplot(data = `2022-10-22_working_melted`[`2022-10-22_working_melted`$variable == "Raw.Kaiju.percent.reads.classified" | `2022-10-22_working_melted`$variable == "Processed.Kaiju.percent.reads.classified",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host_X_season), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Percent reads classified (%)") + ggtitle("Kaiju, 20 errors allowed, percent reads classified pre- and post-processing") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species and\nsample month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + stat_compare_means(method = "wilcox.test", label.y = 25.5, label = 'p.signif') + facet_wrap(~ Host_X_season) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_10_22_all_samples_raw_versus_processed_percent_classified_metaphlan_fig.svg", `2022_10_22_all_samples_raw_versus_processed_percent_classified_metaphlan_fig`)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_10_22_all_samples_raw_versus_processed_percent_classified_kraken_fig.svg", `2022_10_22_all_samples_raw_versus_processed_percent_classified_kraken_fig`)
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_10_22_all_samples_raw_versus_processed_percent_classified_kaiju_fig.svg", `2022_10_22_all_samples_raw_versus_processed_percent_classified_kaiju_fig`)

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-10-22.RData")
toc()

2022-10-24
`2022_10_16_working_melted_dfast`
sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value)
sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value)

phyper(1, sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), 1, lower.tail = FALSE)

phyper(sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), 1, lower.tail = FALSE)

dhyper(sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value))

phyper(27, 27, 87, 114)
phyper(27, 27, 87, 114, lower.tail = FALSE)
phyper(10, 27, 87, 29)
1-(phyper(9, 27, 87, 29))

thingy <- 0:29
thingy_probs <- dhyper(thingy, 27, 87, 29)

qplot(thingy, weight = thingy_probs, geom = "histogram", bins = 29, col = I("#ff8d1e"))

rhyper(50, 27, 87, 29)

thingy <- 0:10
thingy_probs <- dhyper(thingy, sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$Bin_status != "None" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value))

thingy <- 0:10
thingy_probs <- dhyper(thingy, sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "hypothetical.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "hypothetical.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$Bin_status != "None" & `2022_10_16_working_melted_dfast`$variable == "hypothetical.protein", ]$value))

N.acetylmuramoyl.L.alanine.amidase
thingy <- 0:10
thingy_probs <- dhyper(thingy, sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "N.acetylmuramoyl.L.alanine.amidase", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "N.acetylmuramoyl.L.alanine.amidase", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$Bin_status != "None" & `2022_10_16_working_melted_dfast`$variable == "N.acetylmuramoyl.L.alanine.amidase", ]$value))

dhyper(10, sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value))

phyper((27-1), 58, (89-58), 29, lower.tail = FALSE)

phyper(sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value) - 1, sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "None" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value) - sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "None" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein", ]$value), sum(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status != "None" & `2022_10_16_working_melted_dfast`$variable == "gamma.carbonic.anhydrase.family.protein" & `2022_10_16_working_melted_dfast`$Bin_status != "AC_enriched_vs", ]$value), lower.tail = FALSE)

`2022_10_16_working_melted_dfast`$Bin_status
summarize(`2022_10_16_working_melted_dfast`$Bin_status)

str(unique(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "None",]$Product))
 chr [1:164] "Bin_c1005" "Bin_c1007" "Bin_c101" "Bin_c1023" "Bin_c1047" "Bin_c1052" "Bin_c1055" "Bin_c1081" "Bin_c1082" ...
str(unique(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AC_enriched_vs",]$Product))
 chr [1:107] "Bin_c1036" "Bin_c1043" "Bin_c1119" "Bin_c113" "Bin_c1148" "Bin_c1158" "Bin_c1180" "Bin_c1183" "Bin_c1194" ...
str(unique(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "AR_enriched_vs",]$Product))
 chr [1:65] "Bin_c1000" "Bin_c112" "Bin_c1188" "Bin_c1199" "Bin_c1325" "Bin_c1328" "Bin_c1372" "Bin_c1378" "Bin_c1401" ...
str(unique(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "in_AC_effect",]$Product))
 chr [1:9] "Bin_c1070" "Bin_c1111" "Bin_c1167" "Bin_c375" "Bin_c380" "Bin_m1135" "Bin_m1474" "Bin_m293" "Bin_m621"
str(unique(`2022_10_16_working_melted_dfast`[`2022_10_16_working_melted_dfast`$Bin_status == "in_AR_effect",]$Product))
 chr [1:3] "Bin_c1435" "Bin_c796" "Bin_m137"

phyper(q=(107-1), m=27, n=(114-27), 184, lower.tail = FALSE)

unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "gamma.carbonic.anhydrase.family.protein" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product)

str(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "gamma.carbonic.anhydrase.family.protein" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))
 chr [1:112] "Bin_c1007" "Bin_c1023" "Bin_c1036" "Bin_c1047" "Bin_c1052" "Bin_c1055" "Bin_c1081" "Bin_c1119" "Bin_c1138" ...

group_by(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "putative.TspO.MBR.related.protein.precursor",], Bin_status) %>% dplyr::summarise(mean = mean(value, na.rm = TRUE), sd = sd(value, na.rm = TRUE), median = median(value, na.rm = TRUE), IQR = IQR(value, na.rm = TRUE), count = n())

k <- 112
m <- 27
x <- 107
q <- x-1
N <- 114
n <- N-m

p.value <- phyper(q=x-1, m=m, n=n, k=k, lower.tail=FALSE)
p.values <- phyper(q=range(x)-1, m=m, n=n, k=k, lower.tail=FALSE)

plot(range(x), p.values, type="l", lwd=2, col="violet", main="Hypergeometric P-value", xlab="x = marked elements in the selection", ylab="P-value = P(X>=x)", ylim=c(0, 1), panel.first=grid())

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-10-22.RData")
toc()

2022-10-25

yubtub <- nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "gamma.carbonic.anhydrase.family.protein" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product)))
yubtub <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "gamma.carbonic.anhydrase.family.protein" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))

as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None",]$Product))))

as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "anion.permease" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))

tic()
`test_list` <- as.list(sample(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable), 500))

test_products_hypergeometric_pvalues_df <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Product", "Noise.bins.pvalue", "AC.enriched.versus.bins.pvalue", "AR.enriched.versus.bins.pvalue", "AC.enriched.within.bins.pvalue", "AR.enriched.within.bins.pvalue"))))
for (product in `test_list`){
  product.name <- paste(product)
  noise.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None",]$Product))))
  AC.enriched.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs",]$Product))))
  AR.enriched.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs",]$Product))))
  within.AC.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect",]$Product))))
  within.AR.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect",]$Product))))
  summed.noise.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.AC.enriched.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.AR.enriched.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.within.AC.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.within.AR.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.all.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  noise.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  AC.enriched.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  AR.enriched.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  within.AC.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  within.AR.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  total.number.bins.value.product.over.zero <- (noise.bins.product.values.over.zero + AC.enriched.bins.product.values.over.zero + AR.enriched.bins.product.values.over.zero + within.AC.bins.product.values.over.zero + within.AR.bins.product.values.over.zero)
  noise.bins.product.pvalue <- phyper(q = (noise.bins.count - 1), m = summed.noise.product.values, n = (summed.all.bins.product.values - summed.noise.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  AC.enriched.bins.product.pvalue <- phyper(q = (AC.enriched.bins.count - 1), m = summed.AC.enriched.product.values, n = (summed.all.bins.product.values - summed.AC.enriched.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  AR.enriched.bins.product.pvalue <- phyper(q = (AR.enriched.bins.count - 1), m = summed.AR.enriched.product.values, n = (summed.all.bins.product.values - summed.AR.enriched.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  within.AC.bins.product.pvalue <- phyper(q = (within.AC.bins.count - 1), m = summed.within.AC.bins.product.values, n = (summed.all.bins.product.values - summed.within.AC.bins.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  within.AR.bins.product.pvalue <- phyper(q = (within.AR.bins.count - 1), m = summed.within.AR.bins.product.values, n = (summed.all.bins.product.values - summed.within.AR.bins.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  `test_products_hypergeometric_pvalues_df`[nrow(`test_products_hypergeometric_pvalues_df`) +1,] <- c(product.name, noise.bins.product.pvalue, AC.enriched.bins.product.pvalue, AR.enriched.bins.product.pvalue, within.AC.bins.product.pvalue, within.AR.bins.product.pvalue)
  `test_products_hypergeometric_pvalues_df`[,c(2:6)] <- sapply(`test_products_hypergeometric_pvalues_df`[,c(2:6)], as.numeric)
  rm(`product.name`)
  rm(`noise.bins.count`)
  rm(`AC.enriched.bins.count`)
  rm(`AR.enriched.bins.count`)
  rm(`within.AC.bins.count`)
  rm(`within.AR.bins.count`)
  rm(`summed.noise.product.values`)
  rm(`summed.AC.enriched.product.values`)
  rm(`summed.AR.enriched.product.values`)
  rm(`summed.within.AC.bins.product.values`)
  rm(`summed.within.AR.bins.product.values`)
  rm(`summed.all.bins.product.values`)
  rm(`noise.bins.product.values.over.zero`)
  rm(`AC.enriched.bins.product.values.over.zero`)
  rm(`AR.enriched.bins.product.values.over.zero`)
  rm(`within.AC.bins.product.values.over.zero`)
  rm(`within.AR.bins.product.values.over.zero`)
  rm(`total.number.bins.value.product.over.zero`)
  rm(`noise.bins.product.pvalue `)
  rm(`AC.enriched.bins.product.pvalue`)
  rm(`AR.enriched.bins.product.pvalue`)
  rm(`within.AC.bins.product.pvalue`)
  rm(`within.AR.bins.product.pvalue`)
}
view(test_products_hypergeometric_pvalues_df)
toc()

multidrug.ABC.transporter.ATP.binding.protein
as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "multidrug.ABC.transporter.ATP.binding.protein" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))

sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "multidrug.ABC.transporter.ATP.binding.protein" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs",]$value)
sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "multidrug.ABC.transporter.ATP.binding.protein",]$value)

AC_enriched
k <- 215
m <- 253
x <- 107
q <- x-1
N <- 648
n <- N-m

p.value <- phyper(q=x-1, m=m, n=n, k=k, lower.tail=FALSE)
p.values <- phyper(q=(0:x)-1, m=m, n=n, k=k, lower.tail=FALSE)

plot(0:x, p.values, type="l", lwd=2, col="violet", main="Hypergeometric P-value", xlab="x = marked elements in the selection", ylab="P-value = P(X>=x)", ylim=c(0, 1), panel.first=grid())
plot(0:x, p.values,  type="l", lwd=2, col="violet", main="Hypergeometric P-value (log Y scale)", xlab="x = marked elements in the selection", ylab="P-value = P(X>=x); log scale", panel.first=grid(), log='y')

tic()
test_products_hypergeometric_pvalues_df <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Product", "Noise.bins.pvalue", "AC.enriched.versus.bins.pvalue", "AR.enriched.versus.bins.pvalue", "AC.enriched.within.bins.pvalue", "AR.enriched.within.bins.pvalue"))))
for (product in `2022.10.16.Unique_product_names`){
  product.name <- paste(product)
  noise.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None",]$Product))))
  AC.enriched.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs",]$Product))))
  AR.enriched.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs",]$Product))))
  within.AC.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect",]$Product))))
  within.AR.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect",]$Product))))
  summed.noise.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.AC.enriched.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.AR.enriched.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.within.AC.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.within.AR.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.all.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  noise.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  AC.enriched.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  AR.enriched.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  within.AC.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  within.AR.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  total.number.bins.value.product.over.zero <- (noise.bins.product.values.over.zero + AC.enriched.bins.product.values.over.zero + AR.enriched.bins.product.values.over.zero + within.AC.bins.product.values.over.zero + within.AR.bins.product.values.over.zero)
  noise.bins.product.pvalue <- phyper(q = (noise.bins.count - 1), m = summed.noise.product.values, n = (summed.all.bins.product.values - summed.noise.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  AC.enriched.bins.product.pvalue <- phyper(q = (AC.enriched.bins.count - 1), m = summed.AC.enriched.product.values, n = (summed.all.bins.product.values - summed.AC.enriched.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  AR.enriched.bins.product.pvalue <- phyper(q = (AR.enriched.bins.count - 1), m = summed.AR.enriched.product.values, n = (summed.all.bins.product.values - summed.AR.enriched.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  within.AC.bins.product.pvalue <- phyper(q = (within.AC.bins.count - 1), m = summed.within.AC.bins.product.values, n = (summed.all.bins.product.values - summed.within.AC.bins.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  within.AR.bins.product.pvalue <- phyper(q = (within.AR.bins.count - 1), m = summed.within.AR.bins.product.values, n = (summed.all.bins.product.values - summed.within.AR.bins.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  `test_products_hypergeometric_pvalues_df`[nrow(`test_products_hypergeometric_pvalues_df`) +1,] <- c(product.name, noise.bins.product.pvalue, AC.enriched.bins.product.pvalue, AR.enriched.bins.product.pvalue, within.AC.bins.product.pvalue, within.AR.bins.product.pvalue)
  `test_products_hypergeometric_pvalues_df`[,c(2:6)] <- sapply(`test_products_hypergeometric_pvalues_df`[,c(2:6)], as.numeric)
  rm(`product.name`)
  rm(`noise.bins.count`)
  rm(`AC.enriched.bins.count`)
  rm(`AR.enriched.bins.count`)
  rm(`within.AC.bins.count`)
  rm(`within.AR.bins.count`)
  rm(`summed.noise.product.values`)
  rm(`summed.AC.enriched.product.values`)
  rm(`summed.AR.enriched.product.values`)
  rm(`summed.within.AC.bins.product.values`)
  rm(`summed.within.AR.bins.product.values`)
  rm(`summed.all.bins.product.values`)
  rm(`noise.bins.product.values.over.zero`)
  rm(`AC.enriched.bins.product.values.over.zero`)
  rm(`AR.enriched.bins.product.values.over.zero`)
  rm(`within.AC.bins.product.values.over.zero`)
  rm(`within.AR.bins.product.values.over.zero`)
  rm(`total.number.bins.value.product.over.zero`)
  rm(`noise.bins.product.pvalue `)
  rm(`AC.enriched.bins.product.pvalue`)
  rm(`AR.enriched.bins.product.pvalue`)
  rm(`within.AC.bins.product.pvalue`)
  rm(`within.AR.bins.product.pvalue`)
}
view(test_products_hypergeometric_pvalues_df)
`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status` <- test_products_hypergeometric_pvalues_df
write.table(`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, file = "2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status.tsv", sep="\t", row.names=FALSE)
`2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status` <- reshape2::melt(`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, id=c("Product"))
write.table(`2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, file = "2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status.tsv", sep="\t", row.names=FALSE)
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-10-22.RData")
toc()

ggplot(data = `2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, aes(x = variable, y = value, colour = variable, shape = variable)) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`[`2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`$Product %in% `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`[`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$Non.enriched.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$AC.enriched.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$AR.enriched.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$Within.AC.seasonally.different.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$Within.AR.seasonally.different.bin.geometric.mean.dfast.detections > 0,]$Product.name,], aes(x = variable, y = value, colour = variable, shape = variable)) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`[`2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`$Product %in% `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`[`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$Non.enriched.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$AC.enriched.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$AR.enriched.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$Within.AC.seasonally.different.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$Within.AR.seasonally.different.bin.geometric.mean.dfast.detections > 0,]$Product.name,], aes(x = value, y = variable, colour = variable, shape = variable)) + geom_vline(xintercept = 0.05, linetype = "dashed", size = 1.25, colour = "black") + xlim(-1, 1) + scale_x_log10() + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`[`2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`$value < 1 & `2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`$Product %in% `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`[`2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$Non.enriched.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$AC.enriched.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$AR.enriched.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$Within.AC.seasonally.different.bin.geometric.mean.dfast.detections > 0 | `2022_10_17-all_single_unique_products_dfast_strict_bins_shapiro-wilkes_and_levene_test_pvalues_by_bin_enrichment_within_or_across_host_status-Levene_pvalue_sorted`$Within.AR.seasonally.different.bin.geometric.mean.dfast.detections > 0,]$Product.name,], aes(x = value, y = variable, colour = variable, shape = variable)) + geom_vline(xintercept = 0.05, linetype = "dashed", size = 1.25, colour = "black") + xlim(0, 1) + scale_x_log10() + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))


2022-10-28
#Version post chat with Wilfried, might still be wrong plus writing after drinking
#so check tomorrow
tic()
test_products_hypergeometric_pvalues_df <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Product", "Noise.bins.pvalue", "AC.enriched.versus.bins.pvalue", "AR.enriched.versus.bins.pvalue", "AC.enriched.within.bins.pvalue", "AR.enriched.within.bins.pvalue"))))
for (product in `2022.10.16.Unique_product_names`){
product.name <- paste(product)
total.number.bins <- 348
noise.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
AC.enriched.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
AR.enriched.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
within.AC.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
within.AR.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
summed.noise.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
summed.AC.enriched.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
summed.AR.enriched.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
summed.within.AC.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
summed.within.AR.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
summed.all.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
noise.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
AC.enriched.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
AR.enriched.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
within.AC.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
within.AR.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
total.number.bins.value.product.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
noise.bins.product.pvalue <- phyper(q = (noise.bins.count - 1), m = summed.noise.product.values, n = (summed.all.bins.product.values - summed.noise.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
AC.enriched.bins.product.pvalue <- phyper(q = (AC.enriched.bins.count - 1), m = summed.AC.enriched.product.values, n = (summed.all.bins.product.values - summed.AC.enriched.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
AR.enriched.bins.product.pvalue <- phyper(q = (AR.enriched.bins.count - 1), m = summed.AR.enriched.product.values, n = (summed.all.bins.product.values - summed.AR.enriched.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
within.AC.bins.product.pvalue <- phyper(q = (within.AC.bins.count - 1), m = summed.within.AC.bins.product.values, n = (summed.all.bins.product.values - summed.within.AC.bins.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
within.AR.bins.product.pvalue <- phyper(q = (within.AR.bins.count - 1), m = summed.within.AR.bins.product.values, n = (summed.all.bins.product.values - summed.within.AR.bins.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
`test_products_hypergeometric_pvalues_df`[nrow(`test_products_hypergeometric_pvalues_df`) +1,] <- c(product.name, noise.bins.product.pvalue, AC.enriched.bins.product.pvalue, AR.enriched.bins.product.pvalue, within.AC.bins.product.pvalue, within.AR.bins.product.pvalue)
`test_products_hypergeometric_pvalues_df`[,c(2:6)] <- sapply(`test_products_hypergeometric_pvalues_df`[,c(2:6)], as.numeric)
rm(`product.name`)
rm(`total.number.bins`)
rm(`noise.bins.count`)
rm(`AC.enriched.bins.count`)
rm(`AR.enriched.bins.count`)
rm(`within.AC.bins.count`)
rm(`within.AR.bins.count`)
rm(`summed.noise.product.values`)
rm(`summed.AC.enriched.product.values`)
rm(`summed.AR.enriched.product.values`)
rm(`summed.within.AC.bins.product.values`)
rm(`summed.within.AR.bins.product.values`)
rm(`summed.all.bins.product.values`)
rm(`noise.bins.product.values.over.zero`)
rm(`AC.enriched.bins.product.values.over.zero`)
rm(`AR.enriched.bins.product.values.over.zero`)
rm(`within.AC.bins.product.values.over.zero`)
rm(`within.AR.bins.product.values.over.zero`)
rm(`total.number.bins.value.product.over.zero`)
rm(`noise.bins.product.pvalue`)
rm(`AC.enriched.bins.product.pvalue`)
rm(`AR.enriched.bins.product.pvalue`)
rm(`within.AC.bins.product.pvalue`)
rm(`within.AR.bins.product.pvalue`)
}
view(test_products_hypergeometric_pvalues_df)
`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status` <- test_products_hypergeometric_pvalues_df
write.table(`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, file = "2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status.tsv", sep="\t", row.names=FALSE)
`2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status` <- reshape2::melt(`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, id=c("Product"))
write.table(`2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, file = "2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status.tsv", sep="\t", row.names=FALSE)
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-10-22.RData")
toc()

`test_list` <- as.list(sample(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable), 25))

tic()
test_products_hypergeometric_pvalues_df <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Product", "Noise.bins.pvalue", "AC.enriched.versus.bins.pvalue", "AR.enriched.versus.bins.pvalue", "AC.enriched.within.bins.pvalue", "AR.enriched.within.bins.pvalue"))))
for (product in `test_list`){
  product.name <- paste(product)
  total.number.bins <- 348
  noise.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  AC.enriched.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  AR.enriched.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  within.AC.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  within.AR.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  summed.noise.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.AC.enriched.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.AR.enriched.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.within.AC.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.within.AR.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  summed.all.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
  noise.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  AC.enriched.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  AR.enriched.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  within.AC.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  within.AR.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  total.number.bins.value.product.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
  noise.bins.product.pvalue <- phyper(q = (noise.bins.count - 1), m = summed.noise.product.values, n = (summed.all.bins.product.values - summed.noise.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  AC.enriched.bins.product.pvalue <- phyper(q = (AC.enriched.bins.count - 1), m = summed.AC.enriched.product.values, n = (summed.all.bins.product.values - summed.AC.enriched.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  AR.enriched.bins.product.pvalue <- phyper(q = (AR.enriched.bins.count - 1), m = summed.AR.enriched.product.values, n = (summed.all.bins.product.values - summed.AR.enriched.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  within.AC.bins.product.pvalue <- phyper(q = (within.AC.bins.count - 1), m = summed.within.AC.bins.product.values, n = (summed.all.bins.product.values - summed.within.AC.bins.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  within.AR.bins.product.pvalue <- phyper(q = (within.AR.bins.count - 1), m = summed.within.AR.bins.product.values, n = (summed.all.bins.product.values - summed.within.AR.bins.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
  `test_products_hypergeometric_pvalues_df`[nrow(`test_products_hypergeometric_pvalues_df`) +1,] <- c(product.name, noise.bins.product.pvalue, AC.enriched.bins.product.pvalue, AR.enriched.bins.product.pvalue, within.AC.bins.product.pvalue, within.AR.bins.product.pvalue)
  `test_products_hypergeometric_pvalues_df`[,c(2:6)] <- sapply(`test_products_hypergeometric_pvalues_df`[,c(2:6)], as.numeric)
  rm(`product.name`)
  rm(`total.number.bins`)
  rm(`noise.bins.count`)
  rm(`AC.enriched.bins.count`)
  rm(`AR.enriched.bins.count`)
  rm(`within.AC.bins.count`)
  rm(`within.AR.bins.count`)
  rm(`summed.noise.product.values`)
  rm(`summed.AC.enriched.product.values`)
  rm(`summed.AR.enriched.product.values`)
  rm(`summed.within.AC.bins.product.values`)
  rm(`summed.within.AR.bins.product.values`)
  rm(`summed.all.bins.product.values`)
  rm(`noise.bins.product.values.over.zero`)
  rm(`AC.enriched.bins.product.values.over.zero`)
  rm(`AR.enriched.bins.product.values.over.zero`)
  rm(`within.AC.bins.product.values.over.zero`)
  rm(`within.AR.bins.product.values.over.zero`)
  rm(`total.number.bins.value.product.over.zero`)
  rm(`noise.bins.product.pvalue`)
  rm(`AC.enriched.bins.product.pvalue`)
  rm(`AR.enriched.bins.product.pvalue`)
  rm(`within.AC.bins.product.pvalue`)
  rm(`within.AR.bins.product.pvalue`)
}
view(test_products_hypergeometric_pvalues_df)
toc()

as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "multidrug.ABC.transporter.ATP.binding.protein" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))

tic()
test_products_hypergeometric_pvalues_df <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Product", "Noise.bins.pvalue", "AC.enriched.versus.bins.pvalue", "AR.enriched.versus.bins.pvalue", "AC.enriched.within.bins.pvalue", "AR.enriched.within.bins.pvalue"))))
for (product in `2022.10.16.Unique_product_names`){
product.name <- paste(product)
total.number.bins <- 348
noise.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
AC.enriched.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
AR.enriched.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
within.AC.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
within.AR.bins.count <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
summed.noise.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
summed.AC.enriched.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
summed.AR.enriched.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
summed.within.AC.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
summed.within.AR.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
summed.all.bins.product.values <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
noise.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "None" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
AC.enriched.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
AR.enriched.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
within.AC.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
within.AR.bins.product.values.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect" & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
total.number.bins.value.product.over.zero <- as.numeric(nrow(as.data.frame(unique(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value > 0,]$Product))))
noise.bins.product.pvalue <- phyper(q = (noise.bins.count - 1), m = summed.noise.product.values, n = (summed.all.bins.product.values - summed.noise.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
AC.enriched.bins.product.pvalue <- phyper(q = (AC.enriched.bins.count - 1), m = summed.AC.enriched.product.values, n = (summed.all.bins.product.values - summed.AC.enriched.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
AR.enriched.bins.product.pvalue <- phyper(q = (AR.enriched.bins.count - 1), m = summed.AR.enriched.product.values, n = (summed.all.bins.product.values - summed.AR.enriched.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
within.AC.bins.product.pvalue <- phyper(q = (within.AC.bins.count - 1), m = summed.within.AC.bins.product.values, n = (summed.all.bins.product.values - summed.within.AC.bins.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
within.AR.bins.product.pvalue <- phyper(q = (within.AR.bins.count - 1), m = summed.within.AR.bins.product.values, n = (summed.all.bins.product.values - summed.within.AR.bins.product.values), k = total.number.bins.value.product.over.zero, lower.tail = FALSE)
`test_products_hypergeometric_pvalues_df`[nrow(`test_products_hypergeometric_pvalues_df`) +1,] <- c(product.name, noise.bins.product.pvalue, AC.enriched.bins.product.pvalue, AR.enriched.bins.product.pvalue, within.AC.bins.product.pvalue, within.AR.bins.product.pvalue)
`test_products_hypergeometric_pvalues_df`[,c(2:6)] <- sapply(`test_products_hypergeometric_pvalues_df`[,c(2:6)], as.numeric)
rm(`product.name`)
rm(`total.number.bins`)
rm(`noise.bins.count`)
rm(`AC.enriched.bins.count`)
rm(`AR.enriched.bins.count`)
rm(`within.AC.bins.count`)
rm(`within.AR.bins.count`)
rm(`summed.noise.product.values`)
rm(`summed.AC.enriched.product.values`)
rm(`summed.AR.enriched.product.values`)
rm(`summed.within.AC.bins.product.values`)
rm(`summed.within.AR.bins.product.values`)
rm(`summed.all.bins.product.values`)
rm(`noise.bins.product.values.over.zero`)
rm(`AC.enriched.bins.product.values.over.zero`)
rm(`AR.enriched.bins.product.values.over.zero`)
rm(`within.AC.bins.product.values.over.zero`)
rm(`within.AR.bins.product.values.over.zero`)
rm(`total.number.bins.value.product.over.zero`)
rm(`noise.bins.product.pvalue`)
rm(`AC.enriched.bins.product.pvalue`)
rm(`AR.enriched.bins.product.pvalue`)
rm(`within.AC.bins.product.pvalue`)
rm(`within.AR.bins.product.pvalue`)
}
view(test_products_hypergeometric_pvalues_df)
`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status` <- test_products_hypergeometric_pvalues_df
write.table(`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, file = "2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status.tsv", sep="\t", row.names=FALSE)
`2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status` <- reshape2::melt(`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, id=c("Product"))
write.table(`2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, file = "2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status.tsv", sep="\t", row.names=FALSE)
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-10-22.RData")
toc()

`test_products_hypergeometric_pvalues_df`
`2022-10-28_working2` <- `2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`

`2022-10-28_working1` <- `2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`
`2022-10-28_working1`$BH.corrected.noise.bins.Pvalue <- p.adjust(`2022-10-28_working1`[order(`2022-10-28_working1`$Noise.bins.pvalue),]$Noise.bins.pvalue, method = "BH")
`2022-10-28_working1`$BH.corrected.AC.enriched.versus.bins.Pvalue <- p.adjust(`2022-10-28_working1`[order(`2022-10-28_working1`$Noise.bins.pvalue),]$AC.enriched.versus.bins.pvalue, method = "BH")
`2022-10-28_working1`$BH.corrected.AR.enriched.versus.bins.Pvalue <- p.adjust(`2022-10-28_working1`[order(`2022-10-28_working1`$Noise.bins.pvalue),]$AR.enriched.versus.bins.pvalue, method = "BH")
`2022-10-28_working1`$BH.corrected.AC.enriched.within.bins.Pvalue <- p.adjust(`2022-10-28_working1`[order(`2022-10-28_working1`$Noise.bins.pvalue),]$AC.enriched.within.bins.pvalue, method = "BH")
`2022-10-28_working1`$BH.corrected.AR.enriched.within.bins.Pvalue <- p.adjust(`2022-10-28_working1`[order(`2022-10-28_working1`$Noise.bins.pvalue),]$AR.enriched.within.bins.pvalue, method = "BH")

view(`2022-10-28_working1`)

`2022-10-28_melted_working1` <- reshape2::melt(`2022-10-28_working1`, id=c("Product"))

ggplot(data = `2022-10-28_melted_working1`[grepl(glob2rx("BH.corrected*"), `2022-10-28_melted_working1`$variable),], aes(x = value, y = variable, colour = variable, fill = variable)) + geom_jitter() + theme_linedraw()

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-10-22.RData")
toc()

2022-10-31

ggplot(data = `06_09_melted_krak_read_percent`[(`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria"), ], aes(x = Kraken_minimum_confidence_score_required, y = value, group = Kraken_minimum_confidence_score_required)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_violin(color="black", fill='#8a795d', alpha = 0.65, width = 12) + geom_boxplot(width = 1, fill = '#f2f3f4') + xlab("Minimum confidence score required for classification") + ylab("Reads classified as bacteria as percent\nof total input reads (%)") + scale_x_continuous(breaks=seq(0,100,5))
`06_09_met_read_percent`
ggplot(data = `06_10_melted_kaiju_read_percent`, aes(x = Errors_allowed, y = value, group = Errors_allowed)) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + geom_violin(color="black", fill='#8a795d', alpha = 0.65, width = 1) + geom_boxplot(width = 0.5, fill = '#f2f3f4') + scale_x_continuous(breaks=seq(0,20,1)) + xlab("Allowed errors in match") + ylab("Percent input reads reads classified (%)")

Range(`06_09_melted_krak_read_percent`[`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria" & `06_09_melted_krak_read_percent`$Kraken_minimum_confidence_score_required == 10,]$value)

Range(`06_09_melted_krak_read_percent`[`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria" & `06_09_melted_krak_read_percent`$Kraken_minimum_confidence_score_required == 50,]$value)

exp(mean(log((`06_09_melted_krak_read_percent`[`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria" & `06_09_melted_krak_read_percent`$Kraken_minimum_confidence_score_required == 10,]$value))))
exp(mean(log((`06_09_melted_krak_read_percent`[`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria" & `06_09_melted_krak_read_percent`$Kraken_minimum_confidence_score_required == 25,]$value))))
exp(mean(log((`06_09_melted_krak_read_percent`[`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria" & `06_09_melted_krak_read_percent`$Kraken_minimum_confidence_score_required == 50,]$value))))
exp(mean(log((`06_09_melted_krak_read_percent`[`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria" & `06_09_melted_krak_read_percent`$Kraken_minimum_confidence_score_required == 66,]$value))))
exp(mean(log((`06_09_melted_krak_read_percent`[`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria" & `06_09_melted_krak_read_percent`$Kraken_minimum_confidence_score_required == 75,]$value))))
exp(mean(log((`06_09_melted_krak_read_percent`[`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria" & `06_09_melted_krak_read_percent`$Kraken_minimum_confidence_score_required == 95,]$value))))
median(`06_09_melted_krak_read_percent`[`06_09_melted_krak_read_percent`$variable == "Percent_reads_classified_Bacteria" & `06_09_melted_krak_read_percent`$Kraken_minimum_confidence_score_required == 95,]$value)

`06_09_met_read_percent`

exp(mean(log(`06_09_met_read_percent`$Estimated_reads_mapped_to_known_clades_by_Metaphlan_as_percentage_reads_in_subsampled_read_file)))

`06_10_melted_kaiju_read_percent`
Range(`06_10_melted_kaiju_read_percent`[`06_10_melted_kaiju_read_percent`$Errors_allowed == 0 & `06_10_melted_kaiju_read_percent`$variable == "Percent_reads_classified",]$value)

exp(mean(log(`06_10_melted_kaiju_read_percent`[`06_10_melted_kaiju_read_percent`$Errors_allowed == 0 & `06_10_melted_kaiju_read_percent`$variable == "Percent_reads_classified",]$value)))

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-10-31.RData")
toc()

2022-11-01
`2022.11.01_transposed_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-11-01_transposed_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages.txt")
View(`2022.11.01_transposed_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`)

`2022.11.01_melted_transposed_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages` <- reshape2::melt(`2022.11.01_transposed_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`, id=c("Sample", "Species", "Season", "Sequencing", "Species.x.season", "SampleID"))
view(`2022.11.01_melted_transposed_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`)
`2022.11.01.metaphlan.melted` <- `2022.11.01_melted_transposed_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Kingdom",]$Sample,], aes(x = Species.x.season, y = value, fill = Species.x.season, colour = Species.x.season)) + geom_jitter()
ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Phylum",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season, colour = Species.x.season)) + geom_jitter()

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Phylum",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season, colour = Species.x.season)) + geom_jitter() + scale_colour_manual(name = "Species.x.season", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + facet_wrap(~variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Phylum",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season, colour = Species.x.season)) + geom_jitter() + scale_colour_manual(name = "Species.x.season", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + facet_wrap(~variable) + theme(text = element_text(family = "Tex Gyre Heroes")) + stat_compare_means(method = "wilcox.test")


ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Phylum",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, colour = Species.x.season)) + geom_jitter() + scale_colour_manual(name = "Species.x.season", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + stat_compare_means(method = "wilcox.test", label = 'p.signif') + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Class",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, colour = Species.x.season)) + geom_jitter() + scale_colour_manual(name = "Species.x.season", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Class",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.15) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Order",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.15) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Family",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.15) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Family",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.15) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.15) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Species",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.15) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable == "Bifidobacterium pseudolongum" & `2022.11.01.metaphlan.melted`$value > 0,]

nrow(`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Phylum",])

nrow(as.data.frame(unique(`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable == "Dietzia alimentaria" & `2022.11.01.metaphlan.melted`$value > 0,]$Species.x.season)))

ggplot(data = `2022-10-22_working_melted`[`2022-10-22_working_melted`$variable == "Raw.Metaphlan.estimated.percent.reads.classified" | `2022-10-22_working_melted`$variable == "Processed.Metaphlan.estimated.percent.reads.classified",], aes(x = variable, y = value)) + geom_boxplot(aes(fill = Host_X_season), colour = "#0f0f0f", alpha = 0.85) + xlab("File type") + ylab("Estimated percent reads classified (%)") + ggtitle("Metaphlan estimated percent reads classified pre- and post-processing") + scale_x_discrete(labels = c("Raw", "Processed")) + theme_linedraw() + scale_fill_manual(name = "Host species and\nsample month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + stat_compare_means(method = "wilcox.test", label.y = 25.5, label = 'p.signif') + facet_wrap(~ Host_X_season) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022.11.01.metaphlan.melted`$variable <- gsub(".", " ", fixed = TRUE, as.character(`2022.11.01.metaphlan.melted`$variable))

`for_sorting_melted_working`$variable <- gsub("errors allowed", "", as.character(`for_sorting_melted_working`$variable))

2022-11-02
nrow(`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Phylum",])
nrow(`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Class",])
nrow(`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Order",])
nrow(`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Family",])
nrow(`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",])
nrow(`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Species",])

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Phylum",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.10) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-02_metaphlan_all_subsampled_reads_all_possible_phyla_boxplot_species_and_season_relative_abundances` <- ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Phylum",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.10) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",]$Sample & count(`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$),]

`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01.metaphlan.melted`[count(`2022.11.01.metaphlan.melted`$value) >= 2,]$variable,] & `2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,]

`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01.metaphlan.melted`[nrow(as.data.frame(count(`2022.11.01.metaphlan.melted`$variable))) >= 2,]$variable & `2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",]$Sample,]

`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$value > 0 & `2022.11.01.metaphlan.melted`$variable %in% `2022.11.01.metaphlan.melted`[count(`2022.11.01.metaphlan.melted`$value) > 1,]$variable,]


`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",]$Sample & `2022.11.01.metaphlan.melted`$value > 0 & `2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Sample %in% `2022.11.01.metaphlan.melted`[(`2022.11.01.metaphlan.melted` %>% dplyr::count(value))$n > 1,]$variable,]$Sample,]

`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,] %>% dplyr::count(variable)

`2022.11.01.metaphlan.melted.genus.count.samples.detected.in` <- as.data.frame(`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,] %>% dplyr::count(variable))

ggplot(data = `2022.11.01.metaphlan.melted.genus.count.samples.detected.in`, aes(x = reorder(variable, -n), y = n, fill = ifelse(n > 1, 'black', 'red'))) + geom_bar(stat = "identity") + coord_flip() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(values = c("#ff1e90", "#0f0f0f")) + ylab("Genus") + xlab("Number of samples with assigned\nreads to genus") + theme(legend.position = "none")
`2022-11-02_all_subsampled_reads_metaphlan_number_of_samples_each_possible_genus_found_in` <- ggplot(data = `2022.11.01.metaphlan.melted.genus.count.samples.detected.in`, aes(x = reorder(variable, -n), y = n, fill = ifelse(n > 1, 'black', 'red'))) + geom_bar(stat = "identity") + coord_flip() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(values = c("#ff1e90", "#0f0f0f")) + ylab("Genus") + xlab("Number of samples with assigned\nreads to genus") + theme(legend.position = "none")

`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",]$Sample & `2022.11.01.metaphlan.melted`$value > 0 & `2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Sample %in% `2022.11.01.metaphlan.melted`[count(`2022.11.01.metaphlan.melted`$value) > 1,]$variable,]$Sample,]

ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01.metaphlan.melted.genus.count.samples.detected.in`[`2022.11.01.metaphlan.melted.genus.count.samples.detected.in`$n > 1,]$variable & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.15) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))
`2022.11.02.metaphlan_all_subsampled_reads_relative_abundance_genus_level_only_genera_found_at_least_two_samples` <- ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01.metaphlan.melted.genus.count.samples.detected.in`[`2022.11.01.metaphlan.melted.genus.count.samples.detected.in`$n > 1,]$variable & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.15) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$Species == "AR" & `2022.11.01.metaphlan.melted`$variable %in% `2022.11.01.metaphlan.melted.genus.count.samples.detected.in`[`2022.11.01.metaphlan.melted.genus.count.samples.detected.in`$n > 1,]$variable & `2022.11.01.metaphlan.melted`$value > 25 & `2022.11.01.metaphlan.melted`$variable == "Bacteroides",]

#here on 2022-12-13
`2022-11-02_metaphlan_all_subsampled_reads_all_possible_phyla_boxplot_species_and_season_relative_abundances` <- ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Phylum",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.10) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1ec8ff", "ACN" = "#006ad1", "ARJ" = "#ff551e", "ARN" = "#d16700")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(strip.text.x = element_text(size = 12)) + theme(text = element_text(family = "Tex Gyre Heroes"))
`2022.11.02.metaphlan_all_subsampled_reads_relative_abundance_genus_level_only_genera_found_at_least_two_samples` <- ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01.metaphlan.melted.genus.count.samples.detected.in`[`2022.11.01.metaphlan.melted.genus.count.samples.detected.in`$n > 1,]$variable & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", width = 0.15) + scale_fill_manual(name = "Host species and\ncollection month", values = c("ACJ" = "#1ec8ff", "ACN" = "#006ad1", "ARJ" = "#ff551e", "ARN" = "#d16700")) + theme_linedraw() + xlab("Host species and collection month") + ylab("Relative abundance (%)") + facet_wrap(~ variable) + theme(strip.text.x = element_text(size = 10)) + theme(text = element_text(family = "Tex Gyre Heroes"))


tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-10-31.RData")
toc()

2022-11-03
`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`

`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 0.1,]

`2022-11-03_kraken_all_subsampled_reads_05_minimum_confidence_score_only_at_least_percentage_0_point_01_percent` <- `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 0.01,]

`2022-11-03_kraken_all_subsampled_reads_05_minimum_confidence_score_only_at_least_percentage_0_point_01_percent_not_unclassified` <- `2022-11-03_kraken_all_subsampled_reads_05_minimum_confidence_score_only_at_least_percentage_0_point_01_percent`[`2022-11-03_kraken_all_subsampled_reads_05_minimum_confidence_score_only_at_least_percentage_0_point_01_percent`$variable != "unclassified",]

fig_A <- ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0', ], aes(x = Species_x_season, y = value, fill = Species_x_season, colour = Species_x_season)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Host species and sampling month") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Host species and sampling month"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + facet_wrap(~ Sequencing) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

fig_B <- ggplot(data = `melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`[`melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable != 'X0', ], aes(x = Species_x_season, y = value, fill = Species_x_season, colour = Species_x_season)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxID (%)") + xlab("Host species and sampling month") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Host species and sampling month"), fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + facet_wrap(~ Sequencing) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw()

`melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`

nrow(as.data.frame(unique(`melted_2022_09_02_Transposed_010_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable)))
nrow(as.data.frame(unique(`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_taxid_percentages_from_report_file`$variable)))

fig_C <- fig_A + fig_B + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')
2022-11-04_Kraken2_subsampled_reads_A_50_percent_B_10_percent_min_confidence_scores_percent_reads_classified_as_taxID_all_possible_detected_taxIDs

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-11-04_Kraken2_subsampled_reads_A_50_percent_B_10_percent_min_confidence_scores_percent_reads_classified_as_taxID_all_possible_detected_taxIDs.svg", fig_C, device = "svg")

`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified` <- `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$variable != "unclassified" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$variable != "root",]

`unique_kraken_subsampled_taxons_50_percent_min_conf_LIST` <- as.list(unique(`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`$variable))

tic()
`kraken_050_taxa_species_X_season_geometric_means` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Taxon", "ACJ.geometric.mean", "ACN.geometric.mean", "ARJ.geometric.mean", "ARN.geometric.mean"))))
for (taxon in `unique_kraken_subsampled_taxons_50_percent_min_conf_LIST`) {
taxon.name <- paste(taxon)
ACJ.geometric.mean <- exp(mean(log(`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`$variable == taxon.name & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`$Species_x_season == "ACJ",]$value)))
ACN.geometric.mean <- exp(mean(log(`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`$variable == taxon.name & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`$Species_x_season == "ACN",]$value)))
ARJ.geometric.mean <- exp(mean(log(`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`$variable == taxon.name & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`$Species_x_season == "ARJ",]$value)))
ARN.geometric.mean <- exp(mean(log(`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`$variable == taxon.name & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file_excluding_root_and_unclassified`$Species_x_season == "ARN",]$value)))
`kraken_050_taxa_species_X_season_geometric_means`[nrow(`kraken_050_taxa_species_X_season_geometric_means`) +1,] <- c(taxon.name, ACJ.geometric.mean, ACN.geometric.mean, ARJ.geometric.mean, ARN.geometric.mean)
`kraken_050_taxa_species_X_season_geometric_means`[,c(2:5)] <- sapply(`kraken_050_taxa_species_X_season_geometric_means`[,c(2:5)], as.numeric)
rm(taxon.name)
rm(ACJ.geometric.mean)
rm(ACN.geometric.mean)
rm(ARJ.geometric.mean)
rm(ARN.geometric.mean)
}
write.table(`kraken_050_taxa_species_X_season_geometric_means`, file = "2022-11-04_kraken_050_taxa_species_X_season_geometric_means.tsv", sep="\t", row.names=FALSE)
toc()

`kraken_050_taxa_species_X_season_geometric_means_melted` <- reshape2::melt(`kraken_050_taxa_species_X_season_geometric_means`, id = c("Taxon"))

ggplot(data = `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$variable == "Actinomycetia",], aes(x = `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$Species == "AC",]$value, y = `melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`[`melted.2022.06.23_Transposed_0_allowed_errors_combined_results_ONLY_NOVOGENE_PAIRED_subsampled_files_phylum_level_results_all_potential_taxons_percentages`$Species == "AC",]$value)) + geom_point()

ggplot(data = `kraken_050_taxa_species_X_season_geometric_means`, aes(x = ACJ.geometric.mean, y = ACN.geometric.mean)) + geom_point()
ggplot(data = `kraken_050_taxa_species_X_season_geometric_means`, aes(x = ARJ.geometric.mean, y = ARN.geometric.mean)) + geom_point()

ggplot(data = `kraken_050_taxa_species_X_season_geometric_means`, aes(x = ACJ.geometric.mean, y = ARJ.geometric.mean)) + geom_point()
ggplot(data = `kraken_050_taxa_species_X_season_geometric_means`, aes(x = ACN.geometric.mean, y = ARN.geometric.mean)) + geom_point()

`kraken_050_taxa_species_X_season_geometric_means`[`kraken_050_taxa_species_X_season_geometric_means`$ACJ.geometric.mean > 0 & `kraken_050_taxa_species_X_season_geometric_means`$ACN.geometric.mean > 0 & `kraken_050_taxa_species_X_season_geometric_means`$ARJ.geometric.mean > 0 & `kraken_050_taxa_species_X_season_geometric_means`$ARN.geometric.mean > 0,]

write.table(view(`kraken_050_taxa_species_X_season_geometric_means`[`kraken_050_taxa_species_X_season_geometric_means`$ACJ.geometric.mean > 0 & `kraken_050_taxa_species_X_season_geometric_means`$ACN.geometric.mean > 0 & `kraken_050_taxa_species_X_season_geometric_means`$ARJ.geometric.mean > 0 & `kraken_050_taxa_species_X_season_geometric_means`$ARN.geometric.mean > 0,]), file = "2022-11-04_kraken_050_taxa_species_X_season_geometric_means_only_taxons_with_above_zero_values_for_all_geometric_means_all_host_and_season_combos.tsv", sep="\t", row.names=FALSE)

ggplot(data = `kraken_050_taxa_species_X_season_geometric_means_melted`[`kraken_050_taxa_species_X_season_geometric_means_melted`$value > 0,], aes(x = variable, y = value, colour = variable)) + geom_jitter()

ggplot(data = `kraken_050_taxa_species_X_season_geometric_means_melted`[`kraken_050_taxa_species_X_season_geometric_means_melted`$value > 0,], aes(x = variable, y = value, colour = variable)) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_colour_manual(name = "Geometric mean percentage reads classified\nby host species and collection month", values = c("ACJ.geometric.mean" = "#1e90ff", "ACN.geometric.mean" = "#004385", "ARJ.geometric.mean" = "#ff8d1e", "ARN.geometric.mean" = "#522800")) + facet_wrap(~ variable, scales = "free_x") + geom_label_repel(aes(label = ifelse(value >= 0.9, as.character(Taxon), ''), hjust = 0, vjust = 0), colour = "black", min.segment.length = 0)

ggplot(data = `kraken_050_taxa_species_X_season_geometric_means_melted`[`kraken_050_taxa_species_X_season_geometric_means_melted`$value > 0,], aes(x = reorder(Taxon, value), y = value, fill = variable)) + geom_bar(stat = "identity") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(name = "Geometric mean percentage reads classified\nby host species and collection month", values = c("ACJ.geometric.mean" = "#1e90ff", "ACN.geometric.mean" = "#004385", "ARJ.geometric.mean" = "#ff8d1e", "ARN.geometric.mean" = "#522800")) + facet_wrap(~ variable, scales = "free_y") + coord_flip() + ylab("Geometric mean percentage reads classified (%)") + xlab("Taxon")

ggplot(data = `kraken_050_taxa_species_X_season_geometric_means_melted`[`kraken_050_taxa_species_X_season_geometric_means_melted`$value > 0,], aes(x = reorder(Taxon, value), y = value, fill = variable)) + geom_bar(stat = "identity") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(name = "Geometric mean percentage reads classified\nby host species and collection month", values = c("ACJ.geometric.mean" = "#1e90ff", "ACN.geometric.mean" = "#004385", "ARJ.geometric.mean" = "#ff8d1e", "ARN.geometric.mean" = "#522800")) + facet_wrap(~ variable, scales = "free_y") + coord_flip() + ylab("Geometric mean percentage reads classified (%)") + xlab("Taxon")

`2022-11-04_kraken2_all_subsampled_reads_50_percent_min_conf_score_geometric_means_host_and_month_only_at_least_or_above_0_point_9_percent` <- ggplot(data = `kraken_050_taxa_species_X_season_geometric_means_melted`[`kraken_050_taxa_species_X_season_geometric_means_melted`$value > 0,], aes(x = reorder(Taxon, value), y = value, fill = variable)) + geom_bar(stat = "identity") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(name = "Geometric mean percentage reads classified\nby host species and collection month", values = c("ACJ.geometric.mean" = "#1e90ff", "ACN.geometric.mean" = "#004385", "ARJ.geometric.mean" = "#ff8d1e", "ARN.geometric.mean" = "#522800")) + facet_wrap(~ variable, scales = "free_y") + coord_flip() + ylab("Geometric mean percentage reads classified (%)") + xlab("Taxon")

ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$variable != 'unclassified' & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$variable != 'root' & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 2.5, ], aes(x = reorder(variable, -value), y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxon (%)") + xlab("Taxa classified by Kraken2 with >2.5% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + theme(text = element_text(family = "Tex Gyre Heroes")) + ylim(2.5,10.5) + facet_wrap(~ Sequencing) + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 7.5))

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-11-04.RData")
toc()

2022-11-07
ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Sequencing == "Novogene" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Species == "AC" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 0,], aes(x = Sample_ID, y = value, fill = Season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.65, width = 0.2) + scale_fill_manual(values = c("#1ec8ff", "#006ad1")) + scale_y_log10(labels = label_log()) + ylab("Percentage of reads classified to taxon\nLog 10 scale (%)") + xlab("Sample ID") + facet_wrap(~ Taxon, scales = "free_x") + theme_linedraw() + guides(fill = guide_legend(title = "Sampling month"))

ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Sequencing == "Novogene" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Species == "AC" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 0,], aes(x = Sample_ID, y = value, fill = Season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.65, width = 0.2) + scale_fill_manual(values = c("#1ec8ff", "#006ad1")) + scale_y_log10(labels = label_log()) + ylab("Percentage of reads classified to taxon\nLog 10 scale (%)") + xlab("Sample ID") + facet_wrap(~ Season, scales = "free_x") + theme_linedraw() + guides(fill = guide_legend(title = "Sampling month")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5), text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Sequencing == "Novogene" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Species == "AR" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 0,], aes(x = Sample_ID, y = value, fill = Season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.65, width = 0.2) + scale_fill_manual(values = c("#ff551e", "#d16700")) + scale_y_log10(labels = label_log()) + ylab("Percentage of reads classified to taxon\nLog 10 scale (%)") + xlab("Sample ID") + facet_wrap(~ Season, scales = "free_x") + theme_linedraw() + guides(fill = guide_legend(title = "Sampling month")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5), text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Sequencing == "Novogene" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Species == "AR" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 0,], aes(x = Sample_ID, y = value, fill = Season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.65, width = 0.2) + scale_fill_manual(values = c("#ff551e", "#d16700")) + scale_y_log10(labels = label_log()) + ylab("Percentage of reads classified to taxon\nLog 10 scale (%)") + xlab("Sample ID") + facet_wrap(~ Season, scales = "free_x") + theme_classic() + guides(fill = guide_legend(title = "Sampling month")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5), text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Sequencing == "Novogene" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Species == "AC" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 0,], aes(x = Sample_ID, y = value, fill = Season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.65, width = 0.2) + scale_fill_manual(values = c("#1ec8ff", "#006ad1")) + scale_y_log10(labels = label_log()) + ylab("Percentage of reads classified to taxon\nLog 10 scale (%)") + xlab("Sample ID") + facet_wrap(~ Taxon, scales = "free_x") + theme_classic() + guides(fill = guide_legend(title = "Sampling month")) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-07_Kraken_50_percent_min_conf_subsampled_reads_only_novogene_includes_unpaired_violin_and_box_log_percent_reads_classified_AC` <- ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Sequencing == "Novogene" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Species == "AC" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 0,], aes(x = Sample_ID, y = value, fill = Season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.65, width = 0.2) + scale_fill_manual(values = c("#1ec8ff", "#006ad1")) + scale_y_log10(labels = label_log()) + ylab("Percentage of reads classified to taxon\nLog 10 scale (%)") + xlab("Sample ID") + facet_wrap(~ Taxon, scales = "free_x") + theme_classic() + guides(fill = guide_legend(title = "Sampling month")) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-07_Kraken_50_percent_min_conf_subsampled_reads_only_novogene_includes_unpaired_violin_and_box_log_percent_reads_classified_AR` <- ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Sequencing == "Novogene" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$Species == "AR" & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 0,], aes(x = Sample_ID, y = value, fill = Season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.65, width = 0.2) + scale_fill_manual(values = c("#ff551e", "#d16700")) + scale_y_log10(labels = label_log()) + ylab("Percentage of reads classified to taxon\nLog 10 scale (%)") + xlab("Sample ID") + facet_wrap(~ Taxon, scales = "free_x") + theme_classic() + guides(fill = guide_legend(title = "Sampling month")) + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`[`melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$variable != 'unclassified' & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$variable != 'root' & `melted_2022_09_02_Transposed_050_min_confidence_all_processed_subsampled_samples_Kraken_all_potential_TAXONS_percentages_from_report_file`$value > 2.5, ], aes(x = reorder(variable, -value), y = value, fill = Species_x_season, colour = Species_x_season, shape = Sequencing)) + geom_jitter() + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + scale_colour_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + ylab("Percent reads classified as taxon (%)") + xlab("Taxa classified by Kraken2 with >2.5% reads in at least one sample") + guides(colour = guide_legend(title = "Host species and sampling month"), shape = guide_legend(title = "Sequencing run"), fill = guide_legend(title = "Host species and sampling month")) + theme(text = element_text(family = "Tex Gyre Heroes")) + ylim(2.5,10.5) + facet_wrap(~ Sequencing, scales = "free_x") + theme_linedraw() + theme(axis.text.x=element_text(angle = 45, hjust = 1, size = 7.5))

`kraken_050_taxa_species_X_season_geometric_means` <- `kraken_050_taxa_species_X_season_geometric_means` %>% rowwise() %>% mutate(Log2.plus.one.fold.change.ACJ.to.ACN = log2(((ACN.geometric.mean - ACJ.geometric.mean) / ACJ.geometric.mean) + 1))

ggplot(data = `kraken_050_taxa_species_X_season_geometric_means`[`kraken_050_taxa_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN != "Nan" & !is.infinite(`kraken_050_taxa_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN),], aes(x = Log2.plus.one.fold.change.ACJ.to.ACN, y = Taxon)) + geom_bar(stat = "identity")

temp_df <- `kraken_050_taxa_species_X_season_geometric_means`[!is.infinite(`kraken_050_taxa_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN) & !is.na(`kraken_050_taxa_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN),]

`kraken_050_taxa_species_X_season_geometric_means` <- `kraken_050_taxa_species_X_season_geometric_means` %>% rowwise() %>% mutate(Log2.plus.one.fold.change.ARJ.to.ARN = log2(((ARN.geometric.mean - ARJ.geometric.mean) / ARJ.geometric.mean) + 1))

ggplot(`kraken_050_taxa_species_X_season_geometric_means`[!is.infinite(`kraken_050_taxa_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN) & !is.na(`kraken_050_taxa_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN),], aes(x = Log2.plus.one.fold.change.ARJ.to.ARN, y = reorder(Taxon, -Log2.plus.one.fold.change.ARJ.to.ARN), fill = ifelse(Log2.plus.one.fold.change.ARJ.to.ARN < 0, 'red', 'black'))) + geom_col() + theme_linedraw() + ylab("Taxon") + xlab("Log2()+1 change in geometric mean percentage reads\nassigned from June to November") + scale_fill_manual(values = c("#d16700", "#ff551e"), name = "Month enriched", labels = c("November", "June"))

`2022-11-07_kraken2_50_percent_min_conf_all_subsampled_reads_geometric_means_percentage_classified_reads_AC_only_log2_plus_one_change_June_to_November` <- ggplot(data = temp_df, aes(x = Log2.plus.one.fold.change.ACJ.to.ACN, y = reorder(Taxon, -Log2.plus.one.fold.change.ACJ.to.ACN), fill = ifelse(Log2.plus.one.fold.change.ACJ.to.ACN < 0, 'red', 'black'))) + geom_col() + theme_linedraw() + ylab("Taxon") + xlab("Log2()+1 change in geometric mean percentage reads\nassigned from June to November") + scale_fill_manual(values = c("#006ad1" ,"#1ec8ff"), name = "Month enriched", labels = c("November", "June")) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-07_kraken2_50_percent_min_conf_all_subsampled_reads_geometric_means_percentage_classified_reads_AR_only_log2_plus_one_change_June_to_November` <- ggplot(`kraken_050_taxa_species_X_season_geometric_means`[!is.infinite(`kraken_050_taxa_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN) & !is.na(`kraken_050_taxa_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN),], aes(x = Log2.plus.one.fold.change.ARJ.to.ARN, y = reorder(Taxon, -Log2.plus.one.fold.change.ARJ.to.ARN), fill = ifelse(Log2.plus.one.fold.change.ARJ.to.ARN < 0, 'red', 'black'))) + geom_col() + theme_linedraw() + ylab("Taxon") + xlab("Log2()+1 change in geometric mean percentage reads\nassigned from June to November") + scale_fill_manual(values = c("#d16700", "#ff551e"), name = "Month enriched", labels = c("November", "June")) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-07_kraken2_50_percent_min_conf_all_subsampled_reads_geometric_means_percentage_classified_reads_AC_only_log2_plus_one_change_June_to_November` / `2022-11-07_kraken2_50_percent_min_conf_all_subsampled_reads_geometric_means_percentage_classified_reads_AR_only_log2_plus_one_change_June_to_November` + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

2022-11-08
`2022.11.08_Transposed_Kaiju_phylum_0_errors_all_processed_COPY` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-11-08_Transposed_Kaiju_phylum_0_errors_all_processed_COPY.txt")
View(`2022.11.08_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`)

`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY` <- reshape2::melt(`2022.11.08_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`, id=c("Tax_ID", "Species", "Season", "Sequencing", "Species.x.season", "SampleID"))

ggplot(data = `2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`, aes(x = variable, y = value)) + geom_bar(stat = "summary", fun = "mean")

ggplot(data = `2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`[`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$value > 0,], aes(x = variable, y = value)) + geom_bar(stat = "summary", fun = "mean")

tic()
`unique_kaiju_subsampled_phylum_level_0_errors_LIST` <- as.list(unique(`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$variable))
`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Phylum", "ACJ.geometric.mean", "ACN.geometric.mean", "ARJ.geometric.mean", "ARN.geometric.mean"))))
for (taxon in `unique_kaiju_subsampled_phylum_level_0_errors_LIST`) {
taxon.name <- paste(taxon)
ACJ.geometric.mean <- exp(mean(log(`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`[`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$variable == taxon.name & `2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$Species.x.season == "ACJ",]$value)))
ACN.geometric.mean <- exp(mean(log(`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`[`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$variable == taxon.name & `2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$Species.x.season == "ACN",]$value)))
ARJ.geometric.mean <- exp(mean(log(`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`[`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$variable == taxon.name & `2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$Species.x.season == "ARJ",]$value)))
ARN.geometric.mean <- exp(mean(log(`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`[`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$variable == taxon.name & `2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$Species.x.season == "ARN",]$value)))
`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`[nrow(`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`) +1,] <- c(taxon.name, ACJ.geometric.mean, ACN.geometric.mean, ARJ.geometric.mean, ARN.geometric.mean)
`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`[,c(2:5)] <- sapply(`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`[,c(2:5)], as.numeric)
rm(taxon.name)
rm(ACJ.geometric.mean)
rm(ACN.geometric.mean)
rm(ARJ.geometric.mean)
rm(ARN.geometric.mean)
}
write.table(`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`, file = "kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means.tsv", sep="\t", row.names=FALSE)
toc()

`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means` %>% filter_all(all_vars(.>1))

ggplot(data = `2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`[`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$variable %in% (`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means` %>% filter_all(all_vars(.>1)))$Phylum,], aes(x = variable, y = value)) + geom_bar(stat = "summary", fun = "mean")

ggplot(data = `2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`[`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$variable %in% (`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means` %>% filter_all(all_vars(.>1)))$Phylum,], aes(x = variable, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Phylum") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + facet_wrap(~ Species.x.season, scales = "free_x") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`[`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$variable %in% (`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means` %>% filter_all(any_vars(.>1)))$Phylum,], aes(x = variable, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Phylum") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + facet_wrap(~ Species.x.season, scales = "free_x") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-08_all_subsampled_reads_Kaiju_0_errors_phylum_level_phyla_with_geometric_means_above_1_all_species_x_seasons_log_scale_percents_box_and_violin` <- ggplot(data = `2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`[`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY`$variable %in% (`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means` %>% filter_all(all_vars(.>1)))$Phylum,], aes(x = variable, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Phylum") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10() + facet_wrap(~ Species.x.season, scales = "free_x") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022.11.08_Melted_Transposed_Kaiju_phylum_0_errors_all_processed_COPY` %>% gather()

`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`[order(`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`$ACJ.geometric.mean),]
sort(`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`$ACJ.geometric.mean, decreasing = TRUE)

`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`$Phylum %in% (`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means` %>% filter_all(all_vars(.>1)))$Phylum,]
`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`$Phylum %in% (`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means` %>% filter_all(any_vars(.>1)))$Phylum,]

`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`$Phylum %in% (`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means` %>% filter_all(`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`[2:5,](.>1)))$Phylum,]

`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means`$Phylum %in% (`kaiju_subsampled_phylum_level_0_errors_species_X_season_geometric_means` %>% filter_all(select(., ACJ.geometric.mean:ARN.geometric.mean)) >1)))$Phylum,]

`2022.11.08_transposed_Kaiju_0_errors_genus_level_subsampled_reads_taxIDs_percent_reads_classified` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-11-08_transposed_Kaiju_0_errors_genus_level_subsampled_reads_taxIDs_percent_reads_classified.txt")
View(`2022.11.08_transposed_Kaiju_0_errors_genus_level_subsampled_reads_taxIDs_percent_reads_classified`)

`2022.11.08_melted_transposed_Kaiju_0_errors_genus_level_subsampled_reads_taxIDs_percent_reads_classified` <- reshape2::melt(`2022.11.08_transposed_Kaiju_0_errors_genus_level_subsampled_reads_taxIDs_percent_reads_classified`, id=c("Tax_ID", "Season", "Sequencing", "Species_x_season", "Sample_ID"))

ggplot(data = `2022.11.08_melted_transposed_Kaiju_0_errors_genus_level_subsampled_reads_taxIDs_percent_reads_classified`, aes(x = Species_x_season, y = value)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Host species and sampling month") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + facet_wrap(~ Species_x_season, scales = "free_x") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable == "Azorhizobium" | `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable == "Lactobacillus",], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Host species and sampling month") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + facet_wrap(~ Species.x.season, scales = "free_x") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`, aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Host species and sampling month") + ylab("Percent reads classified (%)") + scale_y_log10(labels = label_log()) + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + facet_wrap(~ Species.x.season, scales = "free_x") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-08_Kaiju_0_errors_all_subsampled_files_genus_level_box_and_violin_logscale_percent_reads_classified` <- ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`, aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Host species and sampling month") + ylab("Percent reads classified (%)") + scale_y_log10(labels = label_log()) + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + facet_wrap(~ Species.x.season, scales = "free_x") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(axis.text = element_text(size = 12.25))

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$Species == "AC",], aes(x = SampleID, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Host species and sampling month") + ylab("Percent reads classified (%)") + scale_y_log10(labels = label_log()) + scale_fill_manual(values = c("#1ec8ff", "#006ad1")) + guides(fill = guide_legend(title = "Host species and sampling month")) + facet_wrap(~ SampleID, scales = "free_x") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$Species == "AR",], aes(x = SampleID, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Host species and sampling month") + ylab("Percent reads classified (%)") + scale_y_log10(labels = label_log()) + scale_fill_manual(values = c("#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + facet_wrap(~ SampleID, scales = "free_x") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = temp_bins_genus_top_7_and_other_strict_bins_count, aes(x = Genus, y = Bins, fill = Genus)) + geom_bar(stat = "identity", colour = 'black') + scale_fill_manual(values=cbbPalette) + theme(text = element_text(family = "Tex Gyre Heroes")) + guides(fill = guide_legend(title = "Genera")) + theme_linedraw() + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) + theme(legend.position = "left") + theme(legend.title = element_text(size = 15), legend.text = element_text(size = 15))

`2022.11.08_0_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-11-08_0_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages.txt")
view(`2022.11.08_0_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`)

ggplot(`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`, aes(x = Species.x.season, y = value, fill = Species.x.season)) +

`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages` <- reshape2::melt(`2022.11.08_0_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`, id=c("Tax_ID", "Species", "Season", "Sequencing", "Species.x.season", "SampleID"))

tic()
`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxons_list_taxons` <- as.list(unique(`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable))
`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Genus", "ACJ.geometric.mean", "ACN.geometric.mean", "ARJ.geometric.mean", "ARN.geometric.mean"))))
for (taxon in `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxons_list_taxons`) {
taxon.name <- paste(taxon)
ACJ.geometric.mean <- exp(mean(log(`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable == taxon.name & `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$Species.x.season == "ACJ",]$value)))
ACN.geometric.mean <- exp(mean(log(`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable == taxon.name & `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$Species.x.season == "ACN",]$value)))
ARJ.geometric.mean <- exp(mean(log(`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable == taxon.name & `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$Species.x.season == "ARJ",]$value)))
ARN.geometric.mean <- exp(mean(log(`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable == taxon.name & `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$Species.x.season == "ARN",]$value)))
`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[nrow(`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`) +1,] <- c(taxon.name, ACJ.geometric.mean, ACN.geometric.mean, ARJ.geometric.mean, ARN.geometric.mean)
`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[,c(2:5)] <- sapply(`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[,c(2:5)], as.numeric)
rm(taxon.name)
rm(ACJ.geometric.mean)
rm(ACN.geometric.mean)
rm(ARJ.geometric.mean)
rm(ARN.geometric.mean)
}
write.table(`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`, file = "kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means.tsv", sep="\t", row.names=FALSE)
toc()

`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means` %>% filter_all(all_vars(.>0))

`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means` <- `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means` %>% filter_all(all_vars(.>0))

`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means` <- `all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means` %>% rowwise() %>% mutate(Log2.plus.one.fold.change.ACJ.to.ACN = log2(((ACN.geometric.mean - ACJ.geometric.mean) / ACJ.geometric.mean) + 1))
`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means` <- `all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means` %>% rowwise() %>% mutate(Log2.plus.one.fold.change.ARJ.to.ARN = log2(((ARN.geometric.mean - ARJ.geometric.mean) / ARJ.geometric.mean) + 1))

`kraken_050_taxa_species_X_season_geometric_means` <- `kraken_050_taxa_species_X_season_geometric_means` %>% rowwise() %>% mutate(Log2.plus.one.fold.change.ARJ.to.ARN = log2(((ARN.geometric.mean - ARJ.geometric.mean) / ARJ.geometric.mean) + 1))

ggplot(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[!is.infinite(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN) & !is.na(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN),], aes(x = Log2.plus.one.fold.change.ACJ.to.ACN, y = reorder(Genus, -Log2.plus.one.fold.change.ACJ.to.ACN), fill = ifelse(Log2.plus.one.fold.change.ACJ.to.ACN < 0, 'red', 'black'))) + geom_col() + theme_linedraw() + ylab("Taxon") + xlab("Log2()+1 change in geometric mean percentage reads\nassigned from June to November") + scale_fill_manual(values = c("#006ad1", "#1ec8ff"), name = "Month enriched", labels = c("November", "June"))

ggplot(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[!is.infinite(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN) & !is.na(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN) & (`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN > 0.75 | `all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN < -0.75),], aes(x = Log2.plus.one.fold.change.ACJ.to.ACN, y = reorder(Genus, -Log2.plus.one.fold.change.ACJ.to.ACN), fill = ifelse(Log2.plus.one.fold.change.ACJ.to.ACN < 0, 'red', 'black'))) + geom_col() + theme_linedraw() + ylab("Genus") + xlab("Log2()+1 change in geometric mean percentage reads\nassigned from June to November") + scale_fill_manual(values = c("#006ad1", "#1ec8ff"), name = "Month enriched", labels = c("November", "June")) + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.y = element_text(face = "italic"))

ggplot(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[!is.infinite(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN) & !is.na(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN) & (`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN > 0.75 | `all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN < -0.75),], aes(x = Log2.plus.one.fold.change.ARJ.to.ARN, y = reorder(Genus, -Log2.plus.one.fold.change.ARJ.to.ARN), fill = ifelse(Log2.plus.one.fold.change.ARJ.to.ARN < 0, 'red', 'black'))) + geom_col() + theme_linedraw() + ylab("Genus") + xlab("Log2()+1 change in geometric mean percentage reads\nassigned from June to November") + scale_fill_manual(values = c("#d16700", "#ff551e"), name = "Month enriched", labels = c("November", "June")) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-08_kaiju_0_errors_genus_level_log2_plus_one_change_geometric_means_ACJ_to_ACN_greater_0_point_75_positive_or_negative_bar_plot` <- ggplot(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[!is.infinite(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN) & !is.na(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN) & (`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN > 0.75 | `all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN < -0.75),], aes(x = Log2.plus.one.fold.change.ACJ.to.ACN, y = reorder(Genus, -Log2.plus.one.fold.change.ACJ.to.ACN), fill = ifelse(Log2.plus.one.fold.change.ACJ.to.ACN < 0, 'red', 'black'))) + geom_col() + theme_linedraw() + ylab("Genus") + xlab("") + scale_fill_manual(values = c("#006ad1", "#1ec8ff"), name = "Month enriched", labels = c("November", "June")) + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.y = element_text(face = "italic"))

`2022-11-08_kaiju_0_errors_genus_level_log2_plus_one_change_geometric_means_ARJ_to_ARN_greater_0_point_75_positive_or_negative_bar_plot` <- ggplot(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[!is.infinite(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN) & !is.na(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN) & (`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN > 0.75 | `all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN < -0.75),], aes(x = Log2.plus.one.fold.change.ARJ.to.ARN, y = reorder(Genus, -Log2.plus.one.fold.change.ARJ.to.ARN), fill = ifelse(Log2.plus.one.fold.change.ARJ.to.ARN < 0, 'red', 'black'))) + geom_col() + theme_linedraw() + ylab("Genus") + xlab("Log2()+1 change in geometric mean percentage reads\nassigned from June to November") + scale_fill_manual(values = c("#d16700", "#ff551e"), name = "Month enriched", labels = c("November", "June")) + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-08_kaiju_0_errors_genus_level_log2_plus_one_change_geometric_means_ACJ_to_ACN_greater_0_point_75_positive_or_negative_bar_plot` / `2022-11-08_kaiju_0_errors_genus_level_log2_plus_one_change_geometric_means_ARJ_to_ARN_greater_0_point_75_positive_or_negative_bar_plot` + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

`2022-11-08_kaiju_0_errors_genus_level_log2_plus_one_change_geometric_means_combined_figures_both_species_seasonal_change_greater_0_point_75_positive_or_negative_bar_plot` <- `2022-11-08_kaiju_0_errors_genus_level_log2_plus_one_change_geometric_means_ACJ_to_ACN_greater_0_point_75_positive_or_negative_bar_plot` / `2022-11-08_kaiju_0_errors_genus_level_log2_plus_one_change_geometric_means_ARJ_to_ARN_greater_0_point_75_positive_or_negative_bar_plot` + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% `all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[!is.infinite(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN) & !is.na(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN) & (`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN > 0.75 | `all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ACJ.to.ACN < -0.75),]$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% `all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[!is.infinite(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN) & !is.na(`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN) & (`all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN > 0.75 | `all_host.x.season_combos_larger_than_0_kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$Log2.plus.one.fold.change.ARJ.to.ARN < -0.75),]$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% (`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means` %>% filter_all(all_vars(.>0.5)))$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

#Thing below is bad, needs fixing
ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% (`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means` %>% filter_all(any_vars(.>1)))$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACJ.geometric.mean > 1 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACN.geometric.mean > 1 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARJ.geometric.mean > 1 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARN.geometric.mean > 1,]$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACJ.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACN.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARJ.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARN.geometric.mean > 0.25,]

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACJ.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACN.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARJ.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARN.geometric.mean > 0.25,]$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACJ.geometric.mean > 1,]$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACN.geometric.mean > 1,]$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARJ.geometric.mean > 1,]$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARN.geometric.mean > 1,]$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-10_kaiju_0_errors_genus_level_box_violin_plots_all_subsampled_reads_log10_percentages_only_genera_geometric_mean_over_0_point_25_any_host_and_season` <- ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACJ.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACN.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARJ.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARN.geometric.mean > 0.25,]$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-10_kaiju_0_errors_genus_level_box_violin_plots_all_subsampled_reads_log10_percentages_only_genera_geometric_mean_over_0_point_25_any_host_and_season` <- ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$variable %in% `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`[`kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACJ.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ACN.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARJ.geometric.mean > 0.25 | `kaiju_subsampled_genus_level_0_errors_species_X_season_geometric_means`$ARN.geometric.mean > 0.25,]$Genus,], aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(text = element_text(size = 12)) + theme(strip.text.x = element_text(size = 11)) + theme(text = element_text(family = "Tex Gyre Heroes"))


`2022-11-10_kaiju_0_error_genus_level_processed_reads_AC_only_box_and_violin_percent_classified_log10scale_facet_sampleID` <- ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$Species == "AC",], aes(x = SampleID, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ SampleID, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-10_kaiju_0_error_genus_level_processed_reads_AR_only_box_and_violin_percent_classified_log10scale_facet_sampleID` <- ggplot(data = `2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`[`2022.11.08_0_melted_transposed_allowed_errors_combined_results_all_processed_subsampled_files_genus_level_results_all_potential_taxID_percentages`$Species == "AR",], aes(x = SampleID, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Genus") + ylab("Percent reads classified (%)") + scale_fill_manual(values = c("#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + scale_y_log10(labels = label_log()) + theme_classic() + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ SampleID, scales = "free_x") + theme(text = element_text(family = "Tex Gyre Heroes"))

2022-11-11
`2022.11.10_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-11-10_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping.tsv")

View(`2022.11.10_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`)

`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping` <- reshape2::melt(`2022.11.10_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`, id=c("SampleID", "Species", "Season", "Sequencing", "Species.x.season", "SampleID.1"))

ggplot(data = `2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`, aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Host species and sample month") + ylab("Mapping reads (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + guides(fill = guide_legend(title = "Host species and sampling month")) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-11-11_samtools_flagstat_percent_mapped_bam_files_not_filtered_all_processed_read_files_to_single_contig_per_bin_concatenated_reference_box_and_violin_figure` <- ggplot(data = `2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`, aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Host species and sample month") + ylab("Mapping reads (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + facet_wrap(~ Sequencing) + guides(fill = guide_legend(title = "Host species and sampling month")) + theme_linedraw() + theme(strip.text.x = element_text(size = 15), text = element_text(family = "Tex Gyre Heroes"))

median(`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`[`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`$Sequencing == "Pilot",]$value)

median(`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`[`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`$Sequencing == "Novogene",]$value)

range(`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`[`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`$Sequencing == "Pilot",]$value)
range(`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`[`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`$Sequencing == "Novogene",]$value)

mean(`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`[`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`$Sequencing == "Pilot",]$value)
mean(`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`[`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`$Sequencing == "Novogene",]$value)

t.test(value ~ Season, data = `2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`[`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`$Species == "AC",])
t.test(value ~ Season, data = `2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`[`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`$Species == "AR",])

t.test(value ~ Sequencing, data = `2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`)
ggboxplot(`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`[`2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`$Species == "AC",], x = "Species.x.season", y = "value", colour = "Species.x.season")

`2022-09-14_var_explained_screeplot_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- ggplot(data = `2022-09-14_var_explained_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`[`2022-09-14_var_explained_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$var_explained > 0.010, ], aes(x = reorder(PC, -var_explained), y = var_explained, colour = "black", fill = ifelse(var_explained > 0.05, 'black', 'red'))) + geom_bar(stat = 'identity') + scale_fill_manual(values = c("#ff1e90", "#f2f3f4")) + scale_colour_manual(values = c("#0f0f0f")) + coord_flip() + theme_linedraw() + ylab("Proportion of variance explained") + xlab("Principal Components") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

`2022-09-14_PCA_PC1_vs_PC2_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` <- ggbiplot::ggbiplot(`2022-09-14_PCA_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Season), color=factor(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`
table(`2022-09-14_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged`$Species_x_season)

`2022-09-14_PCA_PC1_vs_PC2_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` + `2022-09-14_PCA_PC1_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` + `2022-09-14_PCA_PC2_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

`2022-11-11_PCA_subsampled_reads_only_novogene_paired_salmon_RPMs_mapping_single_contig_per_bin_concatenated_fasta_seq_and_bias_adjusted_PC1v2_1v3_2v3` <- `2022-09-14_PCA_PC1_vs_PC2_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` + `2022-09-14_PCA_PC1_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` + `2022-09-14_PCA_PC2_vs_PC3_only_novogene_paired_subsampled_reads_mapped_all_strict_bins_Salmon_TPMs_GC_and_bias_flagged` + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

`2022-09-23_within_AC_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure` <- ggplot(data = `2022-0-23_AC_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AC.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AC seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#1e90ff", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AC.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AC.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#004385", min.segment.length = 0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

`2022-09-21_only_novogene_paired_strict_bin_mapping_salmon_GC_and_seq_bias_flagged_bin_level_host_effect_foldchange_and_qvalue_fig` <- ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue), colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

`2022-09-21_only_novogene_paired_strict_bin_mapping_salmon_GC_and_seq_bias_flagged_bin_level_month_effect_foldchange_and_qvalue_fig` <- ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = June.to.November.log2fold.add.one.change, y = -log10(Season.corrected.pvalue), colour = ifelse((June.to.November.log2fold.add.one.change > 1 & -log10(Season.corrected.pvalue) > -log10(0.05)) | (June.to.November.log2fold.add.one.change < -1 & -log10(Season.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Time effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#b5b5b3", "#0f0f0f")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

2022-11-12

`2022-09-21_only_novogene_paired_strict_bin_mapping_salmon_GC_and_seq_bias_flagged_bin_level_host_effect_foldchange_and_qvalue_fig` <- ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue), colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue), colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Host effect bins, AC to AR") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue >= 0.05 & ! is.infinite(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change) & (`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change >= 1 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1),]

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & ! is.infinite(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change >= 1,]

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & ! is.infinite(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1,]

view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & ! is.infinite(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1,]$Bin)

`2022.11.12_AR_enriched_and_corrected_pvalue_sig_bins_and_families` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-11-12_AR_enriched_and_corrected_pvalue_sig_bins_and_families.txt")
View(`2022.11.12_AR_enriched_and_corrected_pvalue_sig_bins_and_families`)

table(`2022.11.12_AR_enriched_and_corrected_pvalue_sig_bins_and_families`$GTDBTK.family)
as.data.frame(table(`2022.11.12_AR_enriched_and_corrected_pvalue_sig_bins_and_families`$GTDBTK.family))
`2022.11.12_AR_enriched_and_corrected_pvalue_sig_bins_count_of_families` <- as.data.frame(table(`2022.11.12_AR_enriched_and_corrected_pvalue_sig_bins_and_families`$GTDBTK.family))

ggplot(data = `2022.11.12_AR_enriched_and_corrected_pvalue_sig_bins_count_of_families`, aes(x = reorder(Var1, - Freq), y = Freq, fill = Var1)) + geom_bar(stat = "identity")

`2022.11.12_AC_enriched_and_corrected_pvalue_sig_bins_and_families` <- read.delim("~/Google Drive/PhD/2022-Paper_2-Chapter_3/2022-11-12_AC_enriched_and_corrected_pvalue_sig_bins_and_families.txt")
View(`2022.11.12_AC_enriched_and_corrected_pvalue_sig_bins_and_families`)
table(`2022.11.12_AC_enriched_and_corrected_pvalue_sig_bins_and_families`$GTDBTK.family)
as.data.frame(table(`2022.11.12_AC_enriched_and_corrected_pvalue_sig_bins_and_families`$GTDBTK.family))
`2022.11.12_AC_enriched_and_corrected_pvalue_sig_bins_count_of_families` <- as.data.frame(table(`2022.11.12_AC_enriched_and_corrected_pvalue_sig_bins_and_families`$GTDBTK.family))

`2022.11.12_AR_enriched_and_corrected_pvalue_sig_bins_count_of_families`[`2022.11.12_AR_enriched_and_corrected_pvalue_sig_bins_count_of_families`$Var1 %notin% `2022.11.12_AC_enriched_and_corrected_pvalue_sig_bins_count_of_families`$Var1,]
`2022.11.12_AC_enriched_and_corrected_pvalue_sig_bins_count_of_families`[`2022.11.12_AC_enriched_and_corrected_pvalue_sig_bins_count_of_families`$Var1 %notin% `2022.11.12_AR_enriched_and_corrected_pvalue_sig_bins_count_of_families`$Var1,]

`2022.11.12_AR_enriched_and_corrected_pvalue_sig_bins_count_of_families`$Var1

view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & ! is.infinite(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1,])

`2022.11.12_AC_enriched_and_corrected_pvalue_sig_bins_and_families`

range(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == "hypothetical.protein" & ! is.infinite(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$value),]$value)

table(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable)
`2022-11-12_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_total_counts_per_product` <- as.data.frame(table(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable))

tic()
`products_sums_and_arithmetic_means` <- data.frame(matrix(ncol = 11, nrow = 0, dimnames = list(NULL, c("Product", "Sum.all.bins", "Sum.AC.enriched.vs", "Sum.AR.enriched.vs", "Sum.AC.within", "Sum.AR.within", "Mean.all.bins", "Mean.AC.enriched.vs", "Mean.AR.enriched.vs", "Mean.AC.within", "Mean.AR.within"))))
for (product in `2022.10.16.Unique_product_names`){
product.name <- paste(product)
all.sum <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
AC.vs.sum <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs",]$value)
AR.vs.sum <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs",]$value)
AC.within.sum <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect",]$value)
AR.within.sum <- sum(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect",]$value)
all.mean <- mean(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name,]$value)
AC.vs.mean <- mean(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AC_enriched_vs",]$value)
AR.vs.mean <- mean(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "AR_enriched_vs",]$value)
AC.within.mean <- mean(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AC_effect",]$value)
AR.within.mean <- mean(`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`[`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$variable == product.name & `2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`$Bin_status == "in_AR_effect",]$value)
`products_sums_and_arithmetic_means`[nrow(`products_sums_and_arithmetic_means`) +1,] <- c(product.name, all.sum, AC.vs.sum, AR.vs.sum, AC.within.sum, AR.within.sum, all.mean, AC.vs.mean, AR.vs.mean, AC.within.mean, AR.within.mean)
`products_sums_and_arithmetic_means`[,c(2:11)] <- sapply(`products_sums_and_arithmetic_means`[,c(2:11)], as.numeric)
rm(all.sum)
rm(AC.vs.sum)
rm(AR.vs.sum)
rm(AC.within.sum)
rm(AR.within.sum)
rm(all.mean)
rm(AC.vs.mean)
rm(AR.vs.mean)
rm(AC.within.mean)
rm(AR.within.mean)
}
view(`products_sums_and_arithmetic_means`)
write.table(`products_sums_and_arithmetic_means`, file = "2022-11-13_dfast_single_highest_scoring_unique_products_strict_bins_sums_and_arithmetic_means_table.tsv", sep="\t", row.names=FALSE)
toc()

tic()
`test_products_hypergeometric_pvalues_df`[nrow(`test_products_hypergeometric_pvalues_df`) +1,] <- c(product.name, noise.bins.product.pvalue, AC.enriched.bins.product.pvalue, AR.enriched.bins.product.pvalue, within.AC.bins.product.pvalue, within.AR.bins.product.pvalue)
`test_products_hypergeometric_pvalues_df`[,c(2:6)] <- sapply(`test_products_hypergeometric_pvalues_df`[,c(2:6)], as.numeric)
}
view(test_products_hypergeometric_pvalues_df)
`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status` <- test_products_hypergeometric_pvalues_df
write.table(`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, file = "2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status.tsv", sep="\t", row.names=FALSE)
`2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status` <- reshape2::melt(`2022-10-25_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, id=c("Product"))
write.table(`2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status`, file = "2022-10-25_melted_all_single_unique_products_dfast_strict_bins_hypergeometric_Pvalues_uncorrected_by_bin_enrichment_status.tsv", sep="\t", row.names=FALSE)

toc()

`2022.10.16_melted_transposed_combined_table_all_good_bins_all_unique_single_products_single_hypothetical_protein_simple_counts`
`2022-10-28_working1`
`2022-10-28_melted_working1` <- reshape2::melt(`2022-10-28_working1`, id=c("Product"))
ggplot(data = `2022-10-28_melted_working1`[grepl(glob2rx("BH.corrected*"), `2022-10-28_melted_working1`$variable),], aes(x = value, y = variable, colour = variable, fill = variable)) + geom_jitter() + theme_linedraw()

ggplot(data = `products_sums_and_arithmetic_means`, aes(x = "Products", y = Sum.all.bins)) + geom_violin(fill = "#8d1eff") + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + ylab("Sum of counts of products across\nall bins on Log10 scale") + xlab("") + scale_y_log10(labels = label_log()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

`melted_products_sums_and_arithmetic_means` <- reshape2::melt(`products_sums_and_arithmetic_means`, id = c("Product"))

`melted_products_sums_and_arithmetic_means`

ggplot(data = `melted_products_sums_and_arithmetic_means`[grepl(glob2rx("Sum*"), `melted_products_sums_and_arithmetic_means`$variable),], aes(x = variable, y = value, fill = variable)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + ylab("Sum of counts of products across\nbins on Log10 scale") + xlab("") + scale_y_log10(labels = label_log()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

ggplot(data = `melted_products_sums_and_arithmetic_means`[grepl(glob2rx("Sum*"), `melted_products_sums_and_arithmetic_means`$variable),], aes(x = variable, y = value, fill = variable)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + ylab("Sum of counts of products across\nbins on Log10 scale") + xlab("") + scale_y_log10(labels = label_log()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 12.5)) + scale_fill_manual(values = c("#8d1eff", "#1e90ff", "#ff8d1e", "#0a0dff", "#bd5d00")) + theme(legend.position = "none")

`2022-11-13_DFAST_unique_highest_scoring_products_sum_counts_bins_by_enrichment_type_of_bins` <- ggplot(data = `melted_products_sums_and_arithmetic_means`[grepl(glob2rx("Sum*"), `melted_products_sums_and_arithmetic_means`$variable),], aes(x = variable, y = value, fill = variable)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + ylab("Sum of counts of products across\nbins on Log10 scale") + xlab("") + scale_y_log10(labels = label_log()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 12.5)) + scale_fill_manual(values = c("#8d1eff", "#1e90ff", "#ff8d1e", "#0a0dff", "#bd5d00")) + theme(legend.position = "none")

ggplot(data = `2022-10-28_melted_working1`[grepl(glob2rx("BH.corrected*"), `2022-10-28_melted_working1`$variable),], aes(x = value, y = variable, colour = variable)) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 12.5)) + xlab("BH corrected P-value") + ylab("Bin enrichment status") + scale_colour_manual(values = c("#8d1eff", "#1e90ff", "#ff8d1e", "#0a0dff", "#bd5d00")) + theme(legend.position = "none") + geom_vline(xintercept=c(0.05), col="#c82b08", linetype = "dashed")

`2022-11-13_hypergeometric_pvalues_plot_single_unique_highest_scoring_products_DFAST_strict_bins_categorised_by_bin_enrichment_type` <- ggplot(data = `2022-10-28_melted_working1`[grepl(glob2rx("BH.corrected*"), `2022-10-28_melted_working1`$variable),], aes(x = value, y = variable, colour = variable)) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 12.5)) + xlab("BH corrected P-value") + ylab("Bin enrichment status") + scale_colour_manual(values = c("#8d1eff", "#1e90ff", "#ff8d1e", "#0a0dff", "#bd5d00")) + theme(legend.position = "none") + geom_vline(xintercept=c(0.05), col="#c82b08", linetype = "dashed")

`2022-10-28_working1`[`2022-10-28_working1`$BH.corrected.AR.enriched.versus.bins.Pvalue <= 0.05,]

`2022-09-21_only_novogene_paired_strict_bin_mapping_salmon_GC_and_seq_bias_flagged_bin_level_month_effect_foldchange_and_qvalue_fig` <- ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = June.to.November.log2fold.add.one.change, y = -log10(Season.corrected.pvalue), colour = ifelse((June.to.November.log2fold.add.one.change > 1 & -log10(Season.corrected.pvalue) > -log10(0.05)) | (June.to.November.log2fold.add.one.change < -1 & -log10(Season.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Time effect bins, June to November") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#b5b5b3", "#0f0f0f")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & (`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change >= 1 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1),]

view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & (`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change >= 1 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1),]

table(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & (`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change >= 1 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1),]$GTDB.TK.taxonomy)

view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue > 0.05,])

as.data.frame(table(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05,]$GTDB.TK.taxonomy))

as.data.frame(table(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 & (`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < 1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > -1),]$GTDB.TK.taxonomy))

nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05,]))

view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1,])

view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & (`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > -1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < 1),])

#Counts below are for Lachnospiraceae
#AC enriched - 51
view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Lachnospiraceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1,])
nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Lachnospiraceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1,]))

#AR enriched - 2
view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change >= 1,])
nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Lachnospiraceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change >= 1,]))


#Not statistically different - 41
view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue > 0.05,])
nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Lachnospiraceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue > 0.05,]))


#Enriched but not enough AC - 8
view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > -1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < 0,])
nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Lachnospiraceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & (`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < 0 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > -1),]))


#Enriched but not enough AR - 2
view(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < 1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 0,])
nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Lachnospiraceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & (`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 0 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < 1),]))


nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Muribaculaceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1,]))

nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Muribaculaceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change >= 1,]))

nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Muribaculaceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue > 0.05,]))

table(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Muribaculaceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1,]$GTDB.TK.taxonomy)

nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Anaeroplasmataceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue >= 0.05,]))
nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue <= 0.05 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change <= -1,]))

nrow(as.data.frame(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[grepl(glob2rx("*f__Lachnospiraceae*"), `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$GTDB.TK.taxonomy) & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05,]))

2022-11-18
nrow(as.data.frame(`products_sums_and_arithmetic_means`[`products_sums_and_arithmetic_means`$Mean.all.bins >= 1 & `products_sums_and_arithmetic_means`$Mean.all.bins <= 10,]))
tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-11-14.RData")
toc()

2022-11-27
setwd("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work")
`2022.04.08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size` <- read.csv("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-04-08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size.csv")
View(`2022.04.08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size`)

checkm_isolates_plot1 <- ggplot(data = `2022.04.08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size`, aes(x = Completeness)) + geom_histogram(binwidth = 1, fill = "#f2f3f4", color = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Completeness (%)") + ylab("Frequency") + geom_vline(xintercept=c(80), col="#1e90ff", linetype = "dashed", size = 1)

checkm_isolates_plot2 <- ggplot(data = `2022.04.08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size`, aes(x = Contamination)) + geom_histogram(binwidth = 1, fill = "#848482", color = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Contamination (%)") + ylab("Frequency") + geom_vline(xintercept=c(5), col="#ff8d1e", linetype = "dashed", size = 1)

checkm_isolates_plot3 <- ggplot(data = `2022.04.08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size`, aes(x = Completeness, y = Contamination, colour = ifelse(Completeness > 80 & Contamination < 5, 'black', 'red'))) + geom_jitter(size = 2.5) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Completeness (%)") + ylab("Contamination (%)") + geom_vline(xintercept=c(80), col="#1e90ff", linetype = "dashed", size = 1.35) + geom_hline(yintercept=c(5), col="#ff8d1e", linetype = "dashed", size = 1.35) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba"))

checkm_isolates_plot1 + checkm_isolates_plot2 + checkm_isolates_plot3 + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') + plot_annotation(title = 'CheckM analysis of 35 isolate assembly fasta files')

`2022-11-27.all_isolates_500bp_min_contigs_CheckM_plots_combined_figure` <- checkm_isolates_plot1 + checkm_isolates_plot2 + checkm_isolates_plot3 + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') + plot_annotation(title = 'CheckM analysis of 35 isolate assembly fasta files')

`2022.11.27_all_isolates_some_checkm_results_plus_host_species` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-11-27_all_isolates_some_checkm_results_plus_host_species.txt")

range(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[! `2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Genome.size.bp)
median(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[! `2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Genome.size.bp)
mean(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[! `2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Genome.size.bp)

range(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Genome.size.bp)
median(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Genome.size.bp)
mean(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Genome.size.bp)

range(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AC",]$Genome.size.bp)
median(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AC",]$Genome.size.bp)
mean(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AC",]$Genome.size.bp)

range(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AR",]$Genome.size.bp)
median(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AR",]$Genome.size.bp)
mean(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AR",]$Genome.size.bp)

`2022.11.27.Melted_all_isolates_some_checkm_results_plus_host_species` <- reshape2::melt(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`, id=c("Isolate.ID", "Isolate.name", "Host.species"))

ggplot(data = `2022.11.27.Melted_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27.Melted_all_isolates_some_checkm_results_plus_host_species`$variable == "Number.contigs",], aes(x = Host.species, y = value, fill = Host.species)) + xlab("Host species or genus") + ylab ("Number of contigs in assembly") + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.35) + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + facet_wrap(~ Host.species, scales = "free_x")

`2022.11.27.all_isolates_500bp_min_contig_size_SPADES_assemblies_number_contigs_box_and_violin_plot_facet_by_host_species_or_genus` <- ggplot(data = `2022.11.27.Melted_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27.Melted_all_isolates_some_checkm_results_plus_host_species`$variable == "Number.contigs",], aes(x = Host.species, y = value, fill = Host.species)) + xlab("Host species or genus") + ylab ("Number of contigs in assembly") + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.35) + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + facet_wrap(~ Host.species, scales = "free_x")

2022-11-27
range(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[! `2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Number.contigs)
median(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[! `2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Number.contigs)
mean(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[! `2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Number.contigs)

range(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Number.contigs)
median(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Number.contigs)
mean(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Number.contigs)

range(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AC",]$Number.contigs)
median(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AC",]$Number.contigs)
mean(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AC",]$Number.contigs)

range(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AR",]$Number.contigs)
median(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AR",]$Number.contigs)
mean(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AR",]$Number.contigs)

checkm_isolates_plot3 <- ggplot(data = `2022.11.27_all_isolates_some_checkm_results_plus_host_species`, aes(x = Completeness, y = Contamination, label = Isolate.ID, colour = ifelse(Completeness > 80 & Contamination < 5, 'black', 'red'))) + geom_jitter(size = 1.5) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Completeness (%)") + ylab("Contamination (%)") + geom_vline(xintercept=c(80), col="#1e90ff", linetype = "dashed", size = 1.35) + geom_hline(yintercept=c(5), col="#ff8d1e", linetype = "dashed", size = 1.35) + geom_label_repel(aes(label = ifelse(Contamination > 5, as.character(Isolate.ID), ''), hjust = 0, vjust = 0), colour = "#363b40", min.segment.length = 0) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba"))

checkm_isolates_plot1 + checkm_isolates_plot2 + checkm_isolates_plot3 + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') + plot_annotation(title = 'CheckM analysis of 35 isolate assembly fasta files')

`2022-11-27.all_isolates_500bp_min_contigs_CheckM_plots_combined_figure` <- checkm_isolates_plot1 + checkm_isolates_plot2 + checkm_isolates_plot3 + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') + plot_annotation(title = 'CheckM analysis of 35 isolate assembly fasta files')

`2022.11.27_GTDB.TK_genera_isolates_passing_QC_and_picked_to_use_when_ANI_over_99_point_9` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-11-27_GTDB-TK_genera_isolates_passing_QC_and_picked_to_use_when_ANI_over_99_point_9.txt")
View(`2022.11.27_GTDB.TK_genera_isolates_passing_QC_and_picked_to_use_when_ANI_over_99_point_9`)

table(`2022.11.27_GTDB.TK_genera_isolates_passing_QC_and_picked_to_use_when_ANI_over_99_point_9`$Genus)
as.data.frame(table(`2022.11.27_GTDB.TK_genera_isolates_passing_QC_and_picked_to_use_when_ANI_over_99_point_9`$Genus))

ggplot(data = as.data.frame(table(`2022.11.27_GTDB.TK_genera_isolates_passing_QC_and_picked_to_use_when_ANI_over_99_point_9`$Genus)), aes(x = reorder(Var1, - Freq), y = Freq, fill = Var1)) + geom_bar(stat = 'identity', colour = "#0f0f0f") + theme_linedraw() + theme(axis.text.x = element_text(angle = 45, hjust=1)) + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + xlab("Genera") + ylab("Number of isolates classified") + scale_fill_manual(values = c("#2D3047", "#C04ABC", "#E0A458", "#419D78", "#FFDBB5", "#a0a7ae"), name = "GTDB-TK assigned genus")

`2022-11-27_all_isolates_passing_QC_and_picked_if_ANI_similar_GTDB-TK_genus_level_counts_per_genus_bar_chart` <- ggplot(data = as.data.frame(table(`2022.11.27_GTDB.TK_genera_isolates_passing_QC_and_picked_to_use_when_ANI_over_99_point_9`$Genus)), aes(x = reorder(Var1, - Freq), y = Freq, fill = Var1)) + geom_bar(stat = 'identity', colour = "#0f0f0f") + theme_linedraw() + theme(axis.text.x = element_text(angle = 45, hjust=1)) + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + xlab("Genera") + ylab("Number of isolates classified") + scale_fill_manual(values = c("#2D3047", "#C04ABC", "#E0A458", "#419D78", "#FFDBB5", "#a0a7ae"), name = "GTDB-TK assigned genus") 

2022-11-28
readDelta <- function(deltafile){
  lines = scan(deltafile, 'a', sep='\n', quiet=TRUE)
  lines = lines[-1]
  lines.l = strsplit(lines, ' ')
  lines.len = lapply(lines.l, length) %>% as.numeric
  lines.l = lines.l[lines.len != 1]
  lines.len = lines.len[lines.len != 1]
  head.pos = which(lines.len == 4)
  head.id = rep(head.pos, c(head.pos[-1], length(lines.l)+1)-head.pos)
  mat = matrix(as.numeric(unlist(lines.l[lines.len==7])), 7)
  res = as.data.frame(t(mat[1:5,]))
  colnames(res) = c('rs','re','qs','qe','error')
  res$qid = unlist(lapply(lines.l[head.id[lines.len==7]], '[', 2))
  res$rid = unlist(lapply(lines.l[head.id[lines.len==7]], '[', 1)) %>% gsub('^>', '', .)
  res$strand = ifelse(res$qe-res$qs > 0, '+', '-')
  res
}
mumgp = readDelta("13E_S5_Nucmer_against_16aD_S27.delta")
filterMum <- function(df, minl=1000, flanks=1e4){
    coord = df %>% filter(abs(re-rs)>minl) %>% group_by(qid, rid) %>%
        summarize(qsL=min(qs)-flanks, qeL=max(qe)+flanks, rs=median(rs)) %>%
        ungroup %>% arrange(desc(rs)) %>%
        mutate(qid=factor(qid, levels=unique(qid))) %>% select(-rs)
    merge(df, coord) %>% filter(qs>qsL, qe<qeL) %>%
        mutate(qid=factor(qid, levels=levels(coord$qid))) %>% select(-qsL, -qeL)
}
mumgp.filt = filterMum(mumgp, minl=1e4)
mumgp.filt %>% head %>% kable

ggplot(mumgp, aes(x=rs, xend=re, y=similarity, yend=similarity)) + geom_segment() +
    theme_bw() + xlab('reference sequence') + ylab('similarity') + ggtitle('All contigs') +
    ylim(0,1)

ggplot(mumgp[mumgp], aes(x=rs, xend=re, y=similarity, yend=similarity)) + geom_segment() +
    theme_bw() + xlab('reference sequence') + ylab('similarity') + ggtitle('All contigs') +
    ylim(0,1)
)

trial_mummerplot=read.table("16aF_S29_Nucmer_against_41B_S7_TRIAL.coords.plot")
ggplot(data = trial_mummerplot[! is.na(trial_mummerplot$V1),], aes(x = V1, y = V2)) + geom_point()


trial_mummerplot=read.table("trial_pt2.plot")
ggplot(data = trial_mummerplot, aes(x = V1, y = V2)) + geom_point()

trial_mummerplot=read.table("trial_pt3.plot")
ggplot(data = trial_mummerplot, aes(x = V1, y = V2)) + geom_line()

`39B_S15_MUMmer_against_18A_S9_forward_reads` <- read.table("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/39B_S15_MUMmer_against_18A_S9_forward_reads.plot", quote="\"", comment.char="")
View(`39B_S15_MUMmer_against_18A_S9_forward_reads`)
ggplot(data = `39B_S15_MUMmer_against_18A_S9_forward_reads`, aes(x = V1, y = V2, colour = "dodger blue")) + geom_point() + theme_linedraw()

testing.coverage <- read.table("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/testing.coverage.plot", quote="\"", comment.char="")
View(testing.coverage)
ggplot(data = testing.coverage, aes(x = V1, y = V3, colour = ifelse(V3 > 90, 'black', 'red'))) + geom_point() + xlab("Reference genome position") + ylab("Percent coverage by query (%)") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba")) + guides(colour = "none") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

`39B_S15_MUMmer_against_18A_S9_forward_reads_coverage` <- read.table("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/39B_S15_MUMmer_against_18A_S9_forward_reads_coverage.plot", quote="\"", comment.char="")
View(`39B_S15_MUMmer_against_18A_S9_forward_reads_coverage`)
ggplot(data = `39B_S15_MUMmer_against_18A_S9_forward_reads_coverage`, aes(x = V1, y = V3, colour = ifelse(V3 > 90, 'black', 'red'))) + geom_point() + xlab("Reference genome position") + ylab("Percent coverage by query (%)") + scale_colour_manual(values = c("#0f0f0f", "#737df4")) + guides(colour = "none") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

`ZUKSWANG_MUMmer_against_ALBATROSS_forward_reads_coverage` <- read.table("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/ZUKSWANG_MUMmer_against_ALBATROSS_forward_reads_coverage.plot", quote="\"", comment.char="")
`ZUKSWANG_MUMmer_against_ALBATROSS_forward_reads_coverage_plot_figure` <- ggplot(data = `ZUKSWANG_MUMmer_against_ALBATROSS_forward_reads_coverage`, aes(x = V1, y = V3, colour = ifelse(V3 >= 90, 'black', 'red'))) + geom_point() + xlab("ALBATROSS assembly position") + ylab("Percent similarity with\nZUKSWANG assembly (%)") + scale_colour_manual(values = c("#0f0f0f", "#737df4")) + ggtitle("Percentage similarity across alignment regions between\nisolates: ZUKSWANG to ALBATROSS") + guides(colour = "none") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))
ggsave(filename="/Users/osbourne/Google lDrive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-11-28_NUCMER_alignment_coverage_plot_ZUKSWANG_against_ALBATROSS.png", `ZUKSWANG_MUMmer_against_ALBATROSS_forward_reads_coverage_plot_figure`)
ggsave(filename="/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-11-28_NUCMER_alignment_coverage_plot_ZUKSWANG_against_ALBATROSS.svg", `ZUKSWANG_MUMmer_against_ALBATROSS_forward_reads_coverage_plot_figure`)

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-Paper_2-Chapter_3/2022_10_22_all_samples_raw_versus_processed_percent_classified_metaphlan_fig.svg", `2022_10_22_all_samples_raw_versus_processed_percent_classified_metaphlan_fig`)

source("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-11-28_Rscript_for_decent_ggplot_coverage_figures_ANI_similarities_acomys_isolates_over_99_point_9_percent_rscript.R", encoding = 'UTF-8')

`13B_S2_MUMmer_against_39A_S14_forward_reads_coverage_plot_figure`
`13D_S4_MUMmer_against_13E_S5_forward_reads_coverage_plot_figure`
`13D_S4_MUMmer_against_16bD_S22_forward_reads_coverage_plot_figure`
`13D_S4_MUMmer_against_18B_S10_forward_reads_coverage_plot_figure`
`13E_S5_MUMmer_against_13D_S4_forward_reads_coverage_plot_figure`
`13E_S5_MUMmer_against_16bD_S22_forward_reads_coverage_plot_figure`
`13E_S5_MUMmer_against_18B_S10_forward_reads_coverage_plot_figure`
`16bD_S22_MUMmer_against_13D_S4_forward_reads_coverage_plot_figure`
`16bD_S22_MUMmer_against_13E_S5_forward_reads_coverage_plot_figure`
`18A_S9_MUMmer_against_39B_S15_forward_reads_coverage_plot_figure`
`18B_S10_MUMmer_against_13D_S4_forward_reads_coverage_plot_figure`
`18B_S10_MUMmer_against_13E_S5_forward_reads_coverage_plot_figure`
`39A_S14_MUMmer_against_13B_S2_forward_reads_coverage_plot_figure`
`39B_S15_MUMmer_against_18A_S9_forward_reads_coverage_plot_figure`

`39B_S15_MUMmer_against_18A_S9_forward_reads` <- read.table("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/39B_S15_MUMmer_against_18A_S9_forward_reads.plot", quote="\"", comment.char="")
View(`39B_S15_MUMmer_against_18A_S9_forward_reads`)

ggplot(data = `39B_S15_MUMmer_against_18A_S9_forward_reads`, aes(x = V1, y = V2, colour = ifelse(V1 - V2 == 0, 'black', 'red'))) + geom_point() + scale_colour_manual(values = c("#0f0f0f", "#737df4")) +  guides(colour = "none") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `this_shite`, aes(x = V1, y = V2, colour = ifelse(V1 - V2 == 0, 'black', 'red'))) + geom_point() + scale_colour_manual(values = c("#0f0f0f", "#737df4")) +  guides(colour = "none") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

`ZUKSWANG_MUMmer_against_ALBATROSS_forward_filtered_reads_coverage` <- read.table("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/ZUKSWANG_MUMmer_against_ALBATROSS_forward_reads_filtered_coverage.plot", quote="\"", comment.char="")
`ZUKSWANG_MUMmer_against_ALBATROSS_forward_reads_filtered_coverage_plot_figure` <- ggplot(data = `ZUKSWANG_MUMmer_against_ALBATROSS_forward_filtered_reads_coverage`, aes(x = V1, y = V3, colour = ifelse(V3 <= 90, 'black', 'red'))) + geom_point() + xlab("ALBATROSS assembly position") + ylab("Percent similarity with\nZUKSWANG assembly (%)") + scale_colour_manual(values = c("#ff8d1e", "#0f0f0f")) + ggtitle("Percentage similarity across alignment regions between\nisolates: ZUKSWANG to ALBATROSS") + guides(colour = "none") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))
`ZUKSWANG_MUMmer_against_ALBATROSS_forward_filtered_reads_delta` <- read.table("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/ZUKSWANG_MUMmer_against_ALBATROSS_filtered_delta.plot", quote="\"", comment.char="")
`ZUKSWANG_MUMmer_against_ALBATROSS_forward_reads_filtered_delta_plot_figure` <- ggplot(data = `ZUKSWANG_MUMmer_against_ALBATROSS_forward_filtered_reads_delta`, aes(x = V1, y = V2, colour = ifelse(V3 <= 90, 'black', 'red'))) + geom_point() + xlab("ALBATROSS assembly position") + ylab("ZUKSWANG assembly position") + scale_colour_manual(values = c("#1e90ff", "#0f0f0f")) + ggtitle("Alignment between\nisolates: ZUKSWANG to ALBATROSS") + guides(colour = "none") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes"))

rm(list=ls(pattern='*MUMmer*'))
source("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-11-28_Rscript_for_decent_ggplot_coverage_figures_ANI_similarities_acomys_isolates_over_99_point_9_percent_rscript.R", encoding = 'UTF-8')

`13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure` + `13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`

2022-11-30
`13B_S2_MUMmer_against_39A_S14_forward_reads_filtered_delta_plot_figure`
`13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`
`13D_S4_MUMmer_against_16bD_S22_forward_reads_filtered_delta_plot_figure`
`13D_S4_MUMmer_against_18B_S10_forward_reads_filtered_delta_plot_figure`
`13E_S5_MUMmer_against_13D_S4_forward_reads_filtered_delta_plot_figure`
`13E_S5_MUMmer_against_16bD_S22_forward_reads_filtered_delta_plot_figure`
`13E_S5_MUMmer_against_18B_S10_forward_reads_filtered_delta_plot_figure`
`16bD_S22_MUMmer_against_13D_S4_forward_reads_filtered_delta_plot_figure`
`16bD_S22_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`
`18A_S9_MUMmer_against_39B_S15_forward_reads_filtered_delta_plot_figure`
`18B_S10_MUMmer_against_13D_S4_forward_reads_filtered_delta_plot_figure`
`18B_S10_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`
`39A_S14_MUMmer_against_13B_S2_forward_reads_filtered_delta_plot_figure`
`39B_S15_MUMmer_against_18A_S9_forward_reads_filtered_delta_plot_figure`

grid.arrange(`13B_S2_MUMmer_against_39A_S14_forward_reads_filtered_coverage_plot_figure`, `13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`, `13D_S4_MUMmer_against_16bD_S22_forward_reads_filtered_coverage_plot_figure`, `13D_S4_MUMmer_against_18B_S10_forward_reads_filtered_coverage_plot_figure`, nrow = 2)

`13E_S5_MUMmer_against_13D_S4_forward_reads_filtered_coverage_plot_figure`
`13E_S5_MUMmer_against_16bD_S22_forward_reads_filtered_coverage_plot_figure`
`13E_S5_MUMmer_against_18B_S10_forward_reads_filtered_coverage_plot_figure`
`16bD_S22_MUMmer_against_13D_S4_forward_reads_filtered_coverage_plot_figure`
`16bD_S22_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`
`18A_S9_MUMmer_against_39B_S15_forward_reads_filtered_coverage_plot_figure`
`18B_S10_MUMmer_against_13D_S4_forward_reads_filtered_coverage_plot_figure`
`18B_S10_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`
`39A_S14_MUMmer_against_13B_S2_forward_reads_filtered_coverage_plot_figure`
`39B_S15_MUMmer_against_18A_S9_forward_reads_filtered_coverage_plot_figure`

grid.arrange(`13B_S2_MUMmer_against_39A_S14_forward_reads_filtered_delta_plot_figure`,
`13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`,
`13D_S4_MUMmer_against_16bD_S22_forward_reads_filtered_delta_plot_figure`,
`13D_S4_MUMmer_against_18B_S10_forward_reads_filtered_delta_plot_figure`,
`13E_S5_MUMmer_against_13D_S4_forward_reads_filtered_delta_plot_figure`,
`13E_S5_MUMmer_against_16bD_S22_forward_reads_filtered_delta_plot_figure`,
`13E_S5_MUMmer_against_18B_S10_forward_reads_filtered_delta_plot_figure`, nrow = 4
)

grid.arrange(`13B_S2_MUMmer_against_39A_S14_forward_reads_filtered_coverage_plot_figure`,
`13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`,
`13D_S4_MUMmer_against_16bD_S22_forward_reads_filtered_coverage_plot_figure`,
`13D_S4_MUMmer_against_18B_S10_forward_reads_filtered_coverage_plot_figure`,
`13E_S5_MUMmer_against_13D_S4_forward_reads_filtered_coverage_plot_figure`,
`13E_S5_MUMmer_against_16bD_S22_forward_reads_filtered_coverage_plot_figure`,
`13E_S5_MUMmer_against_18B_S10_forward_reads_filtered_coverage_plot_figure`, nrow = 4
)

grid.arrange(`13B_S2_MUMmer_against_39A_S14_forward_reads_filtered_delta_plot_figure`, `13B_S2_MUMmer_against_39A_S14_forward_reads_filtered_coverage_plot_figure`, nrow = 1)
grid.arrange(`13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`, `13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`, nrow = 1)
grid.arrange(`13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`, `13D_S4_MUMmer_against_16bD_S22_forward_reads_filtered_coverage_plot_figure`, nrow=1)
grid.arrange()

2022-12-03

grid.arrange(`13B_S2_MUMmer_against_39A_S14_forward_reads_filtered_delta_plot_figure`,
`13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`,
`13D_S4_MUMmer_against_16bD_S22_forward_reads_filtered_delta_plot_figure`,
`13D_S4_MUMmer_against_18B_S10_forward_reads_filtered_delta_plot_figure`, nrow = 2
)

`13B_S2_MUMmer_against_39A_S14_forward_reads_filtered_delta_plot_figure`
`13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`
`13D_S4_MUMmer_against_16bD_S22_forward_reads_filtered_delta_plot_figure`
`13D_S4_MUMmer_against_18B_S10_forward_reads_filtered_delta_plot_figure`
`13E_S5_MUMmer_against_13D_S4_forward_reads_filtered_delta_plot_figure`
`13E_S5_MUMmer_against_16bD_S22_forward_reads_filtered_delta_plot_figure`
`13E_S5_MUMmer_against_18B_S10_forward_reads_filtered_delta_plot_figure`
`16bD_S22_MUMmer_against_13D_S4_forward_reads_filtered_delta_plot_figure`
`16bD_S22_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`
`18A_S9_MUMmer_against_39B_S15_forward_reads_filtered_delta_plot_figure`
`18B_S10_MUMmer_against_13D_S4_forward_reads_filtered_delta_plot_figure`
`18B_S10_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`
`39A_S14_MUMmer_against_13B_S2_forward_reads_filtered_delta_plot_figure`
`39B_S15_MUMmer_against_18A_S9_forward_reads_filtered_delta_plot_figure`

#Moving on from this for now
`2022.12.02_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-12-02_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies.tsv")
View(`2022.12.02_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`)

`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-12-02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies.txt")
View(`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`)

`2022.12.03.melted.transposed.isolates_salmon` <- reshape2::melt(`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`, id=c("Name", "Host", "Month", "Host.x.month"))
`2022.12.03.melted.transposed.isolates_salmon`$variable <- gsub("X", "", as.character(`2022.12.03.melted.transposed.isolates_salmon`$variable))
`2022.12.03.Salmon_isolates_novogene_paired_mapping_Acomys_russatus_derived_isolates` <- ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`[grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable), ], aes(x = variable, y = value, fill = Host.x.month, shape = Host.x.month)) + geom_violin() + ylab("RPM for assembly on Log10 scale") + xlab("") + scale_y_log10(labels = label_log()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 11)) + scale_fill_manual(values = c("#1e90ff", "#002d59", "#ff8d1e", "#934a00"), name = "Read host species\nand sample month") + facet_wrap(~ variable, scales = "free_x")
`2022.12.03.Salmon_isolates_novogene_paired_mapping_Acomys_cahirinus_derived_isolates` <- ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`[! grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & ! grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & ! grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable), ], aes(x = variable, y = value, fill = Host.x.month, shape = Host.x.month)) + geom_violin() + ylab("RPM for assembly on Log10 scale") + xlab("") + scale_y_log10(labels = label_log()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 11)) + scale_fill_manual(values = c("#1e90ff", "#002d59", "#ff8d1e", "#934a00"), name = "Read host species\nand sample month") + facet_wrap(~ variable, scales = "free_x")

`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AC",]$value

as.numeric(nrow(as.data.frame(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$value >= 100 & `2022.12.03.melted.transposed.isolates_salmon`$value <= 1000,])))

ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`, aes(x = Host, y = value, fill = Host)) + geom_violin() +  geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + scale_y_log10(labels = label_log()) + theme_linedraw() +xlab("Acomys species of ") + theme(text = element_text(family = "Tex Gyre Heroes")) + scale_fill_manual(values = c("#1e90ff", "#ff8d1e"), name = "Read host species") + facet_wrap(~ variable, scales = "free_x")

#Cahirinus to all
range(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AC",]$value)
median(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AC",]$value)
mean(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AC",]$value)
#Russatus to all
range(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AR",]$value)
median(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AR",]$value)
mean(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AR",]$value)

#Cahirinus reads to cahrinus assemblies
range(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AC" & (! grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & ! grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & !grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)
median(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AC" & (! grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & ! grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & !grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)
mean(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AC" & (! grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & ! grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & !grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)
#Russatus reads to russatus assemblies
range(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AR" & (grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)
median(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AR" & (grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)
mean(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AR" & (grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)
#Cahirinus reads to russatus assemblies
range(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AC" & (grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)
median(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AC" & (grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)
mean(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AC" & (grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)
#Russatus reads to cahirinus assemblies
range(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AR" & (! grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & ! grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & !grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)
median(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AR" & (! grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & ! grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & !grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)
mean(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$Host == "AR" & (! grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & ! grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & !grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable)),]$value)


`2022.12.03.isolates_salmon_PCA` <- prcomp(`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`[,c(2:26)], center = TRUE, scale = FALSE)
`2022.12.03.isolates_salmon_PCA_var_explained` <- data.frame(PC = paste0("PC", 1:25), var_explained = (`2022.12.03.isolates_salmon_PCA`$sdev)^2/sum((`2022.12.03.isolates_salmon_PCA`$sdev)^2))
`2022.12.03.isolates_salmon_PCA_var_explained_screeplot` <- ggplot(data = `2022.12.03.isolates_salmon_PCA_var_explained`[`2022.12.03.isolates_salmon_PCA_var_explained`$var_explained > 0.10, ], aes(x = reorder(PC, -var_explained), y = var_explained, colour = "black", fill = ifelse(var_explained > 0.05, 'black', 'red'))) + geom_bar(stat = 'identity') + scale_fill_manual(values = c("#ff1e90", "#f2f3f4")) + scale_colour_manual(values = c("#0f0f0f")) + coord_flip() + theme_linedraw() + ylab("Proportion of variance explained") + xlab("Principal Components") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")
`2022.12.03.isolates_salmon_PCA_PC1_vs_PC2` <- ggbiplot(`2022.12.03.isolates_salmon_PCA`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape = factor(`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`$Month), color = factor(`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022.12.03.cahirinus_isolates_salmon_PCA` <- prcomp(`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`[,c(2,6,8,9,12:14,17:18,20,21,23,26)], center = TRUE, scale = FALSE)
`2022.12.03.cahirinus_isolates_salmon_PCA_PC1_vs_PC2` <- ggbiplot(`2022.12.03.cahirinus_isolates_salmon_PCA`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape = factor(`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`$Month), color = factor(`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022.12.03.russatus_isolates_salmon_PCA` <- prcomp(`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`[,c(3,4,5,7,10,11,15,16,19,22,24,25)], center = TRUE, scale = FALSE)
`2022.12.03.russatus_isolates_salmon_PCA_PC1_vs_PC2` <- ggbiplot(`2022.12.03.russatus_isolates_salmon_PCA`, choices = c(1,2), obs.scale = 1, var.scale = 1, ellipse = TRUE, groups = `2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`$Host, ellipse.prob = 0.75, circle = TRUE, var.axes = FALSE, plot.loadings = TRUE) + theme_linedraw() + geom_point(aes(shape = factor(`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`$Month), color = factor(`2022.12.02_Transposed_RPMs_adjusted_seq_and_GC_bias_subsampled_novogene_paired_reads_to_single_contig_per_assembly_concatenated_all_good_Acomys_isolate_assemblies`$Host)), size = 4) + scale_color_manual(values = c("AC" = '#0073ff', "AR" = '#ff8d1e')) + scale_shape_manual(values = c("J" = 17, "N" = 16)) + labs(shape="Month of collection", colour="Host species") + scale_x_continuous(labels = comma) + scale_y_continuous(labels = comma) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022.12.03.isolates_salmon_PCA_PC1_vs_PC2` + `2022.12.03.cahirinus_isolates_salmon_PCA_PC1_vs_PC2` + `2022.12.03.russatus_isolates_salmon_PCA_PC1_vs_PC2` + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') 
`2022.12.03.isolates_salmon_PCAs_PC1_vs_PC2_all_assemblies_and_per_host_derived_from` <- `2022.12.03.isolates_salmon_PCA_PC1_vs_PC2` + `2022.12.03.cahirinus_isolates_salmon_PCA_PC1_vs_PC2` + `2022.12.03.russatus_isolates_salmon_PCA_PC1_vs_PC2` + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') 

`2022.12.03.melted.transposed.isolates_salmon`

unique(`2022.12.03.melted.transposed.isolates_salmon`$variable)
`2022.12.04.all_isolates_name_list` <- as.list(unique(`2022.12.03.melted.transposed.isolates_salmon`$variable))

`isolate_geometric_means_novogene_paired_salmon_TPMs_single_contigs` <- data.frame(matrix(ncol = 7, nrow = 0, dimnames = list(NULL, c("Isolate", "AC.geometric.mean", "AR.geometric.mean", "June.geometric.mean", "November.geometric.mean", "Host.raw.pvalue", "Season.raw.pvalue"))))
for (isolate in `2022.12.04.all_isolates_name_list`){
  isolate_name <- paste(isolate)
  AC.geometric.mean <- exp(mean(log(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$variable == isolate_name & `2022.12.03.melted.transposed.isolates_salmon`$Host == "AC",]$value)))
  AR.geometric.mean <- exp(mean(log(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$variable == isolate_name & `2022.12.03.melted.transposed.isolates_salmon`$Host == "AR",]$value)))
  June.geometric.mean <- exp(mean(log(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$variable == isolate_name & `2022.12.03.melted.transposed.isolates_salmon`$Month == "J",]$value)))
  November.geometric.mean <- exp(mean(log(`2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$variable == isolate_name & `2022.12.03.melted.transposed.isolates_salmon`$Month == "N",]$value)))
  host_kruskal_raw_pvalue <- kruskal.test(value ~ Host, data = `2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$variable == isolate_name,])$p.value
  month_kruskal_raw_pvalue <- kruskal.test(value ~ Month, data = `2022.12.03.melted.transposed.isolates_salmon`[`2022.12.03.melted.transposed.isolates_salmon`$variable == isolate_name,])$p.value
  `isolate_geometric_means_novogene_paired_salmon_TPMs_single_contigs`[nrow(`isolate_geometric_means_novogene_paired_salmon_TPMs_single_contigs`)+1,] <- c(isolate_name, AC.geometric.mean, AR.geometric.mean, June.geometric.mean, November.geometric.mean, host_kruskal_raw_pvalue, month_kruskal_raw_pvalue)
  `isolate_geometric_means_novogene_paired_salmon_TPMs_single_contigs`[, c(2:7)] <- sapply(`isolate_geometric_means_novogene_paired_salmon_TPMs_single_contigs`[, c(2:7)], as.numeric)
  rm(isolate_name)
  rm(AC.geometric.mean)
  rm(AR.geometric.mean)
  rm(June.geometric.mean)
  rm(November.geometric.mean)
  rm(host_kruskal_raw_pvalue)
  rm(month_kruskal_raw_pvalue)
}

novogene_paired_salmon_TPMs_good_isolates_single_contigs_host_pvalues <- `isolate_geometric_means_novogene_paired_salmon_TPMs_single_contigs`[order(`isolate_geometric_means_novogene_paired_salmon_TPMs_single_contigs`$Host.raw.pvalue),]
novogene_paired_salmon_TPMs_good_isolates_single_contigs_month_pvalues <- `isolate_geometric_means_novogene_paired_salmon_TPMs_single_contigs`[order(`isolate_geometric_means_novogene_paired_salmon_TPMs_single_contigs`$Season.raw.pvalue),]
novogene_paired_salmon_TPMs_good_isolates_single_contigs_host_pvalues$BH.corrected.host.pvalues <- p.adjust(novogene_paired_salmon_TPMs_good_isolates_single_contigs_host_pvalues$Host.raw.pvalue, method = "BH")
novogene_paired_salmon_TPMs_good_isolates_single_contigs_month_pvalues$BH.corrected.season.pvalues <- p.adjust(novogene_paired_salmon_TPMs_good_isolates_single_contigs_host_pvalues$Season.raw.pvalue, method = "BH")
novogene_paired_salmon_TPMs_good_isolates_single_contigs_combined_table <- novogene_paired_salmon_TPMs_good_isolates_single_contigs_host_pvalues
novogene_paired_salmon_TPMs_good_isolates_single_contigs_combined_table$BH.corrected.season.pvalues <- novogene_paired_salmon_TPMs_good_isolates_single_contigs_month_pvalues$BH.corrected.season.pvalues

write.table(`novogene_paired_salmon_TPMs_good_isolates_single_contigs_combined_table`, file = "2022-12-06_novogene_paired_salmon_TPMs_good_isolates_single_contigs_combined_table.tsv", sep ="\t", row.names = FALSE)
`2022.12.06_novogene_paired_salmon_TPMs_good_isolates_single_contigs_combined_table` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-12-06_novogene_paired_salmon_TPMs_good_isolates_single_contigs_combined_table.tsv")

ggplot(data = `2022.12.06_novogene_paired_salmon_TPMs_good_isolates_single_contigs_combined_table`, aes(x = BH.corrected.host.pvalues, y = BH.corrected.season.pvalues, colour = Host)) + geom_jitter()

2022-12-04
`2022.11.30_not_subsampled_mapping_percentages` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-11-30_not_subsampled_mapping_percentages.txt")
View(`2022.11.30_not_subsampled_mapping_percentages`)

`2022.12.04.Cleansed_not_subsampled_novogene_paired_reads_mapping_percentage_good_bbmasked_isolates` <- ggplot(data = `2022.11.30_not_subsampled_mapping_percentages`, aes(x = Host.x.month, y = Mapping.percentage, fill = Host.x.month)) + geom_boxplot(alpha = 0.75) + geom_jitter(colour = "#0f0f0f", width = 0.25, size = 2.25) + theme_linedraw() + ylab("Mapping of reads to assembly reference (%)") + xlab("Host species and sampling month") + scale_fill_manual(values = c("#1e90ff", "#002d59", "#ff8d1e", "#934a00"), name = "Read host species\nand sample month") + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022.11.30_subsampled_mapping_percentages` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-11-30_subsampled_mapping_percentages.txt")
View(`2022.11.30_subsampled_mapping_percentages`)
`2022.12.04.Subsampled_novogene_paired_reads_mapping_percentage_good_bbmasked_isolates` <- ggplot(data = `2022.11.30_subsampled_mapping_percentages`, aes(x = Host.x.month, y = Mapping.percentage, fill = Host.x.month)) + geom_boxplot(alpha = 0.75) + geom_jitter(colour = "#0f0f0f", width = 0.25, size = 2.25) + theme_linedraw() + ylab("Mapping of reads to assembly reference (%)") + xlab("Host species and sampling month") + scale_fill_manual(values = c("#1e90ff", "#002d59", "#ff8d1e", "#934a00"), name = "Read host species\nand sample month") + theme(text = element_text(family = "Tex Gyre Heroes"))

`2022.12.04.Cleansed_not_subsampled_novogene_paired_reads_mapping_percentage_good_bbmasked_isolates` + `2022.12.04.Subsampled_novogene_paired_reads_mapping_percentage_good_bbmasked_isolates` + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')
`2022.12.04.Cleansed_A_Subsampled_B_novogene_paired_reads_percentage_mapping_good_bbmasked_isolates_reference` <- `2022.12.04.Cleansed_not_subsampled_novogene_paired_reads_mapping_percentage_good_bbmasked_isolates` + `2022.12.04.Subsampled_novogene_paired_reads_mapping_percentage_good_bbmasked_isolates` + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')


`2022.12.04.bbmasked_isolate_assemblies_alignments_fig_1` <- grid.arrange(`13B_S2_MUMmer_against_39A_S14_forward_reads_filtered_delta_plot_figure`, `13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`, `13D_S4_MUMmer_against_16bD_S22_forward_reads_filtered_delta_plot_figure`, `13D_S4_MUMmer_against_18B_S10_forward_reads_filtered_delta_plot_figure`, nrow = 2)

`2022.12.04.bbmasked_isolate_assemblies_alignments_fig_2` <- grid.arrange(`13E_S5_MUMmer_against_13D_S4_forward_reads_filtered_delta_plot_figure`, `13E_S5_MUMmer_against_16bD_S22_forward_reads_filtered_delta_plot_figure`, `13E_S5_MUMmer_against_18B_S10_forward_reads_filtered_delta_plot_figure`, `16bD_S22_MUMmer_against_13D_S4_forward_reads_filtered_delta_plot_figure`, nrow = 2)

`2022.12.04.bbmasked_isolate_assemblies_alignments_fig_3` <- grid.arrange(`16bD_S22_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`, `18A_S9_MUMmer_against_39B_S15_forward_reads_filtered_delta_plot_figure`, `18B_S10_MUMmer_against_13D_S4_forward_reads_filtered_delta_plot_figure`, nrow = 2)

`2022.12.04.bbmasked_isolate_assemblies_alignments_fig_4` <- grid.arrange(`18B_S10_MUMmer_against_13E_S5_forward_reads_filtered_delta_plot_figure`, `39A_S14_MUMmer_against_13B_S2_forward_reads_filtered_delta_plot_figure`, `39B_S15_MUMmer_against_18A_S9_forward_reads_filtered_delta_plot_figure`, nrow = 2)

grid.arrange(`13B_S2_MUMmer_against_39A_S14_forward_reads_filtered_coverage_plot_figure`, `13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`, `13D_S4_MUMmer_against_16bD_S22_forward_reads_filtered_coverage_plot_figure`, `13D_S4_MUMmer_against_18B_S10_forward_reads_filtered_coverage_plot_figure`, nrow = 2)
grid.arrange(`13E_S5_MUMmer_against_13D_S4_forward_reads_filtered_coverage_plot_figure`,`13E_S5_MUMmer_against_16bD_S22_forward_reads_filtered_coverage_plot_figure`,`13E_S5_MUMmer_against_18B_S10_forward_reads_filtered_coverage_plot_figure`,`16bD_S22_MUMmer_against_13D_S4_forward_reads_filtered_coverage_plot_figure`, nrow = 2)
grid.arrange(`16bD_S22_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`,`18A_S9_MUMmer_against_39B_S15_forward_reads_filtered_coverage_plot_figure`,`18B_S10_MUMmer_against_13D_S4_forward_reads_filtered_coverage_plot_figure`, nrow = 2)
grid.arrange(`18B_S10_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`,`39A_S14_MUMmer_against_13B_S2_forward_reads_filtered_coverage_plot_figure`,`39B_S15_MUMmer_against_18A_S9_forward_reads_filtered_coverage_plot_figure`, nrow = 2)

`2022.12.04.bbmasked_isolate_assemblies_coverage_fig_1` <- grid.arrange(`13B_S2_MUMmer_against_39A_S14_forward_reads_filtered_coverage_plot_figure`, `13D_S4_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`, `13D_S4_MUMmer_against_16bD_S22_forward_reads_filtered_coverage_plot_figure`, `13D_S4_MUMmer_against_18B_S10_forward_reads_filtered_coverage_plot_figure`, nrow = 2)
`2022.12.04.bbmasked_isolate_assemblies_coverage_fig_2` <- grid.arrange(`13E_S5_MUMmer_against_13D_S4_forward_reads_filtered_coverage_plot_figure`,`13E_S5_MUMmer_against_16bD_S22_forward_reads_filtered_coverage_plot_figure`,`13E_S5_MUMmer_against_18B_S10_forward_reads_filtered_coverage_plot_figure`,`16bD_S22_MUMmer_against_13D_S4_forward_reads_filtered_coverage_plot_figure`, nrow = 2)
`2022.12.04.bbmasked_isolate_assemblies_coverage_fig_3` <- grid.arrange(`16bD_S22_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`,`18A_S9_MUMmer_against_39B_S15_forward_reads_filtered_coverage_plot_figure`,`18B_S10_MUMmer_against_13D_S4_forward_reads_filtered_coverage_plot_figure`, nrow = 2)
`2022.12.04.bbmasked_isolate_assemblies_coverage_fig_4` <- grid.arrange(`18B_S10_MUMmer_against_13E_S5_forward_reads_filtered_coverage_plot_figure`,`39A_S14_MUMmer_against_13B_S2_forward_reads_filtered_coverage_plot_figure`,`39B_S15_MUMmer_against_18A_S9_forward_reads_filtered_coverage_plot_figure`, nrow = 2)

`2022.12.04_single_type_psuedogene_count_dfast_all_good_isolates` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-12-04_single_type_psuedogene_count_dfast_all_good_isolates.txt")
View(`2022.12.04_single_type_psuedogene_count_dfast_all_good_isolates`)
`2022.12.04_melted_single_type_psuedogene_count_dfast_all_good_isolates` <- reshape2::melt(`2022.12.04_single_type_psuedogene_count_dfast_all_good_isolates`, id = c("Isolate", "Host_species"))
`2022.12.04_melted_single_type_psuedogene_count_dfast_all_good_isolates`$variable <- gsub("_", " ", as.character(`2022.12.04_melted_single_type_psuedogene_count_dfast_all_good_isolates`$variable))
`2022.12.03.melted.transposed.isolates_salmon`$variable <- gsub("X", "", as.character(`2022.12.03.melted.transposed.isolates_salmon`$variable))

ggplot(data = `2022.12.04_melted_single_type_psuedogene_count_dfast_all_good_isolates`, aes(x = Host_species, y = value, fill = Host_species)) + xlab("Host species or genus") + ylab ("Number of detections of pseudogene\ntype by DFAST") + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.35) + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 12)) + facet_wrap(~ variable, scales = "free_x")
`2022.12.04.good_isolates_all_species_box_and_violin_plot_single_type_psuedogene_counts_by_DFAST_figure` <- ggplot(data = `2022.12.04_melted_single_type_psuedogene_count_dfast_all_good_isolates`, aes(x = Host_species, y = value, fill = Host_species)) + xlab("Host species or genus") + ylab ("Number of detections of pseudogene\ntype by DFAST") + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.35) + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 12)) + facet_wrap(~ variable, scales = "free_x")

`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-12-04_all_proteins_detected_per_sample_count_dfast_all_good_isolates.txt")
View(`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`)
range(`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`[`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`$Host_species == "AP", ]$Number.proteins)
mean(`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`[`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`$Host_species == "AP", ]$Number.proteins)
median(`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`[`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`$Host_species == "AP", ]$Number.proteins)

range(`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`[`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`$Host_species == "AC", ]$Number.proteins)
mean(`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`[`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`$Host_species == "AC", ]$Number.proteins)
median(`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`[`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`$Host_species == "AC", ]$Number.proteins)

range(`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`[`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`$Host_species == "AR", ]$Number.proteins)
mean(`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`[`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`$Host_species == "AR", ]$Number.proteins)
median(`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`[`2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`$Host_species == "AR", ]$Number.proteins)

ggplot(data = `2022.12.04_all_proteins_detected_per_sample_count_dfast_all_good_isolates`, aes(x = Host_species, y = Number.proteins, fill = Host_species)) + xlab("Host species or genus") + ylab ("Total number predicted proteins\nby DFAST") + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.35) + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 12))

`2022.12.04_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-12-04_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts.txt")
View(`2022.12.04_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`)

`2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts` <- reshape2::melt(`2022.12.04_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`, id = c("Protein", "Host.species", "SampleID"))
range(`2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`[`2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`$Host.species == "AP" & `2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`$variable == "hypothetical.protein", ]$value)
range(`2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`[`2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`$Host.species == "AC" & `2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`$variable == "hypothetical.protein", ]$value)
range(`2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`[`2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`$Host.species == "AR" & `2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`$variable == "hypothetical.protein", ]$value)

tic()
`cultures` <- `2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`
`2022.12.04.all_good_isolates_unique_protein_names_list` <- as.list(unique(`2022.12.04_melted_transposed_all_isolates_all_possible_unique_proteins_from_faas_counts`$variable))
`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests` <- data.frame(matrix(ncol = 7, nrow = 0, dimnames = list(NULL, c("Predicted.protein", "AC.assembly.geometric.mean", "AR.assembly.geometric.mean", "AP.assembly.geometric.mean", "AC.phyper.raw.pvalue", "AR.phyper.raw.pvalue", "AP.phyper.raw.pvalue"))))
for (protein.name in `2022.12.04.all_good_isolates_unique_protein_names_list`){
  protein.tambo <- paste(protein.name)
  total.number.assemblies.tambo <- 32
  AC.geometric.tambo <- exp(mean(log(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AC", ]$value)))
  AR.geometric.tambo <- exp(mean(log(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AR", ]$value)))
  AP.geometric.tambo <- exp(mean(log(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AP", ]$value)))
  count.AC.assemblies.tambo <- as.numeric(nrow(as.data.frame(unique(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AC",]))))
  count.AR.assemblies.tambo <- as.numeric(nrow(as.data.frame(unique(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AR",]))))
  count.AP.assemblies.tambo <- as.numeric(nrow(as.data.frame(unique(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AP",]))))
  all.assemblies.summed.value.tambo <- sum(`cultures`[`cultures`$variable == protein.tambo,]$value)
  AC.summed.values.tambo <- sum(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AC",]$value)
  AR.summed.values.tambo <- sum(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AR",]$value)
  AP.summed.values.tambo <- sum(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AP",]$value)
  all.assemblies.value.over.zero.tambo <- as.numeric(nrow(as.data.frame(unique(`cultures`[`cultures`$variable == protein.tambo & `cultures`$value > 0,]))))
  count.AC.assemblies.over.zero.tambo <- as.numeric(nrow(as.data.frame(unique(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AC" & `cultures`$value > 0,]))))
  count.AR.assemblies.over.zero.tambo <- as.numeric(nrow(as.data.frame(unique(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AR" & `cultures`$value > 0,]))))
  count.AP.assemblies.over.zero.tambo <- as.numeric(nrow(as.data.frame(unique(`cultures`[`cultures`$variable == protein.tambo & `cultures`$Host.species == "AP" & `cultures`$value > 0,]))))
  AC.assemblies.pvalue.tambo <- phyper(q = (count.AC.assemblies.tambo - 1), m = AC.summed.values.tambo, n = (all.assemblies.summed.value.tambo - AC.summed.values.tambo), k = all.assemblies.value.over.zero.tambo, lower.tail = FALSE)
  AR.assemblies.pvalue.tambo <- phyper(q = (count.AR.assemblies.tambo - 1), m = AR.summed.values.tambo, n = (all.assemblies.summed.value.tambo - AR.summed.values.tambo), k = all.assemblies.value.over.zero.tambo, lower.tail = FALSE)
  AP.assemblies.pvalue.tambo <- phyper(q = (count.AP.assemblies.tambo - 1), m = AP.summed.values.tambo, n = (all.assemblies.summed.value.tambo - AP.summed.values.tambo), k = all.assemblies.value.over.zero.tambo, lower.tail = FALSE)
  `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[nrow(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`) + 1,] <- c(`protein.tambo`, `AC.geometric.tambo`, `AR.geometric.tambo`, `AP.geometric.tambo`, `AC.assemblies.pvalue.tambo`, `AR.assemblies.pvalue.tambo`, `AP.assemblies.pvalue.tambo`)
  `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[,c(2:7)] <- sapply(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[,c(2:7)], as.numeric)
  rm(list=ls(pattern='*tambo'))
}
view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`)
write.table(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`, file = "2022-12-05-all_good_isolates_unique_proteins_geometric_means_and_phyper_tests.tsv", sep="\t", row.names=FALSE)
`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests` <- `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[order(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.phyper.raw.pvalue),]
`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue <- p.adjust(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.phyper.raw.pvalue, method = "BH")
`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests` <- `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[order(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.phyper.raw.pvalue),]
`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue <- p.adjust(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.phyper.raw.pvalue, method = "BH")
`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests` <- `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[order(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.phyper.raw.pvalue),]
`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AP.phyper.raw.pvalue <- p.adjust(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.phyper.raw.pvalue, method = "BH")
view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`)
write.table(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`, file = "2022-12-05-all_good_isolates_unique_proteins_geometric_means_and_phyper_tests_BH_corrected_added.tsv", sep="\t", row.names=FALSE)
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-11-27.RData")
toc()

as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean >= 10,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean >= 10,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean >= 10,])))

view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean >= 10,])
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean >= 10,])))
view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean >= 5 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$Predicted.protein %notin% `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean >= 5,]$Predicted.protein & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$Predicted.protein %notin% `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean >= 5,]$Predicted.protein,])
view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean >= 10 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$Predicted.protein %notin% `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean >= 10,]$Predicted.protein,])


as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean >= 5,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean >= 5,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean >= 5,])))

as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean >= 1,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean >= 1,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean >= 1,])))

as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean > 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean < 1,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean > 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean < 1,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean > 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean < 1,])))


as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean >= 1 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean < 10,])))

as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean >= 10 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean >= 10 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean >= 10,])))

as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean > `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean > `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean,])))

`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean > `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean > `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean,]

as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean > `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean > `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean > `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean > `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean,])))

`2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests` <- reshape2::melt(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`, id = c("Predicted.protein") )
ggplot(data = `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[grepl(glob2rx("BH.corrected*"), `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$variable) & `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$value != 0,], aes(x = value, y = variable, colour = variable)) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 12.5)) + xlab("BH corrected P-value") + ylab("Assembly species of origin") + scale_colour_manual(values = c("#1e90ff", "#ff8d1e", "#8d1eff")) + theme(legend.position = "none") + geom_vline(xintercept=c(0.05), col="#0f0f0f", linetype = "dashed", size = 1.5, alpha = 0.75)

ggplot(data = `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[grepl(glob2rx("BH.corrected*"), `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$variable) & `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$value != 0,], aes(x = value, y = variable, colour = variable)) + scale_x_log10(labels = label_log()) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 12.5)) + xlab("BH corrected P-value on log10 scale") + ylab("Assembly species of origin") + scale_colour_manual(values = c("#1e90ff", "#ff8d1e", "#8d1eff")) + theme(legend.position = "none") + geom_vline(xintercept=c(0.05), col="#0f0f0f", linetype = "dashed", size = 1.5, alpha = 0.75)

ggplot(data = `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[grepl(glob2rx("BH.corrected*"), `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$variable) & `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$value != 0,], aes(x = value, y = variable, colour = variable)) + scale_x_log10(labels = label_log()) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 12.5)) + xlab("BH corrected P-value on log10 scale") + ylab("") + scale_colour_manual(values = c("#1e90ff", "#ff8d1e", "#8d1eff"), name = "Host species / genus", labels = c("Acomys cahirinus", "Acomys russatus", "Apodemus")) + theme(axis.title.y = element_blank(), axis.text.y = element_blank(), axis.ticks.y = element_blank()) + geom_vline(xintercept=c(0.05), col="#0f0f0f", linetype = "dashed", size = 1.5, alpha = 0.75) + theme(legend.title = element_text(size = 14), legend.text = element_text(size = 12))

`2022.12.05.hypergeometric_test_BH_corrected_pvalues_all_good_isolates_predicted_proteins_dfast_figure` <- ggplot(data = `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[grepl(glob2rx("BH.corrected*"), `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$variable) & `2022.12.05.melted_all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$value != 0,], aes(x = value, y = variable, colour = variable)) + scale_x_log10(labels = label_log()) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 12.5)) + xlab("BH corrected P-value on log10 scale") + ylab("") + scale_colour_manual(values = c("#1e90ff", "#ff8d1e", "#8d1eff"), name = "Host species / genus", labels = c("Acomys cahirinus", "Acomys russatus", "Apodemus")) + theme(axis.title.y = element_blank(), axis.text.y = element_blank(), axis.ticks.y = element_blank()) + geom_vline(xintercept=c(0.05), col="#0f0f0f", linetype = "dashed", size = 1.5, alpha = 0.75) + theme(legend.title = element_text(size = 14), legend.text = element_text(size = 12))

as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue != 0,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue != 0,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AP.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AP.phyper.raw.pvalue != 0,])))

as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean > 0,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean > 0,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AP.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AP.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean > 0,])))

view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean > 0,])
view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean > 0,])
view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AP.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AP.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean > 0,])

view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue != 0) & (`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$variable %notin% `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue != 0,]$variable),])

view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue != 0) & (`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue != 0),])

as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean > 0,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AR.assembly.geometric.mean > 0,])))
as.numeric(nrow(as.data.frame(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AP.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AP.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AP.assembly.geometric.mean > 0,])))

view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue <= 0.05 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$AC.assembly.geometric.mean > 0,])

2022-12-05
#Run 1 commands from Antia after modification
run1v<-read.csv("09-31-36.CSV", skip = 5, ) %>%
  dplyr::rename(time="UNIX.Timestamp") %>%
  dplyr::rename(temp= "Air.temp.") %>%
  mutate(...99=NULL) %>%
  pivot_longer(cols=-c(time, temp), names_to = "Well", values_to = "OD") %>%
  separate(
  col=Well,
  into=c("salinity", "strain"),
  sep = 1,
  remove = FALSE,
  )
run1v_cleaned <- run1v[run1v$salinity != "X",]
run_1_renamed <- run1v_cleaned %>% mutate(salinity=factor(salinity), strain=factor(strain))
levels(run_1_renamed$salinity)<- c("0", "1", "1.75", "2.50", "3.50", "5", "7.50", "10")
levels(run_1_renamed$strain)<- c("Negative control", "Positive control",  "S122",  "S128",  "41B",  "13E",  "16aC", "13D",  "16bD", "18D",  "41A",  "41C")
view(run_1_renamed)
ggplot(run_1_renamed, aes(x=as.POSIXct(run_1_renamed$time, origin = "1970-01-01", tz = "UTC"), y=OD, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + scale_color_manual(name = "Salinity (%)", values = c("#c45161", "#e094a0", "#f2b6c0", "#f2dde1", "#cbc7d8", "#8db7d2", "#5e62a9", "#434279")) + facet_wrap(~strain)
lastimepoint <- run_1_renamed %>% filter(time==1663755213)
blank <- run_1_renamed %>% dplyr::filter(strain == "Negative control") %>% dplyr::select(time, salinity, OD)
without_blank_corrected <- dplyr::left_join(run_1_renamed, blank, by=c("time", "salinity"), keep=F) %>% dplyr::mutate(ODCORRECTED = OD.x - OD.y)
without_blank_corrected
lastimepoint_2 <- without_blank_corrected %>% dplyr::filter(time==1663755213)
ggplot(without_blank_corrected, aes(x=as.POSIXct(without_blank_corrected$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + scale_color_manual(name = "Salinity (%)", values = c("#c45161", "#e094a0", "#f2b6c0", "#f2dde1", "#cbc7d8", "#8db7d2", "#5e62a9", "#434279")) + facet_wrap(~strain)
ggplot(lastimepoint_2, aes(x=strain, y=ODCORRECTED, fill=salinity))+ geom_bar(position="dodge", stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_discrete(guide = guide_axis(n.dodge = 2)) + scale_fill_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + ylab("Assembly")
ggplot(without_blank_corrected, aes(x=as.POSIXct(without_blank_corrected$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + facet_wrap(~strain)
trial_fig_1 <- ggplot(lastimepoint_2, aes(x=strain, y=ODCORRECTED, fill=salinity))+ geom_bar(position="dodge", stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_discrete(guide = guide_axis(n.dodge = 2)) + scale_fill_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + xlab("Assembly") + ylab("Corrected OD600")
trial_fig_2 <- ggplot(without_blank_corrected, aes(x=as.POSIXct(without_blank_corrected$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain)
trial_fig_1 + trial_fig_2 + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')
`2022-12-05_combined_plot_Run_1_OD600_corrected_final_time_point_bar_chart_and_line_plots_figure` <- trial_fig_1 + trial_fig_2 + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

run2v<-read.csv("11-15-41.csv", skip = 5, ) %>%
  dplyr::rename(time="UNIX.Timestamp") %>%
  dplyr::rename(temp= "Air.temp.") %>%
  mutate(...99=NULL) %>%
  pivot_longer(cols=-c(time, temp), names_to = "Well", values_to = "OD") %>%
  separate(
  col=Well,
  into=c("salinity", "strain"),
  sep = 1,
  remove = FALSE,
  )
run2v_cleaned <- run2v[run2v$salinity != "X",]
run_2_renamed <- run2v_cleaned %>% mutate(salinity=factor(salinity), strain=factor(strain))
levels(run_2_renamed$salinity)<- c("0", "1", "1.75", "2.50", "3.50", "5", "7.50", "10")
levels(run_2_renamed$strain)<- c("Negative control", "Positive control",  "S122",  "S128",  "41B",  "13E",  "16aC", "13D",  "16bD", "18D",  "41A",  "41C")
view(run_2_renamed)
lastimepoint <- run_2_renamed %>% filter(time==1663931916)
blank <- run_2_renamed %>% dplyr::filter(strain == "Negative control") %>% dplyr::select(time, salinity, OD)
without_blank_corrected <- dplyr::left_join(run_2_renamed, blank, by=c("time", "salinity"), keep=F) %>% dplyr::mutate(ODCORRECTED = OD.x - OD.y)
without_blank_corrected
lastimepoint_2 <- without_blank_corrected %>% dplyr::filter(time==1663931916)
ggplot(without_blank_corrected, aes(x=as.POSIXct(without_blank_corrected$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + scale_color_manual(name = "Salinity (%)", values = c("#c45161", "#e094a0", "#f2b6c0", "#f2dde1", "#cbc7d8", "#8db7d2", "#5e62a9", "#434279")) + facet_wrap(~strain)
ggplot(lastimepoint_2, aes(x=strain, y=ODCORRECTED, fill=salinity))+ geom_bar(position="dodge", stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_discrete(guide = guide_axis(n.dodge = 2)) + scale_fill_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + ylab("Assembly")
ggplot(without_blank_corrected, aes(x=as.POSIXct(without_blank_corrected$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + facet_wrap(~strain)
run_2_fig_1 <- ggplot(lastimepoint_2, aes(x=strain, y=ODCORRECTED, fill=salinity))+ geom_bar(position="dodge", stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_discrete(guide = guide_axis(n.dodge = 2)) + scale_fill_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + xlab("Assembly") + ylab("Corrected OD600")
run_2_fig_2 <- ggplot(without_blank_corrected, aes(x=as.POSIXct(without_blank_corrected$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain)
run_2_fig_1 + run_2_fig_2 + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')
`2022-12-05_combined_plot_Run_2_OD600_corrected_final_time_point_bar_chart_and_line_plots_figure` <- run_2_fig_1 + run_2_fig_2 + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

run3v<-read.csv("06-48-30.csv", skip = 5, ) %>%
  dplyr::rename(time="UNIX.Timestamp") %>%
  dplyr::rename(temp= "Air.temp.") %>%
  mutate(...99=NULL) %>%
  pivot_longer(cols=-c(time, temp), names_to = "Well", values_to = "OD") %>%
  separate(
  col=Well,
  into=c("salinity", "strain"),
  sep = 1,
  remove = FALSE,
  )
run3v_cleaned <- run3v[run3v$salinity != "X",]
run_3_renamed <- run3v_cleaned %>% mutate(salinity=factor(salinity), strain=factor(strain))
levels(run_3_renamed$salinity)<- c("0", "1", "1.75", "2.50", "3.50", "5", "7.50", "10")
levels(run_3_renamed$strain)<- c("Negative control", "Positive control",  "S122",  "S128",  "41B",  "13E",  "16aC", "13D",  "16bD", "18D",  "41A",  "41C")
view(run_3_renamed)
lastimepoint <- run_3_renamed %>% filter(time==1664437887)
blank <- run_3_renamed %>% dplyr::filter(strain == "Negative control") %>% dplyr::select(time, salinity, OD)
without_blank_corrected <- dplyr::left_join(run_3_renamed, blank, by=c("time", "salinity"), keep=F) %>% dplyr::mutate(ODCORRECTED = OD.x - OD.y)
without_blank_corrected
lastimepoint_2 <- without_blank_corrected %>% dplyr::filter(time==1664437887)
ggplot(without_blank_corrected, aes(x=as.POSIXct(without_blank_corrected$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + scale_color_manual(name = "Salinity (%)", values = c("#c45161", "#e094a0", "#f2b6c0", "#f2dde1", "#cbc7d8", "#8db7d2", "#5e62a9", "#434279")) + facet_wrap(~strain)
ggplot(lastimepoint_2, aes(x=strain, y=ODCORRECTED, fill=salinity))+ geom_bar(position="dodge", stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_discrete(guide = guide_axis(n.dodge = 2)) + scale_fill_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + ylab("Assembly")
ggplot(without_blank_corrected, aes(x=as.POSIXct(without_blank_corrected$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + facet_wrap(~strain)
run_3_fig_1 <- ggplot(lastimepoint_2, aes(x=strain, y=ODCORRECTED, fill=salinity))+ geom_bar(position="dodge", stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_discrete(guide = guide_axis(n.dodge = 2)) + scale_fill_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + xlab("Assembly") + ylab("Corrected OD600")
run_3_fig_2 <- ggplot(without_blank_corrected, aes(x=as.POSIXct(without_blank_corrected$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain)
run_3_fig_1 + run_3_fig_2 + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')
`2022-12-05_combined_plot_Run_3_OD600_corrected_final_time_point_bar_chart_and_line_plots_figure` <- run_3_fig_1 + run_3_fig_2 + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-11-27.RData")
toc()

view(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue != 0,])

`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue

`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue <= 0.05, ]$Predicted.protein

`2022.12.08.AC.BH_corrected.significant.proteins` <- as.list(unique(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AC.phyper.raw.pvalue <= 0.05, ]$Predicted.protein))
AC.signif.protein.counts <- data.frame(matrix(ncol = 4, nrow = 0, dimnames = list(NULL, c("Predicted.protein", "Sum.AC.count", "Mean.AC.count", "BH.corrected.AC.pvalue"))))
for (protein in `2022.12.08.AC.BH_corrected.significant.proteins`){
  protein.tambo <- paste(protein)
  count.in.AC.tambo <- sum(`cultures`[`cultures`$variable == protein.tambo,]$value)
  arith.mean.AC.tambo <- mean(`cultures`[`cultures`$variable == protein.tambo,]$value)
  bh.pvalue.tambo <- `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$Predicted.protein == protein.tambo,]$BH.corrected.AC.phyper.raw.pvalue
  AC.signif.protein.counts[nrow(AC.signif.protein.counts) + 1,] <- c(`protein.tambo`, `count.in.AC.tambo`, `arith.mean.AC.tambo`, `bh.pvalue.tambo`)
  AC.signif.protein.counts[,c(2:4)] <- sapply(AC.signif.protein.counts[,c(2:4)], as.numeric)
  rm(list=ls(pattern='*tambo'))
}

`2022.12.08.AR.BH_corrected.significant.proteins` <- as.list(unique(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue <= 0.05, ]$Predicted.protein))
AR.signif.protein.counts <- data.frame(matrix(ncol = 4, nrow = 0, dimnames = list(NULL, c("Predicted.protein", "Sum.AR.count", "Mean.AR.count", "BH.corrected.AR.pvalue"))))
for (protein in `2022.12.08.AR.BH_corrected.significant.proteins`){
  protein.tambo <- paste(protein)
  count.in.AR.tambo <- sum(`cultures`[`cultures`$variable == protein.tambo,]$value)
  arith.mean.AR.tambo <- mean(`cultures`[`cultures`$variable == protein.tambo,]$value)
  bh.pvalue.tambo <- `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$Predicted.protein == protein.tambo,]$BH.corrected.AR.phyper.raw.pvalue
  AR.signif.protein.counts[nrow(AR.signif.protein.counts) + 1,] <- c(`protein.tambo`, `count.in.AR.tambo`, `arith.mean.AR.tambo`, `bh.pvalue.tambo`)
  AR.signif.protein.counts[,c(2:4)] <- sapply(AR.signif.protein.counts[,c(2:4)], as.numeric)
  rm(list=ls(pattern='*tambo'))
}

`2022.12.08.AP.BH_corrected.significant.proteins` <- as.list(unique(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AP.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AP.phyper.raw.pvalue <= 0.05, ]$Predicted.protein))
AP.signif.protein.counts <- data.frame(matrix(ncol = 4, nrow = 0, dimnames = list(NULL, c("Predicted.protein", "Sum.AP.count", "Mean.AP.count", "BH.corrected.AP.pvalue"))))
for (protein in `2022.12.08.AP.BH_corrected.significant.proteins`){
  protein.tambo <- paste(protein)
  count.in.AP.tambo <- sum(`cultures`[`cultures`$variable == protein.tambo,]$value)
  arith.mean.AP.tambo <- mean(`cultures`[`cultures`$variable == protein.tambo,]$value)
  bh.pvalue.tambo <- `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$Predicted.protein == protein.tambo,]$BH.corrected.AP.phyper.raw.pvalue
  AP.signif.protein.counts[nrow(AP.signif.protein.counts) + 1,] <- c(`protein.tambo`, `count.in.AP.tambo`, `arith.mean.AP.tambo`, `bh.pvalue.tambo`)
  AP.signif.protein.counts[,c(2:4)] <- sapply(AP.signif.protein.counts[,c(2:4)], as.numeric)
  rm(list=ls(pattern='*tambo'))
}


write.table(`AC.signif.protein.counts`, file = "2022-12-08-all_good_isolates_unique_proteins_AC_significant_phyper_test_proteins_summed_counts_and_arithmetic_means.tsv", sep="\t", row.names=FALSE)
write.table(`AR.signif.protein.counts`, file = "2022-12-08-all_good_isolates_unique_proteins_AR_significant_phyper_test_proteins_summed_counts_and_arithmetic_means.tsv", sep="\t", row.names=FALSE)
write.table(`AP.signif.protein.counts`, file = "2022-12-08-all_good_isolates_unique_proteins_AP_significant_phyper_test_proteins_summed_counts_and_arithmetic_means.tsv", sep="\t", row.names=FALSE)


`2022-09-23_within_AR_seasonal_change_novogene_paired_mapping_strict_bins_Salmon_RPMs_GC_and_seq_bias_flagged_fold_change_corrected_pvalue_figure` <- ggplot(data = `2022-0-23_AR_seasonal_raw_pvalue_SORTED`, aes(x = log2.add.one.AR.seasonal.fold.change, y = -log10(Sorted_BH_corrected_pvalues), label = Bin, colour = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ggtitle("Within AR seasonal change in bin mapping") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#ff8d1e", "#b5b5b3")) + theme(text = element_text(size = 14)) + geom_label_repel(aes(label = ifelse((log2.add.one.AR.seasonal.fold.change > 1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)) | (log2.add.one.AR.seasonal.fold.change < -1 & -log10(Sorted_BH_corrected_pvalues) > -log10(0.05)), as.character(Bin), ''), hjust = 0, vjust = 0), colour = "#9e4e00", min.segment.length = 0) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(legend.position = "none")

titleless_host_effect_RPMs <- ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = AC.to.AR.log2fold.add.one.change, y = -log10(Host.corrected.pvalue), colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")
titleless_month_effect_RPMs <- ggplot(data = `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`, aes(x = June.to.November.log2fold.add.one.change, y = -log10(Season.corrected.pvalue), colour = ifelse((June.to.November.log2fold.add.one.change > 1 & -log10(Season.corrected.pvalue) > -log10(0.05)) | (June.to.November.log2fold.add.one.change < -1 & -log10(Season.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + geom_point() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + geom_vline(xintercept=c(-1, 1), col="#c80815") + geom_hline(yintercept=(-log10(0.05)), col="#c80815") + ylab("-Log10 BH Corrected P-values") + xlab("log2()+1 fold change of geometric means per bin") + geom_vline(xintercept=c(-0.5, 0.5), col="#c82b08", linetype = "dashed") + scale_colour_manual(values = c("#b5b5b3", "#0f0f0f")) + theme(text = element_text(size = 14)) + theme(legend.position = "none")

titleless_host_effect_RPMs + titleless_month_effect_RPMs + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`
nrow(`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`[`bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change > 1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05 | `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$AC.to.AR.log2fold.add.one.change < -1 & `bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change`$Host.corrected.pvalue < 0.05,])

`2022_10_12_subsampled_processed_only_paired_novogene_reads_mapping_pvalue_significant_strict_bins_log2_plus_one_fold_change_figure` <- ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05 & is.finite(`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$AC.to.AR.log2fold.add.one.change), ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + ylab("Log2()+1 fold change of geometric means per bin") + facet_wrap(~ GTDB.TK.family, scales = "free_x") + theme_linedraw() + ggtitle("Log2()+1 fold change of significantly different bins by GTDB-TK assigned family") + theme(text = element_text(family = "Tex Gyre Heroes"), strip.text = element_text(face = "italic")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + annotate("rect", ymin = -1, ymax = -Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#1e90ff", alpha = 0.25) + annotate("rect", ymin = 1, ymax = Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#ff8d1e", alpha = 0.25) + geom_hline(yintercept=c(-1), col="#1e90ff") + geom_hline(yintercept=c(1), col="#ff8d1e") + geom_jitter()
ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05 & is.finite(`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$AC.to.AR.log2fold.add.one.change), ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + scale_colour_manual(values = c("#0f0f0f", "#b5b5b3")) + ylab("Log2()+1 fold change of geometric means per bin") + facet_wrap(~ GTDB.TK.family, scales = "free_x") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), strip.text = element_text(face = "italic")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + annotate("rect", ymin = -1, ymax = -Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#1e90ff", alpha = 0.25) + annotate("rect", ymin = 1, ymax = Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#ff8d1e", alpha = 0.25) + geom_hline(yintercept=c(-1), col="#1e90ff") + geom_hline(yintercept=c(1), col="#ff8d1e") + geom_jitter() + theme(strip.text.x = element_text(size = 12))
`2022_12_14_subsampled_processed_only_paired_novogene_reads_mapping_pvalue_significant_strict_bins_log2_plus_one_fold_change_modified_figure` <- ggplot(data = `2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`[`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$Host.corrected.pvalue < 0.05 & is.finite(`2022_10_11_bin_geometric_means_novogene_paired_salmon_TPMs_strict_bin_single_contigs_and_corrected_pvalues_and_log2_add_one_fold_change_with_GTDB_TK_families_alone_in_last_field`$AC.to.AR.log2fold.add.one.change), ], aes(x = GTDB.TK.family, y = AC.to.AR.log2fold.add.one.change, colour = ifelse((AC.to.AR.log2fold.add.one.change > 1 & -log10(Host.corrected.pvalue) > -log10(0.05)) | (AC.to.AR.log2fold.add.one.change < -1 & -log10(Host.corrected.pvalue) > -log10(0.05)), 'black', 'red'))) + scale_colour_manual(values = c("#0f0f0f", "#858581")) + ylab("Log2()+1 fold change of geometric means per bin") + facet_wrap(~ GTDB.TK.family, scales = "free_x") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), strip.text = element_text(face = "italic")) + theme(legend.position = "none", axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + annotate("rect", ymin = -1, ymax = -Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#1e90ff", alpha = 0.25) + annotate("rect", ymin = 1, ymax = Inf, xmin = -Inf, xmax = Inf, colour = "white", fill = "#ff8d1e", alpha = 0.25) + geom_hline(yintercept=c(-1), col="#1e90ff") + geom_hline(yintercept=c(1), col="#ff8d1e") + geom_jitter() + theme(strip.text.x = element_text(size = 12))
`2022-11-13_hypergeometric_pvalues_plot_single_unique_highest_scoring_products_DFAST_strict_bins_categorised_by_bin_enrichment_type` <- ggplot(data = `2022-10-28_melted_working1`[grepl(glob2rx("BH.corrected*"), `2022-10-28_melted_working1`$variable),], aes(x = value, y = variable, colour = variable)) + geom_jitter() + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 12.5)) + xlab("BH corrected P-value") + ylab("Bin enrichment status") + scale_colour_manual(values = c("#8d1eff", "#1e90ff", "#ff8d1e", "#0a0dff", "#bd5d00")) + theme(legend.position = "none") + geom_vline(xintercept=c(0.05), col="#c82b08", linetype = "dashed")

`2022.12.03.melted.transposed.isolates_salmon`$variable <- gsub("X", "", as.character(`2022.12.03.melted.transposed.isolates_salmon`$variable))

`2022-10-28_melted_working1`$variable <- gsub('[.]', ' ', as.character(`2022-10-28_melted_working1`$variable))
`2022-11-13_hypergeometric_pvalues_plot_single_unique_highest_scoring_products_DFAST_strict_bins_categorised_by_bin_enrichment_type_modified_figure` <- ggplot(data = `2022-10-28_melted_working1`[grepl(glob2rx("BH corrected*"), `2022-10-28_melted_working1`$variable),], aes(x = value, y = variable, colour = variable)) + geom_jitter(size = 2.5) + theme_linedraw() + theme(text = element_text(size = 14)) + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 12.5)) + xlab("BH corrected P-value") + ylab("Bin enrichment status") + scale_colour_manual(values = c("BH corrected AC enriched within bins Pvalue" = "#1ec8ff", "BH corrected AC enriched versus bins Pvalue" = "#006ad1", "BH corrected AR enriched within bins Pvalue" = "#ff551e", "BH corrected AR enriched versus bins Pvalue" = "#d16700", "BH corrected noise bins Pvalue" = "#8d1eff")) + theme(legend.position = "none") + geom_vline(xintercept=c(0.05), col="#c82b08", linetype = "dashed")

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2022-11-27.RData")
toc()

checkm_isolates_plot1 <- ggplot(data = `2022.04.08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size`, aes(x = Completeness)) + geom_histogram(binwidth = 1, fill = "#f2f3f4", color = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(text = element_text(size = 14)) + xlab("Completeness (%)") + ylab("Frequency") + geom_vline(xintercept=c(80), col="#1e90ff", linetype = "dashed", size = 1)

checkm_isolates_plot2 <- ggplot(data = `2022.04.08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size`, aes(x = Contamination)) + geom_histogram(binwidth = 1, fill = "#848482", color = "#0f0f0f") + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(text = element_text(size = 14)) + xlab("Contamination (%)") + ylab("Frequency") + geom_vline(xintercept=c(5), col="#ff8d1e", linetype = "dashed", size = 1)

checkm_isolates_plot3 <- ggplot(data = `2022.11.27_all_isolates_some_checkm_results_plus_host_species`, aes(x = Completeness, y = Contamination, label = Isolate.ID, colour = ifelse(Completeness > 80 & Contamination < 5, 'black', 'red'))) + geom_jitter(size = 1.5) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme(text = element_text(size = 14)) + xlab("Completeness (%)") + ylab("Contamination (%)") + geom_vline(xintercept=c(80), col="#1e90ff", linetype = "dashed", size = 1.35) + geom_hline(yintercept=c(5), col="#ff8d1e", linetype = "dashed", size = 1.35) + geom_label_repel(aes(label = ifelse(Contamination > 5, as.character(Isolate.ID), ''), hjust = 0, vjust = 0), colour = "#363b40", min.segment.length = 0) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba"))

checkm_isolates_plot1 + checkm_isolates_plot2 + checkm_isolates_plot3 + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

`2022-11-27.all_isolates_500bp_min_contigs_CheckM_plots_combined_figure` <- checkm_isolates_plot1 + checkm_isolates_plot2 + checkm_isolates_plot3 + guide_area() + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A') + plot_annotation(title = 'CheckM analysis of 35 isolate assembly fasta files')

`2022.12.03.Salmon_isolates_novogene_paired_mapping_Acomys_russatus_derived_isolates` <- ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`[grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) | grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable), ], aes(x = variable, y = value, fill = Host.x.month, shape = Host.x.month)) + geom_violin() + ylab("RPM for assembly on Log10 scale") + xlab("") + scale_y_log10(labels = label_log()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 11)) + scale_fill_manual(values = c("#1e90ff", "#002d59", "#ff8d1e", "#934a00"), name = "Read host species\nand sample month") + facet_wrap(~ variable, scales = "free_x") + theme(strip.text.x = element_text(size = 12)) + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())
`2022.12.03.Salmon_isolates_novogene_paired_mapping_Acomys_cahirinus_derived_isolates` <- ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`[! grepl(glob2rx("39*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & ! grepl(glob2rx("41*"), `2022.12.03.melted.transposed.isolates_salmon`$variable) & ! grepl(glob2rx("13*"), `2022.12.03.melted.transposed.isolates_salmon`$variable), ], aes(x = variable, y = value, fill = Host.x.month, shape = Host.x.month)) + geom_violin() + ylab("RPM for assembly on Log10 scale") + xlab("") + scale_y_log10(labels = label_log()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_text(size = 11)) + scale_fill_manual(values = c("#1e90ff", "#002d59", "#ff8d1e", "#934a00"), name = "Read host species\nand sample month") + facet_wrap(~ variable, scales = "free_x") + theme(strip.text.x = element_text(size = 12)) + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank())


ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`, aes(x = Host.x.month, y = value, fill = Host.x.month)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.25) + theme_linedraw() + scale_fill_manual(name = "Host species and\nsample month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + xlab("") + scale_y_log10(labels = label_log()) + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(strip.text.x = element_text(size = 12))

ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`, aes(x = Host.x.month, y = value, fill = Host.x.month)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.25) + theme_linedraw() + scale_fill_manual(name = "Host species and\nsample month", values = c("ACJ" = "#1e90ff", "ACN" = "#004385", "ARJ" = "#ff8d1e", "ARN" = "#522800")) + xlab("") + scale_y_log10(labels = label_log()) + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + facet_wrap(~ variable, scales = "free_x") + theme(strip.text.x = element_text(size = 12))

ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`, aes(x = value)) + geom_histogram(binwidth = 10000, fill = "#6e7f80", color = "#0f0f0f") + geom_vline(aes(xintercept=mean(value)), color="#c5883d", linetype="dashed", size=0.85) + xlab("Reads per million") + ylab("Number of mappings") + scale_x_continuous(labels = label_comma()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14))

ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`, aes(y = variable, x = Host.x.month, fill = value)) + geom_tile() + scale_fill_gradient(low="#f2f3f4", high="#0f0f0f", name = "RPM", labels = label_comma()) + theme_linedraw() + ylab("Assembly isolate") + xlab("Host species and sampling month for faecal sample")

smlf1 <- ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`, aes(x = value)) + geom_histogram(binwidth = 10000, fill = "#6e7f80", color = "#0f0f0f") + geom_vline(aes(xintercept=mean(value)), color="#c5883d", linetype="dashed", size=0.85) + xlab("Reads per million") + ylab("Number of mappings") + scale_x_continuous(labels = label_comma()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14))
smlf2 <- ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`, aes(y = variable, x = Host.x.month, fill = value)) + geom_tile() + scale_fill_gradient(low="#f2f3f4", high="#0f0f0f", name = "RPM", labels = label_comma()) + theme_linedraw() + ylab("Assembly isolate") + xlab("Host species and sampling month for faecal sample")

smlf1 + smlf2 + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

smlf1 <- ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`, aes(x = value)) + geom_histogram(binwidth = 10000, fill = "#6e7f80", color = "#0f0f0f") + geom_vline(aes(xintercept=mean(value)), color="#ff8d1e", linetype="dashed", size=0.85) + xlab("Reads per million (RPM)") + ylab("Number of mappings") + scale_x_continuous(labels = label_comma()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14))
smlf2 <- ggplot(data = `2022.12.03.melted.transposed.isolates_salmon`, aes(y = variable, x = Host.x.month, fill = value)) + geom_tile() + scale_fill_gradient(low="white", high="#006ad1", name = "RPM", labels = label_comma()) + theme_linedraw() + ylab("Assembly isolate") + xlab("Host species and sampling month for faecal sample") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14))

smlf1 + smlf2 + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

`2022-11-11_samtools_flagstat_percent_mapped_bam_files_not_filtered_all_processed_read_files_to_single_contig_per_bin_concatenated_reference_box_and_violin_figure` <-

ggplot(data = `2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`, aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Host species and sample month") + ylab("Mapping reads (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + facet_wrap(~ Sequencing) + guides(fill = guide_legend(title = "Host species and sampling month")) + theme_linedraw() + theme(strip.text.x = element_text(size = 15), text = element_text(family = "Tex Gyre Heroes"))


`tempor_2022-12-16_mapping` <- `2022.11.10_melted_samtools_flagstats_mapping_percentage_from_BAM_files_merged_contigs_subsampled_read_file_mapping`
`tempor_2022-12-16_mapping`$Sequencing <- gsub('Novogene', 'Greater read depth samples', as.character(`tempor_2022-12-16_mapping`$Sequencing))
`tempor_2022-12-16_mapping`$Sequencing <- gsub('Pilot', 'Lower read depth samples', as.character(`tempor_2022-12-16_mapping`$Sequencing))

ggplot(data = `tempor_2022-12-16_mapping`, aes(x = Species.x.season, y = value, fill = Species.x.season)) + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.15) + xlab("Host species and sample month") + ylab("Mapping reads (%)") + scale_fill_manual(values = c("#1ec8ff", "#006ad1", "#ff551e", "#d16700")) + facet_wrap(~ Sequencing) + guides(fill = guide_legend(title = "Host species and sampling month")) + theme_linedraw() + theme(strip.text.x = element_text(size = 15), text = element_text(family = "Tex Gyre Heroes"))


`2022-10-28_melted_working1`$variable <- gsub('[.]', ' ', as.character(`2022-10-28_melted_working1`$variable))

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2023-01-09.RData")
toc()


2023-01-19
`2023-01-19_TRANSPOSED_All_unicycler_assemblies_All_good_isolates_no_min_contig_size.All_possible_COGs_counts` <- read.delim("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2023-01-19_TRANSPOSED_All_unicycler_assemblies_All_good_isolates_no_min_contig_size.All_possible_COGs_counts.txt")
View(`2023-01-19_TRANSPOSED_All_unicycler_assemblies_All_good_isolates_no_min_contig_size.All_possible_COGs_counts`)

`melted_unicycler_isolate_COG_counts` <- reshape2::melt(`2023-01-19_TRANSPOSED_All_unicycler_assemblies_All_good_isolates_no_min_contig_size.All_possible_COGs_counts`, id = c("COG_ID", "Isolate_ID", "Host_species"))
ggplot(data = `melted_unicycler_isolate_COG_counts`, aes( y = variable, x = Host_species, fill = value)) + geom_tile() + scale_fill_gradient(low="white", high="#006ad1", name = "RPM", labels = label_comma()) + theme_linedraw() + ylab("COG ID") + xlab("Acomys species isolate obtained from") + theme(text = element_text(family = "Tex Gyre Heroes"))

ggplot(data = `melted_unicycler_isolate_COG_counts`[`melted_unicycler_isolate_COG_counts`$value > 1, ], aes( y = variable, x = Host_species, fill = value)) + geom_tile() + scale_fill_gradient(low="white", high="#006ad1", name = "RPM", labels = label_comma()) + theme_linedraw() + ylab("COG ID") + xlab("Acomys species isolate obtained from") + theme(text = element_text(family = "Tex Gyre Heroes"))
ggplot(data = `melted_unicycler_isolate_COG_counts`[mean(`melted_unicycler_isolate_COG_counts`$value) > 0,], aes( y = variable, x = Host_species, fill = value)) + geom_tile() + scale_fill_gradient(low="white", high="#006ad1", name = "RPM", labels = label_comma()) + theme_linedraw() + ylab("COG ID") + xlab("Acomys species isolate obtained from") + theme(text = element_text(family = "Tex Gyre Heroes"))
ggplot(data = `melted_unicycler_isolate_COG_counts`[mean(`melted_unicycler_isolate_COG_counts`$value) > 1,], aes( y = variable, x = Host_species, fill = value)) + geom_tile() + scale_fill_gradient(low="white", high="#006ad1", name = "RPM", labels = label_comma()) + theme_linedraw() + ylab("COG ID") + xlab("Acomys species isolate obtained from") + theme(text = element_text(family = "Tex Gyre Heroes"))

mean(`2022.11.27_all_isolates_some_checkm_results_plus_host_species`[! `2022.11.27_all_isolates_some_checkm_results_plus_host_species`$Host.species == "AP",]$Number.contigs)


2023-02-09
#First three runs
#Run 1 commands from Antia after modification
exp1_run_1_run1v<-read.csv("09-31-36.CSV", skip = 5, ) %>%
  dplyr::rename(time="UNIX.Timestamp") %>%
  dplyr::rename(temp= "Air.temp.") %>%
  mutate(...99=NULL) %>%
  pivot_longer(cols=-c(time, temp), names_to = "Well", values_to = "OD") %>%
  separate(
  col=Well,
  into=c("salinity", "strain"),
  sep = 1,
  remove = FALSE,
  )
exp1_run_1_run1v_cleaned <- exp1_run_1_run1v[exp1_run_1_run1v$salinity != "X",]
exp1_run_1_run_1_renamed <- exp1_run_1_run1v_cleaned %>% mutate(salinity=factor(salinity), strain=factor(strain))
levels(exp1_run_1_run_1_renamed$salinity)<- c("0", "1", "1.75", "2.50", "3.50", "5", "7.50", "10")
levels(exp1_run_1_run_1_renamed$strain)<- c("Negative control", "Positive control",  "S122",  "S128",  "41B",  "13E",  "16aC", "13D",  "16bD", "18D",  "41A",  "41C")
exp1_run_1_lastimepoint <- exp1_run_1_run_1_renamed %>% filter(time==1663755213)
exp1_run_1_blank <- exp1_run_1_run_1_renamed %>% dplyr::filter(strain == "Negative control") %>% dplyr::select(time, salinity, OD)
exp1_run_1_without_blank_corrected <- dplyr::left_join(run_1_renamed, blank, by=c("time", "salinity"), keep=F) %>% dplyr::mutate(ODCORRECTED = OD.x - OD.y)
exp1_run_1_without_blank_corrected
exp1_run_1_lastimepoint_2 <- exp1_run_1_without_blank_corrected %>% dplyr::filter(time==1663755213)
view(exp1_run_1_without_blank_corrected)


exp1_run2_run2v<-read.csv("11-15-41.csv", skip = 5, ) %>%
  dplyr::rename(time="UNIX.Timestamp") %>%
  dplyr::rename(temp= "Air.temp.") %>%
  mutate(...99=NULL) %>%
  pivot_longer(cols=-c(time, temp), names_to = "Well", values_to = "OD") %>%
  separate(
  col=Well,
  into=c("salinity", "strain"),
  sep = 1,
  remove = FALSE,
  )
exp1_run2_run2v_cleaned <- exp1_run2_run2v[exp1_run2_run2v$salinity != "X",]
exp1_run2_run2v_cleaned_renamed <- exp1_run2_run2v_cleaned %>% mutate(salinity=factor(salinity), strain=factor(strain))
levels(exp1_run2_run2v_cleaned_renamed$salinity)<- c("0", "1", "1.75", "2.50", "3.50", "5", "7.50", "10")
levels(exp1_run2_run2v_cleaned_renamed$strain)<- c("Negative control", "Positive control",  "S122",  "S128",  "41B",  "13E",  "16aC", "13D",  "16bD", "18D",  "41A",  "41C")
view(exp1_run2_run2v_cleaned_renamed)
lastimepoint <- exp1_run2_run2v_cleaned_renamed %>% filter(time==1663931916)
exp1_run2_run2v_cleaned_renamed_blank <- exp1_run2_run2v_cleaned_renamed %>% dplyr::filter(strain == "Negative control") %>% dplyr::select(time, salinity, OD)
exp1_run2_run2v_cleaned_renamed_without_blank_corrected <- dplyr::left_join(exp1_run2_run2v_cleaned_renamed, exp1_run2_run2v_cleaned_renamed_blank, by=c("time", "salinity"), keep=F) %>% dplyr::mutate(ODCORRECTED = OD.x - OD.y)
exp1_run2_run2v_cleaned_renamed_lastimepoint_2 <- exp1_run2_run2v_cleaned_renamed_without_blank_corrected %>% dplyr::filter(time==1663931916)

exp1_run3_run3v<-read.csv("06-48-30.csv", skip = 5, ) %>%
  dplyr::rename(time="UNIX.Timestamp") %>%
  dplyr::rename(temp= "Air.temp.") %>%
  mutate(...99=NULL) %>%
  pivot_longer(cols=-c(time, temp), names_to = "Well", values_to = "OD") %>%
  separate(
  col=Well,
  into=c("salinity", "strain"),
  sep = 1,
  remove = FALSE,
  )
exp1_run3_run3v_cleaned <- exp1_run3_run3v[exp1_run3_run3v$salinity != "X",]
exp1_run3_run3v_cleaned_renamed <- exp1_run3_run3v_cleaned %>% mutate(salinity=factor(salinity), strain=factor(strain))
levels(exp1_run3_run3v_cleaned_renamed$salinity)<- c("0", "1", "1.75", "2.50", "3.50", "5", "7.50", "10")
levels(exp1_run3_run3v_cleaned_renamed$strain)<- c("Negative control", "Positive control",  "S122",  "S128",  "41B",  "13E",  "16aC", "13D",  "16bD", "18D",  "41A",  "41C")
view(exp1_run3_run3v_cleaned_renamed)
lastimepoint <- exp1_run3_run3v_cleaned_renamed %>% filter(time==1664437887)
exp1_run3_run3v_cleaned_renamed_blank <- exp1_run3_run3v_cleaned_renamed %>% dplyr::filter(strain == "Negative control") %>% dplyr::select(time, salinity, OD)
exp1_run3_run3v_without_blank_corrected <- dplyr::left_join(exp1_run3_run3v_cleaned_renamed, exp1_run3_run3v_cleaned_renamed_blank, by=c("time", "salinity"), keep=F) %>% dplyr::mutate(ODCORRECTED = OD.x - OD.y)
view(exp1_run3_run3v_without_blank_corrected)
exp1_run3_run3v_lastimepoint <- exp1_run3_run3v_without_blank_corrected %>% dplyr::filter(time==1664437887)

#Line figures here for experiment 1
ggplot(`exp1_run_1_without_blank_corrected`, aes(x=as.POSIXct(`exp1_run_1_without_blank_corrected`$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 1, Run 1")
ggplot(`exp1_run2_run2v_cleaned_renamed_without_blank_corrected`, aes(x=as.POSIXct(`exp1_run2_run2v_cleaned_renamed_without_blank_corrected`$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 1, Run 2")
ggplot(`exp1_run3_run3v_without_blank_corrected`, aes(x=as.POSIXct(`exp1_run3_run3v_without_blank_corrected`$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 1, Run 3")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/halotolerance_experiment_1_run_2_line_plot.pdf")

#Line figures here for experiment 2
ggplot(`2nd_go_Run_1_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 1")
ggplot(`2nd_go_Run_2_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 2")
ggplot(`2nd_go_Run_3_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 3")

#Combining the relevant run plots from each experiment
hal_tol_exp1_run_1_fig <- ggplot(`exp1_run_1_without_blank_corrected`, aes(x=as.POSIXct(`exp1_run_1_without_blank_corrected`$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 1, Run 1")
hal_tol_exp2_run_1_fig <- ggplot(`2nd_go_Run_1_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 1")
hal_tol_exp1_run_1_fig + hal_tol_exp2_run_1_fig + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/halotolerance_both_experiments_run_1_combined.pdf")

hal_tol_exp1_run_2_fig <- ggplot(`exp1_run2_run2v_cleaned_renamed_without_blank_corrected`, aes(x=as.POSIXct(`exp1_run2_run2v_cleaned_renamed_without_blank_corrected`$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 1, Run 2")
hal_tol_exp2_run_2_fig <- ggplot(`2nd_go_Run_2_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 2")
hal_tol_exp1_run_2_fig + hal_tol_exp2_run_2_fig + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/halotolerance_both_experiments_run_2_combined.pdf")

hal_tol_exp1_run_3_fig <- ggplot(`exp1_run3_run3v_without_blank_corrected`, aes(x=as.POSIXct(`exp1_run3_run3v_without_blank_corrected`$time, origin = "1970-01-01", tz = "UTC"),y=ODCORRECTED, colour=salinity)) + geom_line() + scale_x_datetime(date_breaks = "12 hours", date_labels = "%H:%M") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 1, Run 3")
hal_tol_exp2_run_3_fig <- ggplot(`2nd_go_Run_3_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 3")
hal_tol_exp1_run_3_fig + hal_tol_exp2_run_3_fig + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/halotolerance_both_experiments_run_3_combined.pdf")

#Combining the three runs from experiment 2
hal_tol_exp2_run_1_fig <- ggplot(`2nd_go_Run_1_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 1") 
hal_tol_exp2_run_2_fig <- ggplot(`2nd_go_Run_2_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 2")
hal_tol_exp2_run_3_fig <- ggplot(`2nd_go_Run_3_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 3")

hal_tol_exp2_run_1_fig + hal_tol_exp2_run_2_fig + hal_tol_exp2_run_3_fig + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

#Combining the three runs from experiment 2 but after renaming the two controls
`2nd_go_Run_1_clean_corrected_controls_renamed` <- `2nd_go_Run_1_clean_corrected`
`2nd_go_Run_1_clean_corrected_controls_renamed`$strain <- gsub('Negative control', 'NC', `2nd_go_Run_1_clean_corrected_controls_renamed`$strain)
`2nd_go_Run_1_clean_corrected_controls_renamed`$strain <- gsub('Positive control', 'PC', `2nd_go_Run_1_clean_corrected_controls_renamed`$strain)
hal_tol_exp2_run_1_fig <- ggplot(`2nd_go_Run_1_clean_corrected_controls_renamed`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes", size = 9)) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 1") 

`2nd_go_Run_2_clean_corrected_controls_renamed` <- `2nd_go_Run_2_clean_corrected`
`2nd_go_Run_2_clean_corrected_controls_renamed`$strain <- gsub('Negative control', 'NC', `2nd_go_Run_2_clean_corrected_controls_renamed`$strain)
`2nd_go_Run_2_clean_corrected_controls_renamed`$strain <- gsub('Positive control', 'PC', `2nd_go_Run_2_clean_corrected_controls_renamed`$strain)
hal_tol_exp2_run_2_fig <- ggplot(`2nd_go_Run_2_clean_corrected_controls_renamed`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes", size = 9)) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 2") 

`2nd_go_Run_3_clean_corrected_controls_renamed` <- `2nd_go_Run_3_clean_corrected`
`2nd_go_Run_3_clean_corrected_controls_renamed`$strain <- gsub('Negative control', 'NC', `2nd_go_Run_3_clean_corrected_controls_renamed`$strain)
`2nd_go_Run_3_clean_corrected_controls_renamed`$strain <- gsub('Positive control', 'PC', `2nd_go_Run_3_clean_corrected_controls_renamed`$strain)
hal_tol_exp2_run_3_fig <- ggplot(`2nd_go_Run_3_clean_corrected_controls_renamed`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes", size = 9)) + theme_classic() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, Run 3") 

#Using the dataframes generated above but removing yaxis labels from run 2 and 3 figs as well as x axis label from run 1 and 3, also only putting run number not experiment number
hal_tol_exp2_run_1_fig <- ggplot(`2nd_go_Run_1_clean_corrected_controls_renamed`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes", size = 9)) + theme_classic() + xlab("") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Replicate 1") 
hal_tol_exp2_run_2_fig <- ggplot(`2nd_go_Run_2_clean_corrected_controls_renamed`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes", size = 9)) + theme_classic() + xlab("Time (seconds)") + ylab("") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Replicate 2") 
hal_tol_exp2_run_3_fig <- ggplot(`2nd_go_Run_3_clean_corrected_controls_renamed`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes", size = 9)) + theme_classic() + xlab("") + ylab("") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Replicate 3") 

hal_tol_exp2_run_1_fig + hal_tol_exp2_run_2_fig + hal_tol_exp2_run_3_fig + plot_layout(guides = "collect") + plot_annotation(tag_levels = 'A')

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/halotolerance_experiment_2_all_runs_line_plots_combined.pdf")

#Second three runs
setwd("/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work")
#Commands from Antia after modification - the files are not the same format as the earlier ones, in fact she sent xlsx files which need to be opened and then saved as csv
#Comments after this are from her code
#Second experiment, 1st run
`2nd_go_Run_1` <- read_csv("r1071222salt.csv", skip = 3, col_names = FALSE, col_select=-2) %>% slice_head(n=-27)
view(`2nd_go_Run_1`)
`2nd_go_Run_1_colnames` <- slice_tail(`2nd_go_Run_1`, n = 1)
`2nd_go_Run_1_colnames`[1] <- "Time"
names(`2nd_go_Run_1`) <- `2nd_go_Run_1_colnames` %>% c(., recursive=TRUE)
view(`2nd_go_Run_1`)
`2nd_go_Run_1`<-`2nd_go_Run_1` %>% 
  slice_head(n=-1) %>%
  mutate(Time=gsub("s", "", Time)) %>% #remove the s in the time values
  pivot_longer(A1:H12, names_to = c("well"), values_to = c("OD")) #when you pipe you dont need to add the dataset every time (run0) as it goes in automatically.
view(`2nd_go_Run_1`)
`2nd_go_Run_1_clean` <- `2nd_go_Run_1` %>% 
  separate("well", into = c("salinity","strain"), sep = 1 ) 
view(`2nd_go_Run_1_clean`)
`2nd_go_Run_1_clean`<- `2nd_go_Run_1_clean`%>% 
  mutate(
    Time = as.integer(Time),
    strain = as.factor(strain),
    salinity = as.factor(salinity),
    OD = as.double(OD))# BECAUSE THE ODS ARE NOT FULL NUMBERS, YOU USE DOUBLE
view(`2nd_go_Run_1_clean`)
salinity_names<-list(A="0%",
B="1%",
C="1.75%",
D="2.50%",
E="3.50%",
F="5%",
G="7.50%",
H="10%")
`2nd_go_Run_1_clean`$salinity <- recode_factor(`2nd_go_Run_1_clean`$salinity, !!!salinity_names)
strain_names<-list("1" ="Negative control",
     "2"="Positive control",
     "3"="S122",
     "4"="S128",
     "5"="41B",
     "6"="13E",
     "7"="16aC",
     "8"="13D",
     "9"="16bD",
     "10"="18D",
     "11"="41A",
     "12"="41C")
`2nd_go_Run_1_clean`$strain <- recode_factor(`2nd_go_Run_1_clean`$strain, !!!strain_names)
end_time=180012# More or less 48 hours in seconds, corresponds to closest timepoint
`negative_control` <- `2nd_go_Run_1_clean` %>% dplyr::filter(strain == "Negative control") %>% dplyr::select(Time, salinity, OD)
`2nd_go_Run_1_clean_corrected` <- left_join(`2nd_go_Run_1_clean`, `negative_control`, by = c("Time", "salinity"), keep = F) %>% mutate(ODCORRECTED = OD.x - OD.y)
view(`2nd_go_Run_1_clean_corrected`)

#Second experiment, 2nd run
`2nd_go_Run_2` <- read_csv("SALTR2091222.csv", skip = 3, col_names = FALSE, col_select=-2) %>% slice_head(n=-27)
`2nd_go_Run_2_colnames` <- slice_tail(`2nd_go_Run_2`, n = 1)
`2nd_go_Run_2_colnames`[1] <- "Time"
names(`2nd_go_Run_2`) <- `2nd_go_Run_2_colnames` %>% c(., recursive=TRUE)
view(`2nd_go_Run_2`)
`2nd_go_Run_2`<-`2nd_go_Run_2` %>% 
  slice_head(n=-1) %>%
  mutate(Time=gsub("s", "", Time)) %>% #remove the s in the time values
  pivot_longer(A1:H12, names_to = c("well"), values_to = c("OD")) #when you pipe you dont need to add the dataset every time (run0) as it goes in automatically.
view(`2nd_go_Run_2`)
`2nd_go_Run_2_clean` <- `2nd_go_Run_2` %>% 
  separate("well", into = c("salinity","strain"), sep = 1 ) 
view(`2nd_go_Run_2_clean`)
`2nd_go_Run_2_clean`<- `2nd_go_Run_2_clean`%>% 
  mutate(
    Time = as.integer(Time),
    strain = as.factor(strain),
    salinity = as.factor(salinity),
    OD = as.double(OD))# BECAUSE THE ODS ARE NOT FULL NUMBERS, YOU USE DOUBLE
view(`2nd_go_Run_2_clean`)
salinity_names<-list(A="0%",
B="1%",
C="1.75%",
D="2.50%",
E="3.50%",
F="5%",
G="7.50%",
H="10%")
`2nd_go_Run_2_clean`$salinity <- recode_factor(`2nd_go_Run_2_clean`$salinity, !!!salinity_names)
strain_names<-list("1" ="Negative control",
     "2"="Positive control",
     "3"="S122",
     "4"="S128",
     "5"="41B",
     "6"="13E",
     "7"="16aC",
     "8"="13D",
     "9"="16bD",
     "10"="18D",
     "11"="41A",
     "12"="41C")
`2nd_go_Run_2_clean`$strain <- recode_factor(`2nd_go_Run_2_clean`$strain, !!!strain_names)
`2nd_end_time`=239415# My, Peter's, note here - actual last timepoint when measurement was taken
`negative_control` <- `2nd_go_Run_2_clean` %>% dplyr::filter(strain == "Negative control") %>% dplyr::select(Time, salinity, OD)
`2nd_go_Run_2_clean_corrected` <- left_join(`2nd_go_Run_2_clean`, `negative_control`, by = c("Time", "salinity"), keep = F) %>% mutate(ODCORRECTED = OD.x - OD.y)
view(`2nd_go_Run_2_clean_corrected`)

#Second experiment, 3nd run
`2nd_go_Run_3` <- read_csv("SALT R3 ANTIA.csv", skip = 3, col_names = FALSE, col_select=-2) %>% slice_head(n=-27)
`2nd_go_Run_3_colnames` <- slice_tail(`2nd_go_Run_3`, n = 1)
`2nd_go_Run_3_colnames`[1] <- "Time"
names(`2nd_go_Run_3`) <- `2nd_go_Run_3_colnames` %>% c(., recursive=TRUE)
view(`2nd_go_Run_3`)
`2nd_go_Run_3`<-`2nd_go_Run_3` %>% 
  slice_head(n=-1) %>%
  mutate(Time=gsub("s", "", Time)) %>% #remove the s in the time values
  pivot_longer(A1:H12, names_to = c("well"), values_to = c("OD")) #when you pipe you dont need to add the dataset every time (run0) as it goes in automatically.
view(`2nd_go_Run_3`)
`2nd_go_Run_3_clean` <- `2nd_go_Run_3` %>% 
  separate("well", into = c("salinity","strain"), sep = 1 ) 
view(`2nd_go_Run_3_clean`)
`2nd_go_Run_3_clean`<- `2nd_go_Run_3_clean`%>% 
  mutate(
    Time = as.integer(Time),
    strain = as.factor(strain),
    salinity = as.factor(salinity),
    OD = as.double(OD))# BECAUSE THE ODS ARE NOT FULL NUMBERS, YOU USE DOUBLE
view(`2nd_go_Run_3_clean`)
salinity_names<-list(A="0%",
B="1%",
C="1.75%",
D="2.50%",
E="3.50%",
F="5%",
G="7.50%",
H="10%")
`2nd_go_Run_3_clean`$salinity <- recode_factor(`2nd_go_Run_3_clean`$salinity, !!!salinity_names)
strain_names<-list("1" ="Negative control",
     "2"="Positive control",
     "3"="S122",
     "4"="S128",
     "5"="41B",
     "6"="13E",
     "7"="16aC",
     "8"="13D",
     "9"="16bD",
     "10"="18D",
     "11"="41A",
     "12"="41C")
`2nd_go_Run_3_clean`$strain <- recode_factor(`2nd_go_Run_3_clean`$strain, !!!strain_names)
`3rd_end_time`=172811# My, Peter's, note here - last timepoint without NA values for some isolates
`negative_control` <- `2nd_go_Run_3_clean` %>% dplyr::filter(strain == "Negative control") %>% dplyr::select(Time, salinity, OD)
`2nd_go_Run_3_clean_corrected` <- left_join(`2nd_go_Run_3_clean`, `negative_control`, by = c("Time", "salinity"), keep = F) %>% mutate(ODCORRECTED = OD.x - OD.y)
view(`2nd_go_Run_3_clean_corrected`)

#Now trying to get unified measuremnts from times for each of the experiment 2 runs
working_exp2_run1 <- `2nd_go_Run_1_clean_corrected`
working_exp2_run1_relevant <- working_exp2_run1[c("Time", "salinity", "strain", "ODCORRECTED")]
working_exp2_run1_relevant_wide <- pivot_wider(working_exp2_run1_relevant, id_cols = c("Time", "salinity"), names_from = "strain", values_from = "ODCORRECTED")

working_exp2_run2 <- `2nd_go_Run_2_clean_corrected`
working_exp2_run2_relevant <- working_exp2_run2[c("Time", "salinity", "strain", "ODCORRECTED")]
working_exp2_run2_relevant_wide <- pivot_wider(working_exp2_run2_relevant, id_cols = c("Time", "salinity"), names_from = "strain", values_from = "ODCORRECTED")

working_exp2_run3 <- `2nd_go_Run_3_clean_corrected`
working_exp2_run3_relevant <- working_exp2_run3[c("Time", "salinity", "strain", "ODCORRECTED")]
working_exp2_run3_relevant_wide <- pivot_wider(working_exp2_run3_relevant, id_cols = c("Time", "salinity"), names_from = "strain", values_from = "ODCORRECTED")

unique(working_exp2_run1_relevant[c("salinity", "strain")])

#Was a bust, since you would need to convert them all to the time as measurements not the actual time values since they are needed as well and aren't the same
#across the three runs
combo_list <- as.list(unique(working_exp2_run1_relevant[c("salinity", "strain")]))

`experiment_2_averages_table` <- data.frame(matrix(ncol = 6, nrow = 0, dimnames = list(NULL, c("Strain", "Salinity", "Mean.corrected.OD", "Run.1.corrected.OD", "Run.2.corrected.OD", "Run.2.corrected.OD", "Median.corrected.OD"))))
for (combo in `combo_list`){
  isolate <- paste(combo[strain])
  saltiness <- paste(combo[salinity])
  run1_cor_OD <- 
  mean_cor_OD <- 
  print(thingy)
  rm(thingy)
  rm(combo)
}


`2022.12.08.AR.BH_corrected.significant.proteins` <- as.list(unique(`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue != 0 & `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$BH.corrected.AR.phyper.raw.pvalue <= 0.05, ]$Predicted.protein))
AR.signif.protein.counts <- data.frame(matrix(ncol = 4, nrow = 0, dimnames = list(NULL, c("Predicted.protein", "Sum.AR.count", "Mean.AR.count", "BH.corrected.AR.pvalue"))))
for (protein in `2022.12.08.AR.BH_corrected.significant.proteins`){
  protein.tambo <- paste(protein)
  count.in.AR.tambo <- sum(`cultures`[`cultures`$variable == protein.tambo,]$value)
  arith.mean.AR.tambo <- mean(`cultures`[`cultures`$variable == protein.tambo,]$value)
  bh.pvalue.tambo <- `2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`[`2022.12.05.all_good_isolates_unique_proteins_geometric_means_and_phyper_tests`$Predicted.protein == protein.tambo,]$BH.corrected.AR.phyper.raw.pvalue
  AR.signif.protein.counts[nrow(AR.signif.protein.counts) + 1,] <- c(`protein.tambo`, `count.in.AR.tambo`, `arith.mean.AR.tambo`, `bh.pvalue.tambo`)
  AR.signif.protein.counts[,c(2:4)] <- sapply(AR.signif.protein.counts[,c(2:4)], as.numeric)
  rm(list=ls(pattern='*tambo'))
}


#Your version of the figures
#Final time point bar chart
ggplot(`2nd_go_Run_1_clean_corrected`[`2nd_go_Run_1_clean_corrected`$Time == `end_time`,], aes(x = strain, y=ODCORRECTED, fill=salinity)) + geom_bar(position="dodge", stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_discrete(guide = guide_axis(n.dodge = 2)) + scale_fill_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + xlab("Assembly") + ylab("Corrected OD600")
ggplot(`2nd_go_Run_2_clean_corrected`[`2nd_go_Run_2_clean_corrected`$Time == `2nd_end_time`,], aes(x = strain, y=ODCORRECTED, fill=salinity)) + geom_bar(position="dodge", stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_discrete(guide = guide_axis(n.dodge = 2)) + scale_fill_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + xlab("Assembly") + ylab("Corrected OD600")
ggplot(`2nd_go_Run_3_clean_corrected`[`2nd_go_Run_3_clean_corrected`$Time == `3rd_end_time`,], aes(x = strain, y=ODCORRECTED, fill=salinity)) + geom_bar(position="dodge", stat="identity") + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + scale_x_discrete(guide = guide_axis(n.dodge = 2)) + scale_fill_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + xlab("Assembly") + ylab("Corrected OD600")


#Line plots
ggplot(`2nd_go_Run_1_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, run 1")
ggplot(`2nd_go_Run_2_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, run 2")
ggplot(`2nd_go_Run_3_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain) + ggtitle("Experiment 2, run 3")


#Trying dot plot
ggplot(`2nd_go_Run_1_clean_corrected`, aes(x = Time, y = ODCORRECTED, colour = salinity)) + geom_point(size = 0.5) + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Time") + ylab("Corrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~strain)


#A different thing here
Second_experiment_193_measurments_all_three_runs_all_strains_and_controls <- read.csv("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/Second_experiment_193_measurments_all_three_runs_all_strains_and_controls.csv")
View(Second_experiment_193_measurments_all_three_runs_all_strains_and_controls)
`working_2nd_experiment_193_measures` <- reshape2::melt(`Second_experiment_193_measurments_all_three_runs_all_strains_and_controls`, id = c ("Measurement", "Run", "Experiment", "Salinity", "Strain"))
view(`working_2nd_experiment_193_measures`)
`working_2nd_experiment_193_measures`$variable <- gsub("X", "", as.character(`working_2nd_experiment_193_measures`$variable))
`working_2nd_experiment_193_measures`$variable <- as.numeric(`working_2nd_experiment_193_measures`$variable)
`working_2nd_experiment_193_measures`$Salinity <- as.factor(`working_2nd_experiment_193_measures`$Salinity)

as.list(unique(`working_2nd_experiment_193_measures`$Measurement))


ggplot(data = `working_2nd_experiment_193_measures`, aes(x = variable, y = mean(value), colour = Salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Measurement") + ylab("Mean uncorrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~Strain)
ggplot(data = `working_2nd_experiment_193_measures`, aes(x = variable, y = mean(value), colour = Salinity)) + geom_line() + theme(text = element_text(family = "Tex Gyre Heroes")) + theme_linedraw() + xlab("Measurement") + ylab("Mean uncorrected OD600") + scale_color_manual(name = "Salinity (%)", values = c("#0f0f0f", "#2271B2", "#3DB7E9", "#F748A5", "#359B73", "#D55E00", "#E69F00", "#F0E442")) + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + facet_wrap(~Measurement)


tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2023-01-09.RData")
toc()

checkm_isolates_plot3 <- ggplot(data = `2022.04.08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size`, aes(x = Completeness, y = Contamination, colour = ifelse(Completeness > 80 & Contamination < 5, 'black', 'red'))) + geom_jitter(size = 2.5) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + xlab("Completeness (%)") + ylab("Contamination (%)") + geom_vline(xintercept=c(80), col="#1e90ff", linetype = "dashed", size = 1.35) + geom_hline(yintercept=c(5), col="#ff8d1e", linetype = "dashed", size = 1.35) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba"))
ggplot(data = `2022.04.08_CheckM_all_isolate_SPADES_assemblies_min_500bp_contig_size`, aes(x = Completeness, y = Contamination, colour = ifelse(Completeness > 80 & Contamination < 5, 'black', 'red'))) + geom_jitter(size = 3.5, alpha = 0.6) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + xlab("Completeness (%)") + ylab("Contamination (%)") + geom_vline(xintercept=c(80), col="#1e90ff", linetype = "dashed", linewidth = 1.35) + geom_hline(yintercept=c(5), col="#ff8d1e", linetype = "dashed", linewidth = 1.35) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba"))

2023-01-08_culture_assembly_QC_fig

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2023-01-08_culture_assembly_QC_fig.pdf")


#2023-02-17
setwd("/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work")
`2022.11.27.all_isolates_500bp_min_contig_size_SPADES_assemblies_number_contigs_box_and_violin_plot_facet_by_host_species_or_genus` <- ggplot(data = `2022.11.27.Melted_all_isolates_some_checkm_results_plus_host_species`[`2022.11.27.Melted_all_isolates_some_checkm_results_plus_host_species`$variable == "Number.contigs",], aes(x = Host.species, y = value, fill = Host.species)) + xlab("Host species or genus") + ylab ("Number of contigs in assembly") + geom_violin() + geom_boxplot(fill = "#e4e6e8", alpha = 0.6, width = 0.35) + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 16)) + facet_wrap(~ Host.species, scales = "free_x")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022.11.27.all_isolates_500bp_min_contig_size_SPADES_assemblies_number_contigs_box_and_violin_plot_facet_by_host_species_or_genus.pdf", `2022.11.27.all_isolates_500bp_min_contig_size_SPADES_assemblies_number_contigs_box_and_violin_plot_facet_by_host_species_or_genus`)
`2022-11-27_all_isolates_passing_QC_and_picked_if_ANI_similar_GTDB-TK_genus_level_counts_per_genus_bar_chart` <- ggplot(data = as.data.frame(table(`2022.11.27_GTDB.TK_genera_isolates_passing_QC_and_picked_to_use_when_ANI_over_99_point_9`$Genus)), aes(x = reorder(Var1, - Freq), y = Freq, fill = Var1)) + geom_bar(stat = 'identity', colour = "#0f0f0f") + theme_linedraw() + theme(axis.text.x = element_text(angle = 45, hjust=1)) + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + xlab("Genera") + ylab("Number of isolates classified") + scale_fill_manual(values = c("#2D3047", "#C04ABC", "#E0A458", "#419D78", "#FFDBB5", "#a0a7ae"), name = "GTDB-TK assigned genus") 
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2022-11-27_all_isolates_passing_QC_and_picked_if_ANI_similar_GTDB-TK_genus_level_counts_per_genus_bar_chart.pdf", `2022-11-27_all_isolates_passing_QC_and_picked_if_ANI_similar_GTDB-TK_genus_level_counts_per_genus_bar_chart`)

#2023-02-22
`2023.02.21_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly` <- read.csv("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2023-02-21_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly.csv")
`2023.02.21_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly` <- read.csv("~/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2023-02-21_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly.csv")

###Prokka COG stuff
`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly` <- reshape2::melt(`2023.02.21_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`, id=c("COG", "SampleID", "Host", "Halotolerance"))
as.numeric(nrow(as.data.frame(unique(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$variable))))

as.numeric(nrow(as.data.frame(unique(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$Host == "AP" & `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$value > 0,]$variable))))
as.numeric(nrow(as.data.frame(unique(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$Host == "AC" & `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$value > 0,]$variable))))

`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","Halotolerance","variable","value")] %>% group_by(Host, variable, Halotolerance) %>% summarise_all(mean)

`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(Host) %>% summarise_all(mean)
`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_all(mean)

mean(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$Host == "AC",]$value)
range(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$Host == "AC",]$value)
median(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$Host == "AC",]$value)

`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), sum)
`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), sum) %>% arrange(desc(value))
`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), sum) %>% arrange(desc(value))

`2032_02_22_COGs_summed_all_assemblies_descending_order` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), sum) %>% arrange(desc(value))
`2032_02_22_COGs_mean_all_assemblies_descending_order` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), mean) %>% arrange(desc(value))
`2032_02_22_COGs_median_all_assemblies_descending_order` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), median) %>% arrange(desc(value))

ggplot(data = `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$variable %in% `2032_02_22_COGs_summed_all_assemblies_descending_order`[`2032_02_22_COGs_summed_all_assemblies_descending_order`$value >= 100,]$variable, ], aes(x = variable, y = value, fill = Host)) + geom_violin() + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + xlab("") +ylab("Count across all assemblies") + facet_wrap(~ variable, scales = "free_x") + theme_classic() + ggtitle("COGs with summed values from all assemblies of 100 or greater")
ggplot(data = `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$variable %in% `2032_02_22_COGs_mean_all_assemblies_descending_order`[`2032_02_22_COGs_mean_all_assemblies_descending_order`$value >= 5,]$variable, ], aes(x = variable, y = value, fill = Host)) + geom_violin() + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + xlab("") +ylab("Count across all assemblies") + facet_wrap(~ variable, scales = "free_x") + theme_classic() + ggtitle("COGs with mean values from all assemblies of 5 or greater")
ggplot(data = `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$variable %in% `2032_02_22_COGs_median_all_assemblies_descending_order`[`2032_02_22_COGs_median_all_assemblies_descending_order`$value >= 5,]$variable, ], aes(x = variable, y = value, fill = Host)) + geom_violin() + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + xlab("") +ylab("Count across all assemblies") + facet_wrap(~ variable, scales = "free_x") + theme_classic() + ggtitle("COGs with median values from all assemblies of 5 or greater")

`2032_02_22_COGs_summed_all_assemblies_descending_order_by_host` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable,Host) %>% summarise_at(vars(value), sum) %>% arrange(desc(value))
`2032_02_22_COGs_mean_all_assemblies_descending_order_by_host` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable,Host) %>% summarise_at(vars(value), mean) %>% arrange(desc(value))
`2032_02_22_COGs_median_all_assemblies_descending_order_by_host` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable,Host) %>% summarise_at(vars(value), median) %>% arrange(desc(value))

`2032_02_22_COGs_summed_all_assemblies_descending_order_by_halotolerance` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Halotolerance","variable","value")] %>% group_by(Halotolerance,variable) %>% summarise_at(vars(value), sum) %>% arrange(desc(value))
`2032_02_22_COGs_mean_all_assemblies_descending_order_by_halotolerance` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Halotolerance","variable","value")] %>% group_by(Halotolerance,variable) %>% summarise_at(vars(value), mean) %>% arrange(desc(value))
`2032_02_22_COGs_median_all_assemblies_descending_order_by_halotolerance` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[c("Halotolerance","variable","value")] %>% group_by(Halotolerance,variable) %>% summarise_at(vars(value), median) %>% arrange(desc(value))

ggplot(data = `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$Halotolerance != "N/A" & `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$variable %in% `2032_02_22_COGs_mean_all_assemblies_descending_order_by_halotolerance`[`2032_02_22_COGs_mean_all_assemblies_descending_order_by_halotolerance`$value >= 3,]$variable,], aes( y = variable, x = Halotolerance, fill = value)) + geom_tile() + scale_fill_gradient(low="white", high="#ff8d1e", name = "COG count", labels = label_comma()) + theme_classic() + ylab("COG ID") + xlab("Halotolerance") + theme(text = element_text(family = "Tex Gyre Heroes")) + ggtitle("COGs with mean detection count \u2265 3 across all assemblies")

###Prokka gene stuff
`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly` <- reshape2::melt(`2023.02.21_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_Genes_from_all_assemblies_counts_per_assembly`, id=c("Gene", "SampleID", "Host", "Halotolerance"))
as.numeric(nrow(as.data.frame(unique(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$variable))))

as.numeric(nrow(as.data.frame(unique(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$Host == "AP" & `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$value > 0,]$variable))))
as.numeric(nrow(as.data.frame(unique(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$Host == "AC" & `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$value > 0,]$variable))))

`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","Halotolerance","variable","value")] %>% group_by(Host, variable, Halotolerance) %>% summarise_all(mean)

`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(Host) %>% summarise_all(mean)
`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_all(mean)

mean(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$Host == "AC",]$value)
range(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$Host == "AC",]$value)
median(`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$Host == "AC",]$value)

`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), sum)
`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), sum) %>% arrange(desc(value))
`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), sum) %>% arrange(desc(value))

`2032_02_22_Genes_summed_all_assemblies_descending_order` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), sum) %>% arrange(desc(value))
`2032_02_22_Genes_mean_all_assemblies_descending_order` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), mean) %>% arrange(desc(value))
`2032_02_22_Genes_median_all_assemblies_descending_order` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable) %>% summarise_at(vars(value), median) %>% arrange(desc(value))

`2032_02_22_Genes_summed_all_assemblies_descending_order_by_host` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable,Host) %>% summarise_at(vars(value), sum) %>% arrange(desc(value))
`2032_02_22_Genes_mean_all_assemblies_descending_order_by_host` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable,Host) %>% summarise_at(vars(value), mean) %>% arrange(desc(value))
`2032_02_22_Genes_median_all_assemblies_descending_order_by_host` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Host","variable","value")] %>% group_by(variable,Host) %>% summarise_at(vars(value), median) %>% arrange(desc(value))

`2032_02_22_Genes_summed_all_assemblies_descending_order_by_halotolerance` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Halotolerance","variable","value")] %>% group_by(Halotolerance,variable) %>% summarise_at(vars(value), sum) %>% arrange(desc(value))
`2032_02_22_Genes_mean_all_assemblies_descending_order_by_halotolerance` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Halotolerance","variable","value")] %>% group_by(Halotolerance,variable) %>% summarise_at(vars(value), mean) %>% arrange(desc(value))
`2032_02_22_Genes_median_all_assemblies_descending_order_by_halotolerance` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[c("Halotolerance","variable","value")] %>% group_by(Halotolerance,variable) %>% summarise_at(vars(value), median) %>% arrange(desc(value))

#Figures from Prokka COGs and genes
ggplot(data = `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$variable %in% `2032_02_22_Genes_summed_all_assemblies_descending_order`[`2032_02_22_Genes_summed_all_assemblies_descending_order`$value >= 100,]$variable, ], aes(x = variable, y = value, fill = Host)) + geom_violin() + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + xlab("") +ylab("Count across all assemblies") + facet_wrap(~ variable, scales = "free_x") + theme_classic() + ggtitle("Genes with summed values from all assemblies of 100 or greater")
ggplot(data = `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$variable %in% `2032_02_22_Genes_mean_all_assemblies_descending_order`[`2032_02_22_Genes_mean_all_assemblies_descending_order`$value >= 5,]$variable, ], aes(x = variable, y = value, fill = Host)) + geom_violin() + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + xlab("") +ylab("Count across all assemblies") + facet_wrap(~ variable, scales = "free_x") + theme_classic() + ggtitle("Genes with mean values from all assemblies of 5 or greater")
ggplot(data = `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$variable %in% `2032_02_22_Genes_median_all_assemblies_descending_order`[`2032_02_22_Genes_median_all_assemblies_descending_order`$value >= 5,]$variable, ], aes(x = variable, y = value, fill = Host)) + geom_violin() + scale_fill_manual(values = c("#1e90ff", "#8d1eff", "#ff8d1e"), name = "Host species/genus") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + xlab("") +ylab("Count across all assemblies") + facet_wrap(~ variable, scales = "free_x") + theme_classic() + ggtitle("Genes with median values from all assemblies of 5 or greater")
ggplot(data = `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$Halotolerance != "N/A" & `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$variable %in% `2032_02_22_COGs_mean_all_assemblies_descending_order_by_halotolerance`[`2032_02_22_COGs_mean_all_assemblies_descending_order_by_halotolerance`$value >= 3,]$variable,], aes( y = variable, x = Halotolerance, fill = value)) + geom_tile() + scale_fill_gradient(low="white", high="#ff8d1e", name = "COG count", labels = label_comma()) + theme_classic() + ylab("COG ID") + xlab("Halotolerance") + theme(text = element_text(family = "Tex Gyre Heroes")) + ggtitle("COGs with mean detection count \u2265 3 across all assemblies")

ggplot(data = `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$Halotolerance != "N/A" & `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$variable %in% `2032_02_22_Genes_mean_all_assemblies_descending_order_by_halotolerance`[`2032_02_22_Genes_mean_all_assemblies_descending_order_by_halotolerance`$value >= 10,]$variable,], aes( y = variable, x = Halotolerance, fill = value)) + geom_tile() + scale_fill_gradient(low="white", high="#ff8d1e", name = "Gene count", labels = label_comma()) + theme_classic() + ylab("Gene name") + xlab("Halotolerance") + theme(text = element_text(family = "Tex Gyre Heroes")) + ggtitle("Genes with mean detection count \u2265 10 across all assemblies")
ggplot(data = `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$Halotolerance != "N/A" & `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$variable %in% `2032_02_22_Genes_mean_all_assemblies_descending_order_by_halotolerance`[`2032_02_22_Genes_summed_all_assemblies_descending_order_by_halotolerance`$value >= 15,]$variable,], aes( y = variable, x = Halotolerance, fill = value)) + geom_tile() + scale_fill_gradient(low="white", high="#ff8d1e", name = "Gene count", labels = label_comma()) + theme_classic() + ylab("Gene name") + xlab("Halotolerance") + theme(text = element_text(family = "Tex Gyre Heroes")) + ggtitle("Genes with summed detection count \u2265 15 across all assemblies")

#2023-02-24
`melted_prokka_COGs_salt_test_cultures_only` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`$Halotolerance != "N/A",]
`melted_prokka_genes_salt_test_cultures_only` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`[`2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`$Halotolerance != "N/A",]

#Preparing lists of COGs and genes
`prokka_cogs` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_COGs_from_all_assemblies_counts_per_assembly`
`prokka_genes` <- `2023.02.21_melted_transposed_Prokka_all_isolate_assemblies_exS125_all_possible_genes_from_all_assemblies_counts_per_assembly`
`prokka_cogs_list` <- as.list(unique(`prokka_cogs`$variable))
`prokka_genes_list` <- as.list(unique(`prokka_genes`$variable))

#All assemblies, Prokka COGs hypergeometric test loop, sample origin species effect
tic()
`prokka_cogs_phyper_tests` <- data.frame(matrix(ncol = 4, nrow = 0, dimnames = list(NULL, c("COG.ID", "AC.phyper.raw", "AR.phyper.raw", "AP.phyper.raw"))))
for (coggle in `prokka_cogs_list`){
  cog.name.zoot <- paste(coggle)
  total.assemblies.zoot <- 34
  count.AC.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_cogs`[`prokka_cogs`$Host == "AC" & `prokka_cogs`$variable == cog.name.zoot,]))))
  count.AR.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_cogs`[`prokka_cogs`$Host == "AR" & `prokka_cogs`$variable == cog.name.zoot,]))))
  count.AP.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_cogs`[`prokka_cogs`$Host == "AP" & `prokka_cogs`$variable == cog.name.zoot,]))))
  all.assemblies.cog.summed.zoot <- sum(`prokka_cogs`[`prokka_cogs`$variable == cog.name.zoot,]$value)
  AC.cog.summed.zoot <- sum(`prokka_cogs`[`prokka_cogs`$variable == cog.name.zoot & `prokka_cogs`$Host == "AC",]$value)
  AR.cog.summed.zoot <- sum(`prokka_cogs`[`prokka_cogs`$variable == cog.name.zoot & `prokka_cogs`$Host == "AR",]$value)
  AP.cog.summed.zoot <- sum(`prokka_cogs`[`prokka_cogs`$variable == cog.name.zoot & `prokka_cogs`$Host == "AP",]$value)
  count.all.assemblies.cog.over.zero.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_cogs`[`prokka_cogs`$variable == cog.name.zoot & `prokka_cogs`$value > 0,]))))
  count.AC.assemblies.cog.over.zero.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_cogs`[`prokka_cogs`$Host == "AC" & `prokka_cogs`$variable == cog.name.zoot & `prokka_cogs`$value > 0,]))))
  count.AR.assemblies.cog.over.zero.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_cogs`[`prokka_cogs`$Host == "AR" & `prokka_cogs`$variable == cog.name.zoot & `prokka_cogs`$value > 0,]))))
  count.AP.assemblies.cog.over.zero.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_cogs`[`prokka_cogs`$Host == "AP" & `prokka_cogs`$variable == cog.name.zoot & `prokka_cogs`$value > 0,]))))
  AC.assemblies.phyper.zoot <- phyper(q = (count.AC.assemblies.zoot - 1), m = AC.cog.summed.zoot, n = (all.assemblies.cog.summed.zoot - AC.cog.summed.zoot), k = count.all.assemblies.cog.over.zero.zoot, lower.tail = FALSE)
  AR.assemblies.phyper.zoot <- phyper(q = (count.AR.assemblies.zoot - 1), m = AR.cog.summed.zoot, n = (all.assemblies.cog.summed.zoot - AR.cog.summed.zoot), k = count.all.assemblies.cog.over.zero.zoot, lower.tail = FALSE)
  AP.assemblies.phyper.zoot <- phyper(q = (count.AP.assemblies.zoot - 1), m = AP.cog.summed.zoot, n = (all.assemblies.cog.summed.zoot - AP.cog.summed.zoot), k = count.all.assemblies.cog.over.zero.zoot, lower.tail = FALSE)
  `prokka_cogs_phyper_tests`[nrow(`prokka_cogs_phyper_tests`) + 1,] <- c(cog.name.zoot, AC.assemblies.phyper.zoot, AR.assemblies.phyper.zoot, AP.assemblies.phyper.zoot)
  `prokka_cogs_phyper_tests`[,c(2:4)] <- sapply(`prokka_cogs_phyper_tests`[,c(2:4)], as.numeric)
  rm(list=ls(pattern='*zoot'))
}
view(`prokka_cogs_phyper_tests`)
`prokka_cogs_phyper_tests` <- `prokka_cogs_phyper_tests`[order(`prokka_cogs_phyper_tests`$AC.phyper.raw),]
`prokka_cogs_phyper_tests`$BH.corrected.AC.phyper <- p.adjust(`prokka_cogs_phyper_tests`$AC.phyper.raw, method = "BH")
`prokka_cogs_phyper_tests` <- `prokka_cogs_phyper_tests`[order(`prokka_cogs_phyper_tests`$AR.phyper.raw),]
`prokka_cogs_phyper_tests`$BH.corrected.AR.phyper <- p.adjust(`prokka_cogs_phyper_tests`$AR.phyper.raw, method = "BH")
`prokka_cogs_phyper_tests` <- `prokka_cogs_phyper_tests`[order(`prokka_cogs_phyper_tests`$AP.phyper.raw),]
`prokka_cogs_phyper_tests`$BH.corrected.AP.phyper <- p.adjust(`prokka_cogs_phyper_tests`$AP.phyper.raw, method = "BH")
view(`prokka_cogs_phyper_tests`)
write.table(`prokka_cogs_phyper_tests`, , file = "2022-02-25_Prokka_Cogs_all_assemblies_exS125_hypergeometric_corrected_and_raw_Pvalues", sep="\t", row.names=FALSE)
toc()

ggplot(data = `prokka_cogs_phyper_tests`[`prokka_cogs_phyper_tests`$BH.corrected.AC.phyper !=0,], aes(x = "Cogs", y = BH.corrected.AC.phyper, colour = ifelse(BH.corrected.AC.phyper <= 0.05, 'black', 'red'))) + geom_hline(yintercept=c(0.05), col="#cc6600", linetype = "dashed", linewidth = 1.15) + geom_jitter(size = 1.5, alpha = 0.65) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba")) + ggtitle("COG IDs with Q value for hypergeometric test for AC assemblies")
ggplot(data = `prokka_cogs_phyper_tests`[`prokka_cogs_phyper_tests`$BH.corrected.AR.phyper !=0,], aes(x = "Cogs", y = BH.corrected.AR.phyper, colour = ifelse(BH.corrected.AR.phyper <= 0.05, 'black', 'red'))) + geom_hline(yintercept=c(0.05), col="#cc6600", linetype = "dashed", linewidth = 1.15) + geom_jitter(size = 1.5, alpha = 0.65) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba")) + ggtitle("COG IDs with Q value for hypergeometric test for AR assemblies")
ggplot(data = `prokka_cogs_phyper_tests`[`prokka_cogs_phyper_tests`$BH.corrected.AP.phyper !=0,], aes(x = "Cogs", y = BH.corrected.AP.phyper, colour = ifelse(BH.corrected.AP.phyper <= 0.05, 'black', 'red'))) + geom_hline(yintercept=c(0.05), col="#cc6600", linetype = "dashed", linewidth = 1.15) + geom_jitter(size = 1.5, alpha = 0.65) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba")) + ggtitle("COG IDs with Q value for hypergeometric test for AP assemblies")

#All assemblies, Prokka gene names hypergeometric loop, sample origin species effect
tic()
`prokka_genes_phyper_tests` <- data.frame(matrix(ncol = 4, nrow = 0, dimnames = list(NULL, c("Gene.name", "AC.phyper.raw", "AR.phyper.raw", "AP.phyper.raw"))))
for (genegle in `prokka_genes_list`){
  gene.name.zoot <- paste(genegle)
  total.assemblies.zoot <- 34
  count.AC.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_genes`[`prokka_genes`$Host == "AC" & `prokka_genes`$variable == gene.name.zoot,]))))
  count.AR.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_genes`[`prokka_genes`$Host == "AR" & `prokka_genes`$variable == gene.name.zoot,]))))
  count.AP.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_genes`[`prokka_genes`$Host == "AP" & `prokka_genes`$variable == gene.name.zoot,]))))
  all.assemblies.gene.summed.zoot <- sum(`prokka_genes`[`prokka_genes`$variable == gene.name.zoot,]$value)
  AC.gene.summed.zoot <- sum(`prokka_genes`[`prokka_genes`$variable == gene.name.zoot & `prokka_genes`$Host == "AC",]$value)
  AR.gene.summed.zoot <- sum(`prokka_genes`[`prokka_genes`$variable == gene.name.zoot & `prokka_genes`$Host == "AR",]$value)
  AP.gene.summed.zoot <- sum(`prokka_genes`[`prokka_genes`$variable == gene.name.zoot & `prokka_genes`$Host == "AP",]$value)
  count.all.assemblies.gene.over.zero.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_genes`[`prokka_genes`$variable == gene.name.zoot & `prokka_genes`$value > 0,]))))
  count.AC.assemblies.gene.over.zero.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_genes`[`prokka_genes`$Host == "AC" & `prokka_genes`$variable == gene.name.zoot & `prokka_genes`$value > 0,]))))
  count.AR.assemblies.gene.over.zero.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_genes`[`prokka_genes`$Host == "AR" & `prokka_genes`$variable == gene.name.zoot & `prokka_genes`$value > 0,]))))
  count.AP.assemblies.gene.over.zero.zoot <- as.numeric(nrow(as.data.frame(unique(`prokka_genes`[`prokka_genes`$Host == "AP" & `prokka_genes`$variable == gene.name.zoot & `prokka_genes`$value > 0,]))))
  AC.assemblies.phyper.zoot <- phyper(q = (count.AC.assemblies.zoot - 1), m = AC.gene.summed.zoot, n = (all.assemblies.gene.summed.zoot - AC.gene.summed.zoot), k = count.all.assemblies.gene.over.zero.zoot, lower.tail = FALSE)
  AR.assemblies.phyper.zoot <- phyper(q = (count.AR.assemblies.zoot - 1), m = AR.gene.summed.zoot, n = (all.assemblies.gene.summed.zoot - AR.gene.summed.zoot), k = count.all.assemblies.gene.over.zero.zoot, lower.tail = FALSE)
  AP.assemblies.phyper.zoot <- phyper(q = (count.AP.assemblies.zoot - 1), m = AP.gene.summed.zoot, n = (all.assemblies.gene.summed.zoot - AP.gene.summed.zoot), k = count.all.assemblies.gene.over.zero.zoot, lower.tail = FALSE)
  `prokka_genes_phyper_tests`[nrow(`prokka_genes_phyper_tests`) + 1,] <- c(gene.name.zoot, AC.assemblies.phyper.zoot, AR.assemblies.phyper.zoot, AP.assemblies.phyper.zoot)
  `prokka_genes_phyper_tests`[,c(2:4)] <- sapply(`prokka_genes_phyper_tests`[,c(2:4)], as.numeric)
  rm(list=ls(pattern='*zoot'))
}
view(`prokka_genes_phyper_tests`)
`prokka_genes_phyper_tests` <- `prokka_genes_phyper_tests`[order(`prokka_genes_phyper_tests`$AC.phyper.raw),]
`prokka_genes_phyper_tests`$BH.corrected.AC.phyper <- p.adjust(`prokka_genes_phyper_tests`$AC.phyper.raw, method = "BH")
`prokka_genes_phyper_tests` <- `prokka_genes_phyper_tests`[order(`prokka_genes_phyper_tests`$AR.phyper.raw),]
`prokka_genes_phyper_tests`$BH.corrected.AR.phyper <- p.adjust(`prokka_genes_phyper_tests`$AR.phyper.raw, method = "BH")
`prokka_genes_phyper_tests` <- `prokka_genes_phyper_tests`[order(`prokka_genes_phyper_tests`$AP.phyper.raw),]
`prokka_genes_phyper_tests`$BH.corrected.AP.phyper <- p.adjust(`prokka_genes_phyper_tests`$AP.phyper.raw, method = "BH")
view(`prokka_genes_phyper_tests`)
write.table(`prokka_genes_phyper_tests`, file = "2022-02-25_Prokka_genes_all_assemblies_exS125_hypergeometric_corrected_and_raw_Pvalues", sep="\t", row.names=FALSE)
toc()

ggplot(data = `prokka_genes_phyper_tests`[`prokka_genes_phyper_tests`$BH.corrected.AC.phyper !=0,], aes(x = "Genes", y = BH.corrected.AC.phyper, colour = ifelse(BH.corrected.AC.phyper <= 0.05, 'black', 'red'))) + geom_hline(yintercept=c(0.05), col="#cc6600", linetype = "dashed", linewidth = 1.15) + geom_jitter(size = 1.5, alpha = 0.65) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba")) + ggtitle("Genes with Q value for hypergeometric test for AC assemblies")
ggplot(data = `prokka_genes_phyper_tests`[`prokka_genes_phyper_tests`$BH.corrected.AR.phyper !=0,], aes(x = "Genes", y = BH.corrected.AR.phyper, colour = ifelse(BH.corrected.AR.phyper <= 0.05, 'black', 'red'))) + geom_hline(yintercept=c(0.05), col="#cc6600", linetype = "dashed", linewidth = 1.15) + geom_jitter(size = 1.5, alpha = 0.65) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba")) + ggtitle("Genes with Q value for hypergeometric test for AR assemblies")
ggplot(data = `prokka_genes_phyper_tests`[`prokka_genes_phyper_tests`$BH.corrected.AP.phyper !=0,], aes(x = "Genes", y = BH.corrected.AP.phyper, colour = ifelse(BH.corrected.AP.phyper <= 0.05, 'black', 'red'))) + geom_hline(yintercept=c(0.05), col="#cc6600", linetype = "dashed", linewidth = 1.15) + geom_jitter(size = 1.5, alpha = 0.65) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#adb4ba")) + ggtitle("Genes with Q value for hypergeometric test for AP assemblies")

#2023-02-25
#Made the results for the assemblies used in the salt culturing experiment earlier
`melted_prokka_COGs_salt_test_cultures_only`
`melted_prokka_genes_salt_test_cultures_only`

#Preparing lists of COGs and genes
`salt_prokka_cogs` <- `melted_prokka_COGs_salt_test_cultures_only`
`salt_prokka_genes` <- `melted_prokka_genes_salt_test_cultures_only`
`salt_prokka_cogs_list` <- as.list(unique(`prokka_cogs`$variable))
`salt_prokka_genes_list` <- as.list(unique(`prokka_genes`$variable))

#Salt tolerance COGs first
#More factors here, do salinity tolerance and then salinity tolerance x origin species
tic()
prokka_COGs_halotolerance_assemblies <- data.frame(matrix(ncol = 13, nrow = 0, dimnames = list(NULL, c("COG.name", "Halotolerant.phyper.raw", "Slightly.halotolerant.phyper.raw", "Not.halotolerant.phyper.raw", "A.cahirinus.halotolerant.phyper.raw", "A.cahirinus.slightly.halotolerant.phyper.raw", "A.cahirinus.not.halotolerant.phyper.raw", "A.russatus.halotolerant.phyper.raw", "A.russatus.slightly.halotolerant.phyper.raw", "A.russatus.not.halotolerant.phyper.raw", "Apodemus.halotolerant.phyper.raw", "Apodemus.slightly.halotolerant.phyper.raw", "Apodemus.not.halotolerant.phyper.raw"))))
for (cogle in `salt_prokka_cogs_list`){
  cog.name.zoot <- paste(cogle)
  total.assemblies.zoot <- 10
  count.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Halotolerance == "Halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  count.slightly.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  count.not.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Halotolerance == "Not.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  count.AC.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AC" & `salt_prokka_cogs`$Halotolerance == "Halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  count.AR.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AR" & `salt_prokka_cogs`$Halotolerance == "Halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  count.AP.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AP" & `salt_prokka_cogs`$Halotolerance == "Halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  count.slightly.AC.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AC" & `salt_prokka_cogs`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  count.slightly.AR.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AR" & `salt_prokka_cogs`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  count.slightly.AP.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AP" & `salt_prokka_cogs`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  count.not.AC.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AC" & `salt_prokka_cogs`$Halotolerance == "Not.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  count.not.AR.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AR" & `salt_prokka_cogs`$Halotolerance == "Not.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  count.not.AP.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AP" & `salt_prokka_cogs`$Halotolerance == "Not.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]))))
  all.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Halotolerance == "Halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  slightly.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  not.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Halotolerance == "Not.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  AC.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AC" & `salt_prokka_cogs`$Halotolerance == "Halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  AR.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AR" & `salt_prokka_cogs`$Halotolerance == "Halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  AP.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AP" & `salt_prokka_cogs`$Halotolerance == "Halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  AC.slightly.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AC" & `salt_prokka_cogs`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  AR.slightly.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AR" & `salt_prokka_cogs`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  AP.slightly.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AP" & `salt_prokka_cogs`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  AC.not.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AC" & `salt_prokka_cogs`$Halotolerance == "Not.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  AR.not.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AR" & `salt_prokka_cogs`$Halotolerance == "Not.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  AP.not.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_cogs`[`salt_prokka_cogs`$Host == "AP" & `salt_prokka_cogs`$Halotolerance == "Not.halotolerant" & `salt_prokka_cogs`$variable == cog.name.zoot,]$value)
  count.all.assemblies.over.zero.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_cogs`[`salt_prokka_cogs`$variable == cog.name.zoot & `salt_prokka_cogs`$value != 0,]))))
  halotolerant.assemblies.phyper.zoot <- phyper(q = (count.halotolerant.assemblies.zoot - 1), m = halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  slightly.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.slightly.halotolerant.assemblies.zoot - 1), m = slightly.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - slightly.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  not.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.not.halotolerant.assemblies.zoot - 1), m = not.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - not.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AC.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.AC.halotolerant.assemblies.zoot - 1), m = AC.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AC.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AR.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.AR.halotolerant.assemblies.zoot - 1), m = AR.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AR.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AP.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.AP.halotolerant.assemblies.zoot - 1), m = AP.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AP.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AC.slightly.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.slightly.AC.halotolerant.assemblies.zoot - 1), m = AC.slightly.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AC.slightly.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AR.slightly.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.slightly.AR.halotolerant.assemblies.zoot - 1), m = AR.slightly.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AR.slightly.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AP.slightly.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.slightly.AP.halotolerant.assemblies.zoot - 1), m = AP.slightly.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AP.slightly.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AC.not.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.not.AC.halotolerant.assemblies.zoot - 1), m = AC.not.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AC.not.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AR.not.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.not.AR.halotolerant.assemblies.zoot - 1), m = AR.not.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AR.not.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AP.not.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.not.AP.halotolerant.assemblies.zoot - 1), m = AP.not.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AP.not.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  `prokka_COGs_halotolerance_assemblies`[nrow(`prokka_COGs_halotolerance_assemblies`) + 1,] <- c(cog.name.zoot, halotolerant.assemblies.phyper.zoot, slightly.halotolerant.assemblies.phyper.zoot, not.halotolerant.assemblies.phyper.zoot, AC.halotolerant.assemblies.phyper.zoot, AC.slightly.halotolerant.assemblies.phyper.zoot, AC.not.halotolerant.assemblies.phyper.zoot, AR.halotolerant.assemblies.phyper.zoot, AR.slightly.halotolerant.assemblies.phyper.zoot, AR.not.halotolerant.assemblies.phyper.zoot, AP.halotolerant.assemblies.phyper.zoot, AP.slightly.halotolerant.assemblies.phyper.zoot, AP.not.halotolerant.assemblies.phyper.zoot)
  `prokka_COGs_halotolerance_assemblies`[,c(2:13)] <- sapply(`prokka_COGs_halotolerance_assemblies`[,c(2:13)], as.numeric)
  rm(list=ls(pattern='*zoot'))
}
view(`prokka_COGs_halotolerance_assemblies`)
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$Halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.Halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$Halotolerant.phyper.raw, method = "BH")
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$Slightly.halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.Slightly.halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$Slightly.halotolerant.phyper.raw, method = "BH")
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$Not.halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.Not.halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$Not.halotolerant.phyper.raw, method = "BH")
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$A.cahirinus.halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.AC.halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$A.cahirinus.halotolerant.phyper.raw, method = "BH")
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$A.cahirinus.slightly.halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.AC.slightly.halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$A.cahirinus.slightly.halotolerant.phyper.raw, method = "BH")
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$A.cahirinus.not.halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.AC.not.halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$A.cahirinus.not.halotolerant.phyper.raw, method = "BH")
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$A.russatus.halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.AR.halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$A.russatus.halotolerant.phyper.raw, method = "BH")
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$A.russatus.slightly.halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.AR.slightly.halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$A.russatus.slightly.halotolerant.phyper.raw, method = "BH")
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$A.russatus.not.halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.AR.not.halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$A.russatus.not.halotolerant.phyper.raw, method = "BH")
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$Apodemus.halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.AP.halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$Apodemus.halotolerant.phyper.raw, method = "BH")
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$Apodemus.slightly.halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.AP.slightly.halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$Apodemus.slightly.halotolerant.phyper.raw, method = "BH")
`prokka_COGs_halotolerance_assemblies` <- `prokka_COGs_halotolerance_assemblies`[order(`prokka_COGs_halotolerance_assemblies`$Apodemus.not.halotolerant.phyper.raw),]
`prokka_COGs_halotolerance_assemblies`$BH.corrected.AP.not.halotolerant.phyper <- p.adjust(`prokka_COGs_halotolerance_assemblies`$Apodemus.not.halotolerant.phyper.raw, method = "BH")
view(`prokka_COGs_halotolerance_assemblies`)
write.table(`prokka_COGs_halotolerance_assemblies`, , file = "2022-02-27_Prokka_COGs_halotolerance_tested_assemblies_exS125_hypergeometric_corrected_and_raw_Pvalues", sep="\t", row.names=FALSE)

toc()

ggplot(data = `prokka_COGs_halotolerance_assemblies`[`prokka_COGs_halotolerance_assemblies`$BH.corrected.Halotolerant.phyper !=0,], aes(x = "COGs", y = BH.corrected.Halotolerant.phyper, colour = ifelse(BH.corrected.Halotolerant.phyper <= 0.05, 'black', 'red'))) + geom_hline(yintercept=c(0.05), col="#c23b22", linetype = "dashed", linewidth = 1.15) + geom_jitter(size = 1.5, alpha = 0.65) + theme_classic() + xlab("") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#9bddff")) + ggtitle("Halotolerant isolates")
ggplot(data = `prokka_COGs_halotolerance_assemblies`[`prokka_COGs_halotolerance_assemblies`$BH.corrected.Not.halotolerant.phyper !=0,], aes(x = "COGs", y = BH.corrected.Not.halotolerant.phyper, colour = ifelse(BH.corrected.Not.halotolerant.phyper <= 0.05, 'black', 'red'))) + geom_hline(yintercept=c(0.05), col="#c23b22", linetype = "dashed", linewidth = 1.15) + geom_jitter(size = 1.5, alpha = 0.65) + theme_classic() + xlab("") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#9bddff")) + ggtitle("Not halotolerant isolates")
ggplot(data = `prokka_COGs_halotolerance_assemblies`[`prokka_COGs_halotolerance_assemblies`$BH.corrected.Slightly.halotolerant.phyper !=0,], aes(x = "COGs", y = BH.corrected.Slightly.halotolerant.phyper, colour = ifelse(BH.corrected.Slightly.halotolerant.phyper <= 0.05, 'black', 'red'))) + geom_hline(yintercept=c(0.05), col="#c23b22", linetype = "dashed", linewidth = 1.15) + geom_jitter(size = 1.5, alpha = 0.65) + theme_classic() + xlab("") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("#0f0f0f", "#9bddff")) + ggtitle("Slightly halotolerant isolates")

thingi <- reshape2::melt(`prokka_COGs_halotolerance_assemblies`, id = c("COG.name"))

ggplot(data = `thingi`[grepl(glob2rx("BH.corrected.*"), thingi$variable),], aes(x = value)) + geom_density(fill = "#848482", colour = "#0f0f0f") + geom_vline(xintercept=c(0.05), col="#c23b22", linetype = "dashed", linewidth = 1.15) + scale_x_discrete(guide = guide_axis(angle = 90)) + theme_classic() + xlab("Q value") + ylab("Density") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + facet_wrap(variable ~ .)

ggplot(data = `thingi`[grepl(glob2rx("BH.corrected.*AC*"), thingi$variable),], aes(x = value)) + geom_density(fill = "#1e90ff", colour = "#0f0f0f") + geom_vline(xintercept=c(0.05), col="#545b62", linetype = "dashed", linewidth = 1.35) + theme_classic() + xlab("Q value") + ylab("Density") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + facet_wrap(variable ~ .)
ggplot(data = `thingi`[grepl(glob2rx("BH.corrected.*AR*"), thingi$variable),], aes(x = value)) + geom_density(fill = "#ff8d1e", colour = "#0f0f0f") + geom_vline(xintercept=c(0.05), col="#545b62", linetype = "dashed", linewidth = 1.35) + theme_classic() + xlab("Q value") + ylab("Density") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + facet_wrap(variable ~ .)
ggplot(data = `thingi`[grepl(glob2rx("BH.corrected.*AP*"), thingi$variable),], aes(x = value)) + geom_density(fill = "#b46bff", colour = "#0f0f0f") + geom_vline(xintercept=c(0.05), col="#545b62", linetype = "dashed", linewidth = 1.35) + theme_classic() + xlab("Q value") + ylab("Density") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + facet_wrap(variable ~ .)

BH_corrected_halotolerance_names <- list(BH.corrected.Halotolerant.phyper="All HT",
BH.corrected.Slightly.halotolerant.phyper="All SHT",
BH.corrected.Not.halotolerant.phyper="All NHT",
BH.corrected.AC.halotolerant.phyper="AC HT",
BH.corrected.AR.halotolerant.phyper="AR HT",
BH.corrected.AP.halotolerant.phyper="AP HT",
BH.corrected.AC.slightly.halotolerant.phyper="AC SHT",
BH.corrected.AR.slightly.halotolerant.phyper="AR SHT",
BH.corrected.AP.slightly.halotolerant.phyper="AP SHT",
BH.corrected.AC.not.halotolerant.phyper="AC NHT",
BH.corrected.AR.not.halotolerant.phyper="AR NHT",
BH.corrected.AP.not.halotolerant.phyper="AP NHT")

thingi$variable <- recode_factor(thingi$variable, !!!BH_corrected_halotolerance_names)

ggplot(data = `thingi`[grepl(glob2rx("All*"), thingi$variable),], aes(x = value)) + geom_density(fill = "#848482", colour = "#0f0f0f") + geom_vline(xintercept=c(0.05), col="#c23b22", linetype = "dashed", linewidth = 1.15) + theme_classic() + xlab("Q value") + ylab("Density") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + facet_wrap(variable ~ .)
ggplot(data = `thingi`[! grepl(glob2rx("*raw"), thingi$variable),], aes(x = value)) + geom_density(fill = "#848482", colour = "#0f0f0f") + geom_vline(xintercept=c(0.05), col="#c23b22", linetype = "dashed", linewidth = 1.15) + theme_classic() + xlab("Q value") + ylab("Density") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + facet_wrap(variable ~ .)

ggplot(data = `thingi`[! grepl(glob2rx("*raw"), thingi$variable),], aes(x = value, fill = variable)) + geom_density(colour = "#0f0f0f") + geom_vline(xintercept=c(0.05), col="#c23b22", linetype = "dashed", linewidth = 1.15) + theme_classic() + xlab("Q value") + ylab("Density") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_fill_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + facet_wrap(variable ~ .)

scale_fill_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff"))

ggplot(data = `thingi`[! grepl(glob2rx("*raw"), thingi$variable) & thingi$value != 0,], aes(x = value, fill = variable)) + geom_density(colour = "#0f0f0f") + geom_vline(xintercept=c(0.05), col="#c23b22", linetype = "dashed", linewidth = 1.15) + theme_classic() + xlab("Q value") + ylab("Density") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_fill_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + facet_wrap(variable ~ .)

ggplot(data = `thingi`[! grepl(glob2rx("*raw"), thingi$variable) & thingi$value != 0,], aes(x = value, fill = variable)) + geom_density(colour = "#0f0f0f") + geom_vline(xintercept=c(0.05), col="#0f0f0f", linetype = "dashed", linewidth = 1.10) + theme_classic() + xlab("Q value") + ylab("Density") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_fill_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + facet_wrap(~factor(variable, levels=c("All HT", "All SHT", "All NHT", "AC HT", "AC SHT", "AC NHT", "AR HT", "AR SHT", "AR NHT", "AP HT", "AP SHT", "AP NHT"))) + theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(data = `thingi`[! grepl(glob2rx("*raw"), thingi$variable) & thingi$value != 0,], aes(x = factor(variable, level = c("All HT", "All SHT", "All NHT", "AC HT", "AC SHT", "AC NHT", "AR HT", "AR SHT", "AR NHT", "AP HT", "AP SHT", "AP NHT")), y = value, colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(0.05), col="#0f0f0f", linetype = "dashed", linewidth = 0.75) + ggbeeswarm::geom_quasirandom() + theme_classic() + xlab("Halotolerance") + ylab("Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("COGs")

ggplot(data = `thingi`[! grepl(glob2rx("*raw"), thingi$variable) & thingi$value != 0,], aes(x = factor(variable, level = c("All HT", "All SHT", "All NHT", "AC HT", "AC SHT", "AC NHT", "AR HT", "AR SHT", "AR NHT", "AP HT", "AP SHT", "AP NHT")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.25) + ggbeeswarm::geom_quasirandom() + theme_classic() + xlab("Halotolerance") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("COGs")

ggplot(data = `thingi`[! grepl(glob2rx("*raw"), thingi$variable) & thingi$value != 0,], aes(x = factor(variable, level = c("All HT", "All SHT", "All NHT", "AC HT", "AC SHT", "AC NHT", "AR HT", "AR SHT", "AR NHT", "AP HT", "AP SHT", "AP NHT")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.25) + ggbeeswarm::geom_quasirandom() + geom_label_repel(aes(label = ifelse(-log10(value) >= -log10(0.05), as.character(COG.name), ''), hjust = 0, vjust = 0), colour = "black", min.segment.length = 0) + theme_classic() + xlab("Halotolerance") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("Halotolerance assessed isolate assemblies, COGs hypergeometric distribution test results")

+ geom_label_repel(aes(label = ifelse(-log10(value) >= -log10(0.05), as.character(COG.name), ''), hjust = 0, vjust = 0), colour = variable, min.segment.length = 0)

`2023-02-27_Prokka_COGs_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot` <- ggplot(data = `thingi`[! grepl(glob2rx("*raw"), thingi$variable) & thingi$value != 0,], aes(x = factor(variable, level = c("All HT", "All SHT", "All NHT", "AC HT", "AC SHT", "AC NHT", "AR HT", "AR SHT", "AR NHT", "AP HT", "AP SHT", "AP NHT")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.25) + ggbeeswarm::geom_quasirandom() + geom_label_repel(aes(label = ifelse(-log10(value) >= -log10(0.05), as.character(COG.name), ''), hjust = 0, vjust = 0), colour = "black", min.segment.length = 0) + theme_classic() + xlab("Halotolerance") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("Halotolerance assessed isolate assemblies, COGs hypergeometric distribution test results")
`2023-02-27_Prokka_COGs_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot` <- ggplot(data = `thingi`[! grepl(glob2rx("*raw"), thingi$variable) & thingi$value != 0,], aes(x = factor(variable, level = c("All HT", "All SHT", "All NHT", "AC HT", "AC SHT", "AC NHT", "AR HT", "AR SHT", "AR NHT", "AP HT", "AP SHT", "AP NHT")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.25) + ggbeeswarm::geom_quasirandom() + geom_label_repel(aes(label = ifelse(-log10(value) >= -log10(0.05), as.character(COG.name), ''), hjust = 0, vjust = 0), colour = "black", min.segment.length = 0) + theme_classic() + xlab("Halotolerance") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("COGs")

ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2023-02-27_Prokka_COGs_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot.pdf", `2023-02-27_Prokka_COGs_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot`)

`thingi`[! grepl(glob2rx("*raw"), thingi$variable) & thingi$value != 0 & thingi$value <= 0.05,]

`2022-02-27_Melted_prokka_COGs_halotolerance_assemblies_raw_and_BH_corrected_pvalues_hypergeometric_test` <- `thingi`

#Salt tolerance genes now
tic()
prokka_genes_halotolerance_assemblies <- data.frame(matrix(ncol = 13, nrow = 0, dimnames = list(NULL, c("gene.name", "Halotolerant.phyper.raw", "Slightly.halotolerant.phyper.raw", "Not.halotolerant.phyper.raw", "A.cahirinus.halotolerant.phyper.raw", "A.cahirinus.slightly.halotolerant.phyper.raw", "A.cahirinus.not.halotolerant.phyper.raw", "A.russatus.halotolerant.phyper.raw", "A.russatus.slightly.halotolerant.phyper.raw", "A.russatus.not.halotolerant.phyper.raw", "Apodemus.halotolerant.phyper.raw", "Apodemus.slightly.halotolerant.phyper.raw", "Apodemus.not.halotolerant.phyper.raw"))))
for (genele in `salt_prokka_genes_list`){
  gene.name.zoot <- paste(genele)
  total.assemblies.zoot <- 10
  count.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Halotolerance == "Halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  count.slightly.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  count.not.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Halotolerance == "Not.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  count.AC.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AC" & `salt_prokka_genes`$Halotolerance == "Halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  count.AR.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AR" & `salt_prokka_genes`$Halotolerance == "Halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  count.AP.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AP" & `salt_prokka_genes`$Halotolerance == "Halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  count.slightly.AC.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AC" & `salt_prokka_genes`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  count.slightly.AR.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AR" & `salt_prokka_genes`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  count.slightly.AP.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AP" & `salt_prokka_genes`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  count.not.AC.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AC" & `salt_prokka_genes`$Halotolerance == "Not.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  count.not.AR.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AR" & `salt_prokka_genes`$Halotolerance == "Not.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  count.not.AP.halotolerant.assemblies.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AP" & `salt_prokka_genes`$Halotolerance == "Not.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]))))
  all.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$variable == gene.name.zoot,]$value)
  halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Halotolerance == "Halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  slightly.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  not.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Halotolerance == "Not.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  AC.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AC" & `salt_prokka_genes`$Halotolerance == "Halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  AR.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AR" & `salt_prokka_genes`$Halotolerance == "Halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  AP.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AP" & `salt_prokka_genes`$Halotolerance == "Halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  AC.slightly.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AC" & `salt_prokka_genes`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  AR.slightly.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AR" & `salt_prokka_genes`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  AP.slightly.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AP" & `salt_prokka_genes`$Halotolerance == "Slightly.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  AC.not.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AC" & `salt_prokka_genes`$Halotolerance == "Not.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  AR.not.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AR" & `salt_prokka_genes`$Halotolerance == "Not.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  AP.not.halotolerant.assemblies.summed.zoot <- sum(`salt_prokka_genes`[`salt_prokka_genes`$Host == "AP" & `salt_prokka_genes`$Halotolerance == "Not.halotolerant" & `salt_prokka_genes`$variable == gene.name.zoot,]$value)
  count.all.assemblies.over.zero.zoot <- as.numeric(nrow(as.data.frame(unique(`salt_prokka_genes`[`salt_prokka_genes`$variable == gene.name.zoot & `salt_prokka_genes`$value != 0,]))))
  halotolerant.assemblies.phyper.zoot <- phyper(q = (count.halotolerant.assemblies.zoot - 1), m = halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  slightly.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.slightly.halotolerant.assemblies.zoot - 1), m = slightly.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - slightly.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  not.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.not.halotolerant.assemblies.zoot - 1), m = not.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - not.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AC.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.AC.halotolerant.assemblies.zoot - 1), m = AC.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AC.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AR.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.AR.halotolerant.assemblies.zoot - 1), m = AR.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AR.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AP.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.AP.halotolerant.assemblies.zoot - 1), m = AP.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AP.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AC.slightly.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.slightly.AC.halotolerant.assemblies.zoot - 1), m = AC.slightly.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AC.slightly.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AR.slightly.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.slightly.AR.halotolerant.assemblies.zoot - 1), m = AR.slightly.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AR.slightly.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AP.slightly.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.slightly.AP.halotolerant.assemblies.zoot - 1), m = AP.slightly.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AP.slightly.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AC.not.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.not.AC.halotolerant.assemblies.zoot - 1), m = AC.not.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AC.not.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AR.not.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.not.AR.halotolerant.assemblies.zoot - 1), m = AR.not.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AR.not.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  AP.not.halotolerant.assemblies.phyper.zoot <- phyper(q = (count.not.AP.halotolerant.assemblies.zoot - 1), m = AP.not.halotolerant.assemblies.summed.zoot, n = (all.assemblies.summed.zoot - AP.not.halotolerant.assemblies.summed.zoot), k = count.all.assemblies.over.zero.zoot, lower.tail = FALSE)
  `prokka_genes_halotolerance_assemblies`[nrow(`prokka_genes_halotolerance_assemblies`) + 1,] <- c(gene.name.zoot, halotolerant.assemblies.phyper.zoot, slightly.halotolerant.assemblies.phyper.zoot, not.halotolerant.assemblies.phyper.zoot, AC.halotolerant.assemblies.phyper.zoot, AC.slightly.halotolerant.assemblies.phyper.zoot, AC.not.halotolerant.assemblies.phyper.zoot, AR.halotolerant.assemblies.phyper.zoot, AR.slightly.halotolerant.assemblies.phyper.zoot, AR.not.halotolerant.assemblies.phyper.zoot, AP.halotolerant.assemblies.phyper.zoot, AP.slightly.halotolerant.assemblies.phyper.zoot, AP.not.halotolerant.assemblies.phyper.zoot)
  `prokka_genes_halotolerance_assemblies`[,c(2:13)] <- sapply(`prokka_genes_halotolerance_assemblies`[,c(2:13)], as.numeric)
  rm(list=ls(pattern='*zoot'))
}
view(`prokka_genes_halotolerance_assemblies`)
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$Halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.Halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$Halotolerant.phyper.raw, method = "BH")
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$Slightly.halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.Slightly.halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$Slightly.halotolerant.phyper.raw, method = "BH")
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$Not.halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.Not.halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$Not.halotolerant.phyper.raw, method = "BH")
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$A.cahirinus.halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.AC.halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$A.cahirinus.halotolerant.phyper.raw, method = "BH")
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$A.cahirinus.slightly.halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.AC.slightly.halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$A.cahirinus.slightly.halotolerant.phyper.raw, method = "BH")
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$A.cahirinus.not.halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.AC.not.halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$A.cahirinus.not.halotolerant.phyper.raw, method = "BH")
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$A.russatus.halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.AR.halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$A.russatus.halotolerant.phyper.raw, method = "BH")
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$A.russatus.slightly.halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.AR.slightly.halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$A.russatus.slightly.halotolerant.phyper.raw, method = "BH")
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$A.russatus.not.halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.AR.not.halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$A.russatus.not.halotolerant.phyper.raw, method = "BH")
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$Apodemus.halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.AP.halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$Apodemus.halotolerant.phyper.raw, method = "BH")
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$Apodemus.slightly.halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.AP.slightly.halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$Apodemus.slightly.halotolerant.phyper.raw, method = "BH")
`prokka_genes_halotolerance_assemblies` <- `prokka_genes_halotolerance_assemblies`[order(`prokka_genes_halotolerance_assemblies`$Apodemus.not.halotolerant.phyper.raw),]
`prokka_genes_halotolerance_assemblies`$BH.corrected.AP.not.halotolerant.phyper <- p.adjust(`prokka_genes_halotolerance_assemblies`$Apodemus.not.halotolerant.phyper.raw, method = "BH")
view(`prokka_genes_halotolerance_assemblies`)
write.table(`prokka_genes_halotolerance_assemblies`, file = "2022-02-27_Prokka_genes_halotolerance_tested_assemblies_exS125_hypergeometric_corrected_and_raw_Pvalues", sep="\t", row.names=FALSE)
toc()

thingi2 <- reshape2::melt(`prokka_genes_halotolerance_assemblies`, id = c("gene.name"))
thingi2$variable <- recode_factor(thingi2$variable, !!!BH_corrected_halotolerance_names)
ggplot(data = `thingi2`[! grepl(glob2rx("*raw"), thingi2$variable) & thingi2$value != 0,], aes(x = factor(variable, level = c("All HT", "All SHT", "All NHT", "AC HT", "AC SHT", "AC NHT", "AR HT", "AR SHT", "AR NHT", "AP HT", "AP SHT", "AP NHT")), y = value, colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(0.05), col="#0f0f0f", linetype = "dashed", linewidth = 0.75) + ggbeeswarm::geom_quasirandom() + theme_classic() + xlab("Halotolerance") + ylab("Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("Genes")
ggplot(data = `thingi2`[! grepl(glob2rx("*raw"), thingi2$variable) & thingi2$value != 0,], aes(x = factor(variable, level = c("All HT", "All SHT", "All NHT", "AC HT", "AC SHT", "AC NHT", "AR HT", "AR SHT", "AR NHT", "AP HT", "AP SHT", "AP NHT")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.75) + ggbeeswarm::geom_quasirandom() + theme_classic() + xlab("Halotolerance") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("Halotolerance assessed isolate assemblies, gene hypergeometric distribution test results")

`2023-02-27_Prokka_genes_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot` <- ggplot(data = `thingi2`[! grepl(glob2rx("*raw"), thingi2$variable) & thingi2$value != 0,], aes(x = factor(variable, level = c("All HT", "All SHT", "All NHT", "AC HT", "AC SHT", "AC NHT", "AR HT", "AR SHT", "AR NHT", "AP HT", "AP SHT", "AP NHT")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.75) + ggbeeswarm::geom_quasirandom() + theme_classic() + xlab("Halotolerance") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("Genes")
`2023-02-27_Prokka_genes_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot` <- ggplot(data = `thingi2`[! grepl(glob2rx("*raw"), thingi2$variable) & thingi2$value != 0,], aes(x = factor(variable, level = c("All HT", "All SHT", "All NHT", "AC HT", "AC SHT", "AC NHT", "AR HT", "AR SHT", "AR NHT", "AP HT", "AP SHT", "AP NHT")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.75) + ggbeeswarm::geom_quasirandom() + theme_classic() + xlab("Halotolerance") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("All HT" = "#848482", "All SHT" = "#848482", "All NHT" = "#848482", "AC HT" = "#1e90ff", "AC SHT" = "#1e90ff", "AC NHT" = "#1e90ff", "AR HT" = "#ff8d1e", "AR SHT" = "#ff8d1e", "AR NHT" = "#ff8d1e", "AP HT" = "#b46bff", "AP SHT" = "#b46bff", "AP NHT" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("Halotolerance assessed isolate assemblies, gene hypergeometric distribution test results")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2023-02-27_Prokka_genes_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot.pdf", `2023-02-27_Prokka_genes_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot`)

`2023-02-27_Prokka_COGs_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot` + `2023-02-27_Prokka_genes_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot` + plot_annotation(tag_levels = 'A')

`2023-02-28_Prokka_halotolerance_assessed_assemblies_COGS_and_genes_combined_figure_neglog10_qvalue_from_BH_corrected_pvalues_hypergeometric_test` <- `2023-02-27_Prokka_COGs_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot` + `2023-02-27_Prokka_genes_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot` + plot_annotation(tag_levels = 'A')
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2023-02-28_Prokka_halotolerance_assessed_assemblies_COGS_and_genes_combined_figure_neglog10_qvalue_from_BH_corrected_pvalues_hypergeometric_test.pdf", `2023-02-28_Prokka_halotolerance_assessed_assemblies_COGS_and_genes_combined_figure_neglog10_qvalue_from_BH_corrected_pvalues_hypergeometric_test`)


`thingi2`[! grepl(glob2rx("*raw"), thingi2$variable) & thingi2$value != 0 & thingi2$value <= 0.05,]

`2022-02-27_Melted_prokka_genes_halotolerance_assemblies_raw_and_BH_corrected_pvalues_hypergeometric_test` <- `thingi2`

#2023-02-27
#Figures for all assemblies hypergeometric tests
`prokka_cogs_phyper_tests`
thingi3 <- reshape2::melt(`prokka_cogs_phyper_tests`, id = c("COG.ID"))
BH_corrected_all_names <- list(BH.corrected.AC.phyper="AC", BH.corrected.AR.phyper="AR", BH.corrected.AP.phyper="AP")
thingi3$variable <- recode_factor(thingi3$variable, !!!BH_corrected_all_names)
ggplot(data = `thingi3`[! grepl(glob2rx("*raw"), thingi3$variable) & thingi3$value != 0,], aes(x = factor(variable, level = c("AC", "AR", "AP")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.25) + ggbeeswarm::geom_quasirandom() + theme_classic() + xlab("Isolate origin species") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("AC" = "#1e90ff", "AR" = "#ff8d1e", "AP" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("All isolate assemblies COG hypergeometric distribution test results")
`2023-02-27_Prokka_COGS_all_assemblies_exS125_hypergeometric_results_figure` <- ggplot(data = `thingi3`[! grepl(glob2rx("*raw"), thingi3$variable) & thingi3$value != 0,], aes(x = factor(variable, level = c("AC", "AR", "AP")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.25) + ggbeeswarm::geom_quasirandom() + theme_classic() + xlab("Isolate origin species") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("AC" = "#1e90ff", "AR" = "#ff8d1e", "AP" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("All isolate assemblies COG hypergeometric distribution test results")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2023-02-27_Prokka_COGS_all_assemblies_exS125_hypergeometric_results_figure.pdf", `2023-02-27_Prokka_COGS_all_assemblies_exS125_hypergeometric_results_figure`)

`2023-02-27_Prokka_COGS_all_assemblies_exS125_hypergeometric_results_figure` <- ggplot(data = `thingi3`[! grepl(glob2rx("*raw"), thingi3$variable) & thingi3$value != 0,], aes(x = factor(variable, level = c("AC", "AR", "AP")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.25) + ggbeeswarm::geom_quasirandom() + geom_label_repel(aes(label = ifelse(-log10(value) >= -log10(0.0005), as.character(COG.ID), ''), hjust = 0, vjust = 0), colour = "black", min.segment.length = 0) + theme_classic() + xlab("Isolate origin species") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("AC" = "#1e90ff", "AR" = "#ff8d1e", "AP" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("COGs") + labs(caption = "Labelled points Q value \u2264 0.0005")
`2023-02-27_Prokka_COGS_all_assemblies_exS125_hypergeometric_results_figure` <- ggplot(data = `thingi3`[! grepl(glob2rx("*raw"), thingi3$variable) & thingi3$value != 0,], aes(x = factor(variable, level = c("AC", "AR", "AP")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.25) + ggbeeswarm::geom_quasirandom() + geom_label_repel(aes(label = ifelse(-log10(value) >= -log10(0.0001), as.character(COG.ID), ''), hjust = 0, vjust = 0), colour = "black", min.segment.length = 0) + theme_classic() + xlab("Isolate origin species") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("AC" = "#1e90ff", "AR" = "#ff8d1e", "AP" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("COGs") + labs(caption = "Labelled points Q value \u2264 0.0001")


`prokka_genes_phyper_tests`
thingi4 <- reshape2::melt(`prokka_genes_phyper_tests`, id = c("Gene.name"))
thingi4$variable <- recode_factor(thingi4$variable, !!!BH_corrected_all_names)
ggplot(data = `thingi4`[! grepl(glob2rx("*raw"), thingi4$variable) & thingi4$value != 0,], aes(x = factor(variable, level = c("AC", "AR", "AP")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.25) + ggbeeswarm::geom_quasirandom() + theme_classic() + xlab("Isolate origin species") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("AC" = "#1e90ff", "AR" = "#ff8d1e", "AP" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("All isolate assemblies gene hypergeometric distribution test results")
`2023-02-27_Prokka_genes_all_assemblies_exS125_hypergeometric_results_figure` <- ggplot(data = `thingi4`[! grepl(glob2rx("*raw"), thingi4$variable) & thingi4$value != 0,], aes(x = factor(variable, level = c("AC", "AR", "AP")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.25) + ggbeeswarm::geom_quasirandom() + theme_classic() + xlab("Isolate origin species") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("AC" = "#1e90ff", "AR" = "#ff8d1e", "AP" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("All isolate assemblies gene hypergeometric distribution test results")
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2023-02-27_Prokka_genes_all_assemblies_exS125_hypergeometric_results_figure.pdf", `2023-02-27_Prokka_genes_all_assemblies_exS125_hypergeometric_results_figure`)

thingi4[thingi4$value <= 0.05 & ! thingi4$value == 0,]

`2023-02-28_combined_COGs_prokka_hypergeometric_plots_figure` <- `2023-02-27_Prokka_COGS_all_assemblies_exS125_hypergeometric_results_figure` + `2023-02-27_Prokka_COGs_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot` + plot_annotation(tag_levels = 'A')
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2023-02-28_combined_COGs_prokka_hypergeometric_plots_figure.pdf", `2023-02-28_combined_COGs_prokka_hypergeometric_plots_figure`)

`2023-02-28_combined_genes_prokka_hypergeometric_plots_figure` <- `2023-02-27_Prokka_genes_all_assemblies_exS125_hypergeometric_results_figure` + `2023-02-27_Prokka_genes_halotolerance_tested_assemblies_hypergeometric_test_BH_corrected_Pvalues_plot` + plot_annotation(tag_levels = 'A')
ggsave(filename = "/Users/osbourne/Google Drive/PhD/2022-03-26_Paper_3_Chapter_4-Culture_isolates_based_work/2023-02-28_combined_genes_prokka_hypergeometric_plots_figure.pdf", `2023-02-28_combined_genes_prokka_hypergeometric_plots_figure`)

`2023-02-28_transposed_Prokka_COGs_all_assemblies_BH_corrected_pvalues_005_or_below_excluding_0` <- t((thingi3[thingi3$value <= 0.05 & thingi3$value != 0 & ! grepl(glob2rx("*raw"), thingi3$variable),]))
`2023-02-28_Prokka_COGs_all_assemblies_BH_corrected_pvalues_005_or_below_excluding_0` <- thingi3[thingi3$value <= 0.05 & thingi3$value != 0 & ! grepl(glob2rx("*raw"), thingi3$variable),]

write.table(`2023-02-28_transposed_Prokka_COGs_all_assemblies_BH_corrected_pvalues_005_or_below_excluding_0`, file = "2023-02-28_transposed_Prokka_COGs_all_assemblies_BH_corrected_pvalues_005_or_below_excluding_0.csv", sep=",", row.names=TRUE)
write.table(`2023-02-28_Prokka_COGs_all_assemblies_BH_corrected_pvalues_005_or_below_excluding_0`, file = "2023-02-28_Prokka_COGs_all_assemblies_BH_corrected_pvalues_005_or_below_excluding_0.csv", sep=",", row.names=FALSE)

tic()
`prokka_cogs_all_assemblies_summed` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("COG.ID", "All.summed", "AC.summed", "AR.summed", "AP.summed"))))
for (coggle in `prokka_cogs_list`){
  cog.name.zoot <- paste(coggle)
  all.assemblies.cog.summed.zoot <- sum(`prokka_cogs`[`prokka_cogs`$variable == cog.name.zoot,]$value)
  AC.cog.summed.zoot <- sum(`prokka_cogs`[`prokka_cogs`$variable == cog.name.zoot & `prokka_cogs`$Host == "AC",]$value)
  AR.cog.summed.zoot <- sum(`prokka_cogs`[`prokka_cogs`$variable == cog.name.zoot & `prokka_cogs`$Host == "AR",]$value)
  AP.cog.summed.zoot <- sum(`prokka_cogs`[`prokka_cogs`$variable == cog.name.zoot & `prokka_cogs`$Host == "AP",]$value)
  `prokka_cogs_all_assemblies_summed`[nrow(`prokka_cogs_all_assemblies_summed`) + 1,] <- c(cog.name.zoot, all.assemblies.cog.summed.zoot, AC.cog.summed.zoot, AR.cog.summed.zoot, AP.cog.summed.zoot)
  `prokka_cogs_all_assemblies_summed`[,c(2:5)] <- sapply(`prokka_cogs_all_assemblies_summed`[,c(2:5)], as.numeric)
  rm(list=ls(pattern='*zoot'))
}
view(`prokka_cogs_all_assemblies_summed`)
write.table(`prokka_cogs_all_assemblies_summed`, file = "2022-02-28_Prokka_Cogs_all_assemblies_exS125_summed_counts.csv", sep=",", row.names=FALSE)
toc()

tic()
`prokka_genes_all_assemblies_summed` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Gene.name", "All.summed", "AC.summed", "AR.summed", "AP.summed"))))
for (coggle in `prokka_genes_list`){
  cog.name.zoot <- paste(coggle)
  all.assemblies.cog.summed.zoot <- sum(`prokka_genes`[`prokka_genes`$variable == cog.name.zoot,]$value)
  AC.cog.summed.zoot <- sum(`prokka_genes`[`prokka_genes`$variable == cog.name.zoot & `prokka_genes`$Host == "AC",]$value)
  AR.cog.summed.zoot <- sum(`prokka_genes`[`prokka_genes`$variable == cog.name.zoot & `prokka_genes`$Host == "AR",]$value)
  AP.cog.summed.zoot <- sum(`prokka_genes`[`prokka_genes`$variable == cog.name.zoot & `prokka_genes`$Host == "AP",]$value)
  `prokka_genes_all_assemblies_summed`[nrow(`prokka_genes_all_assemblies_summed`) + 1,] <- c(cog.name.zoot, all.assemblies.cog.summed.zoot, AC.cog.summed.zoot, AR.cog.summed.zoot, AP.cog.summed.zoot)
  `prokka_genes_all_assemblies_summed`[,c(2:5)] <- sapply(`prokka_genes_all_assemblies_summed`[,c(2:5)], as.numeric)
  rm(list=ls(pattern='*zoot'))
}
view(`prokka_genes_all_assemblies_summed`)
write.table(`prokka_genes_all_assemblies_summed`, file = "2022-02-28_Prokka_Genes_all_assemblies_exS125_summed_counts.csv", sep=",", row.names=FALSE)
toc()

as.numeric(nrow(as.data.frame(unique(thingi3[thingi3$value <= 0.05 & thingi3$value != 0,]$COG.ID))))

view(thingi3[thingi3$value <=0.05 & thingi3$value != 0 & ! grepl(glob2rx("*raw"), thingi3$variable),]$COG.ID)

view(data.frame(table(thingi3[thingi3$value <=0.05 & thingi3$value != 0 & ! grepl(glob2rx("*raw"), thingi3$variable),]$COG.ID)))

view(data.frame(table(thingi3[thingi3$value <=0.05 & thingi3$value != 0 & ! grepl(glob2rx("*raw"), thingi3$variable),]$COG.ID))) %>% filter(Freq > 1)
view(data.frame(table(thingi3[thingi3$value <=0.05 & thingi3$value != 0 & ! grepl(glob2rx("*raw"), thingi3$variable),]$COG.ID))) %>% filter(Freq <= 1)

#2023-03-03
as.numeric(nrow(as.data.frame(unique(`prokka_genes`$variable))))

#The two below do the same thing, just one uses the dataframe containing already summed values for each of the three species
as.numeric(nrow(as.data.frame(unique(`prokka_genes`[`prokka_genes`$Host == "AC" & `prokka_genes`$value != 0,]$variable))))
as.numeric(nrow(as.data.frame(`prokka_genes_all_assemblies_summed`[`prokka_genes_all_assemblies_summed`$AC.summed != 0,])))

as.numeric(nrow(as.data.frame(`prokka_genes_all_assemblies_summed`[`prokka_genes_all_assemblies_summed`$AR.summed != 0,])))
as.numeric(nrow(as.data.frame(`prokka_genes_all_assemblies_summed`[`prokka_genes_all_assemblies_summed`$AP.summed != 0,])))

scale_colour_manual(values = c("AC" = "#1e90ff", "AR" = "#ff8d1e", "AP" = "#b46bff"))

ggplot(data = `prokka_genes_all_assemblies_summed`, aes(x = AC.summed, y = AR.summed)) + geom_jitter(colour = "#1e90ff") + theme_classic() + xlab("Summed gene counts AC assemblies") + ylab("Summed gene counts AR assemblies") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14))
ggplot(data = `prokka_genes_all_assemblies_summed`, aes(x = AR.summed, y = AP.summed)) + geom_jitter(colour = "#ff8d1e") + theme_classic() + xlab("Summed gene counts AR assemblies") + ylab("Summed gene counts AP assemblies") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14))
ggplot(data = `prokka_genes_all_assemblies_summed`, aes(x = AP.summed, y = AC.summed)) + geom_jitter(colour = "#b46bff") + theme_classic() + xlab("Summed gene counts AP assemblies") + ylab("Summed gene counts AC assemblies") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14))

tic()
`prokka_genes_all_assemblies_num_mean` <- data.frame(matrix(ncol = 5, nrow = 0, dimnames = list(NULL, c("Gene.name", "All.mean", "AC.mean", "AR.mean", "AP.mean"))))
for (coggle in `prokka_genes_list`){
  #Note that this loop calculates numeric means, I just haven't bothered changing the name of the variables created
  cog.name.zoot <- paste(coggle)
  all.assemblies.cog.summed.zoot <- mean(`prokka_genes`[`prokka_genes`$variable == cog.name.zoot,]$value)
  AC.cog.summed.zoot <- mean(`prokka_genes`[`prokka_genes`$variable == cog.name.zoot & `prokka_genes`$Host == "AC",]$value)
  AR.cog.summed.zoot <- mean(`prokka_genes`[`prokka_genes`$variable == cog.name.zoot & `prokka_genes`$Host == "AR",]$value)
  AP.cog.summed.zoot <- mean(`prokka_genes`[`prokka_genes`$variable == cog.name.zoot & `prokka_genes`$Host == "AP",]$value)
  `prokka_genes_all_assemblies_num_mean`[nrow(`prokka_genes_all_assemblies_num_mean`) + 1,] <- c(cog.name.zoot, all.assemblies.cog.summed.zoot, AC.cog.summed.zoot, AR.cog.summed.zoot, AP.cog.summed.zoot)
  `prokka_genes_all_assemblies_num_mean`[,c(2:5)] <- sapply(`prokka_genes_all_assemblies_num_mean`[,c(2:5)], as.numeric)
  rm(list=ls(pattern='*zoot'))
}
view(`prokka_genes_all_assemblies_num_mean`)
write.table(`prokka_genes_all_assemblies_num_mean`, file = "2022-02-28_Prokka_Genes_all_assemblies_exS125_numeric_means.csv", sep=",", row.names=FALSE)
toc()

ggplot(data = `prokka_genes_all_assemblies_num_mean`, aes(x = AC.mean, y = AR.mean)) + geom_jitter(colour = "#1e90ff") + theme_classic() + xlab("Mean gene counts AC assemblies") + ylab("Mean gene counts AR assemblies") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14))
ggplot(data = `prokka_genes_all_assemblies_num_mean`, aes(x = AR.mean, y = AP.mean)) + geom_jitter(colour = "#ff8d1e") + theme_classic() + xlab("Mean gene counts AR assemblies") + ylab("Mean gene counts AP assemblies") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14))
ggplot(data = `prokka_genes_all_assemblies_num_mean`, aes(x = AP.mean, y = AC.mean)) + geom_jitter(colour = "#b46bff") + theme_classic() + xlab("Mean gene counts AP assemblies") + ylab("Mean gene counts AC assemblies") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14))

#Getting number of assemblies from each host species
as.numeric(nrow(as.data.frame(unique(`prokka_genes`[`prokka_genes`$Host == "AC",]$Gene)))) #15
as.numeric(nrow(as.data.frame(unique(`prokka_genes`[`prokka_genes`$Host == "AR",]$Gene)))) #12
as.numeric(nrow(as.data.frame(unique(`prokka_genes`[`prokka_genes`$Host == "AP",]$Gene)))) #7

`melted_prokka_genes_all_assemblies_summed` <- reshape2::melt(`prokka_genes_all_assemblies_summed`, id = c("Gene.name"))
ggplot(data = `melted_prokka_genes_all_assemblies_summed`[`melted_prokka_genes_all_assemblies_summed`$variable != "All.summed",], aes(x = factor(variable, level = c("AC.summed", "AR.summed", "AP.summed")), y = value, colour = variable, alpha = 0.65)) + ggbeeswarm::geom_quasirandom() + theme_classic()

ggplot(data = `melted_prokka_genes_all_assemblies_summed`[`melted_prokka_genes_all_assemblies_summed`$variable != "All.summed",], aes(y = value, fill = variable)) + geom_density() + facet_wrap(~variable)

ggplot(data = `thingi4`[! grepl(glob2rx("*raw"), thingi4$variable) & thingi4$value != 0,], aes(x = factor(variable, level = c("AC", "AR", "AP")), y = -log10(value), colour = variable, alpha = 0.65)) + geom_hline(yintercept=c(-log10(0.05)), col="#0f0f0f", linetype = "dashed", linewidth = 0.25) + ggbeeswarm::geom_quasirandom() + theme_classic() + xlab("Isolate origin species") + ylab("-log(10) Q value") + theme(text = element_text(family = "Tex Gyre Heroes", size = 14)) + theme(legend.position = "none") + scale_colour_manual(values = c("AC" = "#1e90ff", "AR" = "#ff8d1e", "AP" = "#b46bff")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("All isolate assemblies gene hypergeometric distribution test results")

`prokka_genes`[`prokka_genes`$Host == "AC",]

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2023-02-17.RData")
toc()

2023-03-14
#Making missing figures spotted from re-reading thesis
#First is Chapter 3, supp figure 3.S02 which should show:
#"for each genus the number of samples containing reads which Metaphlan assigned to the genus"
ggplot(data = `2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$variable %in% `2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",]$Sample & `2022.11.01.metaphlan.melted`$value > 0,], aes(x = Species.x.season, y = ))

`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",]

`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$value > 0 & `2022.11.01.metaphlan.melted`$variable == "Lactobacillus",]

as.numeric(nrow(as.data.frame(unique(`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$value > 0 & `2022.11.01.metaphlan.melted`$variable == "Lactobacillus",]$SampleID))))

tic()
`metaphlan_genera_number_samples_non_zero_abundance` <- data.frame(matrix(ncol = 3, nrow = 0, dimnames = list(NULL, c("Genus", "Number of samples", "Percentage of all samples"))))
for (genera in unique(`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`[`2022.11.01_all_clades_and_levels_all_subsampled_files_metaphlan_results_all_possible_clades_relative_abundance_percentages`$Taxon.level == "Genus",]$Sample)){
  genus.zoot <- paste(genera)
  all.samples.zoot <- 81
  count.samples.zoot <- as.numeric(nrow(as.data.frame(unique(`2022.11.01.metaphlan.melted`[`2022.11.01.metaphlan.melted`$value > 0 & `2022.11.01.metaphlan.melted`$variable == genus.zoot,]$SampleID))))
  percentage.zoot <- (`count.samples.zoot` / `all.samples.zoot`) * 100
  `metaphlan_genera_number_samples_non_zero_abundance`[nrow(`metaphlan_genera_number_samples_non_zero_abundance`) + 1,] <- c(genus.zoot, count.samples.zoot, percentage.zoot)
  `metaphlan_genera_number_samples_non_zero_abundance`[,c(2:3)] <- sapply(`metaphlan_genera_number_samples_non_zero_abundance`[,c(2:3)], as.numeric)
  rm(list=ls(pattern='*zoot'))
}
view(`metaphlan_genera_number_samples_non_zero_abundance`)
toc()
#This makes more sense as a table actually
write.table(`metaphlan_genera_number_samples_non_zero_abundance`, file = "2023-03-14_missing_chapter3_figure_3s02_as_table_metaphlan_number_samples_non_zero_abundance_all_detected_genera.csv", sep=",", row.names=FALSE)

#Next is Chapter 3, supp figures 3.S05 and 3.S06 - which should really be 3.S06 and 3.S07 respectively
#"show the range of RPM values for those bins which met or exceeded the thresholds for corrected P-value and magnitude of the fold change for differences in the geometric mean within each of the host species from June to November samplings."
`2023_03_14_missing_figure_Chapter_3_3S05_mapping_scores_within_AC_significant_bins` <- ggplot(data = `AC_exclusive_only_novogene_paired_melted_good_tpms`[`AC_exclusive_only_novogene_paired_melted_good_tpms`$variable %in% `2022-0-23_AC_seasonal_raw_pvalue_SORTED`[(`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$log2.add.one.AC.seasonal.fold.change >= 1 & `2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues <= 0.05) | (`2022-0-23_AC_seasonal_raw_pvalue_SORTED`$log2.add.one.AC.seasonal.fold.change <= -1 & `2022-0-23_AC_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues <= 0.05),]$Bin,],
  aes(x = value, y = Season, fill = Season, shape = Season)) +
  geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#1ec8ff", "#006ad1")) +
  scale_colour_manual(values = c ("#1ec8ff", "#006ad1")) +
  theme(text = element_text(family = "Tex Gyre Heroes")) +
  theme_classic() + facet_wrap(~variable, strip.position="top") +
  scale_y_discrete(expand = expansion(mult = c(0.01, .425))) +
  theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) +
  theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) +
  xlab("RPMs from mapping of\nsubsampled reads to concatenated Bins reference") +
  ylab("Sampling month") +
  geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) +
  geom_jitter(width = 0.25, alpha = 0.75) +
  scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) +
  theme(text = element_text(size = 15))
#
ggsave(filename = "2023_03_14_missing_figure_Chapter_3_3S05_mapping_scores_within_AC_significant_bins.pdf", `2023_03_14_missing_figure_Chapter_3_3S05_mapping_scores_within_AC_significant_bins`)

`2023_03_14_missing_figure_Chapter_3_3S06_mapping_scores_within_AR_significant_bins` <- ggplot(data = `AR_exclusive_only_novogene_paired_melted_good_tpms`[`AR_exclusive_only_novogene_paired_melted_good_tpms`$variable %in% `2022-0-23_AR_seasonal_raw_pvalue_SORTED`[(`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$log2.add.one.AR.seasonal.fold.change >= 1 & `2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues <= 0.05) | (`2022-0-23_AR_seasonal_raw_pvalue_SORTED`$log2.add.one.AR.seasonal.fold.change <= -1 & `2022-0-23_AR_seasonal_raw_pvalue_SORTED`$Sorted_BH_corrected_pvalues <= 0.05),]$Bin,],
  aes(x = value, y = Season, fill = Season, shape = Season)) +
  geom_violin(alpha = 0.85) + scale_fill_manual(values = c("#ff551e", "#d16700")) +
  scale_colour_manual(values = c ("#ff551e", "#d16700")) +
  theme(text = element_text(family = "Tex Gyre Heroes")) +
  theme_classic() + facet_wrap(~variable, strip.position="top") +
  scale_y_discrete(expand = expansion(mult = c(0.01, .425))) +
  theme(strip.text.x = element_text(size = 11, colour = "Black", face = "bold")) +
  theme(strip.background = element_rect(colour = "black", fill = "#f2f3f4")) +
  xlab("RPMs from mapping of\nsubsampled reads to concatenated Bins reference") +
  ylab("Sampling month") +
  geom_boxplot(fill = "#f2f3f4", width = 0.175, alpha = 0.75) +
  geom_jitter(width = 0.25, alpha = 0.75) +
  scale_x_continuous(trans = "log10", labels = comma, guide = guide_axis(n.dodge = 2)) +
  theme(text = element_text(size = 15))
#
ggsave(filename = "2023_03_14_missing_figure_Chapter_3_3S06_mapping_scores_within_AR_significant_bins.pdf", `2023_03_14_missing_figure_Chapter_3_3S06_mapping_scores_within_AR_significant_bins`)

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2023-02-17.RData")
toc()

#2023-04-10
setwd("/Users/osbourne/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples")
`2023.04.10_transposed_all_processed_pilot_files_percent_reads_assigned_to_mOTUs_reference` <- read.delim("~/Google Drive/PhD/2022-Paper_1-Chapter_1_Benchmarking_and_pilot_samples/2023-04-10_transposed_all_processed_pilot_files_percent_reads_assigned_to_mOTUs_reference.txt")

`2023.04.10_melted_transposed_all_processed_pilot_files_percent_reads_assigned_to_mOTUs_reference` <- reshape2::melt(`2023.04.10_transposed_all_processed_pilot_files_percent_reads_assigned_to_mOTUs_reference`, id=c("Sample", "Host.species"))

#mOTUs percent pilot reads classified
ggplot(data = `2023.04.10_melted_transposed_all_processed_pilot_files_percent_reads_assigned_to_mOTUs_reference`, aes(x = Sample, y = value, color = Host.species, fill = Host.species))+ geom_bar(colour = "#0f0f0f", stat = "identity", position = position_dodge()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("Reads assigned to a ref-mOTU by mOTUs (%)") + xlab("Sample ID") + scale_fill_manual(values = c("Human" = "#8d1eff", "Mus" = "#1a1a1a", "AC" = "#858585", "AR" = "#c9c9c9")) + scale_color_manual(values = c("Human" = "#8d1eff", "Mus" = "#1a1a1a", "AC" = "#858585", "AR" = "#c9c9c9")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color = "Host species") + labs(fill = "Host species") + scale_y_continuous(limits = c(0, 1), breaks=pretty_breaks(n=12))
#Metaphlan 3 percent pilot reads classified
ggplot(data = subset(`2022.04.12_MELTED_metaphlan_all_samples_most_recent_lab_mouse_estimated_reads_classified_as_percent_total_reads_processed_file`, grepl(glob2rx("*percentage*"), variable)), aes(x = Sample, y = value, color = Host.species, fill = Host.species)) + geom_bar(colour = "#0f0f0f", stat = "identity", position = position_dodge()) + theme_linedraw() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("Estimated reads assigned to known clade by Metaphlan 3 (%)") + xlab("Sample ID") + scale_fill_manual(values = c("HS" = "#8d1eff", "MM" = "#1a1a1a", "AC" = "#858585", "AR" = "#c9c9c9")) + scale_color_manual(values = c("HS" = "#8d1eff", "MM" = "#1a1a1a", "AC" = "#858585", "AR" = "#c9c9c9")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color = "Host species") + labs(fill = "Host species") + scale_y_continuous(limits = c(0, 120), breaks=pretty_breaks(n=12))

#combined mOTUs and metaphlan 3 figure
`pilot_processed_motus_fig` <- ggplot(data = subset(`2023.04.10_melted_transposed_all_processed_pilot_files_percent_reads_assigned_to_mOTUs_reference`, ! grepl(glob2rx("SRR*"), Sample)), aes(x = Sample, y = value, color = Host.species, fill = Host.species)) + geom_bar(colour = "#0f0f0f", stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("Reads assigned to a ref-mOTU by mOTUs (%)") + xlab("") + scale_fill_manual(values = c("Human" = "#8d1eff", "Mus" = "#1a1a1a", "AC" = "#858585", "AR" = "#c9c9c9")) + scale_color_manual(values = c("Human" = "#8d1eff", "Mus" = "#1a1a1a", "AC" = "#858585", "AR" = "#c9c9c9")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color = "Host species") + labs(fill = "Host species") + scale_y_continuous(limits = c(0, 1), breaks=pretty_breaks(n=12))
`pilot_processed_metaphlan_fig` <- ggplot(data = subset(`2022.04.12_MELTED_metaphlan_all_samples_most_recent_lab_mouse_estimated_reads_classified_as_percent_total_reads_processed_file`, grepl(glob2rx("*percentage*"), variable)), aes(x = Sample, y = value, color = Host.species, fill = Host.species)) + geom_bar(colour = "#0f0f0f", stat = "identity", position = position_dodge()) + theme_classic() + theme(text = element_text(family = "Tex Gyre Heroes")) + ylab("Estimated reads assigned to known clade by Metaphlan 3 (%)") + xlab("") + scale_fill_manual(values = c("HS" = "#8d1eff", "MM" = "#1a1a1a", "AC" = "#858585", "AR" = "#c9c9c9")) + scale_color_manual(values = c("HS" = "#8d1eff", "MM" = "#1a1a1a", "AC" = "#858585", "AR" = "#c9c9c9")) + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + labs(color = "Host species") + labs(fill = "Host species") + scale_y_continuous(limits = c(0, 120), breaks=pretty_breaks(n=12)) + theme(legend.position = "none")
`2023-04-10_combined_processed_pilot_samples_metaphlan_motus_percent_reads_classified` <- grid.arrange(patchworkGrob(`pilot_processed_metaphlan_fig` + `pilot_processed_motus_fig` + plot_annotation(tag_levels = 'A') + plot_layout(guides = "collect")), bottom = "Sample ID")
ggsave(filename = "2023-04-10_combined_processed_pilot_samples_metaphlan_motus_percent_reads_classified.pdf", `2023-04-10_combined_processed_pilot_samples_metaphlan_motus_percent_reads_classified`)

#Having to save it as a svg then convert online to pdf for some reason
ggsave(filename = "2023-04-10_combined_processed_pilot_samples_metaphlan_motus_percent_reads_classified.svg", `2023-04-10_combined_processed_pilot_samples_metaphlan_motus_percent_reads_classified`)

#Kraken 2 median percent reads classified processed pilot rodent samples
`pilot_processed_rodents_median_kraken2_fig` <- ggplot(`2022_02_23_rodent_only_kraken_working_file_medians_V2`, aes(x = Host_species, color = variable, fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.25)) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.25)) + theme_classic() + ylab("% reads classified") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + geom_text(data = `2022_02_23_rodent_only_kraken_working_file_medians_V2`, aes(x = Host_species, y = -8, colour = factor(variable, level = c('Kraken 10 % confidence', 'Kraken 25 % confidence', 'Kraken 50 % confidence', 'Kraken 66 % confidence', 'Kraken 75 % confidence', 'Kraken 95 % confidence')), label = label), position=position_dodge(width=1.55), size = 3, show.legend = FALSE) + scale_fill_manual(values = c("Kraken 10 % confidence" = "#7b66e8", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + scale_color_manual(values = c("Kraken 10 % confidence" = "#7b66e8", "Kraken 25 % confidence" = "#3399e0", "Kraken 50 % confidence" = "#1c7abc", "Kraken 66 % confidence" = "#124e78", "Kraken 75 % confidence" = "#0a2d45", "Kraken 95 % confidence" = "#000001")) + guides(fill = guide_legend(title = "Percent confidence score", ncol = 2), color = guide_legend(title = "Percent confidence score", ncol = 2))+ guides(fill = guide_legend(title = "Kraken classification confidence score"), color = guide_legend(title = "Kraken classification confidence score")) + theme(strip.text.x = element_text(face = "italic")) + xlab("") + theme(axis.text.x=element_blank(), axis.ticks.x=element_blank(), text = element_text(family = "Tex Gyre Heroes")) + scale_y_continuous(oob=scales::discard)

#Kaiju median percent reads classified processed pilot rodent samples
`pilot_processed_rodents_median_kaiju_fig` <- ggplot(`2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = "Percent reads mapped", color = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), fill = variable)) + geom_point(aes(x = Host_species, y = Classified_median_percent), position=position_dodge(width=1.05), size = 1.75) + geom_errorbar(aes(x = Host_species, y = Classified_median_percent, ymin=lower_confidence_interval, ymax=upper_confidence_interval), position=position_dodge(width=1.05), width = 1, size = 0.75) + geom_text(data = `2022_02_23_rodent_only_kaiju_working_file_v2`, aes(x = Host_species, y = 56, colour = factor(variable, level = c('Kaiju 0 errors allowed', 'Kaiju 1 error allowed', 'Kaiju 2 errors allowed', 'Kaiju 3 errors allowed', 'Kaiju 4 errors allowed', 'Kaiju 5 errors allowed', 'Kaiju 10 errors allowed', 'Kaiju 20 errors allowed')), label = Label), position=position_dodge(width=1.05), size = 3.5, show.legend = FALSE) + ylab("") + facet_wrap( ~ Host_species, scales="free_x", strip.position="top") + 
  scale_fill_manual(name = "Kaiju allowed errors", values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) +
  scale_color_manual(name = "Kaiju allowed errors", values = c("Kaiju 0 errors allowed" = "#0a0909", "Kaiju 1 error allowed" = "#451a09", "Kaiju 2 errors allowed" = "#782e10", "Kaiju 3 errors allowed" = "#9b3b15", "Kaiju 4 errors allowed" = "#ac4217", "Kaiju 5 errors allowed" = "#e0551e", "Kaiju 10 errors allowed" = "#e87c52", "Kaiju 20 errors allowed" = "#ec9674")) +
  guides(fill = guide_legend(title = "Kaiju allowed errors"), color = guide_legend(title = "Kaiju allowed errors")) +
  xlab("") +
  scale_y_continuous(oob=scales::discard) +
  theme_classic() +
  theme(text = element_text(family = "Tex Gyre Heroes"), axis.text.x = element_blank(), axis.ticks.x = element_blank(), strip.text.x = element_text(face = "italic")) +
  guides(fill = "none")

#Combining kraken2 and kaiju median percent rodent figs
`2023-04-10_combined_median_percentages_classified_pilot_rodents_kraken2_kaiju` <- grid.arrange(patchworkGrob(`pilot_processed_rodents_median_kraken2_fig` / `pilot_processed_rodents_median_kaiju_fig` + plot_annotation(tag_levels = 'A') + plot_layout(guides = "collect")), bottom = "Host species")
ggsave(filename = "2023-04-10_combined_median_percentages_classified_pilot_rodents_kraken2_kaiju.svg", `2023-04-10_combined_median_percentages_classified_pilot_rodents_kraken2_kaiju`)

tic()
save.image("~/Google Drive/PhD/Scripts/R_working_directory/2023-02-17.RData")
toc()
